---
title: "ipsc_mn_als_meta"
font-family: 'Helvetica'
author: |
  | Oliver Ziff
  | Patani Lab
  | The Francis Crick Institute
  | Institute of Neurology
  | University College London, UK
date: "`r Sys.Date()`"
output: 
 rmarkdown::html_document:
   theme: spacelab
   highlight: haddock
   code_folding: hide
   toc: true
   toc_float: true
   smooth_scroll: true
   number_sections: false
   self_contained: true  
knit: (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(dirname(inputFile), 'docs/index.html')) })
---

```{css, echo = FALSE}
h1, #TOC>ul>li {
  color: #000000;
    font-weight: bold;
}
h2, #TOC>ul>ul>li {
  color: #41494D;
}
```

# Setup

```{r setup, include=FALSE}
library(here)
proj_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.metadata.RData")) # load metadata before functions to avoid overwriting with old functions

# Mac
camp_path = here("/Volumes/lab-luscomben/home/users/ziffo")
proj_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
script_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta")
shared_path = here("/Volumes/lab-luscomben/home/users/ziffo/proj-luscombn-patani/working")
collab_path = here("/Volumes/lab-luscomben/home/users/ziffo/patani-collab")
## OnDemand
# proj_path = here("/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
# script_path = here(here(proj_path,"scripts/ipsc_mn_als_meta")
# shared_path = here("/camp/project/proj-luscombn-patani/working")
# collab_path = here("/camp/lab/luscomben/home/users/ziffo//patani-collab")
setwd(proj_path)
set_here(proj_path)
here(proj_path)
# here::i_am("ipsc_mn_als_meta.Rmd")
here()
dr_here()

source("/Volumes/lab-luscomben/home/users/ziffo/scripts/functions/mac_R_functions.R") # source("/camp/lab/luscomben/home/users/ziffo/scripts/functions/R_functions.R")

# load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.RData") # 6.7GB > crashes R
ipsc_mn_als_datasets_with_lee = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.rds")) #2.56GB
ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.rds"))
ipsc_mn_als_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.rds")) # 1.26G
ipsc_mn_als_datasets.polyA = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.rds"))
ipsc_mn_als_datasets.total = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.rds"))
ipsc_mn_als_datasets.noiso = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.noiso.rds"))

ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds")) # 84M
ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds")) # 162M
ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds")) # 85M

nygc_postmortem_spinal_cord.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.tissue.rds"))
nygc_postmortem_spinal_cord.c9orf72_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.c9orf72_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.fus_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.fus_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sod1_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sod1_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sporadic_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sporadic_vs_ctrl.rds"))

```

## Update git / github

```{r}
library(usethis) # usethis & gert https://jeroen.github.io/gert2019 https://www.garrickadenbuie.com/blog/pull-request-flow-usethis/?interactive=1&steps=
use_git_config(user.name = "ojziff", user.email = "oliver.ziff@crick.ac.uk")
usethis::create_github_token()

credentials::set_github_pat("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") 
# make github
library(gert)
setwd(here(proj_path,"scripts/ipsc_mn_als_meta")) # setwd then dont need to specify repo argument
git_commit_all("update") # stage all files & commit in one command - implies git_add(".")
pr_init("figures_and_tables") # initialise a new branch
pr_push() # this will open github in chrome > pull request > merge

git_sitrep() # print git and github info
git_config()
git_info()
git_ls() # list git tree
git_log() # check recent commits
git_status() # check which files are modified
# pr_resume("figures_and_tables") # resume a branch
# git_add(".") #stage the files you want to commmit - "." indicates all files.
# git_commit("update") # commit with comment
# pr_pull()

```


## Metadata

```{r}
ipsc_mn_als_datasets.metadata %>% count(dataset)
ipsc_mn_als_datasets.metadata %>% count(condition)
ipsc_mn_als_datasets.metadata %>% count(mutation)

read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_dataportal_datatable.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_participants.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_released_files.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/Clinical_Dictionary.csv")) %>% clean_names %>% glimpse
first_last_visit_alsfrs.answerals = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/answerals_progression_alsfrs.csv")) # answerals progression based on ALSFRS
AALSHXFX.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSHXFX.csv")) %>% clean_names %>% select(participant_id, subject_uid, onsetdt)
Demographics.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Demographics.csv")) %>% clean_names %>% select(participant_id, subject_uid, dob)
age_onset = AALSHXFX.csv %>% full_join(Demographics.csv) %>% mutate(age_onset = as.numeric((ddays(onsetdt) - ddays(dob)) / (364.25*24*60*60)), sample = gsub("-","_",participant_id)) %>% select(sample, age_onset) %>%
  mutate(early_late_onset = case_when(age_onset < 57 ~ "early", age_onset > 57 ~ "late"))
Mortality.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Mortality.csv")) %>% clean_names %>% filter(!dieddt %in% c("Glioblastoma","Prostate cancer", "physician assisted suicide")) %>% select(participant_id, subject_uid, dieddt) 
mortality = AALSHXFX.csv %>% full_join(Mortality.csv) %>% mutate(survival = as.numeric((ddays(dieddt) - ddays(onsetdt)) / (364.25*24*60*60)), sample = gsub("-","_",participant_id)) %>% select(sample, survival) %>%
  mutate(early_late_death = case_when(survival < 2.8 ~ "early", survival > 2.8 ~ "late"))
onset_site = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSHXFX.csv")) %>% clean_names %>% 
                        select(participant_id, subject_uid, hxblb, hxax, hxgen, hxli, hxot) %>% filter(!(hxblb == 0 & hxax == 0 & hxgen == 0 & hxli == 0 & hxot == 0)) %>%
  mutate(onset_site_detailed = case_when(hxblb == 1 & hxax == 0 & hxgen == 0 & hxli == 0 & hxot == 0 ~ "bulbar", hxblb == 0 & hxax == 0 & hxgen == 0 & hxli == 1 & hxot == 0 ~ "limb", hxblb == 0 & hxax == 0 & hxgen == 1 & hxli == 0 & hxot == 0 ~ "generalised", 
                                hxblb == 0 & hxax == 1 & hxgen == 0 & hxli == 0 & hxot == 0 ~ "axial", TRUE ~ "mixed"), #hxot == 1 ~ "other", 
         onset_site = case_when(onset_site_detailed %in% c("axial","mixed","generalised") ~ "other", TRUE ~ onset_site_detailed),
         bulbar_other = case_when(hxblb == 1 ~ "bulbar", TRUE ~ "other"), limb_other = case_when(hxli == 1 ~ "limb", TRUE ~ "other"), axial_other = case_when(hxax == 1 ~ "axial", TRUE ~ "other"), 
         bulbar_axial = case_when(hxblb == 1 ~ "bulbar", hxax == 1 ~ "axial", TRUE ~ "other"), bulbar_limb = case_when(hxblb == 1 ~ "bulbar", hxli == 1 ~ "limb", TRUE ~ "other"), bulbar_mixed = case_when(hxblb == 1 ~ "bulbar", onset_site == "mixed" ~ "mixed", TRUE ~ "other"), 
         limb_axial = case_when(hxli == 1 ~ "limb", hxax == 1 ~ "axial", TRUE ~ "other"), limb_mixed = case_when(onset_site == "mixed" ~ "mixed", hxli == 1 ~ "limb", TRUE ~ "other"), axial_mixed = case_when(hxax == 1 ~ "axial", onset_site == "mixed" ~ "mixed", TRUE ~ "other"),
         sample = gsub("-","_",participant_id)) %>% select(sample, subject_uid, onset_site, onset_site_detailed, bulbar_other,limb_other,axial_other,bulbar_axial,bulbar_limb,bulbar_mixed,limb_axial,limb_mixed,axial_mixed)
family_history = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Family_History_Log.csv")) %>% clean_names %>% select(participant_id, subject_uid, fhals, fhftd) %>% 
  mutate(family_history = case_when(fhals == 1 | fhftd == 1 ~ "yes", TRUE ~ "no"),sample = gsub("-","_",participant_id)) %>% select(sample, family_history) %>% distinct(sample, .keep_all = TRUE)
subjects = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/subjects.csv")) %>% clean_names %>% 
  mutate(mnd_group = case_when(subject_group_id == 17 ~ "other_mnd", subject_group_id == 1 ~ "als", subject_group_id == 5 ~ "healthy", subject_group_id == 11 ~ "carrier"), mnd_group = factor(mnd_group, levels = c("healthy", "carrier","other_mnd","als")), sample = gsub("-","_",participant_id)) %>% select(sample, mnd_group) %>% distinct(sample, .keep_all = TRUE)
ck_value = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Auxiliary_Chemistry.csv")) %>% clean_names %>% drop_na(acckrslt) %>% mutate(sample = gsub("-","_",participant_id), cknorm = case_when(cknorm == 1 ~ "normal", TRUE ~ "raised")) %>% arrange(-acckrslt) %>% distinct(sample, .keep_all = TRUE) %>% select(sample, ck = acckrslt, cknorm)
AALSHXFX.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSDXFX.csv")) %>% clean_names %>% drop_na(elescrlr) %>% select(participant_id, elescrlr) %>% 
  mutate(sample = gsub("-","_",participant_id)) %>% select(sample, elescrlr) %>% arrange(elescrlr) %>% distinct(sample, .keep_all = TRUE)
answerals_transcriptomic_files = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_released_files.csv")) %>% clean_names %>% filter(omic == "transcriptomics", grepl(".fastq.gz$", path)) %>% 
  mutate(fastq = case_when(grepl("_1.fastq.gz$", path) ~ "fastq_1", grepl("_2.fastq.gz$", path) ~ "fastq_2")) %>% 
  filter(participant_id != "CTRL-NEUEU392AE8") %>% #17 fastq's associated with this ID. Unclear if these are technical repeats or incorrectly labelled.
  pivot_wider(c(participant_id, experiment), names_from = fastq, values_from = c(path,size)) %>% 
  mutate(fastq_1 = paste0(here(camp_path,"proj-luscombn-patani/working/answerals/reads/"),path_fastq_1), fastq_2 = paste0(here(camp_path,"proj-luscombn-patani/working/answerals/reads/"), path_fastq_2))
samplesheet = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_dataportal_datatable.csv")) %>% clean_names %>% filter(transcriptomics == "Yes", participant_id != "CTRL-NEUEU392AE8") %>% # ipsc == "Yes", 
  left_join(answerals_transcriptomic_files) %>% 
  mutate(strandedness = "reverse", 
         condition = case_when(subject_group == "ALS" ~ "als", subject_group == "Healthy Control" ~ "ctrl", grepl("carrier",subject_group) ~ "carrier", subject_group == "Non-ALS MND" ~ "other_mnd"),
         mutation = case_when(self_reported_mutations %in% c("C9orf72","SOD1","SETX", "TARDBP (TDP43)") ~ self_reported_mutations, TRUE ~ wgs_variants), mutation = tolower(gsub(" [(]WGS[)]","",mutation)), mutation = gsub("tardbp [(]tdp43[)]","tardbp", mutation),  
         mutation = case_when(condition == "als" & is.na(mutation) ~ "sporadic", condition == "ctrl" & is.na(mutation) ~ "ctrl",TRUE ~ mutation), mutant = case_when(mutation %in% c("sporadic","ctrl") ~ "no", TRUE ~ "yes"),
         sample = gsub("-","_",participant_id), DIV = 32, instrument = "Illumina NovaSeq 6000") %>%
  select(sample, fastq_1, fastq_2, strandedness, condition, mutation, mutant, participant_id, guid, nygc_cgnd_id, sex, DIV, instrument, ethnicity, race, pls_subgroup, revised_el_escorial_criteria, age_at_sample_collection, age_at_symptom_onset, age_at_death, alsfrs_r_baseline, alsfrs_r_latest, alsfrs_r_progression_slope, cbs_baseline, cbs_latest, cbs_progression_slope, subject_group, ipsc, dimn, i_psc_line_id_root, i_psc_line_name, di_m_ns_line_name, batch_id, self_reported_mutations, wgs_variants, repeat_length_c9, repeat_length_atxn2, size_fastq_1, size_fastq_2) %>%
  filter(!sample %in% c("CASE_NEUMN922PMZ","CASE_NEUYC198DCR")) %>% # R2_fastq.gz corrupt crashing nfcore
  left_join(select(first_last_visit_alsfrs.answerals, sample, alsfrs_slope, progression)) %>% left_join(age_onset) %>% left_join(mortality) %>% left_join(onset_site) %>% left_join(family_history) %>% left_join(subjects) %>% left_join(ck_value) %>% left_join(AALSHXFX.csv) # add clinical data to metadata
samplesheet # Ensure no NA's in sample sheet
samplesheet %>% glimpse
samplesheet %>% distinct(participant_id) # 287
write_csv(samplesheet, file = here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/samplesheet.csv"), na = "")

```


Run the DESeq2 objects scripts to create RData. Takes ~ 2.5 hours

```{bash}
cd /camp/home/ziffo/home/projects/ipsc-mn-als-meta/scripts
mls
sar
sbatch -N 1 -c 8 --mem=80G -t 12:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsc_mn_als_meta_objects.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk --job-name=DESeq2objects 

# OnDemand Terminal Tab
cd /camp/home/ziffo/home/projects/ipsc-mn-als-meta/scripts
sbatch -N 1 -c 8 --mem=80G -t 12:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsc_mn_als_meta_objects.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk --job-name=DESeq2objects 

```

```{r}
ipsc_mn_als_datasets.metadata %>% distinct(dataset) #16 (14)
ipsc_mn_als_datasets.metadata %>% distinct(dataset_sample) # 419
ipsc_mn_als_datasets.metadata %>% count(condition) #318 ALS, 101 ctrl
ipsc_mn_als_datasets.metadata %>% count(dataset, condition) %>% print(n=Inf)
ipsc_mn_als_datasets.metadata %>% count(mutation) %>% arrange(-n)
ipsc_mn_als_datasets.metadata %>% count(mutation) # 11
# Gender % 
ipsc_mn_als_datasets.metadata %>% count(dataset, condition, gender) %>% pivot_wider(names_from = c(condition,gender), values_from = n) %>% mutate_if(is.numeric, funs(replace_na(., 0))) %>% mutate(ctrl_male.pct = ctrl_male/(ctrl_male + ctrl_female) * 100, als_male.pct = als_male/(als_male + als_female) * 100)

ipsc_mn_als_datasets.metadata %>% filter(condition == "ctrl") %>% count(mutation)
ipsc_mn_als_datasets.metadata %>% filter(mutation == "iso") %>% count(condition)

# 104 ctrls: 85 ctrl, 17 iso, 2 have mutations (OPTN and ALS2)

ipsc_mn_als_datasets.metadata %>% filter(condition == "ctrl") %>% count(mutation)

neurolincs.metadata %>% filter(dataset == "neurolincs.iMN") %>% distinct(instrument)

      ipsc_mn_als_datasets.introns_tag_pct$res %>% filter(padj < 0.05) #111
ipsc_mn_als_datasets.introns_tag_pct$irfinder %>% filter(padj < 0.05) #118,553
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) # 38
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) # 25
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) # 74
ipsc_mn_als_datasets.total$irfinder %>% filter(padj < 0.05) #118829

ipsc_mn_als_datasets$dds %>% colData %>% as_tibble %>% glimpse
ipsc_mn_als_datasets$dds %>% colData %>% as_tibble %>% my_summarise(samtools_mqc_generalstats_samtools_raw_total_sequences)

samtools_mqc_generalstats_samtools_raw_total_sequences


```



# Table S2 QC stats

Table S1 is iPSN protocols

Multiqc general stats

```{r}
multiqc_stat_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), multiqc_stat_paths = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project","/Volumes/lab-luscomben/home/users/ziffo",multiqc_stat_paths)) %>% pull(multiqc_stat_paths)
multiqc_stats.tsv = multiqc_stat_paths %>% map(read_tsv, show_col_types = FALSE)                                                                        
names(multiqc_stats.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(dataset)
multiqc_data <- map_dfr(multiqc_stats.tsv, bind_rows, .id = "dataset") %>% clean_names() %>% drop_na(samtools_mqc_generalstats_samtools_flagstat_total) %>% filter(sample %in% ipsc_mn_als_datasets_with_lee.metadata$sample) %>%
  mutate(dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincs0" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS", dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~ dataset)) %>%
  select(dataset, sample, raw_total_sequences = samtools_mqc_generalstats_samtools_raw_total_sequences, flagstat_total = samtools_mqc_generalstats_samtools_flagstat_total, mapped_passed = samtools_mqc_generalstats_samtools_mapped_passed, reads_mapped = samtools_mqc_generalstats_samtools_reads_mapped, reads_mapped_percent = samtools_mqc_generalstats_samtools_reads_mapped_percent, reads_aligned = quali_map_mqc_generalstats_qualimap_reads_aligned, r_rna_percent = biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna, bias_5_3 = quali_map_mqc_generalstats_qualimap_5_3_bias, proper_pairs_percent = r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent, reads_assigned_percent = feature_counts_mqc_generalstats_featurecounts_percent_assigned, reads_assigned = feature_counts_mqc_generalstats_featurecounts_assigned, percent_duplication = picard_mqc_generalstats_picard_percent_duplication, error_rate = samtools_mqc_generalstats_samtools_error_rate, non_primary_alignments = samtools_mqc_generalstats_samtools_non_primary_alignments, reads_properly_paired_percent = samtools_mqc_generalstats_samtools_reads_properly_paired_percent, percent_mapped = salmon_mqc_generalstats_salmon_percent_mapped, num_mapped = salmon_mqc_generalstats_salmon_num_mapped, uniquely_mapped_percent = star_mqc_generalstats_star_uniquely_mapped_percent, uniquely_mapped = star_mqc_generalstats_star_uniquely_mapped, percent_trimmed =cutadapt_mqc_generalstats_cutadapt_percent_trimmed, total_sequences = fast_qc_mqc_generalstats_fastqc_total_sequences, raw_percent_gc = fast_qc_raw_mqc_generalstats_fastqc_raw_percent_gc, raw_avg_sequence_length = fast_qc_raw_mqc_generalstats_fastqc_raw_avg_sequence_length, trimmed_percent_duplicates = fast_qc_trimmed_mqc_generalstats_fastqc_trimmed_percent_duplicates, trimmed_percent_gc = fast_qc_trimmed_mqc_generalstats_fastqc_trimmed_percent_gc)
multiqc_data %>% glimpse
write_csv(multiqc_data, here(proj_path,"alignment/multiqc_general_stats.csv"))

```


# Fig S2-5 PCA

Fig S1 PRISMA flow

## Sample characteristics

```{r}
ipsc_mn_als_datasets_with_lee.pcaData <- plotPCA(ipsc_mn_als_datasets_with_lee$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"),
#                                                                              "samtools_mqc_generalstats_samtools_flagstat_total", "samtools_mqc_generalstats_samtools_mapped_passed", "biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna", "quali_map_mqc_generalstats_qualimap_5_3_bias", "quali_map_mqc_generalstats_qualimap_reads_aligned", "r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent",
# "feature_counts_mqc_generalstats_featurecounts_percent_assigned", "feature_counts_mqc_generalstats_featurecounts_assigned", "picard_mqc_generalstats_picard_percent_duplication", "samtools_mqc_generalstats_samtools_error_rate", "samtools_mqc_generalstats_samtools_non_primary_alignments", "samtools_mqc_generalstats_samtools_reads_mapped", "samtools_mqc_generalstats_samtools_reads_mapped_percent", "samtools_mqc_generalstats_samtools_reads_properly_paired_percent", "samtools_mqc_generalstats_samtools_reads_mq0_percent", "samtools_mqc_generalstats_samtools_raw_total_sequences", "salmon_mqc_generalstats_salmon_percent_mapped", "salmon_mqc_generalstats_salmon_num_mapped", "star_mqc_generalstats_star_uniquely_mapped_percent", "star_mqc_generalstats_star_uniquely_mapped"),
returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang","lee.CTRL") ~ "poly(A)", TRUE ~ "Total Ribo-Zero")) %>%
  left_join(select(ipsc_mn_als_datasets_with_lee.metadata, dataset_sample, gender)) #%>% 
  # mutate(database_dir = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project", "/Volumes/lab-luscomben/home/users/ziffo", database_dir))
  
# multiqc stats
multiqc_general_stats_stat_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_general_stats_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), 
     multiqc_general_stats_stat_paths = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project","/Volumes/lab-luscomben/home/users/ziffo",multiqc_general_stats_stat_paths)) %>% pull(multiqc_general_stats_stat_paths)
multiqc_general_stats.tsv = multiqc_general_stats_stat_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_general_stats.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_general_stats <- map_dfr(multiqc_general_stats.tsv, bind_rows, .id = "database_dir") %>% clean_names()
multiqc_rseqc_read_distribution_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_rseqc_read_distribution_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"),
     multiqc_rseqc_read_distribution_paths = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project","/Volumes/lab-luscomben/home/users/ziffo",multiqc_rseqc_read_distribution_paths)) %>% pull(multiqc_rseqc_read_distribution_paths)
multiqc_rseqc_read_distribution.tsv = multiqc_rseqc_read_distribution_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_rseqc_read_distribution.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_rseqc_read_distribution <- map_dfr(multiqc_rseqc_read_distribution.tsv, bind_rows, .id = "database_dir") %>% clean_names()
ipsc_mn_als_datasets_with_lee.pcaData = ipsc_mn_als_datasets_with_lee.pcaData %>% left_join(multiqc_general_stats) %>% left_join(multiqc_rseqc_read_distribution) # need unique sample names - If multiple dataset names within a single multiqc_general_stats.txt then will only join the first dataset name

# Colour by dataset
colourCount = length(unique(mtcars$hp))
getPalette = colorRampPalette(brewer.pal(8, "Accent"))
getPalette(colourCount)
show_col(getPalette(colourCount))
#7FC97F" "#94C09B" "#A9B7B7" "#BEAED4" "#D3B4BA" "#E8BAA0" "#FDC086" "#FDD58C" "#FEEA92" "#FFFF99" "#BCCEA0" "#7A9DA8" 
# "#386CB0" "#75489F" "#B2258F" "#F0027F" "#DF1F5C" "#CF3D39" "#BF5B17" "#A15E31" "#83624B" "#666666"

# Colour by dataset
ipsc_mn_als_datasets_with_lee.pca_dataset <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  scale_color_manual(values = c("AnswerALS"="#94C09B", "neurolincs.iMN"="#A9B7B7", "neurolincs.diMN"="#BEAED4", "bhinge"= "#7FC97F","catanese"="#7FC97F", "dafinca"="#94C09B", "desantis"="#FEEA92", "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "hawkins"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sommer"="#FFFF99", "sterneckert"="#7A9DA8", "wang"="#BCCEA0","lee.CTRL"="#75489F", "lee.ALS"="#386CB0")) +
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "right", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) + #coord_fixed() +
  ggforce::geom_mark_ellipse(data = filter(ipsc_mn_als_datasets_with_lee.pcaData, dataset %in% c("lee.CTRL","lee.ALS")),  aes(label = dataset), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE)
ipsc_mn_als_datasets_with_lee.pca_dataset

# Colour by library_type
ipsc_mn_als_datasets_with_lee.pca_library_type <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=library_type)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))+labs(colour="Library preparation")
ipsc_mn_als_datasets_with_lee.pca_library_type
# Colour by instrument
ipsc_mn_als_datasets_with_lee.pca_instrument <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=instrument)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_blank(), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_instrument
# Colour by DIV
ipsc_mn_als_datasets_with_lee.pca_DIV <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=DIV)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top"))  + labs(color="Days in vitro") + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_DIV
# Colour by ALS
ipsc_mn_als_datasets_with_lee.pca_als <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=condition)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_als
# Colour by mutation
ipsc_mn_als_datasets_with_lee.pca_mutation <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=mutation)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_mutation
# Colour by gender
ipsc_mn_als_datasets_with_lee.pca_gender <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=gender)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets_with_lee.pca_gender

ipsc_mn_als_datasets_with_lee.pca_merged = ggarrange(
  ipsc_mn_als_datasets_with_lee.pca_dataset, 
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_library_type, ipsc_mn_als_datasets_with_lee.pca_instrument, ncol = 2, labels = c("b","c")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_gender, ipsc_mn_als_datasets_with_lee.pca_DIV, ncol = 2, labels = c("d","e")), 
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_als, ipsc_mn_als_datasets_with_lee.pca_mutation, ncol = 2, labels = c("f","g")),
  nrow = 4, heights = c(1.1,1,1,1), labels = c("a","","",""))
# ipsc_mn_als_datasets_with_lee.pca_merged
ggsave(ipsc_mn_als_datasets_with_lee.pca_merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.pca_merged.png"), height = 12, width = 8)
```

## RNAseq QC metrics

```{r}
### PCA 2

# Colour by strandedness
ipsc_mn_als_datasets_with_lee.pca_strandedness <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=strandedness)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.2, keyheight=0.2, default.unit="inch")) + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_strandedness
# Colour by read depth
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_raw_total_sequences)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6, angle = 0), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(1, 'cm'))  + 
  guides(color = guide_colourbar(title = "read depth", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_raw_total_sequences,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences


# Color by intronic read % intron_read_pct
ipsc_mn_als_datasets_with_lee.pca_read_distribution <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=introns_tag_pct)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "intronic reads %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$introns_tag_pct,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets_with_lee.pca_read_distribution
# Colour by biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna
ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "ribosomal rna %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna
# Colour by quali_map_mqc_generalstats_qualimap_5_3_bias
ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=quali_map_mqc_generalstats_qualimap_5_3_bias)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "5' 3' bias", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=1, low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias
# Colour by quali_map_mqc_generalstats_qualimap_reads_aligned
ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_reads_aligned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=quali_map_mqc_generalstats_qualimap_reads_aligned)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads aligned", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$quali_map_mqc_generalstats_qualimap_reads_aligned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_reads_aligned
# Colour by picard_mqc_generalstats_picard_percent_duplication
ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=picard_mqc_generalstats_picard_percent_duplication)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Read duplication %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$picard_mqc_generalstats_picard_percent_duplication,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +  
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication
# Colour by samtools_mqc_generalstats_samtools_reads_mapped
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mapped)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Reads mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped
# Colour by star_mqc_generalstats_star_uniquely_mapped_percent
ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=star_mqc_generalstats_star_uniquely_mapped_percent)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads uniquely mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$star_mqc_generalstats_star_uniquely_mapped_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent

ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged = ggarrange(
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_strandedness, ipsc_mn_als_datasets_with_lee.pca_read_distribution , ncol = 2, labels = c("a","b")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences, ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias, ncol = 2, labels = c("c","d")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna, ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication, ncol = 2, labels = c("e","f")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent, ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped, ncol = 2, labels = c("g","h")), 
  nrow = 4, heights = c(1,1,1,1))
# ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged
ggsave(ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged.png"), height = 11.75, width = 8.25)


# # Colour by samtools_mqc_generalstats_samtools_reads_mapped_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mapped_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mapped_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped_percent
# # Colour by samtools_mqc_generalstats_samtools_error_rate
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_error_rate <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_error_rate)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Error rate", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_error_rate,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_error_rate
# # Colour by samtools_mqc_generalstats_samtools_non_primary_alignments
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_non_primary_alignments <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_non_primary_alignments)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Non primary alignments", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_non_primary_alignments,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_non_primary_alignments
# # Colour by samtools_mqc_generalstats_samtools_reads_mq0_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mq0_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mq0_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Mapping quality 0 %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mq0_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mq0_percent
# # Colour by samtools_mqc_generalstats_samtools_flagstat_total
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_flagstat_total <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_flagstat_total)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_flagstat_total
# # Colour by samtools_mqc_generalstats_samtools_mapped_passed
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_mapped_passed <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_mapped_passed)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Mapped reads passed", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_mapped_passed,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_mapped_passed
# # Colour by r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent
# ipsc_mn_als_datasets_with_lee.pca_r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Proper pairs %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent
# # Colour by samtools_mqc_generalstats_samtools_reads_properly_paired_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_properly_paired_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_properly_paired_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Properly paired %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_properly_paired_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_properly_paired_percent
# # Colour by feature_counts_mqc_generalstats_featurecounts_percent_assigned
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_percent_assigned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=feature_counts_mqc_generalstats_featurecounts_percent_assigned)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Assigned reads %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$feature_counts_mqc_generalstats_featurecounts_percent_assigned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_percent_assigned
# # Colour by feature_counts_mqc_generalstats_featurecounts_assigned
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_assigned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=feature_counts_mqc_generalstats_featurecounts_assigned)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Assigned reads", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$feature_counts_mqc_generalstats_featurecounts_assigned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_assigned
# # Colour by salmon_mqc_generalstats_salmon_percent_mapped
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_percent_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=salmon_mqc_generalstats_salmon_percent_mapped)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$salmon_mqc_generalstats_salmon_percent_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_percent_mapped
# # Colour by salmon_mqc_generalstats_salmon_num_mapped
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_num_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=salmon_mqc_generalstats_salmon_num_mapped)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$salmon_mqc_generalstats_salmon_num_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_num_mapped
# # Colour by star_mqc_generalstats_star_uniquely_mapped
# ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=star_mqc_generalstats_star_uniquely_mapped)) + geom_point(size=1) + 
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
#   guides(color = guide_colourbar(title = "Reads uniquely mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$star_mqc_generalstats_star_uniquely_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# # ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped

```


## RUVSeq

https://github.com/mikelove/preNivolumabOnNivolumab/blob/main/preNivolumabOnNivolumab.knit.md

```{r}
# # simple design, no adjustment
# # estimate technical variation with RUVg method
# ipsc_mn_als_datasets_with_lee.metadata.salmon_files = ipsc_mn_als_datasets_with_lee.metadata %>% mutate(file_salmon = file.path(database_dir, "nfcore/star_salmon", ipsc_mn_als_datasets_with_lee.metadata[["sample"]], "quant.sf")) %>% pull(file_salmon)
# names(ipsc_mn_als_datasets_with_lee.metadata.salmon_files) = ipsc_mn_als_datasets_with_lee.metadata %>% pull(dataset_sample) # sample filename in nfcore outdir
# rownames(ipsc_mn_als_datasets_with_lee.metadata) <- ipsc_mn_als_datasets_with_lee.metadata %>% pull(dataset_sample)  # unique sample name
# ipsc_mn_als_datasets_with_lee.dds.ruv <- tximport(ipsc_mn_als_datasets_with_lee.metadata.salmon_files, type="salmon", tx2gene=tx2gene) %>% DESeqDataSetFromTximport(colData = ipsc_mn_als_datasets_with_lee.metadata, design = ~condition)
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt <- vst(ipsc_mn_als_datasets_with_lee.dds.ruv, blind=FALSE)
# ipsc_mn_als_datasets_with_lee.dds.ruv.lrt = ipsc_mn_als_datasets_with_lee.dds.ruv %>% DESeq(test = "LRT", reduced = ~1) #, fitType="glmGamPoi")
# ipsc_mn_als_datasets_with_lee.res.ruv.lrt <- DESeq2::results(ipsc_mn_als_datasets_with_lee.dds.ruv.lrt)
# table(ipsc_mn_als_datasets_with_lee.res.ruv.lrt$padj < .1) # FALSE  TRUE # 13385 20117
# set <- newSeqExpressionSet(counts(ipsc_mn_als_datasets_with_lee.dds.ruv.lrt))
# set <- betweenLaneNormalization(set, which="upper")
# not_sig <- rownames(ipsc_mn_als_datasets_with_lee.res.ruv.lrt)[which(ipsc_mn_als_datasets_with_lee.res.ruv.lrt$pvalue > .1)]
# empirical <- rownames(set)[ rownames(set) %in% not_sig ]
# set <- RUVg(set, empirical, k=5)
# pdat <- pData(set) # %>% mutate(condition = case_when) # W_1 and W_2 etc are unwanted variation factors
# pdat$condition <- ipsc_mn_als_datasets_with_lee.metadata$condition # visualize how the factors of unwanted variation describe the samples in the PC1 and PC2 space:
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt$W1 <- pdat$W_1
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt$W2 <- pdat$W_2
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt$W3 <- pdat$W_3
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt$W4 <- pdat$W_4
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt$W5 <- pdat$W_5
# saveRDS(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt, here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.rds"))

ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.rds"))
ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData <- plotPCA(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt, intgroup=c("sample", "W1", "W2", "W3", "W4", "W5"), returnData=TRUE) %>% as_tibble()
ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W1 <- ggplot(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W1)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W1", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData$W1,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W1
ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W2 <- ggplot(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W2)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W2", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData$W2,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W2
ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W3 <- ggplot(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W3)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W3", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData$W3,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W3
ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W4 <- ggplot(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W4)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W4", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData$W4,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W4
ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W5 <- ggplot(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W5)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W5", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData$W5,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W5
ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pca = ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W1 + ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W2 + ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W3 + ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W4 + ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.W5 + 
  plot_layout(ncol = 2, nrow = 3)
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pca
ggsave(ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pca, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.pca.png"), height = 11.75, width = 8.25)


# # Adding the factors to the design, and performing a LRT is fairly simple, first we show this with DESeq2:
# colData(ipsc_mn_als_datasets_with_lee.dds.ruv.lrt) <- cbind(colData(ipsc_mn_als_datasets_with_lee.dds.ruv.lrt), pdat[,1:5])
# design(ipsc_mn_als_datasets_with_lee.dds.ruv.lrt) <- ~W_1 + W_2 + W_3 + W_4 + W_5 + condition
# ipsc_mn_als_datasets_with_lee.dds.ruv.lrt <- DESeq(ipsc_mn_als_datasets_with_lee.dds.ruv.lrt, test="LRT", reduced=~W_1 + W_2 + W_3 + W_4 + W_5)
# # Controlling for technical variation in this case greatly reduces the number of DE genes. Controlling for technical variation can also increase the number of DE genes.
# # That the number is reduced here indicates that some of the previous results were likely due to confounding of technical variation with the condition variable. Ignoring that confounding, again, will result in invalid inference for all methods that look for shifts in the expression values.
# ipsc_mn_als_datasets_with_lee.res.ruv.lrt <- DESeq2::results(ipsc_mn_als_datasets_with_lee.dds.ruv.lrt)
# table(ipsc_mn_als_datasets_with_lee.res.ruv.lrt$padj < .1)
# # FALSE  TRUE 
# # 28633  1712 

```

## PC1 & PC2 gene loadings

https://stackoverflow.com/questions/12760108/principal-components-analysis-how-to-get-the-contribution-of-each-paramete
https://github.com/mikelove/DESeq2/blob/master/R/plots.R

```{r}
ipsc_mn_als_datasets_with_lee.vsd.rv <- rowVars(assay(ipsc_mn_als_datasets_with_lee$vsd)) # calculate the variance for each gene
ipsc_mn_als_datasets_with_lee.vsd.ntop_genes <- order(ipsc_mn_als_datasets_with_lee.vsd.rv, decreasing=TRUE)[seq_len(min(500, length(ipsc_mn_als_datasets_with_lee.vsd.rv)))] # select the ntop genes by variance
ipsc_mn_als_datasets_with_lee.vsd.pca <- prcomp(t(assay(ipsc_mn_als_datasets_with_lee$vsd)[ipsc_mn_als_datasets_with_lee.vsd.ntop_genes,])) # perform a PCA on the data in assay(x) for the selected genes
ipsc_mn_als_datasets_with_lee.vsd.loadings <- as_tibble(ipsc_mn_als_datasets_with_lee.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
ipsc_mn_als_datasets_with_lee.vsd.loadings %>% arrange(PC1)
ipsc_mn_als_datasets_with_lee.vsd.loadings %>% arrange(-PC1)

# ipsc_mn_als_datasets_with_lee.vsd.pca <- prcomp(t(assay(ipsc_mn_als_datasets_with_lee$vsd))) # perform a PCA on the data in assay(x) for the selected genes
# ipsc_mn_als_datasets_with_lee.vsd.loadings <- as_tibble(ipsc_mn_als_datasets_with_lee.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")
HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
neuronal.markers = c(nonspecific.neuronal, dl.domains, V0.domain,V1, V2a.domain, V2b.domain,MN.domain,V3.domain)
# nonpolyadenylated_genes = read_csv(here(camp_path,"genomes/annotation/nonpolyadenylated_genes.csv")) %>% pull(nonpolyadenylated_genes)

# scatter all PC1 vs PC2
ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter = ggplot(ipsc_mn_als_datasets_with_lee.vsd.loadings, aes(x = PC1, y = PC2 )) + 
    labs(x = "PC1 gene loading", y = "PC2 gene loading") +
    geom_hline(yintercept = 0, linetype = 3) + geom_vline(xintercept = 0, linetype = 3) +
    geom_point(size = 1, alpha = 0.1) +
    scale_fill_continuous(type = "viridis") +
    guides(fill = FALSE) +
    theme_oz() +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings, abs(PC1) > 0.04, abs(PC2) > 0.04, grepl("^H[1-9]",gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings, abs(PC1) > 0.04, abs(PC2) > 0.04, grepl("^SNOR|RN7SK|RMRP|RPPH1",gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings,  gene_name %in% c(neuronal.markers, HOX.markers)), aes(x = PC1, y = PC2, label = gene_name), colour = "black", size = 2.3) +
  annotate("text", x = 0.04, y = 0.05, label = "PC1+ PC2+",size = 3, color = "black") + annotate("text", x = -0.07, y = -0.12, label = "PC1- PC2-", size = 3, color = "firebrick2") + 
  annotate("text", x = 0.04, y = -0.12, label = "PC1+ PC2-", size = 3, color = "black") + annotate("text", x = -0.07, y = 0.05, label = "PC1- PC2+", size = 3, color = "black")
ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter
ggsave(ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter.png"), height = 9, width = 9)


# # GO analysis on the 4 quadrants of scatterplot:
# # revigo
# ipsc_mn_als_datasets_with_lee.vsd.loadings_gost <- ipsc_mn_als_datasets_with_lee.vsd.loadings %>% group_split(PC1 > 0, PC2 > 0) %>% map("gene_id") %>% gost() 
# ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt = ipsc_mn_als_datasets_with_lee.vsd.loadings_gost$result %>% filter(str_detect(source, paste0("GO:BP"))) %>% arrange(-log10(p_value)) %>% mutate(query = case_when(query == "query_1" ~ "PC1- PC2-", query == "query_2" ~ "PC1- PC2+", query == "query_3" ~ "PC1+ PC2-", query == "query_4" ~ "PC1+ PC2+"), term_name = as.factor(term_name))
# 
# ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query1 <- ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt %>% filter(query == "PC1- PC2-") %>% top_n(10, wt = -p_value)
# paste('"',unique(ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query1$term_name),'",', sep = "", collapse = '\n') %>% cat()
# 
# ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query2 <- ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt %>% filter(query == "PC1- PC2+") %>% top_n(10, wt = -p_value)
# paste('"',unique(ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query2$term_name),'",', sep = "", collapse = '\n') %>% cat()
# 
# ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query3 <- ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt %>% filter(query == "PC1+ PC2-") %>% top_n(10, wt = -p_value)
# paste('"',unique(ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query3$term_name),'",', sep = "", collapse = '\n') %>% cat()
# 
# ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query4 <- ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt %>% filter(query == "PC1+ PC2+") %>% top_n(10, wt = -p_value)
# paste('"',unique(ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query4$term_name),'",', sep = "", collapse = '\n') %>% cat()
# 
# reducedTerms_combined <- bind_rows(mutate(ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query1, query = "PC1- PC2-"), mutate(ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query2, query = "PC1- PC2+"), mutate(ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query3, query = "PC1+ PC2-"), mutate(ipsc_mn_als_datasets_with_lee.vsd.loadings_gost.res_filt.query4, query = "PC1+ PC2+")) %>% 
#   mutate(query = factor(query, levels = c("PC1- PC2+", "PC1+ PC2+", "PC1+ PC2-", "PC1- PC2-")))
# 
# ipsc_mn_als_datasets_with_lee.vsd.loadings.go_curated <- curated_profiles(gprofiles = reducedTerms_combined, label_angle = 0, cols =  c("gold2", "forestgreen", "dodgerblue2", "firebrick2")) #+ theme_oz()
# ipsc_mn_als_datasets_with_lee.vsd.loadings.go_curated

# pca_gene_loadings = ggarrange(ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter, ipsc_mn_als_datasets_with_lee.vsd.loadings.go_curated, nrow = 2, labels = c("a","b"))
# pca_gene_loadings
# ggsave(pca_gene_loadings, filename = here(proj_path,"expression/deseq2/pca_gene_loadings.png", height = 12, width = 9)

```

Dendogram

```{r}
# Dendrogram
ipsc_mn_als_datasets_with_lee.sampleDists <- dist(t(assay(ipsc_mn_als_datasets_with_lee$vsd)))
ipsc_mn_als_datasets_with_lee.hc <- hclust(ipsc_mn_als_datasets_with_lee.sampleDists, method = "ward.D2")
ipsc_mn_als_datasets_with_lee.hc$labels <- gsub("dafinca.tardbp", "dafinca", ipsc_mn_als_datasets_with_lee.hc$labels) %>% gsub("dafinca.c9orf72", "dafinca", .) %>% gsub("mn_d25_", "", .)

display.brewer.pal(n = 12, name = 'Paired')
brewer.pal(n = 12, name = "Paired")
show_col(brewer_pal(palette = "Paired")(12))
"#A6CEE3" # light blue
"#1F78B4" 
"#B2DF8A" # green
"#33A02C"
"#FB9A99" # red
"#E31A1C" 
"#FDBF6F" # orange
"#FF7F00"
"#CAB2D6" # purple
"#6A3D9A" 
"#B15928 # brown
"#FFFF99" # yellow - replace with darker yellow for readability #F0E442
ipsc_mn_als_datasets_with_lee.a <- c(rep("#386CB0", 3), rep("#75489F", 9), rep("#386CB0", 6), 
                                                   rep("#FEEA92", 6), # desantis
                                                   rep("#BCCEA0", 4), # wang
                                                   rep("#7A9DA8", 6), # sterneckert
                                                   rep("#A9B7B7", 6),# kapeli
                                                   rep("#E8BAA0", 6), # mitchell
                                                   rep("#D3B4BA", 6), #luisier
                                                  rep("#FDC086", 8),#sareen
                                                 rep("#7FC97F", 12), # catanese
                                                 rep("#BEAED4", 5), #kiskinis
                            rep("#94C09B", 12), # dafinca
                            rep("#FDD58C", 5), # smith
                            rep("#94C09B", 10)) # dafinca
# PCA colour assignment:
# c("HB9::GFP CTRL"="#386CB0", "HB9::GFP ALS"="#386CB0", "PHOX2B::GFP CTRL"="#75489F", "PHOX2B::GFP ALS"="#75489F", "desantis"="#FEEA92", "catanese"="#7FC97F", "dafinca"="#94C09B", 
                                # "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "mitchell"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sterneckert"="#7A9DA8", "wang"="#BCCEA0"))
ipsc_mn_als_datasets_with_lee.dendrogram <- ggdendrogram(ipsc_mn_als_datasets_with_lee.hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(size = 6), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
ipsc_mn_als_datasets_with_lee.dendrogram
ggsave(ipsc_mn_als_datasets_with_lee.dendrogram, filename = here(proj_path,"expression/deseq2/dendrogram.png", height = 20, width = 8.25)

# ### confirm hclust(method="single") for fairness - Lee samples are the last to join the other datasets. 
# ipsc_mn_als_datasets_with_lee.hc.single <- hclust(ipsc_mn_als_datasets_with_lee.sampleDists, method = "single")
# ipsc_mn_als_datasets_with_lee.hc.single$labels <- gsub("dafinca.tardbp", "dafinca", ipsc_mn_als_datasets_with_lee.hc.single$labels) %>% gsub("dafinca.c9orf72", "dafinca", .) %>% gsub("mn_d25_", "", .) %>% gsub("REP", "", .) %>% gsub("lee_spinal", "HB9::GFP",.) %>% gsub("lee_ocular", "PHOX2B::GFP", .)
# ipsc_mn_als_datasets_with_lee.a.single =  c(rep("#A6CEE3", 3), rep("#1F78B4", 3), rep("black", 86), 
#                                                    rep("#A6CEE3", 6), # desantis
#                                                    rep("#1F78B4", 6))
# ipsc_mn_als_datasets_with_lee.dendrogram <- ggdendrogram(ipsc_mn_als_datasets_with_lee.hc.single, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = ipsc_mn_als_datasets_with_lee.a.single, size = 8), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
# ipsc_mn_als_datasets_with_lee.dendrogram

```

Sample-to-sample distances

```{r}
rownames(colData(ipsc_mn_als_datasets_with_lee$vsd)) = rownames(colData(ipsc_mn_als_datasets_with_lee$vsd)) %>% gsub("dafinca.tardbp", "dafinca",.) %>% gsub("dafinca.c9orf72", "dafinca", .) %>% gsub("mn_d25_", "", .)
sampleDists <- dist(t(assay(ipsc_mn_als_datasets_with_lee$vsd)))
# library("RColorBrewer")
# PCA colour assignment:
# c("HB9::GFP CTRL"="#386CB0", "HB9::GFP ALS"="#386CB0", "PHOX2B::GFP CTRL"="#75489F", "PHOX2B::GFP ALS"="#75489F", "desantis"="#FEEA92", "catanese"="#7FC97F", "dafinca"="#94C09B", 
                                # "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "mitchell"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sterneckert"="#7A9DA8", "wang"="#BCCEA0"))

sampleDistMatrix <- as.matrix(sampleDists)
colnames(sampleDistMatrix) <- NULL
rownames(sampleDistMatrix)
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
ipsc_mn_als_datasets_with_lee.sample2sample_dists<- Heatmap(sampleDistMatrix,
                                                   border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
        clustering_distance_rows = "euclidean", clustering_distance_columns = "euclidean",
        clustering_method_rows  = "average", clustering_method_columns  = "average",
        col=colors,
        # name="Distance",
        row_names_gp = gpar(fontsize = 6))#, col = c(rep("#7FC97F",12), rep("#386CB0",6), rep("#75489F",9), rep("#386CB0",3), rep("#94C09B",22), rep("#D3B4BA",6), rep("#E8BAA0",6), rep("#BCCEA0",4), rep("#BEAED4",5), rep("#FDC086",8), rep("#7A9DA8",6), rep("#A9B7B7",6), rep("#FEEA92",6), rep("#FDD58C",5))))
ipsc_mn_als_datasets_with_lee.sample2sample_dists
ggsave(as.ggplot(ipsc_mn_als_datasets_with_lee.sample2sample_dists), filename = here(proj_path,"expression/deseq2/sample2sample_dists.png", height = 20, width = 8.25)
```




# Fig S6-9 Heatmap expression

## Cell type markers

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","EDNRB",  "FGFR3",  "SLC4A4", "DIO2")#"SOX9", ,"MLC1"
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4", "PECAM1")
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CCL3","IL1A", "TREM2") # "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "CX3CR1","P2RY12", "TNF",  "TMEM119"
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1")

undifferentiated = c("SOX2","POU5F1","UTF1","REX1","ALPL","TERT","ESRG","CNMD","SFRP2")
neural.precursor = c("NES","SOX1","SOX2","CD133","PAX6","NOTCH1","OCLN","CDH1","CDH2","SOX10","VIM","GFAP")
neural.progenitor = c("MAP2","HUC","HUD","NF","NCAM","DCX","SOX2")
neural.crest <- c("SOX10","NGFR","FOXD3","SNAI1","SNAI2","LMX1A")   #NGFR very low in HB9 samples
dorsal.progenitor <- c("ATOH1","OLIG3","MSX1","PAX3","PAX7","DBX1","DBX2","IRX3","ASCL1")
ho.genes = c("SST","DCX","RFX4","ASCL1","HOXB8","NEFH","SCN1A","TNS1","SPOCK3","SNAP25","TDGF1","POU5F1","L1TD1","FXYD5","PHLDA2","TOP2A","DLGAP5","FZD7","ILF2","MSH2")

nonspecific = c("SOX2")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
pMN.domain = c("NKX6-1", "OLIG2")
ventral.progenitor = c("NKX2-2","NKX2-8")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")

nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

# neuronal.markers = c("NPAS4", "NPY", "SST", "CCK", "GAD1", "SYT1", "STMN2", "SYN1") # "ENO2", 

nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#"NEFH",
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3") #"ETV1", "NOS1",,"RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2"
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

### Cell type markers
celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, oligo.markers, mn.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]
celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte.markers", gene_name %in% endo.markers ~ "endo.markers", 
                           gene_name %in% myeloid.markers ~ "myeloid.markers", gene_name %in% oligo.markers ~ "oligo.markers", 
                           gene_name %in% mn.markers ~ "mn.markers"), 
         group = factor(group, levels = c("astrocyte.markers", "endo.markers", "myeloid.markers", "oligo.markers","mn.markers"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets_with_lee.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = as.factor(dataset)) %>% filter(!grepl("answerals_CASE",dataset_sample))

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
celltype.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  # rename_with(~ gsub("dafinca.tardbp", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafinca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE")) # keep only 100 Answer ALS samples to improve visualisaiton
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))

### Cluster samples ###
celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(celltype.markers.marker.centered_mat)]

celltype.markers.rnaseq.cluster.heatmap <- Heatmap(t(celltype.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = celltype.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = celltype.markers.df.filt$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelial", "Myeloid", "Oligodendrocyte","Motor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), column_title = NULL, 
          show_row_names = FALSE, #row_names_gp = gpar(fontsize = 6), 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)),  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
celltype.markers.rnaseq.cluster.heatmap
ggsave(as.ggplot(celltype.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.celltype.markers.heatmap.png"), width = 8, height = 13)
```


## Neuron type markers

```{r}
nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#,"NEFH"
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3","ETV1", "NOS1","RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2") #
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

neuron.type.markers = c(motor.neuron.progenitor, postmitotic.motor.neuron, nonspecific.neuronal,interneuron.markers) 
neuron.type.markers[!neuron.type.markers %in% gtf$gene_name]
neuron.type.markers.df = tibble("gene_name" = neuron.type.markers) %>% filter(gene_name %in% neuron.type.markers) %>% 
  mutate(group = case_when(gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% motor.neuron.progenitor ~ "motor.neuron.progenitor", 
                           gene_name %in% postmitotic.motor.neuron ~ "postmitotic.motor.neuron", #gene_name %in% oculomotor.markers ~ "oculomotor.markers", 
                           gene_name %in% interneuron.markers ~ "interneuron"), 
         group = factor(group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets_with_lee.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = as.factor(dataset)) %>% filter(!grepl("answerals_CASE",dataset_sample))

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
neuron.type.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% neuron.type.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuron.type.markers.df$gene_name)) %>% select(-gene_id) %>% 
  # rename_with(~ gsub("dafinca.tardbp", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafinca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE")) # keep only 100 Answer ALS samples to improve visualisaiton
neuron.type.markers.df.filt = neuron.type.markers.df %>% filter(gene_name %in% neuron.type.markers.marker$gene_name)
neuron.type.markers.marker.mat <- make_matrix(select(neuron.type.markers.marker,-gene_name), pull(neuron.type.markers.marker,gene_name))

rownames(neuron.type.markers.marker.centered_mat)[!rownames(neuron.type.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(neuron.type.markers.marker.centered_mat)]

### Cluster samples ###
neuron.type.markers.marker.centered_mat = t(scale(t(neuron.type.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
neuron.type.markers.rnaseq.heatmap <- Heatmap(t(neuron.type.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = neuron.type.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = factor(neuron.type.markers.df.filt$group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron")),
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",4)), labels = c("nonspecific\nneuronal", "motor neuron\nprogenitor", "postmitotic\nmotor neuron", "interneuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, 
          show_row_names = FALSE,
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
neuron.type.markers.rnaseq.heatmap
ggsave(as.ggplot(neuron.type.markers.rnaseq.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.neuron.type.markers.heatmap.png"), width = 8, height = 13)

# ## clustering samples
# neuron.type.markers.rnaseq.cluster.heatmap <- Heatmap(t(neuron.type.markers.marker.mat), 
#           border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
#           column_order = neuron.type.markers.df.filt$gene_name,
#           column_names_gp = gpar(fontsize = 6), 
#           column_split = factor(neuron.type.markers.df.filt$group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron")),
#           bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",4)), labels = c("nonspecific\nneuronal", "motor\nneuron\nprogenitor", "postmitotic\nmotor\nneuron", "interneuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           column_title = NULL, 
#           show_row_names = TRUE, row_names_gp = gpar(fontsize = 6), 
#           row_title = NULL, cluster_rows = TRUE, cluster_column_slices = FALSE, cluster_columns = FALSE)
# neuron.type.markers.rnaseq.cluster.heatmap
# ggsave(as.ggplot(neuron.type.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.neuron.type.markers.cluster.heatmap.png"), width = 9, height = 16)
# 
## Motor neuron markers only heatmap
# celltype.marker.df = data.frame("gene_name" = mn.markers) %>% mutate(group = case_when(gene_name %in% mn.markers ~ "motor neuron"))
# ipsc_mn_als_datasets_with_lee$vsd.counts %>% filter(gene_name %in% mn.markers) %>% group_by(gene_name) %>% filter(n()>1) # any duplicate gene_names?
# 
# vsd_counts.celltype.marker <-  ipsc_mn_als_datasets_with_lee$vsd.counts  %>% filter(gene_name %in% mn.markers) %>% select(-gene_id) %>% mutate_if(is.numeric, funs((. - 4.505))) #rowwise() %>%
# vsd_counts.celltype.marker.mat <- make_matrix(select(vsd_counts.celltype.marker,-gene_name), pull(vsd_counts.celltype.marker,gene_name))
# celltype.marker.df.filt <- celltype.marker.df %>% filter(gene_name %in% vsd_counts.celltype.marker$gene_name)
# 
# motor.neuron.marker.heatmap <- Heatmap(t(vsd_counts.celltype.marker.mat), 
#           heatmap_legend_param = list(title = ""), 
#           cluster_columns = TRUE, column_order = vsd_counts.celltype.marker$gene_name,
#           show_column_names = TRUE, column_names_gp = gpar(fontsize = 8, fontface = "italic"), #row_labels = celltype.marker.row.labels,
#           top_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = c("white")), labels = c("Motor neuron markers"), labels_gp = gpar(fontsize = 10))), column_title = NULL,
#           show_row_names = FALSE, #column_names_rot = 90, 
#           row_title = NULL, cluster_rows = FALSE,
#           right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
#                                                                  labels = c("AnswerALS", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Luisier",  "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "bhinge", "Smith", "Sterneckert", "Wang"), 
#                                                                  labels_gp = gpar(fontsize = 7), labels_rot = 0)), row_split = dataset.samples.df$dataset)
# 
# motor.neuron.marker.heatmap

```


## Dorsoventral markers

From Fig 2 in https://journals.biologists.com/dev/article/148/15/dev199711/271192/Single-cell-transcriptome-profiling-of-the-human 

```{r}
nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

neuronal.markers = c(dl.domains, V0.domain,V1, V2a.domain, V2b.domain,MN.domain,V3.domain) #nonspecific.neuronal
neuronal.markers[!neuronal.markers %in% gtf$gene_name]
neuronal.markers.df = tibble("gene_name" = neuronal.markers) %>% 
  mutate(group = case_when(gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% dl.domains ~ "dl.domains", 
                           gene_name %in% V0.domain ~ "V0.domain", gene_name %in% V1 ~ "V1", 
                           gene_name %in% V2a.domain ~ "V2a.domain", gene_name %in% V2b.domain ~ "V2b.domain", gene_name %in% MN.domain ~ "MN.domain", gene_name %in% V3.domain ~ "V3.domain"), 
         group = factor(group, levels = c("dl.domains","V0.domain", "V1", "V2a.domain", "V2b.domain", "MN.domain", "V3.domain"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
neuronal.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% neuronal.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuronal.markers.df$gene_name)) %>% select(-gene_id) %>%
  # rename_with(~ gsub("dafinca.tardbp", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafinca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE")) # keep only 100 Answer ALS samples to improve visualisaiton
neuronal.markers.marker.mat <- make_matrix(select(neuronal.markers.marker,-gene_name), pull(neuronal.markers.marker,gene_name))

### Cluster samples ###
neuronal.markers.marker.centered_mat = t(scale(t(neuronal.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
dorsoventral.markers.heatmap <- Heatmap(t(neuronal.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = neuronal.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = neuronal.markers.df$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",7)), labels =  c("dl\ndomains","V0\ndomain", "V1", "V2a\ndomain", "V2b\ndomain", "MN\ndomain", "V3"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          # show_row_names = TRUE, row_names_gp = gpar(fontsize = 6, col = c(rep("#386CB0",9), rep("#75489F",9), rep("black",90))), #row_names_rot = 90, 
          cluster_rows = FALSE,
          row_title = NULL, 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)))
dorsoventral.markers.heatmap
ggsave(as.ggplot(dorsoventral.markers.heatmap), filename = here(proj_path,"expression/deseq2/dorsoventral.markers.heatmap.png"), width = 8, height = 13)


# # cluster by columns
# Heatmap(neuronal.markers.marker.centered_mat, 
#           border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
#           row_order = neuronal.markers.df$gene_name,
#           row_names_gp = gpar(fontsize = 7), 
#           row_split = factor(neuronal.markers.df$group, levels =  c("nonspecific.neuronal","dl.domains","V0.domain", "V1", "V2a.domain", "V2b.domain", "MN.domain", "V3.domain")),
#           right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = rep("white",8)), labels =  c("nonspecific\nneuronal","dl\ndomains","V0\ndomain", "V1", "V2a\ndomain", "V2b\ndomain", "MN\ndomain", "V3\ndomain"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           row_title = "Neuronal markers", row_title_gp = gpar(fontsize = 8), 
#           show_column_names = TRUE, column_names_gp = gpar(fontsize = 6, col = c(rep("#386CB0",9), rep("#75489F",9), rep("black",90))), #column_names_rot = 90,
#           cluster_rows = FALSE, cluster_row_slices = FALSE, cluster_columns = TRUE,
#           column_title = NULL)

```

Progenitor dosroventral markers


```{r}
# nonspecific = c("SOX2")
# roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
# dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
# intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
# pMN.domain = c("NKX6-1", "OLIG2")
# ventral.progenitor = c("NKX2-2","NKX2-8")
# floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")
# 
# progenitor.markers = c(nonspecific, roof.plate, dorsal.progenitor,intermediate.progenitor, pMN.domain, ventral.progenitor,floor.plate)
# progenitor.markers[!progenitor.markers %in% gtf$gene_name]
# progenitor.markers.df = tibble("gene_name" = progenitor.markers) %>% 
#   mutate(group = case_when(gene_name %in% nonspecific ~ "nonspecific", gene_name %in% roof.plate ~ "roof.plate", 
#                            gene_name %in% dorsal.progenitor ~ "dorsal.progenitor", gene_name %in% intermediate.progenitor ~ "intermediate.progenitor", 
#                            gene_name %in% pMN.domain ~ "pMN.domain", gene_name %in% ventral.progenitor ~ "ventral.progenitor", gene_name %in% floor.plate ~ "floor.plate"), 
#          group = factor(group, levels = c("nonspecific","roof.plate", "dorsal.progenitor", "intermediate.progenitor", "pMN.domain", "ventral.progenitor", "floor.plate"))) %>% 
#   arrange(group) %>% distinct(gene_name, .keep_all = TRUE)
# 
# # together with other datasets
# min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
# progenitor.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
#   filter(gene_name %in% progenitor.markers.df$gene_name) %>% 
#   dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
#   arrange(factor(gene_name, levels = progenitor.markers.df$gene_name)) %>% select(-gene_id) %>%
#   rename_with(~ gsub("dafinca.tardbp", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafinca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
# progenitor.markers.marker.mat <- make_matrix(select(progenitor.markers.marker,-gene_name), pull(progenitor.markers.marker,gene_name))
# 
# ### Cluster samples ###
# progenitor.markers.marker.centered_mat = t(scale(t(progenitor.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
# progenitor.markers.heatmap <- Heatmap(t(progenitor.markers.marker.mat), 
#           border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
#           column_order = progenitor.markers.df$gene_name,
#           column_names_gp = gpar(fontsize = 7), 
#           column_split = progenitor.markers.df$group,
#           bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",7)), labels =  c("nonspecific","roof\nplate", "dorsal\nprogenitor", "intermediate\nprogenitor", "pMN\ndomain", "ventral\nprogenitor", "floor\nplate"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           column_title ="Progenitor markers", column_title_gp = gpar(fontsize = 8), 
#           show_row_names = FALSE, 
#           # row_order = dataset.samples.df$dataset_sample,
#           row_split = dataset.samples.df$dataset,
#           right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",17)), 
#                                                                  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
#                                                                  labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
# progenitor.markers.heatmap
# 

```


## HOX markers

Regional identity of iPSN

```{r}
HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
# forebrain.markers = c("FOXG1","OTX2")
# HOX.markers = c(forebrain.markers, HOX.markers)
HOX.markers.df <- data.frame(row.names = HOX.markers, gene_name = HOX.markers) %>% 
  mutate(group = gsub("HOX[A-D]","", gene_name), group = factor(as.numeric(gsub("-AS$","", group)))) %>% 
  arrange(group) %>% 
  mutate(level = factor(case_when(group %in% c(1,2,3) ~ "Hindbrain", group %in% c(4,5,6,7) ~ "Cervical", group %in% c(8,9) ~ "Thoracic", 
                                  group %in% c(10,11) ~ "Lumbar", TRUE ~ "Sacral"), levels = c("Hindbrain","Cervical","Thoracic","Lumbar","Sacral")))
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
HOX.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% 
  filter(gene_name %in% HOX.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst))) %>% 
  arrange(factor(gene_name, levels = HOX.markers.df$gene_name)) %>% select(-gene_id) %>%
  # rename_with(~ gsub("dafinca.tardbp", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafinca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE")) # keep only 100 Answer ALS samples to improve visualisaiton
HOX.markers.marker.mat <- make_matrix(select(HOX.markers.marker,-gene_name), pull(HOX.markers.marker,gene_name))
HOX.markers.marker.centered.mat = t(scale(t(HOX.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
HOX.markers.heatmap <- Heatmap(t(HOX.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = HOX.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = HOX.markers.df$level,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Hindbrain", "Cervical", "Thoracic","Lumbar","Sacral"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          cluster_rows = FALSE,
          row_title = NULL, 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",17)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)))
HOX.markers.heatmap
ggsave(as.ggplot(HOX.markers.heatmap), filename = here(proj_path,"expression/deseq2/HOX.markers.heatmap.png"), width = 8, height = 13)
```



# Fig 2 hiPSC ALS meta-analysis

## Volcano

```{r}
ipsc_mn_als_datasets$res %>% filter(padj < 0.05)# 43
# ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)#3
ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #23 down, 20 up
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) #0
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # RNASEL

## Volcano
ipsc_mn_als_datasets.als.labels <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.gene.labels <- ipsc_mn_als_datasets$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.volcano <- volcano_plot(ipsc_mn_als_datasets$res, numerator = "ALS", denomenator = "CTRL",
               subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 2, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.als.labels, ipsc_mn_als_datasets.gene.labels))
ipsc_mn_als_datasets.volcano

## MA plot
ipsc_mn_als_datasets.ma <- ma_plot(ipsc_mn_als_datasets$res, 
               numerator = "ALS", denomenator = "CTRL", subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 1000, dpi = 300, 
               gene_labels = c(ipsc_mn_als_datasets.als.labels, ipsc_mn_als_datasets.gene.labels)) #+ xlim(0,10000)
ipsc_mn_als_datasets.ma


ipsc_mn_als_datasets$res %>% filter(gene_name %in% c("H2AX","TP53BP1","PARP1") | grepl("^RPA[1-4]",gene_name)) # RNASEL
ipsc_mn_als_datasets$res %>% filter(grepl("^RPA[1-9]",gene_name)) # RNASEL

```

## GO terms & GSEA

```{r}
ipsc_mn_als_datasets.gost <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_als_datasets.gprofiles <- ipsc_mn_als_datasets.gost$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
ipsc_mn_als_datasets.gprofiles_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")
ipsc_mn_als_datasets.gprofiles_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(ipsc_mn_als_datasets.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "transcription regulator complex",
# "sequence-specific DNA binding",
# "double-stranded DNA binding",
# "sequence-specific double-stranded DNA binding",
# "chromatin",
"neuron fate commitment",
# "transcription regulatory region nucleic acid binding",
# "transcription cis-regulatory region binding",
# "DNA-binding transcription factor activity",
"cell differentiation in spinal cord",
"DNA-binding transcription factor activity, RNA polymerase II-specific",
"ventral spinal cord development",
"cis-regulatory region sequence-specific DNA binding"
# "Aplasia of the musculature"
)
ipsc_mn_als_datasets.gprofiles_filt_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_als_datasets.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"p53 transcriptional gene network",
"Genotoxicity pathway",
"miRNA regulation of DNA damage response",
"DNA damage response",
"p53 signaling pathway")
ipsc_mn_als_datasets.gprofiles_filt_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_als_datasets.gprofiles_filt_combined <- bind_rows(ipsc_mn_als_datasets.gprofiles_filt_down, ipsc_mn_als_datasets.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
ipsc_mn_als_datasets.gprofiles_filt_combined = ipsc_mn_als_datasets.gprofiles_filt_combined %>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
ipsc_mn_als_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_als_datasets.gprofiles_filt_combined) #+ theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
ipsc_mn_als_datasets.go_curated

# which genes are driving the terms
p53.DDR.repair_gene_set = unique(c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`,gprofiler_database$`DNA repair`))
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway)) # CDKN1A
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(gprofiler_database$`DNA Damage Response`)) # TNFRSF10B, CDKN1A, RRM2B, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA Damage Response`) # TNFRSF10B, CDKN1A, RRM2B, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA repair`) # TNFRSF10B, CDKN1A, RRM2B, SESN1

wikipathways.msigdb <- gmtPathways(here(camp_path,"/genomes/gene-ontology/c2.cp.wikipathways.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
hpo.msigdb <- gmtPathways(here(camp_path,"/genomes/gene-ontology/c5.hpo.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_MIRNA_REGULATION_OF_DNA_DAMAGE_RESPONSE) # TNFRSF10B, CDKN1A, MDM2, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`TP53 Network`) # CDKN1A
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`cell fate commitment`) # TBX5, MYOG, LMO4, OLIG1, TBR1, OLIG2, FOXN4, BCL11B
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`neuron fate commitment`) # OLIG1, OLIG2, FOXN4
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_OLIGODENDROCYTE_SPECIFICATION_AND_DIFFERENTIATION_LEADING_TO_MYELIN_COMPONENTS_FOR_CNS) # OLIG1, OLIG2
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% hpo.msigdb$HP_APLASIA_OF_THE_MUSCULATURE) # CHRND, TBX5
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`ventral spinal cord development`) # LMO4, OLIG2, FOXN4
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA-binding transcription activator activity, RNA polymerase II-specific`) # MYOG, TBX5, POU5F1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$chromatin) # MYOG, TBX5, POU5F1, OLIG1, OLIG2, FOXN4

## GSEA: DNA damage response, p53 signalling pathway
ipsc_mn_als_datasets.geneList <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.geneList) <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ipsc_mn_als_datasets.geneList = sort(ipsc_mn_als_datasets.geneList, decreasing = TRUE)

# p53 signalling pathway
p53.signalling_genes.list <- list("p53.signalling" = ipsc_mn_als_datasets$res$gene_name[ipsc_mn_als_datasets$res$gene_name %in% p53.signalling.pathway])
ipsc_mn_als_datasets.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = ipsc_mn_als_datasets.geneList)
ipsc_mn_als_datasets.p53.signalling_fgseaRes
# pathway        pval        padj   log2err        ES      NES size                             leadingEdge
# 1: p53.signalling 0.003792447 0.003792447 0.4317077 0.4789896 1.509032   95 CDKN1A,MDM2,TRIAP1,GADD45A,BBC3,BAX,...
ipsc_mn_als_datasets.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], ipsc_mn_als_datasets.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("p53 signalling") + xlab("") +
  annotate("text", x = 35000, y = 0.3, label = paste0("NES = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$pval, digits = 3)), size = 2.8, hjust = 0) + theme_oz()
ipsc_mn_als_datasets.p53.signalling.gseaplot

```


## DoRoTHEA & PROGENy

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/pw_bk.html 
<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.res.matrix <- ipsc_mn_als_datasets$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_progeny.pathwayActivity <- ggplot(ipsc_mn_als_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "ALS vs CTRL Normalised enrichment") +
    stat_pvalue_manual(ipsc_mn_als_progeny.contrast_acts, label = "p.signif", y.position = 14, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_als_progeny.pathwayActivity

# p53 weights
ipsc_mn_als_progeny.p53weights = net_progeny %>% filter(source == "p53", target %in% ipsc_mn_als_datasets$res$gene_name) %>% left_join(select(ipsc_mn_als_datasets$res, target=gene_name,stat)) %>% 
  mutate(color = case_when(weight > 0 & stat > 0 ~ '1', weight > 0 & stat < 0 ~ '2', weight < 0 & stat > 0 ~ '2', weight < 0 & stat < 0 ~ '1', TRUE ~ "3"))
ipsc_mn_als_progeny.p53weights.plot = ggplot(ipsc_mn_als_progeny.p53weights, aes(x = weight, y = stat, color = color)) + geom_point(size = 0.5) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_progeny.p53weights, weight > 5, stat > 3), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.1) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "p53 pathway weights", x = "p53 pathway gene weight", y = "ALS vs CTRL test statistic") +
  geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted') +
  coord_cartesian(xlim = c(-8,18), ylim = c(-2.5,5)) 
ipsc_mn_als_progeny.p53weights.plot

```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_dorothea.contrast_acts.labels.up = ipsc_mn_als_dorothea.contrast_acts %>% top_n(score, n = 3) %>% pull(source)
ipsc_mn_als_dorothea.contrast_acts.labels.down = ipsc_mn_als_dorothea.contrast_acts %>% top_n(-score, n = 3) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.contrast_acts, source %in% c(ipsc_mn_als_dorothea.contrast_acts.labels.up, ipsc_mn_als_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_dorothea.contrast_acts.volcano

ipsc_mn_als_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_dorothea.contrast_acts %>% arrange(score)

# # TP53 target genes
ipsc_mn_als_dorothea.tp53 <- net_dorothea %>%  filter(source == "TP53") %>% mutate(gene_name = target) %>% left_join(select(ipsc_mn_als_datasets$res, gene_name,stat,log2FoldChange,pvalue)) %>%
  mutate(color = case_when(mor > 0 & stat > 0 ~ '1', mor > 0 & stat < 0~'2',mor < 0 & stat > 0~'2',mor < 0 & stat < 0~'1',TRUE ~ "3"))

ipsc_mn_als_dorothea.tp53.volcano = ggplot(ipsc_mn_als_dorothea.tp53, aes(x = log2FoldChange, y = -log10(pvalue), color = color, size=abs(mor))) + geom_point() + scale_size(range = c(0, 3)) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(data = filter(ipsc_mn_als_dorothea.tp53, abs(stat) > 3), aes(label = gene_name, size=1), fontface = "italic", max.overlaps = Inf, size = 2.3) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +  geom_vline(xintercept = 0, linetype = 'dotted') + geom_hline(yintercept = 0, linetype = 'dotted') +
    labs(title = "TP53 regulon", x = "log2 fold change (ALS vs CTRL)", y = expression(-log[10]~P~value)) + ylim(c(0,2)) +   geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.tp53, log2FoldChange > 1), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5)
ipsc_mn_als_dorothea.tp53.volcano

ipsc_mn_als_dorothea.tp53 %>% arrange(-stat) # TNFRSF10B, SESN1, RRM2B, CDKN1A, MDM2
```

## Merge

```{r}
ipsc_mn_als_datasets.figs.merged = plot_grid(
  plot_grid(ipsc_mn_als_datasets.volcano, ipsc_mn_als_datasets.go_curated, ncol=2, rel_widths = c(1,0.8), labels = c("a","b")), 
  plot_grid(ipsc_mn_als_datasets.p53.signalling.gseaplot, ipsc_mn_als_progeny.pathwayActivity, ncol=2, rel_widths = c(0.6,1), labels = c("c","d")), 
  plot_grid(ipsc_mn_als_progeny.p53weights.plot,ipsc_mn_als_dorothea.contrast_acts.volcano, ncol=2, labels = c("e","f")),
  nrow = 3, rel_heights = c(1.3,1,1))
# ipsc_mn_als_datasets.figs.merged
ggsave(ipsc_mn_als_datasets.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.figs.merged.png"), height = 10, width = 8.25)

```


# Table S3 DGE panALS

```{r}
ipsc_mn_als_datasets.res.sig = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% select(gene_name, everything())
# ipsc_mn_als_datasets.res.sig
write_csv(ipsc_mn_als_datasets.res.sig, here(proj_path,"expression/deseq2/TableS2.ipsc_mn_als_datasets.res.sig.csv"))

```

# Table S4 Dorothea panALS

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% select(-statistic, -condition, transcription_factor = source, normalised_enrichment_score = score) %>% arrange(p_value, -abs(normalised_enrichment_score))
# ipsc_mn_als_dorothea.contrast_acts
write_csv(ipsc_mn_als_dorothea.contrast_acts, here(proj_path,"expression/deseq2/TableS4.ipsc_mn_als_dorothea.csv"))

```


# Fig S10 No isogenics

## Volcano

```{r}
ipsc_mn_als_datasets.noiso.metadata %>% count(condition) #318 ALS, 90 CTRL
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05)# 82
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)#5
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #45 down, 37 up
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) # 0
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # RNASEL, DDX18, INTS1, DIS3

## Volcano
ipsc_mn_als_datasets.noiso.als.labels <- ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.noiso.gene.labels <- ipsc_mn_als_datasets.noiso$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.noiso.volcano <- volcano_plot(ipsc_mn_als_datasets.noiso$res, numerator = "ALS", denomenator = "CTRL",
               title = "Isogenics removed", subtitle = "89 Controls vs 317 ALS", ymax = 8, xmax = 2, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.noiso.als.labels, ipsc_mn_als_datasets.noiso.gene.labels))
ipsc_mn_als_datasets.noiso.volcano

# ## MA plot
# ipsc_mn_als_datasets.noiso.ma <- ma_plot(ipsc_mn_als_datasets.noiso$res, 
#                numerator = "ALS", denomenator = "CTRL", subtitle = "107 Controls vs 324 ALS", ymax = 8, xmax = 1000, dpi = 300, 
#                gene_labels = c(ipsc_mn_als_datasets.noiso.als.labels, ipsc_mn_als_datasets.noiso.gene.labels)) #+ xlim(0,10000)
# ipsc_mn_als_datasets.noiso.ma

```

## GO terms & GSEA

```{r}
ipsc_mn_als_datasets.noiso.gost <- ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_als_datasets.noiso.gprofiles <- ipsc_mn_als_datasets.noiso.gost$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
ipsc_mn_als_datasets.noiso.gprofiles_down <- ipsc_mn_als_datasets.noiso.gprofiles %>% filter(query == " ALS Down ")
ipsc_mn_als_datasets.noiso.gprofiles_up <- ipsc_mn_als_datasets.noiso.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(ipsc_mn_als_datasets.noiso.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"ventral spinal cord development",
"cell fate commitment")
ipsc_mn_als_datasets.noiso.gprofiles_filt_down <- ipsc_mn_als_datasets.noiso.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_als_datasets.noiso.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"Transcriptional Regulation by TP53",
# "lymph node",
"p53 transcriptional gene network",
"miRNA regulation of DNA damage response",
"DNA damage response",
"p53 signaling pathway",
"Genotoxicity pathway")
ipsc_mn_als_datasets.noiso.gprofiles_filt_up <- ipsc_mn_als_datasets.noiso.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_als_datasets.noiso.gprofiles_filt_combined <- bind_rows(ipsc_mn_als_datasets.noiso.gprofiles_filt_down, ipsc_mn_als_datasets.noiso.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
ipsc_mn_als_datasets.noiso.gprofiles_filt_combined = ipsc_mn_als_datasets.noiso.gprofiles_filt_combined %>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name), term_name = gsub("TP53 Regulates Transcription of Cell Death Genes", "TP53 transcription regulation",term_name))
ipsc_mn_als_datasets.noiso.go_curated <- curated_profiles(gprofiles = ipsc_mn_als_datasets.noiso.gprofiles_filt_combined) #+ theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
ipsc_mn_als_datasets.noiso.go_curated

# which genes are driving the terms
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway)) # TRIAP1, CDKN1A, GADD45A
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% c(gprofiler_database$`DNA Damage Response`)) # TNFSF10B, SESN1, CDKN1A, RRM2B, GADD45A
wikipathways.msigdb <- gmtPathways(here(camp_path,"/genomes/gene-ontology/c2.cp.wikipathways.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
hpo.msigdb <- gmtPathways(here(camp_path,"/genomes/gene-ontology/c5.hpo.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_MIRNA_REGULATION_OF_DNA_DAMAGE_RESPONSE) # TNFRSF10B, CDKN1A, MDM2, SESN1
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`TP53 Network`) # CDKN1A
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`cell fate commitment`) # TBX5, MYOG, LMO4, OLIG1, TBR1, OLIG2, FOXN4, BCL11B
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`neuron fate commitment`) # OLIG1, OLIG2, FOXN4
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`ventral spinal cord development`) # LMO4, OLIG2, FOXN4
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA-binding transcription activator activity, RNA polymerase II-specific`) # MAFF, POU5F1
ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$chromatin) # POU5F1, OLIG1, OLIG2, FOXN4

## GSEA: DNA damage response, p53 signalling pathway
ipsc_mn_als_datasets.noiso.geneList <- ipsc_mn_als_datasets.noiso$res %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.noiso.geneList) <- ipsc_mn_als_datasets.noiso$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ipsc_mn_als_datasets.noiso.geneList = sort(ipsc_mn_als_datasets.noiso.geneList, decreasing = TRUE)

# p53 signalling pathway
p53.signalling.pathway = unique(c(gprofiler_database$`TP53 Network`, go.pathways.msigdb$GO_P53_BINDING, go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_SIGNAL_TRANSDUCTION_BY_P53_CLASS_MEDIATOR, go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_DNA_DAMAGE_RESPONSE_SIGNAL_TRANSDUCTION_BY_P53_CLASS_MEDIATOR, go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_INTRINSIC_APOPTOTIC_SIGNALING_PATHWAY_BY_P53_CLASS_MEDIATOR))
p53.signalling_genes.list <- list("p53.signalling" = ipsc_mn_als_datasets.noiso$res$gene_name[ipsc_mn_als_datasets.noiso$res$gene_name %in% p53.signalling.pathway])
ipsc_mn_als_datasets.noiso.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = ipsc_mn_als_datasets.noiso.geneList)
ipsc_mn_als_datasets.noiso.p53.signalling_fgseaRes
#  pathway      pval      padj   log2err        ES      NES size                              leadingEdge
# 1: p53.signalling 0.0133884 0.0133884 0.3807304 0.4554388 1.422513   95 TRIAP1,CDKN1A,GADD45A,BBC3,MDM2,PLK3,...
ipsc_mn_als_datasets.noiso.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], ipsc_mn_als_datasets.noiso.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("p53 signalling") + xlab("") +
  annotate("text", x = 35000, y = 0.3, label = paste0("NES = ", round(ipsc_mn_als_datasets.noiso.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(ipsc_mn_als_datasets.noiso.p53.signalling_fgseaRes$pval, digits = 3)), size = 2.8, hjust = 0) + theme_oz()
ipsc_mn_als_datasets.noiso.p53.signalling.gseaplot

```


## DoRoTHEA & PROGENy

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/pw_bk.html 
<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.noiso.res.matrix <- ipsc_mn_als_datasets.noiso$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als.noiso_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.noiso.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als.noiso_progeny.pathwayActivity <- ggplot(ipsc_mn_als.noiso_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "ALS vs CTRL Normalised enrichment") +
    stat_pvalue_manual(ipsc_mn_als_progeny.contrast_acts, label = "p.signif", y.position = 12, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_als.noiso_progeny.pathwayActivity

# p53 weights
ipsc_mn_als.noiso_progeny.p53weights = net_progeny %>% filter(source == "p53", target %in% ipsc_mn_als_datasets.noiso$res$gene_name) %>% left_join(select(ipsc_mn_als_datasets.noiso$res, target=gene_name,stat)) %>% 
  mutate(color = case_when(weight > 0 & stat > 0 ~ '1', weight > 0 & stat < 0 ~ '2', weight < 0 & stat > 0 ~ '2', weight < 0 & stat < 0 ~ '1', TRUE ~ "3"))
ipsc_mn_als.noiso_progeny.p53weights.plot = ggplot(ipsc_mn_als.noiso_progeny.p53weights, aes(x = weight, y = stat, color = color)) + geom_point(size = 0.5) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_progeny.p53weights, weight > 5, stat > 3), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "p53 pathway weights", x = "p53 pathway gene weight", y = "ALS vs CTRL test statistic") +
  geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted') +
  coord_cartesian(xlim = c(-9,19)) 
ipsc_mn_als.noiso_progeny.p53weights.plot

```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als.noiso_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.noiso.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als.noiso_dorothea.contrast_acts.labels.up = ipsc_mn_als.noiso_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als.noiso_dorothea.contrast_acts.labels.down = ipsc_mn_als.noiso_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als.noiso_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als.noiso_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als.noiso_dorothea.contrast_acts, source %in% c(ipsc_mn_als.noiso_dorothea.contrast_acts.labels.up, ipsc_mn_als.noiso_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als.noiso_dorothea.contrast_acts.volcano

ipsc_mn_als_dorothea.contrast_acts %>% arrange(-score)
 ipsc_mn_als_dorothea.contrast_acts %>% arrange(score)

# # TP53 target genes
ipsc_mn_als.noiso_dorothea.tp53 <- net_dorothea %>%  filter(source == "TP53") %>% mutate(gene_name = target) %>% left_join(select(ipsc_mn_als_datasets.noiso$res, gene_name,stat,log2FoldChange,pvalue)) %>%
  mutate(color = case_when(mor > 0 & stat > 0 ~ '1', mor > 0 & stat < 0~'2',mor < 0 & stat > 0~'2',mor < 0 & stat < 0~'1',TRUE ~ "3"))

ipsc_mn_als.noiso_dorothea.tp53.volcano = ggplot(ipsc_mn_als.noiso_dorothea.tp53, aes(x = log2FoldChange, y = -log10(pvalue), color = color, size=abs(mor))) + geom_point() + scale_size(range = c(0, 3)) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(data = filter(ipsc_mn_als.noiso_dorothea.tp53, abs(stat) > 3), aes(label = gene_name, size=1), fontface = "italic", max.overlaps = Inf, size = 2.3) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +  geom_vline(xintercept = 0, linetype = 'dotted') + geom_hline(yintercept = 0, linetype = 'dotted') +
    labs(title = "TP53 regulon", x = "log2 fold change (ALS vs CTRL)", y = expression(-log[10]~P~value)) + ylim(c(0,2))
ipsc_mn_als.noiso_dorothea.tp53.volcano

ipsc_mn_als_dorothea.tp53 %>% arrange(-stat) # TNFRSF10B, SESN1, RRM2B, CDKN1A, MDM2
```

## Merge

```{r}
ipsc_mn_als_datasets.noiso.figs.merged = plot_grid(
  plot_grid(ipsc_mn_als_datasets.noiso.volcano, ipsc_mn_als_datasets.noiso.go_curated, ncol=2, rel_widths = c(1,0.8), labels = c("a","b")), 
  plot_grid(ipsc_mn_als_datasets.noiso.p53.signalling.gseaplot, ipsc_mn_als.noiso_progeny.pathwayActivity, ncol=2, rel_widths = c(0.6,1), labels = c("c","d")), 
  plot_grid(ipsc_mn_als.noiso_progeny.p53weights.plot,ipsc_mn_als.noiso_dorothea.contrast_acts.volcano, ncol=2, labels = c("e","f")),
  nrow = 3, rel_heights = c(1.3,1,1))
# ipsc_mn_als_datasets.noiso.figs.merged
ggsave(ipsc_mn_als_datasets.noiso.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.noiso.figs.merged.png"), height = 10, width = 8.25)

```

# Table S5 no isogenics DGE

```{r}
ipsc_mn_als_datasets.nosio.res.sig = ipsc_mn_als_datasets.noiso$res %>% filter(padj < 0.05) %>% select(gene_name, everything())
# ipsc_mn_als_datasets.nosio.res.sig
write_csv(ipsc_mn_als_datasets.nosio.res.sig, here(proj_path,"expression/deseq2/TableS5.ipsc_mn_als_datasets.nosio.res.sig.csv"))
```


# Fig S11 polyA & total sensitivity analyses

## polyA

```{r}
# ipsc_mn_als_datasets.polyA = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.rds"))
# ipsc_mn_als_datasets.total = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.rds"))

ipsc_mn_als_datasets.polyA.metadata %>% count(dataset) # 10
ipsc_mn_als_datasets.polyA.metadata %>% count(condition) # 48 ALS, 43 CTRL
ipsc_mn_als_datasets.total.metadata %>% count(dataset) # 5
ipsc_mn_als_datasets.total.metadata %>% count(condition) # 275 ALS, 63 CTRL

# PCA
ipsc_mn_als_datasets.polyA.pcaData <- plotPCA(ipsc_mn_als_datasets.polyA$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "poly(A)"), dataset_sample, gender)) 
  
ipsc_mn_als_datasets.polyA.pca_dataset <- ggplot(ipsc_mn_als_datasets.polyA.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.polyA.pcaData, "percentVar"))[1],"% variance")) + ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.polyA.pcaData, "percentVar"))[2],"% variance")) + labs(color="polyA\ndatasets")
ipsc_mn_als_datasets.polyA.pca_dataset

## Volcano
ipsc_mn_als_datasets.polyA.metadata %>% count(condition)
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) # 57
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)# 4
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #32 down, 25 up
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) # PRPH2
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # POLR2J3
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # POLR2J3

ipsc_mn_als_datasets.polyA.als.labels <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.gene.labels <- ipsc_mn_als_datasets.polyA$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.volcano <- volcano_plot(ipsc_mn_als_datasets.polyA$res, numerator = "ALS", denomenator = "CTRL",
               title = "polyA", subtitle = "43 Controls vs 48 ALS", ymax = 10, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.polyA.als.labels, ipsc_mn_als_datasets.polyA.gene.labels,"SESN1","TCEAL6"), distinct_labels = TRUE)
ipsc_mn_als_datasets.polyA.volcano

## DoRoTHEA & PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.polyA.res.matrix <- ipsc_mn_als_datasets.polyA$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.polyA_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.polyA.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.polyA_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.polyA_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "polyA: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.polyA_progeny.contrast_acts, label = "p.signif", y.position = 12, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_als_datasets.polyA_progeny.pathwayActivity

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.polyA.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% filter(p_value < 0.15) %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "polyA: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano

ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% arrange(score)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% filter(source == "TP53")

```


## Total

```{r}
# PCA
ipsc_mn_als_datasets.total.pcaData <- plotPCA(ipsc_mn_als_datasets.total$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "Total"), dataset_sample, gender)) 

ipsc_mn_als_datasets.total.pca_dataset <- ggplot(ipsc_mn_als_datasets.total.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(nrow=2, keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.total.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.total.pcaData, "percentVar"))[2],"% variance"))  + labs(color="Total\ndatasets") 
ipsc_mn_als_datasets.total.pca_dataset

## Volcano
ipsc_mn_als_datasets.total.metadata %>% count(condition)
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) # 10
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)# 0
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #7 down, 3 up
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) # 
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # RNASEL, MRPL18

## Volcano
filter(ipsc_mn_als_datasets.metadata, library_type == "Total") %>% count(condition)
ipsc_mn_als_datasets.total.als.labels <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.total.gene.labels <- ipsc_mn_als_datasets.total$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.total.volcano <- volcano_plot(ipsc_mn_als_datasets.total$res, numerator = "ALS", denomenator = "CTRL",
               title = "Total", subtitle = "63 Controls vs 275 ALS", ymax = 8, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.total.als.labels, ipsc_mn_als_datasets.total.gene.labels,"RNASEL","SESN1", "TCEAL6","MTCO1P12","MTCO2P12","LMO4"))
ipsc_mn_als_datasets.total.volcano

## DoRoTHEA & PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.total.res.matrix <- ipsc_mn_als_datasets.total$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.total_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.total.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.total_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.total_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Total: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.total_progeny.contrast_acts, label = "p.signif", y.position = 8, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_als_datasets.total_progeny.pathwayActivity

ipsc_mn_als_datasets.total_progeny.contrast_acts 

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als_datasets.total_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.total.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.total_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Total: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.total_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano

ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% filter(source == "TP53")
ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% arrange(score)



## Compare polyA & total
# Venn overlap
ipsc_mn_als_datasets.polyA.res.dge.up <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.res.dge.down <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
ipsc_mn_als_datasets.total.res.dge.up <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.total.res.dge.down <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# # ggvenn
ipsc_mn_als_datasets.polyA.total.up_genes <- list("polyA" = ipsc_mn_als_datasets.polyA.res.dge.up, "Total" = ipsc_mn_als_datasets.total.res.dge.up)
ipsc_mn_als_datasets.polyA.total.down_genes <- list("polyA" = ipsc_mn_als_datasets.polyA.res.dge.down, "Total" = ipsc_mn_als_datasets.total.res.dge.down)
ipsc_mn_als_datasets.polyA.total.up_venn <- ggvenn(ipsc_mn_als_datasets.polyA.total.up_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap Up")  + theme(plot.subtitle = element_text(hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, -1, -1, -1), "lines"))
ipsc_mn_als_datasets.polyA.total.up_venn

ipsc_mn_als_datasets.polyA.total.down_venn <- ggvenn(ipsc_mn_als_datasets.polyA.total.down_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap down")  + theme(plot.subtitle = element_text(hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, -1, -1, -1), "lines"))
ipsc_mn_als_datasets.polyA.total.down_venn

ipsc_mn_als_datasets.polyA.res.dge.up[ipsc_mn_als_datasets.polyA.res.dge.up %in% ipsc_mn_als_datasets.total.res.dge.up]
# 
# # Scatter correlation
ipsc_mn_als_datasets.polyA_total.labels = select(ipsc_mn_als_datasets.total$res, gene_id, gene_name, polyA.stat = stat) %>% left_join(select(ipsc_mn_als_datasets.polyA$res, gene_id, gene_name, total.stat = stat)) %>% filter((polyA.stat > 2.5 & total.stat > 2.5) | (polyA.stat < -2.5 & total.stat < -2.5))
ipsc_mn_als_datasets.polyA_total.list = list("polyA: ALS vs CTRL" = ipsc_mn_als_datasets.polyA$res, "Total: ALS vs CTRL" = ipsc_mn_als_datasets.total$res)
ipsc_mn_als_datasets.polyA_total_stat.corr = plot_de_correlation(model_list1 = ipsc_mn_als_datasets.polyA_total.list, model_list2 = ipsc_mn_als_datasets.polyA_total.list, mutation1 = "polyA: ALS vs CTRL", mutation2 = "Total: ALS vs CTRL", highlight = pull(ipsc_mn_als_datasets.polyA_total.labels,gene_name), plot_corr_line = TRUE)
ipsc_mn_als_datasets.polyA_total_stat.corr

```


## Merge

```{r}
ipsc_mn_als_datasets.polyA_total.figs.merged = plot_grid(
  ipsc_mn_als_datasets.polyA.pca_dataset, ipsc_mn_als_datasets.total.pca_dataset,
  ipsc_mn_als_datasets.polyA.volcano, ipsc_mn_als_datasets.total.volcano,
  ipsc_mn_als_datasets.polyA_progeny.pathwayActivity, ipsc_mn_als_datasets.total_progeny.pathwayActivity,
  ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano, ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano, 
  nrow = 4, ncol=2, rel_heights = c(0.7,1,0.8,1), labels = "auto") 
# ipsc_mn_als_datasets.polyA_total.figs.merged
ggsave(ipsc_mn_als_datasets.polyA_total.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA_total.figs.merged.png"), height = 11.75, width = 8.25)

```


# Table S6 DGE total & polyA

```{r}
ipsc_mn_als_datasets.polyA_total.sig = select(ipsc_mn_als_datasets.polyA$res, gene_id, gene_name, padj.polyA=padj, stat.polyA=stat, lfc.polyA=log2FoldChange) %>% left_join(select(ipsc_mn_als_datasets.total$res, gene_id, gene_name, padj.total=padj,stat.total=stat, lfc.total=log2FoldChange)) %>% filter(padj.polyA < 0.05 | padj.total < 0.05) %>% mutate(changed = case_when(padj.polyA < 0.05 & lfc.polyA > 0 ~ "polyA up", padj.polyA < 0.05 & lfc.polyA < 0 ~ "polyA down", padj.total < 0.05 & lfc.total > 0 ~ "total up", padj.total < 0.05 & lfc.total < 0 ~ "total down"))
ipsc_mn_als_datasets.polyA_total.sig
ipsc_mn_als_datasets.polyA_total.sig %>% count(changed)
write_csv(ipsc_mn_als_datasets.polyA_total.sig, here(proj_path,"expression/deseq2/Table.ipsc_mn_als_datasets.polyA_total.sig.csv"))

```

# Fig 3 Compare mutations

Compare sporadic, C9orf72, SOD1, FUS, TARDBP & VCP

```{r}
ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds")) # 84M
ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds")) # 162M
ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds")) # 85M
ipsc_mn_vcp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_vcp_datasets.rds")) # 59M

```

### Volcano

```{r}
ipsc_mn_mutants_bind_datasets.res = bind_rows(mutate(ipsc_mn_sporadic_datasets$res, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$res, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$res, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$res, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$res, mutation = "FUS"))#, mutate(ipsc_mn_vcp_datasets$res, mutation = "VCP"))

ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% count(mutation) %>% arrange(-n)
# mutation     n
#   <chr>    <int>
# 1 TARDBP    3547
# 2 FUS        202
# 3 C9orf72    161
# 4 SOD1         7
# 5 Sporadic     4
ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05)

ipsc_mn_mutants_join_datasets.res = full_join(select(ipsc_mn_c9orf72_datasets$res, gene_name, padj.c9orf72 = padj, log2FoldChange.c9orf72 = log2FoldChange),
          select(ipsc_mn_sporadic_datasets$res, gene_name, padj.sals = padj, log2FoldChange.sals = log2FoldChange)) %>%
  full_join(select(ipsc_mn_fus_datasets$res, gene_name, padj.fus = padj, log2FoldChange.fus = log2FoldChange)) %>%
  full_join(select(ipsc_mn_tardbp_datasets$res, gene_name, padj.tardbp = padj, log2FoldChange.tardbp = log2FoldChange)) %>%
  full_join(select(ipsc_mn_sod1_datasets$res, gene_name, padj.sod1 = padj, log2FoldChange.sod1 = log2FoldChange)) #%>% full_join(select(ipsc_mn_vcp_datasets$res, gene_name, padj.vcp = padj, log2FoldChange.vcp = log2FoldChange))
ipsc_mn_mutants_join_datasets.res.labels = ipsc_mn_mutants_join_datasets.res %>% filter(c(padj.c9orf72 < 0.05 & padj.fus < 0.05 & padj.tardbp < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.tardbp < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.tardbp < 0.05 & padj.c9orf72 < 0.05)) %>% pull(gene_name)

ipsc_mn_tardbp_datasets.metadata %>% count(dataset)
ipsc_mn_tardbp_datasets.metadata %>% count(condition) 
ipsc_mn_tardbp_datasets.labels <- ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.001, gene_name %in% c(ALS_genes,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_tardbp_datasets.volcano <- volcano_plot(ipsc_mn_tardbp_datasets$res, "TARDBP", numerator = "TARDBP", denomenator = "CTRL", subtitle = "48 Controls vs 10 TARDBP", ymax = 15, xmax = 4.5, dpi = 300, gene_labels = c(ipsc_mn_tardbp_datasets.labels,"UPK3BL1","NPIPA8"))
ipsc_mn_tardbp_datasets.volcano

ipsc_mn_fus_datasets.metadata %>% count(dataset) 
ipsc_mn_fus_datasets.metadata %>% count(condition) 
ipsc_mn_fus_datasets.labels <- ipsc_mn_fus_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_fus_datasets.volcano <- volcano_plot(ipsc_mn_fus_datasets$res, "FUS", numerator = "FUS", denomenator = "CTRL", subtitle = "53 Controls vs 12 FUS", ymax = 12, xmax = 4, dpi = 300, gene_labels = c(ipsc_mn_fus_datasets.labels,"UPK3BL1","NPIPA8"))
# ipsc_mn_fus_datasets.volcano

ipsc_mn_c9orf72_datasets.metadata %>% count(dataset)
ipsc_mn_c9orf72_datasets.metadata %>% count(condition) 
ipsc_mn_c9orf72_datasets.labels <- ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.02, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_c9orf72_datasets.volcano <- volcano_plot(ipsc_mn_c9orf72_datasets$res, "C9orf72", numerator = "C9orf72", denomenator = "CTRL", subtitle = "83 Controls vs 60 C9orf72", ymax = 9, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_c9orf72_datasets.labels, "C9orf72","UPK3BL1","NPIPA8"))
# ipsc_mn_c9orf72_datasets.volcano

ipsc_mn_sod1_datasets.metadata %>% count(dataset) 
ipsc_mn_sod1_datasets.metadata %>% count(condition) 
ipsc_mn_sod1_datasets.labels <- ipsc_mn_sod1_datasets$res %>% filter(padj < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_sod1_datasets.volcano <- volcano_plot(ipsc_mn_sod1_datasets$res, "SOD1", numerator = "SOD1", denomenator = "CTRL", subtitle = "63 Controls vs 20 SOD1", ymax = 12, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_sod1_datasets.labels))
# ipsc_mn_sod1_datasets.volcano

ipsc_mn_sporadic_datasets.metadata %>% count(dataset) 
ipsc_mn_sporadic_datasets.metadata %>% count(condition) 
ipsc_mn_sporadic_datasets.labels <- ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_sporadic_datasets.volcano <- volcano_plot(ipsc_mn_sporadic_datasets$res, "Sporadic", numerator = "Sporadic", denomenator = "CTRL", subtitle = "56 Controls vs 208 Sporadic", ymax = 12, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_sporadic_datasets.labels))
# ipsc_mn_sporadic_datasets.volcano

ipsc_mn_vcp_datasets.metadata %>% count(condition)
ipsc_mn_vcp_datasets.labels <- ipsc_mn_vcp_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS,"NUP210","DPP6","GLYATL2","FAM66B","GATD3A","PCDHGA7")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_vcp_datasets.volcano <- volcano_plot(ipsc_mn_vcp_datasets$res, "VCP", numerator = "VCP", denomenator = "CTRL", subtitle = "6 Controls vs 6 VCP", ymax = 11, xmax = 4.5, dpi = 300, gene_labels = ipsc_mn_vcp_datasets.labels)
# ipsc_mn_vcp_datasets.volcano

# volcano_multiplot <- ipsc_mn_tardbp_datasets.volcano + ipsc_mn_fus_datasets.volcano + ipsc_mn_c9orf72_datasets.volcano + ipsc_mn_sod1_datasets.volcano + ipsc_mn_sporadic_datasets.volcano +
#   plot_layout(ncol = 3, nrow = 2)
# volcano_multiplot

ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% ALS_genes]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% RNA_BINDING_PROTEINS]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% p53.signalling.pathway]

```


### Correlation heatmap

```{r}
# heatmap of correlation coefficients
ipsc_mn_mutations_stat = select(ipsc_mn_sporadic_datasets$res, gene_id, Sporadic = stat) %>% 
  left_join(select(ipsc_mn_c9orf72_datasets$res, gene_id, C9orf72 = stat)) %>%
  left_join(select(ipsc_mn_sod1_datasets$res, gene_id, SOD1 = stat)) %>%
  left_join(select(ipsc_mn_fus_datasets$res, gene_id, FUS = stat)) %>%
  left_join(select(ipsc_mn_tardbp_datasets$res, gene_id, TARDBP = stat)) %>%
  left_join(select(ipsc_mn_als_datasets$res, gene_id, panALS = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_id")
cormat <- round(cor(ipsc_mn_mutations_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
dge_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5))
# dge_correlation_ggheatmap

mutation_volcanos_corr_heatmap <- ipsc_mn_tardbp_datasets.volcano + ipsc_mn_fus_datasets.volcano + ipsc_mn_c9orf72_datasets.volcano + ipsc_mn_sod1_datasets.volcano + ipsc_mn_sporadic_datasets.volcano + dge_correlation_ggheatmap +
  plot_layout(ncol = 3, nrow = 2)
# mutation_volcanos_corr_heatmap


ipsc_mn_mutants_join_datasets.res %>% filter(padj.sod1 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # UPK3BL1, NPIPA8
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.sod1 < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.sod1 < 0.05) # 


# # Overlap ALL DEGs
# ipsc_mn_mutants_bind_datasets.dir.res = bind_rows(mutate(filter(ipsc_mn_c9orf72_datasets$res, log2FoldChange > 0), mutation = "C9orf72 Up"), mutate(filter(ipsc_mn_c9orf72_datasets$res, log2FoldChange < 0), mutation = "C9orf72 Down"),
#                                               mutate(filter(ipsc_mn_tardbp_datasets$res, log2FoldChange > 0), mutation = "TARDBP Up"), mutate(filter(ipsc_mn_tardbp_datasets$res, log2FoldChange < 0), mutation = "TARDBP Down"),
#                                               mutate(filter(ipsc_mn_fus_datasets$res, log2FoldChange > 0), mutation = "FUS Up"), mutate(filter(ipsc_mn_fus_datasets$res, log2FoldChange < 0), mutation = "FUS Down"),
#                                               mutate(filter(ipsc_mn_sod1_datasets$res, log2FoldChange > 0), mutation = "SOD1 Up"), mutate(filter(ipsc_mn_sod1_datasets$res, log2FoldChange < 0), mutation = "SOD1 Down"),
#                                               mutate(filter(ipsc_mn_vcp_datasets$res, log2FoldChange > 0), mutation = "VCP Up"), mutate(filter(ipsc_mn_vcp_datasets$res, log2FoldChange < 0), mutation = "VCP Down"))
# ipsc_mn_mutants_bind_datasets.res.direction.upset = ipsc_mn_mutants_bind_datasets.dir.res %>% filter(padj < 0.05) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#   geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.4, hjust = -0.2, size = 2.5, angle = 45) +
#   scale_x_upset(order_by = "degree")  + # n_intersections = 20
#   scale_y_continuous(breaks = NULL, lim = c(0, 2700), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=6), axis.title=element_text(size=8)) +
#   ylab(expression()) +
#   theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.res.direction.upset
# 
# # Overlap UP DEGs
# ipsc_mn_mutants_bind_datasets.upregulated.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#       geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, size = 2.8) +
#     scale_x_upset(order_by = "degree")  + # n_intersections = 20
#     scale_y_continuous(breaks = NULL, lim = c(0, 2500), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + 
#       ylab(expression()) +
#     theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.upregulated.res.upset
# 
# # Overlap DOWN DEGs
# ipsc_mn_mutants_bind_datasets.downregulated.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#       geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, size = 2.8) +
#     scale_x_upset(order_by = "degree")  + # n_intersections = 20
#     scale_y_continuous(breaks = NULL, lim = c(0, 3000), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + 
#       ylab(expression()) +
#     theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.downregulated.res.upset

```

## GO terms

```{r}
# c9orf72
ipsc_mn_c9orf72_datasets.dge_genes <- ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt <- ipsc_mn_c9orf72_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " C9orf72 Down ", query == "query_2" ~ " C9orf72 Up "), term_name = as.factor(term_name))
ipsc_mn_c9orf72_datasets.dge_gprofiles_down <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ")
ipsc_mn_c9orf72_datasets.dge_gprofiles_up <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")
paste('"',unique(ipsc_mn_c9orf72_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "hedgehog family protein binding",
"microtubule bundle formation",
"cytoplasmic region",
"cytoskeleton organization",
"axoneme assembly",
"cilium assembly")
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_down <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_c9orf72_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest",
"miRNA regulation of p53 pathway in prostate cancer",
# "miR-517 relationship with ARCN1 and USP1",
# "retrograde vesicle-mediated transport, Golgi to endoplasmic reticulum",
# "phospholipid transporter activity",
"Genotoxicity pathway",
"p53 signaling pathway")
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_up <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_down, ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" C9orf72 Up ", " C9orf72 Down ")))
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name <- gsub("DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest", "DNA damage response by p53", ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" pathway in prostate cancer", "", ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_c9orf72_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
# ipsc_mn_c9orf72_datasets.go_curated

# fus
ipsc_mn_fus_datasets.dge_genes <- ipsc_mn_fus_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_fus_datasets.dge_gprofiles_filt <- ipsc_mn_fus_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " FUS Down ", query == "query_2" ~ " FUS Up "), term_name = as.factor(term_name))
ipsc_mn_fus_datasets.dge_gprofiles_down <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Down ")
ipsc_mn_fus_datasets.dge_gprofiles_up <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Up ")
paste('"',unique(ipsc_mn_fus_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "synapse",
"neuron to neuron synapse",
# "postsynaptic specialization",
"trans-synaptic signaling",
# "chemical synaptic transmission",
"anterograde trans-synaptic signaling",
"Calcium signaling pathway",
"synaptic signaling")
ipsc_mn_fus_datasets.dge_gprofiles_filt_down <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_fus_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
# "spinal cord interneuron axon guidance",
"Basal transcription factors",
# "metal ion binding",
# "organic substance biosynthetic process",
"regulation of RNA metabolic process",
"DNA binding",
"RNA Polymerase II Transcription",
"transcription regulator activity")
ipsc_mn_fus_datasets.dge_gprofiles_filt_up <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_fus_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_fus_datasets.dge_gprofiles_filt_down, ipsc_mn_fus_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" FUS Up ", " FUS Down ")))
ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name <- gsub("interneuron ", "", ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name)
# ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" pathway", "", ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_fus_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_fus_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
# ipsc_mn_fus_datasets.go_curated

# tardbp
ipsc_mn_tardbp_datasets.dge_genes <- ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_tardbp_datasets.dge_gprofiles_filt <- ipsc_mn_tardbp_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " TARDBP Down ", query == "query_2" ~ " TARDBP Up "), term_name = as.factor(term_name))
ipsc_mn_tardbp_datasets.dge_gprofiles_down <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Down ")
ipsc_mn_tardbp_datasets.dge_gprofiles_up <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Up ")
paste('"',unique(ipsc_mn_tardbp_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"spliceosomal complex",
"mRNA Splicing",
# "DNA IR-damage and cellular response via ATR",
"regulation of RNA metabolic process",
"nervous system development",
# "DNA metabolic process",
# "microtubule cytoskeleton organization",
"microtubule cytoskeleton",
# "chromosome segregation",
"cell division",
# "mitotic cell cycle",
"protein binding")
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_down <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_tardbp_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
# "neuron differentiation",
"glutamatergic synapse",
# "Neuronal System",
# "neurogenesis",
"synaptic signaling",
# "dendritic tree",
"dendrite",
"axon",
"neuron projection",
"synapse")
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_up <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_tardbp_datasets.dge_gprofiles_filt_down, ipsc_mn_tardbp_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" TARDBP Up ", " TARDBP Down ")))
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub("regulation of RNA metabolic process", "RNA metabolic process", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" and cellular response via ATR", "", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub("cellular response to DNA", "DNA", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
# ipsc_mn_tardbp_datasets.go_curated

# # sod1
# ipsc_mn_sod1_datasets.dge_genes <- ipsc_mn_sod1_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_sod1_datasets.dge_gprofiles_filt <- ipsc_mn_sod1_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in sod1 ", query == "query_2" ~ " Up in sod1 "), term_name = as.factor(term_name))
# ipsc_mn_sod1_datasets.dge_gprofiles_down <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Down in sod1 ")
# ipsc_mn_sod1_datasets.dge_gprofiles_up <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Up in sod1 ")
# paste('"',unique(ipsc_mn_sod1_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# ipsc_mn_sod1_datasets.dge_gprofiles_filt_down <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Down in sod1 ") %>% filter(term_name %in% down_list)
# paste('"',unique(ipsc_mn_sod1_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# #no terms for SOD1
# 
# # vcp
# ipsc_mn_vcp_datasets.dge_genes <- ipsc_mn_vcp_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_vcp_datasets.dge_gprofiles_filt <- ipsc_mn_vcp_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA|WP")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in vcp ", query == "query_2" ~ " Up in vcp "), term_name = as.factor(term_name))
# ipsc_mn_vcp_datasets.dge_gprofiles_down <- ipsc_mn_vcp_datasets.dge_gprofiles_filt %>% filter(query == " Down in vcp ")
# ipsc_mn_vcp_datasets.dge_gprofiles_up <- ipsc_mn_vcp_datasets.dge_gprofiles_filt %>% filter(query == " Up in vcp ")
# paste('"',unique(ipsc_mn_vcp_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# paste('"',unique(ipsc_mn_vcp_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# # no terms for VCP
# 
# # sporadic
# ipsc_mn_sporadic_datasets.dge_genes <- ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_sporadic_datasets.dge_gprofiles_filt <- ipsc_mn_sporadic_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in sporadic ", query == "query_2" ~ " Up in sporadic "), term_name = as.factor(term_name))
# ipsc_mn_sporadic_datasets.dge_gprofiles_down <- ipsc_mn_sporadic_datasets.dge_gprofiles_filt %>% filter(query == " Down in sporadic ")
# ipsc_mn_sporadic_datasets.dge_gprofiles_up <- ipsc_mn_sporadic_datasets.dge_gprofiles_filt %>% filter(query == " Up in sporadic ")
# paste('"',unique(ipsc_mn_sporadic_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# paste('"',unique(ipsc_mn_sporadic_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# # none or sporadic

ipsc_mn_c9orf72_fus_tardbp.go_curated.merged = ipsc_mn_c9orf72_datasets.go_curated + ipsc_mn_fus_datasets.go_curated + ipsc_mn_tardbp_datasets.go_curated
ipsc_mn_c9orf72_fus_tardbp.go_curated.merged


ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_DNA_IRDAMAGE_AND_CELLULAR_RESPONSE_VIA_ATR) # OLIG1, OLIG2

# # fus
# ipsc_mn_fus_datasets.rrvgo = gprofiler2rrvgo(ipsc_mn_fus_datasets$res , query1 = "Down", query2 = "Up", GO_cat = "BP", sig.filter = "padj", species = "human")
# ipsc_mn_fus_datasets.rrvgo =  ipsc_mn_fus_datasets.rrvgo %>%  mutate(term = gsub("regulation of ", "", term), term = gsub("homophilic cell adhesion via plasma membrane adhesion molecules", "plasma membrane adhesion", term), parentTerm = term)
# ipsc_mn_fus_datasets.rrvgo.plot = rrvgo_plot(gprofiler2rrvgo = ipsc_mn_fus_datasets.rrvgo, n_terms = 10, remove_terms = NULL, colours = 2, cols =  c("firebrick2", "dodgerblue2"), label_angle = 0)
# 
# # tardbp
# ipsc_mn_tardbp_datasets.rrvgo = gprofiler2rrvgo(ipsc_mn_tardbp_datasets$res , query1 = "Down", query2 = "Up", GO_cat = "BP", sig.filter = "padj", species = "human")
# ipsc_mn_tardbp_datasets.rrvgo =  ipsc_mn_tardbp_datasets.rrvgo %>%  mutate(term = gsub("regulation of ", "", term), parentTerm = term)
# ipsc_mn_tardbp_datasets.rrvgo.plot = rrvgo_plot(gprofiler2rrvgo = ipsc_mn_tardbp_datasets.rrvgo, n_terms = 10, remove_terms = NULL, colours = 2, cols =  c("firebrick2", "dodgerblue2"), label_angle = 0)

# de_go_terms_mutations_merged = ggarrange(NULL,ipsc_mn_c9orf72_datasets.rrvgo.plot,NULL, ipsc_mn_sod1_datasets.rrvgo.plot,NULL, ipsc_mn_fus_datasets.rrvgo.plot,NULL, ipsc_mn_tardbp_datasets.rrvgo.plot, nrow = 8, ncol = 1, labels = c("C9orf72", "", "SOD1", "", "FUS", "", "TARDBP",""), heights = c(0.1,0.8,0.1,1.3,0.1,0.9,0.1,1.3)) #, ipsc_mn_vcp_datasets.rrvgo.plot) # no terms for VCP
# de_go_terms_mutations_merged
# ggsave(de_go_terms_mutations_merged, filename = here(proj_path,"expression/deseq2/figS5.de_go_terms_mutations_merged.png", height = 11.75, width = 8.25)

# which genes are driving the terms
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GABA_RECEPTOR_COMPLEX)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PLASMA_MEMBRANE_REGION)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GAP_JUNCTION)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT_CONFERRING_COMPRESSION_RESISTANCE)
```



## p53 PROGENy & TP53 Dorothea

```{r}
ipsc_mn_mutations_list = list("Sporadic" = select(ipsc_mn_sporadic_datasets$res,gene_name,stat), "C9orf72" = select(ipsc_mn_c9orf72_datasets$res,gene_name,stat), "TARDBP" = select(ipsc_mn_tardbp_datasets$res,gene_name,stat), "SOD1" = select(ipsc_mn_sod1_datasets$res,gene_name,stat), "FUS" = select(ipsc_mn_fus_datasets$res,gene_name,stat)) #, "VCP" = select(ipsc_mn_vcp_datasets$res,gene_name,stat)) 
mutations <- names(ipsc_mn_mutations_list)
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{drop_na(.x)}) 
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

# ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP","VCP")))) +
#   geom_bar(stat='identity', position = "dodge2") +
#   scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
#   theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
#   geom_hline(yintercept = 0, linetype = 3) +
#   labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment") 
# ipsc_mn_mutations_list.progeny_scores.plot

ipsc_mn_mutations_list.p53.progeny_scores = ipsc_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
ipsc_mn_mutations_list.p53.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.p53.progeny_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
  stat_pvalue_manual(ipsc_mn_mutations_list.p53.progeny_scores, label = "p.signif", y.position = 13, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_mutations_list.p53.progeny_scores.plot



# Dorothea
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

ipsc_mn_mutations_list.TP53.dorothea_scores = ipsc_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
ipsc_mn_mutations_list.TP53.dorothea_scores.plot <- ggplot(ipsc_mn_mutations_list.TP53.dorothea_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") + #ylim(c(-11,11)) +
  stat_pvalue_manual(ipsc_mn_mutations_list.TP53.dorothea_scores, label = "p.signif", y.position = 11, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_mutations_list.TP53.dorothea_scores.plot
```


## Merge

```{r}
mutations_compared_fig.merged = ggarrange(
  ggarrange(ipsc_mn_tardbp_datasets.volcano, ipsc_mn_fus_datasets.volcano, ipsc_mn_c9orf72_datasets.volcano, ipsc_mn_sod1_datasets.volcano, ipsc_mn_sporadic_datasets.volcano, dge_correlation_ggheatmap, ncol = 3, nrow = 2, labels = c("a","b","c","d","e","f")),
  ggarrange(ipsc_mn_c9orf72_datasets.go_curated, ipsc_mn_fus_datasets.go_curated, ipsc_mn_tardbp_datasets.go_curated,ncol=3,labels = c("g","h","i")), # GO plots for each background
  ggarrange(ipsc_mn_mutations_list.p53.progeny_scores.plot, ipsc_mn_mutations_list.TP53.dorothea_scores.plot, ncol=2, labels = c("j","k")),
  nrow = 3, heights = c(1.5,1,0.8))
# mutations_compared_fig.merged
ggsave(mutations_compared_fig.merged, filename = here(proj_path,"expression/deseq2/mutations_compared_fig.merged.png"), height = 13, width = 10.5)

```


# Fig S12 Gene expression overlap mutations 

## Scatters

Scatter correlation of mutants

```{r}
ipsc_mn_mutations_list = list("Sporadic" = ipsc_mn_sporadic_datasets$res, "C9orf72" = ipsc_mn_c9orf72_datasets$res, "TARDBP" = ipsc_mn_tardbp_datasets$res, "SOD1" = ipsc_mn_sod1_datasets$res, "FUS" = ipsc_mn_fus_datasets$res, "panALS" = ipsc_mn_als_datasets$res)
# mutations <- names(ipsc_mn_mutations_list)
# 
# ipsc_mn_mutations_list = purrr::map2(
#       .x = ipsc_mn_mutations_list,
#       .y = names(ipsc_mn_mutations_list), ~{
#         .x$mutation <- .y
#         return(.x)
#       })

ipsc_mn_mutations_list_t_stat_plot <- 
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", highlight = "C9orf72")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", highlight = "SOD1")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", highlight = "FUS")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", highlight = "TARDBP")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", highlight = c("C9orf72","SOD1"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", highlight = c("C9orf72","FUS"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", highlight = c("C9orf72","TARDBP"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", highlight = c("SOD1","FUS"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", highlight = c("SOD1","TARDBP"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", highlight = c("TARDBP","FUS"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "panALS", highlight = c("SOD1"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "panALS")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "panALS", highlight = c("C9orf72"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "panALS", highlight = c("FUS"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "TARDBP", mutation2 = "panALS", highlight = c("TARDBP")) +
  plot_layout(ncol = 4, nrow = 4) &  
  xlim(-8,8) & ylim(-8,8)
# ipsc_mn_mutations_list_t_stat_plot
ggsave(ipsc_mn_mutations_list_t_stat_plot, filename = here(proj_path,"expression/deseq2/dge.mutation.stat.scatters.merged.png"), height = 7, width = 10)

# # log2FoldChange
# lfc_plot <- 
#   plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "TARDBP", mutation2 = "VCP", col = "log2FoldChange") &  
#   xlim(-8,8) & ylim(-8,8)  +
#   plot_layout(ncol = 3, nrow = 5)
# lfc_plot
# ggsave(lfc_plot, filename = here(proj_path,"expression/deseq2/dge.mutation.lfc.scatters.merged.png", height = 13, width = 10)

```

## Upset plot

```{r}
### Upset plot https://github.com/const-ae/ggupset
ipsc_mn_mutants_bind_datasets.res = bind_rows(mutate(ipsc_mn_sporadic_datasets$res, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$res, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$res, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$res, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$res, mutation = "FUS"))# ,mutate(ipsc_mn_vcp_datasets$res, mutation = "VCP"))
ipsc_mn_mutants_bind_datasets.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
  ggplot(aes(x = mutations)) + geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.2, hjust = 0.5, size = 3, angle = 0) +
    scale_x_upset(order_by = "degree")  + # n_intersections = 20
    scale_y_continuous(breaks = NULL, lim = c(0, 4000), name = "") +
    theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + labs(x = "", y = "") +
    theme_combmatrix(combmatrix.label.make_space = TRUE)
ipsc_mn_mutants_bind_datasets.res.upset


```

## Merge

```{r}
genetic_subgroups_scatter_upset = plot_grid(ipsc_mn_mutations_list_t_stat_plot, ipsc_mn_mutants_bind_datasets.res.upset,
                                            nrow = 2, rel_heights = c(2,1), labels = "auto")
# genetic_subgroups_compared.merged
ggsave(genetic_subgroups_scatter_upset, filename = here(proj_path,"expression/deseq2/genetic_subgroups_scatter_upset.png"), height = 11.75, width = 8.25)

```


# Fig S13 DoRoTHEA & PROGENy

<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md
Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
ipsc_mn_mutations_list = list("Sporadic" = select(ipsc_mn_sporadic_datasets$res,gene_name,stat), "C9orf72" = select(ipsc_mn_c9orf72_datasets$res,gene_name,stat), "TARDBP" = select(ipsc_mn_tardbp_datasets$res,gene_name,stat), "SOD1" = select(ipsc_mn_sod1_datasets$res,gene_name,stat), "FUS" = select(ipsc_mn_fus_datasets$res,gene_name,stat))#, "VCP" = select(ipsc_mn_vcp_datasets$res,gene_name,stat)) 
mutations <- names(ipsc_mn_mutations_list)
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{drop_na(.x)}) 
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, source = as.factor(source), source = fct_reorder(source, score))

ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP","VCP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment")# +
  # stat_pvalue_manual(ipsc_mn_mutations_list.progeny_scores, label = "p.signif", y.position = 12, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_mutations_list.progeny_scores.plot


ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=mutation, y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.position = "top") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment") +
  facet_grid(~source) +
  stat_pvalue_manual(ipsc_mn_mutations_list.progeny_scores, label = "p.signif", y.position = 12, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE)
ipsc_mn_mutations_list.progeny_scores.plot


```

Dorothea

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation) %>% 
  mutate(sig = case_when(p_value < 0.1 & abs(score) < 4 ~ "sig", p_value < 0.1 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.1 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

# Volcano of TFs score vs p_value
ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_mutations_list.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", subtitle = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
    geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "C9orf72"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "FUS"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "SOD1"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "Sporadic"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "TARDBP"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "VCP"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
    geom_text_repel(fontface = "italic", data = filter(ipsc_mn_mutations_list.dorothea_scores,source %in% c("TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  facet_wrap(~mutation, scales = "free") & ylim(0,2.8)
ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano

# ,"E2F4","EOMES","MAZ","MITF","SOX2","TFDP1","ZBTB33","ZNF263"

# commonly changed TFs
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05) %>% group_by(mutation) %>% count(source)
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 E2F4       5
#  2 MAZ        5
#  3 NR2C2      5
#  4 TEAD4      5
#  5 ZBTB33     5
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score > 0) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 NR2C2      4
#  2 ZBTB33     4
#  3 ZNF274     4
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score < 0) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 GATA3      4
#  2 MAZ        4
#  3 SOX2       4
#  4 TAL1       4

ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "E2F4")
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "MAZ")
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "NR2C2")
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "TEAD4")

```

Merge

```{r}
ipsc_mn_mutations_list.dorothea_progeny_merged = ggarrange(ipsc_mn_mutations_list.progeny_scores.plot, ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano, nrow = 2, labels = c("a","b"), heights = c(1,1.7))
# ipsc_mn_mutations_list.dorothea_progeny_merged
ggsave(ipsc_mn_mutations_list.dorothea_progeny_merged, filename = here(proj_path, "expression/deseq2/ipsc_mn_mutations_list.dorothea_progeny_merged.png"), height = 11.75, width = 10)

```



# Table S7 DEGs overlap between genetic backgrounds

```{r}
ipsc_mn_mutants_join_datasets.res = 
  mutate(select(ipsc_mn_sporadic_datasets$res, gene_name, gene_id, padj.sporadic = padj, stat.sporadic = stat, lfc.sporadic = log2FoldChange), direction.sporadic = case_when(padj.sporadic < 0.05 & lfc.sporadic > 0 ~ "up", padj.sporadic < 0.05 & lfc.sporadic < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(ipsc_mn_c9orf72_datasets$res, gene_name, gene_id, padj.c9orf72 = padj, stat.c9orf72 = stat, lfc.c9orf72 = log2FoldChange), direction.c9orf72 = case_when(padj.c9orf72 < 0.05 & lfc.c9orf72 > 0 ~ "up", padj.c9orf72 < 0.05 & lfc.c9orf72 < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_fus_datasets$res, gene_name, gene_id, padj.fus = padj, stat.fus = stat, lfc.fus = log2FoldChange), direction.fus = case_when(padj.fus < 0.05 & lfc.fus > 0 ~ "up", padj.fus < 0.05 & lfc.fus < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_tardbp_datasets$res, gene_name, gene_id,  padj.tardbp = padj, stat.tardbp = stat, lfc.tardbp = log2FoldChange), direction.tardbp = case_when(padj.tardbp < 0.05 & lfc.tardbp > 0 ~ "up", padj.tardbp < 0.05 & lfc.tardbp < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_sod1_datasets$res, gene_name, gene_id, padj.sod1 = padj, stat.sod1 = stat, lfc.sod1 = log2FoldChange), direction.sod1 = case_when(padj.sod1 < 0.05 & lfc.sod1 > 0 ~ "up", padj.sod1 < 0.05 & lfc.sod1 < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  # full_join(mutate(select(ipsc_mn_vcp_datasets$res, gene_name, gene_id, padj.vcp = padj, stat.vcp = stat, lfc.vcp = log2FoldChange), direction.vcp = case_when(padj.vcp < 0.05 & lfc.vcp > 0 ~ "up", padj.vcp < 0.05 & lfc.vcp < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  # arrange() %>%
  select(gene_name, gene_id, everything()) %>%
  filter(padj.sporadic < 0.05 | padj.c9orf72 < 0.05 | padj.fus < 0.05 | padj.tardbp < 0.05 | padj.sod1 < 0.05)
ipsc_mn_mutants_join_datasets.res
write_csv(ipsc_mn_mutants_join_datasets.res, here(proj_path,"expression/deseq2/TableS6.overlap_als_individual_datasets.csv"))

```


# Fig 4 Postmortem 

Import NYGC counts from Humphrey et al Zenodo https://zenodo.org/record/6385747#.Yod3Y5PMJqs

## Volcano

```{r}
#### Spinal cord only ###

# metadata
nygc_cervical_cord_metadata = read_tsv(here(shared_path,"nygc-als-consortium/counts/spinal_cord_jack_humphrey/Cervical_Spinal_Cord/Cervical_Spinal_Cord_metadata.tsv")) # 174
nygc_thoracic_cord_metadata = read_tsv(here(shared_path,"nygc-als-consortium/counts/spinal_cord_jack_humphrey/Thoracic_Spinal_Cord/Thoracic_Spinal_Cord_metadata.tsv")) # 52
nygc_lumbar_cord_metadata = read_tsv(here(shared_path,"nygc-als-consortium/counts/spinal_cord_jack_humphrey/Lumbar_Spinal_Cord/Lumbar_Spinal_Cord_metadata.tsv")) # 154
nygc_spinal_cord_metadata.bind = bind_rows(nygc_cervical_cord_metadata, nygc_thoracic_cord_metadata, nygc_lumbar_cord_metadata) %>% # 380
  mutate(condition = case_when(disease == "Control" ~ "ctrl", grepl("ALS",disease) ~ "als"), condition = factor(condition, levels = c("ctrl","als")), library_prep = as.factor(library_prep),
         gender = tolower(sex), mutation = tolower(mutations), mutation = case_when(condition == "als" & mutation == "none" ~ "sporadic", TRUE ~ mutation))
nygc_spinal_cord_metadata.bind %>% count(condition)
nygc_spinal_cord_metadata = nygc_spinal_cord_metadata.bind %>% mutate(tissue = factor(tissue, levels = c("Cervical_Spinal_Cord", "Thoracic_Spinal_Cord", "Lumbar_Spinal_Cord")), sample = rna_id) %>% arrange(tissue) %>% distinct(dna_id, .keep_all = TRUE) %>% column_to_rownames(var = "rna_id") # 203 / 380
nygc_spinal_cord_metadata.sporadic = nygc_spinal_cord_metadata %>% filter(mutation %in% c("sporadic","none"))
nygc_spinal_cord_metadata.c9orf72 = nygc_spinal_cord_metadata %>% filter(mutation %in% c("c9orf72","none"))
nygc_spinal_cord_metadata.sod1 = nygc_spinal_cord_metadata %>% filter(mutation %in% c("sod1","none"))
nygc_spinal_cord_metadata.fus = nygc_spinal_cord_metadata %>% filter(mutation %in% c("fus","none"))

nygc_spinal_cord_metadata %>% count(condition)
nygc_spinal_cord_metadata.sporadic %>% count(condition)
nygc_spinal_cord_metadata.c9orf72 %>% count(condition)
nygc_spinal_cord_metadata.sod1 %>% count(condition)
nygc_spinal_cord_metadata.fus %>% count(condition)

nygc_postmortem_spinal_cord.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.tissue.rds"))
nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) # 14,529
nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #6417 up, 8112 down

## Volcano
nygc_spinal_cord_metadata
nygc_postmortem_spinal_cord.als_vs_ctrl.als.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(gene_name %in% ALS_genes) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.p53.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.ipsn.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj<0.05,gene_name %in% pull(filter(ipsc_mn_als_datasets$res,padj<0.05),gene_name)) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.volcano <- volcano_plot(nygc_postmortem_spinal_cord.als_vs_ctrl$res, numerator = "ALS", denomenator = "CTRL",
               title = "Postmortem spinal cord", subtitle = "80 Controls vs 300 ALS", ymax = 60, xmax = 3, dpi = 300,
               gene_labels = c(nygc_postmortem_spinal_cord.als_vs_ctrl.als.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.ipsn.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.p53.labels,"CFAP410"))
nygc_postmortem_spinal_cord.als_vs_ctrl.volcano

```

## GO

```{r}
nygc_postmortem_spinal_cord.als_vs_ctrl.gost <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles <- nygc_postmortem_spinal_cord.als_vs_ctrl.gost$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_down <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Down ")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_up <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "dendrite",
"transcription by RNA polymerase II",
"regulation of RNA metabolic process",
"cytoskeleton organization",
"axon",
"neuron development",
"synapse",
"neuron projection",
# "generation of neurons",
"neurogenesis",
"nervous system development")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_down <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"cellular response to DNA damage stimulus",
"programmed cell death",
# "cell death",
"endoplasmic reticulum",
# "mitochondrial matrix",
"lysosome",
"protein localization",
"mitochondrial envelope",
# "protein modification process",
"translation",
"response to stress",
# "peptide metabolic process",
"catabolic process",
"mitochondrion",
# "extracellular vesicle",
"protein metabolic process",
"programmed cell death")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_up <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined <- bind_rows(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_down, nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined = nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined %>% mutate(term_name = gsub("regulation of ", "", term_name), term_name = gsub("and","&",term_name), term_name = gsub("cellular ","",term_name))
nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated <- curated_profiles(gprofiles = nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined)# + theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated

```


## PROGENY & Dorothea

```{r}
# nygc_postmortem_spinal_cord.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.tissue.rds"))

# PROGENy & DOROTHEA
net_progeny <- get_progeny(organism = 'human', top = 100)
nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity <- ggplot(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment ALS vs CTRL") +
    stat_pvalue_manual(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 6, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts <- run_wmean(mat=nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(sig = case_when(p_value < 0.06 & abs(score) < 4 ~ "sig", p_value < 0.06 & abs(score) >= 4 ~ "sig_strong", p_value > 0.05 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.up = nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.down = nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano = ggplot(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts, source %in% c(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.up, nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted')
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano

nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts %>% filter(source == "TP53")

```

## Scatter postmortem against iPSN

```{r}
# correlate to meta
postmortem_ipsn.als_vs_ctrl = select(nygc_postmortem_spinal_cord.als_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_als_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.als_vs_ctrl.overlapping = postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name) # 50
postmortem_ipsn_de_list = list("Postmortem spinal cord: ALS vs CTRL" = nygc_postmortem_spinal_cord.als_vs_ctrl$res, "iPSN: ALS vs CTRL" = ipsc_mn_als_datasets$res)
postmortem_ipsn.als_vs_ctrl.scatter = plot_de_correlation(model_list1 = postmortem_ipsn_de_list, model_list2 = postmortem_ipsn_de_list, mutation1 = "iPSN: ALS vs CTRL", mutation2 = "Postmortem spinal cord: ALS vs CTRL", highlight = c(postmortem_ipsn.als_vs_ctrl.overlapping, "CFAP410"))
postmortem_ipsn.als_vs_ctrl.scatter

postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat))
postmortem_ipsn.als_vs_ctrl.overlapping # 
postmortem_ipsn.als_vs_ctrl.overlapping[postmortem_ipsn.als_vs_ctrl.overlapping %in% ALS_genes] # 0
postmortem_ipsn.als_vs_ctrl.overlapping[postmortem_ipsn.als_vs_ctrl.overlapping %in% RNA_BINDING_PROTEINS] # "RNASEL" 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% RNA_BINDING_PROTEINS) 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% gprofiler_database$`DNA Damage Response`) 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% RNA_BINDING_PROTEINS  ) %>% arrange(-nygc.stat + ipsn.stat)
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% group_split(nygc.stat > 0)


nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.up <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.down <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
ipsc_mn_als_datasets.res.dge.up <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.res.dge.down <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
go.obj <- newGeneOverlap(c(ipsc_mn_als_datasets.res.dge.up,ipsc_mn_als_datasets.res.dge.down),
                         c(nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.up,nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.down),
                         genome.size=genome.size)
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=37
# listB size=14501
# Intersection size=21
# Overlapping p-value=1.9e-05
# Jaccard Index=0.0

# Fisher exact overlap test
go.obj <- newGeneOverlap(ipsc_mn_als_datasets.res.dge.up,
                         nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.up,
                         genome.size=genome.size)
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=17
# listB size=6411
# Intersection size=7
# Overlapping p-value=1.1e-03
# Jaccard Index=0.0
go.obj <- newGeneOverlap(ipsc_mn_als_datasets.res.dge.down,
                         nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.down,
                         genome.size=genome.size)
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=20
# listB size=8092
# Intersection size=8
# Overlapping p-value=2.8e-03
# Jaccard Index=0.0
```


## Compare mutations postmortem

```{r}
nygc_postmortem_spinal_cord.c9orf72_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.c9orf72_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.fus_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.fus_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sod1_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sod1_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sporadic_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sporadic_vs_ctrl.rds"))

nygc_postmortem_spinal_cord_mn_mutations_list = list("Sporadic" = select(nygc_postmortem_spinal_cord.sporadic_vs_ctrl$res,gene_name,stat), "C9orf72" = select(nygc_postmortem_spinal_cord.c9orf72_vs_ctrl$res,gene_name,stat), "SOD1" = select(nygc_postmortem_spinal_cord.sod1_vs_ctrl$res,gene_name,stat), "FUS" = select(nygc_postmortem_spinal_cord.fus_vs_ctrl$res,gene_name,stat)) #"TARDBP" = select(nygc_postmortem_spinal_cord.tardbp_vs_ctrl$res,gene_name,stat), 
mutations <- names(nygc_postmortem_spinal_cord_mn_mutations_list)
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{drop_na(.x)}) 
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores = map_df(nygc_postmortem_spinal_cord_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment")
nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores.plot

nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores = nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
  theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
  stat_pvalue_manual(nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores, label = "p.signif", y.position = 6, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) 
nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot



# Dorothea
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord_mn_mutations_list.dorothea_scores = map_df(nygc_postmortem_spinal_cord_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores = nygc_postmortem_spinal_cord_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") + #ylim(c(-6,6)) +
  stat_pvalue_manual(nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores, label = "p.signif", y.position = 5, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) 
nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot

```

## Merge

```{r}
nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged = ggarrange(
  ggarrange(nygc_postmortem_spinal_cord.als_vs_ctrl.volcano, nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated, ncol=2, labels = c("a","b"), widths = c(1,0.8)),
  ggarrange(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity, nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano, ncol=2, labels = c("c","d")),
  ggarrange(postmortem_ipsn.als_vs_ctrl.scatter, nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot, nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot, ncol = 3, labels = c("e","f","g"), widths = c(2,1,1)),
  nrow = 3)
# nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged
ggsave(nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged, filename = here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged.png"), height = 12, width = 9)


```


# Table S8 postmortem & ipsn overlap DGE

```{r}
postmortem_ipsc_overlap.res = 
  mutate(select(nygc_postmortem_spinal_cord.als_vs_ctrl$res, gene_name, gene_id, padj.postmortem = padj, stat.postmortem = stat, lfc.postmortem = log2FoldChange), direction.postmortem = case_when(padj.postmortem < 0.05 & lfc.postmortem > 0 ~ "up", padj.postmortem < 0.05 & lfc.postmortem < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(ipsc_mn_als_datasets$res, gene_name, gene_id, padj.ipsn = padj, stat.ipsn = stat, lfc.ipsn = log2FoldChange), direction.ipsn = case_when(padj.ipsn < 0.05 & lfc.ipsn > 0 ~ "up", padj.ipsn < 0.05 & lfc.ipsn < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  select(gene_name, gene_id, everything()) %>% 
  filter(padj.ipsn < 0.05, padj.postmortem < 0.05, ((stat.ipsn > 0 & stat.postmortem > 0) | (stat.ipsn < 0 & stat.postmortem < 0)))
postmortem_ipsc_overlap.res
write_csv(postmortem_ipsc_overlap.res, here(proj_path,"expression/deseq2/Table.postmortem_ipsc_overlap.csv"))

```


# Fig S14 Scatter postmortem & iPSN subgroups

```{r}
postmortem_ipsc_mn_mutations_list = list("Sporadic iPSN" = ipsc_mn_sporadic_datasets$res, "C9orf72 iPSN" = ipsc_mn_c9orf72_datasets$res, "SOD1 iPSN" = ipsc_mn_sod1_datasets$res, "FUS iPSN" = ipsc_mn_fus_datasets$res, "Sporadic postmortem" = nygc_postmortem_spinal_cord.sporadic_vs_ctrl$res, "C9orf72 postmortem" = nygc_postmortem_spinal_cord.c9orf72_vs_ctrl$res, "SOD1 postmortem" = nygc_postmortem_spinal_cord.sod1_vs_ctrl$res, "FUS postmortem" = nygc_postmortem_spinal_cord.fus_vs_ctrl$res)

postmortem_ipsn.sporadic.join = select(nygc_postmortem_spinal_cord.sporadic_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_sporadic_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.sporadic.overlapping = postmortem_ipsn.sporadic.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name)
postmortem_ipsn.c9orf72.join = select(nygc_postmortem_spinal_cord.c9orf72_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_c9orf72_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.c9orf72.overlapping = postmortem_ipsn.c9orf72.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name)
postmortem_ipsn.sod1.join = select(nygc_postmortem_spinal_cord.sod1_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_sod1_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.sod1.overlapping = postmortem_ipsn.sod1.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name)
postmortem_ipsn.fus.join = select(nygc_postmortem_spinal_cord.fus_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_fus_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.fus.overlapping = postmortem_ipsn.fus.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name)

postmortem_ipsc_mn_mutations_list_t_stat_plot <- plot_grid(
plot_de_correlation(model_list1 = postmortem_ipsc_mn_mutations_list, model_list2 = postmortem_ipsc_mn_mutations_list, mutation1 = "Sporadic iPSN", mutation2 = "Sporadic postmortem", highlight = postmortem_ipsn.sporadic.overlapping),
plot_de_correlation(model_list1 = postmortem_ipsc_mn_mutations_list, model_list2 = postmortem_ipsc_mn_mutations_list, mutation1 = "C9orf72 iPSN", mutation2 = "C9orf72 postmortem", highlight = c(postmortem_ipsn.c9orf72.overlapping, "C9orf72")),
plot_de_correlation(model_list1 = postmortem_ipsc_mn_mutations_list, model_list2 = postmortem_ipsc_mn_mutations_list, mutation1 = "SOD1 iPSN", mutation2 = "SOD1 postmortem", highlight = c(postmortem_ipsn.sod1.overlapping, "SOD1")),
plot_de_correlation(model_list1 = postmortem_ipsc_mn_mutations_list, model_list2 = postmortem_ipsc_mn_mutations_list, mutation1 = "FUS iPSN", mutation2 = "FUS postmortem", highlight = c(postmortem_ipsn.fus.overlapping, "FUS")),
  ncol = 2, nrow = 2, labels = "auto")# & xlim(-8,8) & ylim(-8,8)
postmortem_ipsc_mn_mutations_list_t_stat_plot
ggsave(postmortem_ipsc_mn_mutations_list_t_stat_plot, filename = here(proj_path,"expression/deseq2/postmortem_ipsc_mn_mutations_list_t_stat_plot.png"), height = 8, width = 8)

postmortem_ipsn.sporadic.overlapping[postmortem_ipsn.sporadic.overlapping %in% p53.signalling.pathway]
postmortem_ipsn.c9orf72.overlapping[postmortem_ipsn.c9orf72.overlapping %in% p53.signalling.pathway]
postmortem_ipsn.sod1.overlapping[postmortem_ipsn.sod1.overlapping %in% p53.signalling.pathway]
postmortem_ipsn.fus.overlapping[postmortem_ipsn.fus.overlapping %in% p53.signalling.pathway]
postmortem_ipsn.sporadic.overlapping[postmortem_ipsn.sporadic.overlapping %in% RNA_BINDING_PROTEINS]
postmortem_ipsn.c9orf72.overlapping[postmortem_ipsn.c9orf72.overlapping %in% RNA_BINDING_PROTEINS]
postmortem_ipsn.sod1.overlapping[postmortem_ipsn.sod1.overlapping %in% RNA_BINDING_PROTEINS]
postmortem_ipsn.fus.overlapping[postmortem_ipsn.fus.overlapping %in% RNA_BINDING_PROTEINS]
postmortem_ipsn.sporadic.overlapping[postmortem_ipsn.sporadic.overlapping %in% ALS_genes]
postmortem_ipsn.c9orf72.overlapping[postmortem_ipsn.c9orf72.overlapping %in% ALS_genes]
postmortem_ipsn.sod1.overlapping[postmortem_ipsn.sod1.overlapping %in% ALS_genes]
postmortem_ipsn.fus.overlapping[postmortem_ipsn.fus.overlapping %in% ALS_genes]

postmortem_ipsn.c9orf72.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_id) %>% gost()
# axonemal dynein complex assembly

postmortem_ipsn.c9orf72.overlapping[postmortem_ipsn.c9orf72.overlapping%in%gprofiler_database$`axonemal dynein complex assembly`]

```




# Fig 5 Splicing pan ALS

## MAJIQ 

```{r}
majiq_schematic = magick::image_read(here(proj_path,"splicing/majiq/figures/MAJIQ splicing schematic.png"))
majiq_schematic.gg = ggdraw() + draw_image(majiq_schematic)
majiq_schematic.gg

als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
pan_als.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/pan_als"), skip_rows = 9)

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 264
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 151
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 24
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # 0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) #SIRT2, CASP9, UNC13B
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% sporadic_ALS_genes) #INPP5F
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_GWAS) #0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_EWAS) # SIRT2
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_modifiers) # 0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # BID, HTT, GADD45B
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA repair`)) # BID, HTT, GADD45B
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.signalling.pathway)
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`DNA repair`) # RIF1, SMC6, HUS1, SETMAR, HUWE1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, -log10(tnom) > 4)
## Volcano
ipsc_mn_als_datasets.polyA.metadata %>% count(dataset)
ipsc_mn_als_datasets.polyA.metadata %>% count(condition) #51 ALS, 46 ctrl
majiq.als_vs_ctrl.RBP.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% group_by(gene_name) %>% top_n(5, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.ALS.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes))  %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.DDR.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`DNA repair`) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, -log10(tnom)>3, -log10(tnom)<4) 

majiq.als_vs_ctrl.volcano.junction_labels = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("ADA","MMAB","UNC13B","SIRT2","CASP9","RIF1", "SMC6", "HUS1", "SETMAR", "HUWE1","CHD9","TCF4","TTBK2","METTL22")) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.volcano <- majiq_het_volcano_plot(als_ctrl.voila, "Alternative splicing: pan ALS", numerator = "ALS", denomenator = "CTRL", subtitle = "46 CTRL vs 51 ALS", ymax = 5, xmax = 0.32, dpi = 300, junction_labels =majiq.als_vs_ctrl.volcano.junction_labels)
majiq.als_vs_ctrl.volcano

# GO of splicing events
als_ctrl.voila.gost = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
als_ctrl.voila.gost_filt <- als_ctrl.voila.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC")) %>% arrange( -log10(p_value) )
table(als_ctrl.voila.gost_filt$query) # 
paste('"',unique(als_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
"site of double-strand break",
"NACHT domain binding",
"protein binding")
als_ctrl.voila.gost_filt.terms <- als_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
# als_ctrl.voila.gost_filt.terms$term_name = gsub(" molecule binding| organization| organizing", "", als_ctrl.voila.gost_filt.terms$term_name)
als_ctrl.voila.gost_filt.terms.curated <- curated_profiles(als_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + theme(strip.text.y.right = element_blank())# + theme(legend.position = "none", axis.text = element_text(size = 10))
# als_ctrl.voila.gost_filt.terms.curated

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`protein binding`) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`NACHT domain binding`) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`site of double-strand break`) %>% arrange(-abs(deltapsi))

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_PROTEIN_BINDING) %>% arrange(-abs(deltapsi)) # ARHGEF7, NRP1, ADD1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_RNA_BINDING) %>% arrange(-abs(deltapsi)) %>% distinct(gene_name) # SRRM1, SORBS2, PAN3, MARK2, HUWE1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% arrange(-abs(deltapsi)) # FZNF326, RTF1, AUH, NOVA1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_RNA_SPLICING) %>% arrange(-abs(deltapsi))

# Compare splicing with gene expression
als_splicing_events_genes = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
als_degs = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% pull(gene_name)
als_degs[als_degs %in% als_splicing_events_genes] #THB15B

# # Cryptic Exons == denovo & cassette_exon
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 28 / 264
28/264# 10.6%
pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer) #6 cryptic exons
pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi))  # FMNL2, COL18A1, SNX, SDHAP1, LIMS1, SNX2
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 59 /264
59/264

# Modulizer pie chart of differential events
modulizer_simplfied.sig.summary = pan_als.voila_modulizer %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer_simplified) %>% mutate(modulizer_simplified = gsub("_"," ",modulizer_simplified), modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>% arrange(desc(modulizer_simplified)) %>% 
  mutate(pct = n/sum(n) *100, pct.label = case_when(pct > 8 ~ paste(as.character(n),"\n", as.character(round(pct,1)),"%"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) #%>% print
modulizer_simplfied.sig.summary %>% print
pan_als.voila_modulizer.sig.pie = ggplot(modulizer_simplfied.sig.summary, aes(x = "", y = pct, fill = fct_reorder(modulizer_simplified, n, .desc = TRUE))) + geom_bar(width = 1, stat = "identity", color="white") + coord_polar("y", start=0) + 
  geom_text(aes(x = 1.2, y = ypos, label = pct.label), color = "black", size=2.5) + scale_fill_brewer(palette="Set1") + 
  theme_oz() + theme_void() + theme(legend.title=element_blank(), legend.position = "right", legend.text=element_text(size=7), legend.spacing.y = unit(1, 'mm'), legend.spacing.x = unit(1, 'mm'), legend.key.height = unit(0.4, 'cm'), legend.key.width = unit(0.3, 'cm')) + guides(fill=guide_legend(ncol=1))
pan_als.voila_modulizer.sig.pie

# Intron retention
pan_als.voila_modulizer.ir = pan_als.voila_modulizer %>% filter(modulizer_simplified == "intron_retention")
pan_als.voila_modulizer.ir %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # 60
pan_als.voila_modulizer.ir %>% filter(gene_name %in% ALS_genes, abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # SIRT2

# Voila View violin plots:
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% arrange(-abs(deltapsi)) %>% print(n=Inf)
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "TTBK2") %>% glimpse # lsv_junction_id   "ENSG00000128881:t:42878549-42878684:j:42878684-42919759"
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "PDE5A") %>% glimpse # lsv_junction_id       <chr> "ENSG00000138735:s:119500823-119501253:j:119498738-119501170", "ENSG00000138735:s:119500823-119501253:j:119498739-119500822"
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "ADA") %>% glimpse # lsv_junction_id  "ENSG00000196839:t:44621018-44621147:j:44621147-44622829", "ENSG00000196839:t:44621018-44621147:j:44621147-44622588"

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% arrange(-abs(deltapsi)) %>% print(n=Inf) # UNC13B, CASP9, SIRT2
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "UNC13B") %>% glimpse # lsv_junction_id       <chr> "ENSG00000198722:s:35384246-35384314:j:35384314-35385724", "ENSG00000198722:s:35384246-35384314:j:35384315-35385723"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "CASP9") %>% glimpse # lsv_junction_id   "ENSG00000132906:t:15495273-15495452:j:15495452-15518110", "ENSG00000132906:t:15495273-15495452:j:15495452-15504611"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "SIRT2") %>% glimpse # lsv_junction_id  "ENSG00000068903:s:38898379-38898425:j:38894874-38898378"

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`site of double-strand break`) %>% arrange(-abs(deltapsi)) # SETMAR, RIF1, HUS1, SMC6
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`DNA repair`)  %>% arrange(-abs(deltapsi)) # SETMAR, HUWE1, RIF1, SMC6
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "SETMAR") %>% glimpse # lsv_junction_id   ENSG00000170364:t:4316212-4317567:j:4303526-4316212", "ENSG00000170364:t:4316212-4317567:j:4313761-4316212"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "HUWE1") %>% glimpse # lsv_junction_id   "ENSG00000086758:s:53627410-53627515:j:53626034-53627410", "ENSG00000086758:s:53627410-53627515:j:53625258-53627410"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "RIF1") %>% glimpse # lsv_junction_id    "ENSG00000080345:t:151468641-151468756:j:151468146-151468641", "ENSG00000080345:t:151468641-151468756:j:151468551-151468641"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "SMC6") %>% glimpse # lsv_junction_id  "ENSG00000163029:s:17753626-17753830:j:17753064-17753626", "ENSG00000163029:s:17753626-17753830:j:17745951-17753626"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "METTL22") %>% glimpse # lsv_junction_id  "ENSG00000067365:s:8644437-8645044:j:8644725-8645908", "ENSG00000067365:s:8644437-8645044:j:8644725-8646108"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "MMAB") %>% glimpse #$ lsv_junction_id       <chr> "ENSG00000139428:s:109561420-109561517:j:109561329-109561420",


### import violin from voila view
# edit colour and labels in illustrator > export as png
polyA_majiq_violins_merged.png = magick::image_read(here(proj_path,"splicing/majiq/figures/polyA_majiq_violins_merged.png"))
polyA_majiq_violins_merged.gg = ggdraw() + draw_image(polyA_majiq_violins_merged.png)
polyA_majiq_violins_merged.gg
# BID_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/BID_majiq_violin.png"))
# BID_majiq_violins.gg = ggdraw() + draw_image(BID_majiq_violins.png)
# BID_majiq_violins.gg
# HTT_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/HTT_majiq_violin.png"))
# HTT_majiq_violins.gg = ggdraw() + draw_image(HTT_majiq_violins.png)
# HTT_majiq_violins.gg
# GADD45B_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/GADD45B_majiq_violin.png"))
# GADD45B_majiq_violins.gg = ggdraw() + draw_image(GADD45B_majiq_violins.png)
# GADD45B_majiq_violins.gg
# majiq_violins.gg = GADD45B_majiq_violins.gg + CAMTA1_majiq_violins.gg  + BID_majiq_violins.gg + HTT_majiq_violins.gg
# majiq_violins.gg


# TDP43 knockdown overlap
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
tdp43kd.als_vs_ctrl.brown.voila.sig = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9)# %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.sig 

tdp43kd.als_vs_ctrl.brown.voila.labels = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)


```


## IRFinder 


```{r}
ipsc_mn_als_datasets.polyA = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.rds"))

# proportion of samples that are polyA and total in ALS & control
filter(ipsc_mn_als_datasets.metadata, library_type == "poly(A)") %>% count(condition, library_type)
# condition library_type     n
#   <fct>     <chr>        <int>
# 1 ctrl      poly(A)         28
# 2 ctrl      Total           73
# 3 als       poly(A)         35
# 4 als       Total          283
28/(28+73) #27.2% CTRL
35/(35+283) #11.0% ALS

ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) # 24
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% distinct(gene_name)  # 12
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% group_split(log2FoldChange < 0) # 18 up, 6 down
18/24 # 75%
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% arrange(-stat) %>% filter(gene_name %in% ALS_genes)#
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% arrange(-stat) %>% filter(gene_name %in% ALS_genes)#

ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% filter(gene_name %in% go.pathways.msigdb$GO_RNA_SPLICING) %>% arrange(pvalue) #0
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% filter(gene_name %in% p53.signalling.pathway) %>% arrange(-stat) # 0
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% gprofiler_database$`DNA Damage Response`) %>% arrange(-stat) # 0
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% gprofiler_database$`DNA repair`) %>% arrange(-stat) # 0
ipsc_mn_als_datasets.polyA$irfinder %>% top_n(10, wt = stat)

# # Violin
# ipsc_mn_als_datasets.polyA.irfinder.filt = ipsc_mn_als_datasets.polyA$irfinder %>% filter(log2FoldChange != 0) %>% mutate(condition = "Intron retention\nALS vs CTRL")
# ipsc_mn_als_datasets.polyA.ir.violin <- violin_plot(ipsc_mn_als_datasets.polyA.irfinder.filt, color_by = "condition", cols = "firebrick2", one_sample = TRUE, stats_test = "t_test", arrow_label_x = 0.87, numerator = "ALS", denomenator = "CTRL", ylims = c(-1.5,1.5), dpi = 300)
# ipsc_mn_als_datasets.polyA.ir.violin
# [1] "Using one sample t_test"
# group                           .y.   group1 group2          n statistic     df        p p.signif
#   <chr>                           <chr> <chr>  <chr>       <int>     <dbl>  <dbl>    <dbl> <chr>   
# 1 "Intron retention\nALS vs CTRL" lfc   1      null model 140715     -18.4 140714 7.69e-76 ****    
# # A tibble: 1  7
#   .y.   group1 group2     effsize group                                n magnitude 
# * <chr> <chr>  <chr>        <dbl> <chr>                            <int> <ord>     
# 1 lfc   1      null model -0.0492 "Intron retention\nALS vs CTRL" 140715 negligible

# # Density
# ipsc_mn_als_datasets.polyA.ir.density <- density_plot(ipsc_mn_als_datasets.polyA.irfinder.filt, color_by = "condition", cols = "forestgreen",  xlab_prefix = "Intron Retention", numerator = "ALS", denomenator = "CTRL", xmax = 1.5, dpi = 300)
# ipsc_mn_als_datasets.polyA.ir.density

## Volcano
ipsc_mn_als_datasets.polyA$irfinder %>% filter(abs(stat) > 5, padj < 0.05) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA$irfinder %>% filter(log2FoldChange > 1, padj < 0.05, -log10(pvalue) >4, -log10(pvalue) <6) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA$irfinder %>% filter(log2FoldChange > 1, padj < 0.05, -log10(pvalue) >5.5, -log10(pvalue) <6.5) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% pull(gene_name)

"SNED1","MPP2","HBB","NKAPD1","CACNA2D2","FAM118A","HOXC9","BRAT1"    "DHX30"    "SUPT5H"  
ipsc_mn_als_datasets.polyA.ir.volcano <- volcano_plot(ipsc_mn_als_datasets.polyA$irfinder, numerator = "ALS", denomenator = "CTRL",
               title = "Intron retention: pan ALS", subtitle = "polyA: 42 Controls vs 45 ALS", ymax = 7, xmax = 3, dpi = 300, gene_labels = c("SPHKAP", "SNED1","DBX1", "GSTM1","UGT2A1", "CYP4F29P"), distinct_labels = TRUE, arrow_labels = FALSE) +
        geom_segment(aes(x = 3*0.25, xend = 3*0.95, y= 7*0.60, yend= 7 * 0.60), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -3*0.25, xend = -3*0.95, y= 7*0.60, yend= 7*0.60),arrow=arrow(length=unit(0.3,"cm"))) +
        annotate("text", x = 3*0.6, y = 7*0.65, label = paste0("Up in ALS"), size = 3) + annotate("text", x = -3*0.6, y = 7*0.65, label = paste0("Up in CTRL"), size = 3)
ipsc_mn_als_datasets.polyA.ir.volcano


## GO
ipsc_mn_als_datasets.polyA.gost <- ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% arrange(-abs(stat)) %>% distinct(gene_id, .keep_all = TRUE) %>% group_split(log2FoldChange > 0) %>% map("gene_id") %>% gost()
ipsc_mn_als_datasets.polyA.gost_filt <- ipsc_mn_als_datasets.polyA.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC")) %>% arrange( -log10(p_value) ) %>%
  mutate(query = case_when(query == "query_1" ~ "ALS IR down", query == "query_2" ~ "ALS IR up"), query = factor(query, levels = c("ALS IR down", "ALS IR up")))
table(ipsc_mn_als_datasets.polyA.gost_filt$query)
# ALS IR down   ALS IR up 
#           2           4 
ipsc_mn_als_datasets.polyA.gost_ir.down <- ipsc_mn_als_datasets.polyA.gost_filt %>% filter(query == "ALS IR down")
ipsc_mn_als_datasets.polyA.gost_ir.up <- ipsc_mn_als_datasets.polyA.gost_filt %>% filter(query == "ALS IR up")
paste('"',unique(ipsc_mn_als_datasets.polyA.gost_ir.down$term_name),'",', sep = "", collapse = '\n') %>% cat()
paste('"',unique(ipsc_mn_als_datasets.polyA.gost_ir.up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list = c(
"Drug metabolism - other enzymes",
"Metabolism of xenobiotics by cytochrome P450",
"Drug metabolism - cytochrome P450",
"Chemical carcinogenesis - DNA adducts")
ipsc_mn_als_datasets.polyA.gost_filt_ir_up <- ipsc_mn_als_datasets.polyA.gost_filt %>% filter(query == "ALS IR up")  %>% filter(term_name %in% up_list)
ipsc_mn_als_datasets.polyA.gost_filt_ir = ipsc_mn_als_datasets.polyA.gost_filt_ir_up %>% mutate(query = fct_relevel(query,"ALS IR up"))
ipsc_mn_als_datasets.polyA.gost_filt_ir$term_name = gsub("Drug metabolism - cytochrome P450", "Drug metabolism\ncytochrome P450", ipsc_mn_als_datasets.polyA.gost_filt_ir$term_name) %>% gsub("Drug metabolism - other enzymes", "Drug metabolism\nother enzymes", .) %>% gsub("Chemical carcinogenesis - DNA adducts", "Carcinogenesis\nDNA adduct", .) %>% gsub("Metabolism of xenobiotics by cytochrome P450","Metabolism by\ncytochrome P450", .)
ipsc_mn_als_datasets.polyA.ir.go_curated <- curated_profiles(ipsc_mn_als_datasets.polyA.gost_filt_ir, gprofiles =, cols = c("firebrick2", "dodgerblue2")) + theme(strip.text.y.right = element_text(angle = 90)) 
ipsc_mn_als_datasets.polyA.ir.go_curated

# # which genes are driving the GO terms
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% c( kegg.pathways.msigdb$KEGG_GLUTATHIONE_METABOLISM)) %>% distinct(gene_name, .keep_all = TRUE)# GSTM1
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% c( kegg.pathways.msigdb$KEGG_DRUG_METABOLISM_CYTOCHROME_P450)) %>% distinct(gene_name, .keep_all = TRUE)# GSTM1, UGT2A1
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% c( kegg.pathways.msigdb$KEGG_DRUG_METABOLISM_OTHER_ENZYMES)) %>% distinct(gene_name, .keep_all = TRUE)# GSTM1
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% c(go.pathways.msigdb$GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT)) %>% distinct(gene_name, .keep_all = TRUE)# COL6A3
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_RNA_SPLICING) %>% distinct(gene_name, .keep_all = TRUE)# 0
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`protein binding`) %>% arrange(-abs(stat))
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`mitochondrion`) %>% arrange(-abs(stat)) #SPHKAP
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`RNA metabolic process`) %>% arrange(-abs(stat)) # DBX1
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`cellular response to DNA damage stimulus`) %>% arrange(-abs(stat)) #0
# ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05, stat < 0, gene_name %in% gprofiler_database$`synapse`) %>% arrange(-abs(stat))

## GSEA: p53 signalling pathway
ipsc_mn_als_datasets.polyA.intronList <- ipsc_mn_als_datasets.polyA$irfinder %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.polyA.intronList) <- ipsc_mn_als_datasets.polyA$irfinder %>% drop_na(stat) %>% pull(Intron.GeneName.GeneID.Coords) %>% as.character()
ipsc_mn_als_datasets.polyA.intronList = sort(ipsc_mn_als_datasets.polyA.intronList, decreasing = TRUE)
p53.signalling_intron.list <- list("p53.signalling" = ipsc_mn_als_datasets.polyA$irfinder$Intron.GeneName.GeneID.Coords[ipsc_mn_als_datasets.polyA$irfinder$gene_name %in% p53.signalling.pathway])
ipsc_mn_als_datasets.polyA.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_intron.list, stats = ipsc_mn_als_datasets.polyA.intronList)
ipsc_mn_als_datasets.polyA.p53.signalling_fgseaRes
#  pathway        pval        padj   log2err       ES    NES size
# 1: p53.signalling 0.001315554 0.001315554 0.4550599 0.349433 1.1949 1719
ipsc_mn_als_datasets.polyA.p53.signalling.gseaplot = plotEnrichment(p53.signalling_intron.list[["p53.signalling"]], ipsc_mn_als_datasets.polyA.intronList) + 
  theme_oz() + theme(axis.text=element_text(size=6, angle = 0, vjust = 0), axis.title=element_text(size=8)) + ggtitle(expression(IR~p53~signalling)) + xlab("") +
  annotate("text", x = 150000, y = 0.3, label = paste0("NES = ", round(ipsc_mn_als_datasets.polyA.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(ipsc_mn_als_datasets.polyA.p53.signalling_fgseaRes$pval, digits = 3)), size = 2.8, hjust = 0)
ipsc_mn_als_datasets.polyA.p53.signalling.gseaplot


# ## IR & GE
# # ipsc_mn_als_datasets.polyA$ir.ge = summariseIRjoinGE(ipsc_mn_als_datasets.polyA$irfinder, ipsc_mn_als_datasets.polyA$res)
# 
# ir_ge.ipsc_mn_als_datasets.polyA.filt <- ipsc_mn_als_datasets.polyA$ir.ge %>% #drop_na(GE.direction, IR.direction) %>%
#   mutate(significant = case_when(GE.padj < 0.05 & IR.padj < 0.05 ~ "both", GE.padj < 0.05 ~ "GE only", IR.padj < 0.05 ~ "IR only", TRUE ~ "None"),
#          direction = case_when(IR.direction == "IR up" & GE.direction == "GE down" ~ "IR up mRNA down", IR.direction == "IR up" & GE.direction == "GE up" ~ "IR up mRNA up",
#                                IR.direction == "IR down" & GE.direction == "GE down" ~ "IR down mRNA down", IR.direction == "IR down" & GE.direction == "GE up" ~ "IR down mRNA up"))
# ir_ge.ipsc_mn_als_datasets.polyA.filt %>% filter(GE.padj < 0.05, IR.padj < 0.05) %>% dplyr::count(direction)
# 
# ### Scatterplot
# ir_ge.ipsc_mn_als_datasets.polyA.stat.scatter <- ggscatter(ir_ge.ipsc_mn_als_datasets.polyA.filt, x = "IR.stat", y = "GE.stat", color = "significant", palette = c("firebrick2", "forestgreen", "#A6CEE3", "grey"),
#                                             cor.coef = TRUE, cor.method = "pearson", size = 0.5, cor.coeff.args = list(label.x = 2, label.y = 5, label.sep="\n", size = 2.8),
#           xlab = expression(Intron~Retention~Z~score~ALS~vs~CTRL), ylab = expression(Gene~expression~Z~score~ALS~vs~CTRL)) +
#   theme_oz() + theme(legend.position = "none") +
#   geom_hline(yintercept = 0, linetype = 2) + geom_vline(xintercept = 0, linetype = 2) +
#   coord_cartesian(ylim=c(-5,6), xlim = c(-3,3)) +
#   geom_text_repel(data = filter(ir_ge.ipsc_mn_als_datasets.polyA.filt, GE.padj < 0.05 | IR.padj < 0.05), #, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, nucleocytoplasmic_genes, go.pathways.msigdb$GO_MITOCHONDRION)),
#                   aes(x = IR.stat, y = GE.stat, label = gene_name), fontface = "italic", colour = "black", min.segment.length = 0, box.padding = 0.5, point.padding = 0.5, size = 2.5, max.overlaps = Inf) +
#   geom_smooth(data = ir_ge.ipsc_mn_als_datasets.polyA.filt, aes(x = IR.stat, y = GE.stat), method = "lm", se = F, show.legend = TRUE, colour = "black", linetype = 2)  +
#   annotate(geom="text", x=2, y=-4, label="IR up\nmRNA down\nin ALS", color="black", size = 2.7)  + #annotate(geom="text", x=3, y=9, label="IR up\nmRNA up", color="darkgrey", size = 2.8)  +
#   annotate(geom="text", x=-2, y = 5, label="IR down\nmRNA up\nin ALS", color="black", size = 2.7) #+   annotate(geom="text", x=-5, y=-9, label="IR down\nmRNA down", color="darkgrey", size = 2.8)
# ir_ge.ipsc_mn_als_datasets.polyA.stat.scatter
```

## Merge

```{r}
pan_als_ipsn_splicing_fig_merged = plot_grid(
  majiq_schematic.gg,
  plot_grid(majiq.als_vs_ctrl.volcano, polyA_majiq_violins_merged.gg, ncol = 2, labels = c("b","c"), rel_widths = c(1,0.9)),
  plot_grid(als_ctrl.voila.gost_filt.terms.curated, pan_als.voila_modulizer.sig.pie, ncol = 2, labels = c("d","e")),
  plot_grid(ipsc_mn_als_datasets.polyA.ir.volcano, ipsc_mn_als_datasets.polyA.ir.go_curated, ipsc_mn_als_datasets.polyA.p53.signalling.gseaplot, ncol = 3, labels = c("f","g","h"), rel_widths = c(0.9,0.6,0.7)),
  nrow = 4, rel_heights = c(0.5,1,0.6,1), labels = c("a","","",""))
# pan_als_ipsn_splicing_fig_merged
ggsave(pan_als_ipsn_splicing_fig_merged, filename = here(proj_path,"splicing/majiq/figures/pan_als_ipsn_splicing_fig_merged.png"), height = 10, width = 8.5)

```



# Table S9-10 MAJIQ & IRFinder panALS datasets

```{r}
als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
als_ctrl.voila.sig = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% select(-contains("percentile"),-contains("changing"),-ucsc_lsv_link) %>% arrange(-abs(deltapsi))
als_ctrl.voila.sig
write_csv(als_ctrl.voila.sig, here(proj_path,"splicing/majiq/figures/Table.majiq_splicing_panALS.sig.csv"))


ipsc_mn_als_datasets.polyA.ir.sig = ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% select(Intron.GeneName.GeneID.Coords, log2FoldChange, stat, padj, IRratio.ALS = IRratio.mutant, IRratio.ctrl, IRratio.diff, IRdirection, gene_name, gene_id)
ipsc_mn_als_datasets.polyA.ir.sig
write_csv(ipsc_mn_als_datasets.polyA.ir.sig, here(proj_path,"splicing/irfinder/ipsc_mn_als_datasets.polyA.ir.sig.csv"))

```


# Fig S15 TDP43 knockdown UNC13B event

```{r}
als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
tdp43_positive.als_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/tdp43_positive.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "tdp43_positive")
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
# neuronal_nuclei_tdp43.liu.voila = read_voila_deltapsi("/camp/project/proj-luscombn-patani/working/public-data/neuronal-nuclei-tdp43-liu-2019/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv", grp1 = "tdp43pos", grp2 = "tdp43neg") %>% mutate(mutation = "tdp43pos")
# sfpqkd_vs_scramble.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/patani-data/motor-neuron-rbp-knockdown-rnaseq/splicing/majiq/voila/whole_sfpqkd_vs_scramble.tsv"), grp1 = "sfpqkd_whole", grp2 = "scramble_whole") %>% mutate(mutation = "sfpqkd")


# Volcano TDP-43 knockdown
tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.labels = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
tdp43kd.als_vs_ctrl.brown.voila.volcano = majiq_deltapsi_volcano_plot(tdp43kd.als_vs_ctrl.brown.voila, junction_labels = tdp43kd.als_vs_ctrl.brown.voila.labels, ymax = 4.1, xmax = 0.7, xpos = 0.5, numerator = "TDP-43 KD", denomenator = "Control", dpi = 300)
tdp43kd.als_vs_ctrl.brown.voila.volcano

# Scatter correlation TDP-43 knockdown with pan-ALS
als_tdp43kd_majiq.join = select(als_ctrl.voila, gene_name, lsv_id, lsv_junction_id, lsv_type, junctions_coords, exons_coords, ir_coords, ALS.deltapsi = deltapsi, ALS.changing = changing_dpsi0.1) %>% 
  left_join(select(tdp43kd.als_vs_ctrl.brown.voila, gene_name, lsv_id, lsv_junction_id, lsv_type, junctions_coords, exons_coords, ir_coords, TDP43kd.deltapsi = deltapsi, TDP43kd.changing = changing_dpsi0.1))
als_tdp43kd_majiq.join %>% glimpse # 138,252
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) # 0
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% arrange(-(abs(ALS.deltapsi) + abs(TDP43kd.deltapsi))) %>% print(n=Inf)
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, abs(TDP43kd.deltapsi) > 0.1, ALS.deltapsi > 0 & TDP43kd.deltapsi > 0 | ALS.deltapsi < 0 & TDP43kd.deltapsi < 0) %>% print(n=Inf) # GRIA4, DONSON
# als_tdp43kd_majiq.join = select(als_ctrl.voila, gene_name, lsv_id,  lsv_type, junctions_coords, exons_coords, ir_coords, ALS.deltapsi = deltapsi, ALS.p = tnom) %>% 
#   left_join(select(tdp43kd.als_vs_ctrl.brown.voila, gene_name, lsv_id, lsv_type, junctions_coords, exons_coords, ir_coords, TDP43kd.deltapsi = mean_dpsi_per_lsv_junction, TDP43kd.changing = probability_changing))
# als_tdp43kd_majiq.join %>% glimpse # 229,823

als_tdp43kd_majiq.list = list("pan ALS" = als_ctrl.voila, "TDP-43 KD" = tdp43kd.als_vs_ctrl.brown.voila)
tdp43kd.als_vs_ctrl.brown.voila %>% filter(probability_changing < 0.95, gene_name %in% ALS_genes)

# scatter correlation
als_tdp43kd_majiq.join.labels = als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # "NYNRIN" "SNX2"  
als_tdp43kd_majiq.join.scatter =dpsi_majiq_scatter(model_list1 = als_tdp43kd_majiq.list, model_list2 = als_tdp43kd_majiq.list, mutation1 = "pan ALS", mutation2 = "TDP-43 KD", highlight = als_tdp43kd_majiq.join.labels) + xlim(-0.3,0.3) + ylim(-0.7,0.7)
als_tdp43kd_majiq.join.scatter

# Overlap splicing events in TDP-43 knockdown with pan-ALS
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% print(n=Inf)# NYNRIN
als_tdp43kd_majiq.join.labels = als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # "NYNRIN" "SNX2"  
als_tdp43kd_majiq.join.labels[als_tdp43kd_majiq.join.labels %in% RNA_BINDING_PROTEINS]

# Overlap at gene level:
pan_als_majiq_genes.sig = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) %>% pull(gene_name)
tdp43kd_brown_majiq_genes.sig = tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) %>% pull(gene_name)
pan_als_majiq_genes.sig[pan_als_majiq_genes.sig %in% tdp43kd_brown_majiq_genes.sig] # UNC13B

# where in UNC13B is the cryptic exon?
tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "UNC13B") %>% glimpse # "ENSG00000198722:s:35313899-35313989:j:35313989-35364545", "ENSG00000198722:s:35313899-35313989:j:35313989-35366947"

```

