---
title: "ipsc_mn_als_meta"
font-family: 'Helvetica'
author: |
  | Oliver Ziff
  | Patani Lab
  | The Francis Crick Institute
  | Institute of Neurology
  | University College London, UK
date: "`r Sys.Date()`"
output: 
 rmarkdown::html_document:
   theme: spacelab
   highlight: haddock
   code_folding: hide
   toc: true
   toc_float: true
   smooth_scroll: true
   number_sections: false
   self_contained: true  
knit: (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(dirname(inputFile), 'docs/index.html')) })
---

```{css, echo = FALSE}
h1, #TOC>ul>li {
  color: #000000;
    font-weight: bold;
}
h2, #TOC>ul>ul>li {
  color: #41494D;
}
```

# Setup

```{r setup, include=FALSE}
# Mac
source("/Volumes/lab-luscomben/home/users/ziffo/scripts/functions/mac_R_functions.R") # # source("/camp/lab/luscomben/home/users/ziffo/scripts/functions/mac_R_functions.R")
camp_path = here("/Volumes/lab-luscomben/home/users/ziffo")
proj_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
script_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta")
shared_path = here("/Volumes/lab-luscomben/home/users/ziffo/proj-luscombn-patani/working")

## OnDemand
# source("/camp/lab/luscomben/home/users/ziffo/scripts/functions/R_functions.R") # source("/camp/lab/luscomben/home/users/ziffo/scripts/functions/OnDemand_R_functions.R")
# camp_path = here("/camp/lab/luscomben/home/users/ziffo")
# proj_path = here("/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
# script_path = here(here(proj_path,"scripts/ipsc_mn_als_meta")
# shared_path = here("/camp/project/proj-luscombn-patani/working")

setwd(proj_path)
set_here(proj_path)
here(proj_path)
here::i_am("ipsc_mn_als_meta.Rmd")
here()
dr_here()

load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.metadata.RData")) # load metadata

# load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.RData") # 6.7GB > crashes R
ipsc_mn_als_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.rds")) # 1.25G
ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds")) # 84M
ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds")) # 162M
ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds")) # 85M
# ipsc_mn_vcp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_vcp_datasets.rds")) # 59M
# ipsc_mn_c9orf72_sporadic = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_sporadic.rds"))
# ipsc_mn_als_datasets.noiso = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.noiso.rds"))
# ipsc_mn_answerals.mutation = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.mutation.rds"))
ipsc_mn_familial_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_familial_datasets.rds"))
ipsc_mn_mutant_sporadic = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_mutant_sporadic.rds"))
ipsc_mn_tdp43_positive_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tdp43_positive_datasets.rds"))
ipsc_mn_tdp43_negative_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tdp43_negative_datasets.rds"))
ipsc_mn_tdp_pathology = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tdp_pathology.rds"))

ipsc_mn_als_datasets.polyA = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.rds"))
ipsc_mn_als_datasets.total = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.rds"))
# ipsc_mn_als_datasets.introns_tag_pct = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.introns_tag_pct.rds"))

ipsc_mn_answerals = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.rds"))
# ipsc_mn_answerals.progression = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.progression.rds"))
# ipsc_mn_answerals.alsfrs_slope = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.alsfrs_slope.rds"))
ipsc_mn_answerals.alsfrs_r_progression_slope = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.alsfrs_r_progression_slope.rds"))
# ipsc_mn_answerals.early_late_onset = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.early_late_onset.rds"))
ipsc_mn_answerals.age_onset = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.age_onset.rds"))
# ipsc_mn_answerals.early_late_death = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.early_late_death.rds"))
ipsc_mn_answerals.survival = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.survival.rds"))
ipsc_mn_answerals.onset_site = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.onset_site.rds"))
ipsc_mn_answerals.als_only.onset_site = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.als_only.onset_site.rds"))
ipsc_mn_answerals.bulbar_ctrl = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.bulbar_ctrl.rds"))
ipsc_mn_answerals.limb_ctrl = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.limb_ctrl.rds"))
ipsc_mn_answerals.other_ctrl = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.other_ctrl.rds"))
ipsc_mn_answerals.limb_bulbar = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.limb_bulbar.rds"))
# ipsc_mn_answerals.bulbar_other = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.bulbar_other.rds"))
# ipsc_mn_answerals.limb_other = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.limb_other.rds"))

ipsc_mn_answerals.mutant = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.mutant.rds"))
ipsc_mn_answerals.family_history = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.family_history.rds"))
ipsc_mn_answerals.ck = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.ck.rds"))
ipsc_mn_answerals.el_escorial = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_answerals.el_escorial.rds"))

nygc_postmortem_als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_als_vs_ctrl.rds"))
nygc_postmortem_spinal.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal.als_vs_ctrl.rds"))
nygc_postmortem_cervical.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_cervical.als_vs_ctrl.rds"))
nygc_postmortem_thoracic.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_thoracic.als_vs_ctrl.rds"))
nygc_postmortem_lumbar.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_lumbar.als_vs_ctrl.rds"))
nygc_postmortem_motor_cortex.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_motor_cortex.als_vs_ctrl.rds"))
# saveRDS(nygc_postmortem_cervical.als_vs_ctrl, "/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/expression/deseq2/nygc_postmortem_cervical.als_vs_ctrl.rds")


tdp43_kd_tyzack = readRDS(here(proj_path,"expression/deseq2/tdp43_kd_tyzack.rds"))
fus_kd_tyzack = readRDS(here(proj_path,"expression/deseq2/fus_kd_tyzack.rds"))
sfpq_kd_tyzack = readRDS(here(proj_path,"expression/deseq2/sfpq_kd_tyzack.rds"))
tdp43_kd_brown = readRDS(here(proj_path,"expression/deseq2/tdp43_kd_brown.rds"))
tdp43_kd_klim = readRDS(here(proj_path,"expression/deseq2/tdp43_kd_klim.rds"))
neuronal_nuclei_tdp43_liu = readRDS(here(proj_path,"expression/deseq2/neuronal_nuclei_tdp43_liu.rds"))
tdp43_kd_datasets_notyzack = readRDS(here(proj_path,"expression/deseq2/tdp43_kd_datasets_notyzack.rds"))
tdp43_kd_datasets = readRDS(here(proj_path,"expression/deseq2/tdp43_kd_datasets.rds"))

# source("/camp/lab/luscomben/home/users/ziffo/scripts/functions/all_r_functions.R")
conflict_prefer("unique", "base")
conflict_prefer("rowVars", "matrixStats")
```

## Update git / github

```{r}
library(usethis) # usethis & gert https://jeroen.github.io/gert2019 https://www.garrickadenbuie.com/blog/pull-request-flow-usethis/?interactive=1&steps=
use_git_config(user.name = "ojziff", user.email = "oliver.ziff@crick.ac.uk")
usethis::create_github_token()

credentials::set_github_pat("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") 
# make github
library(gert)
setwd(here(proj_path,"scripts/ipsc_mn_als_meta") # setwd then dont need to specify repo argument
git_commit_all("update") # stage all files & commit in one command - implies git_add(".")
pr_init("figures_and_tables") # initialise a new branch
pr_push() # this will open github in chrome > pull request > merge

git_sitrep() # print git and github info
git_config()
git_info()
git_ls() # list git tree
git_log() # check recent commits
git_status() # check which files are modified
# pr_resume("figures_and_tables") # resume a branch
# git_add(".") #stage the files you want to commmit - "." indicates all files.
# git_commit("update") # commit with comment
# pr_pull()

majiq_bam_symlink = ipsc_mn_als_datasets.metadata %>%
  mutate(nfcore_bam = paste0(database_dir, "/nfcore/star_salmon/",sample,".markdup.sorted.bam"), 
         majiq_bam = paste0(here(proj_path,"alignment/majiq_bamdir/",dataset_sample,".bam"), 
         `#!/bin/bash` = paste0("ln -s ",nfcore_bam," ",majiq_bam)) %>% select(`#!/bin/bash`)
majiq_bam_symlink
majiq_bam_symlink %>% pull(`#!/bin/bash`)
write_tsv(majiq_bam_symlink, here(proj_path,"scripts/majiq_bam_symlink.sh")

majiq_bai_symlink = ipsc_mn_als_datasets.metadata %>%
  mutate(nfcore_bai = paste0(database_dir, "/nfcore/star_salmon/",sample,".markdup.sorted.bam.bai"), 
         majiq_bai = paste0(here(proj_path,"alignment/majiq_bamdir/",dataset_sample,".bam.bai"), 
         `#!/bin/bash` = paste0("ln -s ",nfcore_bai," ",majiq_bai)) %>% select(`#!/bin/bash`)
majiq_bai_symlink
majiq_bai_symlink %>% pull(`#!/bin/bash`)
write_tsv(majiq_bai_symlink, here(proj_path,"scripts/majiq_bai_symlink.sh")


unstranded_samples_majiq = ipsc_mn_als_datasets.metadata %>% filter(strandedness == "unstranded") %>% 
  mutate(majiq_bam_name = paste0(dataset_sample,"=strandness:None")) %>% select(majiq_bam_name)
paste('"',unique(unstranded_samples_majiq$majiq_bam_name),'",', sep = "", collapse = '\n') %>% cat()

unstranded_samples_majiq = ipsc_mn_als_datasets.metadata %>% filter(strandedness == "unstranded") %>% 
  mutate(majiq_bam_name = paste0(dataset_sample,"=strandness:none,")) %>% select(majiq_bam_name)
paste('"',unique(unstranded_samples_majiq$majiq_bam_name),'"', sep = "", collapse = '') %>% cat()

unstranded_samples_majiq = ipsc_mn_als_datasets.metadata %>% filter(strandedness == "unstranded") %>% 
  mutate(majiq_bam_name = paste0(dataset_sample,"=strandness:none")) %>% select(majiq_bam_name)
paste(unique(unstranded_samples_majiq$majiq_bam_name),sep = "", collapse = '\n') %>% cat()

```

## AnswerALS metadata

```{r}
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_dataportal_datatable.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_participants.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_released_files.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/Clinical_Dictionary.csv")) %>% clean_names %>% glimpse
first_last_visit_alsfrs.answerals = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/answerals_progression_alsfrs.csv")) # answerals progression based on ALSFRS
AALSHXFX.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSHXFX.csv")) %>% clean_names %>% select(participant_id, subject_uid, onsetdt)
Demographics.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Demographics.csv")) %>% clean_names %>% select(participant_id, subject_uid, dob)
age_onset = AALSHXFX.csv %>% full_join(Demographics.csv) %>% mutate(age_onset = as.numeric((ddays(onsetdt) - ddays(dob)) / (364.25*24*60*60)), sample = gsub("-","_",participant_id)) %>% select(sample, age_onset) %>%
  mutate(early_late_onset = case_when(age_onset < 57 ~ "early", age_onset > 57 ~ "late"))
Mortality.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Mortality.csv")) %>% clean_names %>% filter(!dieddt %in% c("Glioblastoma","Prostate cancer", "physician assisted suicide")) %>% select(participant_id, subject_uid, dieddt) 
mortality = AALSHXFX.csv %>% full_join(Mortality.csv) %>% mutate(survival = as.numeric((ddays(dieddt) - ddays(onsetdt)) / (364.25*24*60*60)), sample = gsub("-","_",participant_id)) %>% select(sample, survival) %>%
  mutate(early_late_death = case_when(survival < 2.8 ~ "early", survival > 2.8 ~ "late"))
onset_site = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSHXFX.csv")) %>% clean_names %>% 
                        select(participant_id, subject_uid, hxblb, hxax, hxgen, hxli, hxot) %>% filter(!(hxblb == 0 & hxax == 0 & hxgen == 0 & hxli == 0 & hxot == 0)) %>%
  mutate(onset_site_detailed = case_when(hxblb == 1 & hxax == 0 & hxgen == 0 & hxli == 0 & hxot == 0 ~ "bulbar", hxblb == 0 & hxax == 0 & hxgen == 0 & hxli == 1 & hxot == 0 ~ "limb", hxblb == 0 & hxax == 0 & hxgen == 1 & hxli == 0 & hxot == 0 ~ "generalised", 
                                hxblb == 0 & hxax == 1 & hxgen == 0 & hxli == 0 & hxot == 0 ~ "axial", TRUE ~ "mixed"), #hxot == 1 ~ "other", 
         onset_site = case_when(onset_site_detailed %in% c("axial","mixed","generalised") ~ "other", TRUE ~ onset_site_detailed),
         bulbar_other = case_when(hxblb == 1 ~ "bulbar", TRUE ~ "other"), limb_other = case_when(hxli == 1 ~ "limb", TRUE ~ "other"), axial_other = case_when(hxax == 1 ~ "axial", TRUE ~ "other"), 
         bulbar_axial = case_when(hxblb == 1 ~ "bulbar", hxax == 1 ~ "axial", TRUE ~ "other"), bulbar_limb = case_when(hxblb == 1 ~ "bulbar", hxli == 1 ~ "limb", TRUE ~ "other"), bulbar_mixed = case_when(hxblb == 1 ~ "bulbar", onset_site == "mixed" ~ "mixed", TRUE ~ "other"), 
         limb_axial = case_when(hxli == 1 ~ "limb", hxax == 1 ~ "axial", TRUE ~ "other"), limb_mixed = case_when(onset_site == "mixed" ~ "mixed", hxli == 1 ~ "limb", TRUE ~ "other"), axial_mixed = case_when(hxax == 1 ~ "axial", onset_site == "mixed" ~ "mixed", TRUE ~ "other"),
         sample = gsub("-","_",participant_id)) %>% select(sample, subject_uid, onset_site, onset_site_detailed, bulbar_other,limb_other,axial_other,bulbar_axial,bulbar_limb,bulbar_mixed,limb_axial,limb_mixed,axial_mixed)
family_history = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Family_History_Log.csv")) %>% clean_names %>% select(participant_id, subject_uid, fhals, fhftd) %>% 
  mutate(family_history = case_when(fhals == 1 | fhftd == 1 ~ "yes", TRUE ~ "no"),sample = gsub("-","_",participant_id)) %>% select(sample, family_history) %>% distinct(sample, .keep_all = TRUE)
subjects = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/subjects.csv")) %>% clean_names %>% 
  mutate(mnd_group = case_when(subject_group_id == 17 ~ "other_mnd", subject_group_id == 1 ~ "als", subject_group_id == 5 ~ "healthy", subject_group_id == 11 ~ "carrier"), mnd_group = factor(mnd_group, levels = c("healthy", "carrier","other_mnd","als")), sample = gsub("-","_",participant_id)) %>% select(sample, mnd_group) %>% distinct(sample, .keep_all = TRUE)
ck_value = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Auxiliary_Chemistry.csv")) %>% clean_names %>% drop_na(acckrslt) %>% mutate(sample = gsub("-","_",participant_id), cknorm = case_when(cknorm == 1 ~ "normal", TRUE ~ "raised")) %>% arrange(-acckrslt) %>% distinct(sample, .keep_all = TRUE) %>% select(sample, ck = acckrslt, cknorm)
AALSHXFX.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSDXFX.csv")) %>% clean_names %>% drop_na(elescrlr) %>% select(participant_id, elescrlr) %>% 
  mutate(sample = gsub("-","_",participant_id)) %>% select(sample, elescrlr) %>% arrange(elescrlr) %>% distinct(sample, .keep_all = TRUE)
answerals_transcriptomic_files = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_released_files.csv")) %>% clean_names %>% filter(omic == "transcriptomics", grepl(".fastq.gz$", path)) %>% 
  mutate(fastq = case_when(grepl("_1.fastq.gz$", path) ~ "fastq_1", grepl("_2.fastq.gz$", path) ~ "fastq_2")) %>% 
  filter(participant_id != "CTRL-NEUEU392AE8") %>% #17 fastq's associated with this ID. Unclear if these are technical repeats or incorrectly labelled.
  pivot_wider(c(participant_id, experiment), names_from = fastq, values_from = c(path,size)) %>% 
  mutate(fastq_1 = paste0(here(camp_path,"proj-luscombn-patani/working/answerals/reads/"),path_fastq_1), fastq_2 = paste0(here(camp_path,"proj-luscombn-patani/working/answerals/reads/"), path_fastq_2))
samplesheet = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_dataportal_datatable.csv")) %>% clean_names %>% filter(transcriptomics == "Yes", participant_id != "CTRL-NEUEU392AE8") %>% # ipsc == "Yes", 
  left_join(answerals_transcriptomic_files) %>% 
  mutate(strandedness = "reverse", 
         condition = case_when(subject_group == "ALS" ~ "als", subject_group == "Healthy Control" ~ "ctrl", grepl("carrier",subject_group) ~ "carrier", subject_group == "Non-ALS MND" ~ "other_mnd"),
         mutation = case_when(self_reported_mutations %in% c("C9orf72","SOD1","SETX", "TARDBP (TDP43)") ~ self_reported_mutations, TRUE ~ wgs_variants), mutation = tolower(gsub(" [(]WGS[)]","",mutation)), mutation = gsub("tardbp [(]tdp43[)]","tardbp", mutation),  
         mutation = case_when(condition == "als" & is.na(mutation) ~ "sporadic", condition == "ctrl" & is.na(mutation) ~ "ctrl",TRUE ~ mutation), mutant = case_when(mutation %in% c("sporadic","ctrl") ~ "no", TRUE ~ "yes"),
         sample = gsub("-","_",participant_id), DIV = 32, instrument = "Illumina NovaSeq 6000") %>%
  select(sample, fastq_1, fastq_2, strandedness, condition, mutation, mutant, participant_id, guid, nygc_cgnd_id, sex, DIV, instrument, ethnicity, race, pls_subgroup, revised_el_escorial_criteria, age_at_sample_collection, age_at_symptom_onset, age_at_death, alsfrs_r_baseline, alsfrs_r_latest, alsfrs_r_progression_slope, cbs_baseline, cbs_latest, cbs_progression_slope, subject_group, ipsc, dimn, i_psc_line_id_root, i_psc_line_name, di_m_ns_line_name, batch_id, self_reported_mutations, wgs_variants, repeat_length_c9, repeat_length_atxn2, size_fastq_1, size_fastq_2) %>%
  filter(!sample %in% c("CASE_NEUMN922PMZ","CASE_NEUYC198DCR")) %>% # R2_fastq.gz corrupt crashing nfcore
  left_join(select(first_last_visit_alsfrs.answerals, sample, alsfrs_slope, progression)) %>% left_join(age_onset) %>% left_join(mortality) %>% left_join(onset_site) %>% left_join(family_history) %>% left_join(subjects) %>% left_join(ck_value) %>% left_join(AALSHXFX.csv) # add clinical data to metadata
samplesheet # Ensure no NA's in sample sheet
samplesheet %>% glimpse
samplesheet %>% distinct(participant_id) # 287
write_csv(samplesheet, file = here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/samplesheet.csv"), na = "")

```


Run the DESeq2 objects scripts to create RData. Takes ~ 2.5 hours

```{bash}
cd /camp/home/ziffo/home/projects/ipsc-mn-als-meta/scripts
mls
sar
sbatch -N 1 -c 8 --mem=80G -t 12:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsc_mn_als_meta_objects.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk --job-name=DESeq2objects 

# OnDemand Terminal Tab
cd /camp/home/ziffo/home/projects/ipsc-mn-als-meta/scripts
sbatch -N 1 -c 8 --mem=80G -t 12:00:00 --wrap="Rscript /camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsc_mn_als_meta_objects.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk --job-name=DESeq2objects 

```

```{r}
ipsc_mn_als_datasets.metadata %>% distinct(dataset) #16 (14)
ipsc_mn_als_datasets.metadata %>% distinct(dataset_sample) # 419
ipsc_mn_als_datasets.metadata %>% count(condition) #318 ALS, 101 ctrl
ipsc_mn_als_datasets.metadata %>% count(dataset, condition) %>% print(n=Inf)
ipsc_mn_als_datasets.metadata %>% count(mutation) %>% arrange(-n)
ipsc_mn_als_datasets.metadata %>% count(mutation) # 11

ipsc_mn_als_datasets.metadata %>% filter(condition == "ctrl") %>% count(mutation)
ipsc_mn_als_datasets.metadata %>% filter(mutation == "iso") %>% count(condition)

# 104 ctrls: 85 ctrl, 17 iso, 2 have mutations (OPTN and ALS2)

ipsc_mn_als_datasets.metadata %>% filter(condition == "ctrl") %>% count(mutation)

neurolincs.metadata %>% filter(dataset == "neurolincs.iMN") %>% distinct(instrument)

      ipsc_mn_als_datasets.introns_tag_pct$res %>% filter(padj < 0.05) #111
ipsc_mn_als_datasets.introns_tag_pct$irfinder %>% filter(padj < 0.05) #118,553
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) # 38
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) # 25
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) # 74
ipsc_mn_als_datasets.total$irfinder %>% filter(padj < 0.05) #118829

ipsc_mn_als_datasets$dds %>% colData %>% as_tibble %>% glimpse
ipsc_mn_als_datasets$dds %>% colData %>% as_tibble %>% my_summarise(samtools_mqc_generalstats_samtools_raw_total_sequences)

samtools_mqc_generalstats_samtools_raw_total_sequences
```



# Table S2 QC stats

Table S1 is iPSN protocols

Multiqc general stats

```{r}
multiqc_stat_paths = ipsc_mn_als_datasets.metadata %>% distinct(database_dir) %>% mutate(multiqc_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt")) %>% pull(multiqc_stat_paths)
multiqc_stats.tsv = multiqc_stat_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_stats.tsv) =  ipsc_mn_als_datasets.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(dataset)
multiqc_data <- map_dfr(multiqc_stats.tsv, bind_rows, .id = "dataset") %>% clean_names() %>% drop_na(samtools_mqc_generalstats_samtools_flagstat_total) %>% filter(sample %in% ipsc_mn_als_datasets.metadata$sample) %>%
  mutate(dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincs0" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS", dataset %in% c("dafianca.c9orf72","dafianca.tardbp") ~ "dafianca", TRUE ~ dataset)) %>%
  select(-fast_qc_mqc_generalstats_fastqc_percent_duplicates, -fast_qc_mqc_generalstats_fastqc_percent_gc, -fast_qc_mqc_generalstats_fastqc_avg_sequence_length, -fast_qc_mqc_generalstats_fastqc_percent_fails)
write_csv(multiqc_data, here(proj_path,"alignment/multiqc_general_stats.csv")

```


# Fig S2-4 PCA

Fig S1 PRISMA flow

```{r}
ipsc_mn_als_datasets.pcaData <- plotPCA(ipsc_mn_als_datasets$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","samtools_mqc_generalstats_samtools_flagstat_total", "samtools_mqc_generalstats_samtools_mapped_passed",
"biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna", "quali_map_mqc_generalstats_qualimap_5_3_bias", "quali_map_mqc_generalstats_qualimap_reads_aligned", "r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent",
"feature_counts_mqc_generalstats_featurecounts_percent_assigned", "feature_counts_mqc_generalstats_featurecounts_assigned", "picard_mqc_generalstats_picard_percent_duplication", "samtools_mqc_generalstats_samtools_error_rate",
"samtools_mqc_generalstats_samtools_non_primary_alignments", "samtools_mqc_generalstats_samtools_reads_mapped", "samtools_mqc_generalstats_samtools_reads_mapped_percent", "samtools_mqc_generalstats_samtools_reads_properly_paired_percent",
"samtools_mqc_generalstats_samtools_reads_mq0_percent", "samtools_mqc_generalstats_samtools_raw_total_sequences", "salmon_mqc_generalstats_salmon_percent_mapped", "salmon_mqc_generalstats_salmon_num_mapped", "star_mqc_generalstats_star_uniquely_mapped_percent",
"star_mqc_generalstats_star_uniquely_mapped"),
returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), 
         dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS", dataset %in% c("dafianca.c9orf72","dafianca.tardbp") ~ "dafianca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","dafianca","kapeli","kiskinis","luisier","mitchell","sareen","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) #, mutation = gsub("tardbp [(]tdp43[)]","tardbp", mutation))

# Colour by dataset
ipsc_mn_als_datasets.pca_dataset <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "right", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance")) #+ #coord_fixed() + 
  # ggforce::geom_mark_ellipse(data = filter(ipsc_mn_als_datasets.pcaData, dataset %in% c("neurolincs_iMN", "neurolincs_diMN","bhinge","desantis", "wang","dafinca","kiskinis")),  aes(label = dataset), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE)
ipsc_mn_als_datasets.pca_dataset
# Colour by strandedness
ipsc_mn_als_datasets.pca_strandedness <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=strandedness)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.2, keyheight=0.2, default.unit="inch")) + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_strandedness
# Colour by instrument
ipsc_mn_als_datasets.pca_instrument <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=instrument)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_blank(), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_instrument
# Colour by read depth
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_raw_total_sequences <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_raw_total_sequences)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6, angle = 0), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(1, 'cm'))  + 
  guides(color = guide_colourbar(title = "read depth", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$samtools_mqc_generalstats_samtools_raw_total_sequences,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_raw_total_sequences
# Colour by DIV
ipsc_mn_als_datasets.pca_DIV <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=DIV)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top"))  + labs(color="Days in vitro") + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets.pca_DIV
# Colour by ALS
ipsc_mn_als_datasets.pca_als <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=condition)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets.pca_als
# Colour by mutation
ipsc_mn_als_datasets.pca_mutation <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=mutation)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets.pca_mutation


ipsc_mn_als_datasets.pca_merged = ggarrange(
  ipsc_mn_als_datasets.pca_dataset, 
  ggarrange(ipsc_mn_als_datasets.pca_strandedness, ipsc_mn_als_datasets.pca_instrument, ncol = 2, labels = c("b","c")),
  ggarrange(ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_raw_total_sequences, ipsc_mn_als_datasets.pca_DIV, ncol = 2, labels = c("d","e")), 
  ggarrange(ipsc_mn_als_datasets.pca_als, ipsc_mn_als_datasets.pca_mutation, ncol = 2, labels = c("f","g")),
  nrow = 4, heights = c(1.4,1,1,1), labels = c("a","","",""))
# ipsc_mn_als_datasets.pca_merged
ggsave(ipsc_mn_als_datasets.pca_merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.pca_merged.png", height = 12, width = 8)


### PCA 2

# Color by intronic read %
multiqc_stat_paths = ipsc_mn_als_datasets.metadata %>% distinct(database_dir) %>% mutate(multiqc_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt")) %>% pull(multiqc_stat_paths)
multiqc_stats.tsv = multiqc_stat_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_stats.tsv) =  ipsc_mn_als_datasets.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_rseqc_read_distribution <- map_dfr(multiqc_stats.tsv, bind_rows, .id = "database_dir") %>% clean_names()# %>% 
  # mutate(exonic_read_pct = sum(cds_exons_tag_pct, x5_utr_exons_tag_pct, x3_utr_exons_tag_pct, na.rm = TRUE), intronic_read_pct = introns_tag_pct, intergenic_read_pct = sum(tss_up_1kb_tag_pct, tss_up_5kb_tag_pct, tss_up_10kb_tag_pct, tes_down_1kb_tag_pct, tes_down_5kb_tag_pct, tes_down_10kb_tag_pct, other_intergenic_tag_pct, na.rm = TRUE)) %>% select(database_dir, sample, exonic_read_pct, intronic_read_pct, intergenic_read_pct)
ipsc_mn_als_datasets.pcaData.read_distribution = ipsc_mn_als_datasets.pcaData %>% left_join(multiqc_rseqc_read_distribution) # need unique sample names - only works if sample & dataset are unique. If multiple dataset names within a single multiqc_general_stats.txt then will only join the first dataset name
ipsc_mn_als_datasets.pca_read_distribution <- ggplot(ipsc_mn_als_datasets.pcaData.read_distribution, aes(PC1, PC2, color=introns_tag_pct)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "intronic reads %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData.read_distribution$introns_tag_pct,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData.read_distribution, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData.read_distribution, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets.pca_read_distribution

# Colour by library_type
ipsc_mn_als_datasets.pca_library_type <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=library_type)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets.pca_library_type
# Colour by biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna
ipsc_mn_als_datasets.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "ribosomal rna %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna
# Colour by quali_map_mqc_generalstats_qualimap_5_3_bias
ipsc_mn_als_datasets.pca_quali_map_mqc_generalstats_qualimap_5_3_bias <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=quali_map_mqc_generalstats_qualimap_5_3_bias)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "5' 3' bias", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=1, low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_quali_map_mqc_generalstats_qualimap_5_3_bias
# Colour by quali_map_mqc_generalstats_qualimap_reads_aligned
ipsc_mn_als_datasets.pca_quali_map_mqc_generalstats_qualimap_reads_aligned <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=quali_map_mqc_generalstats_qualimap_reads_aligned)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads aligned", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$quali_map_mqc_generalstats_qualimap_reads_aligned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_quali_map_mqc_generalstats_qualimap_reads_aligned
# Colour by picard_mqc_generalstats_picard_percent_duplication
ipsc_mn_als_datasets.pca_picard_mqc_generalstats_picard_percent_duplication <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=picard_mqc_generalstats_picard_percent_duplication)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Read duplication %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$picard_mqc_generalstats_picard_percent_duplication,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +  
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_picard_mqc_generalstats_picard_percent_duplication
# Colour by samtools_mqc_generalstats_samtools_reads_mapped
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_mapped <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mapped)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Reads mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$samtools_mqc_generalstats_samtools_reads_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_mapped
# Colour by star_mqc_generalstats_star_uniquely_mapped_percent
ipsc_mn_als_datasets.pca_star_mqc_generalstats_star_uniquely_mapped_percent <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=star_mqc_generalstats_star_uniquely_mapped_percent)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads uniquely mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$star_mqc_generalstats_star_uniquely_mapped_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_star_mqc_generalstats_star_uniquely_mapped_percent
# Colour by star_mqc_generalstats_star_uniquely_mapped
ipsc_mn_als_datasets.pca_star_mqc_generalstats_star_uniquely_mapped <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=star_mqc_generalstats_star_uniquely_mapped)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads uniquely mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$star_mqc_generalstats_star_uniquely_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_star_mqc_generalstats_star_uniquely_mapped

ipsc_mn_als_datasets.pca_nfcore_qc_metrics.merged = ggarrange(
  ggarrange(ipsc_mn_als_datasets.pca_read_distribution, ipsc_mn_als_datasets.pca_library_type, ncol = 2, labels = c("a","b")),
  ggarrange(ipsc_mn_als_datasets.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna, ipsc_mn_als_datasets.pca_quali_map_mqc_generalstats_qualimap_5_3_bias, ncol = 2, labels = c("c","d")),
  ggarrange(ipsc_mn_als_datasets.pca_picard_mqc_generalstats_picard_percent_duplication, ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_mapped, ncol = 2, labels = c("e","f")), 
  ggarrange(ipsc_mn_als_datasets.pca_star_mqc_generalstats_star_uniquely_mapped_percent, ipsc_mn_als_datasets.pca_star_mqc_generalstats_star_uniquely_mapped, ncol = 2, labels = c("g","h")),
  nrow = 4, heights = c(1,1,1,1))
ipsc_mn_als_datasets.pca_nfcore_qc_metrics.merged
ggsave(ipsc_mn_als_datasets.pca_nfcore_qc_metrics.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.pca_nfcore_qc_metrics.merged.png", height = 11.75, width = 8.25)


# Colour by samtools_mqc_generalstats_samtools_reads_mapped_percent
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_mapped_percent <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mapped_percent)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Reads mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$samtools_mqc_generalstats_samtools_reads_mapped_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_mapped_percent
# Colour by samtools_mqc_generalstats_samtools_error_rate
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_error_rate <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_error_rate)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Error rate", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$samtools_mqc_generalstats_samtools_error_rate,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_error_rate
# Colour by samtools_mqc_generalstats_samtools_non_primary_alignments
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_non_primary_alignments <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_non_primary_alignments)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Non primary alignments", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$samtools_mqc_generalstats_samtools_non_primary_alignments,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_non_primary_alignments
# Colour by samtools_mqc_generalstats_samtools_reads_mq0_percent
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_mq0_percent <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mq0_percent)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Mapping quality 0 %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$samtools_mqc_generalstats_samtools_reads_mq0_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_mq0_percent
# Colour by samtools_mqc_generalstats_samtools_flagstat_total
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_flagstat_total <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_flagstat_total)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_flagstat_total
# Colour by samtools_mqc_generalstats_samtools_mapped_passed
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_mapped_passed <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_mapped_passed)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Mapped reads passed", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$samtools_mqc_generalstats_samtools_mapped_passed,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_mapped_passed
# Colour by r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent
ipsc_mn_als_datasets.pca_r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Proper pairs %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent
# Colour by samtools_mqc_generalstats_samtools_reads_properly_paired_percent
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_properly_paired_percent <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_properly_paired_percent)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Properly paired %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$samtools_mqc_generalstats_samtools_reads_properly_paired_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_samtools_mqc_generalstats_samtools_reads_properly_paired_percent
# Colour by feature_counts_mqc_generalstats_featurecounts_percent_assigned
ipsc_mn_als_datasets.pca_feature_counts_mqc_generalstats_featurecounts_percent_assigned <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=feature_counts_mqc_generalstats_featurecounts_percent_assigned)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Assigned reads %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$feature_counts_mqc_generalstats_featurecounts_percent_assigned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_feature_counts_mqc_generalstats_featurecounts_percent_assigned
# Colour by feature_counts_mqc_generalstats_featurecounts_assigned
ipsc_mn_als_datasets.pca_feature_counts_mqc_generalstats_featurecounts_assigned <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=feature_counts_mqc_generalstats_featurecounts_assigned)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Assigned reads", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$feature_counts_mqc_generalstats_featurecounts_assigned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_feature_counts_mqc_generalstats_featurecounts_assigned
# Colour by salmon_mqc_generalstats_salmon_percent_mapped
ipsc_mn_als_datasets.pca_salmon_mqc_generalstats_salmon_percent_mapped <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=salmon_mqc_generalstats_salmon_percent_mapped)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Reads mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$salmon_mqc_generalstats_salmon_percent_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_salmon_mqc_generalstats_salmon_percent_mapped
# Colour by salmon_mqc_generalstats_salmon_num_mapped
ipsc_mn_als_datasets.pca_salmon_mqc_generalstats_salmon_num_mapped <- ggplot(ipsc_mn_als_datasets.pcaData, aes(PC1, PC2, color=salmon_mqc_generalstats_salmon_num_mapped)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Reads mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.pcaData$salmon_mqc_generalstats_salmon_num_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets.pca_salmon_mqc_generalstats_salmon_num_mapped

```

## PC1 & PC2 gene loadings

https://stackoverflow.com/questions/12760108/principal-components-analysis-how-to-get-the-contribution-of-each-paramete
https://github.com/mikelove/DESeq2/blob/master/R/plots.R

```{r}
ipsc_mn_als_datasets.vsd.rv <- rowVars(assay(ipsc_mn_als_datasets$vsd)) # calculate the variance for each gene
ipsc_mn_als_datasets.vsd.ntop_genes <- order(ipsc_mn_als_datasets.vsd.rv, decreasing=TRUE)[seq_len(min(500, length(ipsc_mn_als_datasets.vsd.rv)))] # select the ntop genes by variance
ipsc_mn_als_datasets.vsd.pca <- prcomp(t(assay(ipsc_mn_als_datasets$vsd)[ipsc_mn_als_datasets.vsd.ntop_genes,])) # perform a PCA on the data in assay(x) for the selected genes
ipsc_mn_als_datasets.vsd.loadings <- as_tibble(ipsc_mn_als_datasets.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
ipsc_mn_als_datasets.vsd.loadings %>% arrange(PC1)
ipsc_mn_als_datasets.vsd.loadings %>% arrange(-PC1)

# ipsc_mn_als_datasets.vsd.pca <- prcomp(t(assay(ipsc_mn_als_datasets$vsd))) # perform a PCA on the data in assay(x) for the selected genes
# ipsc_mn_als_datasets.vsd.loadings <- as_tibble(ipsc_mn_als_datasets.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")
HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
neuronal.markers = c(nonspecific.neuronal, dl.domains, V0.domain,V1, V2a.domain, V2b.domain,MN.domain,V3.domain)

# scatter all PC1 vs PC2
ipsc_mn_als_datasets.vsd.loadings.scatter = ggplot(ipsc_mn_als_datasets.vsd.loadings, aes(x = PC1, y = PC2 )) + 
    labs(x = "PC1 gene loading", y = "PC2 gene loading") +
    geom_hline(yintercept = 0, linetype = 3) + geom_vline(xintercept = 0, linetype = 3) +
    geom_point(size = 1, alpha = 0.1) +
    scale_fill_continuous(type = "viridis") +
    guides(fill = FALSE) +
    theme_oz() +
  geom_text_repel(data = filter(ipsc_mn_als_datasets.vsd.loadings, abs(PC1) > 0.04, abs(PC2) > 0.04, grepl("^H[1-9]",gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) +
  geom_text_repel(data = filter(ipsc_mn_als_datasets.vsd.loadings, abs(PC1) > 0.04, abs(PC2) > 0.04, grepl("^SNOR|RN7SK|RMRP|RPPH1",gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) +
  geom_text_repel(data = filter(ipsc_mn_als_datasets.vsd.loadings,  gene_name %in% c(neuronal.markers, HOX.markers)), aes(x = PC1, y = PC2, label = gene_name), colour = "black", size = 2.3) +
  # geom_text_repel(data = filter(ipsc_mn_als_datasets.vsd.loadings, gene_name %in% c("COL3A1","COL1A2")), aes(x = PC1, y = PC2, label = gene_name), colour = "dodgerblue2", size = 2.3) +
  # geom_text_repel(data = filter(ipsc_mn_als_datasets.vsd.loadings, abs(PC2) > 0.1), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) #+
  # geom_segment(aes(x = 0.015, xend = 0.09, y= -0.1, yend= -0.1), arrow=arrow(length=unit(0.3,"cm"))) +  geom_segment(aes(x = -0.045, xend = -0.13, y= -0.1, yend= -0.1),arrow=arrow(length=unit(0.3,"cm"))) +
  annotate("text", x = 0.07, y = 0.14, label = "PC1+ PC2+",size = 3, color = "black") + annotate("text", x = -0.13, y = -0.04, label = "PC1- PC2-", size = 3, color = "black") + 
  annotate("text", x = 0.07, y = -0.04, label = "PC1+ PC2-", size = 3, color = "black") + annotate("text", x = -0.13, y = 0.14, label = "PC1- PC2+", size = 3, color = "firebrick2")
ipsc_mn_als_datasets.vsd.loadings.scatter
ggsave(ipsc_mn_als_datasets.vsd.loadings.scatter, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.vsd.loadings.scatter.png", height = 9, width = 9)


# # GO analysis on the 4 quadrants of scatterplot:
# # revigo
# ipsc_mn_als_datasets.vsd.loadings_gost <- ipsc_mn_als_datasets.vsd.loadings %>% group_split(PC1 > 0, PC2 > 0) %>% map("gene_id") %>% gost() 
# ipsc_mn_als_datasets.vsd.loadings_gost.res_filt = ipsc_mn_als_datasets.vsd.loadings_gost$result %>% filter(str_detect(source, paste0("GO:BP"))) %>% arrange(-log10(p_value)) %>% mutate(query = case_when(query == "query_1" ~ "PC1- PC2-", query == "query_2" ~ "PC1- PC2+", query == "query_3" ~ "PC1+ PC2-", query == "query_4" ~ "PC1+ PC2+"), term_name = as.factor(term_name))
# 
# ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query1 <- ipsc_mn_als_datasets.vsd.loadings_gost.res_filt %>% filter(query == "PC1- PC2-") %>% top_n(10, wt = -p_value)
# paste('"',unique(ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query1$term_name),'",', sep = "", collapse = '\n') %>% cat()
# 
# ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query2 <- ipsc_mn_als_datasets.vsd.loadings_gost.res_filt %>% filter(query == "PC1- PC2+") %>% top_n(10, wt = -p_value)
# paste('"',unique(ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query2$term_name),'",', sep = "", collapse = '\n') %>% cat()
# 
# ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query3 <- ipsc_mn_als_datasets.vsd.loadings_gost.res_filt %>% filter(query == "PC1+ PC2-") %>% top_n(10, wt = -p_value)
# paste('"',unique(ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query3$term_name),'",', sep = "", collapse = '\n') %>% cat()
# 
# ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query4 <- ipsc_mn_als_datasets.vsd.loadings_gost.res_filt %>% filter(query == "PC1+ PC2+") %>% top_n(10, wt = -p_value)
# paste('"',unique(ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query4$term_name),'",', sep = "", collapse = '\n') %>% cat()
# 
# reducedTerms_combined <- bind_rows(mutate(ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query1, query = "PC1- PC2-"), mutate(ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query2, query = "PC1- PC2+"), mutate(ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query3, query = "PC1+ PC2-"), mutate(ipsc_mn_als_datasets.vsd.loadings_gost.res_filt.query4, query = "PC1+ PC2+")) %>% 
#   mutate(query = factor(query, levels = c("PC1- PC2+", "PC1+ PC2+", "PC1+ PC2-", "PC1- PC2-")))
# 
# ipsc_mn_als_datasets.vsd.loadings.go_curated <- curated_profiles(gprofiles = reducedTerms_combined, label_angle = 0, cols =  c("gold2", "forestgreen", "dodgerblue2", "firebrick2")) #+ theme_oz()
# ipsc_mn_als_datasets.vsd.loadings.go_curated

# pca_gene_loadings = ggarrange(ipsc_mn_als_datasets.vsd.loadings.scatter, ipsc_mn_als_datasets.vsd.loadings.go_curated, nrow = 2, labels = c("a","b"))
# pca_gene_loadings
# ggsave(pca_gene_loadings, filename = here(proj_path,"expression/deseq2/pca_gene_loadings.png", height = 12, width = 9)

```

Dendogram

```{r}
# Dendrogram
ipsc_mn_als_datasets.sampleDists <- dist(t(assay(ipsc_mn_als_datasets$vsd)))
ipsc_mn_als_datasets.hc <- hclust(ipsc_mn_als_datasets.sampleDists, method = "ward.D2")
ipsc_mn_als_datasets.hc$labels <- gsub("dafianca.tardbp", "dafianca", ipsc_mn_als_datasets.hc$labels) %>% gsub("dafianca.c9orf72", "dafinca", .) %>% gsub("mn_d25_", "", .)

display.brewer.pal(n = 12, name = 'Paired')
brewer.pal(n = 12, name = "Paired")
show_col(brewer_pal(palette = "Paired")(12))
"#A6CEE3" # light blue
"#1F78B4" 
"#B2DF8A" # green
"#33A02C"
"#FB9A99" # red
"#E31A1C" 
"#FDBF6F" # orange
"#FF7F00"
"#CAB2D6" # purple
"#6A3D9A" 
"#B15928 # brown
"#FFFF99" # yellow - replace with darker yellow for readability #F0E442
ipsc_mn_als_datasets.a <- c(rep("#386CB0", 3), rep("#75489F", 9), rep("#386CB0", 6), 
                                                   rep("#FEEA92", 6), # desantis
                                                   rep("#BCCEA0", 4), # wang
                                                   rep("#7A9DA8", 6), # sterneckert
                                                   rep("#A9B7B7", 6),# kapeli
                                                   rep("#E8BAA0", 6), # mitchell
                                                   rep("#D3B4BA", 6), #luisier
                                                  rep("#FDC086", 8),#sareen
                                                 rep("#7FC97F", 12), # catanese
                                                 rep("#BEAED4", 5), #kiskinis
                            rep("#94C09B", 12), # dafianca
                            rep("#FDD58C", 5), # smith
                            rep("#94C09B", 10)) # dafianca
# PCA colour assignment:
# c("HB9::GFP CTRL"="#386CB0", "HB9::GFP ALS"="#386CB0", "PHOX2B::GFP CTRL"="#75489F", "PHOX2B::GFP ALS"="#75489F", "desantis"="#FEEA92", "catanese"="#7FC97F", "dafianca"="#94C09B", 
                                # "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "mitchell"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sterneckert"="#7A9DA8", "wang"="#BCCEA0"))
ipsc_mn_als_datasets.dendrogram <- ggdendrogram(ipsc_mn_als_datasets.hc, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(size = 6), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
ipsc_mn_als_datasets.dendrogram
ggsave(ipsc_mn_als_datasets.dendrogram, filename = here(proj_path,"expression/deseq2/dendrogram.png", height = 20, width = 8.25)

# ### confirm hclust(method="single") for fairness - Lee samples are the last to join the other datasets. 
# ipsc_mn_als_datasets.hc.single <- hclust(ipsc_mn_als_datasets.sampleDists, method = "single")
# ipsc_mn_als_datasets.hc.single$labels <- gsub("dafianca.tardbp", "dafianca", ipsc_mn_als_datasets.hc.single$labels) %>% gsub("dafianca.c9orf72", "dafinca", .) %>% gsub("mn_d25_", "", .) %>% gsub("REP", "", .) %>% gsub("lee_spinal", "HB9::GFP",.) %>% gsub("lee_ocular", "PHOX2B::GFP", .)
# ipsc_mn_als_datasets.a.single =  c(rep("#A6CEE3", 3), rep("#1F78B4", 3), rep("black", 86), 
#                                                    rep("#A6CEE3", 6), # desantis
#                                                    rep("#1F78B4", 6))
# ipsc_mn_als_datasets.dendrogram <- ggdendrogram(ipsc_mn_als_datasets.hc.single, rotate = TRUE, theme_dendro = FALSE) +  theme_classic()  + theme(axis.line=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.text.y= element_text(colour = ipsc_mn_als_datasets.a.single, size = 8), strip.background = element_rect(colour="black", fill="white"), panel.grid = element_blank()) 
# ipsc_mn_als_datasets.dendrogram

```

Sample-to-sample distances

```{r}
rownames(colData(ipsc_mn_als_datasets$vsd)) = rownames(colData(ipsc_mn_als_datasets$vsd)) %>% gsub("dafianca.tardbp", "dafianca",.) %>% gsub("dafianca.c9orf72", "dafinca", .) %>% gsub("mn_d25_", "", .)
sampleDists <- dist(t(assay(ipsc_mn_als_datasets$vsd)))
# library("RColorBrewer")
# PCA colour assignment:
# c("HB9::GFP CTRL"="#386CB0", "HB9::GFP ALS"="#386CB0", "PHOX2B::GFP CTRL"="#75489F", "PHOX2B::GFP ALS"="#75489F", "desantis"="#FEEA92", "catanese"="#7FC97F", "dafianca"="#94C09B", 
                                # "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "mitchell"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sterneckert"="#7A9DA8", "wang"="#BCCEA0"))

sampleDistMatrix <- as.matrix(sampleDists)
colnames(sampleDistMatrix) <- NULL
rownames(sampleDistMatrix)
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
ipsc_mn_als_datasets.sample2sample_dists<- Heatmap(sampleDistMatrix,
                                                   border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
        clustering_distance_rows = "euclidean", clustering_distance_columns = "euclidean",
        clustering_method_rows  = "average", clustering_method_columns  = "average",
        col=colors,
        # name="Distance",
        row_names_gp = gpar(fontsize = 6))#, col = c(rep("#7FC97F",12), rep("#386CB0",6), rep("#75489F",9), rep("#386CB0",3), rep("#94C09B",22), rep("#D3B4BA",6), rep("#E8BAA0",6), rep("#BCCEA0",4), rep("#BEAED4",5), rep("#FDC086",8), rep("#7A9DA8",6), rep("#A9B7B7",6), rep("#FEEA92",6), rep("#FDD58C",5))))
ipsc_mn_als_datasets.sample2sample_dists
ggsave(as.ggplot(ipsc_mn_als_datasets.sample2sample_dists), filename = here(proj_path,"expression/deseq2/sample2sample_dists.png", height = 20, width = 8.25)
```


# Fig S5 RUVSeq

https://github.com/mikelove/preNivolumabOnNivolumab/blob/main/preNivolumabOnNivolumab.knit.md

```{r}
# # simple design, no adjustment
# ipsc_mn_als_datasets.metadata.salmon_files = ipsc_mn_als_datasets.metadata %>% mutate(file_salmon = file.path(database_dir, "nfcore/star_salmon", ipsc_mn_als_datasets.metadata[["sample"]], "quant.sf")) %>% pull(file_salmon)
# names(ipsc_mn_als_datasets.metadata.salmon_files) = ipsc_mn_als_datasets.metadata %>% pull(dataset_sample) # sample filename in nfcore outdir
# rownames(ipsc_mn_als_datasets.metadata) <- ipsc_mn_als_datasets.metadata %>% pull(dataset_sample)  # unique sample name
# ipsc_mn_als_datasets.dds.ruv <- tximport(ipsc_mn_als_datasets.metadata.salmon_files, type="salmon", tx2gene=tx2gene) %>% DESeqDataSetFromTximport(colData = ipsc_mn_als_datasets.metadata, design = ~condition) 
# ipsc_mn_als_datasets.vsd.ruv.lrt <- vst(ipsc_mn_als_datasets.dds.ruv, blind=FALSE)
# ipsc_mn_als_datasets.dds.ruv.lrt = ipsc_mn_als_datasets.dds.ruv %>% DESeq(test = "LRT", reduced = ~1) #, fitType="glmGamPoi")
# ipsc_mn_als_datasets.res.ruv.lrt <- DESeq2::results(ipsc_mn_als_datasets.dds.ruv.lrt)
# table(ipsc_mn_als_datasets.res.ruv.lrt$padj < .1)
# # FALSE  TRUE 
# # 37458 17045 
# # estimate technical variation with RUVg method
# # library(RUVSeq)
# set <- newSeqExpressionSet(counts(ipsc_mn_als_datasets.dds.ruv.lrt))
# set <- betweenLaneNormalization(set, which="upper")
# not_sig <- rownames(ipsc_mn_als_datasets.res.ruv.lrt)[which(ipsc_mn_als_datasets.res.ruv.lrt$pvalue > .1)]
# empirical <- rownames(set)[ rownames(set) %in% not_sig ]
# set <- RUVg(set, empirical, k=5)
# 
# # W_1 and W_2 etc are unwanted variation factors
# pdat <- pData(set) %>% mutate(condition = case_when)
# pdat$condition <- ipsc_mn_als_datasets.metadata$condition
# 
# # visualize how the factors of unwanted variation describe the samples in the PC1 and PC2 space:
# ipsc_mn_als_datasets.vsd.ruv.lrt$W1 <- pdat$W_1
# ipsc_mn_als_datasets.vsd.ruv.lrt$W2 <- pdat$W_2
# ipsc_mn_als_datasets.vsd.ruv.lrt$W3 <- pdat$W_3
# ipsc_mn_als_datasets.vsd.ruv.lrt$W4 <- pdat$W_4
# ipsc_mn_als_datasets.vsd.ruv.lrt$W5 <- pdat$W_5
# saveRDS(ipsc_mn_als_datasets.vsd.ruv.lrt, here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.vsd.ruv.lrt.rds")
ipsc_mn_als_datasets.vsd.ruv.lrt = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.vsd.ruv.lrt.rds")

ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData <- plotPCA(ipsc_mn_als_datasets.vsd.ruv.lrt, intgroup=c("sample", "W1", "W2", "W3", "W4", "W5"), returnData=TRUE) %>% as_tibble()
ipsc_mn_als_datasets.vsd.ruv.lrt.W1 <- ggplot(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W1)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W1", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData$W1,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets.vsd.ruv.lrt.W1

ipsc_mn_als_datasets.vsd.ruv.lrt.W2 <- ggplot(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W2)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W2", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData$W2,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets.vsd.ruv.lrt.W2

ipsc_mn_als_datasets.vsd.ruv.lrt.W3 <- ggplot(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W3)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W3", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData$W3,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets.vsd.ruv.lrt.W3

ipsc_mn_als_datasets.vsd.ruv.lrt.W4 <- ggplot(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W4)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W4", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData$W4,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets.vsd.ruv.lrt.W4

ipsc_mn_als_datasets.vsd.ruv.lrt.W5 <- ggplot(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, aes(PC1, PC2, color=W5)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "RUV factor W5", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData$W5,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.vsd.ruv.lrt.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets.vsd.ruv.lrt.W5

ipsc_mn_als_datasets.vsd.ruv.lrt.pca = ipsc_mn_als_datasets.vsd.ruv.lrt.W1 + ipsc_mn_als_datasets.vsd.ruv.lrt.W2 + ipsc_mn_als_datasets.vsd.ruv.lrt.W3 + ipsc_mn_als_datasets.vsd.ruv.lrt.W4 + ipsc_mn_als_datasets.vsd.ruv.lrt.W5 + 
  plot_layout(ncol = 2, nrow = 3)
ipsc_mn_als_datasets.vsd.ruv.lrt.pca
ggsave(ipsc_mn_als_datasets.vsd.ruv.lrt.pca, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.vsd.ruv.lrt.pca.png", height = 11.75, width = 8.25)


# # Adding the factors to the design, and performing a LRT is fairly simple, first we show this with DESeq2:
# colData(ipsc_mn_als_datasets.dds.ruv.lrt) <- cbind(colData(ipsc_mn_als_datasets.dds.ruv.lrt), pdat[,1:5])
# design(ipsc_mn_als_datasets.dds.ruv.lrt) <- ~W_1 + W_2 + W_3 + W_4 + W_5 + condition
# ipsc_mn_als_datasets.dds.ruv.lrt <- DESeq(ipsc_mn_als_datasets.dds.ruv.lrt, test="LRT", reduced=~W_1 + W_2 + W_3 + W_4 + W_5)
# # Controlling for technical variation in this case greatly reduces the number of DE genes. Controlling for technical variation can also increase the number of DE genes.
# # That the number is reduced here indicates that some of the previous results were likely due to confounding of technical variation with the condition variable. Ignoring that confounding, again, will result in invalid inference for all methods that look for shifts in the expression values.
# ipsc_mn_als_datasets.res.ruv.lrt <- DESeq2::results(ipsc_mn_als_datasets.dds.ruv.lrt)
# table(ipsc_mn_als_datasets.res.ruv.lrt$padj < .1)
# # FALSE  TRUE 
# # 28633  1712 

```


# Fig S6-9 Heatmap expression

## Cell type markers

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","EDNRB",  "FGFR3",  "SLC4A4", "DIO2")#"SOX9", ,"MLC1"
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4", "PECAM1")
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CCL3","IL1A", "TREM2") # "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "CX3CR1","P2RY12", "TNF",  "TMEM119"
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1")

undifferentiated = c("SOX2","POU5F1","UTF1","REX1","ALPL","TERT","ESRG","CNMD","SFRP2")
neural.precursor = c("NES","SOX1","SOX2","CD133","PAX6","NOTCH1","OCLN","CDH1","CDH2","SOX10","VIM","GFAP")
neural.progenitor = c("MAP2","HUC","HUD","NF","NCAM","DCX","SOX2")
neural.crest <- c("SOX10","NGFR","FOXD3","SNAI1","SNAI2","LMX1A")   #NGFR very low in HB9 samples
dorsal.progenitor <- c("ATOH1","OLIG3","MSX1","PAX3","PAX7","DBX1","DBX2","IRX3","ASCL1")
ho.genes = c("SST","DCX","RFX4","ASCL1","HOXB8","NEFH","SCN1A","TNS1","SPOCK3","SNAP25","TDGF1","POU5F1","L1TD1","FXYD5","PHLDA2","TOP2A","DLGAP5","FZD7","ILF2","MSH2")

nonspecific = c("SOX2")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
pMN.domain = c("NKX6-1", "OLIG2")
ventral.progenitor = c("NKX2-2","NKX2-8")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")

nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

# neuronal.markers = c("NPAS4", "NPY", "SST", "CCK", "GAD1", "SYT1", "STMN2", "SYN1") # "ENO2", 

nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#"NEFH",
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3") #"ETV1", "NOS1",,"RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2"
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

### Cell type markers
celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, oligo.markers, mn.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]
celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte.markers", gene_name %in% endo.markers ~ "endo.markers", 
                           gene_name %in% myeloid.markers ~ "myeloid.markers", gene_name %in% oligo.markers ~ "oligo.markers", 
                           gene_name %in% mn.markers ~ "mn.markers"), 
         group = factor(group, levels = c("astrocyte.markers", "endo.markers", "myeloid.markers", "oligo.markers","mn.markers"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafianca.tardbp", "dafianca", dataset_sample), dataset_sample = gsub("dafianca.c9orf72", "dafinca", dataset_sample), dataset_sample = gsub("mn_d25_", "",dataset_sample), dataset = as.factor(dataset)) 


# together with other datasets
min_vst = min(ipsc_mn_als_datasets$vsd.counts[,2])
celltype.markers.marker <- ipsc_mn_als_datasets$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("dafianca.tardbp", "dafianca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafianca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))

rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(celltype.markers.marker.centered_mat)]

### Cluster samples ###
celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered

celltype.markers.rnaseq.cluster.heatmap <- Heatmap(t(celltype.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = celltype.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = celltype.markers.df.filt$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelial", "Myeloid", "Oligodendrocyte","Motor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, 
          show_row_names = FALSE, #row_names_gp = gpar(fontsize = 6), 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
celltype.markers.rnaseq.cluster.heatmap
ggsave(as.ggplot(celltype.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.celltype.markers.heatmap.png", width = 8, height = 13)
```


## Neuron type markers

```{r}
nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#,"NEFH"
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3","ETV1", "NOS1","RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2") #
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)


neuron.type.markers = c(motor.neuron.progenitor, postmitotic.motor.neuron, nonspecific.neuronal,interneuron.markers) 
neuron.type.markers[!neuron.type.markers %in% gtf$gene_name]
neuron.type.markers.df = tibble("gene_name" = neuron.type.markers) %>% filter(gene_name %in% neuron.type.markers) %>% 
  mutate(group = case_when(gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% motor.neuron.progenitor ~ "motor.neuron.progenitor", 
                           gene_name %in% postmitotic.motor.neuron ~ "postmitotic.motor.neuron", #gene_name %in% oculomotor.markers ~ "oculomotor.markers", 
                           gene_name %in% interneuron.markers ~ "interneuron"), 
         group = factor(group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafianca.tardbp", "dafianca", dataset_sample), dataset_sample = gsub("dafianca.c9orf72", "dafinca", dataset_sample), dataset_sample = gsub("mn_d25_", "",dataset_sample), dataset = as.factor(dataset)) 


# together with other datasets
min_vst = min(ipsc_mn_als_datasets$vsd.counts[,2])
neuron.type.markers.marker <- ipsc_mn_als_datasets$vsd.counts %>% 
  filter(gene_name %in% neuron.type.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuron.type.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("dafianca.tardbp", "dafianca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafianca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
neuron.type.markers.df.filt = neuron.type.markers.df %>% filter(gene_name %in% neuron.type.markers.marker$gene_name)
neuron.type.markers.marker.mat <- make_matrix(select(neuron.type.markers.marker,-gene_name), pull(neuron.type.markers.marker,gene_name))

rownames(neuron.type.markers.marker.centered_mat)[!rownames(neuron.type.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(neuron.type.markers.marker.centered_mat)]

### Cluster samples ###
neuron.type.markers.marker.centered_mat = t(scale(t(neuron.type.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
neuron.type.markers.rnaseq.heatmap <- Heatmap(t(neuron.type.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = neuron.type.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = factor(neuron.type.markers.df.filt$group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron")),
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",4)), labels = c("nonspecific\nneuronal", "motor neuron\nprogenitor", "postmitotic\nmotor neuron", "interneuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, 
          show_row_names = FALSE,
          row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
neuron.type.markers.rnaseq.heatmap
ggsave(as.ggplot(neuron.type.markers.rnaseq.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.neuron.type.markers.heatmap.png", width = 8, height = 13)

## clustering samples
neuron.type.markers.rnaseq.cluster.heatmap <- Heatmap(t(neuron.type.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = neuron.type.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = factor(neuron.type.markers.df.filt$group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron")),
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",4)), labels = c("nonspecific\nneuronal", "motor\nneuron\nprogenitor", "postmitotic\nmotor\nneuron", "interneuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, 
          show_row_names = TRUE, row_names_gp = gpar(fontsize = 6), 
          # row_order = dataset.samples.df$dataset_sample,
          # row_split = dataset.samples.df$dataset,
          # right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
          #                                                        labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Smith", "Sterneckert", "Wang"), 
          #                                                        labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = TRUE, cluster_column_slices = FALSE, cluster_columns = FALSE)
neuron.type.markers.rnaseq.cluster.heatmap
ggsave(as.ggplot(neuron.type.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.neuron.type.markers.cluster.heatmap.png", width = 9, height = 16)

```

Motor neuron markers only heatmap

```{r}
celltype.marker.df = data.frame("gene_name" = mn.markers) %>% mutate(group = case_when(gene_name %in% mn.markers ~ "motor neuron"))
ipsc_mn_als_datasets$vsd.counts %>% filter(gene_name %in% mn.markers) %>% group_by(gene_name) %>% filter(n()>1) # any duplicate gene_names?

vsd_counts.celltype.marker <-  ipsc_mn_als_datasets$vsd.counts  %>% filter(gene_name %in% mn.markers) %>% select(-gene_id) %>% mutate_if(is.numeric, funs((. - 4.505))) #rowwise() %>%
vsd_counts.celltype.marker.mat <- make_matrix(select(vsd_counts.celltype.marker,-gene_name), pull(vsd_counts.celltype.marker,gene_name))
celltype.marker.df.filt <- celltype.marker.df %>% filter(gene_name %in% vsd_counts.celltype.marker$gene_name)

motor.neuron.marker.heatmap <- Heatmap(t(vsd_counts.celltype.marker.mat), 
          heatmap_legend_param = list(title = ""), 
          cluster_columns = TRUE, column_order = vsd_counts.celltype.marker$gene_name,
          show_column_names = TRUE, column_names_gp = gpar(fontsize = 8, fontface = "italic"), #row_labels = celltype.marker.row.labels,
          top_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = c("white")), labels = c("Motor neuron markers"), labels_gp = gpar(fontsize = 10))), column_title = NULL,
          show_row_names = FALSE, #column_names_rot = 90, 
          row_title = NULL, cluster_rows = FALSE,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
                                                                 labels = c("AnswerALS", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "bhinge", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)), row_split = dataset.samples.df$dataset)

motor.neuron.marker.heatmap

```


## Dorsoventral markers

From Fig 2 in https://journals.biologists.com/dev/article/148/15/dev199711/271192/Single-cell-transcriptome-profiling-of-the-human 

```{r}
nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

neuronal.markers = c(dl.domains, V0.domain,V1, V2a.domain, V2b.domain,MN.domain,V3.domain) #nonspecific.neuronal
neuronal.markers[!neuronal.markers %in% gtf$gene_name]
neuronal.markers.df = tibble("gene_name" = neuronal.markers) %>% 
  mutate(group = case_when(gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% dl.domains ~ "dl.domains", 
                           gene_name %in% V0.domain ~ "V0.domain", gene_name %in% V1 ~ "V1", 
                           gene_name %in% V2a.domain ~ "V2a.domain", gene_name %in% V2b.domain ~ "V2b.domain", gene_name %in% MN.domain ~ "MN.domain", gene_name %in% V3.domain ~ "V3.domain"), 
         group = factor(group, levels = c("dl.domains","V0.domain", "V1", "V2a.domain", "V2b.domain", "MN.domain", "V3.domain"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

# together with other datasets
min_vst = min(ipsc_mn_als_datasets$vsd.counts[,2])
neuronal.markers.marker <- ipsc_mn_als_datasets$vsd.counts %>% 
  filter(gene_name %in% neuronal.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuronal.markers.df$gene_name)) %>% select(-gene_id) %>%
  rename_with(~ gsub("dafianca.tardbp", "dafianca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafianca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
neuronal.markers.marker.mat <- make_matrix(select(neuronal.markers.marker,-gene_name), pull(neuronal.markers.marker,gene_name))

### Cluster samples ###
neuronal.markers.marker.centered_mat = t(scale(t(neuronal.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
dorsoventral.markers.heatmap <- Heatmap(t(neuronal.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = neuronal.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = neuronal.markers.df$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",7)), labels =  c("dl\ndomains","V0\ndomain", "V1", "V2a\ndomain", "V2b\ndomain", "MN\ndomain", "V3"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          # show_row_names = TRUE, row_names_gp = gpar(fontsize = 6, col = c(rep("#386CB0",9), rep("#75489F",9), rep("black",90))), #row_names_rot = 90, 
          cluster_rows = FALSE,
          row_title = NULL, 
          row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)))
dorsoventral.markers.heatmap
ggsave(as.ggplot(dorsoventral.markers.heatmap), filename = here(proj_path,"expression/deseq2/dorsoventral.markers.heatmap.png", width = 8, height = 13)


# cluster by columns
Heatmap(neuronal.markers.marker.centered_mat, 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          row_order = neuronal.markers.df$gene_name,
          row_names_gp = gpar(fontsize = 7), 
          row_split = factor(neuronal.markers.df$group, levels =  c("nonspecific.neuronal","dl.domains","V0.domain", "V1", "V2a.domain", "V2b.domain", "MN.domain", "V3.domain")),
          right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = rep("white",8)), labels =  c("nonspecific\nneuronal","dl\ndomains","V0\ndomain", "V1", "V2a\ndomain", "V2b\ndomain", "MN\ndomain", "V3\ndomain"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = "Neuronal markers", row_title_gp = gpar(fontsize = 8), 
          show_column_names = TRUE, column_names_gp = gpar(fontsize = 6, col = c(rep("#386CB0",9), rep("#75489F",9), rep("black",90))), #column_names_rot = 90,
          cluster_rows = FALSE, cluster_row_slices = FALSE, cluster_columns = TRUE,
          column_title = NULL)

```

Progenitor dosroventral markers


```{r}
nonspecific = c("SOX2")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
pMN.domain = c("NKX6-1", "OLIG2")
ventral.progenitor = c("NKX2-2","NKX2-8")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")

progenitor.markers = c(nonspecific, roof.plate, dorsal.progenitor,intermediate.progenitor, pMN.domain, ventral.progenitor,floor.plate)
progenitor.markers[!progenitor.markers %in% gtf$gene_name]
progenitor.markers.df = tibble("gene_name" = progenitor.markers) %>% 
  mutate(group = case_when(gene_name %in% nonspecific ~ "nonspecific", gene_name %in% roof.plate ~ "roof.plate", 
                           gene_name %in% dorsal.progenitor ~ "dorsal.progenitor", gene_name %in% intermediate.progenitor ~ "intermediate.progenitor", 
                           gene_name %in% pMN.domain ~ "pMN.domain", gene_name %in% ventral.progenitor ~ "ventral.progenitor", gene_name %in% floor.plate ~ "floor.plate"), 
         group = factor(group, levels = c("nonspecific","roof.plate", "dorsal.progenitor", "intermediate.progenitor", "pMN.domain", "ventral.progenitor", "floor.plate"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

# together with other datasets
min_vst = min(ipsc_mn_als_datasets$vsd.counts[,2])
progenitor.markers.marker <- ipsc_mn_als_datasets$vsd.counts %>% 
  filter(gene_name %in% progenitor.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = progenitor.markers.df$gene_name)) %>% select(-gene_id) %>%
  rename_with(~ gsub("dafianca.tardbp", "dafianca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafianca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
progenitor.markers.marker.mat <- make_matrix(select(progenitor.markers.marker,-gene_name), pull(progenitor.markers.marker,gene_name))

### Cluster samples ###
progenitor.markers.marker.centered_mat = t(scale(t(progenitor.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
progenitor.markers.heatmap <- Heatmap(t(progenitor.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = progenitor.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = progenitor.markers.df$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",7)), labels =  c("nonspecific","roof\nplate", "dorsal\nprogenitor", "intermediate\nprogenitor", "pMN\ndomain", "ventral\nprogenitor", "floor\nplate"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title ="Progenitor markers", column_title_gp = gpar(fontsize = 8), 
          show_row_names = FALSE, 
          row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
progenitor.markers.heatmap


# figS5.merged = ggarrange(as.ggplot(progenitor.markers.heatmap),
#                          as.ggplot(neuronal.markers.heatmap),
#                          nrow = 2, labels = c("a","b"), heights = c(0.8,1.2))
# figS5.merged
# ggsave(as.ggplot(figS5.merged), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.neuron.type.markers.cluster.heatmap.png", width = 8, height = 13)


```


## HOX markers

Regional identity of iPSN

```{r}
HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
# forebrain.markers = c("FOXG1","OTX2")
# HOX.markers = c(forebrain.markers, HOX.markers)
HOX.markers.df <- data.frame(row.names = HOX.markers, gene_name = HOX.markers) %>% 
  mutate(group = gsub("HOX[A-D]","", gene_name), group = factor(as.numeric(gsub("-AS$","", group)))) %>% 
  arrange(group) %>% 
  mutate(level = factor(case_when(group %in% c(1,2,3) ~ "Hindbrain", group %in% c(4,5,6,7) ~ "Cervical", group %in% c(8,9) ~ "Thoracic", 
                                  group %in% c(10,11) ~ "Lumbar", TRUE ~ "Sacral"), levels = c("Hindbrain","Cervical","Thoracic","Lumbar","Sacral")))
min_vst = min(ipsc_mn_als_datasets$vsd.counts[,2])
HOX.markers.marker <- ipsc_mn_als_datasets$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% 
  filter(gene_name %in% HOX.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst))) %>% 
  arrange(factor(gene_name, levels = HOX.markers.df$gene_name)) %>% select(-gene_id) %>%
  rename_with(~ gsub("dafianca.tardbp", "dafianca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafianca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
HOX.markers.marker.mat <- make_matrix(select(HOX.markers.marker,-gene_name), pull(HOX.markers.marker,gene_name))
HOX.markers.marker.centered.mat = t(scale(t(HOX.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
HOX.markers.heatmap <- Heatmap(t(HOX.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = HOX.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = HOX.markers.df$level,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Hindbrain", "Cervical", "Thoracic","Lumbar","Sacral"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          cluster_rows = FALSE,
          row_title = NULL, 
          row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)))
HOX.markers.heatmap
ggsave(as.ggplot(HOX.markers.heatmap), filename = here(proj_path,"expression/deseq2/HOX.markers.heatmap.png", width = 8, height = 13)
```

## non-polyadenylated markers

From IRFinder paper https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1184-4

```{r}
nonpolyadenylated_genes = read_csv("/camp/lab/luscomben/home/users/ziffo/genomes/annotation/nonpolyadenylated_genes.csv") %>% pull(nonpolyadenylated_genes)
nonpolyadenylated_genes = read_csv("/camp/lab/luscomben/home/users/ziffo/genomes/annotation/nonpolyadenylated_genes.csv") %>% pull(nonpolyadenylated_genes)
nonpolyadenylated_genes[!nonpolyadenylated_genes %in% gtf$gene_name]

HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)

min_vst = min(ipsc_mn_als_datasets$vsd.counts[,2])
nonpolyadenylated.markers.marker <- ipsc_mn_als_datasets$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% 
  filter(gene_name %in% nonpolyadenylated_genes) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst))) %>% 
  arrange(factor(gene_name, levels = nonpolyadenylated_genes)) %>% select(-gene_id) %>%
  rename_with(~ gsub("dafianca.tardbp", "dafianca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafianca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
nonpolyadenylated.markers.marker.mat <- make_matrix(select(nonpolyadenylated.markers.marker,-gene_name), pull(nonpolyadenylated.markers.marker,gene_name))
nonpolyadenylated.markers.marker.centered.mat = t(scale(t(nonpolyadenylated.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
nonpolyadenylated.markers.heatmap <- Heatmap(t(nonpolyadenylated.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          # column_order = nonpolyadenylated_genes,
          column_names_gp = gpar(fontsize = 7), 
          # column_split = nonpolyadenylated.markers.df$level,
          # bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Hindbrain", "Cervical", "Thoracic","Lumbar","Sacral"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          cluster_rows = FALSE,
          row_title = NULL, 
          row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)),
                                                                 labels = c("AnswerALS", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "bhinge", "Smith", "Sterneckert", "Wang"), labels_gp = gpar(fontsize = 7), labels_rot = 0)))
nonpolyadenylated.markers.heatmap
ggsave(as.ggplot(nonpolyadenylated.markers.heatmap), filename = here(proj_path,"expression/deseq2/nonpolyadenylated.markers.heatmap.png", width = 8, height = 13)


nonpolyadenylated.markers.heatmap <- Heatmap(t(nonpolyadenylated.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          # column_order = nonpolyadenylated_genes,
          column_names_gp = gpar(fontsize = 7), 
          # column_split = nonpolyadenylated.markers.df$level,
          # bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Hindbrain", "Cervical", "Thoracic","Lumbar","Sacral"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          cluster_rows = TRUE,
          row_title = NULL, 
          # row_order = dataset.samples.df$dataset_sample,
          # row_split = dataset.samples.df$dataset,
          show_row_names = TRUE, row_names_gp = gpar(fontsize = 6)) 
# right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)),
                                                                 # labels = c("AnswerALS", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Kapeli", "Kiskinis", "Luisier", "Mitchell", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "bhinge", "Smith", "Sterneckert", "Wang"), labels_gp = gpar(fontsize = 7), labels_rot = 0)))
nonpolyadenylated.markers.heatmap
```



# Fig 2 hiPSC ALS meta-analysis

## Volcano

```{r}
ipsc_mn_als_datasets$res %>% filter(padj < 0.05)
ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0)
ipsc_mn_als_datasets$res %>% filter(grepl("ADAR",gene_name))
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # 53


## Volcano
ipsc_mn_als_datasets.als.labels <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.gene.labels <- ipsc_mn_als_datasets$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.volcano <- volcano_plot(ipsc_mn_als_datasets$res, numerator = "ALS", denomenator = "CTRL",
               subtitle = "101 Controls vs 318 ALS", ymax = 8, xmax = 2, dpi = 300,
               gene_labels = c(ipsc_mn_als_datasets.als.labels, ipsc_mn_als_datasets.gene.labels))
ipsc_mn_als_datasets.volcano

ipsc_mn_als_datasets$res %>% filter(padj<0.05, log2FoldChange > 0) #25
ipsc_mn_als_datasets$res %>% filter(padj<0.05, log2FoldChange < 0) #54

ipsc_mn_als_datasets.noiso$res %>% filter(padj<0.05, log2FoldChange > 0) #43
ipsc_mn_als_datasets.noiso$res %>% filter(padj<0.05, log2FoldChange < 0) #104
```

## GO terms & GSEA

```{r}
ipsc_mn_als_datasets.gost <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_als_datasets.gprofiles <- ipsc_mn_als_datasets.gost$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
ipsc_mn_als_datasets.gprofiles_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")
ipsc_mn_als_datasets.gprofiles_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(ipsc_mn_als_datasets.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"neuron fate commitment",
"cell fate commitment")
ipsc_mn_als_datasets.gprofiles_filt_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_als_datasets.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"death receptor activity",
# "hsa-miR-938",
# "5S rRNA binding",
"TP53 network",
"Senescence and autophagy in cancer",
# "oral mucosa; squamous epithelial cells[Medium]",
"ErbB signaling pathway",
"Genotoxicity pathway",
"rRNA binding",
"p53 signaling pathway",
"miRNA regulation of DNA damage response",
"DNA damage response")
ipsc_mn_als_datasets.gprofiles_filt_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_als_datasets.gprofiles_filt_combined <- bind_rows(ipsc_mn_als_datasets.gprofiles_filt_down, ipsc_mn_als_datasets.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
ipsc_mn_als_datasets.gprofiles_filt_combined = ipsc_mn_als_datasets.gprofiles_filt_combined %>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name))
ipsc_mn_als_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_als_datasets.gprofiles_filt_combined) #+ theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
ipsc_mn_als_datasets.go_curated

# which genes are driving the terms
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA Damage Response`) # TNFRSF10B, CDKN1A, MDM2, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`TP53 Network`) # CDKN1A, MDM2
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`cell fate commitment`) # TBX5, MYOG, LMO4, OLIG1, TBR1, OLIG2, FOXN4, BCL11B
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`neuron fate commitment`) # TBX5, MYOG, LMO4, OLIG1, TBR1, OLIG2, FOXN4, BCL11B


## GSEA: DNA damage response, p53 signalling pathway
ipsc_mn_als_datasets.geneList <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.geneList) <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ipsc_mn_als_datasets.geneList = sort(ipsc_mn_als_datasets.geneList, decreasing = TRUE)

# DNA damage response
dna.damage_genes.list <- list("dna.damage" = ipsc_mn_als_datasets$res$gene_name[ipsc_mn_als_datasets$res$gene_name %in% gprofiler_database$`DNA Damage Response`])
ipsc_mn_als_datasets.dna.damage_fgseaRes <- fgsea(pathways = dna.damage_genes.list, stats = ipsc_mn_als_datasets.geneList)
ipsc_mn_als_datasets.dna.damage_fgseaRes
#       pathway       pval       padj   log2err        ES      NES size                               leadingEdge
# 1: dna.damage 0.02747435 0.02747435 0.3524879 0.4434927 1.351135   67 TNFRSF10B,CDKN1A,MDM2,SESN1,RRM2B,BAX,...

ipsc_mn_als_datasets.dna.damage.gseaplot = plotEnrichment(dna.damage_genes.list[["dna.damage"]], ipsc_mn_als_datasets.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(DNA~damage~response)) + xlab("") +
  annotate("text", x = 25000, y = 0.3, label = paste0("NES = ", round(ipsc_mn_als_datasets.dna.damage_fgseaRes$NES, digits = 3),"\nP = ", round(ipsc_mn_als_datasets.dna.damage_fgseaRes$pval, digits = 3)), size = 2.5, hjust = 0) + theme_oz()
ipsc_mn_als_datasets.dna.damage.gseaplot

# p53 signalling pathway
p53.signalling.pathway = unique(c(gprofiler_database$`TP53 Network`, go.pathways.msigdb$GO_P53_BINDING, go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_SIGNAL_TRANSDUCTION_BY_P53_CLASS_MEDIATOR, go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_DNA_DAMAGE_RESPONSE_SIGNAL_TRANSDUCTION_BY_P53_CLASS_MEDIATOR, go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_INTRINSIC_APOPTOTIC_SIGNALING_PATHWAY_BY_P53_CLASS_MEDIATOR))
p53.signalling_genes.list <- list("p53.signalling" = ipsc_mn_als_datasets$res$gene_name[ipsc_mn_als_datasets$res$gene_name %in% p53.signalling.pathway])
ipsc_mn_als_datasets.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = ipsc_mn_als_datasets.geneList)
ipsc_mn_als_datasets.p53.signalling_fgseaRes
#           pathway        pval        padj   log2err       ES      NES size                          leadingEdge
# 1: p53.signalling 0.005941836 0.005941836 0.4070179 0.455499 1.479662   95 CDKN1A,MDM2,TRIAP1,BAX,TAF3,BBC3,...
ipsc_mn_als_datasets.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], ipsc_mn_als_datasets.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ylab(expression(p53~signalling)) + xlab("") +
  annotate("text", x = 25000, y = 0.3, label = paste0("NES = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$pval, digits = 3)), size = 2.5, hjust = 0) + theme_oz()
ipsc_mn_als_datasets.p53.signalling.gseaplot

```


## DoRoTHEA & PROGENy

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/pw_bk.html 
<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.res.matrix <- ipsc_mn_als_datasets$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_progeny.pathwayActivity <- ggplot(ipsc_mn_als_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "iPSN: Signaling Pathways", x = "", y = "ALS vs CTRL Normalised enrichment") +
    stat_pvalue_manual(ipsc_mn_als_progeny.contrast_acts, label = "p.signif", y.position = 13, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_als_progeny.pathwayActivity

# p53 weights
ipsc_mn_als_progeny.p53weights = net_progeny %>% filter(source == "p53", target %in% ipsc_mn_als_datasets$res$gene_name) %>% left_join(select(ipsc_mn_als_datasets$res, target=gene_name,stat)) %>% 
  mutate(color = case_when(weight > 0 & stat > 0 ~ '1', weight > 0 & stat < 0 ~ '2', weight < 0 & stat > 0 ~ '2', weight < 0 & stat < 0 ~ '1', TRUE ~ "3"))
ipsc_mn_als_progeny.p53weights.plot = ggplot(ipsc_mn_als_progeny.p53weights, aes(x = weight, y = stat, color = color)) + geom_point() +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_progeny.p53weights, weight > 10, stat > 2), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "iPSN: p53 weights", x = "p53 pathway gene weight", y = "ALS vs CTRL test statistic") +
  geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted') +
  coord_cartesian(xlim = c(-9,19)) 
ipsc_mn_als_progeny.p53weights.plot

```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.1 & abs(score) < 4 ~ "sig", p_value < 0.1 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.1 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_dorothea.contrast_acts.labels.up = ipsc_mn_als_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_dorothea.contrast_acts.labels.down = ipsc_mn_als_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "iPSN: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.contrast_acts, source %in% c(ipsc_mn_als_dorothea.contrast_acts.labels.up, ipsc_mn_als_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted')
# ipsc_mn_als_dorothea.contrast_acts.volcano

# TP53 target genes
ipsc_mn_als_dorothea.tp53 <- net_dorothea %>%  filter(source == "TP53") %>% mutate(gene_name = target) %>% left_join(select(ipsc_mn_als_datasets$res, gene_name,stat,log2FoldChange,pvalue)) %>%
  mutate(color = case_when(mor > 0 & stat > 0 ~ '1', mor > 0 & stat < 0~'2',mor < 0 & stat > 0~'2',mor < 0 & stat < 0~'1',TRUE ~ "3"))

ipsc_mn_als_dorothea.tp53.volcano = ggplot(ipsc_mn_als_dorothea.tp53, aes(x = log2FoldChange, y = -log10(pvalue), color = color, size=abs(mor))) + geom_point() + scale_size(range = c(0, 3)) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(data = filter(ipsc_mn_als_dorothea.tp53, abs(stat) > 3), aes(label = gene_name, size=1), fontface = "italic", max.overlaps = Inf, size = 2.3) + 
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +  geom_vline(xintercept = 0, linetype = 'dotted') + geom_hline(yintercept = 0, linetype = 'dotted') +
    labs(title = "TP53 regulon", x = "log2 fold change (ALS vs CTRL)", y = expression(-log[10]~P~value))
# ipsc_mn_als_dorothea.tp53.volcano

```

## Merge

```{r}
ipsc_mn_als_datasets.figs.merged = ggarrange(
  ggarrange(ipsc_mn_als_datasets.volcano, ipsc_mn_als_datasets.go_curated,  ggarrange(ipsc_mn_als_datasets.dna.damage.gseaplot, ipsc_mn_als_datasets.p53.signalling.gseaplot, nrow=2, labels = c("c","d"), heights = c(1,1)), ncol=3, labels = c("a","b",""), widths = c(1.1,0.7,0.5)),
  ggarrange(ipsc_mn_als_progeny.pathwayActivity, ipsc_mn_als_progeny.p53weights.plot,ipsc_mn_als_dorothea.contrast_acts.volcano, ncol=3, labels = c("e","f","g"), widths = c(1,1,1)),
  nrow = 2, heights = c(1.3,1))
# ipsc_mn_als_datasets.figs.merged
ggsave(ipsc_mn_als_datasets.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.figs.merged.png"), height = 8, width = 10)

ipsc_mn_als_datasets.figs.merged = plot_grid(
  plot_grid(ipsc_mn_als_datasets.volcano, ipsc_mn_als_datasets.go_curated, ncol=2, rel_widths = c(1,0.8), labels = c("a","b")), 
  plot_grid(ipsc_mn_als_datasets.dna.damage.gseaplot, ipsc_mn_als_datasets.p53.signalling.gseaplot, ipsc_mn_als_progeny.pathwayActivity, ncol=3, rel_widths = c(0.7,0.7,1), labels = c("c","d","e")), 
  plot_grid(ipsc_mn_als_progeny.p53weights.plot,ipsc_mn_als_dorothea.contrast_acts.volcano, ncol=2, labels = c("f","g")),
  nrow = 3, rel_heights = c(1.3,1,1))
# ipsc_mn_als_datasets.figs.merged
ggsave(ipsc_mn_als_datasets.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.figs.merged.png"), height = 10, width = 8.25)


compare_mutations_irfinder_volcano_multiplot <- plot_grid(
  plot_grid(ipsc_mn_sporadic_datasets.ir.volcano, ipsc_mn_c9orf72_datasets.ir.volcano, ipsc_mn_fus_datasets.ir.volcano, ipsc_mn_sod1_datasets.ir.volcano, ipsc_mn_tardbp_datasets.ir.volcano, ipsc_mn_mutations_irfinder_ggheatmap, labels = "auto", ncol = 2, nrow = 3),
  ipsc_mn_mutants_bind_datasets.irfinder.upset_intron_level, labels = c("","g"), nrow = 2, rel_heights = c(3,1)) 
# compare_mutations_irfinder_volcano_multiplot
ggsave(compare_mutations_irfinder_volcano_multiplot, filename = here(proj_path,"splicing/irfinder/compare_mutations_irfinder_volcano_multiplot.png"), height = 11.75, width = 8.25)


# ipsc_mn_als_datasets.figs.merged = ggarrange(
#   ggarrange(ipsc_mn_als_datasets.volcano, ipsc_mn_als_datasets.go_curated,  ggarrange(ipsc_mn_als_datasets.dna.damage.gseaplot, ipsc_mn_als_datasets.p53.signalling.gseaplot, nrow=2, labels = c("c","d"), heights = c(1,1)), ncol=3, labels = c("a","b",""), widths = c(1.1,0.7,0.5)),
#   ggarrange(ipsc_mn_als_progeny.pathwayActivity, ipsc_mn_als_progeny.p53weights.plot, ncol=2, labels = c("e","f"), widths = c(1,1)),
#   ggarrange(ipsc_mn_als_dorothea.contrast_acts.volcano, ipsc_mn_als_dorothea.tp53.volcano, ncol=2, labels = c("g","h"), widths = c(1,1)),
#   ggarrange(nygc_postmortem_spinal_cord.als_vs_ctrl.p53_TP53.barplot, postmortem_ipsn.als_vs_ctrl.scatter, ncol=2, labels = c("i","j"), widths = c(1,1)),
#   nrow = 4, heights = c(1.3,1,1,1))
# # ipsc_mn_als_datasets.figs.merged
# ggsave(ipsc_mn_als_datasets.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.figs.merged.png"), height = 12, width = 9)


```


# Table S3 DGE panALS

```{r}
ipsc_mn_als_datasets.res.sig = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% select(gene_name, everything())
ipsc_mn_als_datasets.res.sig
write_csv(ipsc_mn_als_datasets.res.sig, here(proj_path,"expression/deseq2/TableS2.ipsc_mn_als_datasets.res.sig.csv")

```

# Table S4 Dorothea panALS

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% select(-statistic, -condition, transcription_factor = source, normalised_enrichment_score = score) %>% arrange(p_value, -abs(normalised_enrichment_score))
ipsc_mn_als_dorothea.contrast_acts
write_csv(ipsc_mn_als_dorothea.contrast_acts, here(proj_path,"expression/deseq2/TableS4.ipsc_mn_als_dorothea.csv"))

```



# Fig 3 Compare mutations

Compare sporadic, C9orf72, SOD1, FUS, TARDBP & VCP

```{r}
ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds")) # 84M
ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds")) # 162M
ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds")) # 85M
# ipsc_mn_vcp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_vcp_datasets.rds")) # 59M

```

### Volcano

```{r}
ipsc_mn_mutants_bind_datasets.res = bind_rows(mutate(ipsc_mn_sporadic_datasets$res, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$res, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$res, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$res, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$res, mutation = "FUS"))#, mutate(ipsc_mn_vcp_datasets$res, mutation = "VCP"))

ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% count(mutation) %>% arrange(-n)
# mutation     n
#   <chr>    <int>
# 1 TARDBP    3623
# 2 FUS        364
# 3 C9orf72    166
# 4 SOD1        43
# 5 VCP         28
# 6 Sporadic     3

ipsc_mn_mutants_join_datasets.res = full_join(select(ipsc_mn_c9orf72_datasets$res, gene_name, padj.c9orf72 = padj, log2FoldChange.c9orf72 = log2FoldChange),
          select(ipsc_mn_sporadic_datasets$res, gene_name, padj.sals = padj, log2FoldChange.sals = log2FoldChange)) %>%
  full_join(select(ipsc_mn_fus_datasets$res, gene_name, padj.fus = padj, log2FoldChange.fus = log2FoldChange)) %>%
  full_join(select(ipsc_mn_tardbp_datasets$res, gene_name, padj.tardbp = padj, log2FoldChange.tardbp = log2FoldChange)) %>%
  full_join(select(ipsc_mn_sod1_datasets$res, gene_name, padj.sod1 = padj, log2FoldChange.sod1 = log2FoldChange)) #%>% full_join(select(ipsc_mn_vcp_datasets$res, gene_name, padj.vcp = padj, log2FoldChange.vcp = log2FoldChange))
ipsc_mn_mutants_join_datasets.res.labels = ipsc_mn_mutants_join_datasets.res %>% filter(c(padj.c9orf72 < 0.05 & padj.fus < 0.05 & padj.tardbp < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.tardbp < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.tardbp < 0.05 & padj.c9orf72 < 0.05)) %>% pull(gene_name)

ipsc_mn_tardbp_datasets.metadata %>% count(condition) 
ipsc_mn_tardbp_datasets.labels <- ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.001, gene_name %in% c(ALS_genes,"XIST","NUP210","JAK1","PCDHB18P","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_tardbp_datasets.volcano <- volcano_plot(ipsc_mn_tardbp_datasets$res, "TARDBP", numerator = "TARDBP", denomenator = "CTRL", subtitle = "48 Controls vs 10 TARDBP", ymax = 18, xmax = 4.5, dpi = 300, gene_labels = c(ipsc_mn_tardbp_datasets.labels,"JAK1","PCDHB18P","NPIPA8"))
# ipsc_mn_tardbp_datasets.volcano

ipsc_mn_fus_datasets.metadata %>% count(condition) 
ipsc_mn_fus_datasets.labels <- ipsc_mn_fus_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes,"XIST","NUP210")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_fus_datasets.volcano <- volcano_plot(ipsc_mn_fus_datasets$res, "FUS", numerator = "FUS", denomenator = "CTRL", subtitle = "53 Controls vs 12 FUS", ymax = 14, xmax = 4, dpi = 300, gene_labels = c(ipsc_mn_fus_datasets.labels,"JAK1","PCDHB18P","NPIPA8"))
# ipsc_mn_fus_datasets.volcano

ipsc_mn_c9orf72_datasets.metadata %>% count(condition) 
ipsc_mn_c9orf72_datasets.labels <- ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.02, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS,"NUP210")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_c9orf72_datasets.volcano <- volcano_plot(ipsc_mn_c9orf72_datasets$res, "C9orf72", numerator = "C9orf72", denomenator = "CTRL", subtitle = "77 Controls vs 54 C9orf72", ymax = 10, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_c9orf72_datasets.labels, "C9orf72","PCDHB18P","NPIPA8"))
# ipsc_mn_c9orf72_datasets.volcano

ipsc_mn_sod1_datasets.metadata %>% count(condition) 
ipsc_mn_sod1_datasets.labels <- ipsc_mn_sod1_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS,"NUP210","SERPINA3","PLK5")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_sod1_datasets.volcano <- volcano_plot(ipsc_mn_sod1_datasets$res, "SOD1", numerator = "SOD1", denomenator = "CTRL", subtitle = "63 Controls vs 20 SOD1", ymax = 10, xmax = 4, dpi = 300, gene_labels = c(ipsc_mn_sod1_datasets.labels,"JAK1"))
# ipsc_mn_sod1_datasets.volcano

# ipsc_mn_vcp_datasets.metadata %>% count(condition) 
# ipsc_mn_vcp_datasets.labels <- ipsc_mn_vcp_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS,"XIST","NUP210","DPP6","GLYATL2","FAM66B","GATD3A","PCDHGA7")) %>% distinct(gene_name) %>% pull(gene_name)
# ipsc_mn_vcp_datasets.volcano <- volcano_plot(ipsc_mn_vcp_datasets$res, "VCP", numerator = "VCP", denomenator = "CTRL", subtitle = "6 Controls vs 6 VCP", ymax = 11, xmax = 4.5, dpi = 300, gene_labels = ipsc_mn_vcp_datasets.labels)
# # ipsc_mn_vcp_datasets.volcano

ipsc_mn_sporadic_datasets.metadata %>% count(condition) 
ipsc_mn_sporadic_datasets.labels <- ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_sporadic_datasets.volcano <- volcano_plot(ipsc_mn_sporadic_datasets$res, "Sporadic", numerator = "Sporadic", denomenator = "CTRL", subtitle = "56 Controls vs 208 Sporadic", ymax = 10, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_sporadic_datasets.labels))
# ipsc_mn_sporadic_datasets.volcano

volcano_multiplot <- ipsc_mn_tardbp_datasets.volcano + ipsc_mn_fus_datasets.volcano + ipsc_mn_c9orf72_datasets.volcano + ipsc_mn_sod1_datasets.volcano + ipsc_mn_sporadic_datasets.volcano +
  plot_layout(ncol = 3, nrow = 2)
# volcano_multiplot



ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% ALS_genes]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% RNA_BINDING_PROTEINS]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% nucleocytoplasmic_genes]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% go.pathways.msigdb$GOBP_MITOCHONDRION_ORGANIZATION]


```


### Correlation heatmap

```{r}
# heatmap of correlation coefficients
ipsc_mn_mutations_stat = select(ipsc_mn_sporadic_datasets$res, gene_id, Sporadic = stat) %>% 
  left_join(select(ipsc_mn_c9orf72_datasets$res, gene_id, C9orf72 = stat)) %>%
  left_join(select(ipsc_mn_sod1_datasets$res, gene_id, SOD1 = stat)) %>%
  left_join(select(ipsc_mn_fus_datasets$res, gene_id, FUS = stat)) %>%
  left_join(select(ipsc_mn_tardbp_datasets$res, gene_id, TARDBP = stat)) %>%
  # left_join(select(ipsc_mn_vcp_datasets$res, gene_id, VCP = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_id")
cormat <- round(cor(ipsc_mn_mutations_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
dge_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=8)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5))
dge_correlation_ggheatmap

mutation_volcanos_corr_heatmap <- ipsc_mn_tardbp_datasets.volcano + ipsc_mn_fus_datasets.volcano + ipsc_mn_c9orf72_datasets.volcano + ipsc_mn_sod1_datasets.volcano + ipsc_mn_sporadic_datasets.volcano + dge_correlation_ggheatmap +
  plot_layout(ncol = 3, nrow = 2)
# mutation_volcanos_corr_heatmap

### Upset plot https://github.com/const-ae/ggupset
ipsc_mn_mutants_bind_datasets.res = bind_rows(mutate(ipsc_mn_sporadic_datasets$res, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$res, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$res, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$res, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$res, mutation = "FUS"))# ,mutate(ipsc_mn_vcp_datasets$res, mutation = "VCP"))
ipsc_mn_mutants_bind_datasets.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
  ggplot(aes(x = mutations)) + geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.2, hjust = 0.5, size = 2.4, angle = 0) +
    scale_x_upset(order_by = "degree")  + # n_intersections = 20
    scale_y_continuous(breaks = NULL, lim = c(0, 4000), name = "") +
    theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + labs(x = "", y = "") +
    theme_combmatrix(combmatrix.label.make_space = TRUE)
ipsc_mn_mutants_bind_datasets.res.upset

ipsc_mn_mutants_join_datasets.res %>% filter(padj.sod1 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # JAK1
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # PCDHB18P, NPIPA8


# # Overlap ALL DEGs
# ipsc_mn_mutants_bind_datasets.dir.res = bind_rows(mutate(filter(ipsc_mn_c9orf72_datasets$res, log2FoldChange > 0), mutation = "C9orf72 Up"), mutate(filter(ipsc_mn_c9orf72_datasets$res, log2FoldChange < 0), mutation = "C9orf72 Down"),
#                                               mutate(filter(ipsc_mn_tardbp_datasets$res, log2FoldChange > 0), mutation = "TARDBP Up"), mutate(filter(ipsc_mn_tardbp_datasets$res, log2FoldChange < 0), mutation = "TARDBP Down"),
#                                               mutate(filter(ipsc_mn_fus_datasets$res, log2FoldChange > 0), mutation = "FUS Up"), mutate(filter(ipsc_mn_fus_datasets$res, log2FoldChange < 0), mutation = "FUS Down"),
#                                               mutate(filter(ipsc_mn_sod1_datasets$res, log2FoldChange > 0), mutation = "SOD1 Up"), mutate(filter(ipsc_mn_sod1_datasets$res, log2FoldChange < 0), mutation = "SOD1 Down"),
#                                               mutate(filter(ipsc_mn_vcp_datasets$res, log2FoldChange > 0), mutation = "VCP Up"), mutate(filter(ipsc_mn_vcp_datasets$res, log2FoldChange < 0), mutation = "VCP Down"))
# ipsc_mn_mutants_bind_datasets.res.direction.upset = ipsc_mn_mutants_bind_datasets.dir.res %>% filter(padj < 0.05) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#   geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.4, hjust = -0.2, size = 2.5, angle = 45) +
#   scale_x_upset(order_by = "degree")  + # n_intersections = 20
#   scale_y_continuous(breaks = NULL, lim = c(0, 2700), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=6), axis.title=element_text(size=8)) +
#   ylab(expression()) +
#   theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.res.direction.upset
# 
# # Overlap UP DEGs
# ipsc_mn_mutants_bind_datasets.upregulated.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#       geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, size = 2.8) +
#     scale_x_upset(order_by = "degree")  + # n_intersections = 20
#     scale_y_continuous(breaks = NULL, lim = c(0, 2500), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + 
#       ylab(expression()) +
#     theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.upregulated.res.upset
# 
# # Overlap DOWN DEGs
# ipsc_mn_mutants_bind_datasets.downregulated.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#       geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, size = 2.8) +
#     scale_x_upset(order_by = "degree")  + # n_intersections = 20
#     scale_y_continuous(breaks = NULL, lim = c(0, 3000), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + 
#       ylab(expression()) +
#     theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.downregulated.res.upset

```

## GO terms

```{r}
# c9orf72
ipsc_mn_c9orf72_datasets.dge_genes <- ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt <- ipsc_mn_c9orf72_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " C9orf72 Down ", query == "query_2" ~ " C9orf72 Up "), term_name = as.factor(term_name))
ipsc_mn_c9orf72_datasets.dge_gprofiles_down <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ")
ipsc_mn_c9orf72_datasets.dge_gprofiles_up <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")
paste('"',unique(ipsc_mn_c9orf72_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"Activation of C3 and C5",
"hedgehog family protein binding",
"smoothened binding")
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_down <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_c9orf72_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"Cytokine Signaling in Immune system",
"intrinsic apoptotic signaling pathway",
"retrograde vesicle-mediated transport, Golgi to endoplasmic reticulum",
"p53 transcriptional gene network",
# "annulate lamellae",
# "miR-517 relationship with ARCN1 and USP1",
# "regulation of signal transduction by p53 class mediator",
"signal transduction by p53 class mediator",
# "hsa-miR-489-5p",
# "negative regulation of signal transduction by p53 class mediator",
"miRNA regulation of DNA damage response",
# "hsa-miR-1323",
# "hsa-miR-548o-3p",
# "miRNA regulation of p53 pathway in prostate cancer",
# "negative regulation of intrinsic apoptotic signaling pathway by p53 class mediator",
"DNA damage response",
"Genotoxicity pathway",
"intrinsic apoptotic signaling pathway by p53 class mediator",
# "regulation of intrinsic apoptotic signaling pathway by p53 class mediator",
"p53 signaling pathway")
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_up <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_down, ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" C9orf72 Up ", " C9orf72 Down ")))
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name <- gsub("retrograde vesicle-mediated transport, Golgi to endoplasmic reticulum", "vesicle-mediated transport", ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name <- gsub("intrinsic apoptotic signaling pathway by p53 class mediator", "apoptotic signaling pathway by p53", ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_c9orf72_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
ipsc_mn_c9orf72_datasets.go_curated

# fus
ipsc_mn_fus_datasets.dge_genes <- ipsc_mn_fus_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_fus_datasets.dge_gprofiles_filt <- ipsc_mn_fus_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " FUS Down ", query == "query_2" ~ " FUS Up "), term_name = as.factor(term_name))
ipsc_mn_fus_datasets.dge_gprofiles_down <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Down ")
ipsc_mn_fus_datasets.dge_gprofiles_up <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Up ")
paste('"',unique(ipsc_mn_fus_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"ubiquitin-dependent ERAD pathway",
"regulation of transport",
"protein localization to lysosome",
"endomembrane system",
"secretory vesicle",
"dendrite",
"somatodendritic compartment",
"intracellular vesicle",
"synaptic signaling")
ipsc_mn_fus_datasets.dge_gprofiles_filt_down <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_fus_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"spinal cord association neuron differentiation",
"histone modification",
"DNA binding",
"transcription regulator activity",
"double-stranded DNA binding",
"regulation of RNA metabolic process",
"sequence-specific DNA binding",
"transcription by RNA polymerase II",
"Gene expression (Transcription)")
ipsc_mn_fus_datasets.dge_gprofiles_filt_up <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_fus_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_fus_datasets.dge_gprofiles_filt_down, ipsc_mn_fus_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" FUS Up ", " FUS Down ")))
ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name <- gsub("spinal cord association neuron differentiation", "spinal cord neuron differentiation", ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name)
# ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" pathway", "", ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_fus_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_fus_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
ipsc_mn_fus_datasets.go_curated

# tardbp
ipsc_mn_tardbp_datasets.dge_genes <- ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_tardbp_datasets.dge_gprofiles_filt <- ipsc_mn_tardbp_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " TARDBP Down ", query == "query_2" ~ " TARDBP Up "), term_name = as.factor(term_name))
ipsc_mn_tardbp_datasets.dge_gprofiles_down <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Down ")
ipsc_mn_tardbp_datasets.dge_gprofiles_up <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Up ")
paste('"',unique(ipsc_mn_tardbp_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"regulation of RNA metabolic process",
"microtubule cytoskeleton",
"nervous system development",
"mitotic nuclear division",
"cell division",
"chromosome organization",
"mitotic cell cycle process",
"cell cycle process",
"nucleoplasm",
"cell cycle",
"protein binding")
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_down <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_tardbp_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"macroautophagy",
"MAPK signaling pathway",
"cytoskeleton",
"microtubule",
"protein binding",
"neuron development",
"voltage-gated channel activity",
"glutamatergic synapse",
"ion transport",
"dendrite",
"transport",
"axon",
"synaptic signaling")
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_up <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_tardbp_datasets.dge_gprofiles_filt_down, ipsc_mn_tardbp_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" TARDBP Up ", " TARDBP Down ")))
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub("regulation of RNA metabolic process", "RNA metabolic process", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
# ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" pathway", "", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
ipsc_mn_tardbp_datasets.go_curated

# # sod1
# ipsc_mn_sod1_datasets.dge_genes <- ipsc_mn_sod1_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_sod1_datasets.dge_gprofiles_filt <- ipsc_mn_sod1_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in sod1 ", query == "query_2" ~ " Up in sod1 "), term_name = as.factor(term_name))
# ipsc_mn_sod1_datasets.dge_gprofiles_down <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Down in sod1 ")
# ipsc_mn_sod1_datasets.dge_gprofiles_up <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Up in sod1 ")
# paste('"',unique(ipsc_mn_sod1_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# ipsc_mn_sod1_datasets.dge_gprofiles_filt_down <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Down in sod1 ") %>% filter(term_name %in% down_list)
# paste('"',unique(ipsc_mn_sod1_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# #no terms for SOD1
# 
# # vcp
# ipsc_mn_vcp_datasets.dge_genes <- ipsc_mn_vcp_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_vcp_datasets.dge_gprofiles_filt <- ipsc_mn_vcp_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA|WP")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in vcp ", query == "query_2" ~ " Up in vcp "), term_name = as.factor(term_name))
# ipsc_mn_vcp_datasets.dge_gprofiles_down <- ipsc_mn_vcp_datasets.dge_gprofiles_filt %>% filter(query == " Down in vcp ")
# ipsc_mn_vcp_datasets.dge_gprofiles_up <- ipsc_mn_vcp_datasets.dge_gprofiles_filt %>% filter(query == " Up in vcp ")
# paste('"',unique(ipsc_mn_vcp_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# paste('"',unique(ipsc_mn_vcp_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# # no terms for VCP
# 
# # sporadic
# ipsc_mn_sporadic_datasets.dge_genes <- ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_sporadic_datasets.dge_gprofiles_filt <- ipsc_mn_sporadic_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in sporadic ", query == "query_2" ~ " Up in sporadic "), term_name = as.factor(term_name))
# ipsc_mn_sporadic_datasets.dge_gprofiles_down <- ipsc_mn_sporadic_datasets.dge_gprofiles_filt %>% filter(query == " Down in sporadic ")
# ipsc_mn_sporadic_datasets.dge_gprofiles_up <- ipsc_mn_sporadic_datasets.dge_gprofiles_filt %>% filter(query == " Up in sporadic ")
# paste('"',unique(ipsc_mn_sporadic_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# paste('"',unique(ipsc_mn_sporadic_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# # none or sporadic

ipsc_mn_c9orf72_fus_tardbp.go_curated.merged = ipsc_mn_c9orf72_datasets.go_curated + ipsc_mn_fus_datasets.go_curated + ipsc_mn_tardbp_datasets.go_curated
ipsc_mn_c9orf72_fus_tardbp.go_curated.merged
# # fus
# ipsc_mn_fus_datasets.rrvgo = gprofiler2rrvgo(ipsc_mn_fus_datasets$res , query1 = "Down", query2 = "Up", GO_cat = "BP", sig.filter = "padj", species = "human")
# ipsc_mn_fus_datasets.rrvgo =  ipsc_mn_fus_datasets.rrvgo %>%  mutate(term = gsub("regulation of ", "", term), term = gsub("homophilic cell adhesion via plasma membrane adhesion molecules", "plasma membrane adhesion", term), parentTerm = term)
# ipsc_mn_fus_datasets.rrvgo.plot = rrvgo_plot(gprofiler2rrvgo = ipsc_mn_fus_datasets.rrvgo, n_terms = 10, remove_terms = NULL, colours = 2, cols =  c("firebrick2", "dodgerblue2"), label_angle = 0)
# 
# # tardbp
# ipsc_mn_tardbp_datasets.rrvgo = gprofiler2rrvgo(ipsc_mn_tardbp_datasets$res , query1 = "Down", query2 = "Up", GO_cat = "BP", sig.filter = "padj", species = "human")
# ipsc_mn_tardbp_datasets.rrvgo =  ipsc_mn_tardbp_datasets.rrvgo %>%  mutate(term = gsub("regulation of ", "", term), parentTerm = term)
# ipsc_mn_tardbp_datasets.rrvgo.plot = rrvgo_plot(gprofiler2rrvgo = ipsc_mn_tardbp_datasets.rrvgo, n_terms = 10, remove_terms = NULL, colours = 2, cols =  c("firebrick2", "dodgerblue2"), label_angle = 0)

# de_go_terms_mutations_merged = ggarrange(NULL,ipsc_mn_c9orf72_datasets.rrvgo.plot,NULL, ipsc_mn_sod1_datasets.rrvgo.plot,NULL, ipsc_mn_fus_datasets.rrvgo.plot,NULL, ipsc_mn_tardbp_datasets.rrvgo.plot, nrow = 8, ncol = 1, labels = c("C9orf72", "", "SOD1", "", "FUS", "", "TARDBP",""), heights = c(0.1,0.8,0.1,1.3,0.1,0.9,0.1,1.3)) #, ipsc_mn_vcp_datasets.rrvgo.plot) # no terms for VCP
# de_go_terms_mutations_merged
# ggsave(de_go_terms_mutations_merged, filename = here(proj_path,"expression/deseq2/figS5.de_go_terms_mutations_merged.png", height = 11.75, width = 8.25)

# which genes are driving the terms
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GABA_RECEPTOR_COMPLEX)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PLASMA_MEMBRANE_REGION)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GAP_JUNCTION)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT_CONFERRING_COMPRESSION_RESISTANCE)
```



## p53 PROGENy & TP53 Dorothea

```{r}
ipsc_mn_mutations_list = list("Sporadic" = select(ipsc_mn_sporadic_datasets$res,gene_name,stat), "C9orf72" = select(ipsc_mn_c9orf72_datasets$res,gene_name,stat), "TARDBP" = select(ipsc_mn_tardbp_datasets$res,gene_name,stat), "SOD1" = select(ipsc_mn_sod1_datasets$res,gene_name,stat), "FUS" = select(ipsc_mn_fus_datasets$res,gene_name,stat)) #, "VCP" = select(ipsc_mn_vcp_datasets$res,gene_name,stat)) 
mutations <- names(ipsc_mn_mutations_list)
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{drop_na(.x)}) 
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

# ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP","VCP")))) +
#   geom_bar(stat='identity', position = "dodge2") +
#   scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
#   theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
#   geom_hline(yintercept = 0, linetype = 3) +
#   labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment") 
# ipsc_mn_mutations_list.progeny_scores.plot

ipsc_mn_mutations_list.p53.progeny_scores = ipsc_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
ipsc_mn_mutations_list.p53.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.p53.progeny_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
  stat_pvalue_manual(ipsc_mn_mutations_list.p53.progeny_scores, label = "p.signif", y.position = 12, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_mutations_list.p53.progeny_scores.plot



# Dorothea
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

ipsc_mn_mutations_list.TP53.dorothea_scores = ipsc_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
ipsc_mn_mutations_list.TP53.dorothea_scores.plot <- ggplot(ipsc_mn_mutations_list.TP53.dorothea_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") + #ylim(c(-11,11)) +
  stat_pvalue_manual(ipsc_mn_mutations_list.TP53.dorothea_scores, label = "p.signif", y.position = 10, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_mutations_list.TP53.dorothea_scores.plot
```


## Merge

```{r}
mutations_compared_fig.merged = ggarrange(
  ggarrange(ipsc_mn_tardbp_datasets.volcano, ipsc_mn_fus_datasets.volcano, ipsc_mn_c9orf72_datasets.volcano, ipsc_mn_sod1_datasets.volcano, ipsc_mn_sporadic_datasets.volcano, dge_correlation_ggheatmap, ncol = 3, nrow = 2, labels = c("a","b","c","d","e","f")),
  ggarrange(ipsc_mn_c9orf72_datasets.go_curated, ipsc_mn_fus_datasets.go_curated, ipsc_mn_tardbp_datasets.go_curated,ncol=3,labels = c("g","h","i")), # GO plots for each background
  ggarrange(ipsc_mn_mutations_list.p53.progeny_scores.plot, ipsc_mn_mutations_list.TP53.dorothea_scores.plot, ncol=2, labels = c("j","k")),
  nrow = 3, heights = c(1.5,1,0.8))
# mutations_compared_fig.merged
ggsave(mutations_compared_fig.merged, filename = here(proj_path,"expression/deseq2/mutations_compared_fig.merged.png"), height = 13, width = 10)



mutations_compared = ggarrange(
  mutation_volcanos_corr_heatmap, 
  ggarrange(dge_correlation_ggheatmap,ipsc_mn_mutants_bind_datasets.res.upset, ncol=2, labels = c("b","c"), widths = c(0.6,1)),
  ipsc_mn_c9orf72_fus_tardbp.go_curated.merged, # GO plots for each background
  ggarrange(ipsc_mn_mutations_list.p53.progeny_scores.plot, ipsc_mn_mutations_list.TP53.dorothea_scores.plot, ncol=2, labels = c("e","f")),
  nrow = 4, heights = c(1.4,0.6,1,0.8), labels = c("a","","d",""))
# mutations_compared
ggsave(mutations_compared, filename = here(proj_path,"expression/deseq2/mutations_compared.png"), height = 15, width = 10)

```


# Fig S11 Gene expression mutations scatters

Scatter correlation of mutants

```{r}
ipsc_mn_mutations_list = list("Sporadic" = ipsc_mn_sporadic_datasets$res, "C9orf72" = ipsc_mn_c9orf72_datasets$res, "TARDBP" = ipsc_mn_tardbp_datasets$res, "SOD1" = ipsc_mn_sod1_datasets$res, "FUS" = ipsc_mn_fus_datasets$res)#, "VCP" = ipsc_mn_vcp_datasets$res)
# mutations <- names(ipsc_mn_mutations_list)
# 
# ipsc_mn_mutations_list = purrr::map2(
#       .x = ipsc_mn_mutations_list,
#       .y = names(ipsc_mn_mutations_list), ~{
#         .x$mutation <- .y
#         return(.x)
#       })

t_stat_plot <- 
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", highlight = "C9orf72")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", highlight = "SOD1")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", highlight = "FUS")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", highlight = "TARDBP")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "VCP", highlight = "VCP")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", highlight = c("C9orf72","SOD1"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", highlight = c("C9orf72","FUS"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", highlight = c("C9orf72","TARDBP"))  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "VCP", highlight = c("C9orf72","VCP"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", highlight = c("SOD1","FUS"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", highlight = c("SOD1","TARDBP"))  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "VCP", highlight = c("SOD1","VCP"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", highlight = c("TARDBP","FUS"))  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "VCP", highlight = c("VCP","FUS"))  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "TARDBP", mutation2 = "VCP", highlight = c("TARDBP","VCP")) + 
  plot_layout(ncol = 5, nrow = 2) &  
  xlim(-8,8) & ylim(-8,8)
t_stat_plot
ggsave(t_stat_plot, filename = here(proj_path,"expression/deseq2/dge.mutation.stat.scatters.merged.png"), height = 7, width = 10)

# log2FoldChange
lfc_plot <- 
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "VCP", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "VCP", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "VCP", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "VCP", col = "log2FoldChange")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "TARDBP", mutation2 = "VCP", col = "log2FoldChange") &  
  xlim(-8,8) & ylim(-8,8)  +
  plot_layout(ncol = 3, nrow = 5)
lfc_plot
ggsave(lfc_plot, filename = here(proj_path,"expression/deseq2/dge.mutation.lfc.scatters.merged.png", height = 13, width = 10)

```


# Fig S12 DoRoTHEA & PROGENy

<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md
Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
ipsc_mn_mutations_list = list("Sporadic" = select(ipsc_mn_sporadic_datasets$res,gene_name,stat), "C9orf72" = select(ipsc_mn_c9orf72_datasets$res,gene_name,stat), "TARDBP" = select(ipsc_mn_tardbp_datasets$res,gene_name,stat), "SOD1" = select(ipsc_mn_sod1_datasets$res,gene_name,stat), "FUS" = select(ipsc_mn_fus_datasets$res,gene_name,stat))#, "VCP" = select(ipsc_mn_vcp_datasets$res,gene_name,stat)) 
mutations <- names(ipsc_mn_mutations_list)
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{drop_na(.x)}) 
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, source = as.factor(source), source = fct_reorder(source, score))

ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP","VCP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment")# +
  # stat_pvalue_manual(ipsc_mn_mutations_list.progeny_scores, label = "p.signif", y.position = 12, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_mutations_list.progeny_scores.plot


ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=mutation, y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.position = "top") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment") +
  facet_grid(~source) +
  stat_pvalue_manual(ipsc_mn_mutations_list.progeny_scores, label = "p.signif", y.position = 12, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE)
ipsc_mn_mutations_list.progeny_scores.plot


```

Dorothea

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation) %>% 
  mutate(sig = case_when(p_value < 0.1 & abs(score) < 4 ~ "sig", p_value < 0.1 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.1 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

# Volcano of TFs score vs p_value
ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_mutations_list.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", subtitle = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
    geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "C9orf72"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "FUS"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "SOD1"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "Sporadic"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "TARDBP"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "VCP"),n = 5, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
    geom_text_repel(fontface = "italic", data = filter(ipsc_mn_mutations_list.dorothea_scores,source %in% c("TP53","E2F4","EOMES","MAZ","MITF","SOX2","TFDP1","ZBTB33","ZNF263")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  facet_wrap(~mutation, scales = "free") & ylim(0,2)
ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano


# commonly changed TFs
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05) %>% group_by(mutation) %>% count(source)
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05) %>% count(source) %>% arrange(-n)
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score > 0) %>% count(source) %>% arrange(-n)
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score < 0) %>% count(source) %>% arrange(-n)


ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source > 0) %>% count(source) %>% arrange(-n)
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source < 0) %>% count(source) %>% arrange(-n)

ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "MAZ")

```

Merge

```{r}
ipsc_mn_mutations_list.dorothea_progeny_merged = ggarrange(ipsc_mn_mutations_list.progeny_scores.plot, ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano, nrow = 2, labels = c("a","b"), heights = c(1,1.7))
# ipsc_mn_mutations_list.dorothea_progeny_merged
ggsave(ipsc_mn_mutations_list.dorothea_progeny_merged, filename = here(proj_path, "expression/deseq2/ipsc_mn_mutations_list.dorothea_progeny_merged.png"), height = 11.75, width = 8.25)

```



# Table S5 DEGs overlap between genetic backgrounds

```{r}
ipsc_mn_mutants_join_datasets.res = 
  mutate(select(ipsc_mn_sporadic_datasets$res, gene_name, gene_id, padj.sporadic = padj, stat.sporadic = stat, lfc.sporadic = log2FoldChange), direction.sporadic = case_when(padj.sporadic < 0.05 & lfc.sporadic > 0 ~ "up", padj.sporadic < 0.05 & lfc.sporadic < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(ipsc_mn_c9orf72_datasets$res, gene_name, gene_id, padj.c9orf72 = padj, stat.c9orf72 = stat, lfc.c9orf72 = log2FoldChange), direction.c9orf72 = case_when(padj.c9orf72 < 0.05 & lfc.c9orf72 > 0 ~ "up", padj.c9orf72 < 0.05 & lfc.c9orf72 < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_fus_datasets$res, gene_name, gene_id, padj.fus = padj, stat.fus = stat, lfc.fus = log2FoldChange), direction.fus = case_when(padj.fus < 0.05 & lfc.fus > 0 ~ "up", padj.fus < 0.05 & lfc.fus < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_tardbp_datasets$res, gene_name, gene_id,  padj.tardbp = padj, stat.tardbp = stat, lfc.tardbp = log2FoldChange), direction.tardbp = case_when(padj.tardbp < 0.05 & lfc.tardbp > 0 ~ "up", padj.tardbp < 0.05 & lfc.tardbp < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_sod1_datasets$res, gene_name, gene_id, padj.sod1 = padj, stat.sod1 = stat, lfc.sod1 = log2FoldChange), direction.sod1 = case_when(padj.sod1 < 0.05 & lfc.sod1 > 0 ~ "up", padj.sod1 < 0.05 & lfc.sod1 < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  # full_join(mutate(select(ipsc_mn_vcp_datasets$res, gene_name, gene_id, padj.vcp = padj, stat.vcp = stat, lfc.vcp = log2FoldChange), direction.vcp = case_when(padj.vcp < 0.05 & lfc.vcp > 0 ~ "up", padj.vcp < 0.05 & lfc.vcp < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  # arrange() %>%
  select(gene_name, gene_id, everything())
ipsc_mn_mutants_join_datasets.res
write_csv(ipsc_mn_mutants_join_datasets.res, here(proj_path,"expression/deseq2/TableS5.overlap_als_individual_datasets.csv"))

```


# Fig S12 Compare genetic subgroups

## sALS vs fALS

Scatters and volcanos

```{r}
ipsc_mn_familial_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_familial_datasets.rds"))
ipsc_mn_mutant_sporadic = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_mutant_sporadic.rds"))
ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
# ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
# ipsc_mn_c9orf72_sporadic = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_sporadic.rds"))
ipsc_mn_tdp43_positive_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tdp43_positive_datasets.rds"))
ipsc_mn_tdp43_negative_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tdp43_negative_datasets.rds"))
ipsc_mn_tdp_pathology = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tdp_pathology.rds"))

# mutant_vs_sporadic
ipsc_mn_familial_datasets$res %>% filter(padj < 0.05) %>% pull(gene_name)
ipsc_mn_sporadic_familial_list = list("Sporadic" = ipsc_mn_sporadic_datasets$res, "Mutant" = ipsc_mn_familial_datasets$res)
ipsc_mn_sporadic_familial_stat.corr = plot_de_correlation(model_list1 = ipsc_mn_sporadic_familial_list, model_list2 = ipsc_mn_sporadic_familial_list, mutation1 = "Sporadic", mutation2 = "Mutant", highlight = pull(filter(ipsc_mn_mutant_sporadic$res,padj < 0.05),gene_name))
ipsc_mn_sporadic_familial_stat.corr

ipsc_mn_mutant_sporadic.metadata %>% count(mutant_sporadic)
ipsc_mn_mutant_sporadic$res %>% filter(padj < 0.05) #24
ipsc_mn_mutant_sporadic.als.labels <- ipsc_mn_mutant_sporadic$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_mutant_sporadic.gene.labels <- ipsc_mn_mutant_sporadic$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_mutant_sporadic.volcano <- volcano_plot(ipsc_mn_mutant_sporadic$res, numerator = "mutant", denomenator = "sporadic",
               subtitle = "208 sporadic vs 110 mutant", ymax = 8, xmax = 2, dpi = 300,
               gene_labels = c(ipsc_mn_mutant_sporadic.als.labels, ipsc_mn_mutant_sporadic.gene.labels))
ipsc_mn_mutant_sporadic.volcano

ipsc_mn_mutant_sporadic.dge_genes <- ipsc_mn_mutant_sporadic$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost() # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
ipsc_mn_mutant_sporadic.dge_genes$result
#    query significant    p_value term_size query_size intersection_size precision      recall    term_id source                               term_name effective_domain_size source_order       parents
# 1 query_1        TRUE 0.04967062         2          2                 1       0.5 0.500000000 CORUM:5389  CORUM                   SERPINA3-CTSG complex                  3627         1666 CORUM:0000000
# 2 query_1        TRUE 0.04983033         1          1                 1       1.0 1.000000000 HP:0011285     HP      Long-segment aganglionic megacolon                  4667         7727    HP:0002251
# 3 query_1        TRUE 0.04810003       352          2                 2       1.0 0.005681818 KEGG:04080   KEGG Neuroactive ligand-receptor interaction                  8014          244    KEGG:00000
ipsc_mn_mutant_sporadic$res %>% filter(padj < 0.05, gene_name %in% kegg.pathways.msigdb$KEGG_NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION) # BRS3, BDKRB2

```

## TDP43 vs non TDP43 status

```{r}
# TDP-43 pathology vs no TDP-43 pathology (SOD1,FUS) tdp_pathology_yes_vs_no
ipsc_mn_tdp43_positive_negative_list = list("TDP-43 positive" = ipsc_mn_tdp43_positive_datasets$res, "TDP-43 negative" = ipsc_mn_tdp43_negative_datasets$res)
ipsc_mn_tdp43_positive_negative_stat.corr = plot_de_correlation(model_list1 = ipsc_mn_tdp43_positive_negative_list, model_list2 = ipsc_mn_tdp43_positive_negative_list, mutation1 = "TDP-43 positive", mutation2 = "TDP-43 negative", 
                                                          highlight = pull(filter(ipsc_mn_tdp_pathology$res,padj < 0.05),gene_name))
ipsc_mn_tdp43_positive_negative_stat.corr

ipsc_mn_tdp_pathology.metadata %>% count(tdp_pathology)
ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05) #108
ipsc_mn_tdp_pathology.als.labels <- ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, go.pathways.msigdb$GO_CELL_CELL_ADHESION)) %>% distinct(gene_name) %>% pull(gene_name)
# ipsc_mn_tdp_pathology.gene.labels <- ipsc_mn_tdp_pathology$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_tdp_pathology.volcano <- volcano_plot(ipsc_mn_tdp_pathology$res, numerator = "TDP-43 pathology", denomenator = "SOD1 and FUS ALS",
               subtitle = "29 SOD1 and FUS vs 283 TDP-43 pathology", ymax = 8, xmax = 2, dpi = 300,
               gene_labels = c(ipsc_mn_tdp_pathology.als.labels, "TARDBP","SOD1","FUS","SNRPE","KCNA6"))
ipsc_mn_tdp_pathology.volcano

ipsc_mn_tdp_pathology$res %>% filter(gene_name %in% c("TARDBP","SOD1","FUS")) #29
# gene_id         baseMean log2FoldChange  lfcSE   stat pvalue  padj gene_name
#   <chr>              <dbl>          <dbl>  <dbl>  <dbl>  <dbl> <dbl> <chr>    
# 1 ENSG00000120948    4118.       -0.0575  0.0493 -1.17   0.243 0.815 TARDBP   
# 2 ENSG00000089280    4994.       -0.0894  0.106  -0.840  0.401 0.883 FUS      
# 3 ENSG00000142168    2772.        0.00621 0.0545  0.114  0.909 0.991 SOD1     
ipsc_mn_tdp_pathology.dge_genes <- ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost() # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
ipsc_mn_tdp_pathology.dge_genes$result
#   query significant      p_value term_size query_size intersection_size precision      recall    term_id source                                                       term_name effective_domain_size
# 1  query_1        TRUE 1.351476e-02         7          8                 2 0.2500000 0.285714286  CORUM:614  CORUM   NRD complex (Nucleosome remodeling and deacetylation complex)                  3627
# 2  query_1        TRUE 1.351476e-02         7          8                 2 0.2500000 0.285714286 CORUM:1150  CORUM                                            Histone H3.3 complex                  3627
# 3  query_1        TRUE 1.799979e-02         8          8                 2 0.2500000 0.250000000 CORUM:1149  CORUM                                            Histone H3.1 complex                  3627
# 4  query_1        TRUE 2.311704e-02         9          8                 2 0.2500000 0.222222222  CORUM:646  CORUM                                HDAC1-associated protein complex                  3627
# 5  query_1        TRUE 3.791370e-03       212         22                 5 0.2272727 0.023584906 GO:0006333  GO:BP                               chromatin assembly or disassembly                 21029
# 6  query_1        TRUE 9.704919e-03        34         22                 3 0.1363636 0.088235294 GO:0034724  GO:BP              DNA replication-independent chromatin organization                 21029
# 7  query_1        TRUE 3.915810e-03       158         23                 4 0.1739130 0.025316456 GO:0005685  GO:CC                                                        U1 snRNP                 21672
# 8  query_1        TRUE 3.579775e-02      2671         23                10 0.4347826 0.003743916 GO:0140513  GO:CC                              nuclear protein-containing complex                 21672
# 9  query_2        TRUE 4.994746e-02         2          1                 1 1.0000000 0.500000000 CORUM:7236  CORUM                                               ARXN1-CIC complex                  3627
# 10 query_2        TRUE 9.148598e-07       168         14                 6 0.4285714 0.035714286 GO:0007156  GO:BP homophilic cell adhesion via plasma membrane adhesion molecules                 21029
# 11 query_2        TRUE 1.838016e-05       277         14                 6 0.4285714 0.021660650 GO:0098742  GO:BP       cell-cell adhesion via plasma-membrane adhesion molecules                 21029
# 12 query_2        TRUE 8.360600e-04       888         14                 7 0.5000000 0.007882883 GO:0098609  GO:BP                                              cell-cell adhesion                 21029
# 13 query_2        TRUE 2.517310e-02      1480         14                 7 0.5000000 0.004729730 GO:0007155  GO:BP                                                   cell adhesion                 21029
# 14 query_2        TRUE 2.596410e-02      1487         14                 7 0.5000000 0.004707465 GO:0022610  GO:BP                                             biological adhesion                 21029
# 15 query_2        TRUE 5.290213e-04      1630         15                 8 0.5333333 0.004907975 GO:0005887  GO:CC                           integral component of plasma membrane                 21672
# 16 query_2        TRUE 7.679771e-04      1713         15                 8 0.5333333 0.004670169 GO:0031226  GO:CC                          intrinsic component of plasma membrane                 21672
# 17 query_2        TRUE 4.992555e-05       724         14                 7 0.5000000 0.009668508 GO:0005509  GO:MF                                             calcium ion binding                 20166
# 18 query_2        TRUE 3.839566e-04        74         15                 4 0.2666667 0.054054054  TF:M06624     TF                               Factor: ZNF2; motif: NSATCGATCCGA                 19959
ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_CHROMATIN_ASSEMBLY_OR_DISASSEMBLY) # CENPV, NASP, RBBP4, SUPT16H
ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_U1_SNRNP) # SNRPE
ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_CELL_CELL_ADHESION) # HLX, 6 PCDH genes





# # C9orf72 vs Sporadic
# ipsc_mn_sporadic_c9orf72_list = list("Sporadic" = ipsc_mn_sporadic_datasets$res, "C9orf72" = ipsc_mn_c9orf72_datasets$res)
# ipsc_mn_sporadic_c9orf72_stat.corr = plot_de_correlation(model_list1 = ipsc_mn_sporadic_c9orf72_list, model_list2 = ipsc_mn_sporadic_c9orf72_list, mutation1 = "Sporadic", mutation2 = "C9orf72", highlight = pull(filter(ipsc_mn_mutant_sporadic$res,padj < 0.05),gene_name))
# ipsc_mn_sporadic_c9orf72_stat.corr
# 
# ipsc_mn_c9orf72_sporadic$res %>% filter(padj < 0.05) #27
# ipsc_mn_c9orf72_sporadic.als.labels <- ipsc_mn_c9orf72_sporadic$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS)) %>% distinct(gene_name) %>% pull(gene_name)
# ipsc_mn_c9orf72_sporadic.gene.labels <- ipsc_mn_c9orf72_sporadic$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
# ipsc_mn_c9orf72_sporadic.volcano <- volcano_plot(ipsc_mn_c9orf72_sporadic$res, numerator = "C9orf72", denomenator = "sALS",
#                subtitle = "208 sporadic vs 51 C9orf72", ymax = 8, xmax = 2, dpi = 300,
#                gene_labels = c(ipsc_mn_c9orf72_sporadic.als.labels, ipsc_mn_c9orf72_sporadic.gene.labels, "C9orf72"))
# ipsc_mn_c9orf72_sporadic.volcano
# ipsc_mn_c9orf72_sporadic$res %>% filter(gene_name == "C9orf72") #29
# # gene_id         baseMean log2FoldChange  lfcSE  stat   pvalue  padj gene_name
# #   <chr>              <dbl>          <dbl>  <dbl> <dbl>    <dbl> <dbl> <chr>    
# # 1 ENSG00000147894    2545.         -0.202 0.0597 -3.38 0.000724 0.157 C9orf72  

```



## Merge

```{r}
genetic_subgroups_compared.merged = ggarrange(ipsc_mn_sporadic_familial_stat.corr, ipsc_mn_mutant_sporadic.volcano,
                                              ipsc_mn_tdp43_positive_negative_stat.corr, ipsc_mn_tdp_pathology.volcano,
                                              nrow = 2, ncol = 2, labels = c("a","b","c","d","e",""))
# genetic_subgroups_compared.merged
ggsave(genetic_subgroups_compared.merged, filename = here(proj_path,"expression/deseq2/genetic_subgroups_compared.merged.png"), height = 15, width = 10)

```


# Fig 4 Postmortem 

Import NYGC counts from Prudencio et al 2020 using RSEM
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE153960 

## Volcano

```{r}
#### Spinal cord only ###
nygc_postmortem_spinal_cord.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.tissue.rds"))
nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) # 17,837
nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0)

## Volcano
nygc_postmortem_spinal_cord.als_vs_ctrl.als.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(gene_name %in% ALS_genes) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.p53.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.ipsn.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj<0.05,gene_name %in% pull(filter(ipsc_mn_als_datasets$res,padj<0.05),gene_name)) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.volcano <- volcano_plot(nygc_postmortem_spinal_cord.als_vs_ctrl$res, numerator = "ALS", denomenator = "CTRL",
               title = "Postmortem spinal cord", subtitle = "56 Controls vs 206 ALS", ymax = 80, xmax = 3, dpi = 300,
               gene_labels = c(nygc_postmortem_spinal_cord.als_vs_ctrl.als.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.ipsn.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.p53.labels,"CFAP410"))
nygc_postmortem_spinal_cord.als_vs_ctrl.volcano

```

## GO

```{r}
nygc_postmortem_spinal_cord.als_vs_ctrl.gost <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles <- nygc_postmortem_spinal_cord.als_vs_ctrl.gost$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_down <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Down ")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_up <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"axon",
"synapse",
"neuron development",
"cytoskeleton",
"neuron projection",
"neuron differentiation",
# "transcription, DNA-templated",
"regulation of RNA metabolic process",
"generation of neurons",
"cell projection organization",
"neurogenesis",
"cell projection",
"nervous system development")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_down <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
# "G1/S DNA Damage Checkpoints",
# "Proteasome degradation",
# "unfolded protein binding",
# "regulation of response to DNA damage stimulus",
# "p53-Independent DNA Damage Response",
# "signal transduction in response to DNA damage",
"p53-Dependent G1/S DNA damage checkpoint",
# "intrinsic apoptotic signaling pathway in response to DNA damage",
"cellular response to DNA damage stimulus",
"Metabolism of RNA",
"programmed cell death",
# "cell death",
"endoplasmic reticulum",
"regulation of immune system process",
"regulation of response to stimulus",
# "regulation of response to stress",
"translation",
"peptide metabolic process",
"response to stress")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_up <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined <- bind_rows(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_down, nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined = nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined %>% mutate(term_name = gsub("regulation of ", "", term_name), term_name = gsub("and","&",term_name))
nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated <- curated_profiles(gprofiles = nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined)# + theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated

```


## PROGENY & Dorothea

```{r}
# nygc_postmortem_spinal_cord.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.tissue.rds"))

# PROGENy & DOROTHEA
net_progeny <- get_progeny(organism = 'human', top = 100)
nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity <- ggplot(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment ALS vs CTRL") +
    stat_pvalue_manual(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 6, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts <- run_wmean(mat=nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(sig = case_when(p_value < 0.1 & abs(score) < 4 ~ "sig", p_value < 0.1 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.1 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.up = nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.down = nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano = ggplot(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts, source %in% c(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.up, nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted')
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano

```


### Compare mutations postmortem

```{r}
nygc_postmortem_spinal_cord.c9orf72_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.c9orf72_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.tardbp_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.tardbp_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.fus_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.fus_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sod1_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sod1_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sporadic_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sporadic_vs_ctrl.rds"))


nygc_postmortem_spinal_cord_mn_mutations_list = list("Sporadic" = select(nygc_postmortem_spinal_cord.sporadic_vs_ctrl$res,gene_name,stat), "C9orf72" = select(nygc_postmortem_spinal_cord.c9orf72_vs_ctrl$res,gene_name,stat), "TARDBP" = select(nygc_postmortem_spinal_cord.tardbp_vs_ctrl$res,gene_name,stat), "SOD1" = select(nygc_postmortem_spinal_cord.sod1_vs_ctrl$res,gene_name,stat), "FUS" = select(nygc_postmortem_spinal_cord.fus_vs_ctrl$res,gene_name,stat)) 
mutations <- names(nygc_postmortem_spinal_cord_mn_mutations_list)
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{drop_na(.x)}) 
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores = map_df(nygc_postmortem_spinal_cord_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment")
nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores.plot

nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores = nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
  stat_pvalue_manual(nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores, label = "p.signif", y.position = 6, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) 
nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot



# Dorothea
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord_mn_mutations_list.dorothea_scores = map_df(nygc_postmortem_spinal_cord_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores = nygc_postmortem_spinal_cord_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") + #ylim(c(-6,6)) +
  stat_pvalue_manual(nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores, label = "p.signif", y.position = 6, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) 
nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot

```

## Scatter postmortem against iPSN

```{r}
# correlate to meta
postmortem_ipsn.als_vs_ctrl = select(nygc_postmortem_spinal_cord.als_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_als_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.als_vs_ctrl.overlapping = postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name) # 50
postmortem_ipsn_de_list = list("Postmortem spinal cord: ALS vs CTRL" = nygc_postmortem_spinal_cord.als_vs_ctrl$res, "iPSN: ALS vs CTRL" = ipsc_mn_als_datasets$res)
postmortem_ipsn.als_vs_ctrl.scatter = plot_de_correlation(model_list1 = postmortem_ipsn_de_list, model_list2 = postmortem_ipsn_de_list, mutation1 = "iPSN: ALS vs CTRL", mutation2 = "Postmortem spinal cord: ALS vs CTRL", highlight = c(postmortem_ipsn.als_vs_ctrl.overlapping, "CFAP410"))
postmortem_ipsn.als_vs_ctrl.scatter

postmortem_ipsn.als_vs_ctrl.overlapping[postmortem_ipsn.als_vs_ctrl.overlapping %in% ALS_genes] # CFAP410
postmortem_ipsn.als_vs_ctrl.overlapping[postmortem_ipsn.als_vs_ctrl.overlapping %in% RNA_BINDING_PROTEINS] # "RNASEL" "DDX18"  "MRPL18" "EEFSEC"
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% RNA_BINDING_PROTEINS) 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% gprofiler_database$`DNA Damage Response`) 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% RNA_BINDING_PROTEINS  ) %>% arrange(-nygc.stat + ipsn.stat)
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% group_split(nygc.stat > 0)

```



## Merge

```{r}
nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged = ggarrange(
  ggarrange(nygc_postmortem_spinal_cord.als_vs_ctrl.volcano, nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated, ncol=2, labels = c("a","b"), widths = c(1,0.8)),
  ggarrange(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity, nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano, ncol=2, labels = c("c","d")),
  ggarrange(nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot, nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot, postmortem_ipsn.als_vs_ctrl.scatter, ncol = 3, labels = c("e","f","g"), widths = c(1,1,2)),
  nrow = 3)
# nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged
ggsave(nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged, filename = here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged.png"), height = 12, width = 9)

# dorothea_progeny_merged = ggarrange(ipsc_mn_als_progeny.pathwayActivity, mn.ctrl_vcp_nuc_vs_cyt.DoRothEA.scatter, nrow = 2, labels = c("a","b"))
# dorothea_progeny_merged
# ggsave(dorothea_progeny_merged, filename = here(proj_path,"expression/deseq2/dorothea_progeny_merged.png", height = 9, width = 8.25)

```


# Table S6 postmortem & ipsn overlap DGE

```{r}
postmortem_ipsc_overlap.res = 
  mutate(select(nygc_postmortem_spinal_cord.als_vs_ctrl$res, gene_name, gene_id, padj.postmortem = padj, stat.postmortem = stat, lfc.postmortem = log2FoldChange), direction.postmortem = case_when(padj.postmortem < 0.05 & lfc.postmortem > 0 ~ "up", padj.postmortem < 0.05 & lfc.postmortem < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(ipsc_mn_als_datasets$res, gene_name, gene_id, padj.ipsn = padj, stat.ipsn = stat, lfc.ipsn = log2FoldChange), direction.ipsn = case_when(padj.ipsn < 0.05 & lfc.ipsn > 0 ~ "up", padj.ipsn < 0.05 & lfc.ipsn < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  select(gene_name, gene_id, everything()) %>% 
  filter(padj.ipsn < 0.05, padj.postmortem < 0.05, ((stat.ipsn > 0 & stat.postmortem > 0) | (stat.ipsn < 0 & stat.postmortem < 0)))
postmortem_ipsc_overlap.res
write_csv(postmortem_ipsc_overlap.res, here(proj_path,"expression/deseq2/TableS6.postmortem_ipsc_overlap.csv"))

```


# Fig 5 Splicing pan ALS

## MAJIQ 

```{r}
als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
# als_ctrl.voila.rev = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/als_vs_ctrl.tsv", grp1 = "ctrl", grp2 = "als")
# majiq.als_vs_ctrl.volcano.rev <- majiq_het_volcano_plot(als_ctrl.voila.rev, "Alternative splicing: pan ALS", numerator = "ALS", denomenator = "CTRL", subtitle = "102 CTRL vs 312 ALS", ymax = 10, xmax = 0.35, dpi = 300, gene_labels = NULL)
# majiq.als_vs_ctrl.volcano.rev
als_ctrl.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq/modulize/als_ctrl"), skip_rows = 9)

als_ctrl.voila %>% glimpse #225,154
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 67
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # INPP5F
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 631
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 356
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) #14
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) # 9
#"ATXN2"   "NOP56"   "TIA1"    "INPP5F"  "INPP5F"  "SLC9A8"  "C9orf72" "C9orf72" "TBP"     "GLT8D1"  "GLT8D1"  "CFAP410" "CFAP410" "CFAP410"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% sporadic_ALS_genes) #INPP5F
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_GWAS) #SLC9A8, c9orf72, CFAP410
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_EWAS) #
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_modifiers) #
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 144/631
144/631
als_ctrl.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer) #25 cryptic exons
als_ctrl.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi))  # FLNB, SDHAP1, SEC31B
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 459/631
459/631
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "ALG13") %>% glimpse #ENSG00000101901:s:111736714-111736933:j:111736879-111744668
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.signalling.pathway) %>% glimpse # HDAC1 ENSG00000116478:s:32329068-32329160:j:32329160-32330578
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`DNA Damage Response`) #0


## Volcano
ipsc_mn_als_datasets.metadata %>% count(condition) #318 ALS, 101 ctrl
majiq.als_vs_ctrl.RBP.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.ALS.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes))  %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.DDR.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.signalling.pathway)  %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)

majiq.als_vs_ctrl.volcano <- majiq_het_volcano_plot(als_ctrl.voila, "Alternative splicing: pan ALS", numerator = "ALS", denomenator = "CTRL", subtitle = "101 CTRL vs 318 ALS", ymax = 7, xmax = 0.3, dpi = 300, junction_labels = c(majiq.als_vs_ctrl.ALS.junction_labels, majiq.als_vs_ctrl.RBP.junction_labels,majiq.als_vs_ctrl.DDR.junction_labels))
# majiq.als_vs_ctrl.volcano

# majiq.als_vs_ctrl.volcano <- majiq_het_volcano_plot(als_ctrl.voila, "Alternative splicing: pan ALS", significance_value = "wilcoxon", significance_threshold = "wilcoxon", numerator = "ALS", denomenator = "CTRL", subtitle = "102 CTRL vs 312 ALS", ymax = 10, xmax = 0.35, dpi = 30, gene_labels = c(majiq.als_vs_ctrl.ALS.labels, majiq.als_vs_ctrl.RBP.labels))
# majiq.als_vs_ctrl.volcano
# majiq_het_volcano_plot(als_ctrl.voila, "MAJIQ", numerator = "ALS", denomenator = "CTRL", subtitle = "102 CTRL vs 312 ALS", ymax = 10, xmax = 0.35, dpi = 30, gene_labels = c(majiq.als_vs_ctrl.ALS.labels, majiq.als_vs_ctrl.RBP.labels), significance_value = "wilcoxon", significance_threshold = "wilcoxon")
# majiq_het_volcano_plot(als_ctrl.voila, "MAJIQ", numerator = "ALS", denomenator = "CTRL", subtitle = "102 CTRL vs 312 ALS", ymax = 10, xmax = 0.35, dpi = 30, gene_labels = c(majiq.als_vs_ctrl.ALS.labels, majiq.als_vs_ctrl.RBP.labels), significance_value = "ttest", significance_threshold = "ttest")

# GO of splicing events
 als_ctrl.voila.gost = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
als_ctrl.voila.gost_filt <- als_ctrl.voila.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC")) %>% arrange( -log10(p_value) )
table(als_ctrl.voila.gost_filt$query) # 66
paste('"',unique(als_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
# "neuron projection development",
"neuron development",
"actin cytoskeleton organization",
"cytoskeletal protein binding",
"localization",
"Spliceosome",
"neuron projection",
"cytoskeleton organization",
"cytosol",
"microtubule organizing center",
"protein binding",
"cell adhesion molecule binding",
"axon",
"neuron development",
"neuron differentiation",
"nuclear speck")
als_ctrl.voila.gost_filt.terms <- als_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
als_ctrl.voila.gost_filt.terms$term_name = gsub(" molecule binding| organization| organizing", "", als_ctrl.voila.gost_filt.terms$term_name)
als_ctrl.voila.gost_filt.terms.curated <- curated_profiles(als_ctrl.voila.gost_filt.terms, cols = c("firebrick2", "dodgerblue2")) + theme(strip.text.y.right = element_blank())# + theme(legend.position = "none", axis.text = element_text(size = 10))
als_ctrl.voila.gost_filt.terms.curated

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_NUCLEAR_SPECK) %>% arrange(-abs(deltapsi)) #SRSF1. IMC45A. SRSF10
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_NEURON_DEVELOPMENT) %>% arrange(-abs(deltapsi)) #KDINS220, NIN, MAPK8, INPP5F
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_NEURONAL_ACTION_POTENTIAL) %>% arrange(-abs(deltapsi)) 
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_RNA_BINDING) %>% arrange(-abs(deltapsi)) # FLNB, PKM, ZNF326
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% arrange(-abs(deltapsi)) # FZNF326, RTF1, AUH, NOVA1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_RNA_SPLICING) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_CYTOSKELETON_ORGANIZATION) %>% arrange(-abs(deltapsi)) # NIN, RUFY3, FLNB, ARHGEG10L

# Compare splicing with gene expression
als_splicing_events_genes = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
als_degs = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% pull(gene_name)
als_degs[als_degs %in% als_splicing_events_genes] #"GYG2"    "ALG13"   "CFAP410"

# Modulizer pie chart of differential events
modulizer_simplfied.sig.summary = als_ctrl.voila_modulizer %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer_simplified) %>% mutate(modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>% arrange(desc(modulizer_simplified)) %>% 
  mutate(pct = n/sum(n) *100, pct.label = case_when(pct > 6.5 ~ paste(as.character(n),"\n", as.character(round(pct,1)),"%"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) #%>% print
modulizer_simplfied.sig.summary %>% print
als_ctrl.voila_modulizer.sig.pie = ggplot(modulizer_simplfied.sig.summary, aes(x = "", y = pct, fill = fct_reorder(modulizer_simplified, n, .desc = TRUE))) + geom_bar(width = 1, stat = "identity", color="white") + coord_polar("y", start=0) + 
  geom_text(aes(x = 1.2, y = ypos, label = pct.label), color = "black", size=2.5) + scale_fill_brewer(palette="Set1") + 
  theme_oz() + theme_void() + theme(legend.title=element_blank(), legend.position = "right", legend.text=element_text(size=7), legend.spacing.y = unit(1, 'mm'), legend.spacing.x = unit(1, 'mm'), legend.key.height = unit(0.4, 'cm'), legend.key.width = unit(0.3, 'cm')) + guides(fill=guide_legend(ncol=1))
als_ctrl.voila_modulizer.sig.pie

# ### Modulizer pie chart of all events
# modulizer_simplfied.summary = als_ctrl.voila_modulizer %>% count(modulizer_simplified) %>% mutate(modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>% arrange(desc(modulizer_simplified)) %>% mutate(pct = n/sum(n) *100, pct.label = case_when(pct > 5 ~ paste(as.character(n),"\n", as.character(round(pct,1)),"%"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) #%>% print
# modulizer_simplfied.summary %>% print
# als_ctrl.voila_modulizer.pie = ggplot(modulizer_simplfied.summary, aes(x = "", y = pct, fill = fct_reorder(modulizer_simplified, n, .desc = TRUE))) + geom_bar(width = 1, stat = "identity", color="white") + coord_polar("y", start=0) + theme_void() + theme(legend.title=element_blank()) + geom_text(aes(y = ypos, label = pct.label), color = "black", size=2.5) + scale_fill_brewer(palette="Set1")
# als_ctrl.voila_modulizer.pie

# # Violin per splicing type
# als_ctrl.voila_modulizer %>% glimpse
# als_ctrl.voila_modulizer %>% my_summarise(als_ctrl_het_tnom)
# als_ctrl.voila_modulizer.violin = violin_plot(filter(als_ctrl.voila_modulizer, modulizer_simplified != "other"), color_by = "modulizer_simplified", continuous = "als_ctrl_het_median_dpsi", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","dodgerblue2","firebrick2","forestgreen","gold2","forestgreen","gold2"), ylims = c(-0.5,0.5), plot_stats = FALSE, stats_test = "wilcoxon_test", one_sample = TRUE, stat_ypos = 0.5, tip_length = 0.01, xlabel = "", arrow_labels = TRUE, numerator = "ALS", denomenator = "CTRL", dpi = 30)
# als_ctrl.voila_modulizer.violin

# Cryptic Exons == denovo cassette_exon
als_ctrl.voila_modulizer %>% filter(denovo == TRUE) %>% count(modulizer)
als_ctrl.voila_modulizer %>% count(modulizer)
als_ctrl.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer)
als_ctrl.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) %>% distinct(gene_name) %>% pull(gene_name) # "FLNB"          "SDHAP1"        "SEC31B"        "LIMS1"         "TPTEP2-CSNK1E" "ZNF721"        "PHKB"          "BCLAF3"        "SNX2"          "UGGT2"         "C1QTNF3-AMACR" "AC007566.1"    "CXXC4"         "FAM53B"        "ITGA6"         "R3HDM2"        "DELE1"         "NPIPA1"        "BABAM2"        "RALGAPA2"     
als_ctrl.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_wilcoxon < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
als_ctrl.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", gene_name %in% ALS_genes, abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_wilcoxon < 0.05) %>% distinct(gene_name) %>% pull(gene_name) # ATXN2

als_ctrl.voila_modulizer %>% filter(gene_name == "SFPQ") %>% arrange(-abs(als_ctrl_het_median_dpsi))
als_ctrl.voila_modulizer %>% filter(gene_name %in% p53.signalling.pathway, abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi))

# Intron retention 
als_ctrl.voila_modulizer.ir = als_ctrl.voila_modulizer %>% filter(modulizer_simplified == "intron_retention")
als_ctrl.voila_modulizer.ir %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # 417
als_ctrl.voila_modulizer.ir %>% filter(gene_name %in% ALS_genes, abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi))


# Voila View deltaPSI violin plot examples of top events:
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes)
# CFAP410 - increased IR in ALS. visually strong
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "CFAP410") %>% glimpse # lsv_junction_id       <chr> "ENSG00000160226:t:44334061-44335804:j:44335805-44336823", "ENSG00000160226:s:44337649-44337667:j:44335804-44337649", "ENSG00000160226:s:44337649-44337667:j:44336921-44337648"

### TIA1 - increased IR (purple) & decreased skipping (blue). visually mod-strong
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "TIA1") %>% glimpse # lsv_junction_id       <chr> "ENSG00000116001:s:70227735-70227839:j:70224629-70227735"

# # TBP - increased IR in ALS. moderate-strong
# als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "TBP") %>% glimpse # lsv_junction_id       <chr> "ENSG00000112592:s:170569612-170569779:j:170569780-170571409"
# 
# ### ATXN2 - less exon skipping. visually moderate
# als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "ATXN2") %>% glimpse # lsv_junction_id       <chr> "ENSG00000204842:s:111555883-111555919:j:111554217-111555883"
# 
# ### NOP56 - increased IR. visually moderate.
# als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "NOP56") %>% glimpse # lsv_junction_id       <chr> "ENSG00000101361:t:2657929-2658393:j:2657693-2657928"
# 
# # INPP5F - increased skipping. visually moderate.
# als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "INPP5F") %>% glimpse # lsv_junction_id       <chr> "ENSG00000198825:t:119820846-119820917:j:119811955-119820846", "ENSG00000198825:t:119820846-119820917:j:119819539-119820846"
# 
# # SLC9A8 - increased IR. visually moderate.
# als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "SLC9A8") %>% glimpse # lsv_junction_id       <chr> "ENSG00000197818:t:49855438-49855581:j:49850845-49855437"
# 
# # C9orf72 - increased IR. visually moderate.
# als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "C9orf72") %>% glimpse # lsv_junction_id       <chr> "ENSG00000147894:s:27560501-27561649:j:27560299-27561585", "ENSG00000147894:s:27560501-27561649:j:27560300-27560500"
# 
# # GLT8D1 - increased IR in ALS. visually moderate
# als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "GLT8D1") %>% glimpse # lsv_junction_id       <chr> "ENSG00000016864:s:52695928-52696040:j:52695587-52695928", "ENSG00000016864:s:52695928-52696040:j:52695588-52695927"


### import violin from voila view
library(magick)
library(cowplot)
img <- system.file("extdata", "cow.jpg", package = "cowplot") %>%
  image_read() %>%
  image_resize("570x380") %>%
  image_colorize(35, "white")

# edit colour and labels in illustrator > export as png
CFAP410_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/CFAP410_majiq_violin.png"))
CFAP410_majiq_violins.gg = ggdraw() + draw_image(CFAP410_majiq_violins.png)
CFAP410_majiq_violins.gg

TIA1_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/TIA1_majiq_violin.png"))
TIA1_majiq_violins.gg = ggdraw() + draw_image(TIA1_majiq_violins.png)
TIA1_majiq_violins.gg

C9orf72_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/C9orf72_majiq_violin.png"))
C9orf72_majiq_violins.gg = ggdraw() + draw_image(C9orf72_majiq_violins.png)
C9orf72_majiq_violins.gg

HDAC1_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/HDAC1_majiq_violin.png"))
HDAC1_majiq_violins.gg = ggdraw() + draw_image(HDAC1_majiq_violins.png)
HDAC1_majiq_violins.gg

majiq_violins.gg = CFAP410_majiq_violins.gg + TIA1_majiq_violins.gg + C9orf72_majiq_violins.gg + HDAC1_majiq_violins.gg


```






## IRFinder 


```{r}
# proportion of samples that are polyA and total in ALS & control
ipsc_mn_als_datasets.metadata %>% count(condition, library_type)
# condition library_type     n
#   <fct>     <chr>        <int>
# 1 ctrl      poly(A)         28
# 2 ctrl      Total           73
# 3 als       poly(A)         35
# 4 als       Total          283
28/(28+73) #27.2% CTRL
35/(35+283) #11.0% ALS

ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05) # 115,280
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05) %>% distinct(gene_name)  # 12,292
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, log2FoldChange < 0) # 2,902
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, log2FoldChange > 0) # 112,378
114918/118410 # 97.1%
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05) %>% arrange(-stat) %>% filter(gene_name %in% ALS_genes) %>% print(n=100) # 1852
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05) %>% arrange(-stat) %>% filter(gene_name %in% ALS_genes) %>% distinct(gene_name) %>% print(n=100) 
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% go.pathways.msigdb$GO_RNA_SPLICING) %>% arrange(pvalue) # 4534
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% p53.signalling.pathway) %>% arrange(-stat) # 1023
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, log2FoldChange > 0) %>% filter(gene_name %in% gprofiler_database$`DNA Damage Response`) %>% arrange(-stat) # 523
ipsc_mn_als_datasets$irfinder %>% top_n(10, wt = stat) # 95,590
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, gene_name == "SFPQ")# %>% mutate(IRratio.mutant - IRratio.ctrl) arrange()
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, gene_name == "ATXN2") %>% print(n=Inf)


# Violin
ipsc_mn_als_datasets.irfinder.filt = ipsc_mn_als_datasets$irfinder %>% filter(log2FoldChange != 0) %>% mutate(condition = "Intron retention ALS vs CTRL")
ipsc_mn_als_datasets.ir.violin <- violin_plot(ipsc_mn_als_datasets.irfinder.filt, color_by = "condition", cols = "forestgreen", one_sample = TRUE, stats_test = "t_test", numerator = "ALS", denomenator = "CTRL", ylims = c(-1.5,1.5), dpi = 300)
# ipsc_mn_als_datasets.ir.violin
# [1] "Using one sample t_test"
# # A tibble: 1  9
#   group       .y.   group1 group2          n statistic     df     p p.signif
#   <chr>       <chr> <chr>  <chr>       <int>     <dbl>  <dbl> <dbl> <chr>   
# 1 ALS vs CTRL lfc   1      null model 221645      423. 221644     0 ****    
# # A tibble: 1  6
#   .y.   group1 group2     effsize      n magnitude
# * <chr> <chr>  <chr>        <dbl>  <int> <ord>    
# 1 lfc   1      null model   0.899 221645 large    

# # Density
# ipsc_mn_als_datasets.ir.density <- density_plot(ipsc_mn_als_datasets.irfinder.filt, color_by = "condition", cols = "forestgreen",  xlab_prefix = "Intron Retention", numerator = "ALS", denomenator = "CTRL", xmax = 1.5, dpi = 300)
# ipsc_mn_als_datasets.ir.density

## Volcano
ipsc_mn_als_datasets.ir.als.labels <- ipsc_mn_als_datasets$irfinder %>% filter(abs(stat) > 5, gene_name %in% c(ALS_genes)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% top_n(10, wt = abs(stat))  %>% pull(gene_name)
ipsc_mn_als_datasets.ir.p53.labels <- ipsc_mn_als_datasets$irfinder %>% filter(abs(stat) > 5, log2FoldChange > 0.5, gene_name %in% c(p53.signalling.pathway)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_als_datasets.ir.synapse.labels <- ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, gene_name %in% c(gprofiler_database$synapse)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% filter(stat < 0) %>% top_n(10, wt = abs(stat)) %>% pull(gene_name)

ipsc_mn_als_datasets.ir.als.labels.filt = c("UNC13A","STMN2","SETX","HNRNPA1","ATXN2","C9orf72","TARDBP")# ,"TIA1","VAPB","CHMP2B",,"UNC13B"", "QKI")     
# ipsc_mn_als_datasets.ir.rbp.labels <- ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.0001, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.ir.volcano <- volcano_plot(ipsc_mn_als_datasets$irfinder, numerator = "ALS", denomenator = "CTRL",
               title = "Intron retention: pan ALS", subtitle = "101 Controls vs 318 ALS", ymax = 16, xmax = 2, dpi = 300,
               gene_labels = c(ipsc_mn_als_datasets.ir.als.labels, ipsc_mn_als_datasets.ir.als.labels.filt, ipsc_mn_als_datasets.ir.p53.labels, "HDAC1", "CFAP410", ipsc_mn_als_datasets.ir.synapse.labels))
ipsc_mn_als_datasets.ir.volcano

ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.001, gene_name %in% c(go.pathways.msigdb$GO_SYNAPSE)) %>% distinct(gene_name, .keep_all = TRUE)# %>% pull(gene_name)
 ipsc_mn_als_datasets$irfinder %>% filter(abs(stat) > 5, log2FoldChange > 0.5, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% top_n(10, wt = abs(stat))
  ipsc_mn_als_datasets$irfinder %>% filter(abs(stat) > 5, log2FoldChange > 0.5, gene_name %in% c(go.pathways.msigdb$GO_RNA_SPLICING)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% top_n(10, wt = abs(stat))

## GO
ipsc_mn_als_datasets.gost <- ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05) %>% arrange(-abs(stat)) %>% distinct(gene_id, .keep_all = TRUE) %>% group_split(log2FoldChange > 0) %>% map("gene_id") %>% gost()
ipsc_mn_als_datasets.gost_filt <- ipsc_mn_als_datasets.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC")) %>% arrange( -log10(p_value) ) %>%
  mutate(query = case_when(query == "query_1" ~ "ALS IR down", query == "query_2" ~ "ALS IR up"), query = factor(query, levels = c("ALS IR down", "ALS IR up")))
table(ipsc_mn_als_datasets.gost_filt$query)
# ALS IR down   ALS IR up 
#         102        1970
ipsc_mn_als_datasets.gost_ir.down <- ipsc_mn_als_datasets.gost_filt %>% filter(query == "ALS IR down")
ipsc_mn_als_datasets.gost_ir.up <- ipsc_mn_als_datasets.gost_filt %>% filter(query == "ALS IR up")
paste('"',unique(ipsc_mn_als_datasets.gost_ir.down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list = c(
"dendrite",
"synaptic membrane",
"Glutamatergic synapse",
"synaptic signaling",
"presynapse",
"neuron projection",
"cation channel complex",
# "ATP binding",
"synapse")
ipsc_mn_als_datasets.gost_filt_ir_down <- ipsc_mn_als_datasets.gost_filt %>% filter(query == "ALS IR down", term_name %in% down_list)
paste('"',unique(ipsc_mn_als_datasets.gost_ir.up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list = c(
# "signal transduction by p53 class mediator",
"programmed cell death",
"autophagy",
# "endoplasmic reticulum",
"protein ubiquitination",
# "RNA biosynthetic process",
# "nervous system development",
# "ubiquitin-dependent protein catabolic process",
"DNA repair",
# "regulation of response to DNA damage stimulus",
"cellular response to DNA damage stimulus",
# "ncRNA metabolic process",
# "translation",
# "protein catabolic process",
"regulation of RNA metabolic process",
# "mitochondrial envelope",
# "microtubule cytoskeleton",
"protein localization",
"cellular response to stress",
"mitochondrion",
"protein metabolic process",
"protein binding",

"spliceosomal complex",
"DNA metabolic process",
"proteasomal protein catabolic process")

ipsc_mn_als_datasets.gost_filt_ir_up <- ipsc_mn_als_datasets.gost_filt %>% filter(query == "ALS IR up")  %>% filter(term_name %in% up_list)
ipsc_mn_als_datasets.gost_filt_ir = bind_rows(ipsc_mn_als_datasets.gost_filt_ir_up, ipsc_mn_als_datasets.gost_filt_ir_down) %>% mutate(query = fct_relevel(query,"ALS IR up"))
ipsc_mn_als_datasets.gost_filt_ir$term_name = gsub("synaptic transmission, glutamatergic", "glutamatergic synapse", ipsc_mn_als_datasets.gost_filt_ir$term_name) %>% gsub("cation channel activity", "channel", .) %>% gsub("regulation of ", "", .) %>% gsub("protein catabolic process", "process", .) %>% gsub("cellular ", "", .) %>% gsub("response to DNA damage stimulus", "DNA damage response",.)
ipsc_mn_als_datasets.ir.go_curated <- curated_profiles(ipsc_mn_als_datasets.gost_filt_ir, gprofiles =, cols = c("firebrick2", "dodgerblue2")) + theme(strip.text.y.right = element_text(angle = 90)) + coord_cartesian(xlim = c(0,6))# + theme(legend.position = "none", axis.text = element_text(size = 10))
ipsc_mn_als_datasets.ir.go_curated


# which genes are driving the GO terms
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`protein binding`) %>% arrange(-abs(stat))
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`mitochondrion`) %>% arrange(-abs(stat))
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`RNA metabolic process`) %>% arrange(-abs(stat))
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`cellular response to DNA damage stimulus`) %>% arrange(-abs(stat))
ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05, stat < 0, gene_name %in% gprofiler_database$`synapse`) %>% arrange(-abs(stat))

```


## Merge

```{r}
pan_als_ipsn_splicing_fig_merged = ggarrange(
  ggarrange(majiq.als_vs_ctrl.volcano, majiq_violins.gg, ncol = 2, labels = c("a","b"), widths = c(1,0.8)),
  ggarrange(als_ctrl.voila.gost_filt.terms.curated, als_ctrl.voila_modulizer.sig.pie, ipsc_mn_als_datasets.ir.violin, ncol = 3, labels = c("c","d","e"), widths = c(0.5,1,0.5)),
  ggarrange(ipsc_mn_als_datasets.ir.volcano, ipsc_mn_als_datasets.ir.go_curated, ncol = 2, labels = c("f","g"), widths = c(1,0.8)),
  nrow = 3, heights = c(1.2,0.8,1.2))
# pan_als_ipsn_splicing_fig_merged
ggsave(pan_als_ipsn_splicing_fig_merged, filename = here(proj_path,"expression/deseq2/pan_als_ipsn_splicing_fig_merged.png"), height = 9.1, width = 8.5)

```



# Table S7-8 MAJIQ & IRFinder panALS datasets

```{r}
als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
als_ctrl.voila.sig = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% select(-contains("percentile"),-contains("changing"),-ucsc_lsv_link)
als_ctrl.voila.sig
write_csv(als_ctrl.voila.sig, here(proj_path,"splicing/majiq/figures/TableS9.majiq_splicing_panALS.sig.csv"))


ipsc_mn_als_datasets.ir.sig = ipsc_mn_als_datasets$irfinder %>% filter(padj < 0.05) %>% select(Intron.GeneName.GeneID.Coords, log2FoldChange, stat, padj, IRratio.ALS = IRratio.mutant, IRratio.ctrl, IRratio.diff, IRdirection, gene_name, gene_id)
ipsc_mn_als_datasets.ir.sig
write_csv(ipsc_mn_als_datasets.ir.sig, here(proj_path,"splicing/irfinder/ipsc_mn_als_datasets.ir.csv"))

```


# Fig 7 Splicing compare mutations

## MAJIQ

```{r}
sporadic_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/sporadic.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "Sporadic")
c9orf72_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/c9orf72.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "C9orf72")
fus_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/fus.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "FUS")
tardbp_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/tardbp.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "TARDBP")
sod1_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/sod1.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "SOD1")
# vcp_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/vcp.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "VCP")
# polyA.als_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/polyA.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
# total.als_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/total.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
mutations_bind.als_vs_ctrl.voila = bind_rows(sporadic_vs_ctrl.voila, c9orf72_vs_ctrl.voila, fus_vs_ctrl.voila, tardbp_vs_ctrl.voila, sod1_vs_ctrl.voila)#, vcp_vs_ctrl.voila)
mutations_bind.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% count(mutation)
# mutation     n
#   <chr>    <int>
# 1 C9orf72    511
# 2 FUS       7309
# 3 SOD1       864
# 4 Sporadic   695
# 5 TARDBP    9826
# 6 VCP        142


### Volcanos
ipsc_mn_tardbp_datasets.metadata %>% count(condition) 
tardbp_vs_ctrl.voila.ALS.labels <- tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, tnom < 0.0001, gene_name %in% c(ALS_genes, "C9orf72","HNRNPA2B1")) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
tardbp_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(tardbp_vs_ctrl.voila, numerator = "TARDBP", denomenator = "CTRL", subtitle = "48 Controls vs 10 TARDBP", ymax = 10, xmax = 0.6, dpi = 300, junction_labels = c(tardbp_vs_ctrl.voila.ALS.labels))
# tardbp_vs_ctrl.voila.volcano

ipsc_mn_fus_datasets.metadata %>% count(condition) 
fus_vs_ctrl.voila.ALS.labels <- fus_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, tnom < 0.001, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
fus_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(fus_vs_ctrl.voila, numerator = "FUS", denomenator = "CTRL", subtitle = "48 Controls vs 10 FUS", ymax = 10, xmax = 0.5, dpi = 300, junction_labels = c(fus_vs_ctrl.voila.ALS.labels))
# fus_vs_ctrl.voila.volcano

ipsc_mn_c9orf72_datasets.metadata %>% count(condition) 
c9orf72_vs_ctrl.voila.ALS.labels <- c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
c9orf72_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(c9orf72_vs_ctrl.voila, numerator = "C9orf72", denomenator = "CTRL", subtitle = "48 Controls vs 10 C9orf72", ymax = 6, xmax = 0.3, dpi = 300, junction_labels = c(c9orf72_vs_ctrl.voila.ALS.labels))
# c9orf72_vs_ctrl.voila.volcano

ipsc_mn_sod1_datasets.metadata %>% count(condition) 
sod1_vs_ctrl.voila.ALS.labels <- sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, tnom < 0.03, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
sod1_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(sod1_vs_ctrl.voila, numerator = "SOD1", denomenator = "CTRL", subtitle = "48 Controls vs 10 SOD1", ymax = 5, xmax = 0.3, dpi = 300, junction_labels = c(sod1_vs_ctrl.voila.ALS.labels))
# sod1_vs_ctrl.voila.volcano

# ipsc_mn_vcp_datasets.metadata %>% count(condition) 
# vcp_vs_ctrl.voila.ALS.labels <- vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
# vcp_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(vcp_vs_ctrl.voila, numerator = "VCP", denomenator = "CTRL", subtitle = "48 Controls vs 10 VCP", ymax = 4, xmax = 0.3, dpi = 300, junction_labels = c(vcp_vs_ctrl.voila.ALS.labels))
# vcp_vs_ctrl.voila.volcano

ipsc_mn_sporadic_datasets.metadata %>% count(condition) 
sporadic_vs_ctrl.voila.ALS.labels <- sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
sporadic_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(sporadic_vs_ctrl.voila, numerator = "Sporadic", denomenator = "CTRL", subtitle = "48 Controls vs 10 Sporadic", ymax = 7, xmax = 0.35, dpi = 300, junction_labels = c(sporadic_vs_ctrl.voila.ALS.labels))
# sporadic_vs_ctrl.voila.volcano

# majiq.volcano_multiplot <- tardbp_vs_ctrl.voila.volcano + fus_vs_ctrl.voila.volcano + c9orf72_vs_ctrl.voila.volcano + sod1_vs_ctrl.voila.volcano + sporadic_vs_ctrl.voila.volcano +
#   plot_layout(ncol = 3, nrow = 2)
# majiq.volcano_multiplot
# ggsave(majiq.volcano_multiplot, filename = here(proj_path,"expression/deseq2/majiq.volcano_multiplot.png"), height = 7, width = 10)

### Correlation heatmap

# heatmap of correlation coefficients
ipsc_mn_mutations_deltapsi = select(sporadic_vs_ctrl.voila, lsv_junction_id, Sporadic = deltapsi) %>% 
  left_join(select(c9orf72_vs_ctrl.voila, lsv_junction_id, C9orf72 = deltapsi)) %>%
  left_join(select(sod1_vs_ctrl.voila, lsv_junction_id, SOD1 = deltapsi)) %>%
  left_join(select(fus_vs_ctrl.voila, lsv_junction_id, FUS = deltapsi)) %>%
  left_join(select(tardbp_vs_ctrl.voila, lsv_junction_id, TARDBP = deltapsi)) %>%
  # left_join(select(vcp_vs_ctrl.voila, lsv_junction_id, VCP = deltapsi)) %>%
  drop_na() %>%
  column_to_rownames("lsv_junction_id")
cormat <- round(cor(ipsc_mn_mutations_deltapsi),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
majiq_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson Correlation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.spacing.y = unit(1, 'mm'), legend.title = element_text(size = 8), axis.text=element_text(size=8)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 0.5, title.position = "top", title.hjust = 0.5))
majiq_correlation_ggheatmap

fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 11471
fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 2706
fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #38
fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 2246
fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 5847

tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 10,429
tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 3364
tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #51
tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 2707
tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 7553

sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 883
sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 540
sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #"SFPQ"   "TIA1"   "CAMTA1" "DCTN1"  "NOP56" 
sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 183
sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 513

sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 671
sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 338
sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% pull(gene_name) # "BID"    "CAMTA1" "CAMTA1" "SFPQ"   "APP"  
sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 136
sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 360

c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 626
c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 318
c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% pull(gene_name) #"TIA1"   "GOLIM4" "PTPRN2" "PTPRN2" "GOLIM4"
c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 132 / 511
c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 338/511

vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 179
vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 88
vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #"ATXN2" "KIF5A" "EWSR1"
vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 24
vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 96

polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 174
polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 108
polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #"HLA-B"
polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 13
polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 86

total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 381
total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 222
total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #  "CAMTA1" "APOE"   "APP"    "SCFD1" 
total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 99
total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 259



```

## IRFinder

```{r}
ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds"))
ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds"))
ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds"))
ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds"))
ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M

### Violins
ipsc_mn_mutations_datasets.irfinder.filt = bind_rows(
  mutate(filter(ipsc_mn_sporadic_datasets$irfinder, log2FoldChange != 0), condition = "Sporadic"),
  mutate(filter(ipsc_mn_c9orf72_datasets$irfinder, log2FoldChange != 0), condition = "C9orf72"),
  mutate(filter(ipsc_mn_sod1_datasets$irfinder, log2FoldChange != 0), condition = "SOD1"),
  mutate(filter(ipsc_mn_fus_datasets$irfinder, log2FoldChange != 0), condition = "FUS"),
  mutate(filter(ipsc_mn_tardbp_datasets$irfinder, log2FoldChange != 0), condition = "TARDBP")) %>%
  mutate(condition = factor(condition, levels = c("Sporadic","C9orf72","SOD1","TARDBP","FUS")))
ipsc_mn_mutations_datasets.irfinder.filt

ipsc_mn_mutations_datasets.irfinder.filt %>% filter(padj < 0.05) %>% count(condition)
# condition      n
#   <fct>      <int>
# 1 Sporadic    7741
# 2 C9orf72        9
# 3 SOD1           3
# 4 TARDBP     45232
# 5 FUS       125454

ipsc_mn_mutations_datasets.irfinder.violin = violin_plot(ipsc_mn_mutations_datasets.irfinder.filt, color_by = "condition", continuous = "log2FoldChange", cols = brewer.pal(5, "Set1"), ylims = c(-2,2), null_value = 0, plot_stats = TRUE, stats_test = "t_test", one_sample = TRUE, ylabel = "Intron retention log2FC ALS vs CTRL", numerator = "ALS", denomenator = "CTRL", dpi=300)
ipsc_mn_mutations_datasets.irfinder.violin
# [1] "Using one sample t_test"
# # A tibble: 5  9
#   group    .y.   group1 group2          n statistic     df     p p.signif
#   <fct>    <chr> <chr>  <chr>       <int>     <dbl>  <dbl> <dbl> <chr>   
# 1 Sporadic lfc   1      null model 212672      333. 212671     0 ****    
# 2 C9orf72  lfc   1      null model 214757     -329. 214756     0 ****    
# 3 SOD1     lfc   1      null model 211567     -253. 211566     0 ****    
# 4 TARDBP   lfc   1      null model 199570     -274. 199569     0 ****    
# 5 FUS      lfc   1      null model 206862     -478. 206861     0 ****    
# # A tibble: 5  7
#   .y.   group1 group2     effsize group         n magnitude
# * <chr> <chr>  <chr>        <dbl> <fct>     <int> <ord>    
# 1 lfc   1      null model   0.723 Sporadic 212672 moderate 
# 2 lfc   1      null model  -0.710 C9orf72  214757 moderate 
# 3 lfc   1      null model  -0.551 SOD1     211567 moderate 
# 4 lfc   1      null model  -0.612 TARDBP   199570 moderate 
# 5 lfc   1      null model  -1.05  FUS      206862 large    
                                                         

## Upset plot https://github.com/const-ae/ggupset
ipsc_mn_mutants_bind_datasets.irfinder = bind_rows(mutate(ipsc_mn_sporadic_datasets$irfinder, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$irfinder, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$irfinder, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$irfinder, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$irfinder, mutation = "FUS"))#,mutate(ipsc_mn_vcp_datasets$irfinder, mutation = "VCP"))
ipsc_mn_mutants_bind_datasets.irfinder.upset = ipsc_mn_mutants_bind_datasets.irfinder %>% filter(padj < 0.05) %>% select(mutation, Intron.GeneName.GeneID.Coords) %>% group_by(Intron.GeneName.GeneID.Coords) %>% summarise(mutations = list(mutation)) %>%
  ggplot(aes(x = mutations)) + geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.2, hjust = 0.5, size = 2.4, angle = 0) +
    scale_x_upset(order_by = "degree")  + # n_intersections = 20
    # scale_y_continuous(breaks = NULL, lim = c(0, 4500), name = "") +
    theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + labs(x = "", y = "") +
    theme_combmatrix(combmatrix.label.make_space = TRUE)
ipsc_mn_mutants_bind_datasets.irfinder.upset

ipsc_mn_mutants_join_datasets.irfinder %>% filter(padj.vcp < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # XIST
ipsc_mn_mutants_join_datasets.irfinder %>% filter(padj.sod1 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # JAK1


```


## Merge


```{r}
compare_mutations_ipsn_splicing_fig_merged = ggarrange(
  ggarrange(tardbp_vs_ctrl.voila.volcano, fus_vs_ctrl.voila.volcano, c9orf72_vs_ctrl.voila.volcano, sod1_vs_ctrl.voila.volcano, sporadic_vs_ctrl.voila.volcano, majiq_correlation_ggheatmap, nrow = 2, ncol = 3, labels = c("a","b","c","d","e","f")),
  ipsc_mn_mutations_datasets.irfinder.violin,
  nrow = 2, heights = c(2,1), labels = c("","g"))
# compare_mutations_ipsn_splicing_fig_merged
ggsave(compare_mutations_ipsn_splicing_fig_merged, filename = here(proj_path,"expression/deseq2/compare_mutations_ipsn_splicing_fig_merged.png"), height = 9, width = 9)

```




# Fig S14 MAJIQ mutations scatters & upset

```{r}
ipsc_mn_mutations_deltapsi = select(sporadic_vs_ctrl.voila, lsv_junction_id, Sporadic = deltapsi) %>% 
  left_join(select(c9orf72_vs_ctrl.voila, lsv_junction_id, C9orf72 = deltapsi)) %>%
  left_join(select(sod1_vs_ctrl.voila, lsv_junction_id, SOD1 = deltapsi)) %>%
  left_join(select(fus_vs_ctrl.voila, lsv_junction_id, FUS = deltapsi)) %>%
  left_join(select(tardbp_vs_ctrl.voila, lsv_junction_id, TARDBP = deltapsi)) %>%
  # left_join(select(vcp_vs_ctrl.voila, lsv_junction_id, VCP = deltapsi)) %>%
  drop_na() %>%
  column_to_rownames("lsv_junction_id")

ipsc_mn_mutations_majiq_list = list("Sporadic" = sporadic_vs_ctrl.voila, "C9orf72" = c9orf72_vs_ctrl.voila, "TARDBP" = tardbp_vs_ctrl.voila, "SOD1" = sod1_vs_ctrl.voila, "FUS" = fus_vs_ctrl.voila)# "VCP" = vcp_vs_ctrl.voila)
# mutations <- names(ipsc_mn_mutations_list)
# 
# ipsc_mn_mutations_list = purrr::map2(
#       .x = ipsc_mn_mutations_list,
#       .y = names(ipsc_mn_mutations_list), ~{
#         .x$mutation <- .y
#         return(.x)
#       })

mutations_dpsi_majiq_scatter_plot <- 
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "Sporadic", mutation2 = "C9orf72")  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "Sporadic", mutation2 = "SOD1")  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "Sporadic", mutation2 = "FUS")  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "Sporadic", mutation2 = "TARDBP")  +
# dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "Sporadic", mutation2 = "VCP", highlight = "VCP")  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "C9orf72", mutation2 = "SOD1")  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "C9orf72", mutation2 = "FUS")  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "C9orf72", mutation2 = "TARDBP")  +
# dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "C9orf72", mutation2 = "VCP")  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "SOD1", mutation2 = "FUS")  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "SOD1", mutation2 = "TARDBP")  +
# dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "SOD1", mutation2 = "VCP", highlight = c("SOD1","VCP"))  +
dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "FUS", mutation2 = "TARDBP")  +
# dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "FUS", mutation2 = "VCP", highlight = c("VCP","FUS"))  +
# dpsi_majiq_scatter(model_list1 = ipsc_mn_mutations_majiq_list, model_list2 = ipsc_mn_mutations_majiq_list, mutation1 = "TARDBP", mutation2 = "VCP", highlight = c("TARDBP","VCP")) + 
  plot_layout(ncol = 4, nrow = 3) #&  xlim(-8,8) & ylim(-8,8)
# mutations_dpsi_majiq_scatter_plot

### Upset plot https://github.com/const-ae/ggupset
ipsc_mn_mutants_bind_datasets.majiq.upset_lsv_level = mutations_bind.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% select(mutation, lsv_junction_id) %>% group_by(lsv_junction_id) %>% summarise(mutations = list(mutation)) %>%
  ggplot(aes(x = mutations)) + geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, hjust = 0.5, size = 3) + #, angle = 30) +
    scale_x_upset(order_by = "degree")  + # n_intersections = 20
    scale_y_continuous(breaks = NULL, lim = c(0, 10000), name = "") +
    theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + labs(x = "", y = "") +
    theme_combmatrix(combmatrix.label.make_space = TRUE)
ipsc_mn_mutants_bind_datasets.majiq.upset_lsv_level

# join at lsv level
ipsc_mn_mutants_join_datasets.majiq.lsv_level = full_join(select(c9orf72_vs_ctrl.voila, gene_name, lsv_junction_id, changing.c9orf72 = changing_dpsi0.1),
                                                select(sporadic_vs_ctrl.voila, gene_name, lsv_junction_id, changing.sals = changing_dpsi0.1)) %>%
  full_join(select(fus_vs_ctrl.voila, gene_name, lsv_junction_id, changing.fus = changing_dpsi0.1)) %>%
  full_join(select(tardbp_vs_ctrl.voila, gene_name, lsv_junction_id, changing.tardbp = changing_dpsi0.1)) %>%
  full_join(select(sod1_vs_ctrl.voila, gene_name, lsv_junction_id, changing.sod1 = changing_dpsi0.1)) %>%
  full_join(select(vcp_vs_ctrl.voila, gene_name, lsv_junction_id, changing.vcp = changing_dpsi0.1))
ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt = ipsc_mn_mutants_join_datasets.majiq.lsv_level %>% 
  filter(changing.c9orf72 == TRUE & changing.sals == TRUE | changing.c9orf72 == TRUE & changing.tardbp == TRUE |changing.c9orf72 == TRUE & changing.sod1 == TRUE |changing.c9orf72 == TRUE & changing.vcp == TRUE |
         changing.sals == TRUE & changing.fus == TRUE | changing.sals == TRUE & changing.tardbp == TRUE | changing.sals == TRUE & changing.sod1 == TRUE | changing.sals == TRUE & changing.vcp == TRUE |
         changing.fus == TRUE & changing.tardbp == TRUE | changing.fus == TRUE & changing.sod1 == TRUE | changing.fus == TRUE & changing.vcp == TRUE | 
         changing.tardbp == TRUE & changing.sod1 == TRUE | changing.tardbp == TRUE & changing.vcp == TRUE | changing.sod1 == TRUE & changing.vcp == TRUE)
ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt

ipsc_mn_mutants_join_datasets.majiq.lsv_level %>% filter(changing.c9orf72 == TRUE, changing.fus == TRUE, changing.tardbp == TRUE, changing.sod1 == TRUE, changing.sals == TRUE, changing.vcp == TRUE)
ipsc_mn_mutants_join_datasets.majiq.lsv_level %>% filter(changing.c9orf72 == TRUE, changing.fus == TRUE, changing.tardbp == TRUE, changing.sod1 == TRUE, changing.sals == TRUE) 
ipsc_mn_mutants_join_datasets.majiq.lsv_level %>% filter(changing.c9orf72 == TRUE, changing.fus == TRUE, changing.tardbp == TRUE, changing.sod1 == TRUE) # SNHG8, CPNE7
ipsc_mn_mutants_join_datasets.majiq.lsv_level %>% filter(changing.fus == TRUE, changing.tardbp == TRUE, changing.sod1 == TRUE, changing.sals == TRUE) # IKBIP, DDRGK1, CLK4

ipsc_mn_mutants_join_datasets.majiq.lsv_level %>% filter(gene_name == "SFPQ") %>% print(n=Inf)

mutations_majiq_scatter_upset_plot = ggarrange(mutations_dpsi_majiq_scatter_plot, ipsc_mn_mutants_bind_datasets.majiq.upset_lsv_level, nrow = 2, heights = c(2.5,1), labels = c("a","b"))
mutations_majiq_scatter_upset_plot
ggsave(mutations_majiq_scatter_upset_plot, filename = here(proj_path,"expression/deseq2/mutations_majiq_scatter_upset_plot.png"), height = 10, width = 10)



```


# Fig 15 IRFinder compare mutations

```{r}
##  Volcanos
ipsc_mn_sporadic_datasets.als.ir.labels <- ipsc_mn_sporadic_datasets$irfinder %>% filter(abs(stat) > 3, gene_name %in% c(ALS_genes)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all=TRUE) %>% 
  top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_sporadic_datasets.ir.p53.labels <- ipsc_mn_sporadic_datasets$irfinder %>% filter(abs(stat) > 3, log2FoldChange > 0.5, gene_name %in% c(p53.signalling.pathway)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_sporadic_datasets.ir.volcano <- volcano_plot(ipsc_mn_sporadic_datasets$irfinder, "Sporadic", numerator = "Sporadic", denomenator = "CTRL", subtitle = "56 Controls vs 208 Sporadic", ymax = 10, xmax = 2, dpi = 300, gene_labels = c(ipsc_mn_sporadic_datasets.als.ir.labels,ipsc_mn_sporadic_datasets.ir.p53.labels))
# ipsc_mn_sporadic_datasets.ir.volcano

ipsc_mn_c9orf72_datasets.als.ir.labels <- ipsc_mn_c9orf72_datasets$irfinder %>% filter(padj < 0.05) %>% distinct(gene_name, .keep_all=TRUE) %>% top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_c9orf72_datasets.ir.labels <- ipsc_mn_c9orf72_datasets$irfinder %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_c9orf72_datasets.ir.volcano <- volcano_plot(ipsc_mn_c9orf72_datasets$irfinder, "C9orf72", numerator = "C9orf72", denomenator = "CTRL", subtitle = "77 Controls vs 51 C9orf72", ymax = 8, xmax = 2.5, dpi = 300, gene_labels = c(ipsc_mn_c9orf72_datasets.als.ir.labels))
# ipsc_mn_c9orf72_datasets.ir.volcano

ipsc_mn_fus_datasets.als.ir.labels <- ipsc_mn_fus_datasets$irfinder %>% filter(abs(stat) > 3, gene_name %in% c(ALS_genes)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all=TRUE) %>% 
  top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_fus_datasets.ir.p53.labels <- ipsc_mn_fus_datasets$irfinder %>% filter(abs(stat) > 3, log2FoldChange > 0.5, gene_name %in% c(p53.signalling.pathway)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_fus_datasets.ir.volcano <- volcano_plot(ipsc_mn_fus_datasets$irfinder, "FUS", numerator = "FUS", denomenator = "CTRL", subtitle = "54 Controls vs 9 FUS", ymax = 23, xmax = 6, dpi = 300, gene_labels = c(ipsc_mn_fus_datasets.als.ir.labels,ipsc_mn_fus_datasets.ir.p53.labels))
# ipsc_mn_fus_datasets.ir.volcano

ipsc_mn_sod1_datasets.als.ir.labels <- ipsc_mn_sod1_datasets$irfinder %>% filter(padj < 0.05) %>% distinct(gene_name, .keep_all=TRUE) %>% top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_sod1_datasets.ir.volcano <- volcano_plot(ipsc_mn_sod1_datasets$irfinder, "SOD1", numerator = "SOD1", denomenator = "CTRL", subtitle = "63 Controls vs 20 SOD1", ymax = 7, xmax = 2.5, dpi = 300, gene_labels = c(ipsc_mn_sod1_datasets.als.ir.labels))
# ipsc_mn_sod1_datasets.ir.volcano

ipsc_mn_tardbp_datasets.als.ir.labels <- ipsc_mn_tardbp_datasets$irfinder %>% filter(gene_name %in% c(ALS_genes)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all=TRUE) %>% 
  top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_tardbp_datasets.ir.p53.labels <- ipsc_mn_tardbp_datasets$irfinder %>% filter(abs(stat) > 5, log2FoldChange > 0.5, gene_name %in% c(p53.signalling.pathway)) %>% arrange(-abs(stat)) %>% distinct(gene_name, .keep_all = TRUE) %>% top_n(10, wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_tardbp_datasets.ir.volcano <- volcano_plot(ipsc_mn_tardbp_datasets$irfinder, "TARDBP", numerator = "TARDBP", denomenator = "CTRL", subtitle = "48 Controls vs 10 TARDBP", ymax = 15, xmax = 5, dpi = 300, gene_labels = c(ipsc_mn_tardbp_datasets.als.ir.labels,ipsc_mn_tardbp_datasets.ir.p53.labels))
# ipsc_mn_tardbp_datasets.ir.volcano

# ipsc_mn_vcp_datasets.ir.labels <- ipsc_mn_vcp_datasets$irfinder %>% filter(padj < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
# ipsc_mn_vcp_datasets.ir.volcano <- volcano_plot(ipsc_mn_vcp_datasets$irfinder, "VCP", numerator = "VCP", denomenator = "CTRL", subtitle = "6 Controls vs 6 VCP", ymax = 5, xmax = 3.5, dpi = 300, gene_labels = ipsc_mn_vcp_datasets.ir.labels)
# # ipsc_mn_vcp_datasets.ir.volcano

### Correlation heatmap

# heatmap of correlation coefficients
ipsc_mn_mutations_irfinder = select(ipsc_mn_sporadic_datasets$irfinder, Intron.GeneName.GeneID.Coords, Sporadic = stat) %>% 
  left_join(select(ipsc_mn_c9orf72_datasets$irfinder, Intron.GeneName.GeneID.Coords, C9orf72 = stat)) %>%
  left_join(select(ipsc_mn_sod1_datasets$irfinder, Intron.GeneName.GeneID.Coords, SOD1 = stat)) %>%
  left_join(select(ipsc_mn_fus_datasets$irfinder, Intron.GeneName.GeneID.Coords, FUS = stat)) %>%
  left_join(select(ipsc_mn_tardbp_datasets$irfinder, Intron.GeneName.GeneID.Coords, TARDBP = stat)) %>%
  drop_na() %>%
  column_to_rownames("Intron.GeneName.GeneID.Coords")
cormat <- round(cor(ipsc_mn_mutations_irfinder),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
ipsc_mn_mutations_irfinder_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson Correlation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.spacing.y = unit(1, 'mm'), legend.title = element_text(size = 8), axis.text=element_text(size=8)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 0.5, title.position = "top", title.hjust = 0.5))
ipsc_mn_mutations_irfinder_ggheatmap

# upset plot
### Upset plot https://github.com/const-ae/ggupset
ipsc_mn_mutations_irfinder_bind = bind_rows(mutate(ipsc_mn_sporadic_datasets$irfinder,mutation="Spoardic"), mutate(ipsc_mn_fus_datasets$irfinder,mutation="FUS"), mutate(ipsc_mn_tardbp_datasets$irfinder,mutation="TARDBP"), mutate(ipsc_mn_sod1_datasets$irfinder,mutation="SOD1"), mutate(ipsc_mn_c9orf72_datasets$irfinder,mutation="C9orf72"))

ipsc_mn_mutants_bind_datasets.irfinder.upset_intron_level = ipsc_mn_mutations_irfinder_bind %>% filter(padj < 0.05) %>% select(mutation, Intron.GeneName.GeneID.Coords) %>% group_by(Intron.GeneName.GeneID.Coords) %>% summarise(mutations = list(mutation)) %>%
  ggplot(aes(x = mutations)) + geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, hjust = 0.5, size = 3) + #, angle = 30) +
    scale_x_upset(order_by = "degree")  + # n_intersections = 20
    scale_y_continuous(breaks = NULL, lim = c(0, 100000), name = "") +
    theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + labs(x = "", y = "") +
    theme_combmatrix(combmatrix.label.make_space = TRUE)
ipsc_mn_mutants_bind_datasets.irfinder.upset_intron_level


compare_mutations_irfinder_volcano_multiplot <- plot_grid(
  plot_grid(ipsc_mn_sporadic_datasets.ir.volcano, ipsc_mn_c9orf72_datasets.ir.volcano, ipsc_mn_fus_datasets.ir.volcano, ipsc_mn_sod1_datasets.ir.volcano, ipsc_mn_tardbp_datasets.ir.volcano, ipsc_mn_mutations_irfinder_ggheatmap, labels = "auto", ncol = 2, nrow = 3),
  ipsc_mn_mutants_bind_datasets.irfinder.upset_intron_level, labels = c("","g"), nrow = 2, rel_heights = c(3,1)) 
# compare_mutations_irfinder_volcano_multiplot
ggsave(compare_mutations_irfinder_volcano_multiplot, filename = here(proj_path,"splicing/irfinder/compare_mutations_irfinder_volcano_multiplot.png"), height = 11.75, width = 8.25)

```

      
      
# Table S9-10 MAJIQ & IRFinder overlap between genetic backgrounds

```{r}
# join at lsv level
ipsc_mn_mutants_join_datasets.majiq.lsv_level = full_join(select(c9orf72_vs_ctrl.voila, gene_name, lsv_junction_id, changing.c9orf72 = changing_dpsi0.1),
                                                select(sporadic_vs_ctrl.voila, gene_name, lsv_junction_id, changing.sals = changing_dpsi0.1)) %>%
  full_join(select(fus_vs_ctrl.voila, gene_name, lsv_junction_id, changing.fus = changing_dpsi0.1)) %>%
  full_join(select(tardbp_vs_ctrl.voila, gene_name, lsv_junction_id, changing.tardbp = changing_dpsi0.1)) %>%
  full_join(select(sod1_vs_ctrl.voila, gene_name, lsv_junction_id, changing.sod1 = changing_dpsi0.1)) #%>%  full_join(select(vcp_vs_ctrl.voila, gene_name, lsv_junction_id, changing.vcp = changing_dpsi0.1))
ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt = ipsc_mn_mutants_join_datasets.majiq.lsv_level %>% 
  filter(changing.c9orf72 == TRUE & changing.sals == TRUE | changing.c9orf72 == TRUE & changing.tardbp == TRUE |changing.c9orf72 == TRUE & changing.sod1 == TRUE | 
#changing.c9orf72 == TRUE & changing.vcp == TRUE |changing.sals == TRUE & changing.vcp == TRUE |changing.fus == TRUE & changing.vcp == TRUE  | changing.sod1 == TRUE & changing.vcp == TRUE | changing.tardbp == TRUE & changing.vcp == TRUE
         changing.sals == TRUE & changing.fus == TRUE | changing.sals == TRUE & changing.tardbp == TRUE | changing.sals == TRUE & changing.sod1 == TRUE | 
         changing.fus == TRUE & changing.tardbp == TRUE | changing.fus == TRUE & changing.sod1 == TRUE | changing.tardbp == TRUE & changing.sod1 == TRUE)
ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt
write_csv(ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt, here(proj_path,"splicing/majiq/figures/TableS13.majiq_splicing_overlap_mutations.csv"))


# join at intron level
ipsc_mn_mutations_irfinder.sig = mutate(select(ipsc_mn_sporadic_datasets$irfinder, Intron.GeneName.GeneID.Coords, Sporadic.lfc = log2FoldChange, Sporadic.padj = padj), Sporadic.change = case_when(Sporadic.padj < 0.05 & Sporadic.lfc < 0 ~ "down",Sporadic.padj < 0.05 & Sporadic.lfc > 0 ~ "up", TRUE ~ "")) %>% 
  full_join(mutate(select(ipsc_mn_c9orf72_datasets$irfinder, Intron.GeneName.GeneID.Coords, C9orf72.lfc = log2FoldChange, C9orf72.padj = padj), C9orf72.change = case_when(C9orf72.padj < 0.05 & C9orf72.lfc < 0 ~ "down",C9orf72.padj < 0.05 & C9orf72.lfc > 0 ~ "up", TRUE ~ ""))) %>%
  full_join(mutate(select(ipsc_mn_sod1_datasets$irfinder, Intron.GeneName.GeneID.Coords, SOD1.lfc = log2FoldChange, SOD1.padj = padj), SOD1.change = case_when(SOD1.padj < 0.05 & SOD1.lfc < 0 ~ "down",SOD1.padj < 0.05  & SOD1.lfc > 0 ~ "up", TRUE ~ ""))) %>%
  full_join(mutate(select(ipsc_mn_fus_datasets$irfinder, Intron.GeneName.GeneID.Coords, FUS.lfc = log2FoldChange, FUS.padj = padj), FUS.change = case_when(FUS.padj < 0.05 & FUS.lfc < 0 ~ "down",FUS.padj < 0.05 & FUS.lfc > 0 ~ "up", TRUE ~ ""))) %>%
  full_join(mutate(select(ipsc_mn_tardbp_datasets$irfinder, Intron.GeneName.GeneID.Coords, TARDBP.lfc = log2FoldChange, TARDBP.padj = padj), TARDBP.change = case_when(TARDBP.padj < 0.05 & TARDBP.lfc < 0 ~ "down",TARDBP.padj < 0.05 & TARDBP.lfc > 0 ~ "up", TRUE ~ ""))) %>%
  filter(!(Sporadic.change == "" & C9orf72.change == "" & SOD1.change == "" & FUS.change == "" & TARDBP.change == "")) 
ipsc_mn_mutations_irfinder.sig %>% count(Sporadic.change, C9orf72.change, SOD1.change, FUS.change, TARDBP.change) %>% print(n=Inf)
write_csv(ipsc_mn_mutations_irfinder.sig, here(proj_path,"splicing/irfinder/ipsc_mn_mutations_irfinder.sig.csv"))



```






# *** ADDITIONAL UNUSED ANALYSES ***



