---
title: "Meta-analysis of ALS iPSNs"
font-family: 'Helvetica'
author: |
  | Dr Oliver Ziff
  | The Francis Crick Institute
  | Institute of Neurology
  | University College London, UK
date: "`r Sys.Date()`"
output: 
 rmarkdown::html_document:
   theme: spacelab
   highlight: haddock
   code_folding: hide
   toc: true
   toc_float: true
   smooth_scroll: true
   number_sections: false
   self_contained: true  
knit: (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(dirname(inputFile), 'docs/index.html')) })
---

# Setup

```{r setup, include=FALSE}
# .libPaths("/Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library") # .libPaths("/Library/Frameworks/R.framework/Versions/4.1/Resources/library")
library(here)
proj_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/ipsc-mn-als-meta")
load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.metadata.RData")) # load metadata before functions to avoid overwriting with old functions

# Mac
camp_path = here("/Volumes/lab-patanir/home/users/ziffo")
proj_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/ipsc-mn-als-meta")
script_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta")
shared_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working")
collab_path = here("/Volumes/lab-patanir/home/users/ziffo/patani-collab")
nygc_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")

## OnDemand
# proj_path = here("/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta")
# script_path = here(here(proj_path,"scripts/ipsc_mn_als_meta")
# shared_path = here("/camp/project/proj-luscombn-patani/working")
# collab_path = here("/camp/lab/patanir/home/users/ziffo/patani-collab")
setwd(proj_path)
set_here(proj_path)
here(proj_path)
# here::i_am("ipsc_mn_als_meta.Rmd")
here()
dr_here()
source(here(camp_path,"scripts/functions/R_workspace.R"))

# load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.RData") # 6.7GB > crashes R
ipsc_mn_als_datasets_with_lee_ipsc = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee_ipsc.rds")) # includes neurolincs iPSCs
ipsc_mn_als_datasets_with_lee = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.rds")) #2.56GB # includes Lee et al samples
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.rds"))
ipsc_mn_als_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.rds")) # 1.26G
ipsc_mn_als_datasets.polyA = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.rds"))
ipsc_mn_als_datasets.total = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.rds"))
# ipsc_mn_als_datasets.noiso = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.noiso.rds"))

load(file = here(proj_path, "expression/deseq2/ipsc_mn_genetic_groups.RData"))
# ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
# ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds")) # 84M
# ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds")) # 162M
# ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
# ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds")) # 85M

load(file = here(proj_path, "expression/deseq2/nygc_postmortem_spinal_cord.RData"))
# nygc_postmortem_spinal_cord = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.tissue.rds"))
# nygc_postmortem_spinal_cord.c9orf72_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.c9orf72_vs_ctrl.rds"))
# nygc_postmortem_spinal_cord.fus_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.fus_vs_ctrl.rds"))
# nygc_postmortem_spinal_cord.sod1_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sod1_vs_ctrl.rds"))
# nygc_postmortem_spinal_cord.sporadic_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sporadic_vs_ctrl.rds"))

# ipsc_mn_als_datasets.res=ipsc_mn_als_datasets$res
# ipsc_mn_sporadic_datasets.res=ipsc_mn_sporadic_datasets$res
# ipsc_mn_fus_datasets.res=ipsc_mn_fus_datasets$res
# ipsc_mn_sod1_datasets.res=ipsc_mn_sod1_datasets$res
# ipsc_mn_c9orf72_datasets.res=ipsc_mn_c9orf72_datasets$res
# ipsc_mn_tardbp_datasets.res=ipsc_mn_tardbp_datasets$res
# nygc_postmortem_spinal_cord.res=nygc_postmortem_spinal_cord$res
# 
# save(ipsc_mn_als_datasets.res, ipsc_mn_sporadic_datasets.res, ipsc_mn_fus_datasets.res, ipsc_mn_sod1_datasets.res, ipsc_mn_c9orf72_datasets.res, ipsc_mn_tardbp_datasets.res, nygc_postmortem_spinal_cord.res, file = here(proj_path,"expression/deseq2/shiny_res.RData"))

nygc_postmortem_spinal_cord %>% filter(gene_name == "SMPD1")

```

## Update git / github

```{r}
library(here)
proj_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/ipsc-mn-als-meta")
library(usethis) # usethis & gert https://jeroen.github.io/gert2019 https://www.garrickadenbuie.com/blog/pull-request-flow-usethis/?interactive=1&steps=
use_git_config(user.name = "ojziff", user.email = "oliver.ziff@crick.ac.uk")
# usethis::create_github_token()
usethis::edit_r_environ()
credentials::set_github_pat("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") 
# make github
library(gert)
setwd(here(proj_path,"scripts/als_genome_instability")) # setwd then dont need to specify repo argument
git_commit_all("update") # stage all files & commit in one command - implies git_add(".")
pr_init("figures_and_tables") # initialise a new branch
pr_push() # this will open github in chrome > pull request > merge

git_sitrep() # print git and github info
git_config()
git_info()
git_ls() # list git tree
git_log() # check recent commits
git_status() # check which files are modified
# pr_resume("figures_and_tables") # resume a branch
# git_add(".") #stage the files you want to commmit - "." indicates all files.
# git_commit("update") # commit with comment
# pr_pull()

### Shiny objects
saveRDS(ipsc_mn_als_datasets$dds, here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_dds.rds"))

```



## Metadata

```{r}
ipsc_mn_als_datasets.metadata %>% count(dataset)
ipsc_mn_als_datasets.metadata %>% count(condition)
ipsc_mn_als_datasets.metadata %>% count(mutation)
ipsc_mn_als_datasets.metadata %>% distinct(dataset) #16 (14)
ipsc_mn_als_datasets.metadata %>% distinct(dataset_sample) # 419
ipsc_mn_als_datasets.metadata %>% count(condition) #318 ALS, 101 ctrl
ipsc_mn_als_datasets.metadata %>% count(dataset, condition) %>% print(n=Inf)
ipsc_mn_als_datasets.metadata %>% count(mutation) %>% arrange(-n)
ipsc_mn_als_datasets.metadata %>% count(mutation) # 11
ipsc_mn_als_datasets.metadata %>% count(dataset, DIV)

ipsc_mn_als_datasets.metadata %>% filter(condition == "ctrl") %>% count(mutation)
ipsc_mn_als_datasets.metadata %>% filter(mutation == "iso") %>% count(condition)
ipsc_mn_als_datasets.metadata %>% filter(condition == "ctrl") %>% count(mutation)

ipsc_mn_als_datasets.introns_tag_pct$res %>% filter(padj < 0.05) #111
ipsc_mn_als_datasets.introns_tag_pct$irfinder %>% filter(padj < 0.05) #118,553
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) # 38
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) # 25
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) # 74
ipsc_mn_als_datasets.total$irfinder %>% filter(padj < 0.05) #118829

ipsc_mn_als_datasets$dds %>% colData %>% as_tibble %>% glimpse
ipsc_mn_als_datasets$dds %>% colData %>% as_tibble %>% my_summarise(samtools_mqc_generalstats_samtools_raw_total_sequences)


# Answer ALS metadata
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_dataportal_datatable.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_participants.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_released_files.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/Clinical_Dictionary.csv")) %>% clean_names %>% glimpse
first_last_visit_alsfrs.answerals = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/answerals_progression_alsfrs.csv")) # answerals progression based on ALSFRS
AALSHXFX.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSHXFX.csv")) %>% clean_names %>% select(participant_id, subject_uid, onsetdt)
Demographics.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Demographics.csv")) %>% clean_names %>% select(participant_id, subject_uid, dob)
age_onset = AALSHXFX.csv %>% full_join(Demographics.csv) %>% mutate(age_onset = as.numeric((ddays(onsetdt) - ddays(dob)) / (364.25*24*60*60)), sample = gsub("-","_",participant_id)) %>% select(sample, age_onset) %>%
  mutate(early_late_onset = case_when(age_onset < 57 ~ "early", age_onset > 57 ~ "late"))
Mortality.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Mortality.csv")) %>% clean_names %>% filter(!dieddt %in% c("Glioblastoma","Prostate cancer", "physician assisted suicide")) %>% select(participant_id, subject_uid, dieddt) 
mortality = AALSHXFX.csv %>% full_join(Mortality.csv) %>% mutate(survival = as.numeric((ddays(dieddt) - ddays(onsetdt)) / (364.25*24*60*60)), sample = gsub("-","_",participant_id)) %>% select(sample, survival) %>%
  mutate(early_late_death = case_when(survival < 2.8 ~ "early", survival > 2.8 ~ "late"))
onset_site = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSHXFX.csv")) %>% clean_names %>% 
                        select(participant_id, subject_uid, hxblb, hxax, hxgen, hxli, hxot) %>% filter(!(hxblb == 0 & hxax == 0 & hxgen == 0 & hxli == 0 & hxot == 0)) %>%
  mutate(onset_site_detailed = case_when(hxblb == 1 & hxax == 0 & hxgen == 0 & hxli == 0 & hxot == 0 ~ "bulbar", hxblb == 0 & hxax == 0 & hxgen == 0 & hxli == 1 & hxot == 0 ~ "limb", hxblb == 0 & hxax == 0 & hxgen == 1 & hxli == 0 & hxot == 0 ~ "generalised", 
                                hxblb == 0 & hxax == 1 & hxgen == 0 & hxli == 0 & hxot == 0 ~ "axial", TRUE ~ "mixed"), #hxot == 1 ~ "other", 
         onset_site = case_when(onset_site_detailed %in% c("axial","mixed","generalised") ~ "other", TRUE ~ onset_site_detailed),
         bulbar_other = case_when(hxblb == 1 ~ "bulbar", TRUE ~ "other"), limb_other = case_when(hxli == 1 ~ "limb", TRUE ~ "other"), axial_other = case_when(hxax == 1 ~ "axial", TRUE ~ "other"), 
         bulbar_axial = case_when(hxblb == 1 ~ "bulbar", hxax == 1 ~ "axial", TRUE ~ "other"), bulbar_limb = case_when(hxblb == 1 ~ "bulbar", hxli == 1 ~ "limb", TRUE ~ "other"), bulbar_mixed = case_when(hxblb == 1 ~ "bulbar", onset_site == "mixed" ~ "mixed", TRUE ~ "other"), 
         limb_axial = case_when(hxli == 1 ~ "limb", hxax == 1 ~ "axial", TRUE ~ "other"), limb_mixed = case_when(onset_site == "mixed" ~ "mixed", hxli == 1 ~ "limb", TRUE ~ "other"), axial_mixed = case_when(hxax == 1 ~ "axial", onset_site == "mixed" ~ "mixed", TRUE ~ "other"),
         sample = gsub("-","_",participant_id)) %>% select(sample, subject_uid, onset_site, onset_site_detailed, bulbar_other,limb_other,axial_other,bulbar_axial,bulbar_limb,bulbar_mixed,limb_axial,limb_mixed,axial_mixed)
family_history = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Family_History_Log.csv")) %>% clean_names %>% select(participant_id, subject_uid, fhals, fhftd) %>% 
  mutate(family_history = case_when(fhals == 1 | fhftd == 1 ~ "yes", TRUE ~ "no"),sample = gsub("-","_",participant_id)) %>% select(sample, family_history) %>% distinct(sample, .keep_all = TRUE)
subjects = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/subjects.csv")) %>% clean_names %>% 
  mutate(mnd_group = case_when(subject_group_id == 17 ~ "other_mnd", subject_group_id == 1 ~ "als", subject_group_id == 5 ~ "healthy", subject_group_id == 11 ~ "carrier"), mnd_group = factor(mnd_group, levels = c("healthy", "carrier","other_mnd","als")), sample = gsub("-","_",participant_id)) %>% select(sample, mnd_group) %>% distinct(sample, .keep_all = TRUE)
ck_value = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Auxiliary_Chemistry.csv")) %>% clean_names %>% drop_na(acckrslt) %>% mutate(sample = gsub("-","_",participant_id), cknorm = case_when(cknorm == 1 ~ "normal", TRUE ~ "raised")) %>% arrange(-acckrslt) %>% distinct(sample, .keep_all = TRUE) %>% select(sample, ck = acckrslt, cknorm)
AALSHXFX.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSDXFX.csv")) %>% clean_names %>% drop_na(elescrlr) %>% select(participant_id, elescrlr) %>% 
  mutate(sample = gsub("-","_",participant_id)) %>% select(sample, elescrlr) %>% arrange(elescrlr) %>% distinct(sample, .keep_all = TRUE)
answerals_transcriptomic_files = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_released_files.csv")) %>% clean_names %>% filter(omic == "transcriptomics", grepl(".fastq.gz$", path)) %>% 
  mutate(fastq = case_when(grepl("_1.fastq.gz$", path) ~ "fastq_1", grepl("_2.fastq.gz$", path) ~ "fastq_2")) %>% 
  filter(participant_id != "CTRL-NEUEU392AE8") %>% #17 fastq's associated with this ID. Unclear if these are technical repeats or incorrectly labelled.
  pivot_wider(c(participant_id, experiment), names_from = fastq, values_from = c(path,size)) %>% 
  mutate(fastq_1 = paste0(here(camp_path,"proj-luscombn-patani/working/answerals/reads/"),path_fastq_1), fastq_2 = paste0(here(camp_path,"proj-luscombn-patani/working/answerals/reads/"), path_fastq_2))
samplesheet = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_dataportal_datatable.csv")) %>% clean_names %>% filter(transcriptomics == "Yes", participant_id != "CTRL-NEUEU392AE8") %>% # ipsc == "Yes", 
  left_join(answerals_transcriptomic_files) %>% 
  mutate(strandedness = "reverse", 
         condition = case_when(subject_group == "ALS" ~ "als", subject_group == "Healthy Control" ~ "ctrl", grepl("carrier",subject_group) ~ "carrier", subject_group == "Non-ALS MND" ~ "other_mnd"),
         mutation = case_when(self_reported_mutations %in% c("C9orf72","SOD1","SETX", "TARDBP (TDP43)") ~ self_reported_mutations, TRUE ~ wgs_variants), mutation = tolower(gsub(" [(]WGS[)]","",mutation)), mutation = gsub("tardbp [(]tdp43[)]","tardbp", mutation),  
         mutation = case_when(condition == "als" & is.na(mutation) ~ "sporadic", condition == "ctrl" & is.na(mutation) ~ "ctrl",TRUE ~ mutation), mutant = case_when(mutation %in% c("sporadic","ctrl") ~ "no", TRUE ~ "yes"),
         sample = gsub("-","_",participant_id), DIV = 32, instrument = "Illumina NovaSeq 6000") %>%
  select(sample, fastq_1, fastq_2, strandedness, condition, mutation, mutant, participant_id, guid, nygc_cgnd_id, sex, DIV, instrument, ethnicity, race, pls_subgroup, revised_el_escorial_criteria, age_at_sample_collection, age_at_symptom_onset, age_at_death, alsfrs_r_baseline, alsfrs_r_latest, alsfrs_r_progression_slope, cbs_baseline, cbs_latest, cbs_progression_slope, subject_group, ipsc, dimn, i_psc_line_id_root, i_psc_line_name, di_m_ns_line_name, batch_id, self_reported_mutations, wgs_variants, repeat_length_c9, repeat_length_atxn2, size_fastq_1, size_fastq_2) %>%
  filter(!sample %in% c("CASE_NEUMN922PMZ","CASE_NEUYC198DCR")) %>% # R2_fastq.gz corrupt crashing nfcore
  left_join(select(first_last_visit_alsfrs.answerals, sample, alsfrs_slope, progression)) %>% left_join(age_onset) %>% left_join(mortality) %>% left_join(onset_site) %>% left_join(family_history) %>% left_join(subjects) %>% left_join(ck_value) %>% left_join(AALSHXFX.csv) # add clinical data to metadata
samplesheet # Ensure no NA's in sample sheet
samplesheet %>% glimpse
samplesheet %>% distinct(participant_id) # 287
write_csv(samplesheet, file = here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/samplesheet.csv"), na = "")

```


# Table S2 QC stats

Table S1 is iPSN differentiation protocols

Multiqc general stats

```{r}
multiqc_stat_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), multiqc_stat_paths = gsub("/camp/lab/patanir/home/users/ziffo|/camp/project","/Volumes/lab-patanir/home/users/ziffo",multiqc_stat_paths)) %>% pull(multiqc_stat_paths)
multiqc_stats.tsv = multiqc_stat_paths %>% map(read_tsv, show_col_types = FALSE)                                                                        
names(multiqc_stats.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(dataset)
multiqc_data <- map_dfr(multiqc_stats.tsv, bind_rows, .id = "dataset") %>% clean_names() %>% drop_na(samtools_mqc_generalstats_samtools_flagstat_total) %>% filter(sample %in% ipsc_mn_als_datasets_with_lee.metadata$sample) %>%
  mutate(dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincs0" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS", dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~ dataset)) %>%
  select(dataset, sample, raw_total_sequences = samtools_mqc_generalstats_samtools_raw_total_sequences, flagstat_total = samtools_mqc_generalstats_samtools_flagstat_total, mapped_passed = samtools_mqc_generalstats_samtools_mapped_passed, reads_mapped = samtools_mqc_generalstats_samtools_reads_mapped, reads_mapped_percent = samtools_mqc_generalstats_samtools_reads_mapped_percent, reads_aligned = quali_map_mqc_generalstats_qualimap_reads_aligned, r_rna_percent = biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna, bias_5_3 = quali_map_mqc_generalstats_qualimap_5_3_bias, proper_pairs_percent = r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent, reads_assigned_percent = feature_counts_mqc_generalstats_featurecounts_percent_assigned, reads_assigned = feature_counts_mqc_generalstats_featurecounts_assigned, percent_duplication = picard_mqc_generalstats_picard_percent_duplication, error_rate = samtools_mqc_generalstats_samtools_error_rate, non_primary_alignments = samtools_mqc_generalstats_samtools_non_primary_alignments, reads_properly_paired_percent = samtools_mqc_generalstats_samtools_reads_properly_paired_percent, percent_mapped = salmon_mqc_generalstats_salmon_percent_mapped, num_mapped = salmon_mqc_generalstats_salmon_num_mapped, uniquely_mapped_percent = star_mqc_generalstats_star_uniquely_mapped_percent, uniquely_mapped = star_mqc_generalstats_star_uniquely_mapped, percent_trimmed =cutadapt_mqc_generalstats_cutadapt_percent_trimmed, total_sequences = fast_qc_mqc_generalstats_fastqc_total_sequences, raw_percent_gc = fast_qc_raw_mqc_generalstats_fastqc_raw_percent_gc, raw_avg_sequence_length = fast_qc_raw_mqc_generalstats_fastqc_raw_avg_sequence_length, trimmed_percent_duplicates = fast_qc_trimmed_mqc_generalstats_fastqc_trimmed_percent_duplicates, trimmed_percent_gc = fast_qc_trimmed_mqc_generalstats_fastqc_trimmed_percent_gc)
multiqc_data %>% glimpse
write_csv(multiqc_data, here(proj_path,"alignment/multiqc_general_stats.csv"))

```


# Fig 1 Pipeline schematic

## PCA

```{r}
ipsc_mn_als_datasets_with_lee.pcaData <- plotPCA(ipsc_mn_als_datasets_with_lee$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","sareen","sommer","smith","sterneckert","wang","lee.CTRL") ~ "poly(A)", TRUE ~ "Total Ribo-Zero"), #dataset = gsub("\\."," ",dataset), #dataset = str_to_title(dataset),
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/patanir/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         ) %>%
  left_join(select(ipsc_mn_als_datasets_with_lee.metadata, dataset_sample, gender))

# Colour by dataset
colourCount = length(unique(mtcars$hp))
getPalette = colorRampPalette(brewer.pal(8, "Accent"))
getPalette(colourCount)
show_col(getPalette(colourCount))
#7FC97F" "#94C09B" "#A9B7B7" "#BEAED4" "#D3B4BA" "#E8BAA0" "#FDC086" "#FDD58C" "#FEEA92" "#FFFF99" "#BCCEA0" "#7A9DA8" 
# "#386CB0" "#75489F" "#B2258F" "#F0027F" "#DF1F5C" "#CF3D39" "#BF5B17" "#A15E31" "#83624B" "#666666"

# Colour by dataset
ipsc_mn_als_datasets_with_lee.pca_dataset <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, colour=dataset)) + geom_point(size=1) +
  # scale_color_manual(values = c("AnswerALS"="#94C09B", "neurolincs.iMN"="#A9B7B7", "neurolincs.diMN"="#BEAED4", "bhinge"= "#7FC97F","catanese"="#7FC97F", "dafinca"="#94C09B", "desantis"="#FEEA92", "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "hawkins"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sommer"="#FFFF99", "sterneckert"="#7A9DA8", "wang"="#BCCEA0","lee.CTRL"="#75489F", "lee.ALS"="#386CB0")) +
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "right", legend.spacing = unit(0.01, 'cm')) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) + #coord_fixed() +
  # ggforce::geom_mark_ellipse(aes(label = library_type), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE) + 
  guides(color=guide_legend(title="Dataset", keywidth=0.1, keyheight=0.15, default.unit="inch"))
ipsc_mn_als_datasets_with_lee.pca_dataset

```

## Cell type marker heatmap

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","EDNRB",  "FGFR3",  "SLC4A4", "DIO2")#"SOX9", ,"MLC1"
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4", "PECAM1")
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CCL3","IL1A", "TREM2") # "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "CX3CR1","P2RY12", "TNF",  "TMEM119"
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1")

undifferentiated = c("SOX2","POU5F1","UTF1","ZFP42","ALPL","TERT","ESRG","CNMD","SFRP2")
neural.precursor = c("NES","SOX1","SOX2","PROM1","PAX6","NOTCH1","OCLN","CDH1","CDH2","SOX10","VIM","GFAP")
neural.progenitor = c("MAP2","HUC","HUD","NF","NCAM","DCX","SOX2")
neural.crest <- c("SOX10","NGFR","FOXD3","SNAI1","SNAI2","LMX1A")   #NGFR very low in HB9 samples
dorsal.progenitor <- c("ATOH1","OLIG3","MSX1","PAX3","PAX7","DBX1","DBX2","IRX3","ASCL1")#"IRX5","PAX6","GBX2","GSX2","GSX1","SP8"
dorsal.progenitor = c("IRX3","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
nonspecific = c("SOX2")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
pMN.domain = c("NKX6-1", "OLIG2")
ventral.progenitor = c("NKX2-2","NKX2-8")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")

ho.genes = c("SST","DCX","RFX4","ASCL1","HOXB8","NEFH","SCN1A","TNS1","SPOCK3","SNAP25","TDGF1","POU5F1","L1TD1","FXYD5","PHLDA2","TOP2A","DLGAP5","FZD7","ILF2","MSH2")

nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

# neuronal.markers = c("NPAS4", "NPY", "SST", "CCK", "GAD1", "SYT1", "STMN2", "SYN1") # "ENO2", 

nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#"NEFH",
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3") #"ETV1", "NOS1",,"RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2"
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

### Cell type markers
celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, oligo.markers, mn.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]
celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte.markers", gene_name %in% endo.markers ~ "endo.markers", 
                           gene_name %in% myeloid.markers ~ "myeloid.markers", gene_name %in% oligo.markers ~ "oligo.markers", 
                           gene_name %in% mn.markers ~ "mn.markers"), 
         group = factor(group, levels = c("astrocyte.markers", "endo.markers", "myeloid.markers", "oligo.markers","mn.markers"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets_with_lee.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = as.factor(dataset)) %>% filter(!grepl("answerals_CASE",dataset_sample))

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
celltype.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))

### Cluster samples ###
celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(celltype.markers.marker.centered_mat)]

celltype.markers.rnaseq.cluster.heatmap <- Heatmap(t(celltype.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = celltype.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = celltype.markers.df.filt$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelial", "Myeloid", "Oligodendrocyte","Motor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), column_title = NULL, 
          show_row_names = FALSE, #row_names_gp = gpar(fontsize = 6), 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)),  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
celltype.markers.rnaseq.cluster.heatmap
ggsave(as.ggplot(celltype.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.celltype.markers.heatmap.png"), width = 8, height = 13)
```

## Merge

```{r}
ipsn_identities_pca_heatmap = plot_grid(ipsc_mn_als_datasets_with_lee.pca_dataset, NULL,
  as.ggplot(celltype.markers.rnaseq.cluster.heatmap),
  nrow = 3, rel_heights = c(1.5,0.1,4), labels = c("a","b",""))
# ipsn_identities_pca_heatmap
ggsave(ipsn_identities_pca_heatmap, filename = here(proj_path,"expression/deseq2/ipsn_identities_pca_heatmap.png"), width = 8.25, height = 12)

```

## Barplot genetic subtypes

```{r}
ipsc_mn_als_datasets.genetic.n = ipsc_mn_als_datasets.metadata %>% count(mutation) %>% filter(mutation != "iso") %>% mutate(n = gsub(89,106,n),n=as.integer(n), mutation = toupper(mutation), mutation = gsub("ORF","orf", mutation), mutation = gsub("SPORADIC","Sporadic",mutation), mutation = factor(mutation, levels = rev(c("Sporadic", "C9orf72", "SOD1", "FUS","TARDBP","VCP","SQSTM1", "PFN1" ,"SETX", "CHCHD1", "FIG4","CTRL"))))
ipsc_mn_als_datasets.genetic.n

ipsc_mn_als_datasets.genetic.n.barplot <- ggplot(ipsc_mn_als_datasets.genetic.n, aes(x=n, y=mutation, fill=mutation)) +
  geom_bar(stat="identity", position = position_dodge()) +
  # scale_fill_npg() +#  
  # scale_fill_brewer(palette = "Paired") + 
  scale_fill_manual(values = c("Sporadic"="#CF3D39", "C9orf72"="#7FC97F", "SOD1"="#386CB0", "FUS"="#75489F","TARDBP"="#83624B","VCP"="#94C09B","SQSTM1"="#BEAED4", "PFN1"="#FDC086" ,"SETX"="#7A9DA8", "CHCHD1"="#A9B7B7", "FIG4"="#D3B4BA", "CTRL"="#CC87AF")) +
  theme_classic() +  theme(axis.line = element_blank(), axis.text.y = element_text(size=10), axis.text.x = element_blank(), axis.ticks = element_blank(), panel.grid = element_blank(), legend.position = "none", plot.margin = unit(c(0,0,-0.6,-0.2), "cm")) + 
  expand_limits(x = 230) + #, 
  geom_text(aes(label=n), hjust=-0.2, size=3.8)+
  labs(y = "", x = "")
ipsc_mn_als_datasets.genetic.n.barplot
ggsave(ipsc_mn_als_datasets.genetic.n.barplot, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.genetic.n.barplot.png"), width = 3.5, height = 4)



# Colour by dataset
colourCount = length(unique(mtcars$hp))
getPalette = colorRampPalette(brewer.pal(8, "Accent"))
getPalette(colourCount)
show_col(getPalette(colourCount))
#7FC97F" "#94C09B" "#A9B7B7" "#BEAED4" "#D3B4BA" "#E8BAA0" "#FDC086" "#FDD58C" "#FEEA92" "#FFFF99" "#BCCEA0" "#7A9DA8" 
# "#386CB0" "#75489F" "#B2258F" "#F0027F" "#DF1F5C" "#CF3D39" "#BF5B17" "#A15E31" "#83624B" "#666666"

```


# Fig S2-3 PCA

Fig S1 is the PRISMA flow diagram

## PC1 & PC2 gene loadings

https://stackoverflow.com/questions/12760108/principal-components-analysis-how-to-get-the-contribution-of-each-paramete
https://github.com/mikelove/DESeq2/blob/master/R/plots.R

```{r}
ipsc_mn_als_datasets_with_lee.vsd.rv <- rowVars(assay(ipsc_mn_als_datasets_with_lee$vsd)) # calculate the variance for each gene
ipsc_mn_als_datasets_with_lee.vsd.ntop_genes <- order(ipsc_mn_als_datasets_with_lee.vsd.rv, decreasing=TRUE)[seq_len(min(500, length(ipsc_mn_als_datasets_with_lee.vsd.rv)))] # select the ntop genes by variance
ipsc_mn_als_datasets_with_lee.vsd.pca <- prcomp(t(assay(ipsc_mn_als_datasets_with_lee$vsd)[ipsc_mn_als_datasets_with_lee.vsd.ntop_genes,])) # perform a PCA on the data in assay(x) for the selected genes
ipsc_mn_als_datasets_with_lee.vsd.loadings <- as_tibble(ipsc_mn_als_datasets_with_lee.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
ipsc_mn_als_datasets_with_lee.vsd.loadings %>% arrange(PC1)
ipsc_mn_als_datasets_with_lee.vsd.loadings %>% arrange(-PC1)

# ipsc_mn_als_datasets_with_lee.vsd.pca <- prcomp(t(assay(ipsc_mn_als_datasets_with_lee$vsd))) # perform a PCA on the data in assay(x) for the selected genes
# ipsc_mn_als_datasets_with_lee.vsd.loadings <- as_tibble(ipsc_mn_als_datasets_with_lee.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2")#,"ARHGAP36")
V3.domain = c("NKX2-2","SIM1")
HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
neuronal.markers = c(nonspecific.neuronal, dl.domains, V0.domain,V1, V2a.domain, V2b.domain,MN.domain,V3.domain)
nonpolyadenylated_genes = read_csv(here(camp_path,"genomes/annotation/nonpolyadenylated_genes.csv")) %>% pull(nonpolyadenylated_genes)

# scatter all PC1 vs PC2
ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter = ggplot(ipsc_mn_als_datasets_with_lee.vsd.loadings, aes(x = PC1, y = PC2 )) + 
    labs(x = "PC1 gene loading", y = "PC2 gene loading") +
    geom_hline(yintercept = 0, linetype = 3) + geom_vline(xintercept = 0, linetype = 3) +
    geom_point(size = 1, alpha = 0.1) +
    scale_fill_continuous(type = "viridis") +
    guides(fill = FALSE) +
    theme_oz() +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings, abs(PC1) > 0.04, abs(PC2) > 0.04, grepl("^H[1-9]",gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings, abs(PC1) > 0.04, abs(PC2) > 0.04, grepl("^SNOR|RN7SK|RMRP|RPPH1",gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings,  gene_name %in% c(neuronal.markers, HOX.markers)), aes(x = PC1, y = PC2, label = gene_name), colour = "black", size = 2.3) +
  annotate("text", x = 0.04, y = 0.05, label = "PC1+ PC2+",size = 3, color = "black") + annotate("text", x = -0.07, y = -0.12, label = "PC1- PC2-", size = 3, color = "firebrick2") + 
  annotate("text", x = 0.04, y = -0.12, label = "PC1+ PC2-", size = 3, color = "black") + annotate("text", x = -0.07, y = 0.05, label = "PC1- PC2+", size = 3, color = "black")
ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter
# ggsave(ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter.png"), height = 9, width = 9)

```

## Sample characteristics

```{r}
# ipsc_mn_als_datasets_with_lee = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.rds")) #2.56GB
ipsc_mn_als_datasets_with_lee.pcaData <- plotPCA(ipsc_mn_als_datasets_with_lee$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/patanir/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang","lee.CTRL") ~ "poly(A)", TRUE ~ "Total Ribo-Zero")) %>%
  left_join(select(ipsc_mn_als_datasets_with_lee.metadata, dataset_sample, gender)) #%>% 
  # mutate(database_dir = gsub("/camp/lab/patanir/home/users/ziffo|/camp/project", "/Volumes/lab-patanir/home/users/ziffo", database_dir))
  
# multiqc stats
multiqc_general_stats_stat_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_general_stats_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), 
     multiqc_general_stats_stat_paths = gsub("/camp/lab/patanir/home/users/ziffo|/camp/project","/Volumes/lab-patanir/home/users/ziffo",multiqc_general_stats_stat_paths)) %>% pull(multiqc_general_stats_stat_paths)
multiqc_general_stats.tsv = multiqc_general_stats_stat_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_general_stats.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_general_stats <- map_dfr(multiqc_general_stats.tsv, bind_rows, .id = "database_dir") %>% clean_names()
multiqc_rseqc_read_distribution_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_rseqc_read_distribution_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"),
     multiqc_rseqc_read_distribution_paths = gsub("/camp/lab/patanir/home/users/ziffo|/camp/project","/Volumes/lab-patanir/home/users/ziffo",multiqc_rseqc_read_distribution_paths)) %>% pull(multiqc_rseqc_read_distribution_paths)
multiqc_rseqc_read_distribution.tsv = multiqc_rseqc_read_distribution_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_rseqc_read_distribution.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_rseqc_read_distribution <- map_dfr(multiqc_rseqc_read_distribution.tsv, bind_rows, .id = "database_dir") %>% clean_names()
ipsc_mn_als_datasets_with_lee.pcaData = ipsc_mn_als_datasets_with_lee.pcaData %>% left_join(multiqc_general_stats) %>% left_join(multiqc_rseqc_read_distribution) # need unique sample names - If multiple dataset names within a single multiqc_general_stats.txt then will only join the first dataset name

# Colour by dataset
colourCount = length(unique(mtcars$hp))
getPalette = colorRampPalette(brewer.pal(8, "Accent"))
getPalette(colourCount)
show_col(getPalette(colourCount))
#7FC97F" "#94C09B" "#A9B7B7" "#BEAED4" "#D3B4BA" "#E8BAA0" "#FDC086" "#FDD58C" "#FEEA92" "#FFFF99" "#BCCEA0" "#7A9DA8" 
# "#386CB0" "#75489F" "#B2258F" "#F0027F" "#DF1F5C" "#CF3D39" "#BF5B17" "#A15E31" "#83624B" "#666666"

# Colour by dataset
ipsc_mn_als_datasets_with_lee.pca_dataset <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) +
  scale_color_manual(values = c("AnswerALS"="#94C09B", "neurolincs.iMN"="#A9B7B7", "neurolincs.diMN"="#BEAED4", "bhinge"= "#7FC97F","catanese"="#7FC97F", "dafinca"="#94C09B", "desantis"="#FEEA92", "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "hawkins"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sommer"="#FFFF99", "sterneckert"="#7A9DA8", "wang"="#BCCEA0","lee.CTRL"="#75489F", "lee.ALS"="#386CB0")) +
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "right", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) #+ #coord_fixed() +
  # ggforce::geom_mark_ellipse(aes(label = library_type), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE) + guides(color=guide_legend(title="Dataset", keywidth=0.1, keyheight=0.15, default.unit="inch"))
ipsc_mn_als_datasets_with_lee.pca_dataset

# Colour by library_type
ipsc_mn_als_datasets_with_lee.pca_library_type <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=library_type)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))+labs(colour="Library preparation")
ipsc_mn_als_datasets_with_lee.pca_library_type
# Colour by instrument
ipsc_mn_als_datasets_with_lee.pca_instrument <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=instrument)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_blank(), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_instrument
# Colour by DIV
ipsc_mn_als_datasets_with_lee.pca_DIV <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=DIV)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top"))  + labs(color="Days in vitro") + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_DIV
# Colour by ALS
ipsc_mn_als_datasets_with_lee.pca_als <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=condition)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_als
# Colour by mutation
ipsc_mn_als_datasets_with_lee.pca_mutation <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=mutation)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_mutation
# Colour by gender
ipsc_mn_als_datasets_with_lee.pca_gender <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=gender)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets_with_lee.pca_gender

```

Fig S2 Merge

```{r}
ipsc_mn_als_datasets_with_lee.pca_merged = plot_grid(
  ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter, ipsc_mn_als_datasets_with_lee.pca_dataset,
  ipsc_mn_als_datasets_with_lee.pca_library_type, ipsc_mn_als_datasets_with_lee.pca_instrument, #ncol = 2, labels = c("b","c")), 
  ipsc_mn_als_datasets_with_lee.pca_gender, ipsc_mn_als_datasets_with_lee.pca_DIV, #ncol = 2, labels = c("d","e")), 
  ipsc_mn_als_datasets_with_lee.pca_als, ipsc_mn_als_datasets_with_lee.pca_mutation, #ncol = 2, labels = c("f","g")),
  nrow = 4, ncol = 2, labels = c("auto"))
# ipsc_mn_als_datasets_with_lee.pca_merged
ggsave(ipsc_mn_als_datasets_with_lee.pca_merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.pca_merged.png"), height = 11.75, width = 8.25)
```

## RNAseq QC metrics

Fig S3

```{r}
### PCA 2

# Colour by strandedness
ipsc_mn_als_datasets_with_lee.pca_strandedness <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=strandedness)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.2, keyheight=0.2, default.unit="inch")) + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_strandedness
# Colour by read depth
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_raw_total_sequences)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6, angle = 0), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(1, 'cm'))  + 
  guides(color = guide_colourbar(title = "read depth", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_raw_total_sequences,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences


# Color by intronic read % intron_read_pct
ipsc_mn_als_datasets_with_lee.pca_read_distribution <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=introns_tag_pct)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "intronic reads %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$introns_tag_pct,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets_with_lee.pca_read_distribution
# Colour by biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna
ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "ribosomal rna %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna
# Colour by quali_map_mqc_generalstats_qualimap_5_3_bias
ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=quali_map_mqc_generalstats_qualimap_5_3_bias)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "5' 3' bias", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=1, low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias
# Colour by quali_map_mqc_generalstats_qualimap_reads_aligned
ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_reads_aligned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=quali_map_mqc_generalstats_qualimap_reads_aligned)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads aligned", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$quali_map_mqc_generalstats_qualimap_reads_aligned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_reads_aligned
# Colour by picard_mqc_generalstats_picard_percent_duplication
ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=picard_mqc_generalstats_picard_percent_duplication)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Read duplication %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$picard_mqc_generalstats_picard_percent_duplication,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +  
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication
# Colour by samtools_mqc_generalstats_samtools_reads_mapped
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mapped)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Reads mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped
# Colour by star_mqc_generalstats_star_uniquely_mapped_percent
ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=star_mqc_generalstats_star_uniquely_mapped_percent)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads uniquely mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$star_mqc_generalstats_star_uniquely_mapped_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent

ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged = ggarrange(
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_strandedness, ipsc_mn_als_datasets_with_lee.pca_read_distribution , ncol = 2, labels = c("a","b")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences, ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias, ncol = 2, labels = c("c","d")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna, ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication, ncol = 2, labels = c("e","f")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent, ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped, ncol = 2, labels = c("g","h")), 
  nrow = 4, heights = c(1,1,1,1))
# ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged
ggsave(ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged.png"), height = 11.75, width = 8.25)


# # Colour by samtools_mqc_generalstats_samtools_reads_mapped_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mapped_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mapped_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped_percent
# # Colour by samtools_mqc_generalstats_samtools_error_rate
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_error_rate <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_error_rate)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Error rate", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_error_rate,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_error_rate
# # Colour by samtools_mqc_generalstats_samtools_non_primary_alignments
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_non_primary_alignments <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_non_primary_alignments)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Non primary alignments", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_non_primary_alignments,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_non_primary_alignments
# # Colour by samtools_mqc_generalstats_samtools_reads_mq0_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mq0_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mq0_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Mapping quality 0 %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mq0_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mq0_percent
# # Colour by samtools_mqc_generalstats_samtools_flagstat_total
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_flagstat_total <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_flagstat_total)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_flagstat_total
# # Colour by samtools_mqc_generalstats_samtools_mapped_passed
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_mapped_passed <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_mapped_passed)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Mapped reads passed", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_mapped_passed,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_mapped_passed
# # Colour by r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent
# ipsc_mn_als_datasets_with_lee.pca_r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Proper pairs %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent
# # Colour by samtools_mqc_generalstats_samtools_reads_properly_paired_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_properly_paired_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_properly_paired_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Properly paired %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_properly_paired_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_properly_paired_percent
# # Colour by feature_counts_mqc_generalstats_featurecounts_percent_assigned
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_percent_assigned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=feature_counts_mqc_generalstats_featurecounts_percent_assigned)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Assigned reads %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$feature_counts_mqc_generalstats_featurecounts_percent_assigned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_percent_assigned
# # Colour by feature_counts_mqc_generalstats_featurecounts_assigned
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_assigned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=feature_counts_mqc_generalstats_featurecounts_assigned)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Assigned reads", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$feature_counts_mqc_generalstats_featurecounts_assigned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_assigned
# # Colour by salmon_mqc_generalstats_salmon_percent_mapped
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_percent_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=salmon_mqc_generalstats_salmon_percent_mapped)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$salmon_mqc_generalstats_salmon_percent_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_percent_mapped
# # Colour by salmon_mqc_generalstats_salmon_num_mapped
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_num_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=salmon_mqc_generalstats_salmon_num_mapped)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$salmon_mqc_generalstats_salmon_num_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_num_mapped
# # Colour by star_mqc_generalstats_star_uniquely_mapped
# ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=star_mqc_generalstats_star_uniquely_mapped)) + geom_point(size=1) + 
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
#   guides(color = guide_colourbar(title = "Reads uniquely mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$star_mqc_generalstats_star_uniquely_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# # ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped

```



# Fig S4-9 Heatmap expression

## Cell type marker heatmap

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","EDNRB",  "FGFR3",  "SLC4A4", "DIO2")#"SOX9", ,"MLC1"
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4", "PECAM1")
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CCL3","IL1A", "TREM2") # "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "CX3CR1","P2RY12", "TNF",  "TMEM119"
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1")

undifferentiated = c("SOX2","POU5F1","UTF1","ZFP42","ALPL","TERT","ESRG","CNMD","SFRP2")
neural.precursor = c("NES","SOX1","SOX2","PROM1","PAX6","NOTCH1","OCLN","CDH1","CDH2","SOX10","VIM","GFAP")
neural.progenitor = c("MAP2","HUC","HUD","NF","NCAM","DCX","SOX2")
neural.crest <- c("SOX10","NGFR","FOXD3","SNAI1","SNAI2","LMX1A")   #NGFR very low in HB9 samples
dorsal.progenitor <- c("ATOH1","OLIG3","MSX1","PAX3","PAX7","DBX1","DBX2","IRX3","ASCL1")
ho.genes = c("SST","DCX","RFX4","ASCL1","HOXB8","NEFH","SCN1A","TNS1","SPOCK3","SNAP25","TDGF1","POU5F1","L1TD1","FXYD5","PHLDA2","TOP2A","DLGAP5","FZD7","ILF2","MSH2")

nonspecific = c("SOX2")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
pMN.domain = c("NKX6-1", "OLIG2")
ventral.progenitor = c("NKX2-2","NKX2-8")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")

nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

# neuronal.markers = c("NPAS4", "NPY", "SST", "CCK", "GAD1", "SYT1", "STMN2", "SYN1") # "ENO2", 

nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#"NEFH",
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3") #"ETV1", "NOS1",,"RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2"
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

### Cell type markers
celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, oligo.markers, mn.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]
celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte.markers", gene_name %in% endo.markers ~ "endo.markers", 
                           gene_name %in% myeloid.markers ~ "myeloid.markers", gene_name %in% oligo.markers ~ "oligo.markers", 
                           gene_name %in% mn.markers ~ "mn.markers"), 
         group = factor(group, levels = c("astrocyte.markers", "endo.markers", "myeloid.markers", "oligo.markers","mn.markers"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets_with_lee.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = as.factor(dataset)) %>% filter(!grepl("answerals_CASE",dataset_sample))

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
celltype.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))

### Cluster samples ###
celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(celltype.markers.marker.centered_mat)]

celltype.markers.rnaseq.cluster.heatmap <- Heatmap(t(celltype.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = celltype.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = celltype.markers.df.filt$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelial", "Myeloid", "Oligodendrocyte","Motor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), column_title = NULL, 
          show_row_names = FALSE, #row_names_gp = gpar(fontsize = 6), 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)),  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
celltype.markers.rnaseq.cluster.heatmap
ggsave(as.ggplot(celltype.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.celltype.markers.heatmap.png"), width = 8.25, height = 9)
```


### removeBatchEffect reubttal

```{r}
mat <- assay(ipsc_mn_als_datasets_with_lee$vsd)
mm <- model.matrix(~condition, colData(ipsc_mn_als_datasets_with_lee$vsd))
mat <- limma::removeBatchEffect(mat, batch=ipsc_mn_als_datasets_with_lee$vsd$dataset, design=mm)
vsd.counts = as_tibble(mat, rownames = "gene_id") %>% left_join(gene2ens)
min_vst = min(vsd.counts[,2])
celltype.markers.marker <- vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))
celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(celltype.markers.marker.centered_mat)]
celltype.markers.rnaseq.cluster.heatmap.removeBatchEffect <- Heatmap(t(celltype.markers.marker.mat), 
                                                   border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
                                                   column_order = celltype.markers.df.filt$gene_name,
                                                   column_names_gp = gpar(fontsize = 6), 
                                                   column_split = celltype.markers.df.filt$group,
                                                   bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelial", "Myeloid", "Oligodendrocyte","Motor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), column_title = NULL, 
                                                   show_row_names = FALSE, #row_names_gp = gpar(fontsize = 6), 
                                                   # row_order = dataset.samples.df$dataset_sample,
                                                   row_split = dataset.samples.df$dataset,
                                                   right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)),  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                                                          labels_gp = gpar(fontsize = 7), labels_rot = 0)),
                                                   row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
celltype.markers.rnaseq.cluster.heatmap.removeBatchEffect
ggsave(as.ggplot(celltype.markers.rnaseq.cluster.heatmap.removeBatchEffect), filename = here(proj_path,"expression/deseq2/celltype.markers.rnaseq.cluster.heatmap.removeBatchEffect.png"), width = 8.25, height = 9)


```


## Neuron differentiation markers

```{r}
undifferentiated = c("NANOG","SOX2","POU5F1","ZFP42","ALPL","ESRG","CNMD","SFRP2")#"UTF1","TERT",
neural.precursor = c("SOX1","SOX2","PROM1","PAX6","NOTCH1","OCLN","CDH1","SOX10","GFAP")#"NES","CDH2","VIM",
neural.progenitor = c("MAP2","ELAVL3","ELAVL4","NF1","NCAM1","DCX")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
neural.crest <- c("SOX10","NGFR","FOXD3","SNAI1","SNAI2","LMX1A")   #NGFR very low in HB9 samples
dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#,"NEFH"
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3","ETV1", "NOS1","RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2") #
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

neuron.type.markers = c(undifferentiated, neural.precursor, neural.progenitor, floor.plate, roof.plate, neural.crest, dorsal.progenitor, motor.neuron.progenitor, postmitotic.motor.neuron, nonspecific.neuronal)#,interneuron.markers) 
neuron.type.markers[!neuron.type.markers %in% gtf$gene_name]
neuron.type.markers.df = tibble("gene_name" = neuron.type.markers) %>% filter(gene_name %in% neuron.type.markers) %>% 
  mutate(group = case_when(gene_name %in% undifferentiated ~ "undifferentiated", gene_name %in% neural.precursor ~ "neural.precursor", gene_name %in% neural.progenitor ~ "neural.progenitor", gene_name %in% floor.plate ~ "floor.plate", gene_name %in% roof.plate ~ "roof.plate", gene_name %in% neural.crest ~ "neural.crest", gene_name %in% dorsal.progenitor ~ "dorsal.progenitor", gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% motor.neuron.progenitor ~ "motor.neuron.progenitor", 
                           gene_name %in% postmitotic.motor.neuron ~ "postmitotic.motor.neuron"), #gene_name %in% oculomotor.markers ~ "oculomotor.markers", 
                           # gene_name %in% interneuron.markers ~ "interneuron"), 
         group = factor(group, levels = c("undifferentiated", "neural.precursor", "neural.progenitor", "floor.plate", "roof.plate", "neural.crest", "dorsal.progenitor", "nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

ipsc_mn_als_datasets_with_lee_ipsc.df = ipsc_mn_als_datasets_with_lee_ipsc.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = factor(dataset, levels = c("answerals", "bhinge", "catanese", "dafinca.c9orf72", "dafinca.tardbp", "desantis", "hawkins", "kapeli", "kiskinis", "lee", "luisier", "sareen","sommer", "smith", "sterneckert", "wang", "neurolincs.diMN", "neurolincs.iMN", "neurolincs.iPSC"))) %>% filter(!grepl("answerals_CASE",dataset_sample))

# together with other datasets
ipsc_mn_als_datasets_with_lee_ipsc$vsd.counts = as_tibble(assay(ipsc_mn_als_datasets_with_lee_ipsc$vsd), rownames = "gene_id") %>% left_join(gene2ens)
min_vst = min(ipsc_mn_als_datasets_with_lee_ipsc$vsd.counts[,2])
neuron.type.markers.marker <- ipsc_mn_als_datasets_with_lee_ipsc$vsd.counts %>% 
  filter(gene_name %in% neuron.type.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuron.type.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
neuron.type.markers.df.filt = neuron.type.markers.df %>% filter(gene_name %in% neuron.type.markers.marker$gene_name)
neuron.type.markers.marker.mat <- make_matrix(select(neuron.type.markers.marker,-gene_name), pull(neuron.type.markers.marker,gene_name))

### Cluster samples ###
neuron.type.markers.rnaseq.heatmap <- Heatmap(t(neuron.type.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = neuron.type.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = factor(neuron.type.markers.df.filt$group, levels = c("undifferentiated", "neural.precursor", "neural.progenitor", "floor.plate", "roof.plate", "neural.crest", "dorsal.progenitor", "nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron")),
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",4)), labels = c("undifferentiat.\npluripotent", "neural\nprecursor", "neural\nprogenit", "floor\nplate", "roof\nplate", "neur\ncrest", "dorsal\nprogenitor", "nonspecific\nneuronal", "MN\nprogenitor", "postmitotic\nmotor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, 
          show_row_names = FALSE,
          row_split = ipsc_mn_als_datasets_with_lee_ipsc.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)),  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "Sareen","Sommer", "Smith", "Sterneckert", "Wang", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "NeuroLINCS\niPSC batch"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
neuron.type.markers.rnaseq.heatmap
ggsave(as.ggplot(print(neuron.type.markers.rnaseq.heatmap)), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee_ipsc.neuron.type.markers.heatmap.png"), width = 9, height = 11)

ggsave(as.ggplot(print(neuron.type.markers.rnaseq.heatmap)), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee_ipsc.neuron.type.markers.heatmap.wide.png"), width = 10, height = 7)

```

### OCT4 POU5F1 expression

```{r}
# violin of POU5F1 TPMs
POU5F1_gene_counts = read_tsv(here(proj_path,"nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("POU5F1"))
neurolincs.ipsc.POU5F1_gene_counts = read_tsv(here(shared_path,"neurolincs/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("POU5F1")) %>% select(gene_name, starts_with("X"), -starts_with("X000")) %>% rename_all(funs(stringr::str_replace_all(., '^X', 'neurolincs.iPSC_')))
ipsn_ipsc.POU5F1_gene_counts = left_join(POU5F1_gene_counts, neurolincs.ipsc.POU5F1_gene_counts)
ipsn_ipsc.POU5F1_gene_counts.metadata = ipsn_ipsc.POU5F1_gene_counts %>% select(-gene_id) %>% 
  pivot_longer(!gene_name, names_to = "dataset_sample", values_to = "value") %>% 
  left_join(mutate(ipsc_mn_als_datasets_with_lee_ipsc.metadata, dataset_sample = gsub("-",".",dataset_sample))) %>% 
  mutate(group = case_when(dataset=="neurolincs.iPSC" ~ "NeuroLINCS\nundifferentiated iPSCs", condition=="als" ~"iPSMN\nALS", condition=="ctrl" ~ "iPSMN\nCTRL"))
ipsn_ipsc.POU5F1_gene_counts.metadata %>% count(group)
ipsn_ipsc.POU5F1_gene_counts.metadata %>% my_summarise(value)

POU5F1_gene_counts.TPM.violin = violin_plot(data = filter(ipsn_ipsc.POU5F1_gene_counts.metadata, group %in% c("iPSMN\nALS","iPSMN\nCTRL","NeuroLINCS\nundifferentiated iPSCs")), color_by = "group", continuous = "value", ylims = c(0,2160), cols = pal_npg("nrc", alpha = 1)(3), ylabel = "OCT4 (POU5F1) TPM", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = FALSE)
POU5F1_gene_counts.TPM.violin
ggsave(POU5F1_gene_counts.TPM.violin, filename = here(proj_path,"expression/deseq2/POU5F1_gene_counts.TPM.violins.png"), width = 5, height = 3)


# Heatmap of OCT4 expressing > 5 TPM iPSMNs
undifferentiated = c("NANOG","SOX2","POU5F1","ZFP42","ALPL","ESRG","CNMD","SFRP2")#"UTF1","TERT",
neural.precursor = c("SOX1","SOX2","PROM1","PAX6","NOTCH1","OCLN","CDH1","NES","CDH2","VIM")
neural.progenitor = c("MAP2","ELAVL3","ELAVL4","NF1","NCAM1","DCX")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
neural.crest <- c("SOX10","NGFR","FOXD3","SNAI1","SNAI2","LMX1A")   #NGFR very low in HB9 samples
dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#,"NEFH"
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3", "NOS1","RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2")
# oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
# interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)
neuron.type.markers = c(undifferentiated, neural.precursor, neural.progenitor, floor.plate, roof.plate, neural.crest, dorsal.progenitor, motor.neuron.progenitor, postmitotic.motor.neuron, nonspecific.neuronal) 
neuron.type.markers[!neuron.type.markers %in% gtf$gene_name]
neuron.type.markers.df = tibble("gene_name" = neuron.type.markers) %>% filter(gene_name %in% neuron.type.markers) %>% 
  mutate(group = case_when(gene_name %in% undifferentiated ~ "undifferentiated", gene_name %in% neural.precursor ~ "neural.precursor", gene_name %in% neural.progenitor ~ "neural.progenitor", gene_name %in% floor.plate ~ "floor.plate", gene_name %in% roof.plate ~ "roof.plate", gene_name %in% neural.crest ~ "neural.crest", gene_name %in% dorsal.progenitor ~ "dorsal.progenitor", gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% motor.neuron.progenitor ~ "motor.neuron.progenitor", 
                           gene_name %in% postmitotic.motor.neuron ~ "postmitotic.motor.neuron", #gene_name %in% oculomotor.markers ~ "oculomotor.markers", 
                           gene_name %in% interneuron.markers ~ "interneuron"), 
         group = factor(group, levels = c("undifferentiated", "neural.precursor", "neural.progenitor", "floor.plate", "roof.plate", "neural.crest", "dorsal.progenitor", "nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

ipsn_ipsc.POU5F1_gene_counts.metadata.df =  filter(ipsn_ipsc.POU5F1_gene_counts.metadata, value > 5)  %>% select(dataset_sample, dataset, mutation) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = as.factor(dataset)) %>% filter(!grepl("answerals_CASE",dataset_sample))

POU5F1_expressing_ipsns = filter(ipsn_ipsc.POU5F1_gene_counts.metadata, value > 5) 
min_vst = min(ipsc_mn_als_datasets_with_lee_ipsc$vsd.counts[,2])
POU5F1_neuron.type.markers.marker <- ipsc_mn_als_datasets_with_lee_ipsc$vsd.counts %>%
  rename_all(funs(stringr::str_replace_all(., '-', '.'))) %>%
  filter(gene_name %in% neuron.type.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuron.type.markers.df$gene_name)) %>% select(-gene_id) %>% 
  # rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(contains(c("gene_name", pull(POU5F1_expressing_ipsns, dataset_sample))))
POU5F1_neuron.type.markers.df.filt = neuron.type.markers.df %>% filter(gene_name %in% POU5F1_neuron.type.markers.marker$gene_name)
POU5F1_neuron.type.markers.marker.mat <- make_matrix(select(POU5F1_neuron.type.markers.marker,-gene_name), pull(POU5F1_neuron.type.markers.marker,gene_name))
# POU5F1_neuron.type.markers.marker.centered_mat = t(scale(t(POU5F1_neuron.type.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
# POU5F1_neuron.type.markers.marker.scaled_mat = t(scale(t(POU5F1_neuron.type.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
# rownames(POU5F1_neuron.type.markers.marker.centered_mat)[!rownames(POU5F1_neuron.type.markers.marker.centered_mat) %in% ipsc_mn_als_datasets_with_lee_ipsc.df$dataset_sample]
POU5F1_neuron.type.markers.heatmap <- Heatmap(t(POU5F1_neuron.type.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = POU5F1_neuron.type.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = factor(POU5F1_neuron.type.markers.df.filt$group, levels = c("undifferentiated", "neural.precursor", "neural.progenitor", "floor.plate", "roof.plate", "neural.crest", "dorsal.progenitor", "nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron")),
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",4)), labels = c("undifferentiat.\npluripotent", "neural\nprecursor", "neural\nprogenit", "floor\nplate", "roof\nplate", "neur\ncrest", "dorsal\nprogenitor", "nonspecific\nneuronal", "MN\nprogenitor", "postmitotic\nmotor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          show_row_names = FALSE, column_title = NULL, 
          row_split = pull(mutate(POU5F1_expressing_ipsns, ipsc = case_when(dataset == "neurolincs.iPSC" ~ "iPSC",TRUE~"iPSMN")),ipsc),
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",2)),  labels = c("NeuroLINCS\nundifferentiated\niPSCs","iPSMNs with\nOCT4 >5 TPM"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), 
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
# POU5F1_neuron.type.markers.heatmap
ggsave(as.ggplot(POU5F1_neuron.type.markers.heatmap), filename = here(proj_path,"expression/deseq2/POU5F1_neuron.type.markers.heatmap.png"), width = 9, height = 5)


# Dorothea TF regulon activity of POU5F1, SOX2, NANOG
net_dorothea <- get_dorothea(organism='human')#, levels=c('A', 'B', 'C'))
Normalised_counts_matrix <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% distinct(gene_name, .keep_all=TRUE) %>% select(-gene_id) %>% 
      dplyr::mutate_if(is.numeric, funs((. - min_vst))) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
    select(contains(c("gene_name", pull(ipsc_mn_als_datasets_with_lee_ipsc.metadata, dataset_sample)))) %>% select(-contains("answerals_CASE")) %>% 
  tibble::column_to_rownames(var = "gene_name") %>% as.matrix()
sample_acts <- run_wmean(mat=Normalised_counts_matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5)
sample_acts.nanog_pou5f1_sox2 = sample_acts %>% filter(statistic == 'norm_wmean') %>% arrange(-score)# %>% #, source %in% c("NANOG", "POU5F1", "SOX2")) %>% 
sample_acts.nanog_pou5f1_sox2.mat = sample_acts.nanog_pou5f1_sox2 %>%  pivot_wider(id_cols = 'condition', names_from = 'source', values_from = 'score') %>% column_to_rownames('condition') %>% as.matrix()

sample_acts %>% filter(statistic == 'norm_wmean') %>% arrange(-score) %>% distinct(source) %>% print(n=50)#, source %in% c("NANOG","POU5F1","SOX2","GATA2")) 
sample_acts.nanog_pou5f1_sox2.violin = violin_plot(mutate(filter(sample_acts.nanog_pou5f1_sox2, source %in% c("NANOG", "POU5F1", "SOX2", "FOXP1","GATA2","TAL1")),source=factor(source,levels=c("NANOG", "POU5F1", "SOX2", "FOXP1","GATA2","TAL1"))), color_by = "source", continuous = "score", ylims = c(0,40), cols = brewer.pal(12, "Paired"), ylabel = "Transcription Factor\nregulon activity score", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = FALSE)
sample_acts.nanog_pou5f1_sox2.violin
ggsave(sample_acts.nanog_pou5f1_sox2.violin, filename = here(proj_path,"expression/deseq2/sample_acts.nanog_pou5f1_sox2.violin.png"), width = 4, height = 3)

# "HOXB7", "HOXB13", "HOXA9", "HES1","LHX4","PBX1",,"NEUROG3", "NEUROD1","ZEB1", "ZEB2", "ASCL1", "LHX3","ONECUT2","PBX3", "PAX6", "SOX10","NR2F1"))
filter(net_dorothea, grepl("ISL",source)) %>% distinct(source)


# sample_acts.nanog_pou5f1_sox2.scale <- scale(sample_acts.nanog_pou5f1_sox2)
# sample_acts.nanog_pou5f1_sox2.centered_mat = t(scale(t(sample_acts.nanog_pou5f1_sox2), center=TRUE,scale=FALSE)) # centered, not scaled
# sample_acts.nanog_pou5f1_sox2.scaled_mat = t(scale(t(sample_acts.nanog_pou5f1_sox2), center=FALSE,scale=TRUE)) # scaled, not centered
myRows <- c("NANOG","POU5F1","SOX2") # Set stylings for row names and make our selected rows unique
row_idx <- which(colnames(sample_acts.nanog_pou5f1_sox2.mat) %in% myRows)
fontsizes <- rep(5, ncol(sample_acts.nanog_pou5f1_sox2.mat))
fontsizes[row_idx] <- 14
fontcolors <- rep('black', ncol(sample_acts.nanog_pou5f1_sox2.mat))
fontcolors[row_idx] <- 'red'
fontfaces <- rep('plain',ncol(sample_acts.nanog_pou5f1_sox2.mat))
fontfaces[row_idx] <- 'bold'

sample_acts.nanog_pou5f1_sox2.mat.heatmap <- Heatmap(sample_acts.nanog_pou5f1_sox2.mat, 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          show_row_names = FALSE, show_column_names = FALSE, row_title = NULL, 
          row_split = factor(ipsc_mn_als_datasets_with_lee_ipsc.df$dataset, levels = c("answerals", "bhinge", "catanese", "dafinca.c9orf72", "dafinca.tardbp", "desantis", "hawkins", "kapeli", "kiskinis", "lee", "luisier", "neurolincs.diMN", "neurolincs.iMN", "sareen","sommer", "smith", "sterneckert", "wang")),
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)),  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"),  labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          # row_order = sporadic_ctrl_sample_names$mutation,
          # row_split = sporadic_ctrl_sample_names$mutation,
          # right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",2)),  labels = c("CTRL", "Sporadic\nALS"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
          bottom_annotation = columnAnnotation(columns = anno_text(colnames(sample_acts.nanog_pou5f1_sox2.mat), gp = gpar(fontsize = fontsizes, fontface = fontfaces, col = fontcolors))),
          column_title = NULL, cluster_columns = FALSE, cluster_row_slices = FALSE, cluster_rows = FALSE)
sample_acts.nanog_pou5f1_sox2.mat.heatmap
ggsave(as.ggplot(sample_acts.nanog_pou5f1_sox2.mat.heatmap), filename = here(proj_path,"expression/deseq2/sample_acts.nanog_pou5f1_sox2.heatmap.png"), width = 13, height = 10)


# Sensitivity analysis excluding POU5F1 TPM > 5. Excludes 26 samples: 429 > 403 
ipsc_mn_als_datasets.noPOU5F1 = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.noPOU5F1.rds"))
POU5F1_gene_counts = read_tsv(here(proj_path,"nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("POU5F1"))
POU5F1_gene_counts.metadata = POU5F1_gene_counts %>% select(-gene_id) %>% pivot_longer(!gene_name, names_to = "dataset_sample", values_to = "value") %>% left_join(ipsc_mn_als_datasets_with_lee_ipsc.metadata)
POU5F1_expressing_ipsns = filter(POU5F1_gene_counts.metadata, value > 5) 
filter(ipsc_mn_als_datasets.metadata, !dataset_sample %in% POU5F1_expressing_ipsns$dataset_sample) %>% count(dataset) # 15
filter(ipsc_mn_als_datasets.metadata, !dataset_sample %in% POU5F1_expressing_ipsns$dataset_sample) %>% count(condition) # 310 ALS, 93 CTRL

# PCA
ipsc_mn_als_datasets.noPOU5F1.pcaData <- plotPCA(ipsc_mn_als_datasets.noPOU5F1$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/patanir/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "poly(A)"), dataset_sample, gender)) 
  
ipsc_mn_als_datasets.noPOU5F1.pca_dataset <- ggplot(ipsc_mn_als_datasets.noPOU5F1.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm'), plot.title = element_text(hjust = 0.5)) + ggtitle("OCT4 expressing iPSMNs removed") + 
  guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.noPOU5F1.pcaData, "percentVar"))[1],"% variance")) + ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.noPOU5F1.pcaData, "percentVar"))[2],"% variance")) + labs(color="")
ipsc_mn_als_datasets.noPOU5F1.pca_dataset

## Volcano
ipsc_mn_als_datasets.noPOU5F1$res %>% filter(padj < 0.05) # 111
ipsc_mn_als_datasets.noPOU5F1$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)# 4
ipsc_mn_als_datasets.noPOU5F1$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #56 down, 55 up
ipsc_mn_als_datasets.noPOU5F1$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) # 0
ipsc_mn_als_datasets.noPOU5F1$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # RNASEL
ipsc_mn_als_datasets.noPOU5F1$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # 

ipsc_mn_als_datasets.noPOU5F1.als.labels <- ipsc_mn_als_datasets.noPOU5F1$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.noPOU5F1.gene.labels <- ipsc_mn_als_datasets.noPOU5F1$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.noPOU5F1.volcano <- volcano_plot(ipsc_mn_als_datasets.noPOU5F1$res, numerator = "ALS", denomenator = "CTRL",
               title = "OCT4 expressing iPSMNs removed", subtitle = "93 Controls vs 310 ALS", ymax = 10, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.noPOU5F1.als.labels, ipsc_mn_als_datasets.noPOU5F1.gene.labels,"SESN1","TCEAL6"), distinct_labels = TRUE)
ipsc_mn_als_datasets.noPOU5F1.volcano

## DoRoTHEA & PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.noPOU5F1.res.matrix <- ipsc_mn_als_datasets.noPOU5F1$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.noPOU5F1_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.noPOU5F1.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.noPOU5F1_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.noPOU5F1_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "OCT4 expressing iPSMNs removed: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.noPOU5F1_progeny.contrast_acts, label = "p.signif", y.position = 11.5, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_als_datasets.noPOU5F1_progeny.pathwayActivity

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.noPOU5F1.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts %>% filter(p_value < 0.15) %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "OCT4 expressing iPSMNs removed: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts.volcano

ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts %>% arrange(score)
ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts %>% filter(source == "TP53")

ipsc_mn_als_datasets.noPOU5F1.figs.merged = plot_grid(
  ipsc_mn_als_datasets.noPOU5F1.pca_dataset, ipsc_mn_als_datasets.noPOU5F1.volcano, 
  ipsc_mn_als_datasets.noPOU5F1_progeny.pathwayActivity, ipsc_mn_als_datasets.noPOU5F1_dorothea.contrast_acts.volcano,
  nrow = 2, ncol=2, rel_heights = c(1,1), labels = "auto") 
# ipsc_mn_als_datasets.noPOU5F1.figs.merged
ggsave(ipsc_mn_als_datasets.noPOU5F1.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.noPOU5F1.figs.merged.png"), height = 8.25, width = 8.25)


```


## Dorso-ventral markers

From Fig 2 in https://journals.biologists.com/dev/article/148/15/dev199711/271192/Single-cell-transcriptome-profiling-of-the-human 

```{r}
nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

neuronal.markers = c(dl.domains, V0.domain,V1, V2a.domain, V2b.domain,MN.domain,V3.domain) #nonspecific.neuronal
neuronal.markers[!neuronal.markers %in% gtf$gene_name]
neuronal.markers.df = tibble("gene_name" = neuronal.markers) %>% 
  mutate(group = case_when(gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% dl.domains ~ "dl.domains", 
                           gene_name %in% V0.domain ~ "V0.domain", gene_name %in% V1 ~ "V1", 
                           gene_name %in% V2a.domain ~ "V2a.domain", gene_name %in% V2b.domain ~ "V2b.domain", gene_name %in% MN.domain ~ "MN.domain", gene_name %in% V3.domain ~ "V3.domain"), 
         group = factor(group, levels = c("dl.domains","V0.domain", "V1", "V2a.domain", "V2b.domain", "MN.domain", "V3.domain"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
neuronal.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% neuronal.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuronal.markers.df$gene_name)) %>% select(-gene_id) %>%
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
neuronal.markers.marker.mat <- make_matrix(select(neuronal.markers.marker,-gene_name), pull(neuronal.markers.marker,gene_name))
neuronal.markers.marker.centered_mat = t(scale(t(neuronal.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
dorsoventral.markers.heatmap <- Heatmap(t(neuronal.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = neuronal.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = neuronal.markers.df$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",7)), labels =  c("dl\ndomains","V0\ndomain", "V1", "V2a\ndomain", "V2b\ndomain", "MN\ndomain", "V3"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          # show_row_names = TRUE, row_names_gp = gpar(fontsize = 6, col = c(rep("#386CB0",9), rep("#75489F",9), rep("black",90))), #row_names_rot = 90, 
          cluster_rows = FALSE,
          row_title = NULL, 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)))
dorsoventral.markers.heatmap
ggsave(as.ggplot(dorsoventral.markers.heatmap), filename = here(proj_path,"expression/deseq2/dorsoventral.markers.heatmap.png"), width = 8.25, height = 9)


# # cluster by columns
# Heatmap(neuronal.markers.marker.centered_mat, 
#           border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
#           row_order = neuronal.markers.df$gene_name,
#           row_names_gp = gpar(fontsize = 7), 
#           row_split = factor(neuronal.markers.df$group, levels =  c("nonspecific.neuronal","dl.domains","V0.domain", "V1", "V2a.domain", "V2b.domain", "MN.domain", "V3.domain")),
#           right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = rep("white",8)), labels =  c("nonspecific\nneuronal","dl\ndomains","V0\ndomain", "V1", "V2a\ndomain", "V2b\ndomain", "MN\ndomain", "V3\ndomain"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           row_title = "Neuronal markers", row_title_gp = gpar(fontsize = 8), 
#           show_column_names = TRUE, column_names_gp = gpar(fontsize = 6, col = c(rep("#386CB0",9), rep("#75489F",9), rep("black",90))), #column_names_rot = 90,
#           cluster_rows = FALSE, cluster_row_slices = FALSE, cluster_columns = TRUE,
#           column_title = NULL)

```

Progenitor dosroventral markers


```{r}
# nonspecific = c("SOX2")
# roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
# dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
# intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
# pMN.domain = c("NKX6-1", "OLIG2")
# ventral.progenitor = c("NKX2-2","NKX2-8")
# floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")
# 
# progenitor.markers = c(nonspecific, roof.plate, dorsal.progenitor,intermediate.progenitor, pMN.domain, ventral.progenitor,floor.plate)
# progenitor.markers[!progenitor.markers %in% gtf$gene_name]
# progenitor.markers.df = tibble("gene_name" = progenitor.markers) %>% 
#   mutate(group = case_when(gene_name %in% nonspecific ~ "nonspecific", gene_name %in% roof.plate ~ "roof.plate", 
#                            gene_name %in% dorsal.progenitor ~ "dorsal.progenitor", gene_name %in% intermediate.progenitor ~ "intermediate.progenitor", 
#                            gene_name %in% pMN.domain ~ "pMN.domain", gene_name %in% ventral.progenitor ~ "ventral.progenitor", gene_name %in% floor.plate ~ "floor.plate"), 
#          group = factor(group, levels = c("nonspecific","roof.plate", "dorsal.progenitor", "intermediate.progenitor", "pMN.domain", "ventral.progenitor", "floor.plate"))) %>% 
#   arrange(group) %>% distinct(gene_name, .keep_all = TRUE)
# 
# # together with other datasets
# min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
# progenitor.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
#   filter(gene_name %in% progenitor.markers.df$gene_name) %>% 
#   dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
#   arrange(factor(gene_name, levels = progenitor.markers.df$gene_name)) %>% select(-gene_id) %>%
#   rename_with(~ gsub("dafinca.tardbp", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafinca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
# progenitor.markers.marker.mat <- make_matrix(select(progenitor.markers.marker,-gene_name), pull(progenitor.markers.marker,gene_name))
# 
# ### Cluster samples ###
# progenitor.markers.marker.centered_mat = t(scale(t(progenitor.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
# progenitor.markers.heatmap <- Heatmap(t(progenitor.markers.marker.mat), 
#           border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
#           column_order = progenitor.markers.df$gene_name,
#           column_names_gp = gpar(fontsize = 7), 
#           column_split = progenitor.markers.df$group,
#           bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",7)), labels =  c("nonspecific","roof\nplate", "dorsal\nprogenitor", "intermediate\nprogenitor", "pMN\ndomain", "ventral\nprogenitor", "floor\nplate"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           column_title ="Progenitor markers", column_title_gp = gpar(fontsize = 8), 
#           show_row_names = FALSE, 
#           # row_order = dataset.samples.df$dataset_sample,
#           row_split = dataset.samples.df$dataset,
#           right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",17)), 
#                                                                  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
#                                                                  labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
# progenitor.markers.heatmap
# 

```


## Rostro-caudal markers

HOX markers

```{r}
HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
# forebrain.markers = c("FOXG1","OTX2")
# HOX.markers = c(forebrain.markers, HOX.markers)
HOX.markers.df <- data.frame(row.names = HOX.markers, gene_name = HOX.markers) %>% 
  mutate(group = gsub("HOX[A-D]","", gene_name), group = factor(as.numeric(gsub("-AS$","", group)))) %>% 
  arrange(group) %>% 
  mutate(level = factor(case_when(group %in% c(1,2,3) ~ "Hindbrain", group %in% c(4,5,6,7) ~ "Cervical", group %in% c(8,9) ~ "Thoracic", 
                                  group %in% c(10,11) ~ "Lumbar", TRUE ~ "Sacral"), levels = c("Hindbrain","Cervical","Thoracic","Lumbar","Sacral")))
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
HOX.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% 
  filter(gene_name %in% HOX.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst))) %>% 
  arrange(factor(gene_name, levels = HOX.markers.df$gene_name)) %>% select(-gene_id) %>%
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
HOX.markers.marker.mat <- make_matrix(select(HOX.markers.marker,-gene_name), pull(HOX.markers.marker,gene_name))
HOX.markers.marker.centered.mat = t(scale(t(HOX.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
HOX.markers.heatmap <- Heatmap(t(HOX.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = HOX.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = HOX.markers.df$level,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Hindbrain", "Cervical", "Thoracic","Lumbar","Sacral"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          cluster_rows = FALSE,
          row_title = NULL, 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",17)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)))
HOX.markers.heatmap
ggsave(as.ggplot(HOX.markers.heatmap), filename = here(proj_path,"expression/deseq2/HOX.markers.heatmap.png"), width = 8.25, height = 9)
```



# Fig 2 hiPSC ALS meta-analysis

## Volcano

```{r}
ipsc_mn_als_datasets.metadata %>% count(mutation)
ipsc_mn_als_datasets$res %>% filter(padj < 0.05)# 43
# ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)#3
ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #23 down, 20 up
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) #0
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # RNASEL
ipsc_mn_als_datasets$res %>% filter(gene_name %in% c("SFPQ","FUS","TARDBP"))# 43

## Volcano
ipsc_mn_als_datasets.als.labels <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.gene.labels <- ipsc_mn_als_datasets$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.volcano <- volcano_plot(ipsc_mn_als_datasets$res, numerator = "ALS", denomenator = "CTRL",
               subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 2, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.als.labels, ipsc_mn_als_datasets.gene.labels))
# ipsc_mn_als_datasets.volcano

# ## MA plot
# ipsc_mn_als_datasets.ma <- ma_plot(ipsc_mn_als_datasets$res, 
#                numerator = "ALS", denomenator = "CTRL", subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 1000, dpi = 300, 
#                gene_labels = c(ipsc_mn_als_datasets.als.labels, ipsc_mn_als_datasets.gene.labels)) #+ xlim(0,10000)
# ipsc_mn_als_datasets.ma

ipsc_mn_als_datasets$res %>% filter(gene_name %in% c("H2AX","TP53BP1","PARP1") | grepl("^RPA[1-4]",gene_name)) # RNASEL
ipsc_mn_als_datasets$res %>% filter(grepl("^RPA[1-9]",gene_name)) # RNASEL

```

## GO terms & GSEA

```{r}
ipsc_mn_als_datasets.gost <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_als_datasets.gprofiles <- ipsc_mn_als_datasets.gost$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
ipsc_mn_als_datasets.gprofiles_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")
ipsc_mn_als_datasets.gprofiles_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(ipsc_mn_als_datasets.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "transcription regulator complex",
# "sequence-specific DNA binding",
# "double-stranded DNA binding",
# "sequence-specific double-stranded DNA binding",
# "chromatin",
"neuron fate commitment",
# "transcription regulatory region nucleic acid binding",
# "transcription cis-regulatory region binding",
# "DNA-binding transcription factor activity",
"cell differentiation in spinal cord",
"DNA-binding transcription factor activity, RNA polymerase II-specific",
"ventral spinal cord development",
"cis-regulatory region sequence-specific DNA binding"
# "Aplasia of the musculature"
)
ipsc_mn_als_datasets.gprofiles_filt_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_als_datasets.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"p53 transcriptional gene network",
"Genotoxicity pathway",
"miRNA regulation of DNA damage response",
"DNA damage response",
"p53 signaling pathway")
ipsc_mn_als_datasets.gprofiles_filt_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_als_datasets.gprofiles_filt_combined <- bind_rows(ipsc_mn_als_datasets.gprofiles_filt_down, ipsc_mn_als_datasets.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
ipsc_mn_als_datasets.gprofiles_filt_combined = ipsc_mn_als_datasets.gprofiles_filt_combined %>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
ipsc_mn_als_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_als_datasets.gprofiles_filt_combined) #+ theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
# ipsc_mn_als_datasets.go_curated


# which genes are driving the terms
p53.DDR.repair_gene_set = unique(c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`,gprofiler_database$`DNA repair`))
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway)) # TNFRSF10B, SESN1, CDKN1A, RRM2B
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(gprofiler_database$`DNA Damage Response`)) # TNFRSF10B, CDKN1A, RRM2B, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA Damage Response`) # TNFRSF10B, CDKN1A, RRM2B, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA repair`) # TNFRSF10B, CDKN1A, RRM2B, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% kegg.pathways.msigdb$KEGG_P53_SIGNALING_PATHWAY)
kegg.pathways.msigdb$KEGG_P53_SIGNALING_PATHWAY

wikipathways.msigdb <- gmtPathways(here(camp_path,"/genomes/gene-ontology/c2.cp.wikipathways.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
hpo.msigdb <- gmtPathways(here(camp_path,"/genomes/gene-ontology/c5.hpo.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_MIRNA_REGULATION_OF_DNA_DAMAGE_RESPONSE) # TNFRSF10B, CDKN1A, MDM2, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`TP53 Network`) # CDKN1A
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`cell fate commitment`) # TBX5, MYOG, LMO4, OLIG1, TBR1, OLIG2, FOXN4, BCL11B
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`neuron fate commitment`) # OLIG1, OLIG2, FOXN4
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_OLIGODENDROCYTE_SPECIFICATION_AND_DIFFERENTIATION_LEADING_TO_MYELIN_COMPONENTS_FOR_CNS) # OLIG1, OLIG2
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% hpo.msigdb$HP_APLASIA_OF_THE_MUSCULATURE) # CHRND, TBX5
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`ventral spinal cord development`) # LMO4, OLIG2, FOXN4
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA-binding transcription activator activity, RNA polymerase II-specific`) # MYOG, TBX5, POU5F1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$chromatin) # MYOG, TBX5, POU5F1, OLIG1, OLIG2, FOXN4

## GSEA: DNA damage response, p53 signalling pathway
ipsc_mn_als_datasets.geneList <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.geneList) <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ipsc_mn_als_datasets.geneList = sort(ipsc_mn_als_datasets.geneList, decreasing = TRUE)

# p53 signalling pathway
p53.signalling_genes.list <- list("p53.signalling" = ipsc_mn_als_datasets$res$gene_name[ipsc_mn_als_datasets$res$gene_name %in% p53.signalling.pathway])
ipsc_mn_als_datasets.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = ipsc_mn_als_datasets.geneList)
ipsc_mn_als_datasets.p53.signalling_fgseaRes
# pathway         pval         padj   log2err        ES      NES size                                leadingEdge
# 1: p53.signalling 0.0004854782 0.0004854782 0.4984931 0.4073582 1.437247  264 RRM2B,CDKN1A,EDA2R,MDM2,TRIAP1,GADD45A,...
ipsc_mn_als_datasets.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], ipsc_mn_als_datasets.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("p53 signalling") + xlab("") +
  annotate("text", x = 35000, y = 0.3, label = paste0("NES = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$pval, digits = 6)), size = 2.8, hjust = 0) + theme_oz()
ipsc_mn_als_datasets.p53.signalling.gseaplot

```


## DoRoTHEA & PROGENy

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/pw_bk.html 
<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.res.matrix <- ipsc_mn_als_datasets$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_progeny.pathwayActivity <- ggplot(ipsc_mn_als_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "ALS vs CTRL Normalised enrichment") +
    stat_pvalue_manual(ipsc_mn_als_progeny.contrast_acts, label = "p.signif", y.position = 14, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_als_progeny.pathwayActivity

# p53 weights
ipsc_mn_als_progeny.p53weights = net_progeny %>% filter(source == "p53", target %in% ipsc_mn_als_datasets$res$gene_name) %>% left_join(select(ipsc_mn_als_datasets$res, target=gene_name,stat)) %>% 
  mutate(color = case_when(weight > 0 & stat > 0 ~ '1', weight > 0 & stat < 0 ~ '2', weight < 0 & stat > 0 ~ '2', weight < 0 & stat < 0 ~ '1', TRUE ~ "3"))
ipsc_mn_als_progeny.p53weights.plot = ggplot(ipsc_mn_als_progeny.p53weights, aes(x = weight, y = stat, color = color)) + geom_point(size = 0.5) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_progeny.p53weights, weight > 5, stat > 3), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.1) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "p53 pathway weights", x = "p53 pathway gene weight", y = "ALS vs CTRL test statistic") +
  geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted') +
  coord_cartesian(xlim = c(-8,18), ylim = c(-2.5,5)) 
ipsc_mn_als_progeny.p53weights.plot


```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_dorothea.contrast_acts.labels.up = ipsc_mn_als_dorothea.contrast_acts %>% top_n(score, n = 3) %>% pull(source)
ipsc_mn_als_dorothea.contrast_acts.labels.down = ipsc_mn_als_dorothea.contrast_acts %>% top_n(-score, n = 3) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.contrast_acts, source %in% c(ipsc_mn_als_dorothea.contrast_acts.labels.up, ipsc_mn_als_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_dorothea.contrast_acts.volcano

ipsc_mn_als_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_dorothea.contrast_acts %>% arrange(score)

# # TP53 weights
# ipsc_mn_als_dorothea.contrast_acts
# ipsc_mn_als_dorothea.TP53weights = net_dorothea %>% filter(source == "TP53", target %in% ipsc_mn_als_datasets$res$gene_name) %>% left_join(select(ipsc_mn_als_datasets$res, target=gene_name,stat)) %>% 
#   mutate(color = case_when(mor > 0 & stat > 0 ~ '1', mor > 0 & stat < 0 ~ '2', mor < 0 & stat > 0 ~ '2', mor < 0 & stat < 0 ~ '1', TRUE ~ "3"))
# ipsc_mn_als_dorothea.TP53weights.plot = ggplot(ipsc_mn_als_dorothea.TP53weights, aes(x = mor, y = stat, color = color)) + geom_point(size = 0.5) +
#   scale_colour_manual(values = c("red","royalblue3","grey")) +
#   geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.TP53weights, mor > 5, stat > 3), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.1) +
#   theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
#   labs(title = "TP53 TF weights", x = "TP53 TF gene regulation", y = "ALS vs CTRL test statistic") +
#   geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted')# +
#   # coord_cartesian(xlim = c(-8,18), ylim = c(-2.5,5)) 
# ipsc_mn_als_dorothea.TP53weights.plot

ipsc_mn_als_dorothea.TP53weights %>% filter(mor==1, stat > 4)

# ipsc_mn_als_dorothea.tp53 <- net_dorothea %>%  filter(source == "TP53") %>% mutate(gene_name = target) %>% left_join(select(ipsc_mn_als_datasets$res, gene_name,stat,log2FoldChange,pvalue)) %>%
#   mutate(color = case_when(mor > 0 & stat > 0 ~ '1', mor > 0 & stat < 0~'2',mor < 0 & stat > 0~'2',mor < 0 & stat < 0~'1',TRUE ~ "3"))
# 
# ipsc_mn_als_dorothea.tp53.volcano = ggplot(ipsc_mn_als_dorothea.tp53, aes(x = log2FoldChange, y = -log10(pvalue), color = color, size=abs(mor))) + geom_point() + scale_size(range = c(0, 3)) +
#   scale_colour_manual(values = c("red","royalblue3","grey")) +
#   geom_text_repel(data = filter(ipsc_mn_als_dorothea.tp53, abs(stat) > 3), aes(label = gene_name, size=1), fontface = "italic", max.overlaps = Inf, size = 2.3) +
#   theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +  geom_vline(xintercept = 0, linetype = 'dotted') + geom_hline(yintercept = 0, linetype = 'dotted') +
#     labs(title = "TP53 regulon", x = "log2 fold change (ALS vs CTRL)", y = expression(-log[10]~P~value)) + ylim(c(0,2)) +   geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.tp53, log2FoldChange > 1), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5)
# ipsc_mn_als_dorothea.tp53.volcano

ipsc_mn_als_dorothea.tp53 %>% arrange(-stat) # TNFRSF10B, SESN1, RRM2B, CDKN1A, MDM2
```


## Merge

```{r}
ipsc_mn_als_datasets.figs.merged = plot_grid(
  plot_grid(ipsc_mn_als_datasets.volcano, ipsc_mn_als_datasets.go_curated, ipsc_mn_als_datasets.p53.signalling.gseaplot, ncol=3, rel_widths = c(1,0.8,0.6), labels = c("a","b","c")), 
  plot_grid(ipsc_mn_als_progeny.pathwayActivity, ipsc_mn_als_progeny.p53weights.plot,ipsc_mn_als_dorothea.contrast_acts.volcano, ncol=3, rel_widths = c(1,0.9,1), labels = c("d","e","f")), 
  nrow = 2, rel_heights = c(1.1,1))
# ipsc_mn_als_datasets.figs.merged
ggsave(ipsc_mn_als_datasets.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.figs.merged.png"), height = 7, width = 10)
ggsave(ipsc_mn_als_datasets.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.figs.merged.pdf"), height = 7, width = 10)

ipsc_mn_als_datasets.volcano_go = plot_grid(ipsc_mn_als_datasets.volcano, ipsc_mn_als_datasets.go_curated, ncol = 2, rel_widths = c(1,0.8))
ggsave(ipsc_mn_als_datasets.volcano_go, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.volcano_go.png"), height = 4, width = 10)

ipsc_mn_als_datasets.decoupleR = plot_grid(ipsc_mn_als_progeny.pathwayActivity, ipsc_mn_als_dorothea.contrast_acts.volcano, ncol=2, rel_widths = c(1,1))
ggsave(ipsc_mn_als_datasets.decoupleR, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.decoupleR.png"), height = 4, width = 10)

```


# Table S3 DGE panALS

```{r}
ipsc_mn_als_datasets.res.sig = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% select(gene_name, everything())
# ipsc_mn_als_datasets.res.sig
write_csv(ipsc_mn_als_datasets.res.sig, here(proj_path,"expression/deseq2/TableS2.ipsc_mn_als_datasets.res.sig.csv"))
write_csv(ipsc_mn_als_datasets$res, here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.als_vs_ctrl.csv"))

```

# Table S4 Dorothea panALS

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% select(-statistic, -condition, transcription_factor = source, normalised_enrichment_score = score) %>% arrange(p_value, -abs(normalised_enrichment_score))
# ipsc_mn_als_dorothea.contrast_acts
write_csv(ipsc_mn_als_dorothea.contrast_acts, here(proj_path,"expression/deseq2/TableS4.ipsc_mn_als_dorothea.csv"))

```


# Fig S8 GSEA DNA damage 

```{r}
ipsc_mn_als_datasets.geneList <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.geneList) <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ipsc_mn_als_datasets.geneList = sort(ipsc_mn_als_datasets.geneList, decreasing = TRUE)

gprofiler_database.DNA_damage_pathways = gprofiler_database[grepl('DNA damage|p53', names(gprofiler_database))]
gprofiler_database.DNA_damage_pathways_fgseaRes <- fgsea(pathways = gprofiler_database.DNA_damage_pathways, stats = ipsc_mn_als_datasets.geneList)
gprofiler_database.DNA_damage_pathways_fgseaRes %>% arrange(pval)
gprofiler_database.DNA_damage.filt = filter(gprofiler_database.DNA_damage_pathways_fgseaRes, size >7, !grepl("uncertain|negative|prostate",pathway)) %>% mutate(p.signif = case_when(pval < 0.003 ~ "***", pval < 0.009 ~ "***",pval < 0.01 ~ "**",pval < 0.05 ~ "*", TRUE ~ ""), group1 = "pathway", group2 = "pathway")
gprofiler_database.DNA_damage_pathways_fgseaRes.barplot <- ggplot(gprofiler_database.DNA_damage.filt, aes(x=reorder(pathway, NES), y = NES)) +
    geom_bar(aes(fill = NES), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "DNA damage pathways", y = "ALS vs CTRL iPSMNs\nNormalised Enrichment Score", x = "") + geom_hline(yintercept = 0, linetype = 3) + 
    stat_pvalue_manual(gprofiler_database.DNA_damage.filt, label = "p.signif", y.position = 2, size = 4, xmin = "pathway", xmax = NULL, hide.ns = TRUE) + coord_flip()
gprofiler_database.DNA_damage_pathways_fgseaRes.barplot
ggsave(gprofiler_database.DNA_damage_pathways_fgseaRes.barplot, filename = here(proj_path,"expression/deseq2/DNA_damage_pathways.fgseaRes.barchart.png"), height = 11, width = 9)

```


# Fig S9 polyA & total subgroup analyses

## polyA

```{r}
# ipsc_mn_als_datasets.polyA = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.rds"))
# ipsc_mn_als_datasets.total = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.rds"))

ipsc_mn_als_datasets.polyA.metadata %>% count(dataset) # 10
ipsc_mn_als_datasets.polyA.metadata %>% count(condition) # 48 ALS, 43 CTRL
ipsc_mn_als_datasets.total.metadata %>% count(dataset) # 5
ipsc_mn_als_datasets.total.metadata %>% count(condition) # 275 ALS, 63 CTRL

# PCA
ipsc_mn_als_datasets.polyA.pcaData <- plotPCA(ipsc_mn_als_datasets.polyA$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/patanir/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "poly(A)"), dataset_sample, gender)) 
  
ipsc_mn_als_datasets.polyA.pca_dataset <- ggplot(ipsc_mn_als_datasets.polyA.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm'), plot.title = element_text(hjust = 0.5)) + ggtitle("polyA") + 
  guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.polyA.pcaData, "percentVar"))[1],"% variance")) + ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.polyA.pcaData, "percentVar"))[2],"% variance")) + labs(color="polyA\ndatasets")
ipsc_mn_als_datasets.polyA.pca_dataset

## Volcano
ipsc_mn_als_datasets.polyA.metadata %>% count(condition)
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) # 57
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)# 4
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #32 down, 25 up
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) # PRPH2
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # POLR2J3
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # POLR2J3

ipsc_mn_als_datasets.polyA.als.labels <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.gene.labels <- ipsc_mn_als_datasets.polyA$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.volcano <- volcano_plot(ipsc_mn_als_datasets.polyA$res, numerator = "ALS", denomenator = "CTRL",
               title = "polyA", subtitle = "43 Controls vs 48 ALS", ymax = 10, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.polyA.als.labels, ipsc_mn_als_datasets.polyA.gene.labels,"SESN1","TCEAL6"), distinct_labels = TRUE)
# ipsc_mn_als_datasets.polyA.volcano

## DoRoTHEA & PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.polyA.res.matrix <- ipsc_mn_als_datasets.polyA$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.polyA_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.polyA.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.polyA_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.polyA_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "polyA: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.polyA_progeny.contrast_acts, label = "p.signif", y.position = 11.5, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
# ipsc_mn_als_datasets.polyA_progeny.pathwayActivity

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.polyA.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% filter(p_value < 0.15) %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "polyA: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano

ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% arrange(score)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% filter(source == "TP53")

```


## Total

```{r}
# PCA
ipsc_mn_als_datasets.total.pcaData <- plotPCA(ipsc_mn_als_datasets.total$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/patanir/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "Total"), dataset_sample, gender)) 

ipsc_mn_als_datasets.total.pca_dataset <- ggplot(ipsc_mn_als_datasets.total.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm'), plot.title = element_text(hjust = 0.5)) + ggtitle("Ribo-zero") + 
  guides(color=guide_legend(nrow=2, keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.total.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.total.pcaData, "percentVar"))[2],"% variance"))  + labs(color="Ribo-zero\ndatasets") 
# ipsc_mn_als_datasets.total.pca_dataset

## Volcano
ipsc_mn_als_datasets.total.metadata %>% count(condition)
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) # 10
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)# 0
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #7 down, 3 up
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) # 
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # RNASEL, MRPL18

filter(ipsc_mn_als_datasets.metadata, library_type == "Total") %>% count(condition)
ipsc_mn_als_datasets.total.als.labels <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.total.gene.labels <- ipsc_mn_als_datasets.total$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.total.volcano <- volcano_plot(ipsc_mn_als_datasets.total$res, numerator = "ALS", denomenator = "CTRL",
               title = "Ribo-zero", subtitle = "63 Controls vs 275 ALS", ymax = 8, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.total.als.labels, ipsc_mn_als_datasets.total.gene.labels,"RNASEL","SESN1", "TCEAL6","MTCO1P12","MTCO2P12","LMO4"))
# ipsc_mn_als_datasets.total.volcano

## DoRoTHEA & PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.total.res.matrix <- ipsc_mn_als_datasets.total$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.total_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.total.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.total_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.total_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Ribo-zero: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.total_progeny.contrast_acts, label = "p.signif", y.position = 7.5, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
ipsc_mn_als_datasets.total_progeny.pathwayActivity


# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als_datasets.total_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.total.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.total_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Ribo-zero: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.total_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano

ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% filter(source == "TP53")
ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% arrange(score)



## Compare polyA & total
# Venn overlap
ipsc_mn_als_datasets.polyA.res.dge.up <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.res.dge.down <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
ipsc_mn_als_datasets.total.res.dge.up <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.total.res.dge.down <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# # ggvenn
ipsc_mn_als_datasets.polyA.total.up_genes <- list("polyA" = ipsc_mn_als_datasets.polyA.res.dge.up, "Total" = ipsc_mn_als_datasets.total.res.dge.up)
ipsc_mn_als_datasets.polyA.total.down_genes <- list("polyA" = ipsc_mn_als_datasets.polyA.res.dge.down, "Total" = ipsc_mn_als_datasets.total.res.dge.down)
ipsc_mn_als_datasets.polyA.total.up_venn <- ggvenn(ipsc_mn_als_datasets.polyA.total.up_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap Up")  + theme(plot.subtitle = element_text(hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, -1, -1, -1), "lines"))
ipsc_mn_als_datasets.polyA.total.up_venn

ipsc_mn_als_datasets.polyA.total.down_venn <- ggvenn(ipsc_mn_als_datasets.polyA.total.down_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap down")  + theme(plot.subtitle = element_text(hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, -1, -1, -1), "lines"))
ipsc_mn_als_datasets.polyA.total.down_venn

ipsc_mn_als_datasets.polyA.res.dge.up[ipsc_mn_als_datasets.polyA.res.dge.up %in% ipsc_mn_als_datasets.total.res.dge.up]
# 
# # Scatter correlation
ipsc_mn_als_datasets.polyA_total.labels = select(ipsc_mn_als_datasets.total$res, gene_id, gene_name, polyA.stat = stat) %>% left_join(select(ipsc_mn_als_datasets.polyA$res, gene_id, gene_name, total.stat = stat)) %>% filter((polyA.stat > 2.5 & total.stat > 2.5) | (polyA.stat < -2.5 & total.stat < -2.5))
ipsc_mn_als_datasets.polyA_total.list = list("polyA: ALS vs CTRL" = ipsc_mn_als_datasets.polyA$res, "Total: ALS vs CTRL" = ipsc_mn_als_datasets.total$res)
ipsc_mn_als_datasets.polyA_total_stat.corr = plot_de_correlation(model_list1 = ipsc_mn_als_datasets.polyA_total.list, model_list2 = ipsc_mn_als_datasets.polyA_total.list, mutation1 = "polyA: ALS vs CTRL", mutation2 = "Total: ALS vs CTRL", gene_labels = pull(ipsc_mn_als_datasets.polyA_total.labels,gene_name), plot_corr_line = TRUE, plot_p_value = TRUE)
ipsc_mn_als_datasets.polyA_total_stat.corr

```

## Merge

```{r}
ipsc_mn_als_datasets.polyA_total.figs.merged = plot_grid(
  ipsc_mn_als_datasets.polyA.pca_dataset, ipsc_mn_als_datasets.total.pca_dataset, #ipsc_mn_answerals.pca_dataset,
  ipsc_mn_als_datasets.polyA.volcano, ipsc_mn_als_datasets.total.volcano, #ipsc_mn_answerals.volcano,
  ipsc_mn_als_datasets.polyA_progeny.pathwayActivity, ipsc_mn_als_datasets.total_progeny.pathwayActivity, #ipsc_mn_answerals_progeny.pathwayActivity,
  ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano, ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano, #ipsc_mn_answerals_dorothea.contrast_acts.volcano,
  nrow = 4, ncol=2, rel_heights = c(0.8,1,0.8,1), labels = "auto") 
# ipsc_mn_als_datasets.polyA_total.figs.merged
ggsave(ipsc_mn_als_datasets.polyA_total.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA_total.figs.merged.png"), height = 11.75, width = 8.25)

# # Table DGE total & polyA
# ipsc_mn_als_datasets.polyA_total.sig = select(ipsc_mn_als_datasets.polyA$res, gene_id, gene_name, padj.polyA=padj, stat.polyA=stat, lfc.polyA=log2FoldChange) %>% left_join(select(ipsc_mn_als_datasets.total$res, gene_id, gene_name, padj.total=padj,stat.total=stat, lfc.total=log2FoldChange)) %>% filter(padj.polyA < 0.05 | padj.total < 0.05) %>% mutate(changed = case_when(padj.polyA < 0.05 & lfc.polyA > 0 ~ "polyA up", padj.polyA < 0.05 & lfc.polyA < 0 ~ "polyA down", padj.total < 0.05 & lfc.total > 0 ~ "total up", padj.total < 0.05 & lfc.total < 0 ~ "total down"))
# ipsc_mn_als_datasets.polyA_total.sig
# ipsc_mn_als_datasets.polyA_total.sig %>% count(changed)
# write_csv(ipsc_mn_als_datasets.polyA_total.sig, here(proj_path,"expression/deseq2/Table.ipsc_mn_als_datasets.polyA_total.sig.csv"))

```

# Fig S10 Datasets independently


```{r}
load(here(proj_path, "expression/deseq2/datasets_separated.RData"))

```

## Volcano

```{r}
ipsc_mn_datasets_list.res = list("AnswerALS" = answerals$res, "neurolincs.diMN" = neurolincs.diMN$res, "neurolincs.iMN" = neurolincs.iMN$res, "catanese" = catanese$res, "dafinca C9orf72" = dafinca.c9orf72$res, "dafinca tardbp" = dafinca.tardbp$res, "luisier" = luisier$res, "wang" = wang$res, "kiskinis" = kiskinis$res, "sareen" = sareen$res, "sommer" = sommer$res, "sterneckert" = sterneckert$res, "kapeli" = kapeli$res, "desantis" = desantis$res, "smith" = smith$res, "bhinge" = bhinge$res, "hawkins" = hawkins$res)
ipsc_mn_datasets.res = bind_rows(ipsc_mn_datasets_list.res, .id = "dataset")

datasets_separated.volcano = volcano_plot(ipsc_mn_datasets.res, arrow_labels = FALSE, dpi = 300) + facet_wrap(~dataset, scales = "free", ncol = 6)
# datasets_separated.volcano

```


## Correlation heatmap

```{r}
# heatmap of correlation coefficients
ipsc_mn_datasets_stat = select(answerals$res, gene_id, answerals = stat) %>% 
  left_join(select(neurolincs.diMN$res, gene_id, neurolincs.diMN = stat)) %>%
  left_join(select(neurolincs.iMN$res, gene_id, neurolincs.iMN = stat)) %>%
  left_join(select(catanese$res, gene_id, catanese = stat)) %>%
  left_join(select(dafinca.c9orf72$res, gene_id, dafinca.c9orf72 = stat)) %>%
  left_join(select(dafinca.tardbp$res, gene_id, dafinca.tardbp = stat)) %>%
  left_join(select(luisier$res, gene_id, luisier = stat)) %>%
  left_join(select(wang$res, gene_id, wang = stat)) %>%
  left_join(select(kiskinis$res, gene_id, kiskinis = stat)) %>%
  left_join(select(sareen$res, gene_id, sareen = stat)) %>%
  left_join(select(sommer$res, gene_id, sommer = stat)) %>%
  left_join(select(sterneckert$res, gene_id, sterneckert = stat)) %>%
  left_join(select(kapeli$res, gene_id, kapeli = stat)) %>%
  left_join(select(desantis$res, gene_id, desantis = stat)) %>%
  left_join(select(smith$res, gene_id, smith = stat)) %>%
  left_join(select(bhinge$res, gene_id, bhinge = stat)) %>%
  left_join(select(hawkins$res, gene_id, hawkins = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_id")
cormat <- round(cor(ipsc_mn_datasets_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
dataset_dge_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = c(0.25,0.75), legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5))
dataset_dge_correlation_ggheatmap


```


## p53 PROGENy & TP53 Dorothea

```{r}
ipsc_mn_datasets_list = list("AnswerALS" = select(answerals$res,gene_name,stat), "neurolincs.diMN" = select(neurolincs.diMN$res,gene_name,stat), "neurolincs.iMN" = select(neurolincs.iMN$res,gene_name,stat), "catanese" = select(catanese$res,gene_name,stat), "dafinca C9orf72" = select(dafinca.c9orf72$res,gene_name,stat), "dafinca tardbp" = select(dafinca.tardbp$res,gene_name,stat), "luisier" = select(luisier$res,gene_name,stat), "wang" = select(wang$res,gene_name,stat), "kiskinis" = select(kiskinis$res,gene_name,stat), "sareen" = select(sareen$res,gene_name,stat), "sommer" = select(sommer$res,gene_name,stat), "sterneckert" = select(sterneckert$res,gene_name,stat), "kapeli" = select(kapeli$res,gene_name,stat), "desantis" = select(desantis$res,gene_name,stat), "smith" = select(smith$res,gene_name,stat), "bhinge" = select(bhinge$res,gene_name,stat), "hawkins" = select(hawkins$res,gene_name,stat))
datasets <- names(ipsc_mn_datasets_list)
ipsc_mn_datasets_list = map(ipsc_mn_datasets_list, ~{drop_na(.x)}) 
ipsc_mn_datasets_list = map(ipsc_mn_datasets_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_datasets_list = map(ipsc_mn_datasets_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_datasets_list.progeny_scores = map_df(ipsc_mn_datasets_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "dataset") %>% filter(statistic == 'norm_wmean')

ipsc_mn_datasets_list.p53.progeny_scores = ipsc_mn_datasets_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = dataset, group2 = dataset)
ipsc_mn_datasets_list.p53.progeny_scores.plot <- ggplot(ipsc_mn_datasets_list.p53.progeny_scores, aes(x=reorder(dataset, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #
  stat_pvalue_manual(ipsc_mn_datasets_list.p53.progeny_scores, label = "p.signif", y.position = 11.5, xmin = "dataset", xmax = NULL, size = 4, hide.ns = TRUE) + ylim(c(-12,12))
ipsc_mn_datasets_list.p53.progeny_scores.plot


# Dorothea
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_datasets_list.dorothea_scores = map_df(ipsc_mn_datasets_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "dataset") %>% filter(statistic == 'norm_wmean')

ipsc_mn_datasets_list.TP53.dorothea_scores = ipsc_mn_datasets_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = dataset, group2 = dataset)
ipsc_mn_datasets_list.TP53.dorothea_scores.plot <- ggplot(ipsc_mn_datasets_list.TP53.dorothea_scores, aes(x=reorder(dataset, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") +
  stat_pvalue_manual(ipsc_mn_datasets_list.TP53.dorothea_scores, label = "p.signif", y.position = 13, xmin = "dataset", xmax = NULL, size = 4, hide.ns = TRUE) + ylim(c(-13,13))
ipsc_mn_datasets_list.TP53.dorothea_scores.plot

```


## Merge

```{r}
ipsc_mn_datasets.merged = plot_grid(
  datasets_separated.volcano,# answerals.volcano, # ? include volcanos???
  dataset_dge_correlation_ggheatmap,
  plot_grid(ipsc_mn_datasets_list.p53.progeny_scores.plot, ipsc_mn_datasets_list.TP53.dorothea_scores.plot, ncol = 2, labels = letters[3:4]),
  labels = c("a","b",""), nrow = 3, rel_heights = c(1,0.8,1))
# ipsc_mn_datasets.merged
ggsave(ipsc_mn_datasets.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_datasets.merged.png"), height = 12, width = 10)


write_csv(answerals$res, here(proj_path,"expression/deseq2/answerals.als_vs_ctrl.csv"))

```


# Fig S11 Proteomics AnswerALS

Process

https://henkjanvdh.github.io/files/limmaPeptideTutorial.pdf 
https://www.bioconductor.org/packages/devel/bioc/vignettes/DEqMS/inst/doc/DEqMS-package-vignette.html
https://support.bioconductor.org/p/118495/

```{r}
answerals.proteomics.als_vs_ctrl.res = read_csv(here(shared_path,"/answerals/proteomics/answerals_proteomics_als_vs_ctrl.csv"))
answerals.proteomics.als_vs_ctrl.res #4486

```


QC Mass Spec Answer ALS

```{r}
answerals_proteomics_intensities = read_csv(here(shared_path,"/answerals/proteomics/AnswerALS-204-P-v1-release5_protein-level-matrix.csv"))
answerals_proteomics_intensities %>% glimpse #204 samples (cols), 4,442 proteins (rows)

# sample annotation
colnames(answerals_proteomics_intensities) = gsub("-","_",colnames(answerals_proteomics_intensities))
colnames(answerals_proteomics_intensities) = str_split_fixed(colnames(answerals_proteomics_intensities), "_[0-9]+",n=2)[,1]
answerals.metadata %>% filter(sample %in% colnames(answerals_proteomics_intensities)) #197 / 204 samples have RNAseq
targets = tibble(sample = colnames(answerals_proteomics_intensities)) %>% filter(sample!="Protein") %>% mutate(condition = case_when(grepl("CASE",sample)~"als",grepl("CTRL",sample)~"ctrl")) %>% group_split(condition) %>% purrr::map_df(~.x %>% group_by(sample) %>% mutate(replicate = cur_group_id())) %>% ungroup
targets %>% count(condition) # 33 ctrl, 171 als

# log2 transform intensities
answerals_proteomics_intensities.mat = answerals_proteomics_intensities %>% tibble_to_matrix(everything(), -Protein, row_names = "Protein")
answerals_proteomics_intensities.mat
boxplot(answerals_proteomics_intensities.mat,las=2,main="AnswerALS intensitites before log2")

intensities = log2(answerals_proteomics_intensities.mat)
intensities[intensities ==-Inf] = NA # intensities = na.omit(intensities) #remove rows with NAs
boxplot(intensities,las=2,main="AnswerALS intensitites after log2")

# design & contrast
library(limma)
# limmaUserGuide()
ALS = targets %>% mutate(ALS = case_when(grepl("CASE",sample)~1,grepl("CTRL",sample)~0)) %>% tibble_to_matrix(ALS)
CTRL = targets %>% mutate(CTRL = case_when(grepl("CASE",sample)~0,grepl("CTRL",sample)~1)) %>% tibble_to_matrix(CTRL)
design = cbind(ALS, CTRL)+0
head(design)
contr = makeContrasts( ALS - CTRL, levels = design)
fit = lmFit(intensities, design) # linear model
fit2 = contrasts.fit(fit, contr)
fit2 = eBayes(fit2, trend = TRUE, robust = TRUE) # https://support.bioconductor.org/p/118495/
fit2 # 4442 peptides compared

# QC figures
proteomics_mds = plotMDS(intensities, pch = 1, cex = 0.5, col=ifelse(targets$condition=="ctrl","blue","red"))
proteomics_mds.toplot <- data.frame(Dim1 = proteomics_mds$x, Dim2 = proteomics_mds$y, Condition = targets$condition)
proteomics_mds.plot = ggplot(proteomics_mds.toplot, aes(Dim1, Dim2, colour = Condition)) + geom_point(size=1) +
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top")) +
  xlab(paste0("PC1: 29% variance")) +  ylab(paste0("PC2: 11% variance"))
proteomics_mds.plot

proteomics_mds.toplot <- data.frame(sigma.sqrt = sqrt(fit2$sigma), Dim2 = res$AveExpr, Condition = targets$condition)


plotSA(fit2)
p = recordPlot()
plot.new()
p

library(gridGraphics)
library(grid)
grid.echo()
mean_var_plot = grid.grab()
grid.newpage()
mean_var_plot = editGrob(mean_var_plot)
grid.draw(mean_var_plot) # mean-variance relationship
as.ggplot(mean_var_plot)
answerals.proteomics.qc.fig.merged = ggarrange(proteomics_mds.plot, NULL, p, nrow = 3, heights = c(1,0.25,1), labels = c("a","","b"))
answerals.proteomics.qc.fig.merged
ggsave(answerals.proteomics.qc.fig.merged, filename = here(shared_path,"answerals/proteomics/answerals_proteomics_qc_mds_sa.png"), height = 10.5, width = 8)
# 
# 
# # results
# topTable(fit2, coef = 'ALS - CTRL', sort.by = "t", lfc=1, n=5)
# topTable(fit2, coef = 'ALS - CTRL', lfc=1,  n=5)
# res = topTable(fit2, coef = 'ALS - CTRL', n = 5000) %>% as_tibble(rownames = "peptide_id") %>% 
#   mutate(protein_name = str_split_fixed(peptide_id,"\\|",2)[,1], gene_name = str_split_fixed(peptide_id,"\\|",2)[,2])
# 
# # annotate using protein names http://www.bioconductor.org/packages/release/bioc/vignettes/UniProt.ws/inst/doc/UniProt.ws.pdf
# library(UniProt.ws)
# availableUniprotSpecies(pattern="Homo")
# humanUp<-UniProt.ws(9606)
# # humanUp
# # keytypes(humanUp)
# # columns(humanUp)
# prot2gene = UniProt.ws::select(humanUp,res$protein_name, "GENENAME", "UNIPROTKB_ID") 
# # prot2gene
# res = res %>% select(-gene_name) %>% left_join(rename(prot2gene, protein_name = UNIPROTKB_ID, gene_name = GENENAME)) %>% left_join(gene2ens)
# res %>% drop_na(gene_id)
# answerals.proteomics.als_vs_ctrl.res = res %>% select(protein_name, gene_name, gene_id, baseMean = AveExpr, log2FoldChange = logFC, pvalue = P.Value, padj = adj.P.Val, stat = t, B)

```


## Volcano & GO & GSEA
 
```{r}
answerals.proteomics.als_vs_ctrl.res %>% filter(log2FoldChange > 0) #2587
2587/4486
filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05) # 276
filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05, gene_name %in% ALS_genes) # 7
filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05, gene_name %in% p53.signalling.pathway) # RBBP7, CSNK2B, PRMT1, CNOT9 
filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05, gene_name %in% pull(filter(answerals.proteomics.als_vs_ctrl.res, padj < 0.05),gene_name)) # EEFSEC, MACROD1
filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05, gene_name %in% pull(filter(als_ctrl.voila, changing_dpsi0.1 == TRUE),gene_name)) # EEFSEC, MACROD1
filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 46
filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05, gene_name %in% nucleocytoplasmic_genes) %>% print(n=Inf) # 46


# volcano
volcanoplot(fit2)
answerals.proteomics.als_vs_ctrl.volcano = volcano_plot(answerals.proteomics.als_vs_ctrl.res, title = "Protein Expression", subtitle = "171 ALS vs 33 CTRL", 
  gene_labels = c(pull(top_n(answerals.proteomics.als_vs_ctrl.res,n=10,wt=stat),gene_name), pull(filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05, gene_name %in% c(ALS_genes,p53.signalling.pathway)),gene_name), "TARDBP","FUS","SFPQ","SOD1", "BRD4", "EEF2"),
                  ymax = 4,xmax = 2,numerator = "ALS",denomenator = "CTRL", significance_threshold = "pvalue", dpi = 300)
answerals.proteomics.als_vs_ctrl.volcano

# MA plot
plotMD(fit2)
ma_plot_deseq(answerals.proteomics.als_vs_ctrl.res,labels_list = pull(top_n(answerals.proteomics.als_vs_ctrl.res,n=10,wt=stat), gene_name))

# GO terms
answerals.proteomics.als_vs_ctrl.gost <- answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
answerals.proteomics.als_vs_ctrl.gprofiles <- answerals.proteomics.als_vs_ctrl.gost$result %>% filter(!str_detect(source, "TF|CORUM|HPA")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "ALS\nDown", query == "query_2" ~ "ALS\nUp"), term_name = as.factor(term_name))
answerals.proteomics.als_vs_ctrl.gprofiles_down <- answerals.proteomics.als_vs_ctrl.gprofiles %>% filter(query == "ALS\nDown")
answerals.proteomics.als_vs_ctrl.gprofiles_up <- answerals.proteomics.als_vs_ctrl.gprofiles %>% filter(query == "ALS\nUp")
paste('"',unique(answerals.proteomics.als_vs_ctrl.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"Schaffer collateral - CA1 synapse",
"Golgi transport complex")
answerals.proteomics.als_vs_ctrl.gprofiles_filt_down <- answerals.proteomics.als_vs_ctrl.gprofiles_down %>% filter(query == "ALS\nDown")  %>% filter(term_name %in% down_list)
paste('"',unique(answerals.proteomics.als_vs_ctrl.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"proteasome complex",
"Axon guidance",
"nucleocytoplasmic transport",
"Cholesterol biosynthesis",
"Amyotrophic lateral sclerosis",
"protein modification process",
"Metabolism of proteins",
"Cellular responses to stress",
"protein localization",
"endoplasmic reticulum",
"protein binding",
"Metabolism of RNA",
"translation",
"protein metabolic process")
# "cytoplasm")
answerals.proteomics.als_vs_ctrl.gprofiles_filt_up <- answerals.proteomics.als_vs_ctrl.gprofiles_up %>% filter(query == "ALS\nUp")  %>% filter(term_name %in% up_list)
answerals.proteomics.als_vs_ctrl.gprofiles_filt_combined <- bind_rows(answerals.proteomics.als_vs_ctrl.gprofiles_filt_down, answerals.proteomics.als_vs_ctrl.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c("ALS\nUp", "ALS\nDown")))
answerals.proteomics.als_vs_ctrl.go_curated <- curated_profiles(gprofiles = answerals.proteomics.als_vs_ctrl.gprofiles_filt_combined,label_angle = 0) + theme(legend.position = "none", axis.text = element_text(size=10))
# answerals.proteomics.als_vs_ctrl.go_curated

plot_grid(answerals.proteomics.als_vs_ctrl.volcano, answerals.proteomics.als_vs_ctrl.go_curated, ncol = 2)

answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05, gene_name %in% go.pathways.msigdb$GO_POST_TRANSLATIONAL_PROTEIN_MODIFICATION)
answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05, gene_name %in% go.pathways.msigdb$GO_RNA_METABOLIC_PROCESS)
answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05, gene_name %in% RNA_BINDING_PROTEINS)
answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05, gene_name %in% ALS_genes)
answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05, gene_name %in% go.pathways.msigdb$GO_RNA_METABOLIC_PROCESS)
answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05, gene_name %in% p53.signalling.pathway)
answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05, gene_name %in% ALS_genes)

## GSEA: DNA damage response, p53 signalling pathway
answerals.proteomics.als_vs_ctrl.geneList <- answerals.proteomics.als_vs_ctrl.res %>% drop_na(stat) %>% pull(stat)
names(answerals.proteomics.als_vs_ctrl.geneList) <- answerals.proteomics.als_vs_ctrl.res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
answerals.proteomics.als_vs_ctrl.geneList = sort(answerals.proteomics.als_vs_ctrl.geneList, decreasing = TRUE)

# p53 signalling pathway
p53.signalling_genes.list <- list("p53.signalling" = answerals.proteomics.als_vs_ctrl.res$gene_name[answerals.proteomics.als_vs_ctrl.res$gene_name %in% p53.signalling.pathway])
answerals.proteomics.als_vs_ctrl.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = answerals.proteomics.als_vs_ctrl.geneList)
answerals.proteomics.als_vs_ctrl.p53.signalling_fgseaRes
# pathway      pval      padj    log2err        ES       NES size                             leadingEdge
# 1: p53.signalling 0.5122995 0.5122995 0.04763873 0.2633203 0.9831828  103 RBBP7,CSNK2B,PRMT1,CNOT9,RPL23,RRM2,...
answerals.proteomics.als_vs_ctrl.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], answerals.proteomics.als_vs_ctrl.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("p53 signalling") + xlab("") +
  annotate("text", x = 3000, y = 0.2, label = paste0("NES = ", round(answerals.proteomics.als_vs_ctrl.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(answerals.proteomics.als_vs_ctrl.p53.signalling_fgseaRes$pval, digits = 6)), size = 2.8, hjust = 0) + theme_oz()
answerals.proteomics.als_vs_ctrl.p53.signalling.gseaplot

```




## Protein & RNA correlation

```{r}
# Compare RNAseq to mass spec
ipsc_mn_rnaseq_mass_spec.join = select(answerals.proteomics.als_vs_ctrl.res, gene_name, gene_id, MS.lfc = log2FoldChange, MS.pvalue = pvalue, MS.padj = padj, MS.stat = stat) %>% 
  left_join(select(ipsc_mn_als_datasets$res, gene_name, gene_id, RNA.lfc = log2FoldChange, RNA.pvalue = pvalue, RNA.padj = padj, RNA.stat = stat))
ipsc_mn_rnaseq_mass_spec.join %>% filter(RNA.padj < 0.05, MS.pvalue < 0.05) # MACROD1, EEFSEC1
ipsc_mn_rnaseq_mass_spec.join %>% filter(RNA.lfc > 0.2, MS.lfc > 0.2)
ipsc_mn_rnaseq_mass_spec.join %>% filter(RNA.lfc < -0.2, MS.lfc < -0.2)

# Correlate RNAseq & Mass Spec
ipsc_mn_rnaseq_mass_spec = list("RNA-seq: ALS vs CTRL" = ipsc_mn_als_datasets$res, "Mass Spec: ALS vs CTRL" = answerals.proteomics.als_vs_ctrl.res)
ipsc_mn_rnaseq_mass_spec.scatter = plot_de_correlation(model_list1 = ipsc_mn_rnaseq_mass_spec, model_list2 = ipsc_mn_rnaseq_mass_spec, mutation1 = "RNA-seq: ALS vs CTRL", mutation2 = "Mass Spec: ALS vs CTRL", col = "log2FoldChange", gene_labels = c("MACROD1","EEFSEC","EIF1AY","H3C4","CEMIP","PTGR1","HOXC8","KRT1","IGFBP5","TPX2", "TARDBP","FUS","SFPQ","SOD1"), plot_p_value = TRUE) & xlim(-0.9,0.6) & ylim(-1.8,1.9)
ipsc_mn_rnaseq_mass_spec.scatter

# known ALS RBPs
ipsc_mn_rnaseq_mass_spec.join %>% filter(gene_name %in% c("TARDBP","FUS","SFPQ"))

# ggsave(ipsc_mn_rnaseq_mass_spec.scatter, filename = here(proj_path,"expression/deseq2/ipsc_mn_rnaseq_mass_spec.scatter.png", height = 14, width = 10)


```


## Merge

```{r}
answerals_proteomics = ggarrange(
  # ggarrange(proteomics_mds.plot, NULL, p, ncol = 3, widths = c(1,0.25,1), labels = c("a","","b")),# QC PCA
  answerals.proteomics.als_vs_ctrl.volcano, answerals.proteomics.als_vs_ctrl.go_curated, 
  answerals.proteomics.als_vs_ctrl.p53.signalling.gseaplot, ipsc_mn_rnaseq_mass_spec.scatter, 
  nrow = 2, ncol = 2, labels = letters[1:4])  # nrow = 2, heights = c(1,2))#, labels = c("","","",""))
# answerals_proteomics
ggsave(answerals_proteomics, filename = here(proj_path,"expression/deseq2/answerals_proteomics.png"), height = 10, width = 10)


```



# Table S5 Proteomics

```{r}
write_csv(answerals.proteomics.als_vs_ctrl.res, here(shared_path,"/answerals/proteomics/answerals_proteomics_als_vs_ctrl.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=650497827")
write_sheet(answerals.proteomics.als_vs_ctrl.res, ss = "https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=650497827", sheet = "S5 Proteomics")


```


# Fig 3 Compare mutations iPSMNs

Compare sporadic, C9orf72, SOD1, FUS & TARDBP

```{r}
load(file = here(proj_path, "expression/deseq2/ipsc_mn_genetic_groups.RData"))
# ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
# ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds")) # 84M
# ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds")) # 162M
# ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
# ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds")) # 85M


ipsc_mn_sporadic_datasets$res %>% filter(gene_name %in% "POU5F1")

```

### Volcano

```{r}
ipsc_mn_mutants_bind_datasets.res = bind_rows(mutate(ipsc_mn_sporadic_datasets$res, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$res, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$res, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$res, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$res, mutation = "FUS"))

ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% count(mutation) %>% arrange(-n)
# mutation     n
#   <chr>    <int>
# 1 TARDBP    3547
# 2 FUS        202
# 3 C9orf72    161
# 4 SOD1         7
# 5 Sporadic     4
ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05)

ipsc_mn_mutants_join_datasets.res = full_join(select(ipsc_mn_c9orf72_datasets$res, gene_name, padj.c9orf72 = padj, log2FoldChange.c9orf72 = log2FoldChange),
          select(ipsc_mn_sporadic_datasets$res, gene_name, padj.sals = padj, log2FoldChange.sals = log2FoldChange)) %>%
  full_join(select(ipsc_mn_fus_datasets$res, gene_name, padj.fus = padj, log2FoldChange.fus = log2FoldChange)) %>%
  full_join(select(ipsc_mn_tardbp_datasets$res, gene_name, padj.tardbp = padj, log2FoldChange.tardbp = log2FoldChange)) %>%
  full_join(select(ipsc_mn_sod1_datasets$res, gene_name, padj.sod1 = padj, log2FoldChange.sod1 = log2FoldChange)) #%>% full_join(select(ipsc_mn_vcp_datasets$res, gene_name, padj.vcp = padj, log2FoldChange.vcp = log2FoldChange))
ipsc_mn_mutants_join_datasets.res.labels = ipsc_mn_mutants_join_datasets.res %>% filter(c(padj.c9orf72 < 0.05 & padj.fus < 0.05 & padj.tardbp < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.tardbp < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.tardbp < 0.05 & padj.c9orf72 < 0.05)) %>% pull(gene_name)

ipsc_mn_tardbp_datasets.metadata %>% count(dataset)
ipsc_mn_tardbp_datasets.metadata %>% count(condition) 
ipsc_mn_tardbp_datasets.labels <- ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.001, gene_name %in% c(ALS_genes,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_tardbp_datasets.volcano <- volcano_plot(ipsc_mn_tardbp_datasets$res, "TARDBP", numerator = "TARDBP", denomenator = "CTRL", subtitle = "48 Controls vs 10 TARDBP", ymax = 15, xmax = 4.5, dpi = 300, gene_labels = c(ipsc_mn_tardbp_datasets.labels,"UPK3BL1","NPIPA8"))
# ipsc_mn_tardbp_datasets.volcano

ipsc_mn_fus_datasets.metadata %>% count(dataset) 
ipsc_mn_fus_datasets.metadata %>% count(condition) 
ipsc_mn_fus_datasets$res %>% filter(padj < 0.01) %>% top_n(n=10, wt = abs(stat))
ipsc_mn_fus_datasets.labels <- ipsc_mn_fus_datasets$res %>% filter(padj < 0.05, gene_name %in% c(p53.DDR.repair_gene_set, ALS_genes,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_fus_datasets.volcano <- volcano_plot(ipsc_mn_fus_datasets$res, "FUS", numerator = "FUS", denomenator = "CTRL", subtitle = "55 Controls vs 14 FUS", ymax = 15, xmax = 4, dpi = 300, gene_labels = c(ipsc_mn_fus_datasets.labels,"UPK3BL1","NPIPA8","ZNF439","HSD11B2","RNF5P1","RNU1-3"))
# ipsc_mn_fus_datasets.volcano

ipsc_mn_c9orf72_datasets.metadata %>% count(dataset)
ipsc_mn_c9orf72_datasets.metadata %>% count(condition) 
ipsc_mn_c9orf72_datasets.labels <- ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.02, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_c9orf72_datasets.volcano <- volcano_plot(ipsc_mn_c9orf72_datasets$res, "C9orf72", numerator = "C9orf72", denomenator = "CTRL", subtitle = "83 Controls vs 60 C9orf72", ymax = 9, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_c9orf72_datasets.labels, "C9orf72","UPK3BL1","NPIPA8"))
# ipsc_mn_c9orf72_datasets.volcano

ipsc_mn_sod1_datasets.metadata %>% count(dataset) 
ipsc_mn_sod1_datasets.metadata %>% count(condition) 
ipsc_mn_sod1_datasets.labels <- ipsc_mn_sod1_datasets$res %>% filter(padj < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_sod1_datasets.volcano <- volcano_plot(ipsc_mn_sod1_datasets$res, "SOD1", numerator = "SOD1", denomenator = "CTRL", subtitle = "63 Controls vs 20 SOD1", ymax = 12, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_sod1_datasets.labels))
# ipsc_mn_sod1_datasets.volcano

ipsc_mn_sporadic_datasets.metadata %>% count(dataset) 
ipsc_mn_sporadic_datasets.metadata %>% count(condition) 
ipsc_mn_sporadic_datasets.labels <- ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_sporadic_datasets.volcano <- volcano_plot(ipsc_mn_sporadic_datasets$res, "Sporadic", numerator = "Sporadic", denomenator = "CTRL", subtitle = "56 Controls vs 208 Sporadic", ymax = 12, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_sporadic_datasets.labels))
# ipsc_mn_sporadic_datasets.volcano

volcano_multiplot <- ipsc_mn_tardbp_datasets.volcano + ipsc_mn_fus_datasets.volcano + ipsc_mn_c9orf72_datasets.volcano + ipsc_mn_sod1_datasets.volcano + ipsc_mn_sporadic_datasets.volcano +
  plot_layout(ncol = 5, nrow = 1)
volcano_multiplot
ggsave(volcano_multiplot, filename = here(proj_path,"expression/deseq2/genetic_subgroups_volcano_multiplot.png"), height = 4, width = 11)

ipsc_mn_mutants_join_datasets.res.labels
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% ALS_genes]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% RNA_BINDING_PROTEINS]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% p53.signalling.pathway]

```


### Correlation heatmap

```{r}
# heatmap of correlation coefficients
ipsc_mn_mutations_stat = select(ipsc_mn_sporadic_datasets$res, gene_id, Sporadic = stat) %>% 
  left_join(select(ipsc_mn_c9orf72_datasets$res, gene_id, C9orf72 = stat)) %>%
  left_join(select(ipsc_mn_sod1_datasets$res, gene_id, SOD1 = stat)) %>%
  left_join(select(ipsc_mn_fus_datasets$res, gene_id, FUS = stat)) %>%
  left_join(select(ipsc_mn_tardbp_datasets$res, gene_id, TARDBP = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_id")
cormat <- round(cor(ipsc_mn_mutations_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
dge_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5))
dge_correlation_ggheatmap

mutation_volcanos_corr_heatmap <- ipsc_mn_tardbp_datasets.volcano + ipsc_mn_fus_datasets.volcano + ipsc_mn_c9orf72_datasets.volcano + ipsc_mn_sod1_datasets.volcano + ipsc_mn_sporadic_datasets.volcano + dge_correlation_ggheatmap +
  plot_layout(ncol = 3, nrow = 2)
# mutation_volcanos_corr_heatmap

ipsc_mn_mutants_join_datasets.res = full_join(select(ipsc_mn_c9orf72_datasets$res, gene_name, gene_id, padj.c9orf72 = padj, log2FoldChange.c9orf72 = log2FoldChange, stat.c9orf72 = stat),
          select(ipsc_mn_sporadic_datasets$res, gene_name, gene_id, padj.sals = padj, log2FoldChange.sals = log2FoldChange, stat.sals = stat)) %>%
  full_join(select(ipsc_mn_fus_datasets$res, gene_name, gene_id, padj.fus = padj, log2FoldChange.fus = log2FoldChange, stat.fus = stat)) %>%
  full_join(select(ipsc_mn_tardbp_datasets$res, gene_name, gene_id, padj.tardbp = padj, log2FoldChange.tardbp = log2FoldChange, stat.tardbp = stat)) %>%
  full_join(select(ipsc_mn_sod1_datasets$res, gene_name, gene_id, padj.sod1 = padj, log2FoldChange.sod1 = log2FoldChange, stat.sod1 = stat)) #%>% full_join(select(ipsc_mn_vcp_datasets$res, gene_name, padj.vcp = padj, log2FoldChange.vcp = log2FoldChange))

ipsc_mn_mutants_join_datasets.res %>% filter(padj.sod1 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # UPK3BL1, NPIPA8
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.sod1 < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.sod1 < 0.05) # 

ipsc_mn_mutants_join_datasets.res %>% filter(stat.sod1 > 1, stat.sals > 1, stat.c9orf72 > 1, stat.fus > 1, stat.tardbp > 1) %>% filter(gene_name %in% c(ALS_genes,p53.signalling.pathway))
ipsc_mn_mutants_join_datasets.res %>% filter(stat.sod1 < -1, stat.sals < -1, stat.c9orf72 < -1, stat.fus < -1, stat.tardbp < -1) %>% filter(gene_name %in% c(ALS_genes,p53.signalling.pathway))


  padj.sod1 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # UPK3BL1, NPIPA8
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.sod1 < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.sod1 < 0.05) # 


# # Overlap ALL DEGs
# ipsc_mn_mutants_bind_datasets.dir.res = bind_rows(mutate(filter(ipsc_mn_c9orf72_datasets$res, log2FoldChange > 0), mutation = "C9orf72 Up"), mutate(filter(ipsc_mn_c9orf72_datasets$res, log2FoldChange < 0), mutation = "C9orf72 Down"),
#                                               mutate(filter(ipsc_mn_tardbp_datasets$res, log2FoldChange > 0), mutation = "TARDBP Up"), mutate(filter(ipsc_mn_tardbp_datasets$res, log2FoldChange < 0), mutation = "TARDBP Down"),
#                                               mutate(filter(ipsc_mn_fus_datasets$res, log2FoldChange > 0), mutation = "FUS Up"), mutate(filter(ipsc_mn_fus_datasets$res, log2FoldChange < 0), mutation = "FUS Down"),
#                                               mutate(filter(ipsc_mn_sod1_datasets$res, log2FoldChange > 0), mutation = "SOD1 Up"), mutate(filter(ipsc_mn_sod1_datasets$res, log2FoldChange < 0), mutation = "SOD1 Down"),
# ipsc_mn_mutants_bind_datasets.res.direction.upset = ipsc_mn_mutants_bind_datasets.dir.res %>% filter(padj < 0.05) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#   geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.4, hjust = -0.2, size = 2.5, angle = 45) +
#   scale_x_upset(order_by = "degree")  + # n_intersections = 20
#   scale_y_continuous(breaks = NULL, lim = c(0, 2700), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=6), axis.title=element_text(size=8)) +
#   ylab(expression()) +
#   theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.res.direction.upset
# 
# # Overlap UP DEGs
# ipsc_mn_mutants_bind_datasets.upregulated.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#       geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, size = 2.8) +
#     scale_x_upset(order_by = "degree")  + # n_intersections = 20
#     scale_y_continuous(breaks = NULL, lim = c(0, 2500), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + 
#       ylab(expression()) +
#     theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.upregulated.res.upset
# 
# # Overlap DOWN DEGs
# ipsc_mn_mutants_bind_datasets.downregulated.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#       geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, size = 2.8) +
#     scale_x_upset(order_by = "degree")  + # n_intersections = 20
#     scale_y_continuous(breaks = NULL, lim = c(0, 3000), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + 
#       ylab(expression()) +
#     theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.downregulated.res.upset

```


## p53 PROGENy & TP53 Dorothea

```{r}
ipsc_mn_mutations_list = list("Sporadic" = select(ipsc_mn_sporadic_datasets$res,gene_name,stat), "C9orf72" = select(ipsc_mn_c9orf72_datasets$res,gene_name,stat), "TARDBP" = select(ipsc_mn_tardbp_datasets$res,gene_name,stat), "SOD1" = select(ipsc_mn_sod1_datasets$res,gene_name,stat), "FUS" = select(ipsc_mn_fus_datasets$res,gene_name,stat))
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{drop_na(.x)}) 
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})


# PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')
ipsc_mn_mutations_list.p53.progeny_scores = ipsc_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(decoupleR = "p53 pathway") 

# Dorothea
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')
ipsc_mn_mutations_list.TP53.dorothea_scores = ipsc_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(decoupleR = "TP53 TF") 

ipsc_mn_mutations_list.p53.TP53 = bind_rows(ipsc_mn_mutations_list.p53.progeny_scores, ipsc_mn_mutations_list.TP53.dorothea_scores) %>% 
  mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, mutation = factor(mutation, levels = c("SOD1","FUS","Sporadic","TARDBP","C9orf72")))

ipsc_mn_mutations_list.p53.TP53.plot <- ggplot(ipsc_mn_mutations_list.p53.TP53, aes(x=mutation, y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
  theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust = 1), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
  stat_pvalue_manual(ipsc_mn_mutations_list.p53.TP53, label = "p.signif", y.position = 11.5, xmin = "mutation", xmax = NULL, size = 5, hide.ns = TRUE) +
  facet_wrap(~decoupleR, scales = "free")
ipsc_mn_mutations_list.p53.TP53.plot


# # PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
# ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')
# 
# # ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP","VCP")))) +
# #   geom_bar(stat='identity', position = "dodge2") +
# #   scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
# #   theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
# #   geom_hline(yintercept = 0, linetype = 3) +
# #   labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment") 
# # ipsc_mn_mutations_list.progeny_scores.plot
# 
# ipsc_mn_mutations_list.p53.progeny_scores = ipsc_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
# ipsc_mn_mutations_list.p53.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.p53.progeny_scores, aes(x=reorder(mutation, score), y = score)) +
#     geom_bar(aes(fill = score), stat = "identity") +
#   scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
#   theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
#   geom_hline(yintercept = 0, linetype = 3) +
#   labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
#   stat_pvalue_manual(ipsc_mn_mutations_list.p53.progeny_scores, label = "p.signif", y.position = 13, xmin = "mutation", xmax = NULL, size = 6, hide.ns = TRUE) 
# ipsc_mn_mutations_list.p53.progeny_scores.plot
# 
# 
# 
# # Dorothea
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')
# 
# ipsc_mn_mutations_list.TP53.dorothea_scores = ipsc_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
# ipsc_mn_mutations_list.TP53.dorothea_scores.plot <- ggplot(ipsc_mn_mutations_list.TP53.dorothea_scores, aes(x=reorder(mutation, score), y = score)) +
#     geom_bar(aes(fill = score), stat = "identity") +
#   scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
#   theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
#   geom_hline(yintercept = 0, linetype = 3) +
#   labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") + #ylim(c(-11,11)) +
#   stat_pvalue_manual(ipsc_mn_mutations_list.TP53.dorothea_scores, label = "p.signif", y.position = 11, xmin = "mutation", xmax = NULL, size = 6, hide.ns = TRUE) 
# ipsc_mn_mutations_list.TP53.dorothea_scores.plot

```


## Merge

```{r}
mutations_compared_fig.merged = plot_grid(
  plot_grid(ipsc_mn_tardbp_datasets.volcano, ipsc_mn_fus_datasets.volcano, ipsc_mn_c9orf72_datasets.volcano, ipsc_mn_sod1_datasets.volcano, ncol = 4, labels = c("a","b","c","d")),
  plot_grid(ipsc_mn_sporadic_datasets.volcano, dge_correlation_ggheatmap, ipsc_mn_mutations_list.p53.TP53.plot, ncol = 3, rel_widths = c(1,1,1.5), labels = c("e","f","g")),
  nrow = 2)
# mutations_compared_fig.merged
ggsave(mutations_compared_fig.merged, filename = here(proj_path,"expression/deseq2/mutations_compared_fig.merged.png"), height = 7, width = 11)
ggsave(mutations_compared_fig.merged, filename = here(proj_path,"expression/deseq2/mutations_compared_fig.merged.pdf"), height = 7, width = 11)



ipsc_mn_mutations_list.decoupleR = plot_grid(ipsc_mn_mutations_list.p53.progeny_scores.plot, ipsc_mn_mutations_list.TP53.dorothea_scores.plot, ncol = 2, rel_widths = c(1,1))
ggsave(ipsc_mn_mutations_list.decoupleR, filename = here(proj_path,"expression/deseq2/ipsc_mn_mutations_list.decoupleR.png"), height = 4, width = 10)



```


# Fig S10 Gene expression overlap mutations 

## Scatters

Scatter correlation of mutants

```{r}
ipsc_mn_mutations_list = list("Sporadic" = ipsc_mn_sporadic_datasets$res, "C9orf72" = ipsc_mn_c9orf72_datasets$res, "TARDBP" = ipsc_mn_tardbp_datasets$res, "SOD1" = ipsc_mn_sod1_datasets$res, "FUS" = ipsc_mn_fus_datasets$res)#, "panALS" = ipsc_mn_als_datasets$res)

ipsc_mn_mutations_list_t_stat_plot <- 
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", gene_labels = "C9orf72", plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", gene_labels = "SOD1", plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", gene_labels = "FUS", plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", gene_labels = "TARDBP", plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", gene_labels = c("C9orf72","SOD1"), plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", gene_labels = c("C9orf72","FUS"), plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", gene_labels = c("C9orf72","TARDBP"), plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", gene_labels = c("SOD1","FUS"), plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", gene_labels = c("SOD1","TARDBP"), plot_p_value = TRUE)  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", gene_labels = c("TARDBP","FUS"), plot_p_value = TRUE)  +
  plot_layout(ncol = 4, nrow = 3) &  
  xlim(-8,8) & ylim(-8,8)
ipsc_mn_mutations_list_t_stat_plot
# ggsave(ipsc_mn_mutations_list_t_stat_plot, filename = here(proj_path,"expression/deseq2/dge.mutation.stat.scatters.merged.png"), height = 10, width = 7)

# # log2FoldChange
# lfc_plot <- 
#   plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "TARDBP", mutation2 = "VCP", col = "log2FoldChange") &  
#   xlim(-8,8) & ylim(-8,8)  +
#   plot_layout(ncol = 3, nrow = 5)
# lfc_plot
# ggsave(lfc_plot, filename = here(proj_path,"expression/deseq2/dge.mutation.lfc.scatters.merged.png", height = 13, width = 10)

```

## Upset plot

```{r}
### Upset plot https://github.com/const-ae/ggupset
ipsc_mn_mutants_bind_datasets.res = bind_rows(mutate(ipsc_mn_sporadic_datasets$res, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$res, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$res, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$res, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$res, mutation = "FUS"))# ,mutate(ipsc_mn_vcp_datasets$res, mutation = "VCP"))
ipsc_mn_mutants_bind_datasets.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
  ggplot(aes(x = mutations)) + geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.2, hjust = 0.5, size = 3, angle = 0) +
    scale_x_upset(order_by = "degree")  + # n_intersections = 20
    scale_y_continuous(breaks = NULL, lim = c(0, 4000), name = "") +
    theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + labs(x = "", y = "") +
    theme_combmatrix(combmatrix.label.make_space = TRUE)
ipsc_mn_mutants_bind_datasets.res.upset

```

## GO terms

```{r}
# c9orf72
ipsc_mn_c9orf72_datasets.dge_genes <- ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt <- ipsc_mn_c9orf72_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " C9orf72 Down ", query == "query_2" ~ " C9orf72 Up "), term_name = as.factor(term_name))
ipsc_mn_c9orf72_datasets.dge_gprofiles_down <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ")
ipsc_mn_c9orf72_datasets.dge_gprofiles_up <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")
paste('"',unique(ipsc_mn_c9orf72_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "hedgehog family protein binding",
"microtubule bundle formation",
"cytoplasmic region",
"cytoskeleton organization",
"axoneme assembly",
"cilium assembly")
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_down <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_c9orf72_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest",
"miRNA regulation of p53 pathway in prostate cancer",
# "miR-517 relationship with ARCN1 and USP1",
# "retrograde vesicle-mediated transport, Golgi to endoplasmic reticulum",
# "phospholipid transporter activity",
"Genotoxicity pathway",
"p53 signaling pathway")
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_up <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_down, ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" C9orf72 Up ", " C9orf72 Down ")))
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name <- gsub("DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest", "DNA damage response by p53", ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" pathway in prostate cancer", "", ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_c9orf72_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
# ipsc_mn_c9orf72_datasets.go_curated

# fus
ipsc_mn_fus_datasets.dge_genes <- ipsc_mn_fus_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_fus_datasets.dge_gprofiles_filt <- ipsc_mn_fus_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " FUS Down ", query == "query_2" ~ " FUS Up "), term_name = as.factor(term_name))
ipsc_mn_fus_datasets.dge_gprofiles_down <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Down ")
ipsc_mn_fus_datasets.dge_gprofiles_up <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Up ")
paste('"',unique(ipsc_mn_fus_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"positive regulation of oxidative phosphorylation",
"BAT3 complex binding",
"cell-cell adhesion via plasma-membrane adhesion molecules",
# "Babinski sign",
# "Abnormal superficial reflex",
# "Abnormality of salivation",
"synaptic signaling",
# "trans-synaptic signaling",
"chemical synaptic transmission")
ipsc_mn_fus_datasets.dge_gprofiles_filt_down <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_fus_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"regulation of RNA metabolic process",
# "DNA binding",
"transcription regulator activity",
# "DNA-binding transcription factor activity, RNA polymerase II-specific",
# "DNA-binding transcription factor activity",
"transcription by RNA polymerase II",
"double-stranded DNA binding",
"transcription cis-regulatory region binding")
ipsc_mn_fus_datasets.dge_gprofiles_filt_up <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_fus_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_fus_datasets.dge_gprofiles_filt_down, ipsc_mn_fus_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" FUS Up ", " FUS Down ")))
ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name <- gsub("transcription cis-regulatory region binding", "transcription regulatory binding", ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name) %>% gsub("positive regulation of ","",.)  %>% gsub("regulation of ","",.) %>% gsub("cell-cell adhesion via plasma-membrane adhesion molecules", "cell membrane adhesion",.)
# ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" pathway", "", ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_fus_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_fus_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
ipsc_mn_fus_datasets.go_curated

# tardbp
ipsc_mn_tardbp_datasets.dge_genes <- ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_tardbp_datasets.dge_gprofiles_filt <- ipsc_mn_tardbp_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " TARDBP Down ", query == "query_2" ~ " TARDBP Up "), term_name = as.factor(term_name))
ipsc_mn_tardbp_datasets.dge_gprofiles_down <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Down ")
ipsc_mn_tardbp_datasets.dge_gprofiles_up <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Up ")
paste('"',unique(ipsc_mn_tardbp_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"spliceosomal complex",
"mRNA Splicing",
# "DNA IR-damage and cellular response via ATR",
"regulation of RNA metabolic process",
"nervous system development",
# "DNA metabolic process",
# "microtubule cytoskeleton organization",
"microtubule cytoskeleton",
# "chromosome segregation",
"cell division",
# "mitotic cell cycle",
"protein binding")
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_down <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_tardbp_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
# "neuron differentiation",
"glutamatergic synapse",
# "Neuronal System",
# "neurogenesis",
"synaptic signaling",
# "dendritic tree",
"dendrite",
"axon",
"neuron projection",
"synapse")
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_up <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_tardbp_datasets.dge_gprofiles_filt_down, ipsc_mn_tardbp_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" TARDBP Up ", " TARDBP Down ")))
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub("regulation of RNA metabolic process", "RNA metabolic process", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" and cellular response via ATR", "", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub("cellular response to DNA", "DNA", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
# ipsc_mn_tardbp_datasets.go_curated

# # sod1
# ipsc_mn_sod1_datasets.dge_genes <- ipsc_mn_sod1_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_sod1_datasets.dge_gprofiles_filt <- ipsc_mn_sod1_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in sod1 ", query == "query_2" ~ " Up in sod1 "), term_name = as.factor(term_name))
# ipsc_mn_sod1_datasets.dge_gprofiles_down <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Down in sod1 ")
# ipsc_mn_sod1_datasets.dge_gprofiles_up <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Up in sod1 ")
# paste('"',unique(ipsc_mn_sod1_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# ipsc_mn_sod1_datasets.dge_gprofiles_filt_down <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Down in sod1 ") %>% filter(term_name %in% down_list)
# paste('"',unique(ipsc_mn_sod1_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# #no terms for SOD1
# 
# # vcp
# ipsc_mn_vcp_datasets.dge_genes <- ipsc_mn_vcp_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_vcp_datasets.dge_gprofiles_filt <- ipsc_mn_vcp_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA|WP")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in vcp ", query == "query_2" ~ " Up in vcp "), term_name = as.factor(term_name))
# ipsc_mn_vcp_datasets.dge_gprofiles_down <- ipsc_mn_vcp_datasets.dge_gprofiles_filt %>% filter(query == " Down in vcp ")
# ipsc_mn_vcp_datasets.dge_gprofiles_up <- ipsc_mn_vcp_datasets.dge_gprofiles_filt %>% filter(query == " Up in vcp ")
# paste('"',unique(ipsc_mn_vcp_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# paste('"',unique(ipsc_mn_vcp_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# # no terms for VCP
# 
# # sporadic
# ipsc_mn_sporadic_datasets.dge_genes <- ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_sporadic_datasets.dge_gprofiles_filt <- ipsc_mn_sporadic_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in sporadic ", query == "query_2" ~ " Up in sporadic "), term_name = as.factor(term_name))
# ipsc_mn_sporadic_datasets.dge_gprofiles_down <- ipsc_mn_sporadic_datasets.dge_gprofiles_filt %>% filter(query == " Down in sporadic ")
# ipsc_mn_sporadic_datasets.dge_gprofiles_up <- ipsc_mn_sporadic_datasets.dge_gprofiles_filt %>% filter(query == " Up in sporadic ")
# paste('"',unique(ipsc_mn_sporadic_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# paste('"',unique(ipsc_mn_sporadic_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# # none or sporadic

# ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_DNA_IRDAMAGE_AND_CELLULAR_RESPONSE_VIA_ATR) # OLIG1, OLIG2

# # fus
# ipsc_mn_fus_datasets.rrvgo = gprofiler2rrvgo(ipsc_mn_fus_datasets$res , query1 = "Down", query2 = "Up", GO_cat = "BP", sig.filter = "padj", species = "human")
# ipsc_mn_fus_datasets.rrvgo =  ipsc_mn_fus_datasets.rrvgo %>%  mutate(term = gsub("regulation of ", "", term), term = gsub("homophilic cell adhesion via plasma membrane adhesion molecules", "plasma membrane adhesion", term), parentTerm = term)
# ipsc_mn_fus_datasets.rrvgo.plot = rrvgo_plot(gprofiler2rrvgo = ipsc_mn_fus_datasets.rrvgo, n_terms = 10, remove_terms = NULL, colours = 2, cols =  c("firebrick2", "dodgerblue2"), label_angle = 0)
# 
# # tardbp
# ipsc_mn_tardbp_datasets.rrvgo = gprofiler2rrvgo(ipsc_mn_tardbp_datasets$res , query1 = "Down", query2 = "Up", GO_cat = "BP", sig.filter = "padj", species = "human")
# ipsc_mn_tardbp_datasets.rrvgo =  ipsc_mn_tardbp_datasets.rrvgo %>%  mutate(term = gsub("regulation of ", "", term), parentTerm = term)
# ipsc_mn_tardbp_datasets.rrvgo.plot = rrvgo_plot(gprofiler2rrvgo = ipsc_mn_tardbp_datasets.rrvgo, n_terms = 10, remove_terms = NULL, colours = 2, cols =  c("firebrick2", "dodgerblue2"), label_angle = 0)

# de_go_terms_mutations_merged = ggarrange(NULL,ipsc_mn_c9orf72_datasets.rrvgo.plot,NULL, ipsc_mn_sod1_datasets.rrvgo.plot,NULL, ipsc_mn_fus_datasets.rrvgo.plot,NULL, ipsc_mn_tardbp_datasets.rrvgo.plot, nrow = 8, ncol = 1, labels = c("C9orf72", "", "SOD1", "", "FUS", "", "TARDBP",""), heights = c(0.1,0.8,0.1,1.3,0.1,0.9,0.1,1.3)) #, ipsc_mn_vcp_datasets.rrvgo.plot) # no terms for VCP
# de_go_terms_mutations_merged
# ggsave(de_go_terms_mutations_merged, filename = here(proj_path,"expression/deseq2/figS5.de_go_terms_mutations_merged.png", height = 11.75, width = 8.25)

# # which genes are driving the terms
# ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GABA_RECEPTOR_COMPLEX)
# ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PLASMA_MEMBRANE_REGION)
# ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GAP_JUNCTION)
# ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT_CONFERRING_COMPRESSION_RESISTANCE)
```


## Merge

```{r}
genetic_subgroups_scatter_upset = plot_grid(
  plot_grid(plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", gene_labels = "C9orf72", plot_p_value = TRUE),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", gene_labels = "SOD1", plot_p_value = TRUE),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", gene_labels = "FUS", plot_p_value = TRUE),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", gene_labels = "TARDBP", plot_p_value = TRUE), ncol = 4, labels = c("a","","","")),
  plot_grid(plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", gene_labels = c("C9orf72","SOD1"), plot_p_value = TRUE),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", gene_labels = c("C9orf72","FUS"), plot_p_value = TRUE),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", gene_labels = c("C9orf72","TARDBP"), plot_p_value = TRUE),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", gene_labels = c("SOD1","FUS"), plot_p_value = TRUE), ncol = 4),
  plot_grid(plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", gene_labels = c("SOD1","TARDBP"), plot_p_value = TRUE),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", gene_labels = c("TARDBP","FUS"), plot_p_value = TRUE), 
  ipsc_mn_mutants_bind_datasets.res.upset, ncol = 3, rel_widths = c(1,1,2), labels = c("","","b")), 
  plot_grid(ipsc_mn_c9orf72_datasets.go_curated, ipsc_mn_fus_datasets.go_curated, ipsc_mn_tardbp_datasets.go_curated, ncol=3, labels = c("c","d","e")), nrow = 4)
# genetic_subgroups_compared.merged
ggsave(genetic_subgroups_scatter_upset, filename = here(proj_path,"expression/deseq2/genetic_subgroups_scatter_upset.png"), height = 9.5, width = 8.25)

```


# Fig S11 DoRoTHEA & PROGENy

<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md
Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
ipsc_mn_mutations_list = list("Sporadic" = select(ipsc_mn_sporadic_datasets$res,gene_name,stat), "C9orf72" = select(ipsc_mn_c9orf72_datasets$res,gene_name,stat), "TARDBP" = select(ipsc_mn_tardbp_datasets$res,gene_name,stat), "SOD1" = select(ipsc_mn_sod1_datasets$res,gene_name,stat), "FUS" = select(ipsc_mn_fus_datasets$res,gene_name,stat))#, "VCP" = select(ipsc_mn_vcp_datasets$res,gene_name,stat)) 
mutations <- names(ipsc_mn_mutations_list)
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{drop_na(.x)}) 
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, source = as.factor(source), source = fct_reorder(source, score))

ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=mutation, y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment") +
  theme_oz() + theme(axis.text.x = element_text(angle = 90, size = 6), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.position = "top") +
  geom_hline(yintercept = 0, linetype = 3) +
  facet_grid(~source) +
  stat_pvalue_manual(ipsc_mn_mutations_list.progeny_scores, label = "p.signif", y.position = 12, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE)
ipsc_mn_mutations_list.progeny_scores.plot

```

Dorothea

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation) %>% 
  mutate(sig = case_when(p_value < 0.1 & abs(score) < 4 ~ "sig", p_value < 0.1 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.1 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

# Volcano of TFs score vs p_value
ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_mutations_list.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
    labs(title = "Transcription Factors", y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
    geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "C9orf72"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "FUS"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "SOD1"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "Sporadic"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "TARDBP"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
    geom_text_repel(fontface = "italic", data = filter(ipsc_mn_mutations_list.dorothea_scores,source %in% c("TP53","ZNF274","GATA3","MAZ","TAL1","TEAD4")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  facet_wrap(~mutation, scales = "free") & ylim(0,2.8)
ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano

# ,"E2F4","EOMES","MAZ","MITF","SOX2","TFDP1","ZBTB33","ZNF263"

ipsc_mn_mutations_list.dorothea_scores %>% filter(source == "TP53")
# mutation statistic  source condition  score p_value p.signif group1   group2   sig        direction class        
#   <chr>    <chr>      <chr>  <chr>      <dbl>   <dbl> <chr>    <chr>    <chr>    <chr>      <chr>     <chr>        
# 1 Sporadic norm_wmean TP53   V1        -0.313   0.254 ""       Sporadic Sporadic non_sig    down      non_sig down 
# 2 C9orf72  norm_wmean TP53   V1         9.27    0.002 "*"      C9orf72  C9orf72  sig_strong up        sig_strong up
# 3 TARDBP   norm_wmean TP53   V1         2.45    0.014 "*"      TARDBP   TARDBP   sig        up        sig up       
# 4 SOD1     norm_wmean TP53   V1        -2.97    0.002 "*"      SOD1     SOD1     sig        down      sig down     
# 5 FUS      norm_wmean TP53   V1         4.02    0.004 "*"      FUS      FUS      sig_strong up        sig_strong up

# commonly changed TFs
# increased
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score > 0) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 ZNF274     4
# decreased
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score < 0) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 GATA3      4
#  2 MAZ        4
#  3 TAL1       4
#  4 TEAD4      4

ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "MAZ")
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "TEAD4")

```

Merge

```{r}
ipsc_mn_mutations_list.dorothea_progeny_merged = ggarrange(ipsc_mn_mutations_list.progeny_scores.plot, ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano, nrow = 2, labels = c("a","b"), heights = c(1,1.7))
# ipsc_mn_mutations_list.dorothea_progeny_merged
ggsave(ipsc_mn_mutations_list.dorothea_progeny_merged, filename = here(proj_path, "expression/deseq2/ipsc_mn_mutations_list.dorothea_progeny_merged.png"), height = 10, width = 9)

```



# Table S5 DEGs overlap between genetic backgrounds

```{r}
ipsc_mn_mutants_join_datasets.res = 
  mutate(select(ipsc_mn_sporadic_datasets$res, gene_name, gene_id, padj.sporadic = padj, stat.sporadic = stat, lfc.sporadic = log2FoldChange), direction.sporadic = case_when(padj.sporadic < 0.05 & lfc.sporadic > 0 ~ "up", padj.sporadic < 0.05 & lfc.sporadic < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(ipsc_mn_c9orf72_datasets$res, gene_name, gene_id, padj.c9orf72 = padj, stat.c9orf72 = stat, lfc.c9orf72 = log2FoldChange), direction.c9orf72 = case_when(padj.c9orf72 < 0.05 & lfc.c9orf72 > 0 ~ "up", padj.c9orf72 < 0.05 & lfc.c9orf72 < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_fus_datasets$res, gene_name, gene_id, padj.fus = padj, stat.fus = stat, lfc.fus = log2FoldChange), direction.fus = case_when(padj.fus < 0.05 & lfc.fus > 0 ~ "up", padj.fus < 0.05 & lfc.fus < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_tardbp_datasets$res, gene_name, gene_id,  padj.tardbp = padj, stat.tardbp = stat, lfc.tardbp = log2FoldChange), direction.tardbp = case_when(padj.tardbp < 0.05 & lfc.tardbp > 0 ~ "up", padj.tardbp < 0.05 & lfc.tardbp < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_sod1_datasets$res, gene_name, gene_id, padj.sod1 = padj, stat.sod1 = stat, lfc.sod1 = log2FoldChange), direction.sod1 = case_when(padj.sod1 < 0.05 & lfc.sod1 > 0 ~ "up", padj.sod1 < 0.05 & lfc.sod1 < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  # arrange() %>%
  select(gene_name, gene_id, everything()) %>%
  filter(padj.sporadic < 0.05 | padj.c9orf72 < 0.05 | padj.fus < 0.05 | padj.tardbp < 0.05 | padj.sod1 < 0.05)
ipsc_mn_mutants_join_datasets.res
write_csv(ipsc_mn_mutants_join_datasets.res, here(proj_path,"expression/deseq2/TableS6.overlap_als_individual_datasets.csv"))

```

# Fig 4 Postmortem 

Import NYGC counts from Humphrey et al Zenodo https://zenodo.org/record/6385747#.Yod3Y5PMJqs

## Metadata and barplot

```{r}
# nygenome_geo_metadata = read_excel(here(shared_path, "nygc-als-consortium/sample-details/20210804 GEO GSE137810 Collaborator Metadata.xlsx")) %>% clean_names() %>% rename(sample = geo, library_prep = prep, mutation.nygc = reported_genomic_mutations_from_sites_not_in_any_way_associated_with_wgs_data_from_nygc)
# nygc_SraRunTable.txt = read_csv(here(shared_path, "nygc-als-consortium/sample-details/SraRunTable.txt")) %>% clean_names() %>% mutate(sample_source = gsub(" ", "_", tissue)) %>% rename(sample = geo_accession_exp, external_subject_id = subject_id, external_sample_id = sample_id_alt) %>% select(-sample_name, -assay_type, -center_name, -consent, -datastore_filetype, -datastore_provider, -library_selection, -library_source, -organism, -platform, -tissue, -tissue_type, -sample_type, -library_preparation_method, -source_name, -group, -sample_group)
# nygeome_samplesheet = nygenome_geo_metadata %>% left_join(nygc_SraRunTable.txt, by = c("sample", "external_subject_id", "external_sample_id", "sample_source")) %>% #%>% get_dupes(geo) %>% glimpse # 2278
#   drop_na(run) # removes 41 rows without a SRR ID not in nygc_SraRunTable.txt
# nygeome_samplesheet %>% glimpse
# # join Target ALS metadata for postmortem interval - only has interval for 932/2237 samples
# target_als_metadata = read_excel(here(shared_path, "nygc-als-consortium/sample-details/Target ALS RNAseq Metadata 07192022.xlsx")) %>% clean_names() %>% select(external_sample_id, external_subject_id, sample_source, sex, subject_group, subject_group_subcategory, site_of_motor_onset, site_of_motor_onset_detail, age_at_symptom_onset, age_at_death, rin, ph = p_h, disease_duration_in_months, post_mortem_interval_in_hours) %>% distinct(external_sample_id, external_subject_id, sample_source, .keep_all = TRUE)
# nygeome_samplesheet %>% left_join(target_als_metadata) %>% drop_na(run)
# # nygc_spinal_cord_samplesheet = nygeome_samplesheet %>% mutate(fastq_1 = paste0("/camp/project/proj-luscombn-patani/working/nygc-als-consortium/reads/fastq/",run,"_1.fastq.gz"), fastq_2 = paste0("/camp/project/proj-luscombn-patani/working/nygc-als-consortium/reads/fastq/",run,"_2.fastq.gz"), strandedness = "reverse") %>%
#   filter(grepl("Cord",sample_source), grepl("ALS Spectrum|Control",subject_group)) %>% # keep only spinal cord samples
#   mutate(condition = case_when(grepl("Control", subject_group) ~ "ctrl", grepl("ALS",subject_group) ~ "als"), condition = factor(condition, levels = c("ctrl","als")),
#          sample_source = factor(sample_source, levels = c("Spinal_Cord_Cervical", "Spinal_Cord_Thoracic", "Spinal_Cord_Lumbar")),
#          c9orf72_status = case_when(grepl("Positive for C9orf72 Repeat Expansion", mutation.nygc) ~ "positive", grepl("Negative for C9orf72 Repeat Expansion", mutation.nygc) ~ "negative", TRUE ~ "unknown"),
#          sod1_status = case_when(grepl("Positive for SOD1", mutation.nygc) ~ "positive", grepl("Negative for SOD1", mutation.nygc) ~ "negative", TRUE ~ "unknown"),
#          mutation = case_when(c9orf72_status == "positive" ~ "c9orf72", sod1_status == "positive" ~ "sod1", grepl("Positive for ANG",mutation.nygc)~"ang", grepl("Positive for FUS",mutation.nygc)~"fus", grepl("Positive for OPTN",mutation.nygc)~"optn", grepl("Positive for VCP",mutation.nygc)~"vcp", grepl("Positive for DCTN1",mutation.nygc)~"dctn1", grepl("Positive for NEFH",mutation.nygc)~"nefh", grepl("Positive for SETX (likely benign), FIG4 (VUS)",mutation.nygc)~"fig4", grepl("Positive for SETX",mutation.nygc)~"setx", grepl("Positive for FIG4",mutation.nygc)~"fig4", grepl("Positive for UBQLN2",mutation.nygc)~"ubqln2", condition == "als" ~ "sporadic", TRUE ~ "ctrl"),
#          age_at_death = as.numeric(gsub("90 or Older","90",age_at_death))) %>%
#   select(sample, fastq_1, fastq_2, strandedness, condition, mutation, sample_source, everything())
# nygc_spinal_cord_samplesheet %>% glimpse # 696 samples
# nygc_spinal_cord_samplesheet %>% get_dupes(sample, external_sample_id, external_subject_id) %>% glimpse # duplicates are due to technical repeats. Multiple SRRIDs per sample. Need merging in nfcore/rnaseq.
# 
# # 226 individuals have multiple spinal cord samples. For these we take only their superior sample i.e. Cervical > Thoracic > Lumbar
# nygc_spinal_cord_samplesheet %>% distinct(sample, .keep_all = TRUE) %>% group_by(external_subject_id) %>% count(sample_source) %>% pivot_wider(names_from = "sample_source", values_from = "n") %>% mutate(sum = sum(Spinal_Cord_Cervical, Spinal_Cord_Lumbar, Spinal_Cord_Thoracic, na.rm = TRUE)) %>% filter(sum > 1) # 226
# multiple_samples_subjects = nygc_spinal_cord_samplesheet %>% distinct(sample, .keep_all = TRUE) %>% group_by(external_subject_id) %>% count(sample_source) %>% pivot_wider(names_from = "sample_source", values_from = "n") %>% mutate(sum = sum(Spinal_Cord_Cervical, Spinal_Cord_Lumbar, Spinal_Cord_Thoracic, na.rm = TRUE)) %>% filter(sum > 1) %>% pull(external_subject_id) # 226
# nygc_spinal_cord_samplesheet %>% distinct(sample, .keep_all = TRUE) %>% group_by(external_subject_id) %>% count(sample_source) %>% pivot_wider(names_from = "sample_source", values_from = "n") %>% mutate(sum = sum(Spinal_Cord_Cervical, Spinal_Cord_Lumbar, Spinal_Cord_Thoracic, na.rm = TRUE)) %>% filter(sum > 1, is.na(Spinal_Cord_Cervical)) # 3 individuals with multiple samples but not cervical cord sample.
# nygc_spinal_cord_samplesheet %<>% mutate(multiple_samples_per_subject = case_when(external_subject_id %in% multiple_samples_subjects ~ TRUE, TRUE ~ FALSE)) %>%
#   filter(multiple_samples_per_subject == FALSE |
#         (multiple_samples_per_subject == TRUE & external_subject_id %in% multiple_samples_subjects & sample_source == "Spinal_Cord_Cervical") |  # keep only cervical cord sample
#         (multiple_samples_per_subject == TRUE & external_subject_id %in% c("JHU73","JHU79","JHU80") & sample_source == "Spinal_Cord_Thoracic")) # if multiple samples but no cervical cord then keep thoracic.
# nygc_spinal_cord_samplesheet %>% glimpse # 355 runs
# nygc_spinal_cord_samplesheet %>% distinct(sample) # 271 unique individuals
# nygc_spinal_cord_samplesheet %>% distinct(sample, .keep_all=TRUE) %>% count(condition) # 214 ALS and 57 CTRL
# write_csv(nygc_spinal_cord_samplesheet, here(shared_path, "nygc-als-consortium/sample-details/nygc_spinal_cord_samplesheet.csv"))
# write_csv(select(nygc_spinal_cord_samplesheet, run), here(shared_path, "nygc-als-consortium/sample-details/nygc_spinal_cord.srr_id.txt"), col_names = FALSE)
# nygc_spinal_cord_samplesheet = read_csv(here(shared_path, "nygc-als-consortium/sample-details/nygc_spinal_cord_samplesheet.csv"))
# nygc_spinal_cord_samplesheet %>% count(condition) # 274 ALS, 81 CTRL
# nygc_spinal_cord_samplesheet %>% get_dupes(sample) #166 samples with 2 runs.
# # make distinct sample metadata, removing multiple runs per sample.
# nygc_spinal_cord_metadata = nygc_spinal_cord_samplesheet %>% distinct(sample, external_sample_id, external_subject_id, .keep_all = TRUE) %>% select(-run, -fastq_1, -fastq_2) # 550 / 696
# write_csv(nygc_spinal_cord_metadata, here(shared_path, "nygc-als-consortium/sample-details/nygc_spinal_cord_metadata.csv"))

nygc_spinal_cord_metadata %>% count(condition) # 214 ALS, 57 CTRL
nygc_spinal_cord_metadata %>% count(library_prep, instrument, avg_spot_len) # 2 batches (186 NovaSeq6000, 85 HiSeq2500)
nygc_spinal_cord_metadata %>% count(sample_source)
# sample_source            n
#   <fct>                <int>
# 1 Spinal_Cord_Cervical   245
# 2 Spinal_Cord_Thoracic     9
# 3 Spinal_Cord_Lumbar      17
nygc_spinal_cord_metadata %>% count(mutation)
#  mutation     n
#    <chr>    <int>
#  1 ang          1
#  2 c9orf72     36
#  3 ctrl        57
#  4 dctn1        1
#  5 fig4         1
#  6 fus          2
#  7 nefh         2
#  8 optn         1
#  9 setx         2
# 10 sod1         5
# 11 sporadic   161
# 12 ubqln2       1
# 13 vcp          1

nygc_spinal_cord_metadata 
nygc_spinal_cord_metadata.genetic.n = nygc_spinal_cord_metadata %>% count(mutation) %>% mutate(mutation = toupper(mutation), mutation = gsub("ORF","orf", mutation), mutation = gsub("SPORADIC","Sporadic",mutation), mutation = factor(mutation, levels = rev(c("Sporadic", "C9orf72", "SOD1", "FUS","SETX","NEFH","OPTN","VCP","UBQLN2", "ANG" ,"DCTN1", "FIG4","CTRL"))))
nygc_spinal_cord_metadata.genetic.n

nygc_spinal_cord_metadata.genetic.n.barplot <- ggplot(nygc_spinal_cord_metadata.genetic.n, aes(x=n, y=mutation, fill=mutation)) +
  geom_bar(stat="identity", position = position_dodge()) +
  scale_fill_manual(values = c("Sporadic"="#CF3D39", "C9orf72"="#7FC97F", "SOD1"="#386CB0", "FUS"="#75489F","SETX"="#83624B","VCP"="#94C09B","NEFH"="#BEAED4", "OPTN"="#FDC086" ,"SETX"="#7A9DA8", "UBQLN2"="#A9B7B7","ANG"="#83624B","DCTN1"="#FDC086", "FIG4"="#D3B4BA", "CTRL"="#CC87AF")) +
  theme_classic() +  theme(axis.line = element_blank(), axis.text.x = element_blank(), axis.text.y = element_text(size=10), axis.ticks = element_blank(), panel.grid = element_blank(), legend.position = "none", plot.margin = unit(c(0,0,-0.6,-0.2), "cm")) + 
  expand_limits(x = 190) + #, 
  geom_text(aes(label=n), hjust=-0.1, size=3.8)+
  labs(y = "", x = "")
nygc_spinal_cord_metadata.genetic.n.barplot
ggsave(nygc_spinal_cord_metadata.genetic.n.barplot, filename = here(proj_path,"expression/deseq2/nygc_spinal_cord_metadata.genetic.n.barplot.png"), width = 3.5, height = 4)

ggsave(plot_grid(ipsc_mn_als_datasets.genetic.n.barplot + ggtitle("iPSMN"), nygc_spinal_cord_metadata.genetic.n.barplot + ggtitle("Postmortem"), ncol = 2), filename = here(proj_path,"expression/deseq2/ipsn_postmortem.genetic.n.barplot.png"), width = 5.4, height = 4)

```


## Volcano

```{r}
load(file = here(proj_path, "expression/deseq2/nygc_postmortem_spinal_cord.RData"))

nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05) # 14,064
nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05) %>% group_split(stat > 0) #6575 up, 7489 down
nygc_postmortem_spinal_cord$res %>% filter(gene_name %in% c("TARDBP","FUS","SFPQ"))

## Volcano
nygc_postmortem_spinal_cord.als.labels <- nygc_postmortem_spinal_cord$res %>% filter(gene_name %in% ALS_genes) %>% top_n(n=5,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.p53.labels <- nygc_postmortem_spinal_cord$res %>% filter(gene_name %in% p53.DDR.repair_gene_set) %>% top_n(n=5,wt = abs(stat)) %>% pull(gene_name)
# ipsc_mn_als_datasets.res.dge.up <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
# ipsc_mn_als_datasets.res.dge.down <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
# nygc_postmortem_spinal_cord.ipsn.up.labels <- nygc_postmortem_spinal_cord$res %>% filter(padj<0.05,stat>0,gene_name %in% ipsc_mn_als_datasets.res.dge.up) %>% pull(gene_name)
# nygc_postmortem_spinal_cord.ipsn.down.labels <- nygc_postmortem_spinal_cord$res %>% filter(padj<0.05,stat<0,gene_name %in% ipsc_mn_als_datasets.res.dge.down) %>% pull(gene_name)
nygc_postmortem_spinal_cord.labels <- nygc_postmortem_spinal_cord$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.volcano <- volcano_plot(nygc_postmortem_spinal_cord$res, numerator = "ALS", denomenator = "CTRL",
               title = "Postmortem spinal cord", subtitle = "57 Controls vs 214 ALS", ymax = 45, xmax = 2.9, dpi = 300,
               gene_labels = c(nygc_postmortem_spinal_cord.als.labels, #nygc_postmortem_spinal_cord.ipsn.up.labels, nygc_postmortem_spinal_cord.ipsn.down.labels,
                               nygc_postmortem_spinal_cord.labels, nygc_postmortem_spinal_cord.p53.labels))
nygc_postmortem_spinal_cord.volcano

```

## GO terms & GSEA

```{r}
nygc_postmortem_spinal_cord.gost <- nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
nygc_postmortem_spinal_cord.gprofiles <- nygc_postmortem_spinal_cord.gost$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
nygc_postmortem_spinal_cord.gprofiles_down <- nygc_postmortem_spinal_cord.gprofiles %>% filter(query == " ALS Down ")
nygc_postmortem_spinal_cord.gprofiles_up <- nygc_postmortem_spinal_cord.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(nygc_postmortem_spinal_cord.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "histone methylation",
# "dendrite",
"synapse",
"axon",
# "neuron development",
# "DNA binding",
# "neuron projection",
# "histone modification",
# "generation of neurons",
"neuron differentiation",
"cytoskeleton organization",
"neurogenesis",
"transcription by RNA polymerase II",
# "transcription, DNA-templated",
"regulation of RNA metabolic process",
"protein modification process")
nygc_postmortem_spinal_cord.gprofiles_filt_down <- nygc_postmortem_spinal_cord.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(nygc_postmortem_spinal_cord.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
# "regulation of intrinsic apoptotic signaling pathway in response to DNA damage",
# "G1/S DNA Damage Checkpoints",
# "intrinsic apoptotic signaling pathway in response to DNA damage",
# "cellular response to DNA damage stimulus",
# "regulation of signal transduction by p53 class mediator",
# "TP53 Regulates Metabolic Genes",
# "intrinsic apoptotic signaling pathway in response to DNA damage by p53 class mediator",
# "p53-Dependent G1 DNA Damage Response",
# "p53-Dependent G1/S DNA damage checkpoint",
# "intrinsic apoptotic signaling pathway by p53 class mediator",
"regulation of response to DNA damage stimulus",
"Stabilization of p53",
# "protein localization",
# "apoptotic process",
"programmed cell death",
"endoplasmic reticulum",
"lysosome",
"cellular response to stress",
"response to stress",
"translation",
"protein metabolic process",
"mitochondrion")
nygc_postmortem_spinal_cord.gprofiles_filt_up <- nygc_postmortem_spinal_cord.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
nygc_postmortem_spinal_cord.gprofiles_filt_combined <- bind_rows(nygc_postmortem_spinal_cord.gprofiles_filt_down, nygc_postmortem_spinal_cord.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
nygc_postmortem_spinal_cord.gprofiles_filt_combined = nygc_postmortem_spinal_cord.gprofiles_filt_combined %>% mutate(term_name = gsub("regulation of ", "", term_name), term_name = gsub("and","&",term_name), term_name = gsub("cellular ","",term_name), term_name = gsub("response to |transcription by ","",term_name)) 
nygc_postmortem_spinal_cord.go_curated <- curated_profiles(gprofiles = nygc_postmortem_spinal_cord.gprofiles_filt_combined)# + theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
nygc_postmortem_spinal_cord.go_curated

# plot_grid(nygc_postmortem_spinal_cord.volcano, nygc_postmortem_spinal_cord.go_curated, ncol = 2)


## GSEA: DNA damage response, p53 signalling pathway
nygc_postmortem_spinal_cord.geneList <- nygc_postmortem_spinal_cord$res %>% drop_na(stat) %>% pull(stat)
names(nygc_postmortem_spinal_cord.geneList) <- nygc_postmortem_spinal_cord$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
nygc_postmortem_spinal_cord.geneList = sort(nygc_postmortem_spinal_cord.geneList, decreasing = TRUE)

# p53 signalling pathway
p53.signalling_genes.list <- list("p53.signalling" = nygc_postmortem_spinal_cord$res$gene_name[nygc_postmortem_spinal_cord$res$gene_name %in% p53.signalling.pathway])
nygc_postmortem_spinal_cord.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = nygc_postmortem_spinal_cord.geneList)
nygc_postmortem_spinal_cord.p53.signalling_fgseaRes
# pathway         pval         padj   log2err        ES      NES size                            leadingEdge
# 1: p53.signalling 8.564086e-06 8.564086e-06 0.5933255 0.4433985 1.578591  308 RPS27L,RBBP8,ZMAT3,FAS,CCND2,CCNG2,...
nygc_postmortem_spinal_cord.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], nygc_postmortem_spinal_cord.geneList) + 
  theme(axis.text=element_text(size=6), axis.title=element_text(size=10)) + ggtitle("p53 signalling") + xlab("") +
  annotate("text", x = 25000, y = 0.3, label = paste0("NES = ", round(nygc_postmortem_spinal_cord.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(nygc_postmortem_spinal_cord.p53.signalling_fgseaRes$pval, digits = 6)), size = 2.8, hjust = 0) + theme_oz()
nygc_postmortem_spinal_cord.p53.signalling.gseaplot

```


## PROGENY & Dorothea

```{r}
# nygc_postmortem_spinal_cord = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.tissue.rds"))

# PROGENy & DOROTHEA
net_progeny <- get_progeny(organism = 'human', top = 100)
# net_progeny = readRDS(here(camp_path, "projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta/ipsn_als_meta/net_progeny.rds"))
nygc_postmortem_spinal_cord.res.matrix <- nygc_postmortem_spinal_cord$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
nygc_postmortem_spinal_cord_progeny.contrast_acts <- run_wmean(mat=nygc_postmortem_spinal_cord.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
nygc_postmortem_spinal_cord_progeny.pathwayActivity <- ggplot(nygc_postmortem_spinal_cord_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment ALS vs CTRL") +
    stat_pvalue_manual(mutate(nygc_postmortem_spinal_cord_progeny.contrast_acts, y.position = case_when(source %in% c("TNFa","Hypoxia")~5,TRUE~5.5)), label = "p.signif", xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
nygc_postmortem_spinal_cord_progeny.pathwayActivity

# p53 weights
nygc_postmortem_spinal_cord_progeny.p53weights = net_progeny %>% filter(source == "p53", target %in% nygc_postmortem_spinal_cord$res$gene_name) %>% left_join(select(nygc_postmortem_spinal_cord$res, target=gene_name,stat)) %>% 
  mutate(color = case_when(weight > 0 & stat > 0 ~ '1', weight > 0 & stat < 0 ~ '2', weight < 0 & stat > 0 ~ '2', weight < 0 & stat < 0 ~ '1', TRUE ~ "3"))
nygc_postmortem_spinal_cord_progeny.p53weights.plot = ggplot(nygc_postmortem_spinal_cord_progeny.p53weights, aes(x = weight, y = stat, color = color)) + geom_point(size = 0.5) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(fontface = "italic", data = filter(nygc_postmortem_spinal_cord_progeny.p53weights, (weight > 10 & stat > 2) | (weight > 0 & stat > 6)), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.1) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "p53 pathway weights", x = "p53 pathway gene weight", y = "ALS vs CTRL test statistic") +
  geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted') +
  coord_cartesian(xlim = c(-8,18), ylim = c(-7,10)) 
# nygc_postmortem_spinal_cord_progeny.p53weights.plot

# plot_grid(nygc_postmortem_spinal_cord_progeny.pathwayActivity, nygc_postmortem_spinal_cord_progeny.p53weights.plot, ncol = 2)

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord_dorothea.contrast_acts <- run_wmean(mat=nygc_postmortem_spinal_cord.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 10000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(sig = case_when(p_value < 0.06 & abs(score) < 4 ~ "sig", p_value < 0.06 & abs(score) >= 4 ~ "sig_strong", p_value > 0.05 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

nygc_postmortem_spinal_cord_dorothea.contrast_acts.labels.up = nygc_postmortem_spinal_cord_dorothea.contrast_acts %>% top_n(score, n = 3) %>% pull(source)
nygc_postmortem_spinal_cord_dorothea.contrast_acts.labels.down = nygc_postmortem_spinal_cord_dorothea.contrast_acts %>% top_n(-score, n = 3) %>% pull(source)

# Volcano of TFs score vs p_value
nygc_postmortem_spinal_cord_dorothea.contrast_acts.volcano = ggplot(nygc_postmortem_spinal_cord_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(nygc_postmortem_spinal_cord_dorothea.contrast_acts, source %in% c(nygc_postmortem_spinal_cord_dorothea.contrast_acts.labels.up, nygc_postmortem_spinal_cord_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
    geom_text_repel(fontface = "italic", data = filter(nygc_postmortem_spinal_cord_dorothea.contrast_acts, source %in% c("TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 3) +  
  geom_vline(xintercept = 0, linetype = 'dotted')
# nygc_postmortem_spinal_cord_dorothea.contrast_acts.volcano

nygc_postmortem_spinal_cord_dorothea.contrast_acts %>% filter(source == "TP53")

```

## Scatter postmortem against iPSMN

```{r}
# correlate to meta
postmortem_ipsn.als_vs_ctrl = select(nygc_postmortem_spinal_cord$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_als_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.als_vs_ctrl.overlapping = postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% pull(gene_name) # 17
postmortem_ipsn_de_list = list("Postmortem spinal cord: ALS vs CTRL" = nygc_postmortem_spinal_cord$res, "iPSMN: ALS vs CTRL" = ipsc_mn_als_datasets$res)
postmortem_ipsn.als_vs_ctrl.scatter = plot_de_correlation(model_list1 = postmortem_ipsn_de_list, model_list2 = postmortem_ipsn_de_list, mutation1 = "iPSMN: ALS vs CTRL", mutation2 = "Postmortem spinal cord: ALS vs CTRL", gene_labels = c(postmortem_ipsn.als_vs_ctrl.overlapping, "CFAP410"), plot_p_value = TRUE)
# postmortem_ipsn.als_vs_ctrl.scatter

postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat))
postmortem_ipsn.als_vs_ctrl.overlapping # 17/43 # 39.5%
postmortem_ipsn.als_vs_ctrl.overlapping[postmortem_ipsn.als_vs_ctrl.overlapping %in% ALS_genes] # 0
postmortem_ipsn.als_vs_ctrl.overlapping[postmortem_ipsn.als_vs_ctrl.overlapping %in% RNA_BINDING_PROTEINS] # "RNASEL" 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% RNA_BINDING_PROTEINS) 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% gprofiler_database$`DNA Damage Response`) 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% RNA_BINDING_PROTEINS  ) %>% arrange(-nygc.stat + ipsn.stat)
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% group_split(nygc.stat > 0)


# Fisher exact overlap test
nygc_postmortem_spinal_cord.res.dge.up <- nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
nygc_postmortem_spinal_cord.res.dge.down <- nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
ipsc_mn_als_datasets.res.dge.up <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.res.dge.down <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
newGeneOverlap(c(ipsc_mn_als_datasets.res.dge.up,ipsc_mn_als_datasets.res.dge.down), c(nygc_postmortem_spinal_cord.res.dge.up,nygc_postmortem_spinal_cord.res.dge.down), genome.size=nrow(nygc_postmortem_spinal_cord$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=43
# listB size=14058
# Intersection size=24
# Overlapping p-value=3.9e-06
# Jaccard Index=0.0

newGeneOverlap(ipsc_mn_als_datasets.res.dge.up, nygc_postmortem_spinal_cord.res.dge.up, genome.size=nrow(nygc_postmortem_spinal_cord$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=20
# listB size=6574
# Intersection size=7
# Overlapping p-value=3.8e-03
# Jaccard Index=0.0
newGeneOverlap(ipsc_mn_als_datasets.res.dge.down, nygc_postmortem_spinal_cord.res.dge.down, genome.size=nrow(nygc_postmortem_spinal_cord$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=23
# listB size=7487
# Intersection size=10
# Overlapping p-value=2e-04
# Jaccard Index=0.0

# hypergeometric: .  Sorry to be pedantic, but an overlap of 15 feels small given the prevalence of differential genes post-mortem (14529) against iPSN (37), and the discrepancy between these two vastly different rates might draw attention to this 
phyper(17-1, 14064, nrow(nygc_postmortem_spinal_cord$res)-14064, 43, lower.tail=FALSE)
# 0.01211124

```


## Compare mutations 

### Correlation heatmap

```{r}
# heatmap of correlation coefficients
postmortem_mutations_stat = select(nygc_postmortem_spinal_cord.sporadic$res, gene_id, Sporadic = stat) %>% 
  left_join(select(nygc_postmortem_spinal_cord.c9orf72$res, gene_id, C9orf72 = stat)) %>%
  left_join(select(nygc_postmortem_spinal_cord.sod1$res, gene_id, SOD1 = stat)) %>%
  left_join(select(nygc_postmortem_spinal_cord.fus$res, gene_id, FUS = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_id")
cormat <- round(cor(postmortem_mutations_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
postmortem.dge_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5))
postmortem.dge_correlation_ggheatmap

```


### p53 and TP53

```{r}
nygc_postmortem_spinal_cord_mn_mutations_list = list("Sporadic" = select(nygc_postmortem_spinal_cord.sporadic$res,gene_name,stat), "C9orf72" = select(nygc_postmortem_spinal_cord.c9orf72$res,gene_name,stat), "SOD1" = select(nygc_postmortem_spinal_cord.sod1$res,gene_name,stat), "FUS" = select(nygc_postmortem_spinal_cord.fus$res,gene_name,stat))
mutations <- names(nygc_postmortem_spinal_cord_mn_mutations_list)
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{drop_na(.x)}) 
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores = map_df(nygc_postmortem_spinal_cord_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')
nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores = nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(decoupleR = "p53 pathway") 

# Dorothea
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord_mn_mutations_list.dorothea_scores = map_df(nygc_postmortem_spinal_cord_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')
nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores = nygc_postmortem_spinal_cord_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(decoupleR = "TP53 TF") 

nygc_postmortem_spinal_cord_mn_mutations_list.p53.TP53 = bind_rows(nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores, nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores) %>% 
  mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)

nygc_postmortem_spinal_cord_mn_mutations_list.p53.TP53.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.p53.TP53, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
  theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust = 1), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
  stat_pvalue_manual(nygc_postmortem_spinal_cord_mn_mutations_list.p53.TP53, label = "p.signif", y.position = 6, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE) +
  facet_wrap(~decoupleR, scales = "free")
# nygc_postmortem_spinal_cord_mn_mutations_list.p53.TP53.plot

```


## Merge

```{r}
nygc_postmortem_spinal_cord.figs.merged = plot_grid(
  plot_grid(nygc_postmortem_spinal_cord.volcano, nygc_postmortem_spinal_cord.go_curated, nygc_postmortem_spinal_cord.p53.signalling.gseaplot, ncol=3, rel_widths = c(1,0.8,0.6), labels = "auto"), 
  plot_grid(nygc_postmortem_spinal_cord_progeny.pathwayActivity, nygc_postmortem_spinal_cord_progeny.p53weights.plot, nygc_postmortem_spinal_cord_dorothea.contrast_acts.volcano, ncol=3, rel_widths = c(1,0.8,0.9), labels = letters[4:6]), 
  plot_grid(postmortem_ipsn.als_vs_ctrl.scatter, postmortem.dge_correlation_ggheatmap, nygc_postmortem_spinal_cord_mn_mutations_list.p53.TP53.plot, ncol = 3, rel_widths = c(1,1,1), labels = letters[7:9]),
  nrow = 3, rel_heights = c(1,1,1))
# nygc_postmortem_spinal_cord.figs.merged
ggsave(nygc_postmortem_spinal_cord.figs.merged, filename = here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.figs.merged.png"), height = 9, width = 10)
ggsave(nygc_postmortem_spinal_cord.figs.merged, filename = here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.figs.merged.pdf"), height = 9, width = 10)



nygc_postmortem_spinal_cord.tweet = plot_grid(nygc_postmortem_spinal_cord.volcano, nygc_postmortem_spinal_cord.go_curated, nygc_postmortem_spinal_cord_progeny.pathwayActivity, ncol=3, rel_widths = c(0.9,0.7,0.8))
ggsave(nygc_postmortem_spinal_cord.tweet, filename = here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.tweet.png"), height = 4, width = 11)

```


# Table S6 postmortem & ipsn overlap DGE

```{r}
postmortem_ipsc_overlap.res = 
  mutate(select(nygc_postmortem_spinal_cord$res, gene_name, gene_id, padj.postmortem = padj, stat.postmortem = stat, lfc.postmortem = log2FoldChange), direction.postmortem = case_when(padj.postmortem < 0.05 & lfc.postmortem > 0 ~ "up", padj.postmortem < 0.05 & lfc.postmortem < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(ipsc_mn_als_datasets$res, gene_name, gene_id, padj.ipsn = padj, stat.ipsn = stat, lfc.ipsn = log2FoldChange), direction.ipsn = case_when(padj.ipsn < 0.05 & lfc.ipsn > 0 ~ "up", padj.ipsn < 0.05 & lfc.ipsn < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  select(gene_name, gene_id, everything()) %>% 
  filter(padj.ipsn < 0.05, padj.postmortem < 0.05, ((stat.ipsn > 0 & stat.postmortem > 0) | (stat.ipsn < 0 & stat.postmortem < 0)))
postmortem_ipsc_overlap.res
write_csv(postmortem_ipsc_overlap.res, here(proj_path,"expression/deseq2/Table.postmortem_ipsc_overlap.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473")
write_sheet(postmortem_ipsc_overlap.res, ss = "https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473", sheet = "S6 Postmortem iPSN overlap")

postmortem_ipsc_overlap.res %>% filter(gene_name %in% p53.signalling.pathway)


write_csv(nygc_postmortem_spinal_cord$res, here(proj_path,"expression/deseq2/nygc.postmortem_spinal_cord.als_vs_ctrl.csv"))

```



# Fig S14 Compare mutations postmortem

Compare sporadic, C9orf72, SOD1, FUS

### Volcano

```{r}
load(file = here(proj_path, "expression/deseq2/nygc_postmortem_spinal_cord.RData"))

postmortem_mutants_bind_datasets.res = bind_rows(mutate(nygc_postmortem_spinal_cord.sporadic$res, mutation = "Sporadic"), mutate(nygc_postmortem_spinal_cord.c9orf72$res, mutation = "C9orf72"), mutate(nygc_postmortem_spinal_cord.sod1$res, mutation = "SOD1"), mutate(nygc_postmortem_spinal_cord.fus$res, mutation = "FUS"))

postmortem_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% count(mutation) %>% arrange(-n)
# 1 Sporadic 13316
# 2 C9orf72  11277
# 3 SOD1      8418
# 4 FUS       2439
nygc_postmortem_spinal_cord.sporadic$res %>% filter(padj < 0.05)

postmortem_mutants_join_datasets.res = full_join(select(nygc_postmortem_spinal_cord.c9orf72$res, gene_name, padj.c9orf72 = padj, log2FoldChange.c9orf72 = log2FoldChange),
          select(nygc_postmortem_spinal_cord.sporadic$res, gene_name, padj.sals = padj, log2FoldChange.sals = log2FoldChange)) %>%
  full_join(select(nygc_postmortem_spinal_cord.fus$res, gene_name, padj.fus = padj, log2FoldChange.fus = log2FoldChange)) %>%
  full_join(select(nygc_postmortem_spinal_cord.sod1$res, gene_name, padj.sod1 = padj, log2FoldChange.sod1 = log2FoldChange)) #%>% full_join(select(postmortem_vcp_datasets$res, gene_name, padj.vcp = padj, log2FoldChange.vcp = log2FoldChange))
postmortem_mutants_join_datasets.res.labels = postmortem_mutants_join_datasets.res %>% filter(c(padj.c9orf72 < 0.05 & padj.fus < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.c9orf72 < 0.05)) %>% pull(gene_name)

nygc_spinal_cord_metadata %>% count(mutation)
#   mutation     n
#    <chr>    <int>
#  1 ang          1
#  2 c9orf72     36
#  3 ctrl        57
#  4 dctn1        1
#  5 fig4         1
#  6 fus          2
#  7 nefh         2
#  8 optn         1
#  9 setx         2
# 10 sod1         5
# 11 sporadic   161
# 12 ubqln2       1
# 13 vcp          1
nygc_postmortem_spinal_cord.fus.labels <- nygc_postmortem_spinal_cord.fus$res %>% filter(-log10(padj)>3, gene_name %in% c(p53.DDR.repair_gene_set, ALS_genes)) %>% distinct(gene_name) %>% pull(gene_name)
nygc_postmortem_spinal_cord.fus.volcano <- volcano_plot(nygc_postmortem_spinal_cord.fus$res, "FUS", numerator = "FUS", denomenator = "CTRL", subtitle = "57 Controls vs 2 FUS", ymax = 15, xmax = 4, dpi = 300, gene_labels = c(nygc_postmortem_spinal_cord.fus.labels))
nygc_postmortem_spinal_cord.fus.volcano

nygc_postmortem_spinal_cord.c9orf72.labels <- nygc_postmortem_spinal_cord.c9orf72$res %>% filter(-log10(padj)>10, gene_name %in% c(p53.DDR.repair_gene_set, ALS_genes)) %>% distinct(gene_name) %>% pull(gene_name)
nygc_postmortem_spinal_cord.c9orf72.volcano <- volcano_plot(nygc_postmortem_spinal_cord.c9orf72$res, "C9orf72", numerator = "C9orf72", denomenator = "CTRL", subtitle = "57 Controls vs 36 C9orf72", ymax = 30, xmax = 3, dpi = 300, gene_labels = c(nygc_postmortem_spinal_cord.c9orf72.labels))
nygc_postmortem_spinal_cord.c9orf72.volcano

nygc_postmortem_spinal_cord.sod1.labels <- nygc_postmortem_spinal_cord.sod1$res %>% filter(-log10(padj)>5, gene_name %in% c(p53.DDR.repair_gene_set, ALS_genes)) %>% distinct(gene_name) %>% pull(gene_name)
nygc_postmortem_spinal_cord.sod1.volcano <- volcano_plot(nygc_postmortem_spinal_cord.sod1$res, "SOD1", numerator = "SOD1", denomenator = "CTRL", subtitle = "57 Controls vs 5 SOD1", ymax = 17, xmax = 3, dpi = 300, gene_labels = c(nygc_postmortem_spinal_cord.sod1.labels))
nygc_postmortem_spinal_cord.sod1.volcano

nygc_postmortem_spinal_cord.sporadic.labels <- nygc_postmortem_spinal_cord.sporadic$res %>% filter(-log10(padj)>15, gene_name %in% c(p53.DDR.repair_gene_set, ALS_genes)) %>% distinct(gene_name) %>% pull(gene_name)
nygc_postmortem_spinal_cord.sporadic.volcano <- volcano_plot(nygc_postmortem_spinal_cord.sporadic$res, "Sporadic", numerator = "Sporadic", denomenator = "CTRL", subtitle = "57 Controls vs 161 Sporadic", ymax = 35, xmax = 3, dpi = 300, gene_labels = c(nygc_postmortem_spinal_cord.sporadic.labels))
nygc_postmortem_spinal_cord.sporadic.volcano

volcano_multiplot <- nygc_postmortem_spinal_cord.fus.volcano + nygc_postmortem_spinal_cord.c9orf72.volcano + nygc_postmortem_spinal_cord.sod1.volcano + nygc_postmortem_spinal_cord.sporadic.volcano +
  plot_layout(ncol = 4, nrow = 1)
# volcano_multiplot
# ggsave(volcano_multiplot, filename = here(proj_path,"expression/deseq2/genetic_subgroups_volcano_multiplot.png"), height = 4, width = 11)

```


## GO terms

```{r}
# c9orf72
nygc_postmortem_spinal_cord.c9orf72.dge_genes <- nygc_postmortem_spinal_cord.c9orf72$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt <- nygc_postmortem_spinal_cord.c9orf72.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|MIRNA|HP")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " C9orf72 Down ", query == "query_2" ~ " C9orf72 Up "), term_name = as.factor(term_name))
nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_down <- nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ")
nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_up <- nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")
paste('"',unique(nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"regulation of RNA metabolic process",
"ribonucleotide binding",
"cellular localization",
"glutamatergic synapse",
"cytoskeleton organization",
"protein modification process",
"synaptic signaling",
"axon",
"nervous system development",
"synapse",
"protein binding")
nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt_down <- nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ") %>% filter(term_name %in% down_list)
paste('"',unique(nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"intrinsic apoptotic signaling pathway by p53 class mediator",
"cellular response to DNA damage stimulus",
"Nonsense-Mediated Decay (NMD)",
"apoptotic process",
"programmed cell death",
"cell death",
"mitochondrion",
"cytoplasmic translation",
"response to stress",
"extracellular exosome",
"protein metabolic process",
"protein binding")
nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt_up <- nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")  %>% filter(term_name %in% up_list)
nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt_combined <- bind_rows(nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt_down, nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" C9orf72 Up ", " C9orf72 Down ")))
nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt_combined$term_name <- gsub("intrinsic apoptotic signaling pathway by p53 class mediator", "apoptotic signaling by p53", nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt_combined$term_name) %>% gsub("regulation of ", "", .) %>% gsub("response to ","",.) %>% gsub(" \\(NMD\\)","",.)
nygc_postmortem_spinal_cord.c9orf72.go_curated <- curated_profiles(gprofiles = nygc_postmortem_spinal_cord.c9orf72.dge_gprofiles_filt_combined) + theme(legend.position = "none")
nygc_postmortem_spinal_cord.c9orf72.go_curated

# FUS
nygc_postmortem_spinal_cord.fus.dge_genes <- nygc_postmortem_spinal_cord.fus$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt <- nygc_postmortem_spinal_cord.fus.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HP|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " FUS Down ", query == "query_2" ~ " FUS Up "), term_name = as.factor(term_name))
nygc_postmortem_spinal_cord.fus.dge_gprofiles_down <- nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt %>% filter(query == " FUS Down ")
nygc_postmortem_spinal_cord.fus.dge_gprofiles_up <- nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt %>% filter(query == " FUS Up ")
paste('"',unique(nygc_postmortem_spinal_cord.fus.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"oligodendrocyte differentiation",
"cellular localization",
"protein modification process",
"cytoskeleton organization",
"localization",
"axon",
"cytoskeleton",
"protein binding",
"neuron development",
"synapse",
"nervous system development")
nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt_down <- nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt %>% filter(query == " FUS Down ") %>% filter(term_name %in% down_list)
paste('"',unique(nygc_postmortem_spinal_cord.fus.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
# "mitochondrion",
"response to heat",
# "regulation of programmed cell death",
# "response to stress",
"programmed cell death",
"apoptotic process",
"cell death",
"Protein processing in endoplasmic reticulum",
"cellular response to stress",
"response to unfolded protein",
"cytoplasmic translation",
"protein metabolic process",
"protein folding",
# "protein binding",
"extracellular exosome")
nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt_up <- nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt %>% filter(query == " FUS Up ")  %>% filter(term_name %in% up_list)
nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt_combined <- bind_rows(nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt_down, nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" FUS Up ", " FUS Down ")))
nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt_combined$term_name <- gsub("Protein processing in e", "E", nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt_combined$term_name) %>% gsub("positive regulation of ","",.)  %>% gsub("regulation of ","",.)
nygc_postmortem_spinal_cord.fus.go_curated <- curated_profiles(gprofiles = nygc_postmortem_spinal_cord.fus.dge_gprofiles_filt_combined) + theme(legend.position = "none")
nygc_postmortem_spinal_cord.fus.go_curated

# sod1
nygc_postmortem_spinal_cord.sod1.dge_genes <- nygc_postmortem_spinal_cord.sod1$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt <- nygc_postmortem_spinal_cord.sod1.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HP|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " SOD1 Down ", query == "query_2" ~ " SOD1 Up "), term_name = as.factor(term_name))
nygc_postmortem_spinal_cord.sod1.dge_gprofiles_down <- nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt %>% filter(query == " SOD1 Down ")
nygc_postmortem_spinal_cord.sod1.dge_gprofiles_up <- nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt %>% filter(query == " SOD1 Up ")
paste('"',unique(nygc_postmortem_spinal_cord.sod1.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"oligodendrocyte differentiation",
# "cellular localization",
"protein modification process",
"cytoskeleton organization",
# "localization",
"axon",
"cytoskeleton",
"protein binding",
"neuron development",
"synapse",
"nervous system development")
nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt_down <- nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt %>% filter(query == " SOD1 Down ") %>% filter(term_name %in% down_list)
paste('"',unique(nygc_postmortem_spinal_cord.sod1.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"cellular response to DNA damage stimulus",
# "p53-Independent DNA Damage Response",
"regulation of immune response",
"cytokine production",
"programmed cell death",
"cell death",
"translation",
"endoplasmic reticulum",
"response to stress",
"protein localization",
# "mitochondrial membrane",
"cellular localization",
"lysosome",
"mitochondrion",
# "extracellular exosome",
"protein metabolic process")
nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt_up <- nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt %>% filter(query == " SOD1 Up ")  %>% filter(term_name %in% up_list)
nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt_combined <- bind_rows(nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt_down, nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" SOD1 Up ", " SOD1 Down ")))
nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt_combined$term_name <- gsub("Protein processing in e", "E", nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt_combined$term_name) %>% gsub("positive regulation of ","",.)  %>% gsub("regulation of ","",.) %>% gsub("cellular response to ","",.)
nygc_postmortem_spinal_cord.sod1.go_curated <- curated_profiles(gprofiles = nygc_postmortem_spinal_cord.sod1.dge_gprofiles_filt_combined) + theme(legend.position = "none")
nygc_postmortem_spinal_cord.sod1.go_curated


# sporadic
nygc_postmortem_spinal_cord.sporadic.dge_genes <- nygc_postmortem_spinal_cord.sporadic$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt <- nygc_postmortem_spinal_cord.sporadic.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " Sporadic Down ", query == "query_2" ~ " Sporadic Up "), term_name = as.factor(term_name))
nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_down <- nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt %>% filter(query == " Sporadic Down ")
nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_up <- nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt %>% filter(query == " Sporadic Up ")
paste('"',unique(nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "axon",
# "DNA binding",
# "chromatin organization",
# "nucleotide binding",
# "neuron differentiation",
"histone modification",
"cytoskeleton organization",
"neurogenesis",
"nervous system development",
"organelle organization",
"positive regulation of cellular process",
"protein modification process",
"DNA-templated transcription",
"protein binding")
nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt_down <- nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt %>% filter(query == " Sporadic Down ") %>% filter(term_name %in% down_list)
paste('"',unique(nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"cellular response to DNA damage stimulus",
"Stabilization of p53",
# "p53-Dependent G1 DNA Damage Response",
"G1/S DNA Damage Checkpoints",
"Metabolism of RNA",
"regulation of immune system process",
"cell death",
"protein localization",
"endoplasmic reticulum",
"lysosome",
"response to stress",
"cytoplasmic translation",
"protein metabolic process",
"mitochondrion",
"protein binding")
nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt_up <- nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt %>% filter(query == " Sporadic Up ")  %>% filter(term_name %in% up_list)
nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt_combined <- bind_rows(nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt_down, nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" Sporadic Up ", " Sporadic Down ")))
nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt_combined$term_name <- gsub("Protein processing in e", "E", nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt_combined$term_name) %>% gsub("positive regulation of ","",.)  %>% gsub("regulation of ","",.) %>% gsub("cellular response to ","",.)
nygc_postmortem_spinal_cord.sporadic.go_curated <- curated_profiles(gprofiles = nygc_postmortem_spinal_cord.sporadic.dge_gprofiles_filt_combined) + theme(legend.position = "none")
nygc_postmortem_spinal_cord.sporadic.go_curated

```



## Upset plot

```{r}
### Upset plot https://github.com/const-ae/ggupset
postmortem_mutants_bind_datasets.res = bind_rows(mutate(nygc_postmortem_spinal_cord.sporadic$res, mutation = "Sporadic"), mutate(nygc_postmortem_spinal_cord.c9orf72$res, mutation = "C9orf72"),mutate(nygc_postmortem_spinal_cord.sod1$res, mutation = "SOD1"), mutate(nygc_postmortem_spinal_cord.fus$res, mutation = "FUS"))

postmortem_mutants_bind_datasets.res.up.upset = upset_plot(filter(postmortem_mutants_bind_datasets.res, stat > 0), overlap_by = "mutation", overlap_level = "gene_id", split_direction = FALSE, ymax = 2800, order = "freq", order_group_list = c("Sporadic", "C9orf72", "SOD1", "FUS"), title = "Upregulated", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))
postmortem_mutants_bind_datasets.res.up.upset

postmortem_mutants_bind_datasets.res.down.upset = upset_plot(filter(postmortem_mutants_bind_datasets.res, stat < 0), overlap_by = "mutation", overlap_level = "gene_id", split_direction = FALSE, ymax = 2500, order = "freq", order_group_list = c("Sporadic", "C9orf72", "SOD1", "FUS"), title = "Downregulated", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))  
postmortem_mutants_bind_datasets.res.down.upset

```


### DoRoTHEA & PROGENy

<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md
Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
postmortem_mutations_list = list("Sporadic" = select(nygc_postmortem_spinal_cord.sporadic$res,gene_name,stat), "C9orf72" = select(nygc_postmortem_spinal_cord.c9orf72$res,gene_name,stat), "SOD1" = select(nygc_postmortem_spinal_cord.sod1$res,gene_name,stat), "FUS" = select(nygc_postmortem_spinal_cord.fus$res,gene_name,stat))
mutations <- names(postmortem_mutations_list)
postmortem_mutations_list = map(postmortem_mutations_list, ~{drop_na(.x)}) 
postmortem_mutations_list = map(postmortem_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
postmortem_mutations_list = map(postmortem_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

net_progeny <- get_progeny(organism = 'human', top = 100)
postmortem_mutations_list.progeny_scores = map_df(postmortem_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, source = as.factor(source), source = fct_reorder(source, score), mutation = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1")))

postmortem_mutations_list.progeny_scores.plot <- ggplot(postmortem_mutations_list.progeny_scores, aes(x=mutation, y = score, fill = mutation)) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  labs(x = "", y = "Signalling Pathways\nALS vs CTRL\nNormalised enrichment") +
  theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 90, size = 6), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  facet_grid(~source) +
  stat_pvalue_manual(mutate(postmortem_mutations_list.progeny_scores, y.position = case_when(mutation=="Sporadic"~6.5,mutation=="C9orf72"~7,TRUE~7)), label = "p.signif", xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE)
postmortem_mutations_list.progeny_scores.plot

```

Dorothea

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
postmortem_mutations_list.dorothea_scores = map_df(postmortem_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation) %>% 
  mutate(sig = case_when(p_value < 0.1 & abs(score) < 4 ~ "sig", p_value < 0.1 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.1 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction), mutation = factor(mutation, levels = c("Sporadic","C9orf72","SOD1","FUS")))

# Volcano of TFs score vs p_value
postmortem_mutations_merged_dorothea.contrast_acts.volcano = ggplot(postmortem_mutations_list.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
    labs(y = expression(-log[10]~P~value), x = "Transcription Factors ALS vs CTRL Normalised enrichment") +
    guides(colour = "none") +
    theme_oz() +  #theme(plot.title = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
    geom_text_repel(fontface = "italic", data = top_n(filter(postmortem_mutations_list.dorothea_scores, p_value < 0.05, mutation == "C9orf72"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(postmortem_mutations_list.dorothea_scores, p_value < 0.05, mutation == "FUS"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(postmortem_mutations_list.dorothea_scores, p_value < 0.05, mutation == "SOD1"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(postmortem_mutations_list.dorothea_scores, p_value < 0.05, mutation == "Sporadic"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
    geom_text_repel(fontface = "italic", data = filter(postmortem_mutations_list.dorothea_scores,source %in% c("TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.8) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  facet_wrap(~mutation, scales = "free", ncol = 4) & ylim(0,2.8)
postmortem_mutations_merged_dorothea.contrast_acts.volcano

# ,"E2F4","EOMES","MAZ","MITF","SOX2","TFDP1","ZBTB33","ZNF263"

postmortem_mutations_list.dorothea_scores %>% filter(source == "TP53")
# mutation statistic  source condition  score p_value p.signif group1   group2   sig        direction class        
#   <chr>    <chr>      <chr>  <chr>      <dbl>   <dbl> <chr>    <chr>    <chr>    <chr>      <chr>     <chr>        
# 1 Sporadic norm_wmean TP53   V1        -0.313   0.254 ""       Sporadic Sporadic non_sig    down      non_sig down 
# 2 C9orf72  norm_wmean TP53   V1         9.27    0.002 "*"      C9orf72  C9orf72  sig_strong up        sig_strong up
# 3 TARDBP   norm_wmean TP53   V1         2.45    0.014 "*"      TARDBP   TARDBP   sig        up        sig up       
# 4 SOD1     norm_wmean TP53   V1        -2.97    0.002 "*"      SOD1     SOD1     sig        down      sig down     
# 5 FUS      norm_wmean TP53   V1         4.02    0.004 "*"      FUS      FUS      sig_strong up        sig_strong up

# commonly changed TFs
# increased
postmortem_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score > 0) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 ZNF274     4
# decreased
postmortem_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score < 0) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 GATA3      4
#  2 MAZ        4
#  3 TAL1       4
#  4 TEAD4      4

postmortem_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "MAZ")
postmortem_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "TEAD4")

# Merge
postmortem_mutations_list.dorothea_progeny_merged = ggarrange(postmortem_mutations_list.progeny_scores.plot, postmortem_mutations_merged_dorothea.contrast_acts.volcano, nrow = 2, labels = c("a","b"), heights = c(1,1.7))
# postmortem_mutations_list.dorothea_progeny_merged
ggsave(postmortem_mutations_list.dorothea_progeny_merged, filename = here(proj_path, "expression/deseq2/postmortem_mutations_list.dorothea_progeny_merged.png"), height = 10, width = 9)

```


## Merge

```{r}
postmortem_mutations_compared_fig.merged = plot_grid(
  plot_grid(nygc_postmortem_spinal_cord.sporadic.volcano, nygc_postmortem_spinal_cord.c9orf72.volcano, nygc_postmortem_spinal_cord.sod1.volcano, nygc_postmortem_spinal_cord.fus.volcano, ncol = 4, labels = "auto"),
  plot_grid(postmortem_mutants_bind_datasets.res.up.upset, postmortem_mutants_bind_datasets.res.down.upset, ncol = 2, rel_widths = c(1,1), labels = letters[5:6]),
  plot_grid(nygc_postmortem_spinal_cord.sporadic.go_curated, nygc_postmortem_spinal_cord.c9orf72.go_curated, nygc_postmortem_spinal_cord.sod1.go_curated, nygc_postmortem_spinal_cord.fus.go_curated, ncol = 4, labels = letters[7:10]),
  postmortem_mutations_list.progeny_scores.plot, 
  postmortem_mutations_merged_dorothea.contrast_acts.volcano,
  nrow = 5, ncol = 1, rel_heights = c(1,0.8,1.2,1,1), labels = c("","","","k","l"))
# postmortem_mutations_compared_fig.merged
ggsave(postmortem_mutations_compared_fig.merged, filename = here(proj_path,"expression/deseq2/postmortem_mutations_compared_fig.merged.png"), height = 12, width = 11)

```



GSEA DNA damage 

```{r}
nygc_postmortem_spinal_cord.geneList <- nygc_postmortem_spinal_cord$res %>% drop_na(stat) %>% pull(stat)
names(nygc_postmortem_spinal_cord.geneList) <- nygc_postmortem_spinal_cord$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
nygc_postmortem_spinal_cord.geneList = sort(nygc_postmortem_spinal_cord.geneList, decreasing = TRUE)

gprofiler_database.DNA_damage_pathways = gprofiler_database[grepl('DNA damage|p53', names(gprofiler_database))]
postmortem.gprofiler_database.DNA_damage_pathways_fgseaRes <- fgsea(pathways = gprofiler_database.DNA_damage_pathways, stats = nygc_postmortem_spinal_cord.geneList)
postmortem.gprofiler_database.DNA_damage_pathways_fgseaRes %>% arrange(pval)
postmortem.gprofiler_database.DNA_damage.filt = filter(postmortem.gprofiler_database.DNA_damage_pathways_fgseaRes, size >7, !grepl("uncertain|negative|prostate",pathway)) %>% mutate(p.signif = case_when(pval < 0.003 ~ "***", pval < 0.009 ~ "***",pval < 0.01 ~ "**",pval < 0.05 ~ "*", TRUE ~ ""), group1 = "pathway", group2 = "pathway")
postmortem.gprofiler_database.DNA_damage_pathways_fgseaRes.barplot <- ggplot(postmortem.gprofiler_database.DNA_damage.filt, aes(x=reorder(pathway, NES), y = NES)) +
    geom_bar(aes(fill = NES), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "DNA damage pathways", y = "ALS vs CTRL Postmortem\nNormalised Enrichment Score", x = "") + geom_hline(yintercept = 0, linetype = 3) + 
    stat_pvalue_manual(gprofiler_database.DNA_damage.filt, label = "p.signif", y.position = 2, size = 4, xmin = "pathway", xmax = NULL, hide.ns = TRUE) + coord_flip()
postmortem.gprofiler_database.DNA_damage_pathways_fgseaRes.barplot
ggsave(postmortem.gprofiler_database.DNA_damage_pathways_fgseaRes.barplot, filename = here(proj_path,"expression/deseq2/postmortem.p53_pathways.fgsea.barchart.png"), height = 11, width = 9)

```


# Fig S15 TDP43 vs non TDP43

## iPSMN

```{r}
load(file = here(proj_path, "expression/deseq2/ipsc_mn_tdp43_status.RData"))

# # Volcanos
# filter(ipsc_mn_als_datasets.metadata, !mutation %in% c("sod1","fus")) %>% count(dataset)
# filter(ipsc_mn_als_datasets.metadata, !mutation %in% c("sod1","fus")) %>% count(condition)
# ipsc_mn_tdp43_positive_datasets.labels <- ipsc_mn_tdp43_positive_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, p53.signalling.pathway)) %>% distinct(gene_name) %>% pull(gene_name)
# ipsc_mn_tdp43_positive_datasets.volcano <- volcano_plot(ipsc_mn_tdp43_positive_datasets$res, "TDP-43 pathology", numerator = "TDP-43 positive", denomenator = "CTRL", subtitle = "106 Controls vs 289 TDP-43 positive", ymax = 10, xmax = 2, dpi = 300, gene_labels = c(ipsc_mn_tdp43_positive_datasets.labels))
# ipsc_mn_tdp43_positive_datasets.volcano

# filter(ipsc_mn_als_datasets.metadata, mutation %in% c("sod1","fus","ctrl"), dataset %in% c("answerals","bhinge","catanese", "desantis", "hawkins", "kapeli", "kiskinis", "neurolincs.diMN", "wang")) %>% count(dataset) 
# filter(ipsc_mn_als_datasets.metadata, mutation %in% c("sod1","fus","ctrl"), dataset %in% c("answerals","bhinge","catanese", "desantis", "hawkins", "kapeli", "kiskinis", "neurolincs.diMN", "wang")) %>% count(condition) 
# ipsc_mn_tdp43_negative_datasets.labels <- ipsc_mn_tdp43_negative_datasets$res %>% filter(padj < 0.05, !grepl("^AC|^AL", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
# ipsc_mn_tdp43_negative_datasets.volcano <- volcano_plot(ipsc_mn_tdp43_negative_datasets$res, "No TDP-43 pathology", numerator = "TDP-43 negative", denomenator = "CTRL", subtitle = "60 Controls vs 34 TDP-43 negative", ymax = 10, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_tdp43_negative_datasets.labels))
# ipsc_mn_tdp43_negative_datasets.volcano

# # Correlation
# ipsc_mn_tdp43_positive_negative_list = list("TDP-43 positive" = ipsc_mn_tdp43_positive_datasets$res, "TDP-43 negative" = ipsc_mn_tdp43_negative_datasets$res)
# ipsc_mn_tdp43_positive_negative_stat.corr = plot_de_correlation(model_list1 = ipsc_mn_tdp43_positive_negative_list, model_list2 = ipsc_mn_tdp43_positive_negative_list, mutation1 = "TDP-43 positive", mutation2 = "TDP-43 negative", 
#                                                           gene_labels = pull(filter(ipsc_mn_tdp_pathology$res,padj < 0.05),gene_name), plot_p_value = TRUE)
# ipsc_mn_tdp43_positive_negative_stat.corr


ipsc_mn_tdp43_pos_neg_list = list("TDP-43 ALS" = select(ipsc_mn_tdp43_positive_datasets$res,gene_name,stat), "non-TDP-43 ALS" = select(ipsc_mn_tdp43_negative_datasets$res,gene_name,stat))
ipsc_mn_tdp43_pos_neg_list = map(ipsc_mn_tdp43_pos_neg_list, ~{drop_na(.x)}) 
ipsc_mn_tdp43_pos_neg_list = map(ipsc_mn_tdp43_pos_neg_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_tdp43_pos_neg_list = map(ipsc_mn_tdp43_pos_neg_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_tdp43_pos_neg_list.progeny_scores = map_df(ipsc_mn_tdp43_pos_neg_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)

ipsc_mn_tdp43_positive_datasets_progeny.pathwayActivity <- ggplot(ipsc_mn_tdp43_pos_neg_list.progeny_scores, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "iPSMN signaling", x = "", y = "ALS vs CTRL\nNormalised enrichment") +
    stat_pvalue_manual(mutate(ipsc_mn_tdp43_pos_neg_list.progeny_scores, y.position = case_when(mutation == "non-TDP-43 ALS" & source %in% c("PI3K","TGFb")~ 4, mutation == "TDP-43 ALS" & source %in% c("MAPK") ~ 14, mutation == "TDP-43 ALS"~15, mutation == "non-TDP-43 ALS"~4.5)), label = "p.signif", xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) +
  facet_wrap(~mutation, scales = "free")
ipsc_mn_tdp43_positive_datasets_progeny.pathwayActivity

# Dorothea
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_tdp43_pos_neg_list.dorothea_scores = map_df(ipsc_mn_tdp43_pos_neg_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_tdp43_positive_datasets_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_tdp43_pos_neg_list.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "iPSMN TFs") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_tdp43_pos_neg_list.dorothea_scores, source %in% "TP53"), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8)) +
  facet_wrap(~mutation)
ipsc_mn_tdp43_positive_datasets_dorothea.contrast_acts.volcano


# # TDP43 positive vs negative ALS
# ipsc_mn_tdp_pathology.metadata %>% count(tdp_pathology)
# # tdp_pathology     n
#   # <chr>         <int>
# # 1 no               34
# # 2 yes             306
# ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05) #110
# ipsc_mn_tdp_pathology.als.labels <- ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway)) %>% distinct(gene_name) %>% pull(gene_name)
# ipsc_mn_tdp_pathology.volcano <- volcano_plot(ipsc_mn_tdp_pathology$res, numerator = "TDP-43 pathology", denomenator = "SOD1 and FUS ALS",
#                subtitle = "34 SOD1 and FUS vs 306 TDP-43 pathology", ymax = 8, xmax = 2, dpi = 300, gene_labels = c(ipsc_mn_tdp_pathology.als.labels, "TARDBP","SOD1","FUS","SNRPE","KCNA6"))
# ipsc_mn_tdp_pathology.volcano
# 
# ipsc_mn_tdp_pathology$res %>% filter(gene_name %in% c("TARDBP","SOD1","FUS")) #29
# #  gene_id         baseMean log2FoldChange  lfcSE   stat pvalue  padj gene_name
# #   <chr>              <dbl>          <dbl>  <dbl>  <dbl>  <dbl> <dbl> <chr>    
# # 1 ENSG00000120948    4161.       -0.0801  0.0421 -1.90  0.0573 0.489 TARDBP   
# # 2 ENSG00000089280    5696.       -0.157   0.0888 -1.77  0.0761 0.533 FUS      
# # 3 ENSG00000142168    2825.       -0.00482 0.0463 -0.104 0.917  0.986 SOD1     
# ipsc_mn_tdp_pathology.dge_genes <- ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost() # run gprofiler2 on all groups - runs on all KEGG, GO, REAC,TF, WP pathways simultaneously
# ipsc_mn_tdp_pathology.dge_genes$result %>% filter(!grepl("HPA|WP",term_id))
# # query significant     p_value term_size query_size intersection_size  precision      recall            term_id source                                  term_name effective_domain_size source_order                            parents
# # 1  query_1        TRUE 0.002813228       309         55                 8 0.14545455 0.025889968         GO:0006260  GO:BP                            DNA replication                 21100         2278                         GO:0006259
# # 2  query_1        TRUE 0.003580182      1023         55                13 0.23636364 0.012707722         GO:0006259  GO:BP                      DNA metabolic process                 21100         2277             GO:0044260, GO:0090304
# # 3  query_1        TRUE 0.005252230      1059         55                13 0.23636364 0.012275732         GO:0051276  GO:BP                    chromosome organization                 21100        15725                         GO:0006996
# # 4  query_1        TRUE 0.005383069         3         56                 2 0.03571429 0.666666667         GO:0031391  GO:CC                      Elg1 RFC-like complex                 21698         1318             GO:0005694, GO:1902494
# # 5  query_1        TRUE 0.012425828      1838         56                15 0.26785714 0.008161045         GO:0005694  GO:CC                                 chromosome                 21698          347                         GO:0043232
# # 6  query_1        TRUE 0.026781634         6         56                 2 0.03571429 0.333333333         GO:0120099  GO:CC           procentriole replication complex                 21698         3868             GO:0005737, GO:0032991
# # 7  query_1        TRUE 0.006774307       113         56                 5 0.08928571 0.044247788         GO:0008094  GO:MF      ATP-dependent activity, acting on DNA                 20183         1961             GO:0140097, GO:0140657
# # 8  query_1        TRUE 0.023420435        74         56                 4 0.07142857 0.054054054         GO:0003678  GO:MF                      DNA helicase activity                 20183          403             GO:0004386, GO:0008094
# # 9  query_2        TRUE 0.009118909         4         21                 2 0.09523810 0.500000000         GO:1902994  GO:BP          regulation of phospholipid efflux                 21100        26109             GO:0033700, GO:2001138
# # 10 query_2        TRUE 0.009118909         4         21                 2 0.09523810 0.500000000         GO:1902995  GO:BP positive regulation of phospholipid efflux                 21100        26110 GO:0033700, GO:1902994, GO:2001140
# # 11 query_2        TRUE 0.040173259        18         11                 2 0.18181818 0.111111111 REAC:R-HSA-1369062   REAC      ABC transporters in lipid homeostasis                 10770            5                  REAC:R-HSA-382556
# ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_CHROMATIN_ASSEMBLY_OR_DISASSEMBLY) # CENPV, NASP, RBBP4, SUPT16H
# ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_U1_SNRNP) # SNRPE
# ipsc_mn_tdp_pathology$res %>% filter(padj < 0.05, gene_name %in% p53.signalling.pathway) # RFC4, ATAD5, SUPT16H
# 
# net_progeny <- get_progeny(organism = 'human', top = 100)
# ipsc_mn_tdp_pathology.res.matrix <- ipsc_mn_tdp_pathology$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
# ipsc_mn_tdp_pathology_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_tdp_pathology.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
# ipsc_mn_tdp_pathology_progeny.pathwayActivity <- ggplot(ipsc_mn_tdp_pathology_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
#     geom_bar(aes(fill = score), stat = "identity") +
#     scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
#     theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
#     labs(title = "Signaling Pathways", x = "", y = "TDP43 vs non-TDP43 ALS Normalised enrichment") +
#     stat_pvalue_manual(ipsc_mn_tdp_pathology_progeny.contrast_acts, label = "p.signif", y.position = 5, xmin = "source", xmax = NULL, size = 3, hide.ns = TRUE)
# ipsc_mn_tdp_pathology_progeny.pathwayActivity
# 
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# ipsc_mn_tdp_pathology_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_tdp_pathology.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>%
#   mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))
# 
# ipsc_mn_tdp_pathology_dorothea.contrast_acts.labels.up = ipsc_mn_tdp_pathology_dorothea.contrast_acts %>% top_n(score, n = 3) %>% pull(source)
# ipsc_mn_tdp_pathology_dorothea.contrast_acts.labels.down = ipsc_mn_tdp_pathology_dorothea.contrast_acts %>% top_n(-score, n = 3) %>% pull(source)
# 
# # Volcano of TFs score vs p_value
# ipsc_mn_tdp_pathology_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_tdp_pathology_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
#     scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
#   labs(y = expression(-log[10]~P~value), x = "Normalised enrichment TDP43 vs non-TDP43", title = "Transcription Factors") +
#     guides(colour = "none") +
#     theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
#   # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
#   geom_text_repel(fontface = "italic", data = filter(ipsc_mn_tdp_pathology_dorothea.contrast_acts, source %in% c(ipsc_mn_tdp_pathology_dorothea.contrast_acts.labels.up, ipsc_mn_tdp_pathology_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +
#   geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
# ipsc_mn_tdp_pathology_dorothea.contrast_acts.volcano

```

## Postmortem

```{r}
load(file = here(proj_path, "expression/deseq2/nygc_postmortem_spinal_cord_tdp43_status.RData"))

nygc_postmortem_spinal_cord_tdp43_pos_neg_list = list("TDP-43 ALS" = select(nygc_postmortem_spinal_cord_tdp43_positive_datasets$res,gene_name,stat), "non-TDP-43 ALS" = select(nygc_postmortem_spinal_cord_tdp43_negative_datasets$res,gene_name,stat))
nygc_postmortem_spinal_cord_tdp43_pos_neg_list = map(nygc_postmortem_spinal_cord_tdp43_pos_neg_list, ~{drop_na(.x)}) 
nygc_postmortem_spinal_cord_tdp43_pos_neg_list = map(nygc_postmortem_spinal_cord_tdp43_pos_neg_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
nygc_postmortem_spinal_cord_tdp43_pos_neg_list = map(nygc_postmortem_spinal_cord_tdp43_pos_neg_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
nygc_postmortem_spinal_cord_tdp43_pos_neg_list.progeny_scores = map_df(nygc_postmortem_spinal_cord_tdp43_pos_neg_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)

nygc_postmortem_spinal_cord_tdp43_positive_datasets_progeny.pathwayActivity <- ggplot(nygc_postmortem_spinal_cord_tdp43_pos_neg_list.progeny_scores, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Postmortem signaling", x = "", y = "ALS vs CTRL\nNormalised enrichment") +
    stat_pvalue_manual(mutate(nygc_postmortem_spinal_cord_tdp43_pos_neg_list.progeny_scores, y.position = case_when(mutation == "non-TDP-43 ALS" & source == "TNFa" ~ 6, mutation == "TDP-43 ALS" & source %in% c("TNFa","NFkB") ~ 5, mutation == "TDP-43 ALS"~5.5, mutation == "non-TDP-43 ALS"~6.5)), label = "p.signif", xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) +
  facet_wrap(~mutation, scales = "free")
nygc_postmortem_spinal_cord_tdp43_positive_datasets_progeny.pathwayActivity

nygc_postmortem_spinal_cord_tdp43_pos_neg_list.progeny_scores %>% filter(source == "p53")


# Dorothea
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord_tdp43_pos_neg_list.dorothea_scores = map_df(nygc_postmortem_spinal_cord_tdp43_pos_neg_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

nygc_postmortem_spinal_cord_tdp43_positive_datasets_dorothea.contrast_acts.volcano = ggplot(nygc_postmortem_spinal_cord_tdp43_pos_neg_list.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Postmortem TFs") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(nygc_postmortem_spinal_cord_tdp43_pos_neg_list.dorothea_scores, source %in% "TP53"), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8)) +
  facet_wrap(~mutation, scales = "free")
nygc_postmortem_spinal_cord_tdp43_positive_datasets_dorothea.contrast_acts.volcano

nygc_postmortem_spinal_cord_tdp43_pos_neg_list.dorothea_scores %>% filter(source == "TP53")



# # p53 in both iPSMN & postmortem
# ipsc_mn_tdp43_pos_neg_list.p53.progeny_scores = ipsc_mn_tdp43_pos_neg_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, model = "iPSMN")
# nygc_postmortem_spinal_cord_tdp43_pos_neg_list.p53.progeny_scores = nygc_postmortem_spinal_cord_tdp43_pos_neg_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, model = "Postmortem")
# ipsmn_postmortem_tdp43_pos_neg_list.p53.progeny_scores.plot <- ggplot(bind_rows(ipsc_mn_tdp43_pos_neg_list.p53.progeny_scores, nygc_postmortem_spinal_cord_tdp43_pos_neg_list.p53.progeny_scores), aes(x=reorder(mutation, score), y = score)) +
#     geom_bar(aes(fill = score), stat = "identity") +
#   scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
#   theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
#   geom_hline(yintercept = 0, linetype = 3) +
#   labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
#   stat_pvalue_manual(bind_rows(ipsc_mn_tdp43_pos_neg_list.p53.progeny_scores, nygc_postmortem_spinal_cord_tdp43_pos_neg_list.p53.progeny_scores), label = "p.signif", y.position = c(15,15,6,6), xmin = "mutation", xmax = NULL, size = 6, hide.ns = TRUE) +
#   facet_wrap(~model, scales = "free")
# ipsmn_postmortem_tdp43_pos_neg_list.p53.progeny_scores.plot
# 
# # TP53 in both iPSMN & postmortem
# ipsc_mn_tdp43_pos_neg_list.TP53.dorothea_scores = ipsc_mn_tdp43_pos_neg_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, model = "iPSMN")
# nygc_postmortem_spinal_cord_tdp43_pos_neg_list.TP53.dorothea_scores = nygc_postmortem_spinal_cord_tdp43_pos_neg_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, model = "Postmortem")
# ipsc_mn_tdp43_pos_neg_list.TP53.dorothea_scores.plot <- ggplot(bind_rows(ipsc_mn_tdp43_pos_neg_list.TP53.dorothea_scores, nygc_postmortem_spinal_cord_tdp43_pos_neg_list.TP53.dorothea_scores), aes(x=reorder(mutation, score), y = score)) +
#     geom_bar(aes(fill = score), stat = "identity") +
#   scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
#   theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
#   geom_hline(yintercept = 0, linetype = 3) +
#   labs(title = "TP53 TF activity", x = "", y = "Normalised enrichment") + #ylim(c(-11,11)) +
#   stat_pvalue_manual(bind_rows(ipsc_mn_tdp43_pos_neg_list.TP53.dorothea_scores, nygc_postmortem_spinal_cord_tdp43_pos_neg_list.TP53.dorothea_scores), label = "p.signif", y.position = c(8,8,3.5,3.5), xmin = "mutation", xmax = NULL, size = 6, hide.ns = TRUE) +
#   facet_wrap(~model, scales = "free")
# ipsc_mn_tdp43_pos_neg_list.TP53.dorothea_scores.plot

```

## TDP43 knockdown

Liu TDP43+ vs - postmortem
TDP43 knockdown models

```{r}
neuronal_nuclei_tdp43_liu = readRDS(here(proj_path,"expression/deseq2/neuronal_nuclei_tdp43_liu.rds"))
tdp43_kd_datasets = readRDS(here(proj_path,"expression/deseq2/tdp43_kd_datasets.rds"))

ipsc_mn_tdp43_neuron_kd_list = list("Neuronal Nuclei Brain" = select(neuronal_nuclei_tdp43_liu$res,gene_name,stat), 
                                  "TDP-43 knockdown cells" = select(tdp43_kd_datasets$res,gene_name,stat))
ipsc_mn_tdp43_neuron_kd_list = map(ipsc_mn_tdp43_neuron_kd_list, ~{drop_na(.x)}) 
ipsc_mn_tdp43_neuron_kd_list = map(ipsc_mn_tdp43_neuron_kd_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_tdp43_neuron_kd_list = map(ipsc_mn_tdp43_neuron_kd_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_tdp43_neuron_kd_list.progeny_scores = map_df(ipsc_mn_tdp43_neuron_kd_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)

ipsc_mn_tdp43_neuron_kd_datasets_progeny.pathwayActivity <- ggplot(ipsc_mn_tdp43_neuron_kd_list.progeny_scores, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "TDP-43 depletion", x = "", y = "TDP-43 depleted vs TDP-43 retained\nNormalised enrichment") +
     stat_pvalue_manual(mutate(ipsc_mn_tdp43_neuron_kd_list.progeny_scores, y.position = case_when(mutation == "Neuronal Nuclei Brain" & source == "p53" ~ 3.5, mutation == "TDP-43 knockdown cells" & source %in% c("TNFa","VEGF") ~ 15, mutation == "TDP-43 knockdown cells"~16, mutation == "Neuronal Nuclei Brain"~4)), label = "p.signif", xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) +
  facet_wrap(~mutation, scales = "free")
ipsc_mn_tdp43_neuron_kd_datasets_progeny.pathwayActivity

# Dorothea
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_tdp43_neuron_kd_list.dorothea_scores = map_df(ipsc_mn_tdp43_neuron_kd_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_tdp43_neuron_kd_datasets_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_tdp43_neuron_kd_list.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment TDP-43 depleted vs TDP-43 positive", title = "TDP-43 depletion TFs") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_tdp43_neuron_kd_list.dorothea_scores, source %in% "TP53"), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8)) +
  facet_wrap(~mutation, scales = "free")
# ipsc_mn_tdp43_neuron_kd_datasets_dorothea.contrast_acts.volcano



# Check TDP-43 knockdown with TARDBP TPM
tdp43_kd_melamed.tardbp.tpm <- read_tsv(here(shared_path,"public-data/shsy5y-tdp43-kd-melamed-2019/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="melamed")
tdp43_kd_kapeli.tardbp.tpm <- read_tsv(here(shared_path,"public-data/ipsc-mn-tdp43-kd-kapeli-2016/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="kapeli")
tdp43_kd_ferguson.tardbp.tpm <- read_tsv(here(shared_path,"public-data/hela-tdp43-kd-ferguson-2019/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="ferguson")
tdp43_kd_tam.tardbp.tpm <- read_tsv(here(shared_path,"public-data/shsy5y-tdp43-kd-tam-2019/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="tam")
tdp43_kd_brown.tardbp.tpm <- read_tsv(here(shared_path,"public-data/ipsc-mn-tardbp-kd-brown-2022/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="brown")
tdp43_kd_klim.tardbp.tpm <- read_tsv(here(shared_path,"public-data/ipsc-mn-tardbp-kd-klim-2019/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="klim")
tdp43_kd_appocher.tardbp.tpm <- read_tsv(here(shared_path,"public-data/shsy5y-tardbp-kd-appocher-2017/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="appocher")
tdp43_kd_dunker.tardbp.tpm <- read_tsv(here(shared_path,"public-data/shsy5y-tdp43-kd-dunker-2021/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="dunker")
tdp43_kd_liu.tardbp.tpm <- read_tsv(here(shared_path,"public-data/neuronal-nuclei-tdp43-liu-2019/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% filter(gene_name %in% c("TARDBP")) %>%  
  pivot_longer(!c(gene_name,gene_id), names_to = "sample", values_to = "value") %>% mutate(dataset="liu")
tdp43_kd_datasets.tardbp.tpm = bind_rows(tdp43_kd_melamed.tardbp.tpm, tdp43_kd_kapeli.tardbp.tpm, tdp43_kd_ferguson.tardbp.tpm, tdp43_kd_tam.tardbp.tpm, tdp43_kd_brown.tardbp.tpm, tdp43_kd_klim.tardbp.tpm, tdp43_kd_appocher.tardbp.tpm, tdp43_kd_dunker.tardbp.tpm) %>% #, tdp43_kd_liu.tardbp.tpm)
  mutate(group = str_split_fixed(sample,"_",2)[,1])

tdp43_kd_datasets.tardbp.tpm %>% count(dataset)
tdp43_kd_datasets.tardbp.tpm %>% my_summarise(value, dataset, group)

TARDBP_gene_counts.TPM.violin = violin_plot(data = tdp43_kd_datasets.tardbp.tpm, color_by = "group", continuous = "value", facet_by = "dataset", facet_nrow = 2,  cols = pal_npg("nrc", alpha = 1)(8), ylabel = "TARDBP TPM", plot_violin = FALSE, arrow_labels = FALSE,  dpi = 300, plot_stats = FALSE)
TARDBP_gene_counts.TPM.violin
ggsave(TARDBP_gene_counts.TPM.violin, filename = here(proj_path,"expression/deseq2/TARDBP_gene_counts.TPM.violin.png"), width = 6, height = 3)

tdp43_kd_datasets.tardbp.tpm %>% my_summarise(value, dataset, group)

# Heatmap of cell types
tdp43_kd_datasets$vsd.counts = assay(tdp43_kd_datasets$vsd) %>% as_tibble(rownames = "gene_id") %>% left_join(gene2ens)
min_vst = min(tdp43_kd_datasets$vsd.counts[,2])
celltype.markers.marker <- tdp43_kd_datasets$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))
celltype.markers.rnaseq.cluster.heatmap <- Heatmap(t(celltype.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = celltype.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = celltype.markers.df.filt$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelial", "Myeloid", "Oligodendrocyte","Motor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), column_title = NULL, 
          show_row_names = TRUE,row_names_gp = gpar(fontsize = 7), 
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
celltype.markers.rnaseq.cluster.heatmap

```


## TDP43 overexpression

```{r}
maor_nof_cell_tdp_vs_ctrl = read_csv(here(proj_path, "expression/maor_nof_cell_tdp/GSE162048_D1CON_vs_D1TDP-ExpDiff.csv")) %>% rename(gene_name.mouse = "Gene") %>% left_join(mus.musculus.gene2ens)

maor_nof_cell_tdp_vs_ctrl.genelist = maor_nof_cell_tdp_vs_ctrl %>% select(gene_name.mouse, stat) %>% drop_na() %>% distinct(gene_name.mouse, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name.mouse")

# PROGENy
net_progeny.mouse <- get_progeny(organism = 'mouse', top = 100)
maor_nof_cell_tdp_vs_ctrl.progeny_scores = run_wmean(maor_nof_cell_tdp_vs_ctrl.genelist, net=net_progeny.mouse, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""))

maor_nof_cell_tdp_vs_ctrl_progeny.pathwayActivity <- ggplot(maor_nof_cell_tdp_vs_ctrl.progeny_scores, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "TDP-43 over-expression", x = "", y = "TDP-43 overexpresison vs CTRL\nNormalised enrichment") +
    stat_pvalue_manual(mutate(maor_nof_cell_tdp_vs_ctrl.progeny_scores, y.position = 9.5, group1=source, group2=source), label = "p.signif", xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE)
maor_nof_cell_tdp_vs_ctrl_progeny.pathwayActivity

# Dorothea
net_dorothea.mouse <- get_dorothea(organism='mouse', levels=c('A', 'B', 'C'))
maor_nof_cell_tdp_vs_ctrl.dorothea_scores = run_wmean(maor_nof_cell_tdp_vs_ctrl.genelist, net=net_dorothea.mouse, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

maor_nof_cell_tdp_vs_ctrl_dorothea.contrast_acts.volcano = ggplot(maor_nof_cell_tdp_vs_ctrl.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment TDP-43 overexpression vs CTRL", title = "TDP-43 overexpression TFs") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(maor_nof_cell_tdp_vs_ctrl.dorothea_scores, source %in% "Tp53"), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
maor_nof_cell_tdp_vs_ctrl_dorothea.contrast_acts.volcano

maor_nof_cell_tdp_vs_ctrl.dorothea_scores %>% filter(source == "Tp53")

```



## Merge

```{r}
tdp43_subgroups_neurons_kd.merged = plot_grid(
  ipsc_mn_tdp43_positive_datasets_progeny.pathwayActivity, ipsc_mn_tdp43_positive_datasets_dorothea.contrast_acts.volcano, 
  nygc_postmortem_spinal_cord_tdp43_positive_datasets_progeny.pathwayActivity, nygc_postmortem_spinal_cord_tdp43_positive_datasets_dorothea.contrast_acts.volcano, 
  ipsc_mn_tdp43_neuron_kd_datasets_progeny.pathwayActivity, ipsc_mn_tdp43_neuron_kd_datasets_dorothea.contrast_acts.volcano,
  maor_nof_cell_tdp_vs_ctrl_progeny.pathwayActivity, maor_nof_cell_tdp_vs_ctrl_dorothea.contrast_acts.volcano, 
  nrow = 4, ncol = 2, rel_widths = c(1.2,1), labels = "auto")
# tdp43_subgroups_neurons_kd.merged
ggsave(tdp43_subgroups_neurons_kd.merged, filename = here(proj_path,"expression/deseq2/tdp43_subgroups_neurons_kd.merged.png"), height = 12, width = 11)

```


# Fig 5 Splicing

MAJIQ 

## Volcano & GO terms

```{r}
majiq_schematic = magick::image_read(here(proj_path,"splicing/majiq/figures/MAJIQ splicing schematic.png"))
majiq_schematic.gg = ggdraw() + draw_image(majiq_schematic)
majiq_schematic.gg

# als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
# als_ctrl.changing_only.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.changing_only.tsv"), grp1 = "als", grp2 = "ctrl")
# pan_als.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/pan_als"), skip_rows = 9)
# save(als_ctrl.voila, als_ctrl.changing_only.voila, pan_als.voila_modulizer, file = here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.RData"))
load(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.RData"))

als_ctrl.changing_only.voila
als_ctrl.voila
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 264
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 161
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 27
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.DDR.repair_gene_set) # MTA1, POLM, PMS1 HUWE1, HDAC1, ZWIM7
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) # MTA1, POLM, PMS1 HUWE1, HDAC1, ZWIM7
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # 0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) # HLA-B
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% sporadic_ALS_genes) #0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_GWAS) #0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_EWAS) # SIRT2
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_modifiers) # 0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% TDP43_spliced_genes) # POLDIP3, CAMK2B, PPP6R3, CEP290

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # HDAC1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`DNA repair`) #  MTA1, POLM, PMS1 HUWE1, ZWIM7
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, -log10(tnom) > 4) # TVP23C-CDRT4, SREK1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("SETMAR", "RIF1", "HUS1", "SMC6", "METTL22")) # METTL22


# Fisher exact overlap test
als_ctrl.voila.p53 = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% unique(c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, p53.DDR.repair_gene_set))) %>% pull(gene_name)
als_ctrl.voila.p53 #"MTA1"   "POLM"   "POLM"   "PMS1"   "CASP2"  "CASP2"  "HUWE1"  "HUWE1"  "HDAC1"  "HDAC1"  "ZSWIM7" "ZSWIM7"
newGeneOverlap(als_ctrl.voila.p53, unique(c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, p53.DDR.repair_gene_set)), genome.size=nrow(distinct(als_ctrl.voila, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=7
# listB size=819
# Intersection size=7
# Overlapping p-value=2.3e-08
# Jaccard Index=0.0
als_ctrl.voila.RBP = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% pull(gene_name)
newGeneOverlap(als_ctrl.voila.RBP, RNA_BINDING_PROTEINS, genome.size=nrow(distinct(als_ctrl.voila, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=21
# listB size=1542
# Intersection size=21
# Overlapping p-value=6.8e-18
# Jaccard Index=0.0

## Volcano
ipsc_mn_als_datasets.polyA.metadata %>% count(dataset) # 10
ipsc_mn_als_datasets.polyA.metadata %>% count(condition) #48 ALS, 43 ctrl
# majiq.als_vs_ctrl.RBP.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.RBP.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("YTHDC2", "THOC1", "PRR3", "STAU2", "PTBP3", "SREK1", "POLDIP3")) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.ALS.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.DDR.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.DDR.repair_gene_set) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.extra_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("METTL22","LINC00665")) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, -log10(tnom)>3, -log10(tnom)<4) 

majiq.als_vs_ctrl.volcano.junction_labels = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(majiq.als_vs_ctrl.RBP.junction_labels, majiq.als_vs_ctrl.ALS.junction_labels, majiq.als_vs_ctrl.DDR.junction_labels)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.volcano <- majiq_het_volcano_plot(als_ctrl.voila, "Alternative splicing: iPSMN pan ALS", numerator = "ALS", denomenator = "CTRL", subtitle = "polyA: 43 CTRL vs 48 ALS", ymax = 5, xmax = 0.32, dpi = 300, junction_labels = c(majiq.als_vs_ctrl.RBP.junction_labels, majiq.als_vs_ctrl.ALS.junction_labels, majiq.als_vs_ctrl.DDR.junction_labels, majiq.als_vs_ctrl.extra_labels))
# majiq.als_vs_ctrl.volcano

# GO of splicing events
als_ctrl.voila.gost = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
als_ctrl.voila.gost_filt <- als_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(als_ctrl.voila.gost_filt$query) # 
paste('"',unique(als_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
"Hedgehog signaling pathway",
"regulation of nervous system development",
"positive regulation of synapse maturation",
# "latrotoxin receptor activity",
"protein binding")
als_ctrl.voila.gost_filt.terms <- als_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
als_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", als_ctrl.voila.gost_filt.terms$term_name) 
als_ctrl.voila.gost_filt.terms.curated <- curated_profiles(als_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "iPSMN splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
# als_ctrl.voila.gost_filt.terms.curated

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`protein binding`) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`regulation of nervous system development`) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`positive regulation of synapse maturation`) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% kegg.pathways.msigdb$KEGG_HEDGEHOG_SIGNALING_PATHWAY) %>% arrange(-abs(deltapsi))

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_PROTEIN_BINDING) %>% arrange(-abs(deltapsi)) # ARHGEF7, NRP1, ADD1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_AXON) %>% arrange(-abs(deltapsi)) %>% distinct(gene_name) # SRRM1, SORBS2, PAN3, MARK2, HUWE1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% arrange(-abs(deltapsi)) %>% print(n=Inf) # FZNF326, RTF1, AUH, NOVA1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_RNA_SPLICING) %>% arrange(-abs(deltapsi)) #THOC1, ZC3H13, PTBP3, SREK1


# Compare splicing with gene expression
als_splicing_events_genes = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
als_degs = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% pull(gene_name)
als_degs[als_degs %in% als_splicing_events_genes] #THB15B


# TDP43 knockdown overlap
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
tdp43kd.als_vs_ctrl.brown.voila.sig = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9)# %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.sig 

tdp43kd.als_vs_ctrl.brown.voila.labels = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)

# # Cryptic Exons == denovo & cassette_exon
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 28 / 264
28/264# 10.6%
pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer) #7 cryptic exons
pan_als.voila_modulizer.cryptic = pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) %>% pull(gene_name)  # "RELCH"         "HOXC4"         "RBM26"         "SLC35B3"       "TENM3"         "TPTEP2-CSNK1E" "ZSCAN29"   
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 59 /264
50/264
# overlap cryptic exons with Brown et al TDP43 knockdown
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
tdp43kd.als_vs_ctrl.brown.voila.sig.cryptic = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, de_novo_junctions == TRUE)# %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.sig.cryptic #121

tdp43kd.als_vs_ctrl.brown.voila.sig.cryptic$gene_name[tdp43kd.als_vs_ctrl.brown.voila.sig.cryptic$gene_name %in% "UNC13A"]
tdp43kd.als_vs_ctrl.brown.voila.labels = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
```

## Modulizer

```{r}
# pan_als.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/pan_als"), skip_rows = 9)

# Modulizer pie chart of differential events
modulizer_simplfied.sig.summary = pan_als.voila_modulizer %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer_simplified) %>% mutate(modulizer_simplified = gsub("_"," ",modulizer_simplified), modulizer_accronym = case_when(modulizer_simplified == "alt3and5prime" ~ "A3&A5SS", modulizer_simplified == "alt5prime" ~ "A5SS", modulizer_simplified == "alt3prime" ~ "A3SS", modulizer_simplified == "alternate first exon" ~ "AFE", modulizer_simplified == "alternate last exon" ~ "ALE", modulizer_simplified == "multi exon spanning" ~ "multi-exon", modulizer_simplified == "intron retention" ~ "retained intron", TRUE ~ modulizer_simplified), modulizer_accronym = fct_reorder(modulizer_accronym, n, .desc = TRUE)) %>% arrange(desc(modulizer_accronym)) %>% 
  mutate(pct = n/sum(n) *100, pct.label = case_when(pct > 8 ~ paste(as.character(n),"\n", as.character(round(pct,1)),"%"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) #%>% print
modulizer_simplfied.sig.summary %>% print
pan_als.voila_modulizer.sig.pie = ggplot(modulizer_simplfied.sig.summary, aes(x = "", y = pct, fill = fct_reorder(modulizer_accronym, n, .desc = TRUE))) + geom_bar(width = 1, stat = "identity", color="white") + coord_polar("y", start=0) + 
  geom_text(aes(x = 1.2, y = ypos, label = pct.label), color = "black", size=2.5) + scale_fill_brewer(palette="Set1") + 
  theme_oz() + theme_void() + theme(legend.title=element_blank(), legend.position = "top", legend.text=element_text(size=7), legend.spacing.y = unit(1, 'mm'), legend.spacing.x = unit(1, 'mm'), legend.key.height = unit(0.4, 'cm'), legend.key.width = unit(0.3, 'cm')) + guides(fill=guide_legend(nrow=3))
pan_als.voila_modulizer.sig.pie

# Intron retention
pan_als.voila_modulizer.ir = pan_als.voila_modulizer %>% filter(modulizer_simplified == "intron_retention")
pan_als.voila_modulizer.ir %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # 60
pan_als.voila_modulizer.ir %>% filter(gene_name %in% ALS_genes, abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # SIRT2
pan_als.voila_modulizer.ir %>% filter(gene_name %in% "SFPQ")


```

## Voila view

```{r}
# Voila View violin plots:
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% arrange(-abs(deltapsi)) %>% print(n=Inf)
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "SYNJ1") %>% glimpse # lsv_junction_id   "ENSG00000159082:t:32673340-32673531:j:32673531-32678645", "ENSG00000159082:t:32673340-32673531:j:32673531-32676332"
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "AGAP1") %>% glimpse # lsv_junction_id   ENSG00000157985:t:235968462-235968623:j:235908906-235968462", "ENSG00000157985:t:235968462-235968623:j:235930923-235968462"
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "LINC00665") %>% glimpse # lsv_junction_id "ENSG00000232677:t:36321993-36322030:j:36322030-36330253", "ENSG00000232677:t:36321993-36322030:j:36322030-36322130"


als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% arrange(-abs(deltapsi)) %>% print(n=Inf) # HLA-B
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "HLA-B") %>% glimpse # lsv_junction_id   "ENSG00000234745:s:31356688-31356957:j:31356442-31356688", "ENSG00000234745:s:31356688-31356957:j:31271348-31356688"

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.DDR.repair_gene_set) # MTA1, POLM, PMS1 HUWE1, HDAC1, ZWIM7, METTL22
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "METTL22") %>% glimpse # lsv_junction_id  "ENSG00000067365:s:8644437-8645044:j:8644725-8645908", "ENSG00000067365:s:8644437-8645044:j:8644725-8646108"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "MTA1") %>% glimpse # lsv_junction_id  "ENSG00000182979:s:105466707-105467192:j:105467193-105468108"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLM") %>% glimpse # lsv_junction_id  "ENSG00000122678:s:44073625-44073708:j:44073430-44073625", "ENSG00000122678:s:44073625-44073708:j:44073377-44073625"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "PMS1") %>% glimpse # lsv_junction_id "ENSG00000064933:t:189867799-189867929:j:189864360-189867798"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "HUWE1") %>% glimpse # lsv_junction_id "ENSG00000086758:s:53627410-53627515:j:53626034-53627410", "ENSG00000086758:s:53627410-53627515:j:53625258-53627410"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "HDAC1") %>% glimpse # lsv_junction_id  "ENSG00000116478:s:32327536-32327677:j:32327677-32329068", "ENSG00000116478:s:32327536-32327677:j:32327678-32329067"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "ZSWIM7") %>% glimpse # lsv_junction_id  "ENSG00000214941:t:15976560-15977092:j:15977092-15978044", "ENSG00000214941:t:15976560-15977092:j:15977092-15977579"

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% arrange(-abs(deltapsi)) %>% print(n=Inf) # YTHDC2, THOC1, PRR3, STAU2, PTBP3, SREK1, POLDIP3
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLDIP3") %>% glimpse # lsv_junction_id  "ENSG00000100227:s:42602768-42603160:j:42599793-42602770"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "YTHDC2") %>% glimpse # lsv_junction_id"ENSG00000047188:t:113532879-113533045:j:113515362-113532879", "ENSG00000047188:t:113532879-113533045:j:113526785-113532879"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "THOC1") %>% glimpse # lsv_junction_id ENSG00000079134:s:263493-264092:j:260304-264026"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "PRR3") %>% glimpse # lsv_junction_id"ENSG00000204576:s:30558150-30558436:j:30558212-30561328", "ENSG00000204576:s:30558150-30558436:j:30558212-30561834"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "STAU2") %>% glimpse # lsv_junction_id"ENSG00000040341:s:73549604-73552319:j:73422702-73552012"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "PTBP3") %>% glimpse # lsv_junction_id "ENSG00000119314:s:112297832-112297916:j:112276013-112297832"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "SREK1") %>% glimpse # lsv_junction_id  "ENSG00000153914:s:66174624-66175041:j:66175041-66177517"


### import violin from voila view
# edit colour and labels in illustrator > export as png
polyA_majiq_violins_merged.png = magick::image_read(here(proj_path,"splicing/majiq/figures/polyA_majiq_violins_nro1.png"))
polyA_majiq_violins_merged.gg = ggdraw() + draw_image(polyA_majiq_violins_merged.png)
polyA_majiq_violins_merged.gg
# BID_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/BID_majiq_violin.png"))
# BID_majiq_violins.gg = ggdraw() + draw_image(BID_majiq_violins.png)
# BID_majiq_violins.gg
# HTT_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/HTT_majiq_violin.png"))
# HTT_majiq_violins.gg = ggdraw() + draw_image(HTT_majiq_violins.png)
# HTT_majiq_violins.gg
# GADD45B_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/GADD45B_majiq_violin.png"))
# GADD45B_majiq_violins.gg = ggdraw() + draw_image(GADD45B_majiq_violins.png)
# GADD45B_majiq_violins.gg
# majiq_violins.gg = GADD45B_majiq_violins.gg + CAMTA1_majiq_violins.gg  + BID_majiq_violins.gg + HTT_majiq_violins.gg
# majiq_violins.gg



```


## Postmortem splicing

```{r}
# nygc_path = here("/camp/lab/patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
# nygc.als_ctrl.voila = read_voila_het(here(nygc_path,"splicing/majiq/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
# nygc.pan_als.voila_modulizer <- read_voila_modulizer(here(nygc_path,"splicing/majiq/modulize/pan_als"), skip_rows = 9)
# save(nygc.als_ctrl.voila, nygc.pan_als.voila_modulizer, file = here(nygc_path,"splicing/majiq/voila-tsv/als_vs_ctrl.RData"))
load(here(nygc_path,"splicing/majiq/voila-tsv/als_vs_ctrl.RData"))

nygc.als_ctrl.voila %>% filter(gene_name %in% "SFPQ")  %>% glimpse
nygc.als_ctrl.voila %>% filter(gene_name %in% "UNC13B")  %>% glimpse
# nygc.als_ctrl.voila %>% filter(gene_name %in% "SFPQ") %>% write_csv(here(nygc_path, "splicing/majiq/voila-tsv/sfpq_events.als_vs_ctrl.tsv"))

nygc.als_ctrl.voila
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 842
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 445
nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 250
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.DDR.repair_gene_set) %>% print(n=Inf) # 22
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% distinct(gene_name)  %>% print(n=Inf) #47
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) # TTC7A, DNM1
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) # CAMTA1, GRIN1, NEK1, ATXN1, DNM1, VEGF
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% sporadic_ALS_genes) #0
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_GWAS) # CAMTA1, NEK1
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_EWAS) # TTC7A, DNM1
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_modifiers) # VEGFA
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% TDP43_spliced_genes) %>% print(n=Inf)  # KALRN, KMT2C, PRUNE2, EPB41L1, APLP2, PLEKHA1, EIF4Ge, NCAM1, RAPGEF4, PTPRD, KIF3A

nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # CNOT3, EI24
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`DNA repair`) #  APTX, CENPX, RIF1, RPAIN, KAT7
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, -log10(tnom) > 4) # TVP23C-CDRT4, SREK1
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLDIP3") # 0

# TDP-43 neuronal nuclei depleted overlap
neuronal_nuclei_tdp43.liu.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/neuronal-nuclei-tdp43-liu-2019/splicing/majiq/voila-tsv/tdp43neg_vs_tdp43pos.tsv"), grp1 = "tdp43neg", grp2 = "tdp43pos") %>% mutate(mutation = "tdp43_neuronal_nuclei")
neuronal_nuclei_tdp43.liu.voila.sig = neuronal_nuclei_tdp43.liu.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9)# %>% print(n=50)
neuronal_nuclei_tdp43.liu.voila.sig %>% filter(gene_name %in% pull(filter(nygc.als_ctrl.voila, changing_dpsi0.1 == TRUE),gene_name)) %>% distinct(gene_name)

# TDP43 knockdown overlap
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
tdp43kd.als_vs_ctrl.brown.voila.sig = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9)# %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.sig %>% filter(gene_name %in% pull(filter(nygc.als_ctrl.voila, changing_dpsi0.1 == TRUE),gene_name)) %>% distinct(gene_name)

# Volcano
majiq.als_vs_ctrl.TDP43.junction_labels <- nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% TDP43_spliced_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.ALS.junction_labels <- nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.DDR.junction_labels <- nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% p53.DDR.repair_gene_set) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
# majiq.als_vs_ctrl.extra_labels <- nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("METTL22","LINC00665")) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, -log10(tnom)>3, -log10(tnom)<4) 

majiq.nygc.als_vs_ctrl.volcano <- majiq_het_volcano_plot(nygc.als_ctrl.voila, "Alternative splicing: Post-mortem ALS", numerator = "ALS", denomenator = "CTRL", subtitle = "57 CTRL vs 214 ALS", ymax = 11, xmax = 0.5, dpi = 300, junction_labels = c(majiq.als_vs_ctrl.ALS.junction_labels, majiq.als_vs_ctrl.DDR.junction_labels, majiq.als_vs_ctrl.TDP43.junction_labels))
majiq.nygc.als_vs_ctrl.volcano

# Fisher exact overlap test
nygc.als_ctrl.voila.p53 = nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% unique(c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, p53.DDR.repair_gene_set))) %>% pull(gene_name)
nygc.als_ctrl.voila.p53 # "APTX"   "CENPX"  "RIF1"   "RIF1"   "RPAIN"  "RPAIN"  "RPAIN"  "KAT7"   "KAT7"   "SMC6"   "CNOT3"  "CNOT3"  "PAXX"   "PAXX"   "UVSSA"  "UVSSA"  "EI24"   "EI24"   "CEP164" "CEP164" "TCEA1"  "INO80E"
newGeneOverlap(nygc.als_ctrl.voila.p53, unique(c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, p53.DDR.repair_gene_set)), genome.size=nrow(distinct(nygc.als_ctrl.voila, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=13
# listB size=819
# Intersection size=13
# Overlapping p-value=6.8e-16
# Jaccard Index=0.0

nygc.als_ctrl.voila.RBP = nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% pull(gene_name)
newGeneOverlap(nygc.als_ctrl.voila.RBP, RNA_BINDING_PROTEINS, genome.size=nrow(distinct(nygc.als_ctrl.voila, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=29
# listB size=1542
# Intersection size=29
# Overlapping p-value=1.3e-26
# Jaccard Index=0.0

# GO of splicing events
nygc.als_ctrl.voilagost = nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
nygc.als_ctrl.voilagost_filt <- nygc.als_ctrl.voilagost$result %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(nygc.als_ctrl.voilagost_filt$query) # 165
paste('"',unique(nygc.als_ctrl.voilagost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
"nervous system development",
"vesicle-mediated transport",
"cytoskeleton organization",
"phosphatidylinositol binding",
"synapse",
"actin cytoskeleton",
"axon",
"GTPase regulator activity",
"protein binding"
  )
nygc.als_ctrl.voilagost_filt.terms <- nygc.als_ctrl.voilagost_filt %>% filter(term_name %in% majiq_als_go_list)
# nygc.als_ctrl.voilagost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", nygc.als_ctrl.voilagost_filt.terms$term_name) 
nygc.als_ctrl.voilagost_filt.terms.curated <- curated_profiles(nygc.als_ctrl.voilagost_filt.terms, cols = c("forestgreen")) + labs(title = "Postmortem splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
nygc.als_ctrl.voilagost_filt.terms.curated


# Modulize
nygc.als_ctrl.voila.modulizer.count = nygc.pan_als.voila_modulizer %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>%
  count(modulizer_simplified) %>% 
  mutate(modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>%  
  add_tally(n) %>% mutate(pct = n/nn *100, pct.label = case_when(pct > 14 ~ paste0(as.character(n)," (", as.character(round(pct,1)),"%)"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) %>% ungroup()
nygc.als_ctrl.voila.modulizer.count %>% print(n=Inf)

nygc.als_ctrl.voila.modulizer.barplot = ggbarplot(mutate(nygc.als_ctrl.voila.modulizer.count,mutation=""), x="mutation", y="pct", fill="modulizer_simplified", position = position_fill(), xlab = "",  label = nygc.als_ctrl.voila.modulizer.count$pct.label, lab.size = 3, lab.vjust = 0.5, lab.hjust = 1.5)  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8), strip.text.y = element_text(angle = 0)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 2)) +
  # facet_grid(mutation ~ .,scales="free_y", labeller = "label_parsed") +
  coord_flip() + ylab(expression(Splicing~type~proportions)) + scale_x_discrete(labels = scales::parse_format()) 
nygc.als_ctrl.voila.modulizer.barplot


nygc.modulizer_simplfied.sig.summary = nygc.pan_als.voila_modulizer %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05, modulizer_simplified != "orphan_junction") %>% count(modulizer_simplified) %>% mutate(modulizer_simplified = gsub("_"," ",modulizer_simplified), modulizer_accronym = case_when(modulizer_simplified == "alt3and5prime" ~ "A3&A5SS", modulizer_simplified == "alt5prime" ~ "A5SS", modulizer_simplified == "alt3prime" ~ "A3SS", modulizer_simplified == "alternate first exon" ~ "AFE", modulizer_simplified == "alternate last exon" ~ "ALE", modulizer_simplified == "multi exon spanning" ~ "multi-exon", modulizer_simplified == "intron retention" ~ "retained intron", modulizer_simplified == "mutually exclusive exon" ~ "MXE", TRUE ~ modulizer_simplified), modulizer_accronym = fct_reorder(modulizer_accronym, n, .desc = TRUE)) %>% arrange(desc(modulizer_accronym)) %>% 
  mutate(pct = n/sum(n) *100, pct.label = case_when(pct > 8 ~ paste(as.character(n),"\n", as.character(round(pct,1)),"%"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) #%>% print
nygc.modulizer_simplfied.sig.summary %>% print
nygc.pan_als.voila_modulizer.sig.pie = ggplot(nygc.modulizer_simplfied.sig.summary, aes(x = "", y = pct, fill = fct_reorder(modulizer_accronym, n, .desc = TRUE))) + geom_bar(width = 1, stat = "identity", color="white") + coord_polar("y", start=0) + 
  geom_text(aes(x = 1.2, y = ypos, label = pct.label), color = "black", size=2.5) + scale_fill_brewer(palette="Set1") + 
  theme_oz() + theme_void() + theme(legend.title=element_blank(), legend.position = "top", legend.text=element_text(size=7), legend.spacing.y = unit(1, 'mm'), legend.spacing.x = unit(1, 'mm'), legend.key.height = unit(0.4, 'cm'), legend.key.width = unit(0.3, 'cm')) + guides(fill=guide_legend(nrow=3))
nygc.pan_als.voila_modulizer.sig.pie



# # Cryptic Exons == denovo & cassette_exon
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 178
178/842
nygc.pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer) # 21 cryptic exons
21/178
nygc.pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% filter(modulizer == "cassette_exon", gene_name %in% TDP43_spliced_genes)
nygc.pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% filter(modulizer == "cassette_exon", gene_name %in% tdp43kd.als_vs_ctrl.brown.voila.sig$gene_name) # EP400
nygc.pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% filter(modulizer == "cassette_exon", gene_name %in% neuronal_nuclei_tdp43.liu.voila.sig$gene_name) # BMP2K, KMT2C



nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, tnom < 0.05, grepl("i$",lsv_type)) # 462
462/842 # 55%
nygc.pan_als.voila_modulizer.cryptic = nygc.pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) %>% pull(gene_name)  # "RELCH"         "HOXC4"         "RBM26"         "SLC35B3"       "TENM3"         "TPTEP2-CSNK1E" "ZSCAN29"   
50/264

# Overlap postmortem and ipsmn
nygc.als_ctrl.voila_splicing_events_genes =  nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% filter(gene_name %in% nygc.als_ctrl.voila_splicing_events_genes) #20 
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% filter(gene_name %in% nygc.als_ctrl.voila_splicing_events_genes) %>% distinct(gene_name) #12

# Fisher exact overlap test
nygc.als_ctrl.splicing_events_genes = nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
ipsn.als_ctrl.splicing_events_genes = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
newGeneOverlap(nygc.als_ctrl.splicing_events_genes, ipsn.als_ctrl.splicing_events_genes, genome.size=nrow(distinct(gtf,gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=445
# listB size=161
# Intersection size=12
# Overlapping p-value=3.3e-09
# Jaccard Index=0.0
nygc.als_ctrl.splicing_events_genes[nygc.als_ctrl.splicing_events_genes %in% ipsn.als_ctrl.splicing_events_genes] #"PLEKHA6" "PLEKHA6" "PLEKHA6" "PLEKHA6" "DNM2"    "DNM2"    "DNM2"    "SUZ12P1" "SUZ12P1" "MGRN1"   "TPD52L2" "TPD52L2" "TPD52L2" "ILKAP"   "ILKAP"   "CYTH1"   "CYTH1"   "SYNJ1"   "SYNJ1"   "KIF1B"   "AGAP1"   "NCOR2"   "PAN3"    "PAN3" 

nygc.ipmn.nygc.als_ctrl.splicing_events_genes = nygc.als_ctrl.splicing_events_genes[nygc.als_ctrl.splicing_events_genes %in% ipsn.als_ctrl.splicing_events_genes]
nygc.ipmn.nygc.als_ctrl.splicing_events_genes[nygc.ipmn.nygc.als_ctrl.splicing_events_genes %in% TDP43_spliced_genes] #0


# Voila view
nygc_majiq_violins_merged.png = magick::image_read(here(proj_path,"splicing/majiq/figures/nygc_majiq_violins_merged.png"))
nygc_majiq_violins_merged.gg = ggdraw() + draw_image(nygc_majiq_violins_merged.png)
nygc_majiq_violins_merged.gg

```




## Merge

```{r}
pan_als_ipsn_splicing_fig_merged = plot_grid(
  majiq_schematic.gg,
  plot_grid(majiq.als_vs_ctrl.volcano, als_ctrl.voila.gost_filt.terms.curated, pan_als.voila_modulizer.sig.pie, ncol = 3, labels = letters[2:4], rel_widths = c(1,0.6,0.6)),
  polyA_majiq_violins_merged.gg,
  plot_grid(majiq.nygc.als_vs_ctrl.volcano, nygc.als_ctrl.voilagost_filt.terms.curated, nygc.pan_als.voila_modulizer.sig.pie, ncol = 3, rel_widths = c(1,0.6,0.6), labels = letters[6:8]),
  nygc_majiq_violins_merged.gg, #NYGC voila view
  nrow = 5, rel_heights = c(0.5,1,0.6,1,0.6), labels = c("a","","e","","i"))
# pan_als_ipsn_splicing_fig_merged
ggsave(pan_als_ipsn_splicing_fig_merged, filename = here(proj_path,"splicing/majiq/figures/pan_als_ipsn_splicing_fig_merged.png"), height = 9, width = 9)
ggsave(pan_als_ipsn_splicing_fig_merged, filename = here(proj_path,"splicing/majiq/figures/pan_als_ipsn_splicing_fig_merged.pdf"), height = 9, width = 9)

```


# Table S9 MAJIQ iPSMN datasets

```{r}
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
neuronal_nuclei_tdp43.liu.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/neuronal-nuclei-tdp43-liu-2019/splicing/majiq/voila-tsv/tdp43neg_vs_tdp43pos.tsv"), grp1 = "tdp43neg", grp2 = "tdp43pos") %>% mutate(mutation = "tdp43_neuronal_nuclei")
tdp43neg_majiq_genes.sig = neuronal_nuclei_tdp43.liu.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
tdp43kd_brown_majiq_genes.sig = tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)

# als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
als_ctrl.voila.sig = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% select(-contains("percentile"),-contains("changing"),-ucsc_lsv_link) %>% arrange(-abs(deltapsi)) %>%
  mutate(neuronal_nuclei_tdp43 = case_when(gene_name %in% tdp43neg_majiq_genes.sig ~ "overlap", TRUE ~ ""), tdp43kd = case_when(gene_name %in% tdp43kd_brown_majiq_genes.sig ~ "overlap", TRUE ~ "")) %>%
  select(gene_name, lsv_junction_id, deltapsi, pvalue=tnom, neuronal_nuclei_tdp43, tdp43kd, everything(), -ttest, -wilcoxon, -lsv_type,-tnom_quantile, -ttest_quantile, -wilcoxon_quantile, -tnom_score)
als_ctrl.voila.sig %>% glimpse
write_csv(als_ctrl.voila.sig, here(proj_path,"splicing/majiq/figures/Table.majiq_splicing_panALS.sig.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473")
write_sheet(als_ctrl.voila.sig, ss = "https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473", sheet = "S9 Splicing iPSMN")

```


# Table S10 MAJIQ genetic backgrounds

```{r}
# join at lsv level
ipsc_mn_mutants_join_datasets.majiq.lsv_level = select(c9orf72_vs_ctrl.voila, gene_name, lsv_junction_id, dpsi.c9orf72 = deltapsi, pval.c9orf72 = tnom, changing.c9orf72 = changing_dpsi0.1) %>%
  full_join(select(fus_vs_ctrl.voila, gene_name, lsv_junction_id, dpsi.fus = deltapsi, pval.fus = tnom, changing.fus = changing_dpsi0.1)) %>%
  full_join(select(tardbp_vs_ctrl.voila, gene_name, lsv_junction_id, dpsi.tardbp = deltapsi, pval.tardbp = tnom, changing.tardbp = changing_dpsi0.1)) %>%
  full_join(select(sod1_vs_ctrl.voila, gene_name, lsv_junction_id, dpsi.sod1 = deltapsi, pval.sod1 = tnom, changing.sod1 = changing_dpsi0.1)) %>%
  replace_na(list(changing.c9orf72 = FALSE, changing.fus = FALSE, changing.tardbp = FALSE, changing.sod1 = FALSE))
ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt = ipsc_mn_mutants_join_datasets.majiq.lsv_level %>% 
  filter((changing.c9orf72 == TRUE & changing.sod1 == TRUE & dpsi.c9orf72 > 0 & dpsi.sod1 > 0) | (changing.c9orf72 == TRUE & changing.fus == TRUE & dpsi.c9orf72 > 0 & dpsi.fus > 0) | 
           (changing.c9orf72 == TRUE & changing.tardbp == TRUE & dpsi.c9orf72 > 0 & dpsi.tardbp > 0) | 
         (changing.tardbp == TRUE & changing.sod1 == TRUE & dpsi.tardbp > 0 & dpsi.sod1 > 0) | (changing.fus == TRUE & changing.tardbp == TRUE & dpsi.fus > 0 & dpsi.tardbp > 0) | (changing.fus == TRUE & changing.sod1 == TRUE & dpsi.fus > 0 & dpsi.sod1 > 0) | 
           (changing.c9orf72 == TRUE & changing.sod1 == TRUE & dpsi.c9orf72 < 0 & dpsi.sod1 < 0) | (changing.c9orf72 == TRUE & changing.fus == TRUE & dpsi.c9orf72 < 0 & dpsi.fus < 0) | 
           (changing.c9orf72 == TRUE & changing.tardbp == TRUE & dpsi.c9orf72 < 0 & dpsi.tardbp < 0) | 
         (changing.tardbp == TRUE & changing.sod1 == TRUE & dpsi.tardbp < 0 & dpsi.sod1 < 0) | (changing.fus == TRUE & changing.tardbp == TRUE & dpsi.fus < 0 & dpsi.tardbp < 0) | (changing.fus == TRUE & changing.sod1 == TRUE & dpsi.fus < 0 & dpsi.sod1 < 0)) 
ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt
write_csv(ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt, here(proj_path,"splicing/majiq/figures/TableS11.majiq_splicing_overlap_mutations.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473")
write_sheet(ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt, ss = "https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473", sheet = "S8 Splicing subgroups")

ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt %>% filter(gene_name %in% p53.signalling.pathway)
ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt %>% filter(gene_name %in% p53.DDR.repair_gene_set)
ipsc_mn_mutants_join_datasets.majiq.lsv_level.filt %>% filter(gene_name %in% RNA_BINDING_PROTEINS)


```

# Table S11 MAJIQ postmortem NYGC

```{r}
load(here(nygc_path,"splicing/majiq/voila-tsv/als_vs_ctrl.RData"))

nygc.als_ctrl.voila %>% filter(gene_name %in% "SFPQ")  %>% glimpse
nygc.als_ctrl.voila %>% filter(gene_name %in% "UNC13B")  %>% glimpse
# nygc.als_ctrl.voila %>% filter(gene_name %in% "SFPQ") %>% write_csv(here(nygc_path, "splicing/majiq/voila-tsv/sfpq_events.als_vs_ctrl.tsv"))

nygc.als_ctrl.voila

# nygc.als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
nygc.als_ctrl.voila.sig = nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% select(-contains("percentile"),-contains("changing"),-ucsc_lsv_link) %>% arrange(-abs(deltapsi)) %>%
  mutate(neuronal_nuclei_tdp43 = case_when(gene_name %in% tdp43neg_majiq_genes.sig ~ "overlap", TRUE ~ ""), tdp43kd = case_when(gene_name %in% tdp43kd_brown_majiq_genes.sig ~ "overlap", TRUE ~ ""), ipsmn_als = case_when(gene_name %in% als_ctrl.voila.sig$gene_name ~ "overlap", TRUE ~ "")) %>%
  select(gene_name, lsv_junction_id, deltapsi, pvalue=tnom, neuronal_nuclei_tdp43, tdp43kd, ipsmn_als, everything(), -ttest, -wilcoxon, -lsv_type,-tnom_quantile, -ttest_quantile, -wilcoxon_quantile, -tnom_score)
nygc.als_ctrl.voila.sig %>% glimpse
write_csv(nygc.als_ctrl.voila.sig, here(proj_path,"splicing/majiq/figures/Table.nygc.majiq_splicing_nALS.sig.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473")
write_sheet(nygc.als_ctrl.voila.sig, ss = "https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473", sheet = "S11 Splicing PM")



```

# Fig S13 TDP43 knockdown POLDIP event

```{r}
als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
# tdp43_positive.als_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/tdp43_positive.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "tdp43_positive")
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
neuronal_nuclei_tdp43.liu.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/neuronal-nuclei-tdp43-liu-2019/splicing/majiq/voila-tsv/tdp43neg_vs_tdp43pos.tsv"), grp1 = "tdp43neg", grp2 = "tdp43pos") %>% mutate(mutation = "tdp43_neuronal_nuclei")


# Liu et al TDP43 pos neg neuronal nuclei
neuronal_nuclei_tdp43.liu.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) %>% print(n=50)
neuronal_nuclei_tdp43.liu.voila.labels = neuronal_nuclei_tdp43.liu.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
neuronal_nuclei_tdp43.liu.voila.volcano = majiq_deltapsi_volcano_plot(neuronal_nuclei_tdp43.liu.voila, junction_labels = neuronal_nuclei_tdp43.liu.voila.labels, ymax = 4.1, xmax = 0.7, xpos = 0.5, numerator = "TDP-43 neg", denomenator = "TDP-43 pos", dpi = 300)
neuronal_nuclei_tdp43.liu.voila.volcano

# Scatter correlation TDP-43 knockdown with pan-ALS
als_tdp43neg_majiq.join = select(als_ctrl.voila, gene_name, lsv_id, lsv_junction_id, lsv_type, junctions_coords, exons_coords, ir_coords, ALS.deltapsi = deltapsi, ALS.changing = changing_dpsi0.1) %>% 
  left_join(select(neuronal_nuclei_tdp43.liu.voila, gene_name, lsv_id, lsv_junction_id, lsv_type, junctions_coords, exons_coords, ir_coords, tdp43neg.deltapsi = deltapsi, tdp43neg.changing = changing_dpsi0.1))
als_tdp43neg_majiq.join %>% glimpse # 137,586
als_tdp43neg_majiq.join %>% filter(ALS.changing == TRUE, tdp43neg.changing == TRUE) # 2 POLDIP3
als_tdp43neg_majiq.join %>% filter(ALS.changing == TRUE, tdp43neg.changing == TRUE) %>% arrange(-(abs(ALS.deltapsi) + abs(tdp43neg.deltapsi))) %>% print(n=Inf) # POLDIP3
als_tdp43neg_majiq.join %>% filter(ALS.changing == TRUE, abs(tdp43neg.deltapsi) > 0.1, ALS.deltapsi > 0 & tdp43neg.deltapsi > 0 | ALS.deltapsi < 0 & tdp43neg.deltapsi < 0) %>% print(n=Inf) # POLDIP3, ROBO2, PTBP3

als_tdp43neg_majiq.list = list("pan ALS" = als_ctrl.voila, "TDP-43 KD" = neuronal_nuclei_tdp43.liu.voila)
neuronal_nuclei_tdp43.liu.voila %>% filter(probability_changing < 0.95, gene_name %in% ALS_genes) 

# scatter correlation
als_tdp43neg_majiq.join.labels = als_tdp43neg_majiq.join %>% filter(ALS.changing == TRUE, tdp43neg.changing == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # "NYNRIN" "SNX2"  
als_tdp43neg_majiq.join.scatter =dpsi_majiq_scatter(model_list1 = als_tdp43neg_majiq.list, model_list2 = als_tdp43neg_majiq.list, mutation1 = "pan ALS", mutation2 = "TDP-43 neuronal nuclei depletion", gene_labels = als_tdp43neg_majiq.join.labels) + xlim(-0.3,0.3) + ylim(-0.7,0.7)
als_tdp43neg_majiq.join.scatter

# Overlap splicing events in TDP-43 knockdown with pan-ALS
als_tdp43neg_majiq.join %>% filter(ALS.changing == TRUE, tdp43neg.changing == TRUE) %>% print(n=Inf)# POLDIP3
als_tdp43neg_majiq.join.labels = als_tdp43neg_majiq.join %>% filter(ALS.changing == TRUE, tdp43neg.changing == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # "NYNRIN" "SNX2"  
als_tdp43neg_majiq.join.labels #POLDIP3
als_tdp43neg_majiq.join.labels[als_tdp43neg_majiq.join.labels %in% RNA_BINDING_PROTEINS]

# Overlap at gene level:
pan_als_majiq_genes.sig = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
tdp43neg_majiq_genes.sig = neuronal_nuclei_tdp43.liu.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
pan_als_majiq_genes.sig[pan_als_majiq_genes.sig %in% tdp43neg_majiq_genes.sig] # 17
 # "ZNF846"    "ROBO2"     "POLDIP3"   "NRXN2"     "PTBP3"     "DPY19L2P2" "PPP6R3"    "CAMK2B"    "CEP290"    "CERS5"     "PRKRIP1"   "OSBPL1A"  


# Volcano TDP-43 knockdown
tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.labels = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
tdp43kd.als_vs_ctrl.brown.voila.volcano = majiq_deltapsi_volcano_plot(tdp43kd.als_vs_ctrl.brown.voila, junction_labels = tdp43kd.als_vs_ctrl.brown.voila.labels, ymax = 4.1, xmax = 0.7, xpos = 0.5, numerator = "TDP-43 KD", denomenator = "Control", dpi = 300)
tdp43kd.als_vs_ctrl.brown.voila.volcano

# Scatter correlation TDP-43 knockdown with pan-ALS
als_tdp43kd_majiq.join = select(als_ctrl.voila, gene_name, lsv_id, lsv_junction_id, lsv_type, junctions_coords, exons_coords, ir_coords, ALS.deltapsi = deltapsi, ALS.changing = changing_dpsi0.1) %>% 
  left_join(select(tdp43kd.als_vs_ctrl.brown.voila, gene_name, lsv_id, lsv_junction_id, lsv_type, junctions_coords, exons_coords, ir_coords, TDP43kd.deltapsi = deltapsi, TDP43kd.changing = changing_dpsi0.1))
als_tdp43kd_majiq.join %>% glimpse # 137,586
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) # POLDIP3
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% arrange(-(abs(ALS.deltapsi) + abs(TDP43kd.deltapsi))) %>% print(n=Inf)
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, abs(TDP43kd.deltapsi) > 0.1, ALS.deltapsi > 0 & TDP43kd.deltapsi > 0 | ALS.deltapsi < 0 & TDP43kd.deltapsi < 0) %>% print(n=Inf) # GRIA4, DONSON

als_tdp43kd_majiq.list = list("pan ALS" = als_ctrl.voila, "TDP-43 KD" = tdp43kd.als_vs_ctrl.brown.voila)
tdp43kd.als_vs_ctrl.brown.voila %>% filter(probability_changing < 0.95, gene_name %in% ALS_genes)

# scatter correlation
als_tdp43kd_majiq.join.labels = als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # "NYNRIN" "SNX2"  
als_tdp43kd_majiq.join.scatter =dpsi_majiq_scatter(model_list1 = als_tdp43kd_majiq.list, model_list2 = als_tdp43kd_majiq.list, mutation1 = "pan ALS", mutation2 = "TDP-43 KD", gene_labels = als_tdp43kd_majiq.join.labels) + xlim(-0.3,0.3) + ylim(-0.7,0.7)
als_tdp43kd_majiq.join.scatter

# Overlap splicing events in TDP-43 knockdown with pan-ALS
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% print(n=Inf)# NYNRIN
als_tdp43kd_majiq.join.labels = als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # "NYNRIN" "SNX2"  
als_tdp43kd_majiq.join.labels #POLDIP3
als_tdp43kd_majiq.join.labels[als_tdp43kd_majiq.join.labels %in% RNA_BINDING_PROTEINS]

# Overlap at gene level:
pan_als_majiq_genes.sig = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
tdp43kd_brown_majiq_genes.sig = tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
pan_als_majiq_genes.sig[pan_als_majiq_genes.sig %in% tdp43kd_brown_majiq_genes.sig] # "POLDIP3" "CAMK2B"  "CEP290"  "HERC2P3"

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "CAMK2B") %>% glimpse #"ENSG00000058404:s:44234390-44234461:j:44232866-44234390", "ENSG00000058404:s:44234390-44234461:j:44231054-44234390"
tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "CAMK2B") %>% glimpse #"ENSG00000058404:t:44220826-44220953:j:44220901-44231006", "ENSG00000058404:t:44254542-44254607:j:44254607-44258872", "ENSG00000058404:t:44254542-44254607:j:44254607-44258490"


# where in POLDIP3 is the cryptic exon?
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLDIP3") %>% glimpse # ENSG00000100227:s:42602768-42603160:j:42602056-42602770", "ENSG00000100227:s:42602768-42603160:j:42599793-42602770"
tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLDIP3") %>% glimpse # "ENSG00000100227:t:42599698-42599793:j:42599793-42602770", "ENSG00000100227:t:42599698-42599793:j:42599793-42601970", "ENSG00000100227:s:42602768-42603160:j:42602056-42602770"
neuronal_nuclei_tdp43.liu.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLDIP3") %>% glimpse # "ENSG00000100227:s:42602768-42603160:j:42602056-42602770", "ENSG00000100227:s:42602768-42603160:j:42599793-42602770", "ENSG00000100227:t:42599698-42599793:j:42599793-42602770", "ENSG00000100227:t:42599698-42599793:j:42599793-42601970"


```


# Fig S16 Genetic groups splicing

## Volcanos


```{r}
# # sporadic_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/sporadic.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "Sporadic")
# c9orf72_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/c9orf72.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "C9orf72")
# fus_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/fus.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "FUS")
# tardbp_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/tardbp.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "TARDBP")
# sod1_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/sod1.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "SOD1")
# # sporadic.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq/modulize/sporadic"), skip_rows = 9) # still running 17/5
# c9orf72.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/c9orf72"), skip_rows = 9)
# fus.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/fus"), skip_rows = 9)
# tardbp.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/tardbp"), skip_rows = 9)
# sod1.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/sod1"), skip_rows = 9)
# save(c9orf72_vs_ctrl.voila, fus_vs_ctrl.voila, tardbp_vs_ctrl.voila, c9orf72.voila_modulizer, fus.voila_modulizer, tardbp.voila_modulizer, sod1.voila_modulizer,
#      file = here(proj_path,"splicing/majiq_polyA/voila-tsv/genetic_backgrounds.RData"))
load(file = here(proj_path,"splicing/majiq_polyA/voila-tsv/genetic_backgrounds.RData"))

#MAJIQ Volcanos
mutations_bind.als_vs_ctrl.voila = bind_rows(c9orf72_vs_ctrl.voila, fus_vs_ctrl.voila, tardbp_vs_ctrl.voila, sod1_vs_ctrl.voila)#sporadic_vs_ctrl.voila, vcp_vs_ctrl.voila)
mutations_bind.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% count(mutation)
# mutation     n
#   <chr>    <int>
# 1 C9orf72    429
# 2 FUS       1099
# 3 SOD1       256
# 4 TARDBP    1435

tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`))
fus_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`))
sod1_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`))
c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`))

### Volcanos
filter(ipsc_mn_tardbp_datasets.metadata, library_type == "poly(A)") %>% count(condition) 
tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`))
tardbp_vs_ctrl.voila.ALS.labels <- tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
tardbp_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(tardbp_vs_ctrl.voila, numerator = "TARDBP", denomenator = "CTRL", subtitle = "6 Controls vs 9 TARDBP", ymax = 4, xmax = 0.6, dpi = 300, junction_labels = c(tardbp_vs_ctrl.voila.ALS.labels))
# tardbp_vs_ctrl.voila.volcano

filter(ipsc_mn_fus_datasets.metadata, library_type == "poly(A)") %>% count(condition) 
fus_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`)) 
fus_vs_ctrl.voila.ALS.labels <- fus_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
fus_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(fus_vs_ctrl.voila, numerator = "FUS", denomenator = "CTRL", subtitle = "8 Controls vs 9 FUS", ymax = 4, xmax = 0.5, dpi = 300, junction_labels = c(fus_vs_ctrl.voila.ALS.labels))
# fus_vs_ctrl.voila.volcano

filter(ipsc_mn_c9orf72_datasets.metadata, library_type == "poly(A)") %>% count(condition) 
c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`)) 
c9orf72_vs_ctrl.voila.ALS.labels <- c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
c9orf72_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(c9orf72_vs_ctrl.voila, numerator = "C9orf72", denomenator = "CTRL", subtitle = "27 Controls vs 23 C9orf72", ymax = 5, xmax = 0.3, dpi = 300, junction_labels = c(c9orf72_vs_ctrl.voila.ALS.labels))
# c9orf72_vs_ctrl.voila.volcano

filter(ipsc_mn_sod1_datasets.metadata, library_type == "poly(A)") %>% count(condition) 
sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`)) 
sod1_vs_ctrl.voila.ALS.labels <- sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`, gprofiler_database$`DNA repair`))  %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
sod1_vs_ctrl.voila.volcano <- majiq_het_volcano_plot(sod1_vs_ctrl.voila, numerator = "SOD1", denomenator = "CTRL", subtitle = "5 Controls vs 4 SOD1", ymax = 3, xmax = 0.3, dpi = 300, junction_labels = c(sod1_vs_ctrl.voila.ALS.labels))
# sod1_vs_ctrl.voila.volcano


# majiq.volcano_multiplot <- tardbp_vs_ctrl.voila.volcano + fus_vs_ctrl.voila.volcano + c9orf72_vs_ctrl.voila.volcano + sod1_vs_ctrl.voila.volcano + sporadic_vs_ctrl.voila.volcano +
#   plot_layout(ncol = 3, nrow = 2)
# majiq.volcano_multiplot
# ggsave(majiq.volcano_multiplot, filename = here(proj_path,"expression/deseq2/majiq.volcano_multiplot.png"), height = 7, width = 10)

```

## GO terms

```{r}
c9orf72_vs_ctrl.voila.gost = c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
c9orf72_vs_ctrl.voila.gost_filt <- c9orf72_vs_ctrl.voila.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC")) %>% arrange( -log10(p_value) )
table(c9orf72_vs_ctrl.voila.gost_filt$query) # 
paste('"',unique(c9orf72_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_c9orf72_go_list = c(
"neuron projection morphogenesis",
"protein binding",
"Golgi organization",
"cytoplasm")
c9orf72_vs_ctrl.voila.gost_filt.terms <- c9orf72_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_c9orf72_go_list)
# c9orf72_vs_ctrl.voila.gost_filt.terms$term_name = gsub(" molecule binding| organization| organizing", "", c9orf72_vs_ctrl.voila.gost_filt.terms$term_name)
c9orf72_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(c9orf72_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen"), title = "C9orf72")+ theme(strip.text.y.right = element_blank())
c9orf72_vs_ctrl.voila.gost_filt.terms.curated

tardbp_vs_ctrl.voila.gost = tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
tardbp_vs_ctrl.voila.gost_filt <- tardbp_vs_ctrl.voila.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC")) %>% arrange( -log10(p_value) )
table(tardbp_vs_ctrl.voila.gost_filt$query) # 
paste('"',unique(tardbp_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_tardbp_go_list = c(
"synapse",
"nervous system development",
"cytoskeleton organization",
# "cytoskeletal protein binding",
"cell junction organization",
# "intracellular anatomical structure",
"neuron projection",
"axon",
# "organelle organization",
"protein binding",
"cytoplasm")
tardbp_vs_ctrl.voila.gost_filt.terms <- tardbp_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_tardbp_go_list)
# tardbp_vs_ctrl.voila.gost_filt.terms$term_name = gsub(" molecule binding| organization| organizing", "", tardbp_vs_ctrl.voila.gost_filt.terms$term_name)
tardbp_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(tardbp_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen"), title = "TARDBP") + theme(strip.text.y.right = element_blank())
tardbp_vs_ctrl.voila.gost_filt.terms.curated

fus_vs_ctrl.voila.gost = fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
fus_vs_ctrl.voila.gost_filt <- fus_vs_ctrl.voila.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC")) %>% arrange( -log10(p_value) )
table(fus_vs_ctrl.voila.gost_filt$query) # 
paste('"',unique(fus_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_fus_go_list = c(
"transcription, DNA-templated",
# "regulation of nucleic acid-templated transcription",
# "regulation of transcription, DNA-templated",
"microtubule cytoskeleton",
"regulation of RNA metabolic process",
# "intracellular membrane-bounded organelle",
"membrane-bounded organelle",
"organelle organization",
# "intracellular anatomical structure",
"cytoplasm",
"protein binding")
fus_vs_ctrl.voila.gost_filt.terms <- fus_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_fus_go_list)
# fus_vs_ctrl.voila.gost_filt.terms$term_name = gsub(" molecule binding| organization| organizing", "", fus_vs_ctrl.voila.gost_filt.terms$term_name)
fus_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(fus_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen"), title = "FUS") + theme(strip.text.y.right = element_blank())# + theme(legend.position = "none", axis.text = element_text(size = 10))
fus_vs_ctrl.voila.gost_filt.terms.curated

sod1_vs_ctrl.voila.gost = sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
sod1_vs_ctrl.voila.gost_filt <- sod1_vs_ctrl.voila.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC")) %>% arrange( -log10(p_value) )
table(sod1_vs_ctrl.voila.gost_filt$query) # 
paste('"',unique(sod1_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_sod1_go_list = c(
"Spliceosome",
"negative regulation of mRNA processing",
"negative regulation of RNA splicing",
# "regulation of mRNA processing",
"protein binding",
# "regulation of mRNA splicing, via spliceosome",
"regulation of RNA splicing")
sod1_vs_ctrl.voila.gost_filt.terms <- sod1_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_sod1_go_list)
sod1_vs_ctrl.voila.gost_filt.terms$term_name = gsub("negative regulation of ", "", sod1_vs_ctrl.voila.gost_filt.terms$term_name)
sod1_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(sod1_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen"), title = "SOD1") + theme(strip.text.y.right = element_blank())# + theme(legend.position = "none", axis.text = element_text(size = 10))
sod1_vs_ctrl.voila.gost_filt.terms.curated
```

## Correlation heatmap

```{r}
# heatmap of correlation coefficients
ipsc_mn_mutations_deltapsi = select(c9orf72_vs_ctrl.voila, lsv_junction_id, C9orf72 = deltapsi) %>%
  left_join(select(sod1_vs_ctrl.voila, lsv_junction_id, SOD1 = deltapsi)) %>%
  left_join(select(fus_vs_ctrl.voila, lsv_junction_id, FUS = deltapsi)) %>%
  left_join(select(tardbp_vs_ctrl.voila, lsv_junction_id, TARDBP = deltapsi)) %>%
  drop_na() %>%
  column_to_rownames("lsv_junction_id")
cormat <- round(cor(ipsc_mn_mutations_deltapsi),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
majiq_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson Correlation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.spacing.y = unit(1, 'mm'), legend.title = element_text(size = 8), axis.text=element_text(size=8)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 0.5, title.position = "top", title.hjust = 0.5))
majiq_correlation_ggheatmap

```


## Modulizer barchart

````{r}
genetic_subgroups.modulizer.bind = bind_rows(
  mutate(rename(c9orf72.voila_modulizer, dpsi = als_ctrl_het_median_dpsi, tnom = als_ctrl_het_tnom), mutation = "C9orf72"), 
  mutate(rename(fus.voila_modulizer, dpsi = als_ctrl_het_median_dpsi, tnom = als_ctrl_het_tnom), mutation = "FUS"),
  mutate(rename(tardbp.voila_modulizer, dpsi = als_ctrl_het_median_dpsi, tnom = als_ctrl_het_tnom), mutation = "TARDBP"), 
  mutate(rename(sod1.voila_modulizer, dpsi = als_ctrl_het_median_dpsi, tnom = als_ctrl_het_tnom), mutation = "SOD1")) %>% 
  mutate(mutation = factor(mutation, levels = c("C9orf72","FUS", "TARDBP","SOD1")), modulizer_simplified = gsub("_"," ",modulizer_simplified)) %>%
  filter(tnom < 0.05, abs(dpsi) > 0.1)

genetic_subgroups.modulizer.bind.count = genetic_subgroups.modulizer.bind %>% count(mutation, modulizer_simplified) %>% 
  mutate(modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>%  
  group_by(mutation) %>% add_tally(n) %>% mutate(pct = n/nn *100, pct.label = case_when(pct > 11 ~ paste0(as.character(n)," (", as.character(round(pct,1)),"%)"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) %>% ungroup()
genetic_subgroups.modulizer.bind.count %>% print(n=Inf)

genetic_subgroups.modulizer.bind.barplot = ggbarplot(genetic_subgroups.modulizer.bind.count, x="mutation", y="pct", fill="modulizer_simplified", position = position_fill(), xlab = "",  label = genetic_subgroups.modulizer.bind.count$pct.label, lab.size = 3, lab.vjust = 0.5, lab.hjust = 1.5)  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8), strip.text.y = element_text(angle = 0)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 2)) +
  # facet_grid(mutation ~ .,scales="free_y", labeller = "label_parsed") +
  coord_flip() + ylab(expression(Splicing~type~proportions)) + scale_x_discrete(labels = scales::parse_format()) 
genetic_subgroups.modulizer.bind.barplot




# fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 11471
# fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 2706
# fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #38
# fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 2246
# fus_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 5847
# 
# tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 10,429
# tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 3364
# tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #51
# tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 2707
# tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 7553
# 
# sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 883
# sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 540
# sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #"SFPQ"   "TIA1"   "CAMTA1" "DCTN1"  "NOP56" 
# sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 183
# sod1_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 513
# 
# sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 671
# sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 338
# sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% pull(gene_name) # "BID"    "CAMTA1" "CAMTA1" "SFPQ"   "APP"  
# sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 136
# sporadic_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 360
# 
# c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 626
# c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 318
# c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% pull(gene_name) #"TIA1"   "GOLIM4" "PTPRN2" "PTPRN2" "GOLIM4"
# c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 132 / 511
# c9orf72_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 338/511
# 
# vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 179
# vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 88
# vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #"ATXN2" "KIF5A" "EWSR1"
# vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 24
# vcp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 96
# 
# polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 174
# polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 108
# polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #"HLA-B"
# polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 13
# polyA.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 86
# 
# total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 381
# total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 222
# total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) %>% distinct(gene_name) %>% pull(gene_name) #  "CAMTA1" "APOE"   "APP"    "SCFD1" 
# total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 99
# total.als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 259

```


## Upset plot

```{r}
genetic_subgroups.voila.bind = bind_rows(mutate(tardbp_vs_ctrl.voila, mutation = "TARDBP"), mutate(fus_vs_ctrl.voila, mutation = "FUS"), mutate(sod1_vs_ctrl.voila, mutation = "SOD1"), mutate(c9orf72_vs_ctrl.voila, mutation = "C9orf72"))

genetic_subgroups.voila.bind.upset = majiq_upset_plot(genetic_subgroups.voila.bind, overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 1500, order = "freq", order_group_list = c("TARDBP", "FUS", "SOD1", "C9orf72"), label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"), significance_threshold = "tnom", deltapsi_threshold = 0.1)
genetic_subgroups.voila.bind.upset

# genetic_subgroups.voila.bind.upset = majiq_upset_plot(genetic_subgroups.voila.bind, overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = TRUE, ymax = 6800, order = "freq", order_group_list = c("TARDBP", "FUS", "SOD1", "C9orf72"), label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"), significance_threshold = "tnom", deltapsi_threshold = 0.1)
# genetic_subgroups.voila.bind.upset

genetic_subgroups.voila.bind.up.upset = majiq_upset_plot(filter(genetic_subgroups.voila.bind, deltapsi > 0), overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 750, order = "freq", order_group_list = c("TARDBP", "FUS", "SOD1", "C9orf72"), title = "ALS splicing increased", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"), significance_threshold = "tnom", deltapsi_threshold = 0.1)
genetic_subgroups.voila.bind.up.upset

genetic_subgroups.voila.bind.down.upset = majiq_upset_plot(filter(genetic_subgroups.voila.bind, deltapsi < 0), overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 770, order = "freq", order_group_list = c("TARDBP", "FUS", "SOD1", "C9orf72"), title = "ALS splicing decreased", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"), significance_threshold = "tnom", deltapsi_threshold = 0.1)  
genetic_subgroups.voila.bind.down.upset

```

## Merge

```{r}
majiq.genetic_subgroups.merged = plot_grid(
  plot_grid(tardbp_vs_ctrl.voila.volcano, fus_vs_ctrl.voila.volcano, sod1_vs_ctrl.voila.volcano, c9orf72_vs_ctrl.voila.volcano, ncol = 4, labels = "auto"),
  plot_grid(tardbp_vs_ctrl.voila.gost_filt.terms.curated, fus_vs_ctrl.voila.gost_filt.terms.curated, sod1_vs_ctrl.voila.gost_filt.terms.curated, c9orf72_vs_ctrl.voila.gost_filt.terms.curated, ncol = 4, labels = letters[5:8]),
  genetic_subgroups.modulizer.bind.barplot,
  plot_grid(majiq_correlation_ggheatmap, genetic_subgroups.voila.bind.up.upset, genetic_subgroups.voila.bind.down.upset, ncol = 3, rel_widths = c(0.8,1,1), labels = letters[10:12]),
  nrow = 4, rel_heights = c(1,0.9,0.9,0.8), labels = c("","","i",""))
# majiq.genetic_subgroups.merged
ggsave(majiq.genetic_subgroups.merged, filename = here(proj_path, "splicing/majiq/majiq.genetic_subgroups.merged.png"), height = 11, width = 12)

```




# Fig 6 Variants and fusions

## Mutations

https://cran.r-project.org/web/packages/vcfR/vignettes/intro_to_vcfR.html

### iPSNs

```{r}
# camp_path = here("/camp/lab/patanir/home/users/ziffo")
# read_vcfannotate = function(metadata){
#   print("reading in vcf-annotate files with read_csv")
#   vcf.annotate.csv = metadata$rnavar_path %>% map(read_csv, col_names = c("n","type"), col_types = list(n="n",type="c"))
#   names(vcf.annotate.csv) = metadata$dataset_sample # needs to be unique names e.g. dataset_sample
#   vcf.annotate_tab <- map_dfr(vcf.annotate.csv, bind_rows, .id = "dataset_sample") %>%
#     mutate(variant_type = case_when(grepl("del", type) ~ "Deletion", grepl("ins", type) ~ "Insertion", grepl("snp", type) ~ "SNP", grepl("complex", type) ~ "Complex", TRUE ~ "Unknown")) %>%
#     left_join(metadata, by = "dataset_sample")
#   return(vcf.annotate_tab)
# }
# ipsc_mn_als_datasets.metadata = ipsc_mn_als_datasets.metadata %>% mutate(rnavar_path = paste0("/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate/dbsnp-rnaedit-filtered/",dataset_sample,".dbsnp.redi.filtered.vcfstats.csv"))
# ipsc_mn_als_datasets.vcfannotate = read_vcfannotate(ipsc_mn_als_datasets.metadata)
# ipsc_mn_als_datasets.vcfannotate
# write_csv(ipsc_mn_als_datasets.vcfannotate, here(camp_path,"projects/ipsc-mn-als-meta/rnavar/ipsc_mn_als_datasets.vcfannotate.dbsnp.redi.filtered.csv"))
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
ipsc_mn_als_datasets.vcfannotate = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.dbsnp.redi.filtered.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(dataset_sample)%>% add_tally(n, name = "total") %>% ungroup %>% select(dataset_sample, n, type, variant_type, total, everything())  %>% # add a column of sum of mutations per sample
  left_join(multiqc_rseqc_read_distribution.tsv) # left_join multiqc stats
ipsc_mn_als_datasets.vcfannotate
answerals.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% filter(dataset == "answerals") %>%
  mutate(age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ age_at_death, TRUE ~ age_at_sample_collection), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ age_at_symptom_onset, TRUE ~ age_at_sample_collection), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 59, TRUE ~ age_at_sample_collection)) # add age as covariate to adjust for (deal with few NAs)
answerals.metadata %>% count(condition) # 238 ALS, 42 CTRL
answerals.vcfannotate.samples_no_variants = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% answerals.vcfannotate$dataset_sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

### ALL VARIANT TYPES
answerals.vcfannotate.totals = answerals.vcfannotate %>% select(dataset_sample, variant_type, n, CONDITION, total_reads, age_at_sample_collection) %>% pivot_wider(c(dataset_sample, CONDITION, total_reads, age_at_sample_collection), names_from = "variant_type", values_from = "n") %>% mutate(total = Complex + Deletion + Insertion + SNP) #keep 1 row per sample for total number of variants
answerals.vcfannotate.totals %>% glimpse
answerals.vcfannotate.totals %>% my_summarise(Deletion)

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
answerals.vcfannotate.totals.fitg = glm(total ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.totals, family = poisson)
summ(answerals.vcfannotate.totals.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.    z val.             p
# (Intercept)                                                     8.659831e+00 9.435873e-03 917.75619  0.000000e+00
# CONDITIONALS                                                    4.207523e-02 1.631781e-03  25.78485 1.311328e-146
plot_summs(answerals.vcfannotate.totals.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

answerals.vcfannotate.totals.fitg_plot = glm(total ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.vcfannotate.totals, family = poisson) # cant plot rcs spline
# partialize(answerals.vcfannotate.totals.fitg, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(answerals.vcfannotate.totals.fitg_plot, vars = "CONDITION") %>% my_summarise(total, CONDITION)
# CONDITION mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total se_total
#   <fct>          <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>    <dbl>
# 1 CTRL          11709.       11236.    1882.   10419.   12332.     9343.    17187.   468354.      40     298.
# 2 ALS           12254.       11780.    1858.   11062.   13197.     9428.    21756.  2806278.     229     123.
# answerals.vcfannotate.totals.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 65000, xmin = 1, xmax = 2)
# answerals.vcfannotate.total.violin <- violin_plot(partialize(answerals.vcfannotate.totals.fitg_plot, vars = "CONDITION"), color_by = "CONDITION", continuous = "total", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(54000,65000), dpi = 300, ylabel = "Adjusted Number of Variants", title = "iPSMN: All Variant types") +
#   stat_pvalue_manual(answerals.vcfannotate.totals.stats, tip.length = 0.01, size = 4)
# answerals.vcfannotate.total.violin
effect_plot(answerals.vcfannotate.totals.fitg_plot, pred = CONDITION, plot.points = TRUE, interval = TRUE, jitter = 0.2, partial.residuals = TRUE, x.label = "", y.label = "Adjusted number of variants detected", line.colors = c("dodgerblue2", "firebrick2")) + theme_oz(xaxis_arrow = FALSE) + scale_colour_manual(values = c("dodgerblue2", "firebrick2"))
effect_plot(answerals.vcfannotate.totals.fitg_plot, pred = total_reads, interval = TRUE, plot.points = TRUE,  jitter = 0, x.label = "Coverage Depth", y.label = "Raw number of variants detected")  + theme_oz() # read effect
effect_plot(answerals.vcfannotate.totals.fitg_plot, pred = age_at_sample_collection, interval = TRUE, plot.points = TRUE,  jitter = 0)
fit <- glm(total ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), family = poisson, data=answerals.vcfannotate.totals)
predicted <- predict(fit, type="response")
ggplot(cbind(answerals.vcfannotate.totals, predicted=predicted), aes(x=total_reads, y=predicted, colour= CONDITION, group = interaction(CONDITION))) +
  scale_colour_manual(values = c("dodgerblue2", "firebrick2")) + 
  geom_line() + theme_oz() + labs(y = "Adjusted number of variants", x = "Coverage Depth") + xlim(c(60000000,110000000)) + ylim(c(52500,61000))
#### Variant types
answerals.vcfannotate %>% group_by(variant_type) %>% tally(n)
# variant_type       n
#   <fct>          <dbl>
# 1 SNP          2422491
# 2 Insertion     608164
# 3 Deletion      342038
# 4 Complex          899


### SNP
answerals.vcfannotate.snp = answerals.vcfannotate %>% filter(variant_type == "SNP") 
# add samples with 0 variants for each variant type
samples_no_SNP = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% pull(answerals.vcfannotate.snp,dataset_sample)] 
samples_no_SNP # 0 samples with 0 SNPS

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
answerals.vcfannotate.snp.fitg = glm(n ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.snp, family = poisson)
summ(answerals.vcfannotate.snp.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.     z val.             p
# (Intercept)                                                     8.290175e+00 1.091895e-02 759.246591  0.000000e+00
# CONDITIONALS                                                    4.975676e-02 1.884780e-03  26.399236 1.398116e-153
plot_summs(answerals.vcfannotate.snp.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

answerals.vcfannotate.snp.fitg_plot = glm(n ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.vcfannotate.snp, family = poisson) # cant plot rcs spline
partialize(answerals.vcfannotate.snp.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(answerals.vcfannotate.snp.fitg_plot, vars = "CONDITION") %>% my_summarise(n, CONDITION)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n  max_n    sum_n   n_n  se_n
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl>  <dbl>    <dbl> <int> <dbl>
# 1 CTRL       8354.    7964. 1550. 7381. 9016. 6494. 13599.  350888.    42  239.
# 2 ALS        8854.    8343. 1713. 7747. 9581. 6316. 19182. 2107174.   238  111.
effect_plot(answerals.vcfannotate.snp.fitg_plot, pred = CONDITION, plot.points = TRUE, interval = TRUE, jitter = 0.2, partial.residuals = TRUE)


### Insertions
answerals.vcfannotate.insertion = answerals.vcfannotate %>% filter(variant_type == "Insertion") 
samples_no_Insertion = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% pull(answerals.vcfannotate.insertion,dataset_sample)] # add samples with 0 variants for each variant type
samples_no_Insertion # 0 samples with 0 Insertions

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
answerals.vcfannotate.insertion.fitg = glm(n ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.insertion, family = poisson)
summ(answerals.vcfannotate.insertion.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.     z val.            p
# (Intercept)                                                     7.219755e+00 2.149395e-02 335.897129 0.000000e+00
# CONDITIONALS                                                    2.641514e-02 3.734622e-03   7.073041 1.515747e-12
plot_summs(answerals.vcfannotate.insertion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

answerals.vcfannotate.insertion.fitg_plot = glm(n ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.vcfannotate.insertion, family = poisson) # cant plot rcs spline
partialize(answerals.vcfannotate.insertion.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(answerals.vcfannotate.insertion.fitg_plot, vars = "CONDITION") %>% my_summarise(n, CONDITION)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n   sum_n   n_n  se_n
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>   <dbl> <int> <dbl>
# 1 CTRL       2142.    2097.  374. 1931. 2272. 1706. 3663.  89974.    42  57.6
# 2 ALS        2199.    2163.  303. 2023. 2322. 1506. 3609. 523270.   238  19.7
effect_plot(answerals.vcfannotate.insertion.fitg_plot, pred = CONDITION, plot.points = TRUE, interval = TRUE, jitter = 0.2, partial.residuals = TRUE)


### Deletions
answerals.vcfannotate.deletion = answerals.vcfannotate %>% filter(variant_type == "Deletion") 
samples_no_Deletion = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% pull(answerals.vcfannotate.deletion,dataset_sample)] # add samples with 0 variants for each variant type
samples_no_Deletion # 0 samples with 0 Deletions

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
answerals.vcfannotate.deletion.fitg = glm(n ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.deletion, family = poisson)
summ(answerals.vcfannotate.deletion.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.      z val.             p
# (Intercept)                                                     6.511775e+00 2.887506e-02 225.5155688  0.000000e+00
# CONDITIONALS                                                    3.862437e-02 4.997163e-03   7.7292597  1.081738e-14
plot_summs(answerals.vcfannotate.deletion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

answerals.vcfannotate.deletion.fitg_plot = glm(n ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.vcfannotate.deletion, family = poisson) # cant plot rcs spline
partialize(answerals.vcfannotate.deletion.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(answerals.vcfannotate.deletion.fitg_plot, vars = "CONDITION") %>% my_summarise(n, CONDITION)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n   sum_n   n_n  se_n
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>   <dbl> <int> <dbl>
# 1 CTRL       1183.    1160.  111. 1103. 1265.  989. 1465.  49688.    42 17.1 
# 2 ALS        1232.    1217.  112. 1163. 1278.  988. 1761. 293243.   238  7.23
effect_plot(answerals.vcfannotate.deletion.fitg_plot, pred = CONDITION, plot.points = TRUE, interval = TRUE, jitter = 0.2, partial.residuals = TRUE)

# Facet voilin
answerals.vcfannotate.totals.fitg_plot.bind = bind_rows(mutate(partialize(answerals.vcfannotate.totals.fitg_plot, vars = "CONDITION"), type = "All mutations",n=total), mutate(partialize(answerals.vcfannotate.snp.fitg_plot, vars = "CONDITION"), type = "SNV"),
mutate(partialize(answerals.vcfannotate.insertion.fitg_plot, vars = "CONDITION"), type = "Insertion"), mutate(partialize(answerals.vcfannotate.deletion.fitg_plot, vars = "CONDITION"), type = "Deletion")) %>% 
  mutate(type = factor(type, levels = c("All mutations", "SNV", "Insertion", "Deletion")))
answerals.vcfannotate.bind.stats = tibble(facet =  c("All mutations", "SNV", "Insertion", "Deletion"), group1 = "CTRL", group2 = "ALS",  p.signif = "****", 
                                          y.position = c(17000,13000,2800,1540), xmin = 1, xmax = 2)
answerals.vcfannotate.totals.fitg_plot.bind.filt = answerals.vcfannotate.totals.fitg_plot.bind %>% filter((type == "All mutations" & n < 18000) |(type == "Deletion" & n < 1600) |(type == "Insertion" & n < 3000) |(type == "SNV" & n < 14000))
answerals.vcfannotate.totals.fitg_plot.bind.violin <- violin_plot(answerals.vcfannotate.totals.fitg_plot.bind.filt, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, facet_by = "type", facet_order = c("All mutations", "SNV", "Insertion", "Deletion"),dpi = 300, ylabel = "Adjusted Number of Mutations", title = "iPSMN: Somatic Mutations", facet_ncol = 4) +
  stat_pvalue_manual(answerals.vcfannotate.bind.stats, tip.length = 0.01, size = 4)
answerals.vcfannotate.totals.fitg_plot.bind.violin



violin_plot(partialize(answerals.vcfannotate.totals.fitg_plot, vars = "CONDITION"), color_by = "CONDITION", continuous = "total", cols = c("dodgerblue2", "firebrick2"), plot_stats = TRUE, arrow_labels = FALSE, dpi = 300, ylabel = "Adjusted Number of Variants", ylims = c(8000,23000))
```


#### Genetic subgroups

```{r}
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
ipsc_mn_als_datasets.vcfannotate = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.dbsnp.redi.filtered.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(dataset_sample)%>% add_tally(n, name = "total") %>% ungroup %>% select(dataset_sample, n, type, variant_type, total, everything())  %>% # add a column of sum of variants per sample
  left_join(multiqc_rseqc_read_distribution.tsv) # left_join multiqc stats
ipsc_mn_als_datasets.vcfannotate
answerals.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% filter(dataset == "answerals") %>%
  mutate(age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ age_at_sample_collection, TRUE ~ age_at_sample_collection), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ age_at_symptom_onset, TRUE ~ age_at_sample_collection), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 59, TRUE ~ age_at_sample_collection)) # add age as covariate to adjust for (deal with few NAs)

answerals.vcfannotate.mutation = answerals.vcfannotate %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic",mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "C9orf72", "SOD1"))) %>% #mutation == "fus" ~ "FUS","FUS",  mutation == "tardbp" ~ "TARDBP", "TARDBP", 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, total, genetic_group, CONDITION, total_reads, mutation, age_at_sample_collection) # keep 1 row per sample

answerals.metadata_genetic_subgroup = answerals.metadata %>% filter(mutation %in% c("ctrl", "iso", "sporadic", "sod1", "c9orf72")) #"fus", "tardbp", 
answerals.vcfannotate.samples_no_variants = answerals.metadata_genetic_subgroup$dataset_sample[!answerals.metadata_genetic_subgroup$dataset_sample %in% answerals.vcfannotate.mutation$dataset_sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

# All variants
answerals.vcfannotate.mutation.fitg = glm(total ~ genetic_group + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.mutation, family=poisson)
summ(answerals.vcfannotate.mutation.fitg, digits = 5)$coeftable 
#                                                                        Est.         S.E.      z val.             p
# (Intercept)                                                     8.723720e+00 9.706056e-03 898.7913938  0.000000e+00
# genetic_groupSporadic                                           5.059270e-02 1.618597e-03  31.2571340 1.785927e-214
# genetic_groupC9orf72                                            2.560660e-02 2.537760e-03  10.0902343  6.102366e-24
# genetic_groupSOD1                                              -3.075098e-04 3.580625e-03  -0.0858816  9.315605e-01
plot_summs(answerals.vcfannotate.mutation.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


### SNV
answerals.vcfannotate.mutation.SNV = answerals.vcfannotate %>% filter(variant_type == "SNP") %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation %in% c("ctrl","iso") ~ "CTRL")) %>%  drop_na(genetic_group)
answerals.vcfannotate.SNV = bind_rows(answerals.vcfannotate.mutation.SNV, mutate(filter(answerals.vcfannotate, variant_type == "SNP", CONDITION!="CTRL"),genetic_group="pan ALS")) %>% mutate(genetic_group = factor(genetic_group, levels = c("CTRL", "pan ALS", "Sporadic", "C9orf72", "SOD1"))) %>% select(sample, total, genetic_group, CONDITION, n, mutation, age_at_sample_collection, variant_type, total_reads)

answerals.vcfannotate.samples_no_variants = answerals.metadata_genetic_subgroup$sample[!answerals.metadata_genetic_subgroup$sample %in% answerals.vcfannotate.SNV$sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

answerals.vcfannotate.SNV.fitg = glm(n ~ genetic_group + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.SNV, family = poisson)
summ(answerals.vcfannotate.SNV.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.     z val.             p
# (Intercept)                                                     8.314380e+00 1.148082e-02 724.197403  0.000000e+00
# genetic_groupSporadic                                           5.689798e-02 1.912980e-03  29.743108 2.129229e-194
# genetic_groupC9orf72                                            2.540886e-02 3.001093e-03   8.466534  2.528032e-17
# genetic_groupSOD1                                              -1.404521e-02 4.252654e-03  -3.302693  9.576106e-04
plot_summs(answerals.vcfannotate.SNV.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


### Insertions
answerals.vcfannotate.mutation.insertion = answerals.vcfannotate %>% filter(variant_type == "Insertion") %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation %in% c("ctrl","iso") ~ "CTRL")) %>% drop_na(genetic_group)
answerals.vcfannotate.insertion = bind_rows(answerals.vcfannotate.mutation.insertion, mutate(filter(answerals.vcfannotate, variant_type == "Insertion", CONDITION!="CTRL"),genetic_group="pan ALS")) %>% mutate(genetic_group = factor(genetic_group, levels = c("CTRL", "pan ALS", "Sporadic", "C9orf72", "SOD1"))) %>% 
  select(sample, total, genetic_group, CONDITION, n, mutation, age_at_sample_collection, variant_type, total_reads)

answerals.vcfannotate.samples_no_variants = answerals.metadata_genetic_subgroup$sample[!answerals.metadata_genetic_subgroup$sample %in% answerals.vcfannotate.insertion$sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

answerals.vcfannotate.insertion.fitg = glm(n ~ genetic_group + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.insertion, family = poisson)
summ(answerals.vcfannotate.insertion.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.     z val.            p
# (Intercept)                                                     7.274023e+00 2.267417e-02 320.806662 0.000000e+00
# genetic_groupSporadic                                           2.963446e-02 3.793902e-03   7.811075 5.670216e-15
# genetic_groupC9orf72                                            3.690921e-02 5.911363e-03   6.243772 4.271413e-10
# genetic_groupSOD1                                               5.077407e-02 8.243004e-03   6.159657 7.290289e-10
plot_summs(answerals.vcfannotate.insertion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


### Deletions
answerals.vcfannotate.mutation.Deletion = answerals.vcfannotate %>% filter(variant_type == "Deletion") %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation %in% c("ctrl","iso") ~ "CTRL")) %>% drop_na(genetic_group)
answerals.vcfannotate.Deletion = bind_rows(answerals.vcfannotate.mutation.Deletion, mutate(filter(answerals.vcfannotate, variant_type == "Deletion", CONDITION!="CTRL"),genetic_group="pan ALS")) %>% 
  mutate(genetic_group = factor(genetic_group, levels = c("CTRL", "pan ALS", "Sporadic", "C9orf72", "SOD1"))) %>% 
  select(sample, total, genetic_group, CONDITION, n, mutation, age_at_sample_collection, variant_type, total_reads)
answerals.vcfannotate.samples_no_variants = answerals.metadata_genetic_subgroup$sample[!answerals.metadata_genetic_subgroup$sample %in% answerals.vcfannotate.Deletion$sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

answerals.vcfannotate.Deletion.fitg = glm(n ~ genetic_group + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.Deletion, family = poisson)
summ(answerals.vcfannotate.Deletion.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.      z val.             p
# (Intercept)                                                     6.508977e+00 3.044112e-02 213.8218554  0.000000e+00
# genetic_groupSporadic                                           4.361831e-02 5.073318e-03   8.5975921  8.140605e-18
# genetic_groupC9orf72                                            7.257786e-03 8.008789e-03   0.9062277  3.648153e-01
# genetic_groupSOD1                                               2.950872e-03 1.120849e-02   0.2632712  7.923416e-01
plot_summs(answerals.vcfannotate.Deletion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


answerals.vcfannotate.mutation.fitg.forrest_plot = plot_summs(answerals.vcfannotate.SNV.fitg, answerals.vcfannotate.insertion.fitg, answerals.vcfannotate.Deletion.fitg, 
           model.names = c("SNV","Insertion","Deletion"),
           coefs = c("pan ALS"="genetic_grouppan ALS","Sporadic"="genetic_groupSporadic","C9orf72"="genetic_groupC9orf72","SOD1"="genetic_groupSOD1"), plot.distributions = TRUE, rescale.distributions = TRUE, 
           colors = c("dodgerblue2","firebrick2","forestgreen","gold2"), legend.title	= "",exp = FALSE,point.shape = FALSE, point.size = 1) + 
  coord_cartesian(xlim = c(-0.07,0.07)) + #ggtitle("iPSMN") +
 theme_classic() + theme(legend.text=element_text(size=8), legend.spacing = unit(0.01, 'cm'), legend.position = "top", axis.text = element_text(colour = "black", size = 8), axis.title = element_text(colour = "black", size = 9, face ="plain"), strip.text.y = element_text(colour = "black", size = 7), plot.title = element_text(hjust = 0.5, size = 9)) + guides(color=guide_legend(keywidth=0.1, keyheight=0.15, default.unit="inch")) +
  annotation_custom(grid::textGrob(label = paste0("increase in ALS"), y = unit(0.1, "npc"), x = unit(0.8, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 0)) + 
  annotation_custom(grid::textGrob(label = paste0("increase in CTRL"), y = unit(0.1, "npc"), x = unit(0.2, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 0)) + 
  annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), x = c(0.95,0.65), y = unit(0.05, "npc"))) +
  annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), x = c(0.05,0.35), y = unit(0.05, "npc")))
answerals.vcfannotate.mutation.fitg.forrest_plot



# OLD VIOLINS
# 
# # generalised linear model spline
# 
# ## sporadic
# answerals.vcfannotate.mutation.sporadic = filter(answerals.vcfannotate.mutation, genetic_group %in% c("Sporadic", "CTRL"))
# answerals.vcfannotate.mutation.sporadic.fitg = glm(total ~ genetic_group + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.vcfannotate.mutation.sporadic, family=poisson)
# summ(answerals.vcfannotate.mutation.sporadic.fitg, digits = 5)$coeftable 
# #                                                                        Est.         S.E.     z val.             p
# # (Intercept)                                                     1.067592e+01 4.478339e-03 2383.90238  0.000000e+00
# # genetic_groupSporadic                                           1.608387e-02 7.261189e-04   22.15047 1.032431e-108
# plot_summs(answerals.vcfannotate.mutation.sporadic.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
# answerals.vcfannotate.mutation.sporadic.fitg_plot = glm(total ~ genetic_group + total_reads + age_at_sample_collection, data=answerals.vcfannotate.mutation.sporadic, family = poisson) # cant plot rcs spline
# partialize(answerals.vcfannotate.mutation.sporadic.fitg_plot, vars = "genetic_group") %>% my_summarise(total, genetic_group)
# # genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total
# #   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>
# # 1 CTRL              58116.       57448.    2754.   56734.   58606.    54606.    70968.  2440855.      42
# # 2 Sporadic          59131.       58351.    3197.   57300.   59663.    54538.    72591. 11826188.     200
# 
# ## c9orf72
# answerals.vcfannotate.mutation.c9orf72 = filter(answerals.vcfannotate.mutation, genetic_group %in% c("C9orf72", "CTRL"))
# answerals.vcfannotate.mutation.c9orf72.fitg = glm(data=answerals.vcfannotate.mutation.c9orf72, total ~ genetic_group + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), family=poisson)
# summ(answerals.vcfannotate.mutation.c9orf72.fitg, digits = 5)$coeftable 
# # Coefficients:
# #                                                                  Estimate Std. Error  z value Pr(>|z|)    
# # (Intercept)                                                     1.073e+01  8.830e-03 1215.009  < 2e-16 ***
# # genetic_groupC9orf72                                            5.037e-03  1.179e-03    4.272 1.94e-05 ***
# plot_summs(answerals.vcfannotate.mutation.c9orf72.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
# answerals.vcfannotate.mutation.c9orf72.fitg_plot = glm(total ~ genetic_group + total_reads + age_at_sample_collection, data=answerals.vcfannotate.mutation.c9orf72, family = poisson) # cant plot rcs spline
# partialize(answerals.vcfannotate.mutation.c9orf72.fitg_plot, vars = "genetic_group") %>% my_summarise(total, genetic_group)
# # genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total
# #   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>
# # 1 CTRL              57510.       56924.    2601.   56072.   58445.    53849.    69239.  2415400.      42
# # 2 C9orf72           57638.       57472.    1573.   56556.   58158.    54997.    61591.  1210389.      21
# 
# ## SOD1
# answerals.vcfannotate.mutation.sod1 = filter(answerals.vcfannotate.mutation, genetic_group %in% c("SOD1", "CTRL"))
# answerals.vcfannotate.mutation.sod1.fitg = glm(data=answerals.vcfannotate.mutation.sod1, total ~ genetic_group + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), family=poisson)
# summ(answerals.vcfannotate.mutation.sod1.fitg, digits = 5)$coeftable 
# # Coefficients:
# #                                                                  Estimate Std. Error z value Pr(>|z|)    
# # (Intercept)                                                     1.070e+01  9.573e-03 1117.65  < 2e-16 ***
# # genetic_groupSOD1                                              -4.858e-03  1.728e-03   -2.81  0.00495 ** 
# plot_summs(answerals.vcfannotate.mutation.sod1.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
# answerals.vcfannotate.mutation.sod1.fitg_plot = glm(total ~ genetic_group + total_reads + age_at_sample_collection, data=answerals.vcfannotate.mutation.sod1, family = poisson) # cant plot rcs spline
# partialize(answerals.vcfannotate.mutation.sod1.fitg_plot, vars = "genetic_group") %>% my_summarise(total, genetic_group)
# # genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total
# #   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>
# # 1 CTRL              57907.       57345.    2623.   56480.   58810.    54294.    69852.  2432076.      42
# # 2 SOD1              57263.       57265.    2183.   55976.   59069.    53732.    59989.   458108.       8
# 
# # plot fitted values
# answerals.vcfannotate.mutation.bind= bind_rows(partialize(answerals.vcfannotate.mutation.sporadic.fitg_plot, vars = "genetic_group"), 
#                                                filter(partialize(answerals.vcfannotate.mutation.c9orf72.fitg_plot, vars = "genetic_group"), genetic_group!="CTRL"), 
#                                                filter(partialize(answerals.vcfannotate.mutation.sod1.fitg_plot, vars = "genetic_group"), genetic_group!="CTRL"))
# answerals.vcfannotate.mutation.bind %>% my_summarise(total, genetic_group)
# # genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total
# #   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>
# # 1 CTRL              58116.       57448.    2754.   56734.   58606.    54606.    70968.  2440855.      42
# # 2 Sporadic          59131.       58351.    3197.   57300.   59663.    54538.    72591. 11826188.     200
# # 3 C9orf72           57638.       57472.    1573.   56556.   58158.    54997.    61591.  1210389.      21
# # 4 SOD1              57263.       57265.    2183.   55976.   59069.    53732.    59989.   458108.       8
# 
# # Violin total variant events
# answerals.vcfannotate.mutation.stat <- answerals.vcfannotate.mutation.bind %>% wilcox_test(total ~ genetic_group) %>% add_significance() %>% add_xy_position() %>% filter(group1 == "CTRL") %>%
#   mutate(p.adj.signif = case_when(group2 %in% c("Sporadic","C9orf72") ~ "****", group2 == "SOD1" ~ "ns", TRUE ~ p.adj.signif), y.position = case_when(group2 == "Sporadic" ~ 64000, group2 == "C9orf72" ~ 65000, group2 == "SOD1" ~ 67000, TRUE ~ y.position))
# answerals.vcfannotate.mutation.stat
# # .y.   group1 group2      n1    n2 statistic          p      p.adj p.adj.signif y.position groups        xmin  xmax
# #   <chr> <chr>  <chr>    <int> <int>     <dbl>      <dbl>      <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# # 1 total CTRL   Sporadic   126   200      8580 0.00000123 0.00000738 ****              63000 <chr [2]>        1     2
# # 2 total CTRL   C9orf72    126    21      1302 0.91       1          ****              65000 <chr [2]>        1     3
# # 3 total CTRL   SOD1       126     8       532 0.796      1          ns                67000 <chr [2]>        1     4
# answerals.vcfannotate.mutation.violin <- violin_plot(answerals.vcfannotate.mutation.bind, color_by = "genetic_group", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(53000,65000), dpi = 300, ylabel = "Adjusted Number of Variants", title = "iPSMN: Genetic backgrounds") + stat_pvalue_manual(answerals.vcfannotate.mutation.stat, tip.length = 0.01, size = 4, hide.ns = TRUE)
# answerals.vcfannotate.mutation.violin

```



### Postmortem

```{r}
# read_vcfannotate = function(metadata){
#   print("reading in vcf-annotate files with read_csv")
#   vcf.annotate.csv = metadata$rnavar_path %>% map(read_csv, col_names = c("n","type"), col_types = list(n="n",type="c"))
#   names(vcf.annotate.csv) = metadata$sample # needs to be unique names e.g. sample
#   vcf.annotate_tab <- map_dfr(vcf.annotate.csv, bind_rows, .id = "sample") %>%
#     mutate(variant_type = case_when(grepl("del", type) ~ "Deletion", grepl("ins", type) ~ "Insertion", grepl("snp", type) ~ "SNP", grepl("complex", type) ~ "Complex", TRUE ~ "Unknown")) %>%
#     left_join(metadata, by = "sample")
#   return(vcf.annotate_tab)
# }
# nygc_spinal_cord_metadata %<>% mutate(rnavar_path = paste0("/camp/project/proj-luscombn-patani/working/nygc-als-consortium/rnavar/variant_calling/vcfannotate/rna-editing-filtered/",sample,".redi.filtered.vcfstats.csv"))
# nygc_spinal_cord.vcfannotate = read_vcfannotate(nygc_spinal_cord_metadata)
# nygc_spinal_cord.vcfannotate
# write_csv(nygc_spinal_cord.vcfannotate, "/camp/project/proj-luscombn-patani/working/nygc-als-consortium/rnavar/nygc_spinal_cord.vcfannotate.dbsnp.redi.filtered.csv")
nygc_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
nygc.rnavar.multiqc_star.txt = read_tsv(here(nygc_path,"rnavar/reports/multiqc_data/multiqc_star.txt")) %>% clean_names()
nygc_spinal_cord.vcfannotate = read_csv(here(nygc_path,"rnavar/nygc_spinal_cord.vcfannotate.dbsnp.redi.filtered.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(sample) %>% add_tally(n, name = "total") %>% ungroup %>% select(sample, n, type, variant_type, total, everything()) %>%  # add a column of sum of variants per sample
  left_join(nygc.rnavar.multiqc_star.txt) # left_join multiqc stats
nygc_spinal_cord.vcfannotate
nygc_spinal_cord_metadata %>% count(condition) # 214 ALS, 57 CTRL
nygc_spinal_cord.vcfannotate.samples_no_variants = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% nygc_spinal_cord.vcfannotate$sample] # 0 i.e. all samples have at least 1 variant
nygc_spinal_cord.vcfannotate.samples_no_variants # 0

### ALL VARIANT TYPES
nygc_spinal_cord.vcfannotate.totals = nygc_spinal_cord.vcfannotate %>% select(sample, CONDITION, variant_type, n, total_reads, age_at_death, sample_source, library_prep, rin) %>% pivot_wider(c(sample, CONDITION, total_reads, age_at_death, sample_source, library_prep, rin), names_from = "variant_type", values_from = "n") %>% mutate(Complex = case_when(is.na(Complex) ~ 0, TRUE ~ Complex), total = Complex + Deletion + Insertion + SNP) #keep 1 row per sample for total number of variants
nygc_spinal_cord.vcfannotate.totals
nygc_spinal_cord.vcfannotate.totals %>% glimpse

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
nygc_spinal_cord.vcfannotate.totals.fitg = glm(total ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3) + rms::rcs(rin, 3), data=nygc_spinal_cord.vcfannotate.totals, family = poisson)
summ(nygc_spinal_cord.vcfannotate.totals.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.      z val.             p
# (Intercept)                             8.853923e+00 7.327248e-03 1208.355936  0.000000e+00
# CONDITIONALS                            5.392562e-02 1.352342e-03   39.875736  0.000000e+00
nygc_spinal_cord.vcfannotate.totals.fitg_plot = glm(total ~ CONDITION + library_prep + total_reads + age_at_death + rin, data=nygc_spinal_cord.vcfannotate.totals, family = poisson) # cant plot rcs spline
partialize(nygc_spinal_cord.vcfannotate.totals.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(nygc_spinal_cord.vcfannotate.totals.fitg_plot, vars = "CONDITION") %>% my_summarise(total, CONDITION)
# CONDITION mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total se_total
#   <fct>          <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>    <dbl>
# 1 CTRL          13549.       13232.    1447.   12600.   14349.    10288.    17241.   772312.      57    192. 
# 2 ALS           14024.       14143.    1421.   13176.   14893.    10609.    18192.  3001225.     214     97.1
plot_summs(nygc_spinal_cord.vcfannotate.totals.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
# nygc_spinal_cord.vcfannotate.totals.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 78000., xmin = 1, xmax = 2)
# nygc_spinal_cord.vcfannotate.total.violin <- violin_plot(partialize(nygc_spinal_cord.vcfannotate.totals.fitg_plot, vars = "CONDITION"), color_by = "CONDITION", continuous = "total", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(52980.,78000.), dpi = 300, ylabel = "Adjusted Number of Variants", title = "Postmortem: All Variant types") +
#   stat_pvalue_manual(nygc_spinal_cord.vcfannotate.totals.stats, tip.length = 0.01, size = 4)
# nygc_spinal_cord.vcfannotate.total.violin
effect_plot(nygc_spinal_cord.vcfannotate.totals.fitg_plot, pred = CONDITION, plot.points = TRUE, interval = TRUE, jitter = 0.2, partial.residuals = TRUE)
effect_plot(nygc_spinal_cord.vcfannotate.totals.fitg_plot, pred = total_reads, interval = TRUE, plot.points = TRUE,  jitter = 0) # read effect
effect_plot(nygc_spinal_cord.vcfannotate.totals.fitg_plot, pred = age_at_death, interval = TRUE, plot.points = TRUE,  jitter = 0)
effect_plot(nygc_spinal_cord.vcfannotate.totals.fitg_plot, pred = rin, interval = TRUE, plot.points = TRUE,  jitter = 0) + ggpubr::stat_cor(method = "pearson") + geom_smooth(method='lm') 
fit <- glm(total ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), family = poisson, data=nygc_spinal_cord.vcfannotate.totals)
predicted <- predict(fit, type="response")
ggplot(cbind(nygc_spinal_cord.vcfannotate.totals, predicted=predicted), aes(x=total_reads, y=predicted, colour= CONDITION, group = interaction(CONDITION))) + geom_line() + theme_oz()

fit <- glm(total ~ CONDITION + rms::rcs(rin, 3), family = poisson, data=nygc_spinal_cord.vcfannotate.totals)
predicted <- predict(fit, type="response")
ggplot(cbind(nygc_spinal_cord.vcfannotate.totals, predicted=predicted), aes(x=rin, y=predicted, colour= CONDITION, group = interaction(CONDITION))) + geom_line() + theme_oz()

#### Variant types
nygc_spinal_cord.vcfannotate %>% group_by(variant_type) %>% tally(n)
# variant_type       n
#   <fct>          <dbl>
# 1 SNP          2758427
# 2 Insertion     909250
# 3 Deletion      200656
# 4 Complex          672

### SNP
nygc_spinal_cord.vcfannotate.snp = nygc_spinal_cord.vcfannotate %>% filter(variant_type == "SNP") 
samples_no_SNP = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(nygc_spinal_cord.vcfannotate.snp,sample)] # add samples with 0 variants for each variant type
samples_no_SNP # 0 samples with 0 SNPS

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
nygc_spinal_cord.vcfannotate.snp.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.vcfannotate.snp, family = poisson)
summ(nygc_spinal_cord.vcfannotate.snp.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.      z val.             p
# (Intercept)                             8.342918e+00 7.151475e-03 1166.600991  0.000000e+00
# CONDITIONALS                            4.066343e-02 1.557398e-03   26.109856 2.817650e-150
plot_summs(nygc_spinal_cord.vcfannotate.snp.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

nygc_spinal_cord.vcfannotate.snp.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate.snp, family = poisson) # cant plot rcs spline
partialize(nygc_spinal_cord.vcfannotate.snp.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(nygc_spinal_cord.vcfannotate.snp.fitg_plot, vars = "CONDITION") %>% my_summarise(n, CONDITION)
# CONDITION mean_n median_n  sd_n  Q1_n   Q3_n min_n  max_n    sum_n   n_n  se_n
#   <fct>      <dbl>    <dbl> <dbl> <dbl>  <dbl> <dbl>  <dbl>    <dbl> <int> <dbl>
# 1 CTRL       9570.    9447. 1094. 8753. 10079. 7372. 12464.  545473.    57 145. 
# 2 ALS        9733.    9593. 1178. 8953. 10389. 7291. 12960. 2082928.   214  80.5


### Insertions
nygc_spinal_cord.vcfannotate.insertion = nygc_spinal_cord.vcfannotate %>% filter(variant_type == "Insertion") 
# add samples with 0 variants for each variant type
samples_no_Insertion = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(nygc_spinal_cord.vcfannotate.insertion,sample)]
samples_no_Insertion # 0 samples with 0 Insertions

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
nygc_spinal_cord.vcfannotate.insertion.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.vcfannotate.insertion, family = poisson)
summ(nygc_spinal_cord.vcfannotate.insertion.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.             p
# (Intercept)                             7.768376e+00 1.254492e-02 619.244604  0.000000e+00
# CONDITIONALS                            6.956602e-02 2.722444e-03  25.552781 5.113434e-144
plot_summs(nygc_spinal_cord.vcfannotate.insertion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

nygc_spinal_cord.vcfannotate.insertion.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate.insertion, family = poisson) # cant plot rcs spline
partialize(nygc_spinal_cord.vcfannotate.insertion.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(nygc_spinal_cord.vcfannotate.insertion.fitg_plot, vars = "CONDITION") %>% my_summarise(n, CONDITION)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n   sum_n   n_n  se_n
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>   <dbl> <int> <dbl>
# 1 CTRL       3430.    3424.  790. 3045. 3739. 1947. 7043. 195489.    57 105. 
# 2 ALS        3662.    3654.  789. 3116. 4059. 1974. 6846. 783715.   214  53.9


### Deletions
nygc_spinal_cord.vcfannotate.deletion = nygc_spinal_cord.vcfannotate %>% filter(variant_type == "Deletion") 

# add samples with 0 variants for each variant type
samples_no_Deletion = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(nygc_spinal_cord.vcfannotate.deletion,sample)] 
samples_no_Deletion # 0 samples with 0 Deletions

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
nygc_spinal_cord.vcfannotate.deletion.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.vcfannotate.deletion, family = poisson)
summ(nygc_spinal_cord.vcfannotate.deletion.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.             p
# (Intercept)                             5.837663e+00 2.616149e-02 223.139593  0.000000e+00
# CONDITIONALS                            2.645573e-02 5.774111e-03   4.581784  4.610265e-06
plot_summs(nygc_spinal_cord.vcfannotate.deletion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
nygc_spinal_cord.vcfannotate.deletion.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate.deletion, family = poisson) # cant plot rcs spline
partialize(nygc_spinal_cord.vcfannotate.deletion.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(nygc_spinal_cord.vcfannotate.deletion.fitg_plot, vars = "CONDITION") %>% my_summarise(n, CONDITION)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n   sum_n   n_n  se_n
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>   <dbl> <int> <dbl>
# 1 CTRL        704.     695. 112.   643.  753.  487. 1101.  40148.    57 14.9 
# 2 ALS         710.     704.  86.7  647.  767.  537. 1039. 151870.   214  5.92
nygc_spinal_cord.vcfannotate.deletion.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 3600, xmin = 1, xmax = 2)
nygc_spinal_cord.vcfannotate.deletion.violin <- violin_plot(partialize(nygc_spinal_cord.vcfannotate.deletion.fitg_plot, vars = "CONDITION"), color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(2285.,3600), dpi = 300, ylabel = "Adjusted Number of Deletions", title = "Postmortem: Deletion") +
  stat_pvalue_manual(nygc_spinal_cord.vcfannotate.deletion.stats, tip.length = 0.01, size = 4)
# nygc_spinal_cord.vcfannotate.deletion.violin


# Facet voilin
nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind = bind_rows(mutate(partialize(nygc_spinal_cord.vcfannotate.totals.fitg_plot, vars = "CONDITION"), type = "All mutations",n=total), mutate(partialize(nygc_spinal_cord.vcfannotate.snp.fitg_plot, vars = "CONDITION"), type = "SNV"),
mutate(partialize(nygc_spinal_cord.vcfannotate.insertion.fitg_plot, vars = "CONDITION"), type = "Insertion"), mutate(partialize(nygc_spinal_cord.vcfannotate.deletion.fitg_plot, vars = "CONDITION"), type = "Deletion")) %>% mutate(type = factor(type, levels = c("All mutations", "SNV", "Insertion", "Deletion")))
nygc_spinal_cord.vcfannotate.bind.stats = tibble(facet = c("All mutations", "SNV", "Insertion", "Deletion"), group1 = "CTRL", group2 = "ALS",  p.signif = c("****", "****","****","***"),
                                          y.position = c(18192.,12960., 5800,960), xmin = 1, xmax = 2)
nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind.filt = nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind %>% filter((type == "All mutations" & n < 20000.) |(type == "Deletion" & n < 1000) |(type == "Insertion" & n < 6000) |(type == "SNV" & n < 14000))
nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind.violin <- violin_plot(nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind.filt, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, facet_by = "type",facet_order = c("All mutations", "SNV", "Insertion", "Deletion"), dpi = 300, ylabel = "Adjusted Number of Mutations", title = "Postmortem: Somatic Mutations", facet_ncol = 4) +
  stat_pvalue_manual(nygc_spinal_cord.vcfannotate.bind.stats, tip.length = 0.01, size = 4)
nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind.violin

```


#### Genetic subgroups

```{r}
nygc_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
nygc.rnavar.multiqc_star.txt = read_tsv(here(nygc_path,"rnavar/reports/multiqc_data/multiqc_star.txt")) %>% clean_names()
nygc_spinal_cord.vcfannotate = read_csv(here(nygc_path,"rnavar/nygc_spinal_cord.vcfannotate.dbsnp.redi.filtered.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(sample) %>% add_tally(n, name = "total") %>% ungroup %>% 
  select(sample, n, type, variant_type, total, everything()) %>%  # add a column of sum of variants per sample
  left_join(nygc.rnavar.multiqc_star.txt) # left_join multiqc stats
nygc_spinal_cord_metadata %>% count(condition) # 214 ALS, 57 CTRL
nygc_spinal_cord.vcfannotate.mutation = nygc_spinal_cord.vcfannotate %>%  distinct(sample, .keep_all = TRUE) %>% # keep 1 row per sample (using total number)
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "C9orf72", "SOD1"))) %>% #  mutation == "tardbp" ~ "TARDBP", "TARDBP", mutation == "fus" ~ "FUS","FUS", 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  select(sample, total, genetic_group, CONDITION, total_reads, library_prep, mutation, age_at_death, variant_type)
nygc_spinal_cord.metadata_genetic_subgroup = nygc_spinal_cord_metadata %>% filter(mutation %in%  c("ctrl", "sporadic", "sod1", "c9orf72")) #"fus", "tardbp", 
nygc_spinal_cord.vcfannotate.samples_no_variants = nygc_spinal_cord.metadata_genetic_subgroup$sample[!nygc_spinal_cord.metadata_genetic_subgroup$sample %in% nygc_spinal_cord.vcfannotate.mutation$sample] # 0 i.e. all samples have at least 1 variant
nygc_spinal_cord.vcfannotate.samples_no_variants # 0


# All variants
nygc_spinal_cord.vcfannotate.mutation.fitg = glm(total ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.vcfannotate.mutation, family=poisson)
summ(nygc_spinal_cord.vcfannotate.mutation.fitg, digits = 5)$coeftable 
# (Intercept)                             8.801898e+00 6.569161e-03 1339.881642  0.000000e+00
# genetic_groupSporadic                   4.021629e-02 1.357600e-03   29.623086 7.537128e-193
# genetic_groupC9orf72                    7.168656e-02 1.924938e-03   37.240970 1.483548e-303
# genetic_groupSOD1                       1.637490e-02 4.374619e-03    3.743161  1.817200e-04
plot_summs(nygc_spinal_cord.vcfannotate.mutation.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


### SNV
nygc_spinal_cord.vcfannotate.mutation.SNV = nygc_spinal_cord.vcfannotate %>% filter(variant_type == "SNP") %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation %in% c("ctrl","iso") ~ "CTRL")) %>% drop_na(genetic_group)
nygc_spinal_cord.vcfannotate.SNV = bind_rows(nygc_spinal_cord.vcfannotate.mutation.SNV, mutate(filter(nygc_spinal_cord.vcfannotate, variant_type == "SNP", CONDITION!="CTRL"),genetic_group="pan ALS")) %>% mutate(genetic_group = factor(genetic_group, levels = c("CTRL", "pan ALS", "Sporadic", "C9orf72", "SOD1"))) %>% select(sample, total, genetic_group, CONDITION, n, library_prep, mutation, age_at_death, variant_type, total_reads)

nygc_spinal_cord.vcfannotate.samples_no_variants = nygc_spinal_cord.metadata_genetic_subgroup$sample[!nygc_spinal_cord.metadata_genetic_subgroup$sample %in% nygc_spinal_cord.vcfannotate.SNV$sample] # 0 i.e. all samples have at least 1 variant
nygc_spinal_cord.vcfannotate.samples_no_variants # 0

nygc_spinal_cord.vcfannotate.SNV.fitg = glm(n ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.vcfannotate.SNV, family = poisson)
summ(nygc_spinal_cord.vcfannotate.SNV.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.      z val.             p
# (Intercept)                             8.320256e+00 7.747228e-03 1073.965550  0.000000e+00
# genetic_groupSporadic                   3.316424e-02 1.607642e-03   20.629118  1.503684e-94
# genetic_groupC9orf72                    7.685806e-02 2.275476e-03   33.776695 4.337326e-250
# genetic_groupSOD1                       2.481177e-03 5.177834e-03    0.479192  6.318020e-01
plot_summs(nygc_spinal_cord.vcfannotate.SNV.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


### Insertions
nygc_spinal_cord.vcfannotate.mutation.insertion = nygc_spinal_cord.vcfannotate %>% filter(variant_type == "Insertion") %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation %in% c("ctrl","iso") ~ "CTRL")) %>% drop_na(genetic_group)
nygc_spinal_cord.vcfannotate.insertion = bind_rows(nygc_spinal_cord.vcfannotate.mutation.insertion, mutate(filter(nygc_spinal_cord.vcfannotate, variant_type == "Insertion", CONDITION!="CTRL"),genetic_group="pan ALS")) %>% mutate(genetic_group = factor(genetic_group, levels = c("CTRL", "pan ALS", "Sporadic", "C9orf72", "SOD1"))) %>% select(sample, total, genetic_group, CONDITION, n, library_prep, mutation, age_at_death, variant_type, total_reads)

nygc_spinal_cord.vcfannotate.samples_no_variants = nygc_spinal_cord.metadata_genetic_subgroup$sample[!nygc_spinal_cord.metadata_genetic_subgroup$sample %in% nygc_spinal_cord.vcfannotate.insertion$sample] # 0 i.e. all samples have at least 1 variant
nygc_spinal_cord.vcfannotate.samples_no_variants # 0

nygc_spinal_cord.vcfannotate.insertion.fitg = glm(n ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.vcfannotate.insertion, family = poisson)
summ(nygc_spinal_cord.vcfannotate.insertion.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.             p
# (Intercept)                             7.765267e+00 1.384844e-02 560.732111  0.000000e+00
# genetic_groupSporadic                   6.320541e-02 2.804355e-03  22.538303 1.748938e-112
# genetic_groupC9orf72                    6.300419e-02 3.996991e-03  15.762905  5.599851e-56
# genetic_groupSOD1                       4.009367e-02 9.170021e-03   4.372255  1.229696e-05
plot_summs(nygc_spinal_cord.vcfannotate.insertion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


### Deletions
nygc_spinal_cord.vcfannotate.mutation.Deletion = nygc_spinal_cord.vcfannotate %>% filter(variant_type == "Deletion") %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation %in% c("ctrl","iso") ~ "CTRL")) %>% drop_na(genetic_group)
nygc_spinal_cord.vcfannotate.Deletion = bind_rows(nygc_spinal_cord.vcfannotate.mutation.Deletion, mutate(filter(nygc_spinal_cord.vcfannotate, variant_type == "Deletion", CONDITION!="CTRL"),genetic_group="pan ALS")) %>% mutate(genetic_group = factor(genetic_group, levels = c("CTRL", "pan ALS", "Sporadic", "C9orf72", "SOD1"))) %>% select(sample, total, genetic_group, CONDITION, n, library_prep, mutation, age_at_death, variant_type, total_reads)

nygc_spinal_cord.vcfannotate.samples_no_variants = nygc_spinal_cord.metadata_genetic_subgroup$sample[!nygc_spinal_cord.metadata_genetic_subgroup$sample %in% nygc_spinal_cord.vcfannotate.Deletion$sample] # 0 i.e. all samples have at least 1 variant
nygc_spinal_cord.vcfannotate.samples_no_variants # 0

nygc_spinal_cord.vcfannotate.Deletion.fitg = glm(n ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.vcfannotate.Deletion, family = poisson)
summ(nygc_spinal_cord.vcfannotate.Deletion.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.             p
# (Intercept)                             5.834998e+00 2.828224e-02 206.313128  0.000000e+00
# genetic_groupSporadic                   2.415765e-02 5.962906e-03   4.051322  5.092914e-05
# genetic_groupC9orf72                    2.977393e-02 8.430833e-03   3.531553  4.131272e-04
# genetic_groupSOD1                       6.110906e-02 1.817627e-02   3.362024  7.737345e-04
plot_summs(nygc_spinal_cord.vcfannotate.Deletion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


nygc_spinal_cord.vcfannotate.mutation.fitg.forrest_plot = plot_summs(nygc_spinal_cord.vcfannotate.SNV.fitg, nygc_spinal_cord.vcfannotate.insertion.fitg, nygc_spinal_cord.vcfannotate.Deletion.fitg, 
           model.names = c("SNV","Insertion","Deletion"),
           coefs = c("pan ALS"="genetic_grouppan ALS","Sporadic"="genetic_groupSporadic","C9orf72"="genetic_groupC9orf72","SOD1"="genetic_groupSOD1"), plot.distributions = TRUE, rescale.distributions = TRUE, 
           colors = c("dodgerblue2","firebrick2","forestgreen","gold2"), legend.title	= "",exp = FALSE,point.shape = FALSE, point.size = 1) + 
  coord_cartesian(xlim = c(-0.1,0.1)) + #ggtitle("Postmortem") +
  theme_classic() + theme(legend.text=element_text(size=8), legend.spacing = unit(0.01, 'cm'), legend.position = "top", axis.text = element_text(colour = "black", size = 8), axis.title = element_text(colour = "black", size = 9, face ="plain"), strip.text.y = element_text(colour = "black", size = 7), plot.title = element_text(hjust = 0.5, size = 9)) + guides(color=guide_legend(keywidth=0.1, keyheight=0.15, default.unit="inch")) +
  annotation_custom(grid::textGrob(label = paste0("increase in ALS"), y = unit(0.1, "npc"), x = unit(0.8, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 0)) + 
  annotation_custom(grid::textGrob(label = paste0("increase in CTRL"), y = unit(0.1, "npc"), x = unit(0.2, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 0)) + 
  annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), x = c(0.95,0.65), y = unit(0.05, "npc"))) +
  annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), x = c(0.05,0.35), y = unit(0.05, "npc")))
nygc_spinal_cord.vcfannotate.mutation.fitg.forrest_plot



# OLD VIOLINS
# # generalised linear model spline
# 
# ## sporadic
# nygc_spinal_cord.vcfannotate.mutation.sporadic = filter(nygc_spinal_cord.vcfannotate.mutation, genetic_group %in% c("Sporadic", "CTRL"))
# nygc_spinal_cord.vcfannotate.mutation.sporadic.fitg = glm(total ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.vcfannotate.mutation.sporadic, family=poisson)
# summ(nygc_spinal_cord.vcfannotate.mutation.sporadic.fitg, digits = 5)$coeftable 
# #                                                Est.         S.E.      z val.            p
# # (Intercept)                             1.065596e+01 3.059188e-03 3483.263893 0.000000e+00
# # genetic_groupSporadic                   8.967134e-04 6.482253e-04    1.383336 1.665618e-01
# nygc_spinal_cord.vcfannotate.mutation.sporadic.fitg_plot = glm(total ~ genetic_group + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate.mutation.sporadic, family = poisson) # cant plot rcs spline
# partialize(nygc_spinal_cord.vcfannotate.mutation.sporadic.fitg_plot, vars = "genetic_group") %>% my_summarise(total, genetic_group)
# # genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total se_total
# #   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>    <dbl>
# # 1 CTRL              62181.       61674.    4427.   59209.   64874.    53281.    77443.  3544334.      57     586.
# # 2 Sporadic          61541.       61868.    3320.   58905.   63972.    52645.    69518.  9908163.     161     262.
# plot_summs(nygc_spinal_cord.vcfannotate.mutation.sporadic.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
# 
# ## c9orf72
# nygc_spinal_cord.vcfannotate.mutation.c9orf72 = filter(nygc_spinal_cord.vcfannotate.mutation, genetic_group %in% c("C9orf72", "CTRL"))
# nygc_spinal_cord.vcfannotate.mutation.c9orf72.fitg = glm(data=nygc_spinal_cord.vcfannotate.mutation.c9orf72, total ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), family=poisson)
# summ(nygc_spinal_cord.vcfannotate.mutation.c9orf72.fitg, digits = 5)$coeftable 
# #                                          Est.         S.E.      z val.            p
# # (Intercept)                             1.076574e+01 5.139619e-03 2094.657134 0.000000e+00
# # genetic_groupC9orf72                   -4.352365e-03 9.640555e-04   -4.514641 6.342400e-06
# nygc_spinal_cord.vcfannotate.mutation.c9orf72.fitg_plot = glm(total ~ genetic_group + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate.mutation.c9orf72, family = poisson) # cant plot rcs spline
# partialize(nygc_spinal_cord.vcfannotate.mutation.c9orf72.fitg_plot, vars = "genetic_group") %>% my_summarise(total, genetic_group)
# # genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total se_total
# #   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>    <dbl>
# # 1 CTRL              62316.       61944.    4289.   59505.   64933.    52864.    76836.  3552039.      57     568.
# # 2 C9orf72           61332.       61311.    2673.   59607.   63281.    55617.    66717.  2207958.      36     445.
# plot_summs(nygc_spinal_cord.vcfannotate.mutation.c9orf72.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
# 
# ## SOD1
# nygc_spinal_cord.vcfannotate.mutation.sod1 = filter(nygc_spinal_cord.vcfannotate.mutation, genetic_group %in% c("SOD1", "CTRL"))
# nygc_spinal_cord.vcfannotate.mutation.sod1.fitg = glm(data=nygc_spinal_cord.vcfannotate.mutation.sod1, total ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), family=poisson)
# summ(nygc_spinal_cord.vcfannotate.mutation.sod1.fitg, digits = 5)$coeftable 
# #                                                Est.         S.E.        z val.            p
# # (Intercept)                             1.066487e+01 5.679772e-03 1877.69335429 0.000000e+00
# # genetic_groupSOD1                       1.994022e-02 2.143238e-03    9.30378378 1.355346e-20
# nygc_spinal_cord.vcfannotate.mutation.sod1.fitg_plot = glm(total ~ genetic_group + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate.mutation.sod1, family = poisson) # cant plot rcs spline
# partialize(nygc_spinal_cord.vcfannotate.mutation.sod1.fitg_plot, vars = "genetic_group") %>% my_summarise(total, genetic_group)
# # genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total se_total
# #   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>    <dbl>
# # 1 CTRL              62478.       62634.    4298.   59652.   65088.    52212.    78060.  3561248.      57     569.
# # 2 SOD1              61599.       58478.    6718.   58381.   60219.    57435.    73482.   307995.       5    3005.
# plot_summs(nygc_spinal_cord.vcfannotate.mutation.sod1.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
# 
# # plot fitted values
# nygc_spinal_cord.vcfannotate.mutation.bind= bind_rows(partialize(nygc_spinal_cord.vcfannotate.mutation.sporadic.fitg_plot, vars = "genetic_group"), 
#                                                filter(partialize(nygc_spinal_cord.vcfannotate.mutation.c9orf72.fitg_plot, vars = "genetic_group"), genetic_group!="CTRL"), 
#                                                filter(partialize(nygc_spinal_cord.vcfannotate.mutation.sod1.fitg_plot, vars = "genetic_group"), genetic_group!="CTRL"))
# nygc_spinal_cord.vcfannotate.mutation.bind %>% my_summarise(total, genetic_group)
# # genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total se_total
# #   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>    <dbl>
# # 1 CTRL              62181.       61674.    4427.   59209.   64874.    53281.    77443.  3544334.      57     586.
# # 2 Sporadic          61541.       61868.    3320.   58905.   63972.    52645.    69518.  9908163.     161     262.
# # 3 C9orf72           61332.       61311.    2673.   59607.   63281.    55617.    66717.  2207958.      36     445.
# # 4 SOD1              61599.       58478.    6718.   58381.   60219.    57435.    73482.   307995.       5    3005.
# 
# # Violin total variant events
# nygc_spinal_cord.vcfannotate.mutation.stat <- nygc_spinal_cord.vcfannotate.mutation.bind %>% wilcox_test(total ~ genetic_group) %>% add_significance() %>% add_xy_position() %>% filter(group1 == "CTRL") %>%
#   mutate(p.adj.signif = case_when(group2 %in% c("Sporadic") ~ "ns", group2 %in%  c("SOD1","C9orf72") ~ "****", TRUE ~ p.adj.signif), y.position = case_when(group2 == "Sporadic" ~ 72000, group2 == "C9orf72" ~ 70000, group2 == "SOD1" ~ 72000, TRUE ~ y.position))
# nygc_spinal_cord.vcfannotate.mutation.stat
# # .y.   group1 group2      n1    n2 statistic          p      p.adj p.adj.signif y.position groups        xmin  xmax
# #   <chr> <chr>  <chr>    <int> <int>     <dbl>      <dbl>      <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# # 1 total CTRL   Sporadic   126   200      8580 0.00000123 0.00000738 ****              63000 <chr [2]>        1     2
# # 2 total CTRL   C9orf72    126    21      1302 0.91       1          ****              65000 <chr [2]>        1     3
# # 3 total CTRL   SOD1       126     8       532 0.796      1          ns                67000 <chr [2]>        1     4
# nygc_spinal_cord.vcfannotate.mutation.violin <- violin_plot(nygc_spinal_cord.vcfannotate.mutation.bind, color_by = "genetic_group", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(52645.,72000), dpi = 300, ylabel = "Adjusted Number of Variants", title = "Postmortem: Genetic backgrounds") + stat_pvalue_manual(nygc_spinal_cord.vcfannotate.mutation.stat, tip.length = 0.01, size = 4, hide.ns = TRUE)
# nygc_spinal_cord.vcfannotate.mutation.violin


```





## Fusions

```{r}
# fusion_schematic = magick::image_read(here(proj_path,"rnafusion/gene fusion schematic.png"))
# fusion_schematic.gg = ggdraw() + draw_image(fusion_schematic)
# fusion_schematic.gg

```

### iPSNs

```{r}
# read_starfusion = function(metadata){
#   print("reading in STAR Fusion files with read_tsv")
#   starfusion.tsv = metadata$starfusion_path %>% map(read_tsv, col_types = list(JunctionReadCount="n",SpanningFragCount="n",est_J="n",est_S="n",FFPM="n",LeftBreakEntropy="n",RightBreakEntropy="n",.default="c"))
#   names(starfusion.tsv) = metadata$dataset_sample # needs to be unique names e.g. dataset_sample
#   starfusion_tab <- map_dfr(starfusion.tsv, bind_rows, .id = "dataset_sample") %>% rename(fusion_name = '#FusionName') %>% clean_names() %>%
#     mutate(left_chr = str_split_fixed(left_breakpoint,":",3)[,1], left_coord = str_split_fixed(left_breakpoint,":",3)[,2], left_strand = str_split_fixed(left_breakpoint,":",3)[,3],
#            right_chr = str_split_fixed(right_breakpoint,":",3)[,1], right_coord = str_split_fixed(right_breakpoint,":",3)[,2], right_strand = str_split_fixed(right_breakpoint,":",3)[,3],
#            fusion_type = case_when(grepl("INTERCHROMOSOMAL", annots) ~ "Inter\nChromosomal", grepl("OVERLAP", annots) ~ "Overlapping\nNeighbours", grepl("NEIGHBORS", annots) ~ "Neighbours", grepl("LOCAL_INVERSION", annots) ~ "Local\nInversion", grepl("LOCAL_REARRANGEMENT", annots) ~ "Local\nRearrangement", TRUE ~ "Distant\nProximity"),
#            breakpoints = paste(left_breakpoint, right_breakpoint, sep="_")) %>%
#     left_join(metadata, by = "dataset_sample")
#   return(starfusion_tab)
# }
# ipsc_mn_als_datasets.paired_end.metadata = ipsc_mn_als_datasets.paired_end.metadata %>% mutate(starfusion_path = paste0("/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/rnafusion/starfusion/",dataset_sample,".starfusion.abridged.coding_effect.tsv"))
# ipsc_mn_als_datasets.starfusion = read_starfusion(ipsc_mn_als_datasets.paired_end.metadata)
# ipsc_mn_als_datasets.starfusion
# write_csv(ipsc_mn_als_datasets.starfusion, here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv"))
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
ipsc_mn_als_datasets.starfusion = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>%
  left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset)
ipsc_mn_als_datasets.paired_end.metadata %>% count(condition) # 306 ALS, 90 CTRL
ipsc_mn_als_datasets.paired_end.metadata %>% count(mutation)

### ALS vs CTRL unique fusion events
ipsc_mn_als_datasets.starfusion.n = ipsc_mn_als_datasets.starfusion %>% 
  add_count(dataset_sample) %>% # count number of fusions per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, dataset, total_reads, age_at_sample_collection) #keep 1 row per sample

# add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.samples_no_fusions = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% ipsc_mn_als_datasets.starfusion.n$dataset_sample] 
ipsc_mn_als_datasets.starfusion.samples_no_fusions # "smith_tardbp_1" "smith_ctrl_1"  
ipsc_mn_als_datasets.paired_end.metadata.no_fusions = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.samples_no_fusions) %>% 
  left_join(multiqc_rseqc_read_distribution.tsv) %>%
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), n = 0) %>% select(dataset_sample, n, CONDITION, dataset, total_reads)
ipsc_mn_als_datasets.starfusion.n = bind_rows(ipsc_mn_als_datasets.starfusion.n, ipsc_mn_als_datasets.paired_end.metadata.no_fusions)

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
ipsc_mn_als_datasets.starfusion.n.fitg = glm(n ~ CONDITION + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=ipsc_mn_als_datasets.starfusion.n, family = poisson)
summ(ipsc_mn_als_datasets.starfusion.n.fitg, digits = 5)$coeftable 
#                                                                       Est.         S.E.    z val.            p
# (Intercept)                                                     1.770149e+00 3.819275e-01  4.634776 3.573236e-06
# CONDITIONALS                                                    1.205603e-01 4.349381e-02  2.771895 5.573094e-03
ipsc_mn_als_datasets.starfusion.n.fitg_plot = glm(n ~ CONDITION + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.n, family = poisson) # cant plot rcs spline
partialize(ipsc_mn_als_datasets.starfusion.n.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(ipsc_mn_als_datasets.starfusion.n.fitg_plot, vars = "CONDITION") %>% my_summarise(n, CONDITION)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# 1 CTRL        9.68     7.94  5.66  6.13  11.5  3.60  32.2  861.    89
# 2 ALS        13.9      9.57 22.9   7.20  12.9  4.35 316.  4243.   305
ipsc_mn_als_datasets.starfusion.n.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 29, xmin = 1, xmax = 2)
ipsc_mn_als_datasets.starfusion.violin <- violin_plot(partialize(ipsc_mn_als_datasets.starfusion.n.fitg_plot, vars = "CONDITION"), color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(3.60,30), dpi = 300, ylabel = "Adjusted Number of Fusions", title = "iPSMN: Gene Fusions") +
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.n.stats, tip.length = 0.01, size = 4)
ipsc_mn_als_datasets.starfusion.violin

effect_plot(ipsc_mn_als_datasets.starfusion.n.fitg_plot, pred = CONDITION, plot.points = TRUE, interval = TRUE, jitter = 0.2, partial.residuals = TRUE)
effect_plot(ipsc_mn_als_datasets.starfusion.n.fitg_plot, pred = total_reads, interval = TRUE, plot.points = TRUE,  jitter = 0) # read effect
effect_plot(ipsc_mn_als_datasets.starfusion.n.fitg_plot, pred = age_at_sample_collection, interval = TRUE, plot.points = TRUE,  jitter = 0)
plot_summs(ipsc_mn_als_datasets.starfusion.n.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
fit <- glm(n ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), family = poisson, data=ipsc_mn_als_datasets.starfusion.n)
predicted <- predict(fit, type="response")
ggplot(cbind(ipsc_mn_als_datasets.starfusion.n, predicted=predicted), aes(x=total_reads, y=predicted, colour= CONDITION, group = interaction(CONDITION))) + geom_line() + theme_oz()


# # PCA of FFPM
# ipsc_mn_als_datasets.starfusion.vsd = ipsc_mn_als_datasets.starfusion %>% mutate(breakpoints = paste(left_breakpoint, right_breakpoint, sep = "_")) %>% select(breakpoints, dataset_sample, ffpm) %>% pivot_wider(names_from = dataset_sample, values_from = ffpm) %>% replace(is.na(.),0)  # samples = dataset_sample, rows = fusion_name, values = ffpm
# ipsc_mn_als_datasets.starfusion.vsd
# library(ggfortify)
# ipsc_mn_als_datasets.starfusion.pca <- prcomp(select(ipsc_mn_als_datasets.starfusion.vsd,-breakpoints), scale. = TRUE)
# autoplot(ipsc_mn_als_datasets.starfusion.pca, data = ipsc_mn_als_datasets.starfusion, colour = "condition")

# # break entropy
# ipsc_mn_als_datasets.starfusion %>% glimpse
# ipsc_mn_als_datasets.starfusion.violin <- violin_plot(ipsc_mn_als_datasets.starfusion, color_by = "CONDITION", continuous = "left_break_entropy", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, stats_test = "t_test", arrow_labels = FALSE, ylims = c(0.9,2.1), dpi = 300, ylabel = "Number of Unique Fusion Events")
# ipsc_mn_als_datasets.starfusion.violin
# ipsc_mn_als_datasets.starfusion.violin <- violin_plot(ipsc_mn_als_datasets.starfusion, color_by = "CONDITION", continuous = "right_break_entropy", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, stats_test = "t_test", arrow_labels = FALSE, ylims = c(0.9,2.1), dpi = 300, ylabel = "Number of Unique Fusion Events")
# ipsc_mn_als_datasets.starfusion.violin
```


#### Genetic subgroups


```{r}
ipsc_mn_als_datasets.starfusion.csv = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) 

# pan ALS
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(!mutation %in% c("ctrl","iso"))

ipsc_mn_als_datasets.starfusion.pan_als = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>%
  left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.pan_als

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% count(condition) # 306 ALS, 90 CTRL
ipsc_mn_als_datasets.starfusion.pan_als.n = ipsc_mn_als_datasets.starfusion.pan_als %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, mutation, dataset, total_reads, age_at_sample_collection) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.pan_als.n %>% count(CONDITION)

ipsc_mn_als_datasets.starfusion.pan_als.fitg = glm(n ~ CONDITION + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.pan_als.n, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.pan_als.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.    z val.            p
# (Intercept)                                                     1.779547e+00 3.799520e-01  4.683611 2.818650e-06
# CONDITIONALS                                                    1.205463e-01 4.349373e-02  2.771578 5.578531e-03
plot_summs(ipsc_mn_als_datasets.starfusion.pan_als.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction



# Sporadic
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "sporadic")

ipsc_mn_als_datasets.starfusion.sporadic = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>%
  filter(dataset %in% c("answerals","neurolincs.diMN"), mutation %in% c("ctrl","iso","sporadic")) %>% 
  left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.sporadic

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","neurolincs.diMN"), mutation %in% c("ctrl","iso","sporadic")) %>% count(condition) # 208 ALS, 50 CTRL
ipsc_mn_als_datasets.starfusion.sporadic.n = ipsc_mn_als_datasets.starfusion.sporadic %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group, mutation, dataset, total_reads, age_at_sample_collection) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.sporadic.n %>% count(genetic_group)

ipsc_mn_als_datasets.starfusion.sporadic.fitg = glm(n ~ genetic_group + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.sporadic.n, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.sporadic.fitg, digits = 5)$coeftable 
#                                                                       Est.         S.E.     z val.            p
# (Intercept)                                                     2.422762e+00 3.197400e-01  7.5772887 3.528511e-14
# genetic_groupSporadic                                           5.801951e-02 5.137953e-02  1.1292341 2.587991e-01
plot_summs(ipsc_mn_als_datasets.starfusion.sporadic.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

# ipsc_mn_als_datasets.starfusion.sporadic.fitg_plot = glm(n ~ genetic_group + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.sporadic.n, family = poisson)
# partialize(ipsc_mn_als_datasets.starfusion.sporadic.fitg_plot, vars = "genetic_group") %>% my_summarise(n, genetic_group)
# # genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# # 1 CTRL            11.1     10.0  4.06  7.61  13.1  6.54  20.9  556.    50
# # 2 Sporadic        12.7     10.4  8.82  8.18  13.5  4.84  97.6 2634.   208
# 
# # Violin
# ipsc_mn_als_datasets.starfusion.sporadic.n.stats = tibble(group1 = "CTRL", group2 = "Sporadic", p.signif = "ns", y.position = 40, xmin = 1, xmax = 2)
# ipsc_mn_als_datasets.starfusion.sporadic.violin <- violin_plot(partialize(ipsc_mn_als_datasets.starfusion.sporadic.fitg_plot, vars = "genetic_group"), color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "firebrick2"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(4,40), dpi = 300, ylabel = "Adjusted Number of Fusions") +
#   stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.sporadic.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
# ipsc_mn_als_datasets.starfusion.sporadic.violin



### c9orf72
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "c9orf72")

ipsc_mn_als_datasets.starfusion.c9orf72 = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>%
  filter(dataset %in% c("answerals","neurolincs.diMN","catanese","dafinca.c9orf72","neurolincs.iMN","sommer"), mutation %in% c("ctrl","iso","c9orf72")) %>% 
  left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.c9orf72

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","neurolincs.diMN","catanese","dafinca.c9orf72","neurolincs.iMN","sommer"), mutation %in% c("ctrl","iso","c9orf72")) %>% count(condition) # 53 ALS, 76 CTRL

ipsc_mn_als_datasets.starfusion.c9orf72.n = ipsc_mn_als_datasets.starfusion.c9orf72 %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group,  mutation, dataset, total_reads, age_at_sample_collection) #keep 1 row per sample
 # add samples with 0 fusions

ipsc_mn_als_datasets.starfusion.c9orf72.fitg = glm(n ~ genetic_group + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.c9orf72.n, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.c9orf72.fitg, digits = 5)$coeftable 
# Coefficients:
#                                                                  Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                                                     2.469e+00  5.985e-01   4.126 3.70e-05 ***
# genetic_groupC9orf72                                            2.013e-01  6.392e-02   3.149  0.00164 ** 
plot_summs(ipsc_mn_als_datasets.starfusion.c9orf72.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

# ipsc_mn_als_datasets.starfusion.c9orf72.fitg_plot = glm(n ~ genetic_group + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.c9orf72.n, family = poisson)
# partialize(ipsc_mn_als_datasets.starfusion.c9orf72.fitg_plot, vars = "genetic_group") %>% my_summarise(n, genetic_group)
# # genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# # 1 CTRL            14.2     12.5  6.70  9.17  17.3  6.57  40.6 1079.    76
# # 2 C9orf72         16.7     14.3  9.52 11.0   19.0  6.94  61.9  884.    53
# 
# # Violin
# ipsc_mn_als_datasets.starfusion.c9orf72.n.stats = tibble(group1 = "CTRL", group2 = "c9orf72", p.signif = "ns", y.position = 40, xmin = 1, xmax = 2)
# ipsc_mn_als_datasets.starfusion.c9orf72.violin <- violin_plot(partialize(ipsc_mn_als_datasets.starfusion.c9orf72.fitg_plot, vars = "genetic_group"), color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "forestgreen"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(4,40), dpi = 300, ylabel = "") +
#   stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.c9orf72.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
# ipsc_mn_als_datasets.starfusion.c9orf72.violin



###TARDBP
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "tardbp")

ipsc_mn_als_datasets.starfusion.tardbp = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>%
  filter(dataset %in% c("answerals","dafinca.tardbp","smith"), mutation %in% c("ctrl","iso","tardbp")) %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.tardbp

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","dafinca.tardbp","smith"), mutation %in% c("ctrl","iso","tardbp")) %>% count(condition) # 48 ALS, 10 CTRL

ipsc_mn_als_datasets.starfusion.tardbp.n = ipsc_mn_als_datasets.starfusion.tardbp %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group, mutation, dataset, total_reads, age_at_sample_collection) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.paired_end.metadata.tardbp = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","dafinca.tardbp","smith"), mutation %in% c("ctrl","iso","tardbp"))
ipsc_mn_als_datasets.starfusion.samples_no_fusions.tardbp = ipsc_mn_als_datasets.paired_end.metadata.tardbp$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.tardbp$dataset_sample %in% ipsc_mn_als_datasets.starfusion.tardbp.n$dataset_sample] 
ipsc_mn_als_datasets.starfusion.samples_no_fusions.tardbp # "smith_tardbp_1" "smith_ctrl_1"  

ipsc_mn_als_datasets.paired_end.metadata.no_fusions.tardbp = ipsc_mn_als_datasets.paired_end.metadata.tardbp %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.samples_no_fusions.tardbp) %>% 
  left_join(multiqc_rseqc_read_distribution.tsv) %>%
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), n = 0, genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP"))) %>% select(dataset_sample, n, CONDITION, mutation, dataset, total_reads, genetic_group)
ipsc_mn_als_datasets.starfusion.tardbp.n = bind_rows(ipsc_mn_als_datasets.starfusion.tardbp.n, ipsc_mn_als_datasets.paired_end.metadata.no_fusions.tardbp)

ipsc_mn_als_datasets.starfusion.tardbp.fitg = glm(n ~ genetic_group + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.tardbp.n, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.tardbp.fitg, digits = 5)$coeftable 
# Coefficients:
#                                                                  Estimate Std. Error z value Pr(>|z|)   
# (Intercept)                                                     3.758e+00  1.683e+00   2.233  0.02552 * 
# genetic_groupTARDBP                                             2.639e-01  1.921e-01   1.374  0.16952   
plot_summs(ipsc_mn_als_datasets.starfusion.tardbp.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

# ipsc_mn_als_datasets.starfusion.tardbp.fitg_plot = glm(n ~ genetic_group + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.tardbp.n, family = poisson)
# partialize(ipsc_mn_als_datasets.starfusion.tardbp.fitg_plot, vars = "genetic_group") %>% my_summarise(n, genetic_group)
# # genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# # 1 CTRL            12.0     10.7  5.27  8.23  13.9  6.81  34.5  563.    47
# # 2 TARDBP          13.5     12.9  4.72 11.0   13.0  8.15  24.2  122.     9
# 
# # Violin
# ipsc_mn_als_datasets.starfusion.tardbp.n.stats = tibble(group1 = "CTRL", group2 = "tardbp", p.signif = "ns", y.position = 30, xmin = 1, xmax = 2)
# ipsc_mn_als_datasets.starfusion.tardbp.violin <- violin_plot(partialize(ipsc_mn_als_datasets.starfusion.tardbp.fitg_plot, vars = "genetic_group"), color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "purple"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(4,25), dpi = 300, ylabel = "") +
#   stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.tardbp.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
# ipsc_mn_als_datasets.starfusion.tardbp.violin



### fus
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "fus")

ipsc_mn_als_datasets.starfusion.fus = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>%
  filter(dataset %in% c("catanese","desantis"), mutation %in% c("ctrl","iso","fus")) %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.fus

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("catanese","desantis"), mutation %in% c("ctrl","iso","fus")) %>% count(condition) # 9 ALS, 9 CTRL

ipsc_mn_als_datasets.starfusion.fus.n = ipsc_mn_als_datasets.starfusion.fus %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group, mutation, dataset, total_reads, age_at_sample_collection) #keep 1 row per sample
 # add samples with 0 fusions

ipsc_mn_als_datasets.starfusion.fus.fitg = glm(n ~ genetic_group + dataset + rms::rcs(total_reads, 3), data = ipsc_mn_als_datasets.starfusion.fus.n, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.fus.fitg, digits = 5)$coeftable 
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)  
# (Intercept)                          -2.586e+00  1.819e+00  -1.421   0.1552  
# CONDITIONALS                          1.124e-01  1.842e-01   0.610   0.5416 
plot_summs(ipsc_mn_als_datasets.starfusion.fus.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

# ipsc_mn_als_datasets.starfusion.fus.fitg_plot = glm(n ~ genetic_group + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.fus.n, family = poisson)
# partialize(ipsc_mn_als_datasets.starfusion.fus.fitg_plot, vars = "genetic_group") %>% my_summarise(n, genetic_group)
# # genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# # 1 CTRL            3.90     3.29  1.46  2.77  4.41  2.52  6.34  35.1     9
# # 2 FUS             4.20     4.16  1.20  3.02  5.21  2.32  5.64  37.8     9
# 
# # Violin
# ipsc_mn_als_datasets.starfusion.fus.n.stats = tibble(group1 = "CTRL", group2 = "fus", p.signif = "ns", y.position = 35, xmin = 1, xmax = 2)
# ipsc_mn_als_datasets.starfusion.fus.violin <- violin_plot(partialize(ipsc_mn_als_datasets.starfusion.fus.fitg_plot, vars = "genetic_group"), color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "grey"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(2,7), dpi = 300, ylabel = "") +
#   stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fus.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
# ipsc_mn_als_datasets.starfusion.fus.violin


### sod1
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "sod1")

ipsc_mn_als_datasets.starfusion.sod1 = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>%
  filter(dataset %in% c("answerals","neurolincs.diMN", "kiskinis","wang"), mutation %in% c("ctrl","iso","sod1")) %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.sod1

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","neurolincs.diMN", "kiskinis","wang"), mutation %in% c("ctrl","iso","sod1")) %>% count(condition) # 208 ALS, 50 CTRL

ipsc_mn_als_datasets.starfusion.sod1.n = ipsc_mn_als_datasets.starfusion.sod1 %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group, mutation, dataset, total_reads, age_at_sample_collection) #keep 1 row per sample
 # add samples with 0 fusions

ipsc_mn_als_datasets.starfusion.sod1.fitg = glm(n ~ genetic_group + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.sod1.n, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.sod1.fitg, digits = 5)$coeftable 
# Coefficients:
#                                                                  Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                                                     2.729e+00  1.158e+00   2.357 0.018401 *  
# genetic_groupSOD1                                               3.092e-01  8.583e-02   3.602 0.000315 ***
plot_summs(ipsc_mn_als_datasets.starfusion.sod1.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

# ipsc_mn_als_datasets.starfusion.sod1.fitg_plot = glm(n ~ genetic_group + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.sod1.n, family = poisson)
# partialize(ipsc_mn_als_datasets.starfusion.sod1.fitg_plot, vars = "genetic_group") %>% my_summarise(n, genetic_group)
# #  genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# # 1 CTRL            11.9     10.7  4.45  8.15  15.2  6.14  21.8  653.    55
# # 2 SOD1            15.3     13.2  6.47 11.8   16.5  9.33  32.7  275.    18
# 
# # Violin
# ipsc_mn_als_datasets.starfusion.sod1.n.stats = tibble(group1 = "CTRL", group2 = "sod1", p.signif = "****", y.position = 32, xmin = 1, xmax = 2)
# ipsc_mn_als_datasets.starfusion.sod1.violin <- violin_plot(partialize(ipsc_mn_als_datasets.starfusion.sod1.fitg_plot, vars = "genetic_group"), color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "gold2"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(6,32), dpi = 300, ylabel = "") +
#   stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.sod1.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
# ipsc_mn_als_datasets.starfusion.sod1.violin

# 
# 
# # Merge
# ipsc_mn_als_datasets.starfusion.mutation.violin = plot_grid(ipsc_mn_als_datasets.starfusion.sporadic.violin + theme(axis.text.x = element_text(angle = 30, hjust = 1)), ipsc_mn_als_datasets.starfusion.c9orf72.violin + theme(axis.title.y = element_blank(),axis.text.x = element_text(angle = 30, hjust = 1)), ipsc_mn_als_datasets.starfusion.sod1.violin + theme(axis.title.y = element_blank(),axis.text.x = element_text(angle = 30, hjust = 1)), ipsc_mn_als_datasets.starfusion.tardbp.violin + theme(axis.title.y = element_blank(),axis.text.x = element_text(angle = 30, hjust = 1)), ipsc_mn_als_datasets.starfusion.fus.violin + theme(axis.title.y = element_blank(),axis.text.x = element_text(angle = 30, hjust = 1)), ncol = 5, rel_widths = c(1.1,1,1,1,1), align = "h")
# ipsc_mn_als_datasets.starfusion.sod1.violin.title <- ggdraw() + draw_label("iPSMN: Genetic backgrounds", fontface = 'plain', hjust = 0.5, size = 9) + theme(plot.margin = margin(0,0,0,14))
# ipsc_mn_als_datasets.starfusion.mutation.violin = plot_grid(ipsc_mn_als_datasets.starfusion.sod1.violin.title, ipsc_mn_als_datasets.starfusion.mutation.violin, nrow = 2, rel_heights = c(0.1,1))
# # ipsc_mn_als_datasets.starfusion.mutation.violin

```

merge into Forrest plot

```{r}
ipsc_mn_als_datasets.starfusion.mutation.fitg.forrest_plot = plot_summs(ipsc_mn_als_datasets.starfusion.pan_als.fitg, ipsc_mn_als_datasets.starfusion.sporadic.fitg, ipsc_mn_als_datasets.starfusion.c9orf72.fitg, ipsc_mn_als_datasets.starfusion.sod1.fitg, ipsc_mn_als_datasets.starfusion.tardbp.fitg, ipsc_mn_als_datasets.starfusion.fus.fitg,
           coefs = c("pan ALS"="CONDITIONALS","Sporadic"="genetic_groupSporadic","C9orf72"="genetic_groupC9orf72","SOD1"="genetic_groupSOD1","TARDBP"="genetic_groupTARDBP","FUS"="genetic_groupFUS"),
           plot.distributions = TRUE, rescale.distributions = TRUE, 
           colors = c("purple","dodgerblue2","firebrick2","forestgreen","gold2","grey"), legend.title	= "",exp = FALSE,point.shape = FALSE, point.size = 1) + 
  coord_cartesian(xlim = c(-0.7,0.7)) + ggtitle("iPSMN: Gene Fusions") + 
  theme_classic() + theme(legend.position = "none", axis.text = element_text(colour = "black", size = 8), axis.title = element_text(colour = "black", size = 9, face ="plain"), strip.text.y = element_text(colour = "black", size = 7), plot.title = element_text(hjust = 0.5, size = 9)) + guides(color=guide_legend(keywidth=0.1, keyheight=0.15, default.unit="inch")) +
  annotation_custom(grid::textGrob(label = paste0("increase in ALS"), y = unit(0.08, "npc"), x = unit(0.8, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 0)) + 
  annotation_custom(grid::textGrob(label = paste0("increase in CTRL"), y = unit(0.08, "npc"), x = unit(0.2, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 0)) + 
  annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), x = c(0.95,0.65), y = unit(0.03, "npc"))) +
  annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), x = c(0.05,0.35), y = unit(0.03, "npc")))
ipsc_mn_als_datasets.starfusion.mutation.fitg.forrest_plot

```

### Postmortem

```{r}
# read_starfusion = function(metadata){
#   print("reading in STAR Fusion files with read_tsv")
#   starfusion.tsv = metadata$starfusion_path %>% map(read_tsv, col_types = list(JunctionReadCount="n",SpanningFragCount="n",est_J="n",est_S="n",FFPM="n",LeftBreakEntropy="n",RightBreakEntropy="n",.default="c"))
#   names(starfusion.tsv) = metadata$sample # needs to be unique names e.g. sample
#   starfusion_tab <- map_dfr(starfusion.tsv, bind_rows, .id = "sample") %>% rename(fusion_name = '#FusionName') %>% clean_names() %>%
#     mutate(left_chr = str_split_fixed(left_breakpoint,":",3)[,1], left_coord = str_split_fixed(left_breakpoint,":",3)[,2], left_strand = str_split_fixed(left_breakpoint,":",3)[,3],
#            right_chr = str_split_fixed(right_breakpoint,":",3)[,1], right_coord = str_split_fixed(right_breakpoint,":",3)[,2], right_strand = str_split_fixed(right_breakpoint,":",3)[,3],
#            fusion_type = case_when(grepl("INTERCHROMOSOMAL", annots) ~ "Inter\nChromosomal", grepl("OVERLAP", annots) ~ "Overlapping\nNeighbours", grepl("NEIGHBORS", annots) ~ "Neighbours", grepl("LOCAL_INVERSION", annots) ~ "Local\nInversion", grepl("LOCAL_REARRANGEMENT", annots) ~ "Local\nRearrangement", TRUE ~ "Distant\nProximity"),
#            breakpoints = paste(left_breakpoint, right_breakpoint, sep="_")) %>%
#     left_join(metadata, by = "sample")
#   return(starfusion_tab)
# }
# nygc_spinal_cord_metadata = read_csv(here(shared_path, "nygc-als-consortium/sample-details/nygc_spinal_cord_metadata.csv")) %>%
#   mutate(starfusion_path = paste0("/camp/project/proj-luscombn-patani/working/nygc-als-consortium/rnafusion/starfusion/",sample,".starfusion.abridged.coding_effect.tsv"))
# nygc_spinal_cord.starfusion = read_starfusion(nygc_spinal_cord_metadata)
# nygc_spinal_cord.starfusion
# write_csv(nygc_spinal_cord.starfusion, "/camp/project/proj-luscombn-patani/working/nygc-als-consortium/rnafusion/nygc_spinal_cord.starfusion.csv")
nygc_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
nygc.multiqc_star.txt = read_tsv(here(nygc_path,"rnafusion/multiqc/multiqc_data/multiqc_star.txt")) %>% clean_names()
nygc.multiqc_star.txt %>% glimpse
nygc_spinal_cord.starfusion = read_csv(here(nygc_path,"rnafusion/nygc_spinal_cord.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))  %>% 
  left_join(nygc.multiqc_star.txt) # left_join multiqc stats
nygc_spinal_cord.starfusion
nygc_spinal_cord_metadata = read_csv(here(shared_path, "nygc-als-consortium/sample-details/nygc_spinal_cord_metadata.csv"))
nygc_spinal_cord_metadata %>% count(condition) # 214 ALS, 57 CTRL
nygc_spinal_cord_metadata %>% count(mutation)

nygc_spinal_cord.starfusion.n = nygc_spinal_cord.starfusion %>% 
  add_count(sample) %>% # count number of fusions per sample
  distinct(sample, .keep_all = TRUE) %>% select(sample, n, CONDITION, total_reads, age_at_death, library_prep, sample_source) #keep 1 row per sample

nygc_spinal_cord.starfusion.samples_no_fusions = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% nygc_spinal_cord.starfusion.n$sample] # add samples with 0 fusions (none in NYGC)
nygc_spinal_cord.starfusion.samples_no_fusions # 0

# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
nygc_spinal_cord.starfusion.n.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.starfusion.n, family = poisson)
summ(nygc_spinal_cord.starfusion.n.fitg, digits = 5)$coeftable 
#                                                Est.         S.E.     z val.            p
# (Intercept)                             2.133027e+00 2.225114e-01  9.5861446 9.143808e-22
# CONDITIONALS                            2.501201e-01 5.574883e-02  4.4865544 7.238424e-06
nygc_spinal_cord.starfusion.n.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.n, family = poisson) # cant plot rcs spline
partialize(nygc_spinal_cord.starfusion.n.fitg_plot, vars = "CONDITION") # the fitted values adjust for the model #https://jtools.jacob-long.com/articles/effect_plot.html#partial-residuals-plots
partialize(nygc_spinal_cord.starfusion.n.fitg_plot, vars = "CONDITION") %>% my_summarise(n, CONDITION)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n  se_n
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL        6.76     5.97  2.82  5.02  7.70  3.73  16.6  385.    57 0.374
# 2 ALS         8.85     7.94  3.17  6.81  9.92  4.39  20.4 1893.   214 0.217
nygc_spinal_cord.starfusion.n.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 20, xmin = 1, xmax = 2)
nygc_spinal_cord.starfusion.violin <- violin_plot(partialize(nygc_spinal_cord.starfusion.n.fitg_plot, vars = "CONDITION"), color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(3.73,20.4), dpi = 300, ylabel = "Adjusted Number of Fusions", title = "Postmortem: Fusions") +
  stat_pvalue_manual(nygc_spinal_cord.starfusion.n.stats, tip.length = 0.01, size = 4)
# nygc_spinal_cord.starfusion.violin

effect_plot(nygc_spinal_cord.starfusion.n.fitg_plot, pred = CONDITION, plot.points = TRUE, interval = TRUE, jitter = 0.2, partial.residuals = TRUE)
effect_plot(nygc_spinal_cord.starfusion.n.fitg_plot, pred = total_reads, interval = TRUE, plot.points = TRUE,  jitter = 0) # read effect
effect_plot(nygc_spinal_cord.starfusion.n.fitg_plot, pred = age_at_death, interval = TRUE, plot.points = TRUE,  jitter = 0)
plot_summs(nygc_spinal_cord.starfusion.n.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
fit <- glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), family = poisson, data=nygc_spinal_cord.starfusion.n)
predicted <- predict(fit, type="response")
ggplot(cbind(nygc_spinal_cord.starfusion.n, predicted=predicted), aes(x=total_reads, y=predicted, colour= CONDITION, group = interaction(CONDITION))) + geom_line() + theme_oz()

```


#### Genetic subgroups

```{r}
nygc_spinal_cord_metadata %>% count(mutation)

nygc_spinal_cord.starfusion.mutation = read_csv(here(nygc_path,"rnafusion/nygc_spinal_cord.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic",mutation %in% c("ctrl") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72"))) %>% # mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", , "FUS", "TARDBP"
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  left_join(nygc.multiqc_star.txt)# %>% 
  # select(sample, total, genetic_group, CONDITION, total_reads, mutation, age_at_death) # keep 1 row per sample
nygc_spinal_cord.starfusion.mutation %>% glimpse

# sum fusions types
nygc_spinal_cord.starfusion.mutation.n = nygc_spinal_cord.starfusion.mutation %>% 
  add_count(sample) %>% # count number of fusion per sample
  distinct(sample, .keep_all = TRUE) %>% select(sample, n, CONDITION, genetic_group, mutation, total_reads, age_at_death, library_prep) #keep 1 row per sample
 # add samples with 0 fusions
nygc_spinal_cord_metadata_genetic_subgroup = nygc_spinal_cord_metadata %>% filter(mutation %in%  c("ctrl", "sporadic", "sod1", "c9orf72")) #"fus", "tardbp", 
nygc_spinal_cord.starfusion.samples_no_variants = nygc_spinal_cord_metadata_genetic_subgroup$sample[!nygc_spinal_cord_metadata_genetic_subgroup$sample %in% nygc_spinal_cord.starfusion.mutation$sample] # 0 i.e. all samples have at least 1 variant
nygc_spinal_cord.starfusion.samples_no_variants # 0

nygc_spinal_cord.starfusion.mutation.n = bind_rows(nygc_spinal_cord.starfusion.mutation.n, mutate(filter(nygc_spinal_cord.starfusion.mutation.n, CONDITION!="CTRL"), genetic_group = "pan ALS"))

# All fusion types
nygc_spinal_cord.starfusion.mutation.fitg = glm(n ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.starfusion.mutation.n, family=poisson)
summ(nygc_spinal_cord.starfusion.mutation.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.            p
# (Intercept)                             2.064461e+00 2.446267e-01  8.4392323 3.194286e-17
# genetic_groupSporadic                   2.603162e-01 5.728481e-02  4.5442453 5.513242e-06
# genetic_groupSOD1                       4.800773e-01 1.383894e-01  3.4690330 5.223353e-04
# genetic_groupC9orf72                    1.619835e-01 7.781659e-02  2.0816061 3.737846e-02
plot_summs(nygc_spinal_cord.starfusion.mutation.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

## pan ALS
nygc_spinal_cord.starfusion.pan_als.n = filter(nygc_spinal_cord.starfusion.mutation.n, genetic_group %in% c("pan ALS", "CTRL"))
nygc_spinal_cord.starfusion.pan_als.fitg = glm(n ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.pan_als.n, family=poisson)
summ(nygc_spinal_cord.starfusion.pan_als.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.      z val.            p
# (Intercept)                             2.202096e+00 2.320431e-01  9.49003193 2.309635e-21
# genetic_grouppan ALS                    2.598073e-01 5.632142e-02  4.61293915 3.970146e-06
plot_summs(nygc_spinal_cord.starfusion.pan_als.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


## sporadic
nygc_spinal_cord.starfusion.sporadic.n = filter(nygc_spinal_cord.starfusion.mutation.n, genetic_group %in% c("Sporadic", "CTRL"))
nygc_spinal_cord.starfusion.sporadic.fitg = glm(n ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.sporadic.n, family=poisson)
summ(nygc_spinal_cord.starfusion.sporadic.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.            p
# (Intercept)                             2.065442e+00 2.466781e-01  8.3730232 5.615854e-17
# genetic_groupSporadic                   2.560328e-01 5.742293e-02  4.4587208 8.245022e-06
plot_summs(nygc_spinal_cord.starfusion.sporadic.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

## c9orf72
nygc_spinal_cord.starfusion.c9orf72.n = filter(nygc_spinal_cord.starfusion.mutation.n, genetic_group %in% c("C9orf72", "CTRL"))
nygc_spinal_cord.starfusion.c9orf72.fitg = glm(n ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.c9orf72.n, family=poisson)
summ(nygc_spinal_cord.starfusion.c9orf72.fitg, digits = 5)$coeftable 
#                                                Est.         S.E.     z val.            p
# (Intercept)                             1.338855e+00 4.524673e-01  2.9590093 3.086298e-03
# genetic_groupC9orf72                    1.836743e-01 8.237449e-02  2.2297469 2.576425e-02
plot_summs(nygc_spinal_cord.starfusion.c9orf72.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

## SOD1
nygc_spinal_cord.starfusion.sod1.n = filter(nygc_spinal_cord.starfusion.mutation.n, genetic_group %in% c("SOD1", "CTRL"))
nygc_spinal_cord.starfusion.sod1.fitg = glm(n ~ genetic_group + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.sod1.n, family=poisson)
summ(nygc_spinal_cord.starfusion.sod1.fitg, digits = 5)$coeftable 
#                                               Est.         S.E.     z val.            p
# (Intercept)                             2.132323e+00 4.734508e-01  4.5037910 6.675186e-06
# genetic_groupSOD1                       5.005710e-01 1.511318e-01  3.3121490 9.258222e-04
plot_summs(nygc_spinal_cord.starfusion.sod1.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction


nygc_spinal_cord.starfusion.mutation.fitg.forrest_plot = plot_summs(nygc_spinal_cord.starfusion.pan_als.fitg, nygc_spinal_cord.starfusion.sporadic.fitg, nygc_spinal_cord.starfusion.c9orf72.fitg, nygc_spinal_cord.starfusion.sod1.fitg,
           coefs = c("pan ALS"="genetic_grouppan ALS", "Sporadic"="genetic_groupSporadic","C9orf72"="genetic_groupC9orf72","SOD1"="genetic_groupSOD1"), plot.distributions = TRUE, rescale.distributions = TRUE, 
           colors = c("purple", "dodgerblue2","firebrick2","forestgreen","gold2","grey"), legend.title	= "",exp = FALSE,point.shape = FALSE, point.size = 1) + 
  coord_cartesian(xlim = c(-0.7,0.7)) + ggtitle("Postmortem: Gene Fusions") +
 theme_classic() + theme(legend.position = "none", axis.text = element_text(colour = "black", size = 8), axis.title = element_text(colour = "black", size = 9, face ="plain"), strip.text.y = element_text(colour = "black", size = 7), plot.title = element_text(hjust = 0.5, size = 9)) + guides(color=guide_legend(keywidth=0.1, keyheight=0.15, default.unit="inch")) +
  annotation_custom(grid::textGrob(label = paste0("increase in ALS"), y = unit(0.1, "npc"), x = unit(0.8, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 0)) + 
  annotation_custom(grid::textGrob(label = paste0("increase in CTRL"), y = unit(0.1, "npc"), x = unit(0.2, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 0)) + 
  annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), x = c(0.95,0.65), y = unit(0.04, "npc"))) +
  annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), x = c(0.05,0.35), y = unit(0.04, "npc")))
nygc_spinal_cord.starfusion.mutation.fitg.forrest_plot




# # generalised linear model spline
# 
# 
# nygc_spinal_cord.starfusion.sporadic.fitg_plot = glm(n ~ genetic_group + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.sporadic.n, family = poisson)
# partialize(nygc_spinal_cord.starfusion.sporadic.fitg_plot, vars = "genetic_group") %>% my_summarise(n, genetic_group)
# # genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# # 1 CTRL            8.61     6.58  5.84  5.30  9.18  4.02  35.1  491.    57
# # 2 Sporadic       10.7      9.31  5.12  7.53 12.2   4.89  43.4 1727.   161
# 
# 
# nygc_spinal_cord.starfusion.c9orf72.fitg_plot = glm(n ~ genetic_group + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.c9orf72.n, family = poisson)
# partialize(nygc_spinal_cord.starfusion.c9orf72.fitg_plot, vars = "genetic_group") %>% my_summarise(n, genetic_group)
# # genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# # 1 CTRL            8.59     6.60  5.90  5.30  9.17  4.01  35.3  490.    57
# # 2 C9orf72         9.40     8.24  4.42  6.63  9.86  5.32  28.0  338.    36
# 
# 
# nygc_spinal_cord.starfusion.sod1.fitg_plot = glm(n ~ genetic_group + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.sod1.n, family = poisson)
# partialize(nygc_spinal_cord.starfusion.sod1.fitg_plot, vars = "genetic_group") %>% my_summarise(n, genetic_group)
# # genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int>
# # 1 CTRL            8.33     6.32  5.49  5.16  8.92  3.88  33.4 475.     57
# # 2 SOD1           12.8     12.7   1.94 11.7  14.1  10.3   15.3  64.1     5
# 
# # plot fitted values
# nygc_spinal_cord.starfusion.mutation.bind = bind_rows(partialize(nygc_spinal_cord.starfusion.sporadic.fitg_plot, vars = "genetic_group"),
#                                               filter(partialize(nygc_spinal_cord.starfusion.c9orf72.fitg_plot, vars = "genetic_group"), genetic_group!="CTRL"),
#                                               filter(partialize(nygc_spinal_cord.starfusion.sod1.fitg_plot, vars = "genetic_group"), genetic_group!="CTRL"))
# nygc_spinal_cord.starfusion.mutation.bind %>% my_summarise(n, genetic_group)
# # genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n  sum_n   n_n
# #   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>  <dbl> <int>
# # 1 CTRL            8.61     6.58  5.84  5.30  9.18  4.02  35.1  491.     57
# # 2 Sporadic       10.7      9.31  5.12  7.53 12.2   4.89  43.4 1727.    161
# # 3 C9orf72         9.40     8.24  4.42  6.63  9.86  5.32  28.0  338.     36
# # 4 SOD1           12.8     12.7   1.94 11.7  14.1  10.3   15.3   64.1     5
# 
# nygc_spinal_cord.starfusion.mutation.bind.stat <- nygc_spinal_cord.starfusion.mutation.bind %>% t_test(n ~ genetic_group) %>% filter(group1=="CTRL") %>% mutate(p.adj.signif = c("****","ns","**"), y.position = c(28,28,30)) 
# nygc_spinal_cord.starfusion.mutation.bind.stat
# # .y.   group1 group2      n1    n2 statistic    df     p p.adj p.adj.signif y.position
# #   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl>
# # 1 n     CTRL   Sporadic    57   161    -2.43   88.3 0.017 0.069 ****                 28
# # 2 n     CTRL   C9orf72     57    36    -0.739  87.9 0.462 0.462 ns                   28
# # 3 n     CTRL   SOD1        57     5    -3.63   12.4 0.003 0.02  **                   30
# 
# nygc_spinal_cord.starfusion.mutation.violin <- violin_plot(nygc_spinal_cord.starfusion.mutation.bind, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,30), dpi = 300, ylabel = "Adjusted Number of Fusions", title = "Postmortem: Genetic backgrounds") +
#   stat_pvalue_manual(nygc_spinal_cord.starfusion.mutation.bind.stat, tip.length = 0.01, size = 4, hide.ns = TRUE)
# nygc_spinal_cord.starfusion.mutation.violin

```



## Merge

```{r}
ipsc_mn_als_datasets.gene_fusion_merged = plot_grid(
  plot_grid(answerals.vcfannotate.totals.fitg_plot.bind.violin, nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind.violin,ncol = 2, rel_widths = c(1,1), labels = c("a","c")),
  plot_grid(answerals.vcfannotate.mutation.fitg.forrest_plot, nygc_spinal_cord.vcfannotate.mutation.fitg.forrest_plot, ncol = 2, rel_widths = c(1,1), labels = c("b","d")),
  plot_grid(ipsc_mn_als_datasets.starfusion.violin, ipsc_mn_als_datasets.starfusion.mutation.fitg.forrest_plot, nygc_spinal_cord.starfusion.violin, nygc_spinal_cord.starfusion.mutation.fitg.forrest_plot, ncol = 4, rel_widths = c(1,2,1,2), labels = c("e","f","g","h")),
  nrow = 3, rel_heights = c(1,1.1,1.1))
# ipsc_mn_als_datasets.gene_fusion_merged
ggsave(ipsc_mn_als_datasets.gene_fusion_merged, filename = here(proj_path,"rnafusion/ipsc_mn_als_datasets.gene_fusion_merged.png"), height = 9, width = 10)
ggsave(ipsc_mn_als_datasets.gene_fusion_merged, filename = here(proj_path,"rnafusion/ipsc_mn_als_datasets.gene_fusion_merged.pdf"), height = 9, width = 10)

# for ppt
ggsave(plot_grid(answerals.vcfannotate.totals.fitg_plot.bind.violin, answerals.vcfannotate.mutation.fitg.forrest_plot, ncol = 2, rel_widths = c(2,1), nrow = 1), filename = here(proj_path,"rnafusion/ipsc_mn_als_datasets.variants.png"), height = 4.5, width = 10)
ggsave(plot_grid(nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind.violin, nygc_spinal_cord.vcfannotate.mutation.fitg.forrest_plot, ncol = 2, rel_widths = c(2,1), nrow = 1), filename = here(proj_path,"rnafusion/nygc_spinal_cord.variants.png"), height = 4.5, width = 10)

ggsave(plot_grid(ipsc_mn_als_datasets.starfusion.violin, ipsc_mn_als_datasets.starfusion.mutation.fitg.forrest_plot, nygc_spinal_cord.starfusion.violin, nygc_spinal_cord.starfusion.mutation.fitg.forrest_plot, ncol = 4, rel_widths = c(0.8,1,0.8,1)),
       filename = here(proj_path,"rnafusion/ipsc_nygc.fusions.png"), height = 4.5, width = 10)


ipsc_mn_als_datasets.variants = plot_grid(answerals.vcfannotate.variant_types.violin, nygc_spinal_cord.starfusion.violin, ncol = 2, rel_widths = c(3,1))
ggsave(ipsc_mn_als_datasets.variants, filename = here(proj_path,"rnafusion/ipsc_mn_als_datasets.variants.png"), height = 4, width = 9)

```


# Table S12-13 Somatic mutations

```{r}
### iPSMN
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
answerals.vcfannotate = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.dbsnp.redi.filtered.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  left_join(multiqc_rseqc_read_distribution.tsv) %>% filter(dataset == "answerals") %>% select(dataset, sample, n, variant_type, CONDITION, mutation, total_reads, age_at_sample_collection, age_at_death, age_at_symptom_onset) %>% 
  pivot_wider(names_from = "variant_type", values_from = "n", values_fill = 0) %>% 
  mutate(Total = SNP+Insertion+Deletion+Complex, age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ age_at_death, TRUE ~ age_at_sample_collection), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ age_at_symptom_onset, TRUE ~ age_at_sample_collection), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 59, TRUE ~ age_at_sample_collection)) %>% 
  select(sample, CONDITION, mutation, Total.raw=Total, SNP.raw=SNP, Insertion.raw=Insertion, Deletion.raw=Deletion, total_reads, age_at_sample_collection) %>% #, Complex
  arrange(CONDITION)
answerals.vcfannotate.Total.raw.fitg = glm(Total.raw ~ CONDITION +total_reads + age_at_sample_collection, data=answerals.vcfannotate, family = poisson)
answerals.vcfannotate.SNP.raw.fitg = glm(SNP.raw ~ CONDITION +total_reads + age_at_sample_collection, data=answerals.vcfannotate, family = poisson)
answerals.vcfannotate.Insertion.raw.fitg = glm(Insertion.raw ~ CONDITION +total_reads + age_at_sample_collection, data=answerals.vcfannotate, family = poisson)
answerals.vcfannotate.Deletion.raw.fitg = glm(Deletion.raw ~ CONDITION +total_reads + age_at_sample_collection, data=answerals.vcfannotate, family = poisson)
answerals.vcfannotate$Total.adj = partialize(answerals.vcfannotate.Total.raw.fitg, vars = "CONDITION")$Total.raw
answerals.vcfannotate$SNP.adj = partialize(answerals.vcfannotate.SNP.raw.fitg, vars = "CONDITION")$SNP.raw
answerals.vcfannotate$Insertion.adj = partialize(answerals.vcfannotate.Insertion.raw.fitg, vars = "CONDITION")$Insertion.raw
answerals.vcfannotate$Deletion.adj = partialize(answerals.vcfannotate.Deletion.raw.fitg, vars = "CONDITION")$Deletion.raw
answerals.vcfannotate
answerals.vcfannotate %>% print(n=Inf)
write_csv(answerals.vcfannotate, here(proj_path,"rnavar/answerals.vcfannotate.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473")
write_sheet(answerals.vcfannotate, ss = "https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473", sheet = "S12 Variants iPSMN")


### Postmortem
nygc_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
nygc.rnavar.multiqc_star.txt = read_tsv(here(nygc_path,"rnavar/reports/multiqc_data/multiqc_star.txt")) %>% clean_names()
nygc_spinal_cord.vcfannotate = read_csv(here(nygc_path,"rnavar/nygc_spinal_cord.vcfannotate.dbsnp.redi.filtered.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(sample) %>% add_tally(n, name = "total") %>% ungroup %>% select(sample, n, type, variant_type, total, everything()) %>%  # add a column of sum of variants per sample
  left_join(nygc.rnavar.multiqc_star.txt) %>% select(sample, CONDITION, mutation, variant_type, n, total_reads, age_at_death,library_prep) %>% 
  pivot_wider(c(sample, CONDITION, mutation, total_reads, age_at_death, library_prep), names_from = "variant_type", values_from = "n", values_fill = 0) %>% 
  mutate(Complex = case_when(is.na(Complex) ~ 0, TRUE ~ Complex), Total = Complex + Deletion + Insertion + SNP) %>%
  select(sample, CONDITION, mutation, Total.raw=Total, SNP.raw=SNP, Insertion.raw=Insertion, Deletion.raw=Deletion, total_reads, age_at_death, library_prep) %>% #, Complex
  arrange(CONDITION)
nygc_spinal_cord.vcfannotate.Total.raw.fitg = glm(Total.raw ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate, family = poisson)
nygc_spinal_cord.vcfannotate.SNP.raw.fitg = glm(SNP.raw ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate, family = poisson)
nygc_spinal_cord.vcfannotate.Insertion.raw.fitg = glm(Insertion.raw ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate, family = poisson)
nygc_spinal_cord.vcfannotate.Deletion.raw.fitg = glm(Deletion.raw ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.vcfannotate, family = poisson)
nygc_spinal_cord.vcfannotate$Total.adj = partialize(nygc_spinal_cord.vcfannotate.Total.raw.fitg, vars = "CONDITION")$Total.raw
nygc_spinal_cord.vcfannotate$SNP.adj = partialize(nygc_spinal_cord.vcfannotate.SNP.raw.fitg, vars = "CONDITION")$SNP.raw
nygc_spinal_cord.vcfannotate$Insertion.adj = partialize(nygc_spinal_cord.vcfannotate.Insertion.raw.fitg, vars = "CONDITION")$Insertion.raw
nygc_spinal_cord.vcfannotate$Deletion.adj = partialize(nygc_spinal_cord.vcfannotate.Deletion.raw.fitg, vars = "CONDITION")$Deletion.raw
nygc_spinal_cord.vcfannotate
nygc_spinal_cord.vcfannotate %>% print(n=Inf)
write_csv(nygc_spinal_cord.vcfannotate, here(proj_path,"rnavar/nygc_spinal_cord.vcfannotate.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473")
write_sheet(nygc_spinal_cord.vcfannotate, ss = "https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473", sheet = "S11 Variants Postmortem")
  
```


# Table S14 gene fusions iPSMNs and postmortem

RNA fusion events in each sample

```{r}
# iPSMN
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
ipsc_mn_als_datasets.starfusion = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))%>% left_join(multiqc_rseqc_read_distribution.tsv)

## Fusions with greatest frequency
ipsc_mn_als_datasets.starfusion.als.count = ipsc_mn_als_datasets.starfusion %>% filter(condition == "als") %>% count(breakpoints, name = "als_yes")
ipsc_mn_als_datasets.starfusion.ctrl.count = ipsc_mn_als_datasets.starfusion %>% filter(condition == "ctrl") %>% count(breakpoints, name = "ctrl_yes")
ipsc_mn_als_datasets.starfusion.als_ctrl.prop = ipsc_mn_als_datasets.starfusion.als.count %>% full_join(ipsc_mn_als_datasets.starfusion.ctrl.count)  %>% replace(is.na(.),0) %>% 
  mutate(als_total = 360, ctrl_total = 90) 

ipsc_mn_als_datasets.starfusion.als_ctrl.burden = ipsc_mn_als_datasets.starfusion.als_ctrl.prop %>% rowwise() %>% 
  mutate(p_value = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total), 
         odds_ratio = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$estimate[1],
         lower_ci = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$conf.int[1], 
         upper_ci = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$conf.int[2])
ipsc_mn_als_datasets.starfusion.als_ctrl.burden

als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
als_ctrl.voila.splicing_genes = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # 264

# add back gene fusion names
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table = ipsc_mn_als_datasets.starfusion.als_ctrl.burden %>% left_join(distinct(select(ipsc_mn_als_datasets.starfusion, breakpoints, fusion_name, fusion_type))) %>% arrange(p_value) %>% 
  mutate(gene1 = str_split_fixed(fusion_name, "--",2)[,1], gene2 = str_split_fixed(fusion_name, "--",2)[,2], alterated_splicing = case_when(gene1 %in% als_ctrl.voila.splicing_genes | gene2 %in% als_ctrl.voila.splicing_genes ~ "yes", TRUE ~ "")) %>%
  select(fusion_name, breakpoints, fusion_type, ipsn_als_number = als_yes, ipsn_ctrl_number = ctrl_yes, ipsn_odds_ratio = odds_ratio, ipsn_lower_ci = lower_ci, ipsn_upper_ci = upper_ci, ipsn_p_value = p_value, alterated_splicing)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table


## Postmortem
nygc_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
nygc.multiqc_star.txt = read_tsv(here(nygc_path,"rnafusion/multiqc/multiqc_data/multiqc_star.txt")) %>% clean_names()
nygc.multiqc_star.txt %>% glimpse
nygc_spinal_cord.starfusion = read_csv(here(nygc_path,"rnafusion/nygc_spinal_cord.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))  %>% 
  left_join(nygc.multiqc_star.txt) # left_join multiqc stats

## Fusions with greatest frequency
nygc_spinal_cord.starfusion.als.count = nygc_spinal_cord.starfusion %>% filter(condition == "als") %>% count(breakpoints, name = "als_yes")
nygc_spinal_cord.starfusion.ctrl.count = nygc_spinal_cord.starfusion %>% filter(condition == "ctrl") %>% count(breakpoints, name = "ctrl_yes")
nygc_spinal_cord.starfusion.als_ctrl.prop = nygc_spinal_cord.starfusion.als.count %>% full_join(nygc_spinal_cord.starfusion.ctrl.count)  %>% replace(is.na(.),0) %>% 
  mutate(als_total = 214, ctrl_total = 57) 

nygc_spinal_cord.starfusion.als_ctrl.burden = nygc_spinal_cord.starfusion.als_ctrl.prop %>% rowwise() %>% 
  mutate(p_value = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total), 
         odds_ratio = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$estimate[1],
         lower_ci = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$conf.int[1], 
         upper_ci = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$conf.int[2])
nygc_spinal_cord.starfusion.als_ctrl.burden

# add back gene fusion names
nygc_spinal_cord.starfusion.als_ctrl.burden_table = nygc_spinal_cord.starfusion.als_ctrl.burden %>% left_join(distinct(select(nygc_spinal_cord.starfusion, breakpoints, fusion_name, fusion_type))) %>% arrange(p_value) %>% 
  mutate(gene1 = str_split_fixed(fusion_name, "--",2)[,1], gene2 = str_split_fixed(fusion_name, "--",2)[,2], pm_lower_ci = as.character(lower_ci), pm_upper_ci = as.character(upper_ci)) %>%
  select(fusion_name, breakpoints, fusion_type, pm_als_number = als_yes, pm_ctrl_number = ctrl_yes, pm_odds_ratio = odds_ratio, pm_lower_ci, pm_upper_ci, pm_p_value = p_value) 
nygc_spinal_cord.starfusion.als_ctrl.burden_table %>% filter(pm_p_value < 0.05)


ipsc_postmortem.starfusion.als_ctrl.burden_table = ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table %>% full_join(nygc_spinal_cord.starfusion.als_ctrl.burden_table) %>% 
  mutate(ipsn_odds_ratio = as.character(ipsn_odds_ratio), ipsn_lower_ci = as.character(ipsn_lower_ci), ipsn_upper_ci = as.character(ipsn_upper_ci), pm_odds_ratio = as.character(pm_odds_ratio), pm_lower_ci = as.character(pm_lower_ci), pm_upper_ci = as.character(pm_upper_ci))
ipsc_postmortem.starfusion.als_ctrl.burden_table
write_csv(ipsc_postmortem.starfusion.als_ctrl.burden_table, here(proj_path,"rnafusion/ipsc_postmortem.starfusion.als_ctrl.burden_table.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473")
write_sheet(ipsc_postmortem.starfusion.als_ctrl.burden_table, ss = "https://docs.google.com/spreadsheets/d/1M7_nHWVV-QRvV-i0m1sQ-XRodzRodBQEFqsdmNCFGOw/edit#gid=514935473", sheet = "S12 fusions")




ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split = ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table %>% mutate(gene1 = str_split_fixed(fusion_name, "--",2)[,1], gene2 = str_split_fixed(fusion_name, "--",2)[,2])
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split

ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene1 %in% ALS_genes)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene2 %in% ALS_genes)

# overlap fusions with degs
ipsc_mn_als_datasets.degs = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% pull(gene_name)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene1 %in% ipsc_mn_als_datasets.degs)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene2 %in% ipsc_mn_als_datasets.degs)

# overlap fusions with splice events
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene1 %in% als_ctrl.voila.splicing_genes)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene2 %in% als_ctrl.voila.splicing_genes)

nygc_spinal_cord.starfusion.als_ctrl.burden_table

nygc_spinal_cord.starfusion.als_ctrl.burden_table.genes_split = nygc_spinal_cord.starfusion.als_ctrl.burden_table %>% mutate(gene1 = str_split_fixed(fusion_name, "--",2)[,1], gene2 = str_split_fixed(fusion_name, "--",2)[,2])
nygc_spinal_cord.starfusion.als_ctrl.burden_table.genes_split

nygc_spinal_cord.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene1 %in% ALS_genes)
nygc_spinal_cord.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene2 %in% ALS_genes)

# overlap fusions with degs
nygc_postmortem_spinal_cord.degs = nygc_postmortem_spinal_cord$res %>% filter(padj < 0.05) %>% pull(gene_name)
nygc_spinal_cord.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene1 %in% nygc_postmortem_spinal_cord.degs)
nygc_spinal_cord.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene2 %in% nygc_postmortem_spinal_cord.degs)

```


# Fig S18 Variants & fusion types & genetic groups

## Base substitution types

Mutation patterns https://bioconductor.org/packages/release/bioc/vignettes/MutationalPatterns/inst/doc/Introduction_to_MutationalPatterns.html#base-substitution-types 

96 mutational profile 
96 trinucleotide mutation count matrix.

```{r}
# BiocManager::install("MutationalPatterns")
library(MutationalPatterns)
# ref_genome <- "BSgenome.Hsapiens.UCSC.hg38"
# library(ref_genome, character.only = TRUE)
ipsmn_mut_mat = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/mut_mat.RDS"))
# ipsmn_mut_mat_ext_context = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/mut_mat_ext_context.RDS"))
ipsmn_pooled_mut_mat <- pool_mut_mat(ipsmn_mut_mat, grouping = answerals.metadata$condition)
colnames(ipsmn_pooled_mut_mat) = c("CTRL", "ALS")
head(ipsmn_pooled_mut_mat)
# ipsmn_p96_profile_pooled = plot_96_profile(pooled_mut_mat, condensed = FALSE)
# ipsmn_p96_profile_pooled
ipsmn_pooled_mut_tb <- ipsmn_pooled_mut_mat %>% as.data.frame() %>% rownames_to_column("full_context") %>% 
  mutate(substitution = stringr::str_replace(full_context, "\\w\\[(.*)\\]\\w", "\\1"), context = stringr::str_replace(full_context, "\\[.*\\]", "\\.")) %>% select(-full_context) %>%
  mutate_if(is.numeric, funs(./sum(.))) %>% # Make contribution relative
  pivot_longer(c(-substitution, -context), names_to = "sample", values_to = "freq") %>% mutate(sample = factor(sample, levels = unique(sample)))
ipsmn_p96_profile_pooled <- ggplot(data = ipsmn_pooled_mut_tb, aes(x = context, y = freq, fill = substitution, width = 0.6 )) +
    geom_bar(stat = "identity", colour = "black", size = .2) +
    scale_fill_manual(values =  c("#2EBAED", "#000000", "#DE1C14", "#D4D2D2", "#ADCC54", "#F0D0CE")) +
    facet_grid(sample ~ substitution) +
    ylab("Relative contribution") + xlab("") + ggtitle("iPSMN: Base substitution types") + coord_cartesian(ylim = c(0, 0.15)) +   scale_y_continuous(breaks = seq(0, 0.15, 0.05)) +
    guides(fill = FALSE) + guides(fill = FALSE) + theme_oz(xaxis_arrow = FALSE) + theme(legend.position = "none", plot.title = element_text(hjust = 0.5), axis.text.x = element_text(size = 5, angle = 90, vjust = 0.5), strip.text.x = element_text(size = 9))
ipsmn_p96_profile_pooled
  
  
# NYGC
nygc_mut_mat = readRDS(here(shared_path, "nygc-als-consortium/rnavar/mutationalpatterns/mut_mat.RDS"))
nygc_pooled_mut_mat <- pool_mut_mat(nygc_mut_mat, grouping = nygc_spinal_cord_metadata$condition)
colnames(nygc_pooled_mut_mat) = c("CTRL", "ALS")
head(nygc_pooled_mut_mat)
# ipsmn_p96_profile_pooled = plot_96_profile(pooled_mut_mat, condensed = FALSE)
# ipsmn_p96_profile_pooled
nygc_pooled_mut_tb <- nygc_pooled_mut_mat %>% as.data.frame() %>% rownames_to_column("full_context") %>% 
  mutate(substitution = stringr::str_replace(full_context, "\\w\\[(.*)\\]\\w", "\\1"), context = stringr::str_replace(full_context, "\\[.*\\]", "\\.")) %>% select(-full_context) %>%
  mutate_if(is.numeric, funs(./sum(.))) %>% # Make contribution relative
  pivot_longer(c(-substitution, -context), names_to = "sample", values_to = "freq") %>% mutate(sample = factor(sample, levels = unique(sample)))
nygc_p96_profile_pooled <- ggplot(data = nygc_pooled_mut_tb, aes(x = context, y = freq, fill = substitution, width = 0.6 )) +
    geom_bar(stat = "identity", colour = "black", size = .2) +
    scale_fill_manual(values =  c("#2EBAED", "#000000", "#DE1C14", "#D4D2D2", "#ADCC54", "#F0D0CE")) +
    facet_grid(sample ~ substitution) +
    ylab("Relative contribution") + xlab("") + ggtitle("Postmortem: Base substitution types") + coord_cartesian(ylim = c(0, 0.18)) +  scale_y_continuous(breaks = seq(0, 0.18, 0.05)) +
    guides(fill = FALSE) + theme_oz(xaxis_arrow = FALSE) + theme(legend.position = "none", plot.title = element_text(hjust = 0.5), axis.text.x = element_text(size = 5, angle = 90, vjust = 0.5), strip.text.x = element_text(size = 9))
nygc_p96_profile_pooled

```


Number of  Substitutions

```{r}
# http://bioconductor.org/packages/release/bioc/vignettes/MutationalPatterns/inst/doc/Introduction_to_MutationalPatterns.html
# BiocManager::install("MutationalPatterns")
# library(MutationalPatterns)
# ref_genome <- "BSgenome.Hsapiens.UCSC.hg38"
# library(ref_genome, character.only = TRUE)

# # read in vcf files as granges
# answerals.metadata = answerals.metadata %>% mutate(vcf_path = paste0("/camp/project/proj-luscombn-patani/working/answerals/rnavar/variant_calling/",sample, "/", sample, ".redi.filtered.recode.vcf.gz"))
# grl <- read_vcfs_as_granges(answerals.metadata$vcf_path, answerals.metadata$sample, ref_genome, type = "all")
# saveRDS(grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/grl.RDS")
# snv_grl <- get_mut_type(grl, type = "snv") # get types of mutations SNV, indels
# saveRDS(snv_grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/snv_grl.RDS")
# indel_grl <- get_mut_type(grl, type = "indel")
# saveRDS(indel_grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/indel_grl.RDS")
# dbs_grl <- get_mut_type(grl, type = "dbs")
# saveRDS(dbs_grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/dbs_grl.RDS")
# mbs_grl <- get_mut_type(grl, type = "mbs")
# saveRDS(mbs_grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/mbs_grl.RDS")
# type_occurrences <- mut_type_occurrences(snv_grl, ref_genome) # count mutation type occurences and base substitutions 
# saveRDS(type_occurrences, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/type_occurrences.RDS")

# grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/grl.RDS"))
# snv_grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/snv_grl.RDS"))
# indel_grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/indel_grl.RDS"))
# dbs_grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/dbs_grl.RDS"))
# mbs_grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/mbs_grl.RDS"))
answerals.type_occurrences = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/type_occurrences.RDS"))
answerals.type_occurrences

# left_join multiqc stats
answerals.multiqc_rseqc_read_distribution.tsv = read_tsv(here(shared_path,"answerals/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names()
answerals.base_substitutions = answerals.type_occurrences %>% as_tibble(rownames = "sample") %>% select(-`C>T at CpG`, -`C>T other`) %>%
  left_join(select(answerals.metadata, sample, condition, age_at_sample_collection, age_at_death, age_at_symptom_onset)) %>% 
  pivot_longer(!c(condition, sample, age_at_sample_collection, age_at_death, age_at_symptom_onset), names_to = "Substitution", values_to = "count") %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), Substitution = as.factor(Substitution), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ age_at_death, TRUE ~ age_at_sample_collection), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ age_at_symptom_onset, TRUE ~ age_at_sample_collection), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 59, TRUE ~ age_at_sample_collection)) %>%
  left_join(select(answerals.multiqc_rseqc_read_distribution.tsv, sample, total_reads))
answerals.base_substitutions %>% glimpse  

# For count data use glm(count ~ condition, family=poisson) which gives estimates and p-values of the underlying 'rate's of fusion per sample 
### C>A
answerals.base_substitutions.C2A = filter(answerals.base_substitutions, Substitution == "C>A")
answerals.base_substitutions.C2A.fitg = glm(count ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.base_substitutions.C2A, family = poisson)
summ(answerals.base_substitutions.C2A.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.    z val.             p
# (Intercept)                                                     4.423371e+00 5.498011e-02  80.45402  0.000000e+00
# CONDITIONALS                                                    1.033970e-01 9.243957e-03  11.18536  4.809231e-29
# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
answerals.base_substitutions.C2A.fitg_plot = glm(count ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.base_substitutions.C2A, family = poisson) # cant plot rcs spline

### C>G
answerals.base_substitutions.C2G = filter(answerals.base_substitutions, Substitution == "C>G")
answerals.base_substitutions.C2G.fitg = glm(count ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.base_substitutions.C2G, family = poisson)
summ(answerals.base_substitutions.C2G.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.      z val.             p
# (Intercept)                                                     5.114225e+00 4.997700e-02 102.3315767  0.000000e+00
# CONDITIONALS                                                    5.415824e-02 8.545865e-03   6.3373614  2.337333e-10
answerals.base_substitutions.C2G.fitg_plot = glm(count ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.base_substitutions.C2G, family = poisson) # cant plot rcs spline

### C>T
answerals.base_substitutions.C2T = filter(answerals.base_substitutions, Substitution == "C>T")
answerals.base_substitutions.C2T.fitg = glm(count ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.base_substitutions.C2T, family = poisson)
summ(answerals.base_substitutions.C2T.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.     z val.             p
# (Intercept)                                                     6.097941e+00 2.864142e-02 212.906346  0.000000e+00
# CONDITIONALS                                                    5.553352e-02 4.870596e-03  11.401793  4.095934e-30
answerals.base_substitutions.C2T.fitg_plot = glm(count ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.base_substitutions.C2T, family = poisson) # cant plot rcs spline

### T>A
answerals.base_substitutions.T2A = filter(answerals.base_substitutions, Substitution == "T>A")
answerals.base_substitutions.T2A.fitg = glm(count ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.base_substitutions.T2A, family = poisson)
summ(answerals.base_substitutions.T2A.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.      z val.             p
# (Intercept)                                                     4.175797e+00 6.622510e-02  63.0546005  0.000000e+00
# CONDITIONALS                                                    7.695741e-02 1.116850e-02   6.8905768  5.556663e-12
answerals.base_substitutions.T2A.fitg_plot = glm(count ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.base_substitutions.T2A, family = poisson) # cant plot rcs spline

### T>C
answerals.base_substitutions.T2C = filter(answerals.base_substitutions, Substitution == "T>C")
answerals.base_substitutions.T2C.fitg = glm(count ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.base_substitutions.T2C, family = poisson)
summ(answerals.base_substitutions.T2C.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.    z val.             p
# (Intercept)                                                     7.998753e+00 1.355218e-02 590.21906  0.000000e+00
# CONDITIONALS                                                    4.425405e-02 2.356557e-03  18.77911  1.119455e-78
answerals.base_substitutions.T2C.fitg_plot = glm(count ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.base_substitutions.T2C, family = poisson) # cant plot rcs spline

### T>G
answerals.base_substitutions.T2G = filter(answerals.base_substitutions, Substitution == "T>G")
answerals.base_substitutions.T2G.fitg = glm(count ~ CONDITION + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data=answerals.base_substitutions.T2G, family = poisson)
summ(answerals.base_substitutions.T2G.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.     z val.            p
# (Intercept)                                                     4.890101e+00 5.810173e-02  84.164466 0.000000e+00
# CONDITIONALS                                                    2.774698e-02 9.918008e-03   2.797637 5.147799e-03
answerals.base_substitutions.T2G.fitg_plot = glm(count ~ CONDITION + total_reads + age_at_sample_collection, data=answerals.base_substitutions.T2G, family = poisson) # cant plot rcs spline

# plot fitted values
answerals.base_substitutions.bind = bind_rows(mutate(partialize(answerals.base_substitutions.C2A.fitg_plot, vars = "CONDITION"), Substitution = "C>A"),
                                              mutate(partialize(answerals.base_substitutions.C2G.fitg_plot, vars = "CONDITION"), Substitution = "C>G"),
                                              mutate(partialize(answerals.base_substitutions.C2T.fitg_plot, vars = "CONDITION"), Substitution = "C>T"),
                                              mutate(partialize(answerals.base_substitutions.T2A.fitg_plot, vars = "CONDITION"), Substitution = "T>A"),
                                              mutate(partialize(answerals.base_substitutions.T2C.fitg_plot, vars = "CONDITION"), Substitution = "T>C"),
                                              mutate(partialize(answerals.base_substitutions.T2G.fitg_plot, vars = "CONDITION"), Substitution = "T>G"))
answerals.base_substitutions.bind %>% my_summarise(count, Substitution, CONDITION)

answerals.Substitution.stat <- answerals.base_substitutions.bind %>% group_by(Substitution) %>% t_test(count ~ CONDITION) %>% mutate(p.signif = c("****","*","****","****","****","ns"), y.position = c(3800,4850,18000,2400,23000,3600)) 
answerals.Substitution.stat
# Substitution .y.           group1 group2    n1    n2 statistic    df          p p.signif y.position
#   <fct>        <chr>         <chr>  <chr>  <int> <int>     <dbl> <dbl>      <dbl> <chr>         <dbl>
# 1 C>A          fitted.values CTRL   ALS       42   238     -3.28  55.0 0.00179    ****           4500
# 2 C>G          fitted.values CTRL   ALS       42   238     -2.76  55.3 0.00789    ***            5500
# 3 C>T          fitted.values CTRL   ALS       42   238     -2.90  55.0 0.00531    ****          20500
# 4 T>A          fitted.values CTRL   ALS       42   238     -3.88  55.7 0.00028    ****           2700
# 5 T>C          fitted.values CTRL   ALS       42   238     -4.88  53.7 0.00000997 ****          25000
# 6 T>G          fitted.values CTRL   ALS       42   238     -2.64  54.9 0.0108     ns             3900
answerals.base_substitutions.bind.filt = answerals.base_substitutions.bind %>%
  filter(Substitution == "C>A" & count < 3800 |  Substitution == "C>G" & count < 4900 |  Substitution == "C>T" & count < 18000 |  Substitution == "T>A" & count < 2400 |  Substitution == "T>C" & count < 23000 |  Substitution == "T>G" & count < 3600) # to scale the y axis with facets need to filter outliers

answerals.base_substitutions.violin <- ggplot(answerals.base_substitutions.bind.filt, aes(x = CONDITION, y = count, colour = CONDITION)) +
    ggrastr::rasterise(geom_violin(adjust = 1), dpi = 300) +
    ggrastr::rasterise(geom_sina(size = 0.5), dpi = 300) +
    ggrastr::rasterise(geom_boxplot(aes(fill = CONDITION), colour = "black", width=0.2, alpha = 0.5, outlier.shape = NA), dpi = 300) + 
    scale_colour_manual(values = c("dodgerblue2", "firebrick2")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
    facet_wrap(~ Substitution, scales = "free", nrow = 1)  + labs(y = "Adjusted Number of Mutations", x = "", title = "iPSMN: Base Substitutions") +
    stat_pvalue_manual(answerals.Substitution.stat, label = "p.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) + 
  theme_oz(xaxis_arrow = FALSE) + theme(legend.position = "none", plot.title = element_text(hjust = 0.5))
answerals.base_substitutions.violin

```

Postmortem numbers of substitutions

```{r}
# Run Rscript /camp/project/proj-luscombn-patani/working/nygc-als-consortium/rnavar/nygc_mutationalpatterns.R
# nygc_spinal_cord_metadata = read_csv(here(shared_path, "nygc-als-consortium/sample-details/nygc_spinal_cord_metadata.csv")) %>% 
#   mutate(vcf_path = paste0("/camp/project/proj-luscombn-patani/working/nygc-als-consortium/rnavar/variant_calling/",sample, "/", sample, ".redi.filtered.recode.vcf.gz"))
# # http://bioconductor.org/packages/release/bioc/vignettes/MutationalPatterns/inst/doc/Introduction_to_MutationalPatterns.html
# # BiocManager::install("MutationalPatterns")
# library(MutationalPatterns)
# ref_genome <- "BSgenome.Hsapiens.UCSC.hg38"
# library(ref_genome, character.only = TRUE)
# 
# # read in vcf files as granges
# grl <- read_vcfs_as_granges(nygc_spinal_cord_metadata$vcf_path, nygc_spinal_cord_metadata$sample, ref_genome, type = "all")
# snv_grl <- get_mut_type(grl, type = "snv") # get types of mutations SNV, indels
# indel_grl <- get_mut_type(grl, type = "indel")
# dbs_grl <- get_mut_type(grl, type = "dbs")
# mbs_grl <- get_mut_type(grl, type = "mbs")
# type_occurrences <- mut_type_occurrences(snv_grl, ref_genome) # count mutation type occurences and base substitutions
# 
# save(grl, snv_grl, indel_grl, dbs_grl, mbs_grl, type_occurrences,
#      file =  "/camp/project/proj-luscombn-patani/working/nygc-als-consortium/rnavar/mutationalpatterns/mutationalpatterns.RData")
nygc_spinal_cord.type_occurrences = readRDS(here(nygc_path, "rnavar/mutationalpatterns/type_occurrences.rds"))
nygc.rnavar.multiqc_star.txt = read_tsv(here(nygc_path,"rnavar/reports/multiqc_data/multiqc_star.txt")) %>% clean_names()
nygc_spinal_cord.base_substitutions = nygc_spinal_cord.type_occurrences %>% as_tibble(rownames = "sample") %>% select(-`C>T at CpG`, -`C>T other`) %>%
  left_join(select(nygc_spinal_cord_metadata, sample, condition, age_at_death, library_prep)) %>% 
  pivot_longer(!c(condition, sample, age_at_death, library_prep), names_to = "Substitution", values_to = "count") %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), Substitution = as.factor(Substitution)) %>%
  left_join(select(nygc.rnavar.multiqc_star.txt, sample, total_reads)) # # left_join multiqc stats
nygc_spinal_cord.base_substitutions  

# For count data use glm(count ~ condition, family=poisson) which gives estimates and p-values of the underlying 'rate's of fusion per sample 
### C>A
nygc_spinal_cord.base_substitutions.C2A = filter(nygc_spinal_cord.base_substitutions, Substitution == "C>A")
nygc_spinal_cord.base_substitutions.C2A.fitg = glm(count ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.base_substitutions.C2A, family = poisson)
summ(nygc_spinal_cord.base_substitutions.C2A.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.    z val.            p
# (Intercept)                             4.543133e+00 4.722124e-02 96.209511 0.000000e+00
# CONDITIONALS                           -6.897747e-02 1.040862e-02 -6.626956 3.426793e-11
# generalised linear model spline # library(rms) # https://cran.r-project.org/web/packages/jtools/vignettes/summ.html
nygc_spinal_cord.base_substitutions.C2A.fitg_plot = glm(count ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.base_substitutions.C2A, family = poisson) # cant plot rcs spline
plot_summs(nygc_spinal_cord.base_substitutions.C2A.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

### C>G
nygc_spinal_cord.base_substitutions.C2G = filter(nygc_spinal_cord.base_substitutions, Substitution == "C>G")
nygc_spinal_cord.base_substitutions.C2G.fitg = glm(count ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.base_substitutions.C2G, family = poisson)
summ(nygc_spinal_cord.base_substitutions.C2G.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.      z val.            p
# (Intercept)                             5.225806e+00 4.034701e-02 129.5215206 0.000000e+00
# CONDITIONALS                           -3.077788e-02 8.820840e-03  -3.4892233 4.844262e-04
nygc_spinal_cord.base_substitutions.C2G.fitg_plot = glm(count ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.base_substitutions.C2G, family = poisson) # cant plot rcs spline
plot_summs(nygc_spinal_cord.base_substitutions.C2G.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

### C>T
nygc_spinal_cord.base_substitutions.C2T = filter(nygc_spinal_cord.base_substitutions, Substitution == "C>T")
nygc_spinal_cord.base_substitutions.C2T.fitg = glm(count ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.base_substitutions.C2T, family = poisson)
summ(nygc_spinal_cord.base_substitutions.C2T.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.             p
# (Intercept)                             6.191884e+00 2.384509e-02 259.671236  0.000000e+00
# CONDITIONALS                           -4.731242e-02 5.218973e-03  -9.065465  1.240735e-19
nygc_spinal_cord.base_substitutions.C2T.fitg_plot = glm(count ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.base_substitutions.C2T, family = poisson) # cant plot rcs spline
plot_summs(nygc_spinal_cord.base_substitutions.C2T.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

### T>A
nygc_spinal_cord.base_substitutions.T2A = filter(nygc_spinal_cord.base_substitutions, Substitution == "T>A")
nygc_spinal_cord.base_substitutions.T2A.fitg = glm(count ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.base_substitutions.T2A, family = poisson)
summ(nygc_spinal_cord.base_substitutions.T2A.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.            p
# (Intercept)                             4.367688e+00 5.495137e-02 79.4827897 0.000000e+00
# CONDITIONALS                            7.559077e-03 1.231462e-02  0.6138294 5.393280e-01
nygc_spinal_cord.base_substitutions.T2A.fitg_plot = glm(count ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.base_substitutions.T2A, family = poisson) # cant plot rcs spline
plot_summs(nygc_spinal_cord.base_substitutions.T2A.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

### T>C
nygc_spinal_cord.base_substitutions.T2C = filter(nygc_spinal_cord.base_substitutions, Substitution == "T>C")
nygc_spinal_cord.base_substitutions.T2C.fitg = glm(count ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.base_substitutions.T2C, family = poisson)
summ(nygc_spinal_cord.base_substitutions.T2C.fitg, digits = 5)$coeftable 
#                                                Est.         S.E.     z val.             p
# (Intercept)                             8.001965e+00 8.166529e-03  979.84903  0.000000e+00
# CONDITIONALS                            6.078008e-02 1.772091e-03   34.29852 8.257111e-258
nygc_spinal_cord.base_substitutions.T2C.fitg_plot = glm(count ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.base_substitutions.T2C, family = poisson) # cant plot rcs spline
plot_summs(nygc_spinal_cord.base_substitutions.T2C.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

### T>G
nygc_spinal_cord.base_substitutions.T2G = filter(nygc_spinal_cord.base_substitutions, Substitution == "T>G")
nygc_spinal_cord.base_substitutions.T2G.fitg = glm(count ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data=nygc_spinal_cord.base_substitutions.T2G, family = poisson)
summ(nygc_spinal_cord.base_substitutions.T2G.fitg, digits = 5)$coeftable 
#                                                Est.         S.E.      z val.            p
# (Intercept)                             5.200214e+00 4.461146e-02 116.5667653 0.000000e+00
# CONDITIONALS                            2.607917e-02 1.004530e-02   2.5961579 9.427274e-03
nygc_spinal_cord.base_substitutions.T2G.fitg_plot = glm(count ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.base_substitutions.T2G, family = poisson) # cant plot rcs spline
plot_summs(nygc_spinal_cord.base_substitutions.T2G.fitg_plot, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction

# plot fitted values
nygc_spinal_cord.base_substitutions.bind = bind_rows(mutate(partialize(nygc_spinal_cord.base_substitutions.C2A.fitg_plot, vars = "CONDITION"), Substitution = "C>A"),
                                              mutate(partialize(nygc_spinal_cord.base_substitutions.C2G.fitg_plot, vars = "CONDITION"), Substitution = "C>G"),
                                              mutate(partialize(nygc_spinal_cord.base_substitutions.C2T.fitg_plot, vars = "CONDITION"), Substitution = "C>T"),
                                              mutate(partialize(nygc_spinal_cord.base_substitutions.T2A.fitg_plot, vars = "CONDITION"), Substitution = "T>A"),
                                              mutate(partialize(nygc_spinal_cord.base_substitutions.T2C.fitg_plot, vars = "CONDITION"), Substitution = "T>C"),
                                              mutate(partialize(nygc_spinal_cord.base_substitutions.T2G.fitg_plot, vars = "CONDITION"), Substitution = "T>G"))
nygc_spinal_cord.base_substitutions.bind %>% my_summarise(count, Substitution, CONDITION)

nygc_spinal_cord.Substitution.stat <- nygc_spinal_cord.base_substitutions.bind %>% group_by(Substitution) %>% t_test(count ~ CONDITION) %>% mutate(p.signif = c("****","****","****","****","****","*"), y.position = c(4000,5500,20000,2600,26500,3800)) 
nygc_spinal_cord.Substitution.stat
# Substitution .y.   group1 group2    n1    n2 statistic    df       p p.signif y.position
#   <chr>        <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>   <dbl> <chr>         <dbl>
# 1 C>A          count CTRL   ALS       57   214     1.96   68.1 0.0541  ****           4000
# 2 C>G          count CTRL   ALS       57   214     3.01   71.8 0.00365 ****           5500
# 3 C>T          count CTRL   ALS       57   214     2.38   68.5 0.02    ****          20000
# 4 T>A          count CTRL   ALS       57   214    -0.459  68.1 0.647   ****           2600
# 5 T>C          count CTRL   ALS       57   214     0.144  81.0 0.886   ****          28000
# 6 T>G          count CTRL   ALS       57   214     1.09   70.3 0.278   *              3800
nygc_spinal_cord.base_substitutions.bind.filt = nygc_spinal_cord.base_substitutions.bind %>%
  filter(Substitution == "C>A" & count < 4000 |  Substitution == "C>G" & count < 5500 |  Substitution == "C>T" & count < 20000 |  Substitution == "T>A" & count < 2600 |  Substitution == "T>C" & count < 26500 |  Substitution == "T>G" & count < 3800) # to scale the y axis with facets need to filter outliers

nygc_spinal_cord.base_substitutions.violin <- ggplot(nygc_spinal_cord.base_substitutions.bind.filt, aes(x = CONDITION, y = count, colour = CONDITION)) +
    ggrastr::rasterise(geom_violin(adjust = 1), dpi = 300) +
    ggrastr::rasterise(geom_sina(size = 0.5), dpi = 300) +
    ggrastr::rasterise(geom_boxplot(aes(fill = CONDITION), colour = "black", width=0.2, alpha = 0.5, outlier.shape = NA), dpi = 300) + 
    scale_colour_manual(values = c("dodgerblue2", "firebrick2")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
    facet_wrap(~ Substitution, scales = "free", nrow = 1)  + labs(y = "Adjusted Number of Mutations", x = "", title = "Postmortem: Base Substitutions") +
    stat_pvalue_manual(nygc_spinal_cord.Substitution.stat, label = "p.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) + 
  theme_oz(xaxis_arrow = FALSE) + theme(legend.position = "none", plot.title = element_text(hjust = 0.5))
# nygc_spinal_cord.base_substitutions.violin

```



## Fusions types

### iPSMNs

```{r}
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
ipsc_mn_als_datasets.starfusion = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>%
  left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion

ipsc_mn_als_datasets.starfusion.fusion_type_n = ipsc_mn_als_datasets.starfusion %>%
  add_count(dataset_sample, fusion_type) %>% # count number of fusion types per sample
  distinct(dataset_sample, fusion_type, .keep_all = TRUE) %>% select(dataset_sample, fusion_type, n, condition, dataset, total_reads, age_at_sample_collection) #keep 1 row per sample fusion type
ipsc_mn_als_datasets.starfusion.fusion_type_n

# fusions types detected in X samples in ALS & CTRL:
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Neighbours") %>% count(condition) # 89/90 98.9% 302/306 98.7%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Local\nRearrangement") %>% count(condition) # 46/90 51.1% 253/306 82.7%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Distant\nProximity") %>% count(condition) # 64/90 71.1% 254/306 83.0%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Overlapping\nNeighbours") %>% count(condition) #37/90 41.1% 197/306 64.4%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Inter\nChromosomal") %>% count(condition) # 27/90 30% 105/306 34%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Local\nInversion") %>% count(condition) #8/90 8.9% 5/306 1.6%

# add samples with 0 fusions for each fusion type
samples_no_neighbours = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Neighbours"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_neighbours = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_neighbours) %>% mutate(fusion_type = "Neighbours", n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type, age_at_sample_collection)
ipsc_mn_als_datasets.paired_end.metadata.no_neighbours

samples_no_local_rearrangement = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Local\nRearrangement"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_local_rearrangement = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_local_rearrangement) %>% mutate(fusion_type = "Local\nRearrangement", n = 0) %>%
  select(dataset_sample, n, condition, fusion_type, age_at_sample_collection)
ipsc_mn_als_datasets.paired_end.metadata.no_local_rearrangement

samples_no_local_inversion = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Local\nInversion"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_local_inversion = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_local_inversion) %>% mutate(fusion_type = "Local\nInversion", n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type, age_at_sample_collection)
ipsc_mn_als_datasets.paired_end.metadata.no_local_inversion

samples_no_overlapping_neighbours = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Overlapping\nNeighbours"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_overlapping_neighbours = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_overlapping_neighbours) %>% mutate(fusion_type = "Overlapping\nNeighbours", n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type, age_at_sample_collection)
ipsc_mn_als_datasets.paired_end.metadata.no_overlapping_neighbours

samples_no_distant_proximity = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Distant\nProximity" ),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_distant_proximity = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_distant_proximity) %>% mutate(fusion_type = "Distant\nProximity" , n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type, age_at_sample_collection)
ipsc_mn_als_datasets.paired_end.metadata.no_distant_proximity

samples_no_inter_chromosomal = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Inter\nChromosomal"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_inter_chromosomal = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_inter_chromosomal) %>% mutate(fusion_type = "Inter\nChromosomal", n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type, age_at_sample_collection)
ipsc_mn_als_datasets.paired_end.metadata.no_inter_chromosomal

ipsc_mn_als_datasets.starfusion.fusion_type_n = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type_n, ipsc_mn_als_datasets.paired_end.metadata.no_neighbours, ipsc_mn_als_datasets.paired_end.metadata.no_local_rearrangement, ipsc_mn_als_datasets.paired_end.metadata.no_local_inversion,  ipsc_mn_als_datasets.paired_end.metadata.no_overlapping_neighbours, ipsc_mn_als_datasets.paired_end.metadata.no_distant_proximity, ipsc_mn_als_datasets.paired_end.metadata.no_inter_chromosomal) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), 
         fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")), age_at_sample_collection = case_when(is.na(age_at_sample_collection) ~ 58, TRUE ~ age_at_sample_collection)) %>% 
  select(-total_reads) %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.fusion_type_n

ipsc_mn_als_datasets.starfusion.fusion_type_n %>% my_summarise(n, CONDITION, fusion_type)


# For count data use glm(count ~ condition, family=poisson) which gives estimates and p-values of the underlying 'rate's of fusion per sample 
### Neighbours
ipsc_mn_als_datasets.starfusion.fusion_neighbours = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Neighbours")
ipsc_mn_als_datasets.starfusion.fusion_neighbours.fitg = glm(n ~ CONDITION + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.fusion_neighbours, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.fusion_neighbours.fitg, digits = 5)$coeftable 
#                                                                      Est.         S.E.     z val.            p
# (Intercept)                                                     9.236723e-01 5.490430e-01  1.6823314 9.250460e-02
# CONDITIONALS                                                    7.786268e-02 6.063703e-02  1.2840781 1.991146e-01
plot_summs(ipsc_mn_als_datasets.starfusion.fusion_neighbours.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
ipsc_mn_als_datasets.starfusion.fusion_neighbours.fitg_plot = glm(n ~ CONDITION + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.fusion_neighbours, family = poisson) 

### Local_Inversion
ipsc_mn_als_datasets.starfusion.fusion_local_inversion = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Local\nInversion")
ipsc_mn_als_datasets.starfusion.fusion_local_inversion.fitg = glm(n ~ CONDITION + dataset + rms::rcs(total_reads, 3), data = ipsc_mn_als_datasets.starfusion.fusion_local_inversion, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.fusion_local_inversion.fitg, digits = 5)$coeftable 
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)
# (Intercept)                          -1.068e+00  7.421e+00  -0.144    0.886
# CONDITIONALS                          4.430e-02  6.734e-01   0.066    0.948
plot_summs(ipsc_mn_als_datasets.starfusion.fusion_local_inversion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
ipsc_mn_als_datasets.starfusion.fusion_local_inversion.fitg_plot = glm(n ~ CONDITION + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.fusion_local_inversion, family = poisson) 

### Local_Rearrangement
ipsc_mn_als_datasets.starfusion.fusion_local_rearrangement = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Local\nRearrangement")
ipsc_mn_als_datasets.starfusion.fusion_local_rearrangement.fitg = glm(n ~ CONDITION + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.fusion_local_rearrangement, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.fusion_local_rearrangement.fitg, digits = 5)$coeftable 
#                                                                         Est.         S.E.     z val.            p
# (Intercept)                                                     1.145374e+00 7.756622e-01  1.4766406 0.1397719960
# CONDITIONALS                                                   -2.214344e-02 9.991560e-02 -0.2216215 0.8246085794
plot_summs(ipsc_mn_als_datasets.starfusion.fusion_local_rearrangement.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
ipsc_mn_als_datasets.starfusion.fusion_local_rearrangement.fitg_plot = glm(n ~ CONDITION + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.fusion_local_rearrangement, family = poisson) 

### Overlapping_overlapping_neighbours
ipsc_mn_als_datasets.starfusion.fusion_overlapping_neighbours = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Overlapping\nNeighbours")
ipsc_mn_als_datasets.starfusion.fusion_overlapping_neighbours.fitg = glm(n ~ CONDITION + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.fusion_overlapping_neighbours, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.fusion_overlapping_neighbours.fitg, digits = 5)$coeftable 
# Coefficients:
#                                                                  Estimate Std. Error z value Pr(>|z|)
# (Intercept)                                                    -4.542e-01  1.426e+00  -0.319    0.750
# CONDITIONALS                                                    3.301e-02  1.895e-01   0.174    0.862
plot_summs(ipsc_mn_als_datasets.starfusion.fusion_overlapping_neighbours.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
ipsc_mn_als_datasets.starfusion.fusion_overlapping_neighbours.fitg_plot = glm(n ~ CONDITION + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.fusion_overlapping_neighbours, family = poisson) 

### Distant_Proximity
ipsc_mn_als_datasets.starfusion.fusion_distant_proximity = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Distant\nProximity")
ipsc_mn_als_datasets.starfusion.fusion_distant_proximity.fitg = glm(n ~ CONDITION + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.fusion_distant_proximity, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.fusion_distant_proximity.fitg, digits = 5)$coeftable 
#                                                                        Est.         S.E.     z val.          p
# (Intercept)                                                     5.289742e-01 9.958564e-01  0.5311751 0.59529741
# CONDITIONALS                                                    1.273978e-02 1.051734e-01  0.1211312 0.90358712
plot_summs(ipsc_mn_als_datasets.starfusion.fusion_distant_proximity.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
ipsc_mn_als_datasets.starfusion.fusion_distant_proximity.fitg_plot = glm(n ~ CONDITION + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.fusion_distant_proximity, family = poisson) 

### Inter_Chromosomal
ipsc_mn_als_datasets.starfusion.fusion_inter_chromosomal = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Inter\nChromosomal")
ipsc_mn_als_datasets.starfusion.fusion_inter_chromosomal.fitg = glm(n ~ CONDITION + dataset + rms::rcs(total_reads, 3) + rms::rcs(age_at_sample_collection, 3), data = ipsc_mn_als_datasets.starfusion.fusion_inter_chromosomal, family=poisson)
summ(ipsc_mn_als_datasets.starfusion.fusion_inter_chromosomal.fitg, digits = 5)$coeftable 
# Coefficients:
#                                                                  Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                                                     3.193e-02  1.652e+00   0.019 0.984585    
# CONDITIONALS                                                    6.412e-01  1.726e-01   3.716 0.000203 ***
plot_summs(ipsc_mn_als_datasets.starfusion.fusion_inter_chromosomal.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
ipsc_mn_als_datasets.starfusion.fusion_inter_chromosomal.fitg_plot = glm(n ~ CONDITION + dataset + total_reads + age_at_sample_collection, data=ipsc_mn_als_datasets.starfusion.fusion_inter_chromosomal, family = poisson) 

# plot fitted values
ipsc_mn_als_datasets.starfusion.fusion_types.bind = bind_rows(mutate(partialize(ipsc_mn_als_datasets.starfusion.fusion_neighbours.fitg_plot, vars = "CONDITION"), fusion_type = "Neighbours"),
                                              # mutate(partialize(ipsc_mn_als_datasets.starfusion.fusion_local_inversion.fitg_plot, vars = "CONDITION"), fusion_type = "Local\nInversion"),
                                              mutate(partialize(ipsc_mn_als_datasets.starfusion.fusion_local_rearrangement.fitg_plot, vars = "CONDITION"), fusion_type = "Local\nRearrangement"),
                                              mutate(partialize(ipsc_mn_als_datasets.starfusion.fusion_overlapping_neighbours.fitg_plot, vars = "CONDITION"), fusion_type = "Overlapping\nNeighbours"),
                                              mutate(partialize(ipsc_mn_als_datasets.starfusion.fusion_distant_proximity.fitg_plot, vars = "CONDITION"), fusion_type = "Distant\nProximity"),
                                              mutate(partialize(ipsc_mn_als_datasets.starfusion.fusion_inter_chromosomal.fitg_plot, vars = "CONDITION"), fusion_type = "Inter\nChromosomal"))
ipsc_mn_als_datasets.starfusion.fusion_types.bind %>% my_summarise(n, fusion_type, CONDITION)

ipsc_mn_als_datasets.starfusion.fusion_types.stat <- ipsc_mn_als_datasets.starfusion.fusion_types.bind %>% #filter(!fusion_type == "Local\nInversion") %>%  
  group_by(fusion_type) %>% t_test(n ~ CONDITION) %>% mutate(p.signif = c("ns","****","ns","ns","ns"), y.position = 6) 
ipsc_mn_als_datasets.starfusion.fusion_types.stat
# fusion_type               .y.   group1 group2    n1    n2 statistic    df      p p.signif y.position
#   <chr>                     <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>  <dbl> <chr>         <dbl>
# 1 "Distant\nProximity"      n     CTRL   ALS       64   254  -0.223   116.  0.824  ns               24
# 2 "Inter\nChromosomal"      n     CTRL   ALS       27   105  -0.0826   31.0 0.935  ****             24
# 3 "Local\nInversion"        n     CTRL   ALS        8     5   0.00244  10.9 0.998  ns               24
# 4 "Local\nRearrangement"    n     CTRL   ALS       46   253  -0.190    66.0 0.85   ns               24
# 5 "Neighbours"              n     CTRL   ALS       89   302  -1.95    311.  0.0517 ns               24
# 6 "Overlapping\nNeighbours" n     CTRL   ALS       37   197  -1.36    126.  0.175  ns               24
ipsc_mn_als_datasets.starfusion.fusion_types.bind.filt = ipsc_mn_als_datasets.starfusion.fusion_types.bind %>%
  filter(fusion_type == "Neighbours" & n < 40 | fusion_type == "Local\nRearrangement" & n < 30 | fusion_type == "Overlapping\nNeighbours" & n < 3 | fusion_type == "Distant\nProximity" & n < 10 | fusion_type == "Inter\nChromosomal" & n < 6) # to scale the y axis with facets need to filter outliers
ipsc_mn_als_datasets.starfusion.fusion_type.violin <- ggplot(ipsc_mn_als_datasets.starfusion.fusion_types.bind.filt, aes(x = CONDITION, y = n, colour = CONDITION)) +
    # ggrastr::rasterise(geom_violin(adjust = 1), dpi = 300) +
    ggrastr::rasterise(geom_sina(size = 0.5), dpi = 300) +
    ggrastr::rasterise(geom_boxplot(aes(fill = CONDITION), colour = "black", width=0.2, alpha = 0.5, outlier.shape = NA), dpi = 300) + 
    scale_colour_manual(values = c("dodgerblue2", "firebrick2")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
    facet_wrap(~ fusion_type, scales = "free", nrow = 1)  + labs(y = "Adjusted Number of Fusions", x = "", title = "iPSMN: Fusion Types") +
    stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_types.stat, label = "p.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) + 
  theme_oz(xaxis_arrow = FALSE) + theme(legend.position = "none", plot.title = element_text(hjust = 0.5))
# ipsc_mn_als_datasets.starfusion.fusion_type.violin

```


### Postmortem

```{r}
nygc_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
nygc.multiqc_star.txt = read_tsv(here(nygc_path,"rnafusion/multiqc/multiqc_data/multiqc_star.txt")) %>% clean_names()
nygc.multiqc_star.txt %>% glimpse
nygc_spinal_cord.starfusion = read_csv(here(nygc_path,"rnafusion/nygc_spinal_cord.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))  %>% 
  left_join(nygc.multiqc_star.txt) # left_join multiqc stats
nygc_spinal_cord.starfusion.fusion_type_n = nygc_spinal_cord.starfusion %>% 
  add_count(sample, fusion_type) %>% # count number of fusion types per sample
  distinct(sample, fusion_type, .keep_all = TRUE) %>% select(sample, fusion_type, n, condition, total_reads, age_at_death, library_prep) #keep 1 row per sample fusion type
nygc_spinal_cord.starfusion.fusion_type_n

# fusions types detected in X samples in ALS & CTRL:
nygc_spinal_cord.starfusion %>% count(condition)
filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Neighbours") %>% count(condition) # 57/81 98.9% 214/274 98.7%
filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Local\nRearrangement") %>% count(condition) # 71/81 51.1% 230/274 82.7%
filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Distant\nProximity") %>% count(condition) # 59/81 71.1% 217/274 83.0%
filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Overlapping\nNeighbours") %>% count(condition) #71/81 41.1% 246/274 64.4%
filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Inter\nChromosomal") %>% count(condition) # 79/81 30% 265/274 34%
filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Local\nInversion") %>% count(condition) # 81/81 8.9% 274/274 1.6%

# add samples with 0 fusions for each fusion type
samples_no_neighbours = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Neighbours"),sample)]
nygc_spinal_cord_metadata.no_neighbours = nygc_spinal_cord_metadata %>% filter(sample %in% samples_no_neighbours) %>% mutate(fusion_type = "Neighbours", n = 0) %>% 
  select(sample, n, condition, fusion_type, age_at_death, library_prep)
nygc_spinal_cord_metadata.no_neighbours

samples_no_local_rearrangement = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Local\nRearrangement"),sample)]
nygc_spinal_cord_metadata.no_local_rearrangement = nygc_spinal_cord_metadata %>% filter(sample %in% samples_no_local_rearrangement) %>% mutate(fusion_type = "Local\nRearrangement", n = 0) %>%
  select(sample, n, condition, fusion_type, age_at_death, library_prep)
nygc_spinal_cord_metadata.no_local_rearrangement

samples_no_local_inversion = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Local\nInversion"),sample)]
nygc_spinal_cord_metadata.no_local_inversion = nygc_spinal_cord_metadata %>% filter(sample %in% samples_no_local_inversion) %>% mutate(fusion_type = "Local\nInversion", n = 0) %>% 
  select(sample, n, condition, fusion_type, age_at_death, library_prep)
nygc_spinal_cord_metadata.no_local_inversion

samples_no_overlapping_neighbours = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Overlapping\nNeighbours"),sample)]
nygc_spinal_cord_metadata.no_overlapping_neighbours = nygc_spinal_cord_metadata %>% filter(sample %in% samples_no_overlapping_neighbours) %>% mutate(fusion_type = "Overlapping\nNeighbours", n = 0) %>% 
  select(sample, n, condition, fusion_type, age_at_death, library_prep)
nygc_spinal_cord_metadata.no_overlapping_neighbours

samples_no_distant_proximity = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Distant\nProximity" ),sample)]
nygc_spinal_cord_metadata.no_distant_proximity = nygc_spinal_cord_metadata %>% filter(sample %in% samples_no_distant_proximity) %>% mutate(fusion_type = "Distant\nProximity" , n = 0) %>% 
  select(sample, n, condition, fusion_type, age_at_death, library_prep)
nygc_spinal_cord_metadata.no_distant_proximity

samples_no_inter_chromosomal = nygc_spinal_cord_metadata$sample[!nygc_spinal_cord_metadata$sample %in% pull(filter(nygc_spinal_cord.starfusion.fusion_type_n,fusion_type == "Inter\nChromosomal"),sample)]
nygc_spinal_cord_metadata.no_inter_chromosomal = nygc_spinal_cord_metadata %>% filter(sample %in% samples_no_inter_chromosomal) %>% mutate(fusion_type = "Inter\nChromosomal", n = 0) %>% 
  select(sample, n, condition, fusion_type, age_at_death, library_prep)
nygc_spinal_cord_metadata.no_inter_chromosomal

nygc_spinal_cord.starfusion.fusion_type_n = bind_rows(nygc_spinal_cord.starfusion.fusion_type_n, nygc_spinal_cord_metadata.no_neighbours, nygc_spinal_cord_metadata.no_local_rearrangement, nygc_spinal_cord_metadata.no_local_inversion, nygc_spinal_cord_metadata.no_overlapping_neighbours, nygc_spinal_cord_metadata.no_distant_proximity, nygc_spinal_cord_metadata.no_inter_chromosomal) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), 
         fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal"))) %>% 
  select(-total_reads) %>% left_join(nygc.multiqc_star.txt) %>% # left_join multiqc stats
  select(-age_at_death) %>% left_join(select(nygc_spinal_cord_metadata,sample,age_at_death, library_prep)) # age at death
nygc_spinal_cord.starfusion.fusion_type_n 

nygc_spinal_cord.starfusion.fusion_type_n %>% my_summarise(n, CONDITION, fusion_type) %>% mutate(IQR = Q3_n - Q1_n)

### Neighbours
nygc_spinal_cord.starfusion.fusion_neighbours = filter(nygc_spinal_cord.starfusion.fusion_type_n, fusion_type == "Neighbours")
nygc_spinal_cord.starfusion.fusion_neighbours.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.fusion_neighbours, family=poisson)
summ(nygc_spinal_cord.starfusion.fusion_neighbours.fitg, digits = 5)$coeftable 
#                                                Est.         S.E.     z val.            p
# (Intercept)                             1.628785e+00 3.052927e-01  5.3351563 9.546228e-08
# CONDITIONALS                            2.976713e-01 7.817125e-02  3.8079382 1.401303e-04
plot_summs(nygc_spinal_cord.starfusion.fusion_neighbours.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
nygc_spinal_cord.starfusion.fusion_neighbours.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.fusion_neighbours, family = poisson) 

### Local_Inversion
nygc_spinal_cord.starfusion.fusion_local_inversion = filter(nygc_spinal_cord.starfusion.fusion_type_n, fusion_type == "Local\nInversion")
nygc_spinal_cord.starfusion.fusion_local_inversion.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.fusion_local_inversion, family=poisson)
summ(nygc_spinal_cord.starfusion.fusion_local_inversion.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.        z val.         p
# (Intercept)                            -2.730259e+01 3.578189e+05 -7.630281e-05 0.9999391
# CONDITIONALS                           -1.986557e-14 8.081681e+04 -2.458099e-19 1.0000000
# plot_summs(nygc_spinal_cord.starfusion.fusion_local_inversion.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
nygc_spinal_cord.starfusion.fusion_local_inversion.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.fusion_local_inversion, family = poisson) 

### Local_Rearrangement
nygc_spinal_cord.starfusion.fusion_local_rearrangement = filter(nygc_spinal_cord.starfusion.fusion_type_n, fusion_type == "Local\nRearrangement")
nygc_spinal_cord.starfusion.fusion_local_rearrangement.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.fusion_local_rearrangement, family=poisson)
summ(nygc_spinal_cord.starfusion.fusion_local_rearrangement.fitg, digits = 5)$coeftable 
#                                                 Est.         S.E.     z val.            p
# (Intercept)                             4.357344e-01 5.525502e-01  0.7885879 4.303529e-01
# CONDITIONALS                            3.226472e-01 1.417770e-01  2.2757375 2.286173e-02
plot_summs(nygc_spinal_cord.starfusion.fusion_local_rearrangement.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
nygc_spinal_cord.starfusion.fusion_local_rearrangement.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.fusion_local_rearrangement, family = poisson) 

### Overlapping_neighbours
nygc_spinal_cord.starfusion.fusion_overlapping_neighbours = filter(nygc_spinal_cord.starfusion.fusion_type_n, fusion_type == "Overlapping\nNeighbours")
nygc_spinal_cord.starfusion.fusion_overlapping_neighbours.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.fusion_overlapping_neighbours, family=poisson)
summ(nygc_spinal_cord.starfusion.fusion_overlapping_neighbours.fitg, digits = 5)$coeftable 
#                                                Est.         S.E.     z val.            p
# (Intercept)                            -5.033177e-01 6.808116e-01 -0.7392908 4.597305e-01
# CONDITIONALS                            8.256064e-02 1.631676e-01  0.5059869 6.128659e-01
plot_summs(nygc_spinal_cord.starfusion.fusion_overlapping_neighbours.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
nygc_spinal_cord.starfusion.fusion_overlapping_neighbours.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.fusion_overlapping_neighbours, family = poisson) 

### Distant_Proximity
nygc_spinal_cord.starfusion.fusion_distant_proximity = filter(nygc_spinal_cord.starfusion.fusion_type_n, fusion_type == "Distant\nProximity")
nygc_spinal_cord.starfusion.fusion_distant_proximity.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.fusion_distant_proximity, family=poisson)
summ(nygc_spinal_cord.starfusion.fusion_distant_proximity.fitg, digits = 5)$coeftable 
#                                                Est.         S.E.     z val.            p
# (Intercept)                            -9.983401e-02 5.245102e-01 -0.1903376 0.8490445921
# CONDITIONALS                            1.897734e-01 1.223668e-01  1.5508569 0.1209359711
plot_summs(nygc_spinal_cord.starfusion.fusion_distant_proximity.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
nygc_spinal_cord.starfusion.fusion_distant_proximity.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.fusion_distant_proximity, family = poisson) 

### Inter_Chromosomal
nygc_spinal_cord.starfusion.fusion_inter_chromosomal = filter(nygc_spinal_cord.starfusion.fusion_type_n, fusion_type == "Inter\nChromosomal")
nygc_spinal_cord.starfusion.fusion_inter_chromosomal.fitg = glm(n ~ CONDITION + library_prep + rms::rcs(total_reads, 3) + rms::rcs(age_at_death, 3), data = nygc_spinal_cord.starfusion.fusion_inter_chromosomal, family=poisson)
summ(nygc_spinal_cord.starfusion.fusion_inter_chromosomal.fitg, digits = 5)$coeftable 
#                                                Est.         S.E.      z val.         p
# (Intercept)                            -3.347617e+00 2.542251e+00 -1.31679258 0.1879082
# CONDITIONALS                            1.003871e-01 5.616813e-01  0.17872609 0.8581528
plot_summs(nygc_spinal_cord.starfusion.fusion_inter_chromosomal.fitg, plot.distributions = TRUE, rescale.distributions = TRUE) # effect direction
nygc_spinal_cord.starfusion.fusion_inter_chromosomal.fitg_plot = glm(n ~ CONDITION + library_prep + total_reads + age_at_death, data=nygc_spinal_cord.starfusion.fusion_inter_chromosomal, family = poisson) 

# plot fitted values
nygc_spinal_cord.starfusion.fusion_types.bind = bind_rows(mutate(partialize(nygc_spinal_cord.starfusion.fusion_neighbours.fitg_plot, vars = "CONDITION"), fusion_type = "Neighbours"),
                                              # mutate(partialize(nygc_spinal_cord.starfusion.fusion_local_inversion.fitg_plot, vars = "CONDITION"), fusion_type = "Local\nInversion"),
                                              mutate(partialize(nygc_spinal_cord.starfusion.fusion_local_rearrangement.fitg_plot, vars = "CONDITION"), fusion_type = "Local\nRearrangement"),
                                              mutate(partialize(nygc_spinal_cord.starfusion.fusion_overlapping_neighbours.fitg_plot, vars = "CONDITION"), fusion_type = "Overlapping\nNeighbours"),
                                              mutate(partialize(nygc_spinal_cord.starfusion.fusion_distant_proximity.fitg_plot, vars = "CONDITION"), fusion_type = "Distant\nProximity"),
                                              mutate(partialize(nygc_spinal_cord.starfusion.fusion_inter_chromosomal.fitg_plot, vars = "CONDITION"), fusion_type = "Inter\nChromosomal"))
nygc_spinal_cord.starfusion.fusion_types.bind %>% my_summarise(n, fusion_type, CONDITION)

nygc_spinal_cord.starfusion.fusion_types.stat <- nygc_spinal_cord.starfusion.fusion_types.bind %>% #filter(!fusion_type == "Local\nInversion") %>%  
  group_by(fusion_type) %>% t_test(n ~ CONDITION) %>% mutate(p.signif = c("ns","ns","*","****","ns"), y.position = c(15,1,40,25,20)) 
nygc_spinal_cord.starfusion.fusion_types.stat
# fusion_type               .y.   group1 group2    n1    n2 statistic    df      p p.signif y.position
#   <chr>                     <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>  <dbl> <chr>         <dbl>
# 1 "Distant\nProximity"      n     CTRL   ALS       64   254  -0.223   116.  0.824  ns               24
# 2 "Inter\nChromosomal"      n     CTRL   ALS       27   105  -0.0826   31.0 0.935  ****             24
# 3 "Local\nInversion"        n     CTRL   ALS        8     5   0.00244  10.9 0.998  ns               24
# 4 "Local\nRearrangement"    n     CTRL   ALS       46   253  -0.190    66.0 0.85   ns               24
# 5 "Neighbours"              n     CTRL   ALS       89   302  -1.95    311.  0.0517 ns               24
# 6 "Overlapping\nNeighbours" n     CTRL   ALS       37   197  -1.36    126.  0.175  ns               24

nygc_spinal_cord.starfusion.fusion_types.bind.filt = nygc_spinal_cord.starfusion.fusion_types.bind %>%
  filter(fusion_type == "Neighbours" & n < 25 | fusion_type == "Local\nRearrangement" & n < 40 | fusion_type == "Overlapping\nNeighbours" & n < 25 | fusion_type == "Distant\nProximity" & n < 11 | fusion_type == "Inter\nChromosomal" & n < 6000) # to scale the y axis with facets need to filter outliers
nygc_spinal_cord.starfusion.fusion_type.violin <- ggplot(nygc_spinal_cord.starfusion.fusion_types.bind.filt, aes(x = CONDITION, y = n, colour = CONDITION)) +
    # ggrastr::rasterise(geom_violin(adjust = 1), dpi = 300) +
    ggrastr::rasterise(geom_sina(size = 0.5), dpi = 300) +
    ggrastr::rasterise(geom_boxplot(aes(fill = CONDITION), colour = "black", width=0.2, alpha = 0.5, outlier.shape = NA), dpi = 300) + 
    scale_colour_manual(values = c("dodgerblue2", "firebrick2")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
    facet_wrap(~ fusion_type, scales = "free", nrow = 1)  + #coord_equal() +
labs(y = "Adjusted Number of Fusions", x = "", title = "Postmortem: Fusion Types") +
    stat_pvalue_manual(nygc_spinal_cord.starfusion.fusion_types.stat, label = "p.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) + 
  theme_oz(xaxis_arrow = FALSE) + theme(legend.position = "none", plot.title = element_text(hjust = 0.5))
# nygc_spinal_cord.starfusion.fusion_type.violin


```


## UpSet plot Fusions iPSMNs and postmortem

```{r}
nygc_spinal_cord.starfusion.unique = nygc_spinal_cord.starfusion %>% group_by(CONDITION) %>% distinct(fusion_name, .keep_all = TRUE) %>% mutate(group = paste("Postmortem", CONDITION))
ipsc_mn_als_datasets.starfusion.unique = ipsc_mn_als_datasets.starfusion %>% group_by(CONDITION) %>% distinct(fusion_name, .keep_all = TRUE) %>% mutate(group = paste("iPSMN", CONDITION))
ipsn_nygc.starfusion = bind_rows(mutate(select(nygc_spinal_cord.starfusion.unique, fusion_name, group), padj = 0.01), mutate(select(ipsc_mn_als_datasets.starfusion.unique, fusion_name, group), padj = 0.01))
ipsn_nygc.starfusion.upset = upset_plot(ipsn_nygc.starfusion, overlap_by = "group", overlap_level = "fusion_name", split_direction = FALSE, ymax = 220, order = "freq", order_group_list = c("Postmortem ALS", "Postmortem CTRL", "iPSMN ALS", "iPSMN CTRL"), label_colours = c("dodgerblue2","firebrick2"), title = "Intersection of Gene Fusions") + 
  theme_combmatrix(combmatrix.panel.striped_background.color.two = "dodgerblue2", combmatrix.panel.striped_background.color.one = "firebrick2")
ipsn_nygc.starfusion.upset


# # Venn overlap unqiue gene fusions
# ipsc_mn_als_datasets.starfusion %>% distinct(fusion_name) # 353
# ipsc_mn_als_datasets.starfusion %>% distinct(fusion_type) # 6
# ipsc_mn_als_datasets.starfusion %>% group_by(condition) %>% distinct(fusion_name) %>% count(condition)
# # condition     n
# #   <chr>     <int>
# # 1 als         292
# # 2 ctrl        152
# ipsc_mn_als_datasets.starfusion.als_events <- ipsc_mn_als_datasets.starfusion %>% filter(condition == "als") %>% distinct(fusion_name) %>% pull(fusion_name)
# ipsc_mn_als_datasets.starfusion.ctrl_events <- ipsc_mn_als_datasets.starfusion %>% filter(condition == "ctrl") %>% distinct(fusion_name) %>% pull(fusion_name)
# 
# # # ggvenn unique fusion events in CTRL & ALS
# ipsc_mn_als_datasets.starfusion.als_ctrl_events <- list("ALS" = ipsc_mn_als_datasets.starfusion.als_events, "CTRL" = ipsc_mn_als_datasets.starfusion.ctrl_events)
# ipsc_mn_als_datasets.starfusion.als_ctrl_events_venn <- ggvenn(ipsc_mn_als_datasets.starfusion.als_ctrl_events, auto_scale = TRUE, fill_color = c("firebrick2", "dodgerblue2"), set_name_color = c("firebrick2", "dodgerblue2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5) + labs(subtitle = "Unique Gene Fusions")  + 
#   theme(plot.subtitle = element_text(hjust = 0.5, size = 9), plot.margin = unit(c(0, -1, 0, -1), "lines")) #+ theme_oz()
# ipsc_mn_als_datasets.starfusion.als_ctrl_events_venn


# # Venn overlap unqiue gene fusions
# nygc_spinal_cord.starfusion %>% distinct(fusion_name) # 202
# nygc_spinal_cord.starfusion %>% distinct(fusion_type) # 5
# nygc_spinal_cord.starfusion %>% group_by(condition) %>% distinct(fusion_name) %>% count(condition)
# condition     n
#   <chr>     <int>
# 1 als         177
# 2 ctrl         96
# nygc_spinal_cord.starfusion.als_events <- nygc_spinal_cord.starfusion %>% filter(condition == "als") %>% distinct(fusion_name) %>% pull(fusion_name)
# nygc_spinal_cord.starfusion.ctrl_events <- nygc_spinal_cord.starfusion %>% filter(condition == "ctrl") %>% distinct(fusion_name) %>% pull(fusion_name)
# 
# # # ggvenn unique fusion events in CTRL & ALS
# nygc_spinal_cord.starfusion.als_ctrl_events <- list("ALS" = nygc_spinal_cord.starfusion.als_events, "CTRL" = nygc_spinal_cord.starfusion.ctrl_events)
# nygc_spinal_cord.starfusion.als_ctrl_events_venn <- ggvenn(nygc_spinal_cord.starfusion.als_ctrl_events, auto_scale = TRUE, fill_color = c("firebrick2", "dodgerblue2"), set_name_color = c("firebrick2", "dodgerblue2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5) + labs(subtitle = "Postmortem Gene Fusions")  + 
#   theme(plot.subtitle = element_text(hjust = 0.5, size = 9), plot.margin = unit(c(0, -1, 0, -1), "lines")) #+ theme_oz()
# nygc_spinal_cord.starfusion.als_ctrl_events_venn

```


## Merge

```{r}
variant_fusion.genetic_groups = plot_grid(
  ipsmn_p96_profile_pooled, #answerals.base_substitutions.violin
  nygc_p96_profile_pooled, #nygc_spinal_cord.base_substitutions.violin, 
  ipsn_nygc.starfusion.upset,
  ipsc_mn_als_datasets.starfusion.fusion_type.violin,
  nygc_spinal_cord.starfusion.fusion_type.violin,
  nrow = 5, rel_heights = c(1,1,0.8,1,1), labels = "auto")
# variant_fusion.genetic_groups
ggsave(variant_fusion.genetic_groups, filename = here(proj_path,"rnafusion/variant_fusion.genetic_groups.png"), height = 11.75, width = 8.25)

```


# Source Data

```{r}
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0")

# Fig 2A
ipsc_mn_als_datasets$res %>% select(gene_name, pvalue, log2FoldChange) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 2A")

# Fig 2B
ipsc_mn_als_datasets.gprofiles_filt_combined %>% select(query, term_name, fdr = p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 2B")

# Fig 2D
ipsc_mn_als_progeny.contrast_acts %>% select(pathway = source, score, p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 2D")

# Fig 2E
ipsc_mn_als_progeny.p53weights %>% select(target, stat, p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 2E")

# Fig 2F
ipsc_mn_als_dorothea.contrast_acts %>% select(TF = source, score, p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 2F")

# Fig 3A-E
ipsc_mn_tardbp_datasets$res %>% select(gene_name, pvalue, log2FoldChange) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 3A")
ipsc_mn_fus_datasets$res %>% select(gene_name, pvalue, log2FoldChange) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 3B")
ipsc_mn_c9orf72_datasets$res %>% select(gene_name, pvalue, log2FoldChange) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 3C")
ipsc_mn_sod1_datasets$res %>% select(gene_name, pvalue, log2FoldChange) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 3D")
ipsc_mn_sporadic_datasets$res %>% select(gene_name, pvalue, log2FoldChange) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 3E")

# Fig 3F
melted_cormat %>% select(subgroup1=Var1, subgroup2=Var2, pearson=value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 3F")

# Fig 3G
ipsc_mn_mutations_list.p53.TP53 %>% select(decoupleR, mutation, score, p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 3G")


# Fig 4A
nygc_postmortem_spinal_cord$res %>% select(gene_name, pvalue, log2FoldChange) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 4A")

# Fig 4B
nygc_postmortem_spinal_cord.gprofiles_filt_combined %>% select(query, term_name, fdr = p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 4B")

# Fig 4D
nygc_postmortem_spinal_cord_progeny.contrast_acts %>% select(pathway = source, score, p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 4D")

# Fig 4E
nygc_postmortem_spinal_cord_progeny.p53weights %>% select(target, stat, p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 4E")

# Fig 4F
nygc_postmortem_spinal_cord_dorothea.contrast_acts %>% select(TF = source, score, p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 4F")

# Fig 4G
postmortem_ipsn.als_vs_ctrl %>% select(gene_name, nygc.stat, ipsn.stat) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 4G")

# Fig 4H
melted_cormat  %>% select(subgroup1=Var1, subgroup2=Var2, pearson=value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 4H")

# Fig 4I
nygc_postmortem_spinal_cord_mn_mutations_list.p53.TP53 %>% select(decoupleR, mutation, score, p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 4I")


# Fig 5B
als_ctrl.voila %>% select(gene_name, lsv_junction_id, deltapsi, tnom) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 5B")

# Fig 5C
als_ctrl.voila.gost_filt.terms %>% select(query, term_name, fdr = p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 5C")

# Fig 5D
modulizer_simplfied.sig.summary %>% select(modulizer_simplified, n, pct) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 5D")

# Fig 5F
nygc.als_ctrl.voila %>% select(gene_name, lsv_junction_id, deltapsi, tnom) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 5F")

# Fig 5G
nygc.als_ctrl.voilagost_filt.terms %>% select(query, term_name, fdr = p_value) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 5G")

# Fig 5H
nygc.als_ctrl.voila.modulizer.count  %>% select(modulizer_simplified, n, pct) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 5H")


# Fig 6A
answerals.vcfannotate.totals.fitg_plot.bind %>% select(condition = CONDITION, mutation_type = type, mutation_n = n) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 6A")

# Fig 6C
nygc_spinal_cord.vcfannotate.totals.fitg_plot.bind.filt %>% select(condition = CONDITION, mutation_type = type, mutation_n = n) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 6C")

# Fig 6E
partialize(ipsc_mn_als_datasets.starfusion.n.fitg_plot, vars = "CONDITION") %>% select(condition = CONDITION, fusion_n = n) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 6E")

# Fig 6G
partialize(nygc_spinal_cord.starfusion.n.fitg_plot, vars = "CONDITION") %>% select(condition = CONDITION, fusion_n = n) %>% write_sheet(ss = "https://docs.google.com/spreadsheets/d/1EolqaTAcIoXe956ZRuTLvyFsBMtv9jqo_IpeLOesus0/edit#gid=0", sheet = "Fig 6G")

```





# *** ADDITIONAL UNUSED ***

## TDP-43 and p53

```{r}
neuronal_nuclei_tdp43_liu = readRDS(here(proj_path,"expression/deseq2/neuronal_nuclei_tdp43_liu.rds"))
## PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
neuronal_nuclei_tdp43_liu.res.matrix <- neuronal_nuclei_tdp43_liu$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
neuronal_nuclei_tdp43_liu.progeny.contrast_acts <- run_wmean(mat=neuronal_nuclei_tdp43_liu.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
neuronal_nuclei_tdp43_liu.progeny.pathwayActivity <- ggplot(neuronal_nuclei_tdp43_liu.progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Postmortem TDP43 status: Signaling Pathways", x = "", y = "TDP43 - vs TDP43 + nuclei Normalised enrichment") +
    stat_pvalue_manual(neuronal_nuclei_tdp43_liu.progeny.contrast_acts, label = "p.signif", y.position = 6, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
neuronal_nuclei_tdp43_liu.progeny.pathwayActivity

# Dorothea
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
neuronal_nuclei_tdp43_liu_dorothea.contrast_acts <- run_wmean(mat=neuronal_nuclei_tdp43_liu.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.1 & abs(score) < 4 ~ "sig", p_value < 0.1 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.1 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))
neuronal_nuclei_tdp43_liu_dorothea.contrast_acts.labels.up = neuronal_nuclei_tdp43_liu_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
neuronal_nuclei_tdp43_liu_dorothea.contrast_acts.labels.down = neuronal_nuclei_tdp43_liu_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)
# Volcano of TFs score vs p_value
neuronal_nuclei_tdp43_liu_dorothea.contrast_acts.volcano = ggplot(neuronal_nuclei_tdp43_liu_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment TDP43 - vs TDP43 +", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(neuronal_nuclei_tdp43_liu_dorothea.contrast_acts, source %in% c(neuronal_nuclei_tdp43_liu_dorothea.contrast_acts.labels.up, neuronal_nuclei_tdp43_liu_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,1.9))
neuronal_nuclei_tdp43_liu_dorothea.contrast_acts.volcano

```

## RNA Editing

https://github.com/jumphone/SPRINT
https://doi.org/10.1093/bioinformatics/btx473 
https://github.com/jumphone/SPRINT/blob/master/SPRINT_manual.pdf

3 outputs: hyper (realigned), regular, all (both hyper & regular)

```{r}
# count_all_sprint_edits = function(metadata){
#   print("reading in SPRINT files with read_tsv")
#   SPRINT_identified_all.res = metadata$sprint_all_path %>% map(read_tsv, col_types = list(`#Chrom`="c",`Start(0base)`="n",`End(1base)`="n",Type="c",Supporting_reads="n",Strand="c",`AD:DP`="c",.default="c"))
#   names(SPRINT_identified_all.res) = pull(metadata, dataset_sample) # needs to be unique names e.g. sample
#   sprint_all_tab <- map_dfr(SPRINT_identified_all.res, bind_rows, .id = "dataset_sample") #%>%
#     # rename(chr = `#Chrom`, start = `Start(0base)`, end = `End(1base)`) %>% clean_names() %>%
#     # mutate(ad = str_split_fixed(ad_dp,":",2)[,1], dp = str_split_fixed(ad_dp,":",2)[,2], edit_type = "all") %>% 
#     count(dataset_sample, name = "count")
#   return(sprint_all_tab)
# }
# 
# ipsc_mn_als_datasets.metadata %<>% mutate(sprint_all_path = paste0("/camp/project/proj-luscombn-patani/working/ipsc-mn-als-meta/editing/sprint/",dataset_sample,"/SPRINT_identified_all.res"))
# ipsc_mn_als_datasets.sprint = count_all_sprint_edits(ipsc_mn_als_datasets.metadata)
# ipsc_mn_als_datasets.sprint
# write_csv(ipsc_mn_als_datasets.sprint, here(proj_path,"editing/ipsc_mn_als_datasets.sprint.csv"))
ipsc_mn_als_datasets.sprint = read_csv(here(proj_path,"editing/ipsc_mn_als_datasets.sprint.csv"))
ipsc_mn_als_datasets.sprint %<>% left_join(ipsc_mn_als_datasets.metadata)
ipsc_mn_als_datasets.sprint %<>% mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS"))) 

# left_join multiqc stats
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
multiqc_rseqc_read_distribution.tsv
ipsc_mn_als_datasets.sprint %<>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.sprint

sprint.samples_no_rnaedits = ipsc_mn_als_datasets.metadata$dataset_sample[!ipsc_mn_als_datasets.metadata$dataset_sample %in% ipsc_mn_als_datasets.sprint$dataset_sample] # 0 i.e. all samples have at least 1 variant
sprint.samples_no_rnaedits # 0

# generalised linear model spline
# library(rms)
ipsc_mn_als_datasets.sprint.glm = glm(data=ipsc_mn_als_datasets.sprint, count ~ CONDITION + rms::rcs(total_reads, 3), family = poisson) # + rms::rcs(total_reads, 3)
summary(ipsc_mn_als_datasets.sprint.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           9.436e+00  7.218e-04 13072.0   <2e-16 ***
# CONDITIONALS                          1.249e-01  1.807e-04   690.9   <2e-16 ***
fit <- glm(count ~ CONDITION * rms::rcs(total_reads, 3), family = poisson, data=ipsc_mn_als_datasets.sprint)
predicted <- predict(fit, type="response")
ggplot(cbind(ipsc_mn_als_datasets.sprint, predicted=predicted), aes(x=total_reads, y=predicted, colour= CONDITION, group = interaction(CONDITION))) + geom_line() + theme_oz()

# Violin
ipsc_mn_als_datasets.sprint.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 2500000, xmin = 1, xmax = 2)
ipsc_mn_als_datasets.sprint.total.violin <- violin_plot(ipsc_mn_als_datasets.sprint, color_by = "CONDITION", continuous = "count", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,2500000), dpi = 72, ylabel = "Total Number of Edits per sample", title = "RNA editing") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.sprint.stats, tip.length = 0.01, size = 4)
ipsc_mn_als_datasets.sprint.total.violin


ipsc_mn_als_datasets.sprint %>% my_summarise(count, condition) %>% mutate(IQR = Q3_count - Q1_count)
# condition mean_count median_count sd_count Q1_count Q3_count min_count max_count sum_count n_count    IQR
#   <fct>          <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>  <dbl>
# 1 ctrl         358940.       195164  352680.   48662.  653990.      2309   1490633  38047599     106 605327
# 2 als          638377.       671394  380590.  277622.  880176.      5257   2386559 206195677     323 602553

```

### AnswerALS

```{r}
answerals.sprint = answerals.sprint %>% filter(dataset == "answerals")

answerals.sprint.glm = glm(data=answerals.sprint, count ~ CONDITION + rms::rcs(total_reads, 3), family = poisson) # + rms::rcs(total_reads, 3)
summary(answerals.sprint.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.280e+01  9.557e-04 13395.4   <2e-16 ***
# CONDITIONALS                          5.322e-02  1.937e-04   274.8   <2e-16 ***
# rms::rcs(total_reads, 3)total_reads   8.944e-09  1.215e-11   736.1   <2e-16 ***
# rms::rcs(total_reads, 3)total_reads' -2.756e-09  1.395e-11  -197.6   <2e-16 ***
fit <- glm(count ~ CONDITION * rms::rcs(total_reads, 3), family = poisson, data=answerals.sprint)
predicted <- predict(fit, type="response")
ggplot(cbind(answerals.sprint, predicted=predicted), aes(x=total_reads, y=predicted, colour= CONDITION, group = interaction(CONDITION))) + geom_line() + theme_oz()

# Violin
answerals.sprint.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 2500000, xmin = 1, xmax = 2)
answerals.sprint.total.violin <- violin_plot(answerals.sprint, color_by = "CONDITION", continuous = "count", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,2500000), dpi = 72, ylabel = "Total Number of Edits per sample", title = "RNA editing") + 
  stat_pvalue_manual(answerals.sprint.stats, tip.length = 0.01, size = 4)
answerals.sprint.total.violin


answerals.sprint %>% my_summarise(count, condition) %>% mutate(IQR = Q3_count - Q1_count)
# condition mean_count median_count sd_count Q1_count Q3_count min_count max_count sum_count n_count     IQR
#   <fct>          <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int>   <dbl>
# 1 ctrl         751198.      732262   215372.  622436.  866018.    405909   1490633  31550314      42 243583.
# 2 als          819397.      790096.  261894.  633270.  941508.    251034   2386559 195016456     238 308238 

```

# Fusion other tools

```{r}
# test Catanese dataset

# Arriba
c9orf72_1.arriba.fusions.tsv = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/arriba/c9orf72_1.arriba.fusions.tsv"))
c9orf72_1.arriba.fusions.tsv
c9orf72_1.arriba.fusions.tsv %>% glimpse

# FusionCatcher
c9orf72_1.fusioncatcher.fusion_genes.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/fusioncatcher/c9orf72_1.fusioncatcher.fusion-genes.txt"))
c9orf72_1.fusioncatcher.fusion_genes.txt
c9orf72_1.fusioncatcher.fusion_genes.txt %>% glimpse

# kallisto
c9orf72_1.kallisto_quant.fusions.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/kallisto/c9orf72_1.kallisto_quant.fusions.txt"))
c9orf72_1.kallisto_quant.fusions.txt
c9orf72_1.kallisto_quant.fusions.txt %>% glimpse

# pizzly
c9orf72_1.pizzly.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/pizzly/c9orf72_1.pizzly.txt"))
c9orf72_1.pizzly.txt
c9orf72_1.pizzly.txt %>% glimpse

# squid
c9orf72_1.squid.fusions.annotated.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/squid/c9orf72_1.squid.fusions.annotated.txt"))
c9orf72_1.squid.fusions.annotated.txt
c9orf72_1.squid.fusions.annotated.txt %>% glimpse

# squid
c9orf72_1.squid.fusions_sv.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/squid/c9orf72_1.squid.fusions_sv.txt"))
c9orf72_1.squid.fusions_sv.txt

# starfusion
c9orf72_1.starfusion.abridged.coding_effect.tsv = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/starfusion/c9orf72_1.starfusion.abridged.coding_effect.tsv"))
c9orf72_1.starfusion.abridged.coding_effect.tsv
c9orf72_1.starfusion.abridged.coding_effect.tsv %>% glimpse

read_starfusion = function(metadata){
  print("reading in STAR Fusion files with read_tsv")
  starfusion.tsv = metadata$starfusion_path %>% map(read_tsv, col_types = list(JunctionReadCount="n",SpanningFragCount="n",est_J="n",est_S="n",FFPM="n",LeftBreakEntropy="n",RightBreakEntropy="n",.default="c"))#, show_col_types = FALSE) # ! Can't combine `ExcludedBases` <double> and `ExcludedBases` <character>.
  names(starfusion.tsv) = metadata$sample #dataset_sample # needs to be unique names e.g. dataset_sample
  starfusion_tab <- map_dfr(starfusion.tsv, bind_rows, .id = "sample") %>% rename(fusion_name = '#FusionName') %>% clean_names() %>% #dataset_sample
    mutate(left_chr = str_split_fixed(left_breakpoint,":",3)[,1], left_coord = str_split_fixed(left_breakpoint,":",3)[,2], left_strand = str_split_fixed(left_breakpoint,":",3)[,3],
           right_chr = str_split_fixed(right_breakpoint,":",3)[,1], right_coord = str_split_fixed(right_breakpoint,":",3)[,2], right_strand = str_split_fixed(right_breakpoint,":",3)[,3],
           fusion_type = case_when(grepl("INTERCHROMOSOMAL", annots) ~ "inter_chromosomal", grepl("NEIGHBORS_OVERLAP", annots) ~ "overlapping_neighbors", grepl("NEIGHBORS", annots) ~ "neighbors", grepl("LOCAL_INVERSION", annots) ~ "local_inversion", grepl("LOCAL_REARRANGEMENT", annots) ~ "local_rearrangement", TRUE ~ "distant_proximity")) %>%
    left_join(metadata, by = "sample") #dataset_sample
  return(starfusion_tab)
}
catanese.metadata = catanese.metadata %>% mutate(starfusion_path = paste0(database_dir,"/rnafusion/starfusion/",sample,".starfusion.abridged.coding_effect.tsv"))
catanese.starfusion = read_starfusion(catanese.metadata)
catanese.starfusion


# The intra-chromosomal fusions were further classified into four subtypes: (1) local rearrangements, which were fusions where the genes are in an unexpected order given the strand of each gene in the pair; (2) not close proximity, which encompassed genes >100 kb apart; (3) neighbors, which were fusions that encompassed genes <100 kb apart and did not show evidence of gene orientation rearrangement; and (4) overlapping neighbors, which encompassed genes whose spans overlapped by at least one base pair.

# Violin unique fusion events in CTRL & ALS
catanese.starfusion %>% add_count(sample) %>% violin_plot(color_by = "condition", continuous = "n", stats_test = FALSE)

catanese.starfusion.n = catanese.starfusion %>% add_count(sample)
catanese.starfusion.violin <- violin_plot(catanese.starfusion.n, color_by = "condition", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, stats_test = "t_test", arrow_labels = FALSE, ylims = c(0,20), dpi = 300, ylabel = "Number of Unique Fusion Events")
catanese.starfusion.violin
```


# Fig S15 Fusions genetic subgroups

```{r}
# left_join multiqc stats
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
multiqc_rseqc_read_distribution.tsv
```



## Fusion Types

Neighbours

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Neighbours") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Neighbours = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Neighbours = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Neighbours) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Neighbours)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.stat$y.position = c(22,21,21,21,21)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.stat
# .y.   group1 group2      n1    n2 statistic     df        p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl>  <dbl>    <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    -3.64  209.   0.000349 0.005 **                 22.4 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -1.23   23.6  0.232    1     ns                 24.0 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9    -0.654   9.52 0.529    1     ns                 25.5 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10     1.53   10.2  0.156    1     ns                 27.1 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53    -1.97   68.6  0.053    0.641 ns                 28.6 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,22), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Neighbours") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL            4.2       4    2.43  3     5        0    12   378    90  2   
# 2 Sporadic        5.40      5    3.03  3     7        0    21  1124   208  4   
# 3 SOD1            5         5    2.54  3     5.75     2    13    90    18  2.75
# 4 FUS             4.78      5    2.54  3     6        2     9    43     9  3   
# 5 TARDBP          2.6       1.5  3.20  0.25  3        0    10    26    10  2.75
# 6 C9orf72         5.57      3    4.70  3     8        1    20   295    53  5   

```

Local Rearrangement

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Local\nRearrangement") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Local_Rearrangement = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Local_Rearrangement = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Local_Rearrangement) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Local_Rearrangement)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.stat$y.position = c(9,21,21,21,21)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.stat
# .y.   group1 group2      n1    n2 statistic    df            p      p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl>        <dbl>      <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    -5.67  153.  0.0000000678 0.00000102 ****               10.8 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -1.33   24.4 0.197        0.985      ns                 11.7 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9     2.48   13.5 0.027        0.245      ns                 12.6 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10     1.27   10.8 0.23         0.985      ns                 13.5 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53    -0.794  92.2 0.429        1          ns                 14.4 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,10), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Local Rearrangement") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL           1.4        1    1.77     0  2.75     0     7   126    90  2.75
# 2 Sporadic       2.62       2    1.58     2  4        0     8   546   208  2   
# 3 SOD1           2          1.5  1.75     1  3        0     6    36    18  2   
# 4 FUS            0.444      0    1.01     0  0        0     3     4     9  0   
# 5 TARDBP         0.6        0    1.90     0  0        0     6     6    10  0   
# 6 C9orf72        1.68       1    2.17     0  2        0    10    89    53  2   

```

Distant Proximity

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Distant\nProximity") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Distant_Proximity = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Distant_Proximity = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Distant_Proximity) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Distant_Proximity)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.stat$y.position = c(22,21,8,21,21)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.stat
#  .y.   group1 group2      n1    n2 statistic    df     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    -2.21  154.  0.028 0.283 ns                   22 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -1.31   22.3 0.203 1     ns                   21 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9     3.61   23.7 0.001 0.02  *                    21 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10     0.134  10.8 0.896 1     ns                   21 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53     0.526 108.  0.6   1     ns                   21 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,8), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Distant Proximity") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL           1.47         1  1.38  0        2     0     6   132    90  2   
# 2 Sporadic       1.84         2  1.24  1        3     0     7   383   208  2   
# 3 SOD1           2            2  1.61  1        2     0     7    36    18  1   
# 4 FUS            0.667        1  0.5   0        1     0     1     6     9  1   
# 5 TARDBP         1.4          1  1.51  0.25     2     0     5    14    10  1.75
# 6 C9orf72        1.34         1  1.40  0        2     0     5    71    53  2   

```


Overlapping Neighbours

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Overlapping\nNeighbours") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Overlapping_Neighbours = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Overlapping_Neighbours = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Overlapping_Neighbours) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Overlapping_Neighbours)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.stat$y.position = c(3.2,21,3.4,21,21)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.stat
# .y.   group1 group2      n1    n2 statistic    df        p    p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    -5.34  173.  2.95e- 7 3.54e- 6 ****                 22 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -2.05   22.9 5.2 e- 2 4.06e- 1 ns                   21 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9     7.72   89   1.62e-11 2.27e-10 ****                 21 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10     0.753  11.4 4.66e- 1 1   e+ 0 ns                   21 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53    -0.945 105.  3.47e- 1 1   e+ 0 ns                   21 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,3.5), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Overlapping Neighbours") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) + 
  geom_jitter(size = 0.5, width = 0.2, height = 0.2) # geom_sina doesnt work if all values the same so specify geom_jitter https://githubmemory.com/repo/thomasp85/ggforce/issues/212
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL           0.422        0 0.519     0  1        0     2    38    90  1   
# 2 Sporadic       0.774        1 0.531     0  1        0     3   161   208  1   
# 3 SOD1           0.722        1 0.575     0  1        0     2    13    18  1   
# 4 FUS            0            0 0         0  0        0     0     0     9  0   
# 5 TARDBP         0.3          0 0.483     0  0.75     0     1     3    10  0.75
# 6 C9orf72        0.509        0 0.541     0  1        0     2    27    53  1   

```

Inter Chromosomal

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Inter\nChromosomal") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Inter_Chromosomal = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Inter_Chromosomal = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Inter_Chromosomal) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Inter_Chromosomal)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.stat
#  .y.   group1 group2      n1    n2 statistic    df     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208     1.19  108.  0.237     1 ns                 24.4 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -1.80   17.2 0.089     1 ns                 27.0 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9    -0.565  11.4 0.583     1 ns                 29.5 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10    -0.411  15.1 0.687     1 ns                 32.1 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53     0.172 141.  0.864     1 ns                 34.7 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,4), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Inter Chromosomal") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL           0.578      0   1.36      0   1       0     8    52    90   1  
# 2 Sporadic       0.399      0   0.659     0   1       0     3    83   208   1  
# 3 SOD1           3.61       0   7.12      0   2.5     0    22    65    18   2.5
# 4 FUS            0.778      1   0.972     0   1       0     3     7     9   1  
# 5 TARDBP         0.7        0.5 0.823     0   1       0     2     7    10   1  
# 6 C9orf72        0.547      0   0.774     0   1       0     3    29    53   1  
```

Local Inversion

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Local\nInversion") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Local_Inversion = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Local_Inversion = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Local_Inversion) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Local_Inversion)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.stat
#  .y.   group1 group2      n1    n2 statistic    df     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    2.50   95.6  0.014 0.169 ns                 2.24 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18   -0.0953 20.6  0.925 1     ns                 2.50 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9   -0.808   8.96 0.44  1     ns                 2.75 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10    2.82   89    0.006 0.084 ns                 3.01 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53    2.82   89    0.006 0.084 ns                 3.27 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,2), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Local Inversion") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) +
    geom_jitter(size = 0.5, width = 0.2, height = 0.2) # geom_sina doesnt work if all values the same so specify geom_jitter https://githubmemory.com/repo/thomasp85/ggforce/issues/212
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
#  genetic_group  mean_n median_n   sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>           <dbl>    <dbl>  <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL          0.1            0 0.337      0     0     0     2     9    90     0
# 2 Sporadic      0.00962        0 0.0978     0     0     0     1     2   208     0
# 3 SOD1          0.111          0 0.471      0     0     0     2     2    18     0
# 4 FUS           0.222          0 0.441      0     0     0     1     2     9     0
# 5 TARDBP        0              0 0          0     0     0     0     0    10     0
# 6 C9orf72       0              0 0          0     0     0     0     0    53     0


# # histogram
# ipsc_mn_als_datasets.starfusion.fusion_type.histogram = ggplot(ipsc_mn_als_datasets.starfusion.fusion_type_n, aes(x=n, color=CONDITION, fill=CONDITION)) + 
#   geom_histogram(alpha=0.5, position="identity", binwidth = 1)+
#   scale_color_manual(values=c("dodgerblue2", "firebrick2"))+ scale_fill_manual(values=c("dodgerblue2", "firebrick2"))+
#   facet_wrap(~ fusion_type, scales = "free", nrow = 1) + theme_oz() #+ ggtitle("Zeros filled")
# ipsc_mn_als_datasets.starfusion.fusion_type.histogram

```


## Merge

```{r}
ipsc_mn_als_datasets.fusion.genetic_groups =  plot_grid(ipsc_mn_als_datasets.starfusion.mutation.violin,
                                                        plot_grid(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.violin, 
                                                                  ncol = 2, nrow = 3, labels = c("b","c","d","e","f","g")),
                                                        nrow = 2, rel_heights = c(1,3), labels = c("a",""))
# ipsc_mn_als_datasets.fusion.genetic_groups
ggsave(ipsc_mn_als_datasets.fusion.genetic_groups, filename = here(proj_path,"rnafusion/ipsc_mn_als_datasets.fusion.genetic_groups.png"), height = 11.75, width = 8.25)

                                                        
```



# Answer ALS alone

## mutations

```{r}
# ipsc_mn_als_datasets.vcfannotate = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.dbsnp.redi.filtered.csv")) %>% 
#   mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
#   group_by(dataset_sample)%>% add_tally(n, name = "total") %>% ungroup %>% select(dataset_sample, n, type, variant_type, total, everything()) # add a column of sum of mutations per sample
# ipsc_mn_als_datasets.vcfannotate

ipsc_mn_als_datasets.vcfannotate.answerals = ipsc_mn_als_datasets.vcfannotate %>% filter(dataset == "answerals")

# Violin total variant events in CTRL & ALS
ipsc_mn_als_datasets.vcfannotate.answerals.totals = ipsc_mn_als_datasets.vcfannotate.answerals %>% distinct(dataset_sample, .keep_all = TRUE) #keep 1 row per sample for total number of variants
ipsc_mn_als_datasets.vcfannotate.answerals.totals

ipsc_mn_als_datasets.vcfannotate.answerals.totals.glm = glm(data = ipsc_mn_als_datasets.vcfannotate.answerals.totals, total ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(ipsc_mn_als_datasets.vcfannotate.answerals.totals.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.067e+01  3.343e-03 3191.93   <2e-16 ***
# CONDITIONALS                          1.072e-02  6.892e-04   15.55   <2e-16 ***


ipsc_mn_als_datasets.vcfannotate.answerals.total.violin <- violin_plot(ipsc_mn_als_datasets.vcfannotate.answerals.totals, color_by = "CONDITION", continuous = "total", cols = c("dodgerblue2", "firebrick2"), 
                                                             one_sample = FALSE, stats_test = "wilcox_test", arrow_labels = FALSE, ylims = c(50000,80000), dpi = 300, ylabel = "Total Number of Mutations", title = "AnswerALS: All Variant types") 
ipsc_mn_als_datasets.vcfannotate.answerals.total.violin
#  .y.   group1 group2    n1    n2 statistic      p p.signif
#   <chr> <chr>  <chr>  <int> <int>     <dbl>  <dbl> <chr>   
# 1 lfc   CTRL   ALS       42   238     3762. 0.0106 *       
# # A tibble: 1  7
#   .y.   group1 group2 effsize    n1    n2 magnitude
# * <chr> <chr>  <chr>    <dbl> <int> <int> <ord>    
# 1 lfc   CTRL   ALS     -0.394    42   238 small    

ipsc_mn_als_datasets.vcfannotate.answerals.totals %>% my_summarise(total, condition) %>% mutate(IQR = Q3_total - Q1_total)
# condition mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <chr>          <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1 als           61291.       60689     3546.   59168.   62950.     48715     73472  14587359     238 3782.
# 2 ctrl          59890.       59646.    3601.   57313    61359.     54971     72087   2515399      42 4046.



ipsc_mn_als_datasets.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.vcfannotate

answerals.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% filter(dataset == "answerals")

### ALS vs CTRL variant number
answerals.vcfannotate.samples_no_variants = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% answerals.vcfannotate$dataset_sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

# Violin total variant events in CTRL & ALS
answerals.vcfannotate.totals = answerals.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) #keep 1 row per sample for total number of variants
answerals.vcfannotate.totals %>% glimpse

# generalised linear model spline
# library(rms)
answerals.vcfannotate.totals.glm = glm(total ~ CONDITION + rms::rcs(total_reads, 3), data=answerals.vcfannotate.totals, family=poisson) # + rms::rcs(total_reads, 3)
summary(answerals.vcfannotate.totals.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.067e+01  3.343e-03 3192.16   <2e-16 ***
# CONDITIONALS                          1.072e-02  6.892e-04   15.55   <2e-16 ***


```


## Fusions


```{r}
# ipsc_mn_als_datasets.starfusion = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
#   mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))
# ipsc_mn_als_datasets.starfusion

ipsc_mn_als_datasets.starfusion.answerals = ipsc_mn_als_datasets.starfusion %>% filter(dataset == "answerals")

# Venn overlap unqiue gene fusions
ipsc_mn_als_datasets.starfusion.answerals %>% distinct(fusion_name) # 173
ipsc_mn_als_datasets.starfusion.answerals %>% distinct(fusion_type) # 6
ipsc_mn_als_datasets.starfusion.answerals %>% group_by(condition) %>% distinct(fusion_name) %>% count(condition)
# condition     n
#   <chr>     <int>
# 1 als         166
# 2 ctrl         71
ipsc_mn_als_datasets.starfusion.answerals.als_events <- ipsc_mn_als_datasets.starfusion.answerals %>% filter(condition == "als") %>% distinct(fusion_name) %>% pull(fusion_name)
ipsc_mn_als_datasets.starfusion.answerals.ctrl_events <- ipsc_mn_als_datasets.starfusion.answerals %>% filter(condition == "ctrl") %>% distinct(fusion_name) %>% pull(fusion_name)

# # ggvenn unique fusion events in CTRL & ALS
ipsc_mn_als_datasets.starfusion.answerals.als_ctrl_events <- list("ALS" = ipsc_mn_als_datasets.starfusion.answerals.als_events, "CTRL" = ipsc_mn_als_datasets.starfusion.answerals.ctrl_events)
ipsc_mn_als_datasets.starfusion.answerals.als_ctrl_events_venn <- ggvenn(ipsc_mn_als_datasets.starfusion.answerals.als_ctrl_events, auto_scale = TRUE, fill_color = c("firebrick2", "dodgerblue2"), set_name_color = c("firebrick2", "dodgerblue2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5) + labs(subtitle = "Unique Gene Fusions")  + 
  theme(plot.subtitle = element_text(hjust = 0.5, size = 8), plot.margin = unit(c(0, -1, 0, -1), "lines")) #+ theme_oz()
ipsc_mn_als_datasets.starfusion.answerals.als_ctrl_events_venn

### ALS vs CTRL unique fusion events
ipsc_mn_als_datasets.starfusion.answerals.n = ipsc_mn_als_datasets.starfusion.answerals %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, total_reads) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.answerals.samples_no_fusions = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% ipsc_mn_als_datasets.starfusion.answerals.n$dataset_sample] # 0 
# answerals.metadata.no_fusions = answerals.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.answerals.samples_no_fusions) %>% 
  # mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), n = 0) %>% select(dataset_sample, n, CONDITION)
# ipsc_mn_als_datasets.starfusion.answerals.n = bind_rows(ipsc_mn_als_datasets.starfusion.answerals.n, answerals.metadata.no_fusions)

glm_starfusion = glm(data = ipsc_mn_als_datasets.starfusion.answerals.n, n ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           2.746e+00  2.282e-01  12.034   <2e-16 ***
# CONDITIONALS                          9.116e-02  5.096e-02   1.789   0.0736 .

ipsc_mn_als_datasets.starfusion.answerals.n %>% my_summarise(n, CONDITION) %>% mutate(IQR = Q3_n - Q1_n)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <int> <int> <int> <int> <dbl>
# 1 CTRL        10.9     10    3.83     8    13     6    19   457    42     5
# 2 ALS         11.5     10.5  5.61     8    14     2    39  2747   238     6

# Violin
ipsc_mn_als_datasets.starfusion.answerals.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.answerals.n, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, stats_test = "t_test", arrow_labels = FALSE, ylims = c(0,43), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Answer ALS: All Fusion types")
ipsc_mn_als_datasets.starfusion.answerals.violin
#  .y.   group1 group2    n1    n2 statistic    df     p p.signif
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <chr>   
# 1 lfc   CTRL   ALS       42   238    -0.952  75.9 0.344 ns      
# # A tibble: 1  7
#   .y.   group1 group2 effsize    n1    n2 magnitude 
# * <chr> <chr>  <chr>    <dbl> <int> <int> <ord>     
# 1 lfc   CTRL   ALS     -0.123    42   238 negligible

```


# scatter number of variants aginst read coverage

```{r}
# left_join multiqc
# multiqc stats
multiqc_general_stats_stat_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_general_stats_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), 
     multiqc_general_stats_stat_paths = gsub("/camp/lab/patanir/home/users/ziffo|/camp/project","/Volumes/lab-patanir/home/users/ziffo",multiqc_general_stats_stat_paths)) %>% pull(multiqc_general_stats_stat_paths)
multiqc_general_stats.tsv = multiqc_general_stats_stat_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_general_stats.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_general_stats <- map_dfr(multiqc_general_stats.tsv, bind_rows, .id = "database_dir") %>% clean_names()
multiqc_rseqc_read_distribution_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_rseqc_read_distribution_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"),
     multiqc_rseqc_read_distribution_paths = gsub("/camp/lab/patanir/home/users/ziffo|/camp/project","/Volumes/lab-patanir/home/users/ziffo",multiqc_rseqc_read_distribution_paths)) %>% pull(multiqc_rseqc_read_distribution_paths)
multiqc_rseqc_read_distribution.tsv = multiqc_rseqc_read_distribution_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_rseqc_read_distribution.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_rseqc_read_distribution <- map_dfr(multiqc_rseqc_read_distribution.tsv, bind_rows, .id = "database_dir") %>% clean_names()
ipsc_mn_als_datasets_with_lee.pcaData = ipsc_mn_als_datasets_with_lee.pcaData %>% left_join(multiqc_general_stats) %>% left_join(multiqc_rseqc_read_distribution) # need unique sample names - If multiple dataset names within a single multiqc_general_stats.txt then will only join the first dataset name

# variants
ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads = select(distinct(ipsc_mn_als_datasets.vcfannotate, dataset_sample, .keep_all = TRUE), database_dir, sample, total, dataset, condition, library_type) %>% left_join(select(multiqc_rseqc_read_distribution, database_dir, sample, total_reads))
ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads

ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads.scatter <- ggscatter(ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads, x = "total_reads", y = "total", color = "condition", cor.coef = TRUE, cor.method = "pearson", size = 0.5, cor.coeff.args = list(label.x = 90000000, label.y = 20000, label.sep="\n", size = 2.8),
          xlab = expression(Total~reads~per~sample), ylab = expression(Total~number~of~mutations~per~sample)) +
  theme_oz() + theme(legend.position = "right") +
  geom_hline(yintercept = 0, linetype = 2) + geom_vline(xintercept = 0, linetype = 2) +
  coord_cartesian(ylim=c(0,80000))
ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads.scatter

# fusions

ipsc_mn_als_datasets.starfusion
ipsc_mn_als_datasets.starfusion.n = ipsc_mn_als_datasets.starfusion %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, dataset, total_reads) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.samples_no_fusions = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% ipsc_mn_als_datasets.starfusion.n$dataset_sample] # "smith_tardbp_1" "smith_ctrl_1"  
ipsc_mn_als_datasets.paired_end.metadata.no_fusions = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.samples_no_fusions) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), n = 0) %>% select(dataset_sample, n, CONDITION, dataset)
ipsc_mn_als_datasets.starfusion.n = bind_rows(ipsc_mn_als_datasets.starfusion.n, ipsc_mn_als_datasets.paired_end.metadata.no_fusions)


ipsc_mn_als_datasets.starfusion.scatter <- ggscatter(ipsc_mn_als_datasets.starfusion.n, x = "total_reads", y = "n", color = "dataset", cor.coef = TRUE, cor.method = "pearson", size = 0.5, cor.coeff.args = list(label.x = 90000000, label.y = 50, label.sep="\n", size = 2.8),
          xlab = expression(Total~reads~per~sample), ylab = expression(Total~number~of~fusions~per~sample)) +
  theme_oz() + theme(legend.position = "right") +
  geom_hline(yintercept = 0, linetype = 2) + geom_vline(xintercept = 0, linetype = 2) #+  coord_cartesian(ylim=c(0,80000))
ipsc_mn_als_datasets.starfusion.scatter


```


# NeuroLINCS iPSC vs iMNs d32 vs diMN d18

Metadata of iPSC & iMN batches from same donors

```{r}
neurolincs.ipsc_iMN_diMN.metadata = read_csv(here(shared_path,"neurolincs/sample-details/samplesheet.csv")) %>% distinct(sample, .keep_all = TRUE) %>%
  mutate(database_dir = here(shared_path,"neurolincs"), condition = factor(condition, levels = c("ctrl", "als")),
         dataset = case_when(dataset == "neurolincs0" ~ "neurolincs.diMN",dataset == "neurolincsA" ~ "neurolincs.iMN", dataset == "neurolincsI" ~ "neurolincs.iPSC"), dataset_sample = paste0(dataset,"_",sample), study_accession = dataset)
neurolincs.ipsc_iMN_diMN.metadata %>% glimpse
neurolincs.ipsc_iMN_diMN.metadata %>% group_by(gap_subject_id)
neurolincs.ipsc_iMN_diMN.metadata %>% group_by(submitted_subject_id) # submitted_subject_id is equivelent to gap_subject_id
neurolincs.ipsc_iMN_diMN.metadata %>% group_by(DIV) %>% count(submitted_subject_id) %>% print(n=Inf)
neurolincs.ipsc_iMN_diMN.metadata %>% select(submitted_subject_id, gap_subject_id) %>% distinct()

```

ALS vs CTRL in iPSC

```{r}
# neurolincs.ipsc = DESeq.analysis(metadata = neurolincs.ipsc.metadata, unique_names = "sample", design = ~ condition, contrast = "condition_als_vs_ctrl")
# saveRDS(neurolincs.ipsc, here(proj_path,"expression/deseq2/neurolincs.ipsc.rds"))
neurolincs.ipsc = readRDS(here(proj_path,"expression/deseq2/neurolincs.ipsc.rds"))
net_progeny <- get_progeny(organism = 'human', top = 100)
neurolincs.ipsc.res.matrix <- neurolincs.ipsc$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
neurolincs.ipsc_progeny.contrast_acts <- run_wmean(mat=neurolincs.ipsc.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
neurolincs.ipsc_progeny.pathwayActivity <- ggplot(neurolincs.ipsc_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
  geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
  theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "NeuroLINCS iPSC Signaling Pathways", x = "", y = "ALS vs CTRL Normalised enrichment") +
  stat_pvalue_manual(neurolincs.ipsc_progeny.contrast_acts, label = "p.signif", y.position = 14, xmin = "source", xmax = NULL, size = 5, hide.ns = TRUE) 
neurolincs.ipsc_progeny.pathwayActivity

```

## rnavar

vcf-annotate

```{r}
# vcf.annotate_script = neurolincs.ipsc_iMN_diMN.metadata %>% filter(DIV == 0) %>%
#   mutate(`#!/bin/bash` = paste0("zcat /camp/project/proj-luscombn-patani/working/neurolincs/rnavar/variant_calling/", sample,"/", sample, ".haplotypecaller.filtered.vcf.gz | vcf-annotate --fill-type | grep -oP 'TYPE=\\w+' | sort | uniq -c | sed 's/ TYPE=/,/' > ", "/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate/",dataset_sample, ".vcfstats.csv")) %>% select(`#!/bin/bash`)
# vcf.annotate_script
# vcf.annotate_script %>% pull(`#!/bin/bash`)    # check script: fastq order, job name
# write_tsv(vcf.annotate_script, "/camp/project/proj-luscombn-patani/working/neurolincs/sample-details/neurolincs.ipsc.vcf.annotate_script.sh")

# # run script
# ml VCFtools
# cd /camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate
# bash /camp/project/proj-luscombn-patani/working/neurolincs/sample-details/neurolincs.ipsc.vcf.annotate_script.sh

# read_vcfannotate = function(metadata){
#   print("reading in vcf-annotate files with read_csv")
#   vcf.annotate.csv = metadata$rnavar_path %>% map(read_csv, col_names = c("n","type"), col_types = list(n="n",type="c"))
#   names(vcf.annotate.csv) = metadata$dataset_sample # needs to be unique names e.g. dataset_sample
#   vcf.annotate_tab <- map_dfr(vcf.annotate.csv, bind_rows, .id = "dataset_sample") %>%
#     mutate(variant_type = case_when(grepl("del", type) ~ "Deletion", grepl("ins", type) ~ "Insertion", grepl("snp", type) ~ "SNP", grepl("complex", type) ~ "Complex", TRUE ~ "Unknown")) %>%
#     left_join(metadata, by = "dataset_sample")
#   return(vcf.annotate_tab)
# }
# neurolincs.ipsc.metadata = neurolincs.ipsc_iMN_diMN.metadata %>% filter(DIV == 0)
# neurolincs.ipsc.metadata = neurolincs.ipsc.metadata %>% mutate(rnavar_path = paste0("/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate/",dataset_sample,".vcfstats.csv"))
# neurolincs.ipsc.metadata.vcfannotate = read_vcfannotate(neurolincs.ipsc.metadata)
# neurolincs.ipsc.metadata.vcfannotate
# write_csv(neurolincs.ipsc.metadata.vcfannotate, here(proj_path,"rnavar/neurolincs.ipsc.metadata.vcfannotate.csv"))

neurolincs.ipsc.metadata.vcfannotate.csv = read_csv(here(proj_path,"rnavar/neurolincs.ipsc.metadata.vcfannotate.csv"))
neurolincs.imn_dimn.metadata.vcfannotate.csv = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.dbsnp.redi.filtered.csv")) %>% filter(dataset %in% c("neurolincs.diMN","neurolincs.iMN"))

neurolincs.ipsc_imn_dimn.metadata.vcfannotate = bind_rows(neurolincs.ipsc.metadata.vcfannotate.csv, neurolincs.imn_dimn.metadata.vcfannotate.csv) %>%
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(dataset_sample)%>% add_tally(n, name = "total") %>% ungroup %>% select(dataset_sample, n, type, variant_type, total, everything()) # add a column of sum of variants per sample
neurolincs.ipsc_imn_dimn.metadata.vcfannotate
```

Total Fusions

```{r}
# Distinct row per samples
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals = neurolincs.ipsc_imn_dimn.metadata.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, total, DIV, condition, submitted_subject_id) # keep 1 row per sample

# Fill in samples with 0 counts
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.samples_no_variants = neurolincs.ipsc_iMN_diMN.metadata$dataset_sample[!neurolincs.ipsc_iMN_diMN.metadata$dataset_sample %in% neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals$dataset_sample] # 0 i.e. all samples have at least 1 variant
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.samples_no_variants # 0

# Violin total variant events in CTRL & ALS at iPSC, iMN and dIMN
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat <- neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals %>% wilcox_test(total ~ DIV) %>% add_significance()  %>% add_xy_position() #%>% filter(group1 == "0")
# neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat$y.position = c(75000,80000,80000,80000,80000)
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat
# .y.   group1 group2    n1    n2 statistic     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 total 0      18        20    30      276  0.645     1 ns               67487. <chr [2]>        1     2
# 2 total 0      32        20    14      125  0.616     1 ns               69069. <chr [2]>        1     3
# 3 total 18     32        30    14      190. 0.614     1 ns               70650. <chr [2]>        2     3
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.violin <- violin_plot(mutate(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals, DIV = as.factor(DIV)), color_by = "DIV", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"),  plot_stats = FALSE, arrow_labels = FALSE, ylims = c(40000,70000), dpi = 300, ylabel = "Number of mutations", title = "NeuroLINCS iPSC, diMN, iMN: Total mutations", xlab = "DIV") + 
  stat_pvalue_manual(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat, label = "p.adj.signif", tip.length = 0.001, size = 4, hide.ns = TRUE)
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.violin

ggline(mutate(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals,submitted_subject_id=as.factor(submitted_subject_id)), "DIV", "total", lintype = "submitted_subject_id", color = "submitted_subject_id")

neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals %>% filter(condition == "als") %>% my_summarise(total, DIV) %>% mutate(IQR = Q3_total - Q1_total)
# DIV mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <dbl>      <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1     0     54008        53930.    1646.   52845    55419.     51093     56690    648096      12 2574.
# 2    18     53870.       54638     3194.   52325.   55914.     42761     58352   1185133      22 3589.
# 3    32     54014.       54492     2304.   53279.   54866.     49614     57671    432111       8 1586.

# restrict to ALS samples only
# Distinct row per samples
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.totals = neurolincs.ipsc_imn_dimn.metadata.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) %>% filter(condition == "als") %>% select(dataset_sample, total, DIV) # keep 1 row per sample

# Violin total variant events in ALS at iPSC, iMN and dIMN
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.stat <- neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.totals %>% wilcox_test(total ~ DIV) %>% add_significance()  %>% add_xy_position() #%>% filter(group1 == "0")
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.stat
# .y.   group1 group2    n1    n2 statistic     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 total 0      18        12    22       126 0.845     1 ns               58551. <chr [2]>        1     2
# 2 total 0      32        12     8        46 0.91      1 ns               58851. <chr [2]>        1     3
# 3 total 18     32        22     8        93 0.836     1 ns               59150. <chr [2]>        2     3
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.violin <- violin_plot(mutate(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.totals, DIV = as.factor(DIV)), color_by = "DIV", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"),  plot_stats = FALSE, arrow_labels = FALSE, ylims = c(40000,60000), dpi = 300, ylabel = "Number of mutations", title = "NeuroLINCS ALS only iPSC, diMN, iMN: Total mutations", xlab = "DIV")
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.violin

neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.totals %>% my_summarise(total, DIV) %>% mutate(IQR = Q3_total - Q1_total)
# DIV mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <dbl>      <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1     0     54008        53930.    1646.   52845    55419.     51093     56690    648096      12 2574.
# 2    18     53870.       54638     3194.   52325.   55914.     42761     58352   1185133      22 3589.
# 3    32     54014.       54492     2304.   53279.   54866.     49614     57671    432111       8 1586.

# merge iMN & diMN
# Distinct row per samples
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals = neurolincs.ipsc_imn_dimn.metadata.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) %>% mutate(celltype = case_when(DIV == 0 ~ "iPSC", TRUE ~ "iPSN")) %>% select(dataset_sample, total, celltype, condition) # keep 1 row per sample

# Violin total variant events in CTRL & ALS at iPSC, iMN and dIMN
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat <- neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals %>% wilcox_test(total ~ celltype) %>% add_significance()  %>% add_xy_position() #%>% filter(group1 == "0")
# neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat$y.position = c(75000,80000,80000,80000,80000)
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat
# .y.   group1 group2    n1    n2 statistic     p p.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 total iPSC   iPSN      20    44       401 0.577 ns           67487. <chr [2]>        1     2
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.violin <- violin_plot(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals, color_by = "celltype", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"),  plot_stats = FALSE, arrow_labels = FALSE, ylims = c(40000,70000), dpi = 300, ylabel = "Number of mutations", title = "NeuroLINCS iPSC vs iPSN: Total mutations", xlab = "celltype") 
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.violin

neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals %>% filter(condition == "als") %>% my_summarise(total, celltype) %>% mutate(IQR = Q3_total - Q1_total)
# celltype mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <chr>         <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1 iPSC         54008        53930.    1646.   52845    55419.     51093     56690    648096      12 2574.
# 2 iPSN         53908.       54498     2945.   52394.   55224.     42761     58352   1617244      30 2830 
```


## rnafusion

```{r}
# read_starfusion = function(metadata){
#   print("reading in STAR Fusion files with read_tsv")
#   starfusion.tsv = metadata$starfusion_path %>% map(read_tsv, col_types = list(JunctionReadCount="n",SpanningFragCount="n",est_J="n",est_S="n",FFPM="n",LeftBreakEntropy="n",RightBreakEntropy="n",.default="c"))
#   names(starfusion.tsv) = metadata$dataset_sample # needs to be unique names e.g. dataset_sample
#   starfusion_tab <- map_dfr(starfusion.tsv, bind_rows, .id = "dataset_sample") %>% rename(fusion_name = '#FusionName') %>% clean_names() %>%
#     mutate(left_chr = str_split_fixed(left_breakpoint,":",3)[,1], left_coord = str_split_fixed(left_breakpoint,":",3)[,2], left_strand = str_split_fixed(left_breakpoint,":",3)[,3],
#            right_chr = str_split_fixed(right_breakpoint,":",3)[,1], right_coord = str_split_fixed(right_breakpoint,":",3)[,2], right_strand = str_split_fixed(right_breakpoint,":",3)[,3],
#            fusion_type = case_when(grepl("INTERCHROMOSOMAL", annots) ~ "Inter\nChromosomal", grepl("OVERLAP", annots) ~ "Overlapping\nNeighbours", grepl("NEIGHBORS", annots) ~ "Neighbours", grepl("LOCAL_INVERSION", annots) ~ "Local\nInversion", grepl("LOCAL_REARRANGEMENT", annots) ~ "Local\nRearrangement", TRUE ~ "Distant\nProximity"),
#            breakpoints = paste(left_breakpoint, right_breakpoint, sep="_")) %>%
#     left_join(metadata, by = "dataset_sample")
#   return(starfusion_tab)
# }
# neurolincs.ipsc.metadata = neurolincs.ipsc_iMN_diMN.metadata %>% filter(DIV == 0)
# neurolincs.ipsc.metadata = neurolincs.ipsc.metadata %>% mutate(starfusion_path = paste0("/camp/project/proj-luscombn-patani/working/neurolincs/rnafusion/starfusion/",sample,".starfusion.abridged.coding_effect.tsv"))
# neurolincs.ipsc.metadata.starfusion = read_starfusion(neurolincs.ipsc.metadata)
# neurolincs.ipsc.metadata.starfusion
# write_csv(neurolincs.ipsc.metadata.starfusion, here(proj_path,"rnafusion/neurolincs.ipsc.metadata.starfusion.csv"))

neurolincs.ipsc.metadata.starfusion.csv = read_csv(here(proj_path,"rnafusion/neurolincs.ipsc.metadata.starfusion.csv")) 
neurolincs.imn_dimn.metadata.starfusion.csv = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% filter(dataset %in% c("neurolincs.diMN","neurolincs.iMN"))

neurolincs.ipsc_imn_dimn.starfusion = bind_rows(neurolincs.ipsc.metadata.starfusion.csv, neurolincs.imn_dimn.metadata.starfusion.csv) %>%
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))
neurolincs.ipsc_imn_dimn.starfusion

```

```{r}
neurolincs.ipsc_imn_dimn.starfusion.n = neurolincs.ipsc_imn_dimn.starfusion %>% add_count(dataset_sample) %>% # count number of fusions per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, DIV, condition, submitted_subject_id) # keep 1 row per sample

# Fill in samples with 0 counts
neurolincs.ipsc_imn_dimn.metadata.starfusion.samples_no_fusion = neurolincs.ipsc_iMN_diMN.metadata$dataset_sample[!neurolincs.ipsc_iMN_diMN.metadata$dataset_sample %in% neurolincs.ipsc_imn_dimn.starfusion.n$dataset_sample] # 0 i.e. all samples have at least 1 fusion
neurolincs.ipsc_imn_dimn.metadata.starfusion.samples_no_fusion # 0


neurolincs.ipsc_imn_dimn.starfusion.stat <- neurolincs.ipsc_imn_dimn.starfusion.n %>% t_test(n ~ DIV) %>% add_significance()  %>% add_xy_position()
# neurolincs.ipsc_imn_dimn.starfusion.stat$y.position = c(40,38,40,42,42)
neurolincs.ipsc_imn_dimn.starfusion.stat
# .y.   group1 group2    n1    n2 statistic    df     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     0      18        20    30     -2.04  47.7 0.047 0.095 ns                 16.8 <chr [2]>        1     2
# 2 n     0      32        20    14      1.52  23.8 0.142 0.142 ns                 18.1 <chr [2]>        1     3
# 3 n     18     32        30    14      3.05  32.7 0.005 0.014 *                  19.4 <chr [2]>        2     3
neurolincs.ipsc_imn_dimn.starfusion.violin <- violin_plot(mutate(neurolincs.ipsc_imn_dimn.starfusion.n, DIV = as.factor(DIV)), color_by = "DIV", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,18), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "NeuroLINCS iPSC, diMN, iMN: All gene Fusion Types") + 
  stat_pvalue_manual(neurolincs.ipsc_imn_dimn.starfusion.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) #y.position = 0.95*{{ylims[2]}}, 
neurolincs.ipsc_imn_dimn.starfusion.violin
#   .y.   group1 group2      n1    n2 statistic     df          p    p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl>  <dbl>      <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    89   208    -4.92  192.   0.00000187 0.000028 ****                 36 <chr [2]>        1     2
# 2 n     CTRL   SOD1        89    18    -2.41   18.6  0.026      0.292    ns                   38 <chr [2]>        1     3
# 3 n     CTRL   FUS         89     9     1.23   11.4  0.244      1        ns                   40 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      89     9     0.942   8.75 0.371      1        ns                   42 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     89    53    -1.18   70.4  0.243      1        ns                   42 <chr [2]>        1     6

ggline(neurolincs.ipsc_imn_dimn.starfusion.n, "DIV", "n", lintype = "submitted_subject_id")

neurolincs.ipsc_imn_dimn.starfusion.n %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
#   genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL            8.17      8    4.33   5   10        0    19   735    90  5   
# 2 Sporadic       11.1      10    4.96   8   13        2    36  2299   208  5   
# 3 SOD1           13.4      11.5  8.93   9   13.8      4    38   242    18  4.75
# 4 FUS             6.89      7    3.06   5    9        2    11    62     9  4   
# 5 TARDBP          5.6       5    6.29   1.5  5.75     0    22    56    10  4.25
# 6 C9orf72         9.64      8    7.90   3   12        2    39   511    53  9   


```


# Copy number variations

## RNAseqCNV

https://github.com/honzee/RNAseqCNV
https://www.nature.com/articles/s41375-022-01547-8

```{r}
devtools::install_github(repo = "honzee/RNAseqCNV")
library(RNAseqCNV)

# inputs: gene counts & GATK SNV MAFs

#config:
out_dir = "/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/rnaseqcnv"
count_dir = "/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/nfcore/star_salmon"
snv_dir = "/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling" 

# metadata: dataset_sample; count_file; vcf_file


RNAseqCNV_wrapper(config = "path/to/config", metadata = "path/to/metadata", snv_format = "vcf", genom_version = "hg38")




```

## CNVkit

## SuperFreq

## CaSpER

https://github.com/akdess/CaSpER/

```{r}

ipsc_mn_als_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.rds")) # 1.26G
centromere <- read.delim("/camp/lab/patanir/home/users/ziffo/genomes/CaSpER/cytoBand_centromere.txt", header=F)
annotation <- CaSpER::generateAnnotation(id_type="ensembl_gene_id", genes=rownames(assay(ipsc_mn_als_datasets$vsd)), ishg19=FALSE, centromere = centromere)

loh <- CaSpER::readBAFExtractOutput(path="/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/cnv", sequencing.type="bulk", suffix = "baf")
names(loh) <- gsub(".baf", "", names(loh))

object <- CaSpER::CreateCasperObject(raw.data=counts(ipsc_mn_als_datasets$dds,normalized=FALSE),annotation=annotation, control.sample.ids=pull(filter(ipsc_mn_als_datasets.metadata,condition == "ctrl"),dataset_sample), 
                                     cytoband=cytoband, loh.name.mapping=pull(ipsc_mn_als_datasets.metadata,dataset_sample), method="iterative",
                                     cnv.scale=3, loh.scale=3, loh=loh, sequencing.type="bulk", expr.cutoff=0.1, log.transformed = FALSE, genomeVersion = "hg38",
                                     matrix.type="raw",  filter="median")
object

final.objects <- CaSpER::runCaSpER(object, removeCentromere=T, cytoband=cytoband, method="iterative")
final.objects
# Large-Scale CNV Summarisation
finalChrMat <- CaSpER::extractLargeScaleEvents(final.objects, thr=0.75) 
finalChrMat
# Segment based CNV Summarisation
gamma <- 6
all.segments <- do.call(rbind, lapply(final.objects, function(x) x@segments))
segment.summary <- extractSegmentSummary (final.objects)
loss <- segment.summary$all.summary.loss
gain <- segment.summary$all.summary.gain
loh <- segment.summary$all.summary.loh
loss.final <- loss[loss$count>gamma, ]
gain.final <- gain[gain$count>gamma, ]
loh.final <- loh[loh$count>gamma, ]
# Gene based CNV Summarisation
all.summary<- rbind(loss.final, gain.final)
colnames(all.summary) [2:4] <- c("Chromosome", "Start",   "End")
rna <-  GRanges(seqnames = Rle(gsub("q", "", gsub("p", "", all.summary$Chromosome))), 
                IRanges(all.summary$Start, all.summary$End))   
ann.gr <- makeGRangesFromDataFrame(final.objects[[1]]@annotation.filt, keep.extra.columns = TRUE, seqnames.field="Chr")
hits <- findOverlaps(geno.rna, ann.gr)
genes <- splitByOverlap(ann.gr, geno.rna, "GeneSymbol")
genes.ann <- lapply(genes, function(x) x[!(x=="")])
all.genes <- unique(final.objects[[1]]@annotation.filt[,2])
all.samples <- unique(as.character(final.objects[[1]]@segments$ID))
rna.matrix <- gene.matrix(seg=all.summary, all.genes=all.genes, all.samples=all.samples, genes.ann=genes.ann)

# Visualisation
obj <- final.objects[[9]]
plotHeatmap(object=obj, fileName="/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/heatmap.png",cnv.scale= 3, cluster_cols = F, cluster_rows = T, show_rownames = T, only_soi = T)
plotLargeScaleEvent(object=obj, fileName="/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/large.scale.events.png") 
plotGEAndGT(chrMat=finalChrMat, genoMat=genoMat, fileName="/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/RNASeqAndGT.png") # only for small sample size
plotBAFAllSamples(loh = obj@loh.median.filtered.data, fileName="/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/LOHAllSamples.png") 
# plotBAFOneSample(object, fileName="/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/LOHPlotsAllScales.pdf") 
# plotBAFInSeperatePages (loh=obj@loh.median.filtered.data, folderName="/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/LOHPlots") # plot each sample on separate page
plotGEAndBAFOneSample (object=obj, cnv.scale=3, loh.scale=3, sample= "MN-5") # visualise gene expression and BAF signal for one sample
## calculate significant mutual exclusive and co-occurent events
results <- extractMUAndCooccurence(finalChrMat, loh, loh.name.mapping)
## visualize mutual exclusive and co-occurent events
plotMUAndCooccurence(results)



ipsc_mn_als_datasets$vsd.counts %>% select(-gene_name)

generateAnnotation <- function(id_type = "ensembl_gene_id", genes, ishg19, centromere, host="www.ensembl.org") {

    if (ishg19) {
        mart <- useDataset("hsapiens_gene_ensembl", useEnsembl(biomart = "ensembl", GRCh = 37, host = host))
    } else {
        mart <- useDataset("hsapiens_gene_ensembl", useEnsembl(biomart = "ensembl", host = host))
    }
    G_list <- getBM(filters = id_type, attributes = c(id_type, 
        "hgnc_symbol", "chromosome_name", "start_position", 
        "end_position", "band"), values = genes, 
        mart = mart, useCache=F)
    common <- intersect(genes, G_list[, id_type])
    ord <- match(common, G_list[, id_type])
    annotation <- G_list[ord, ]
    annotation <- annotation[order(annotation$start_position), ]
    annotation$cytoband <- paste0(annotation$chromosome_name, substr((annotation$band), 0, 1))
    
    chr.list <- as.character(1:22, "X")
    idx <- unlist(unique(as.vector(sapply(chr.list, function(x) as.vector(unlist(which(as.character(annotation$chromosome_name) == 
        x)))))))
    annotation <- as.data.frame(annotation[idx, ])
    
    colnames(annotation)[c(1:5, 7)] <- c("Gene", "GeneSymbol",  "Chr", "start", "end", "cytoband")
    
    annotation$isCentromer <- rep("no", nrow(annotation))
    
    centromere_snps <- NULL
    for (k in 1:(dim(centromere)[1])) {
        annotation$isCentromer[which(as.character(annotation$Chr) == gsub("chr", "", as.character(centromere$V1[k])) & (as.numeric(as.character(annotation$Position)) >= 
            centromere$V2[k] & as.numeric(as.character(annotation$Position)) <= centromere$V3[k]))] <- "yes"
    }
    

```

# Microsatelite instability

https://github.com/WangX-Lab/PreMSIm

```{r}
premsim_genes = c("DDX27", "EPM2AIP1", "HENMT1", "LYG1", "MLH1", "MSH4", "NHLRC1", "NOL4L", "RNLS", "RPL22L1", "RTF2", "SHROOM4", "SMAP1", "TTC30A", "ZSWIM3")

# import FPKM/TPM counts per sample
premsim_gene_counts = read_tsv(here(shared_path,"answerals/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% 
  filter(gene_name %in% premsim_genes)

# ipsc_mn_als_datasets.metadata = ipsc_mn_als_datasets.metadata %>% mutate(salmon_path = paste0(database_dir, "/nfcore/star_salmon/salmon.merged.gene_tpm.tsv"))
# ipsc_mn_als_datasets.metadata.filt = ipsc_mn_als_datasets.metadata %>% distinct(salmon_path, .keep_all = TRUE) 
# salmon.merged.gene_tpm.tsv = ipsc_mn_als_datasets.metadata.filt %>% pull(salmon_path) %>% map(read_tsv) #, col_names = c("n","type"), col_types = list(n="n",type="c"))
# names(salmon.merged.gene_tpm.tsv) = ipsc_mn_als_datasets.metadata.filt$dataset
# salmon.merged.gene_tpm.tab <- reduce(salmon.merged.gene_tpm.tsv, left_join, by = c("gene_id", "gene_name"), suffix = c())
# write_csv(salmon.merged.gene_tpm.tab, here(proj_path,"nfcore/ipsc_mn_als_datasets.tpm_counts.csv"))
salmon.merged.gene_tpm.tab = read_csv(here(proj_path,"nfcore/ipsc_mn_als_datasets.tpm_counts.csv"))
salmon.merged.gene_tpm.premsim = salmon.merged.gene_tpm.tab %>% filter(gene_name %in% premsim_genes)

salmon.merged.gene_tpm.premsim.data_pre = PreMSIm::data_pre(salmon.merged.gene_tpm.premsim)

filter(ipsc_mn_als_datasets$res, gene_name %in% premsim_genes)
ipsc_mn_als_datasets.premism.volcano <- volcano_plot(filter(ipsc_mn_als_datasets$res, gene_name %in% premsim_genes), numerator = "ALS", denomenator = "CTRL",
               subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 2, dpi = 300, gene_labels = c(premsim_genes))
ipsc_mn_als_datasets.premism.volcano


library(PreMSIm)


premsim_genes

LYG1, MSH4, and RPL22L1
data_pre <- function(deseq_res, input.path, type = c("gene_name", "gene_id")) {

	## Check arguments
	if (missing(input.path) || class(input.path) != "character")
		stop("'input.path' is missing or incorrect")
        type <- match.arg(type)   
	# ## Read input data
	# a <- read.table(input.path, stringsAsFactors = FALSE, header = TRUE, row.names = 1, sep = "\t", check.names = FALSE)
	
	## Filter and min-max scaling
	if (TRUE %in% is.na(match(feature[,type], rownames(a)))) {
		if (sum(!is.na(match(feature[,type], rownames(a)))) >= 5) {
			feature <- feature[!is.na(match(feature[,type], rownames(a))),]
			rownames(feature) <- NULL
		} else {
		stop("Some features of the current test set are missing!")
		}
	}
	a <- a[match(feature[,type], rownames(a)),, drop = FALSE]
	if (FALSE %in% complete.cases(a))
		stop("Predictor variables with missing values are presented in the current test set")
	if (dim(a)[2] != 1) {
		a <- apply(t(a), 2, function(x) {
			(x - min(x)) / (max(x) - min(x))
		})
		a[is.nan(a)] <- 1
	} else {
		a <- t(a)
	}
	if (type == "ID") {
		colnames(a) <- as.character(feature[,"Symbol"])
	}
	return(a)
}

```



