---
title: "Meta-analysis of ALS iPSNs"
font-family: 'Helvetica'
author: |
  | Dr Oliver Ziff
  | The Francis Crick Institute
  | Institute of Neurology
  | University College London, UK
date: "`r Sys.Date()`"
output: 
 rmarkdown::html_document:
   theme: spacelab
   highlight: haddock
   code_folding: hide
   toc: true
   toc_float: true
   smooth_scroll: true
   number_sections: false
   self_contained: true  
knit: (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(dirname(inputFile), 'docs/index.html')) })
---

```{css, echo = FALSE}
h1, #TOC>ul>li {
  color: #000000;
    font-weight: bold;
}
h2, #TOC>ul>ul>li {
  color: #41494D;
}
```

# Setup

```{r setup, include=FALSE}
library(here)
proj_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.metadata.RData")) # load metadata before functions to avoid overwriting with old functions

# Mac
camp_path = here("/Volumes/lab-luscomben/home/users/ziffo")
proj_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
script_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/scripts/ipsc_mn_als_meta")
shared_path = here("/Volumes/lab-luscomben/home/users/ziffo/proj-luscombn-patani/working")
collab_path = here("/Volumes/lab-luscomben/home/users/ziffo/patani-collab")
## OnDemand
# proj_path = here("/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
# script_path = here(here(proj_path,"scripts/ipsc_mn_als_meta")
# shared_path = here("/camp/project/proj-luscombn-patani/working")
# collab_path = here("/camp/lab/luscomben/home/users/ziffo/patani-collab")
setwd(proj_path)
set_here(proj_path)
here(proj_path)
# here::i_am("ipsc_mn_als_meta.Rmd")
here()
dr_here()

source("/Volumes/lab-luscomben/home/users/ziffo/scripts/functions/mac_R_functions.R") # source("/camp/lab/luscomben/home/users/ziffo/scripts/functions/R_functions.R")

# load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.RData") # 6.7GB > crashes R
ipsc_mn_als_datasets_with_lee = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.rds")) #2.56GB
# ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.ruv.lrt.rds"))
ipsc_mn_als_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.rds")) # 1.26G
ipsc_mn_als_datasets.polyA = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.rds"))
ipsc_mn_als_datasets.total = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.rds"))
# ipsc_mn_als_datasets.noiso = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.noiso.rds"))

ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds")) # 84M
ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds")) # 162M
ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds")) # 85M

nygc_postmortem_spinal_cord.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.tissue.rds"))
nygc_postmortem_spinal_cord.c9orf72_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.c9orf72_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.fus_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.fus_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sod1_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sod1_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sporadic_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sporadic_vs_ctrl.rds"))


```

## Update git / github

```{r}
library(here)
proj_path = here("/Volumes/lab-luscomben/home/users/ziffo/projects/ipsc-mn-als-meta")
library(usethis) # usethis & gert https://jeroen.github.io/gert2019 https://www.garrickadenbuie.com/blog/pull-request-flow-usethis/?interactive=1&steps=
use_git_config(user.name = "ojziff", user.email = "oliver.ziff@crick.ac.uk")
# usethis::create_github_token()

credentials::set_github_pat("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") 
# make github
library(gert)
setwd(here(proj_path,"scripts/ipsc_mn_als_meta")) # setwd then dont need to specify repo argument
git_commit_all("update") # stage all files & commit in one command - implies git_add(".")
pr_init("figures_and_tables") # initialise a new branch
pr_push() # this will open github in chrome > pull request > merge

git_sitrep() # print git and github info
git_config()
git_info()
git_ls() # list git tree
git_log() # check recent commits
git_status() # check which files are modified
# pr_resume("figures_and_tables") # resume a branch
# git_add(".") #stage the files you want to commmit - "." indicates all files.
# git_commit("update") # commit with comment
# pr_pull()


### Shiny objects
saveRDS(ipsc_mn_als_datasets$dds, here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_dds.rds"))


```



## Metadata

```{r}
ipsc_mn_als_datasets.metadata %>% count(dataset)
ipsc_mn_als_datasets.metadata %>% count(condition)
ipsc_mn_als_datasets.metadata %>% count(mutation)
ipsc_mn_als_datasets.metadata %>% distinct(dataset) #16 (14)
ipsc_mn_als_datasets.metadata %>% distinct(dataset_sample) # 419
ipsc_mn_als_datasets.metadata %>% count(condition) #318 ALS, 101 ctrl
ipsc_mn_als_datasets.metadata %>% count(dataset, condition) %>% print(n=Inf)
ipsc_mn_als_datasets.metadata %>% count(mutation) %>% arrange(-n)
ipsc_mn_als_datasets.metadata %>% count(mutation) # 11

ipsc_mn_als_datasets.metadata %>% filter(condition == "ctrl") %>% count(mutation)
ipsc_mn_als_datasets.metadata %>% filter(mutation == "iso") %>% count(condition)
ipsc_mn_als_datasets.metadata %>% filter(condition == "ctrl") %>% count(mutation)

ipsc_mn_als_datasets.introns_tag_pct$res %>% filter(padj < 0.05) #111
ipsc_mn_als_datasets.introns_tag_pct$irfinder %>% filter(padj < 0.05) #118,553
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) # 38
ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) # 25
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) # 74
ipsc_mn_als_datasets.total$irfinder %>% filter(padj < 0.05) #118829

ipsc_mn_als_datasets$dds %>% colData %>% as_tibble %>% glimpse
ipsc_mn_als_datasets$dds %>% colData %>% as_tibble %>% my_summarise(samtools_mqc_generalstats_samtools_raw_total_sequences)


# Answer ALS metadata
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_dataportal_datatable.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_participants.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_released_files.csv")) %>% clean_names %>% glimpse
read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/Clinical_Dictionary.csv")) %>% clean_names %>% glimpse
first_last_visit_alsfrs.answerals = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/answerals_progression_alsfrs.csv")) # answerals progression based on ALSFRS
AALSHXFX.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSHXFX.csv")) %>% clean_names %>% select(participant_id, subject_uid, onsetdt)
Demographics.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Demographics.csv")) %>% clean_names %>% select(participant_id, subject_uid, dob)
age_onset = AALSHXFX.csv %>% full_join(Demographics.csv) %>% mutate(age_onset = as.numeric((ddays(onsetdt) - ddays(dob)) / (364.25*24*60*60)), sample = gsub("-","_",participant_id)) %>% select(sample, age_onset) %>%
  mutate(early_late_onset = case_when(age_onset < 57 ~ "early", age_onset > 57 ~ "late"))
Mortality.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Mortality.csv")) %>% clean_names %>% filter(!dieddt %in% c("Glioblastoma","Prostate cancer", "physician assisted suicide")) %>% select(participant_id, subject_uid, dieddt) 
mortality = AALSHXFX.csv %>% full_join(Mortality.csv) %>% mutate(survival = as.numeric((ddays(dieddt) - ddays(onsetdt)) / (364.25*24*60*60)), sample = gsub("-","_",participant_id)) %>% select(sample, survival) %>%
  mutate(early_late_death = case_when(survival < 2.8 ~ "early", survival > 2.8 ~ "late"))
onset_site = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSHXFX.csv")) %>% clean_names %>% 
                        select(participant_id, subject_uid, hxblb, hxax, hxgen, hxli, hxot) %>% filter(!(hxblb == 0 & hxax == 0 & hxgen == 0 & hxli == 0 & hxot == 0)) %>%
  mutate(onset_site_detailed = case_when(hxblb == 1 & hxax == 0 & hxgen == 0 & hxli == 0 & hxot == 0 ~ "bulbar", hxblb == 0 & hxax == 0 & hxgen == 0 & hxli == 1 & hxot == 0 ~ "limb", hxblb == 0 & hxax == 0 & hxgen == 1 & hxli == 0 & hxot == 0 ~ "generalised", 
                                hxblb == 0 & hxax == 1 & hxgen == 0 & hxli == 0 & hxot == 0 ~ "axial", TRUE ~ "mixed"), #hxot == 1 ~ "other", 
         onset_site = case_when(onset_site_detailed %in% c("axial","mixed","generalised") ~ "other", TRUE ~ onset_site_detailed),
         bulbar_other = case_when(hxblb == 1 ~ "bulbar", TRUE ~ "other"), limb_other = case_when(hxli == 1 ~ "limb", TRUE ~ "other"), axial_other = case_when(hxax == 1 ~ "axial", TRUE ~ "other"), 
         bulbar_axial = case_when(hxblb == 1 ~ "bulbar", hxax == 1 ~ "axial", TRUE ~ "other"), bulbar_limb = case_when(hxblb == 1 ~ "bulbar", hxli == 1 ~ "limb", TRUE ~ "other"), bulbar_mixed = case_when(hxblb == 1 ~ "bulbar", onset_site == "mixed" ~ "mixed", TRUE ~ "other"), 
         limb_axial = case_when(hxli == 1 ~ "limb", hxax == 1 ~ "axial", TRUE ~ "other"), limb_mixed = case_when(onset_site == "mixed" ~ "mixed", hxli == 1 ~ "limb", TRUE ~ "other"), axial_mixed = case_when(hxax == 1 ~ "axial", onset_site == "mixed" ~ "mixed", TRUE ~ "other"),
         sample = gsub("-","_",participant_id)) %>% select(sample, subject_uid, onset_site, onset_site_detailed, bulbar_other,limb_other,axial_other,bulbar_axial,bulbar_limb,bulbar_mixed,limb_axial,limb_mixed,axial_mixed)
family_history = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Family_History_Log.csv")) %>% clean_names %>% select(participant_id, subject_uid, fhals, fhftd) %>% 
  mutate(family_history = case_when(fhals == 1 | fhftd == 1 ~ "yes", TRUE ~ "no"),sample = gsub("-","_",participant_id)) %>% select(sample, family_history) %>% distinct(sample, .keep_all = TRUE)
subjects = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/subjects.csv")) %>% clean_names %>% 
  mutate(mnd_group = case_when(subject_group_id == 17 ~ "other_mnd", subject_group_id == 1 ~ "als", subject_group_id == 5 ~ "healthy", subject_group_id == 11 ~ "carrier"), mnd_group = factor(mnd_group, levels = c("healthy", "carrier","other_mnd","als")), sample = gsub("-","_",participant_id)) %>% select(sample, mnd_group) %>% distinct(sample, .keep_all = TRUE)
ck_value = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/Auxiliary_Chemistry.csv")) %>% clean_names %>% drop_na(acckrslt) %>% mutate(sample = gsub("-","_",participant_id), cknorm = case_when(cknorm == 1 ~ "normal", TRUE ~ "raised")) %>% arrange(-acckrslt) %>% distinct(sample, .keep_all = TRUE) %>% select(sample, ck = acckrslt, cknorm)
AALSHXFX.csv = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/clinical/AALSDXFX.csv")) %>% clean_names %>% drop_na(elescrlr) %>% select(participant_id, elescrlr) %>% 
  mutate(sample = gsub("-","_",participant_id)) %>% select(sample, elescrlr) %>% arrange(elescrlr) %>% distinct(sample, .keep_all = TRUE)
answerals_transcriptomic_files = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_released_files.csv")) %>% clean_names %>% filter(omic == "transcriptomics", grepl(".fastq.gz$", path)) %>% 
  mutate(fastq = case_when(grepl("_1.fastq.gz$", path) ~ "fastq_1", grepl("_2.fastq.gz$", path) ~ "fastq_2")) %>% 
  filter(participant_id != "CTRL-NEUEU392AE8") %>% #17 fastq's associated with this ID. Unclear if these are technical repeats or incorrectly labelled.
  pivot_wider(c(participant_id, experiment), names_from = fastq, values_from = c(path,size)) %>% 
  mutate(fastq_1 = paste0(here(camp_path,"proj-luscombn-patani/working/answerals/reads/"),path_fastq_1), fastq_2 = paste0(here(camp_path,"proj-luscombn-patani/working/answerals/reads/"), path_fastq_2))
samplesheet = read_csv(here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/metadata/aals_dataportal_datatable.csv")) %>% clean_names %>% filter(transcriptomics == "Yes", participant_id != "CTRL-NEUEU392AE8") %>% # ipsc == "Yes", 
  left_join(answerals_transcriptomic_files) %>% 
  mutate(strandedness = "reverse", 
         condition = case_when(subject_group == "ALS" ~ "als", subject_group == "Healthy Control" ~ "ctrl", grepl("carrier",subject_group) ~ "carrier", subject_group == "Non-ALS MND" ~ "other_mnd"),
         mutation = case_when(self_reported_mutations %in% c("C9orf72","SOD1","SETX", "TARDBP (TDP43)") ~ self_reported_mutations, TRUE ~ wgs_variants), mutation = tolower(gsub(" [(]WGS[)]","",mutation)), mutation = gsub("tardbp [(]tdp43[)]","tardbp", mutation),  
         mutation = case_when(condition == "als" & is.na(mutation) ~ "sporadic", condition == "ctrl" & is.na(mutation) ~ "ctrl",TRUE ~ mutation), mutant = case_when(mutation %in% c("sporadic","ctrl") ~ "no", TRUE ~ "yes"),
         sample = gsub("-","_",participant_id), DIV = 32, instrument = "Illumina NovaSeq 6000") %>%
  select(sample, fastq_1, fastq_2, strandedness, condition, mutation, mutant, participant_id, guid, nygc_cgnd_id, sex, DIV, instrument, ethnicity, race, pls_subgroup, revised_el_escorial_criteria, age_at_sample_collection, age_at_symptom_onset, age_at_death, alsfrs_r_baseline, alsfrs_r_latest, alsfrs_r_progression_slope, cbs_baseline, cbs_latest, cbs_progression_slope, subject_group, ipsc, dimn, i_psc_line_id_root, i_psc_line_name, di_m_ns_line_name, batch_id, self_reported_mutations, wgs_variants, repeat_length_c9, repeat_length_atxn2, size_fastq_1, size_fastq_2) %>%
  filter(!sample %in% c("CASE_NEUMN922PMZ","CASE_NEUYC198DCR")) %>% # R2_fastq.gz corrupt crashing nfcore
  left_join(select(first_last_visit_alsfrs.answerals, sample, alsfrs_slope, progression)) %>% left_join(age_onset) %>% left_join(mortality) %>% left_join(onset_site) %>% left_join(family_history) %>% left_join(subjects) %>% left_join(ck_value) %>% left_join(AALSHXFX.csv) # add clinical data to metadata
samplesheet # Ensure no NA's in sample sheet
samplesheet %>% glimpse
samplesheet %>% distinct(participant_id) # 287
write_csv(samplesheet, file = here(camp_path,"proj-luscombn-patani/working/answerals/sample-details/samplesheet.csv"), na = "")

```


# Table S2 QC stats

Table S1 is iPSN differentiation protocols

Multiqc general stats

```{r}
multiqc_stat_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), multiqc_stat_paths = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project","/Volumes/lab-luscomben/home/users/ziffo",multiqc_stat_paths)) %>% pull(multiqc_stat_paths)
multiqc_stats.tsv = multiqc_stat_paths %>% map(read_tsv, show_col_types = FALSE)                                                                        
names(multiqc_stats.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(dataset)
multiqc_data <- map_dfr(multiqc_stats.tsv, bind_rows, .id = "dataset") %>% clean_names() %>% drop_na(samtools_mqc_generalstats_samtools_flagstat_total) %>% filter(sample %in% ipsc_mn_als_datasets_with_lee.metadata$sample) %>%
  mutate(dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincs0" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS", dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~ dataset)) %>%
  select(dataset, sample, raw_total_sequences = samtools_mqc_generalstats_samtools_raw_total_sequences, flagstat_total = samtools_mqc_generalstats_samtools_flagstat_total, mapped_passed = samtools_mqc_generalstats_samtools_mapped_passed, reads_mapped = samtools_mqc_generalstats_samtools_reads_mapped, reads_mapped_percent = samtools_mqc_generalstats_samtools_reads_mapped_percent, reads_aligned = quali_map_mqc_generalstats_qualimap_reads_aligned, r_rna_percent = biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna, bias_5_3 = quali_map_mqc_generalstats_qualimap_5_3_bias, proper_pairs_percent = r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent, reads_assigned_percent = feature_counts_mqc_generalstats_featurecounts_percent_assigned, reads_assigned = feature_counts_mqc_generalstats_featurecounts_assigned, percent_duplication = picard_mqc_generalstats_picard_percent_duplication, error_rate = samtools_mqc_generalstats_samtools_error_rate, non_primary_alignments = samtools_mqc_generalstats_samtools_non_primary_alignments, reads_properly_paired_percent = samtools_mqc_generalstats_samtools_reads_properly_paired_percent, percent_mapped = salmon_mqc_generalstats_salmon_percent_mapped, num_mapped = salmon_mqc_generalstats_salmon_num_mapped, uniquely_mapped_percent = star_mqc_generalstats_star_uniquely_mapped_percent, uniquely_mapped = star_mqc_generalstats_star_uniquely_mapped, percent_trimmed =cutadapt_mqc_generalstats_cutadapt_percent_trimmed, total_sequences = fast_qc_mqc_generalstats_fastqc_total_sequences, raw_percent_gc = fast_qc_raw_mqc_generalstats_fastqc_raw_percent_gc, raw_avg_sequence_length = fast_qc_raw_mqc_generalstats_fastqc_raw_avg_sequence_length, trimmed_percent_duplicates = fast_qc_trimmed_mqc_generalstats_fastqc_trimmed_percent_duplicates, trimmed_percent_gc = fast_qc_trimmed_mqc_generalstats_fastqc_trimmed_percent_gc)
multiqc_data %>% glimpse
write_csv(multiqc_data, here(proj_path,"alignment/multiqc_general_stats.csv"))

```


# Fig 1 Pipeline schematic

## PCA

```{r}
ipsc_mn_als_datasets_with_lee.pcaData <- plotPCA(ipsc_mn_als_datasets_with_lee$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","sareen","sommer","smith","sterneckert","wang","lee.CTRL") ~ "poly(A)", TRUE ~ "Total Ribo-Zero"), #dataset = gsub("\\."," ",dataset), #dataset = str_to_title(dataset),
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         ) %>%
  left_join(select(ipsc_mn_als_datasets_with_lee.metadata, dataset_sample, gender))

# Colour by dataset
colourCount = length(unique(mtcars$hp))
getPalette = colorRampPalette(brewer.pal(8, "Accent"))
getPalette(colourCount)
show_col(getPalette(colourCount))
#7FC97F" "#94C09B" "#A9B7B7" "#BEAED4" "#D3B4BA" "#E8BAA0" "#FDC086" "#FDD58C" "#FEEA92" "#FFFF99" "#BCCEA0" "#7A9DA8" 
# "#386CB0" "#75489F" "#B2258F" "#F0027F" "#DF1F5C" "#CF3D39" "#BF5B17" "#A15E31" "#83624B" "#666666"

# Colour by dataset
ipsc_mn_als_datasets_with_lee.pca_dataset <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) +
  scale_color_manual(values = c("AnswerALS"="#94C09B", "neurolincs.iMN"="#A9B7B7", "neurolincs.diMN"="#BEAED4", "bhinge"= "#7FC97F","catanese"="#7FC97F", "dafinca"="#94C09B", "desantis"="#FEEA92", "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "hawkins"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sommer"="#FFFF99", "sterneckert"="#7A9DA8", "wang"="#BCCEA0","lee.CTRL"="#75489F", "lee.ALS"="#386CB0")) +
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "right", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) + #coord_fixed() +
  ggforce::geom_mark_ellipse(aes(label = library_type), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE) + guides(color=guide_legend(title="Dataset", keywidth=0.1, keyheight=0.15, default.unit="inch"))
ipsc_mn_als_datasets_with_lee.pca_dataset

```

## Cell type marker heatmap

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","EDNRB",  "FGFR3",  "SLC4A4", "DIO2")#"SOX9", ,"MLC1"
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4", "PECAM1")
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CCL3","IL1A", "TREM2") # "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "CX3CR1","P2RY12", "TNF",  "TMEM119"
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1")

undifferentiated = c("SOX2","POU5F1","UTF1","REX1","ALPL","TERT","ESRG","CNMD","SFRP2")
neural.precursor = c("NES","SOX1","SOX2","CD133","PAX6","NOTCH1","OCLN","CDH1","CDH2","SOX10","VIM","GFAP")
neural.progenitor = c("MAP2","HUC","HUD","NF","NCAM","DCX","SOX2")
neural.crest <- c("SOX10","NGFR","FOXD3","SNAI1","SNAI2","LMX1A")   #NGFR very low in HB9 samples
dorsal.progenitor <- c("ATOH1","OLIG3","MSX1","PAX3","PAX7","DBX1","DBX2","IRX3","ASCL1")
ho.genes = c("SST","DCX","RFX4","ASCL1","HOXB8","NEFH","SCN1A","TNS1","SPOCK3","SNAP25","TDGF1","POU5F1","L1TD1","FXYD5","PHLDA2","TOP2A","DLGAP5","FZD7","ILF2","MSH2")

nonspecific = c("SOX2")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
pMN.domain = c("NKX6-1", "OLIG2")
ventral.progenitor = c("NKX2-2","NKX2-8")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")

nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

# neuronal.markers = c("NPAS4", "NPY", "SST", "CCK", "GAD1", "SYT1", "STMN2", "SYN1") # "ENO2", 

nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#"NEFH",
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3") #"ETV1", "NOS1",,"RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2"
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

### Cell type markers
celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, oligo.markers, mn.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]
celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte.markers", gene_name %in% endo.markers ~ "endo.markers", 
                           gene_name %in% myeloid.markers ~ "myeloid.markers", gene_name %in% oligo.markers ~ "oligo.markers", 
                           gene_name %in% mn.markers ~ "mn.markers"), 
         group = factor(group, levels = c("astrocyte.markers", "endo.markers", "myeloid.markers", "oligo.markers","mn.markers"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets_with_lee.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = as.factor(dataset)) %>% filter(!grepl("answerals_CASE",dataset_sample))

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
celltype.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))

### Cluster samples ###
celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(celltype.markers.marker.centered_mat)]

celltype.markers.rnaseq.cluster.heatmap <- Heatmap(t(celltype.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = celltype.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = celltype.markers.df.filt$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelial", "Myeloid", "Oligodendrocyte","Motor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), column_title = NULL, 
          show_row_names = FALSE, #row_names_gp = gpar(fontsize = 6), 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)),  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
celltype.markers.rnaseq.cluster.heatmap
ggsave(as.ggplot(celltype.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.celltype.markers.heatmap.png"), width = 8, height = 13)
```

## Merge

```{r}
ipsn_identities_pca_heatmap = plot_grid(ipsc_mn_als_datasets_with_lee.pca_dataset, NULL,
  as.ggplot(celltype.markers.rnaseq.cluster.heatmap),
  nrow = 3, rel_heights = c(1.5,0.1,4), labels = c("a","b",""))
# ipsn_identities_pca_heatmap
ggsave(ipsn_identities_pca_heatmap, filename = here(proj_path,"expression/deseq2/ipsn_identities_pca_heatmap.png"), width = 8.25, height = 12)

```


# Fig S2-3 PCA

Fig S1 is the PRISMA flow diagram

## PC1 & PC2 gene loadings

https://stackoverflow.com/questions/12760108/principal-components-analysis-how-to-get-the-contribution-of-each-paramete
https://github.com/mikelove/DESeq2/blob/master/R/plots.R

```{r}
ipsc_mn_als_datasets_with_lee.vsd.rv <- rowVars(assay(ipsc_mn_als_datasets_with_lee$vsd)) # calculate the variance for each gene
ipsc_mn_als_datasets_with_lee.vsd.ntop_genes <- order(ipsc_mn_als_datasets_with_lee.vsd.rv, decreasing=TRUE)[seq_len(min(500, length(ipsc_mn_als_datasets_with_lee.vsd.rv)))] # select the ntop genes by variance
ipsc_mn_als_datasets_with_lee.vsd.pca <- prcomp(t(assay(ipsc_mn_als_datasets_with_lee$vsd)[ipsc_mn_als_datasets_with_lee.vsd.ntop_genes,])) # perform a PCA on the data in assay(x) for the selected genes
ipsc_mn_als_datasets_with_lee.vsd.loadings <- as_tibble(ipsc_mn_als_datasets_with_lee.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
ipsc_mn_als_datasets_with_lee.vsd.loadings %>% arrange(PC1)
ipsc_mn_als_datasets_with_lee.vsd.loadings %>% arrange(-PC1)

# ipsc_mn_als_datasets_with_lee.vsd.pca <- prcomp(t(assay(ipsc_mn_als_datasets_with_lee$vsd))) # perform a PCA on the data in assay(x) for the selected genes
# ipsc_mn_als_datasets_with_lee.vsd.loadings <- as_tibble(ipsc_mn_als_datasets_with_lee.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2")#,"ARHGAP36")
V3.domain = c("NKX2-2","SIM1")
HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
neuronal.markers = c(nonspecific.neuronal, dl.domains, V0.domain,V1, V2a.domain, V2b.domain,MN.domain,V3.domain)
# nonpolyadenylated_genes = read_csv(here(camp_path,"genomes/annotation/nonpolyadenylated_genes.csv")) %>% pull(nonpolyadenylated_genes)

# scatter all PC1 vs PC2
ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter = ggplot(ipsc_mn_als_datasets_with_lee.vsd.loadings, aes(x = PC1, y = PC2 )) + 
    labs(x = "PC1 gene loading", y = "PC2 gene loading") +
    geom_hline(yintercept = 0, linetype = 3) + geom_vline(xintercept = 0, linetype = 3) +
    geom_point(size = 1, alpha = 0.1) +
    scale_fill_continuous(type = "viridis") +
    guides(fill = FALSE) +
    theme_oz() +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings, abs(PC1) > 0.04, abs(PC2) > 0.04, grepl("^H[1-9]",gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings, abs(PC1) > 0.04, abs(PC2) > 0.04, grepl("^SNOR|RN7SK|RMRP|RPPH1",gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", size = 2.3) +
  geom_text_repel(data = filter(ipsc_mn_als_datasets_with_lee.vsd.loadings,  gene_name %in% c(neuronal.markers, HOX.markers)), aes(x = PC1, y = PC2, label = gene_name), colour = "black", size = 2.3) +
  annotate("text", x = 0.04, y = 0.05, label = "PC1+ PC2+",size = 3, color = "black") + annotate("text", x = -0.07, y = -0.12, label = "PC1- PC2-", size = 3, color = "firebrick2") + 
  annotate("text", x = 0.04, y = -0.12, label = "PC1+ PC2-", size = 3, color = "black") + annotate("text", x = -0.07, y = 0.05, label = "PC1- PC2+", size = 3, color = "black")
ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter
# ggsave(ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter.png"), height = 9, width = 9)

```

## Sample characteristics

```{r}
# ipsc_mn_als_datasets_with_lee = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.rds")) #2.56GB
ipsc_mn_als_datasets_with_lee.pcaData <- plotPCA(ipsc_mn_als_datasets_with_lee$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang","lee.CTRL") ~ "poly(A)", TRUE ~ "Total Ribo-Zero")) %>%
  left_join(select(ipsc_mn_als_datasets_with_lee.metadata, dataset_sample, gender)) #%>% 
  # mutate(database_dir = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project", "/Volumes/lab-luscomben/home/users/ziffo", database_dir))
  
# multiqc stats
multiqc_general_stats_stat_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_general_stats_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), 
     multiqc_general_stats_stat_paths = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project","/Volumes/lab-luscomben/home/users/ziffo",multiqc_general_stats_stat_paths)) %>% pull(multiqc_general_stats_stat_paths)
multiqc_general_stats.tsv = multiqc_general_stats_stat_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_general_stats.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_general_stats <- map_dfr(multiqc_general_stats.tsv, bind_rows, .id = "database_dir") %>% clean_names()
multiqc_rseqc_read_distribution_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_rseqc_read_distribution_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"),
     multiqc_rseqc_read_distribution_paths = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project","/Volumes/lab-luscomben/home/users/ziffo",multiqc_rseqc_read_distribution_paths)) %>% pull(multiqc_rseqc_read_distribution_paths)
multiqc_rseqc_read_distribution.tsv = multiqc_rseqc_read_distribution_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_rseqc_read_distribution.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_rseqc_read_distribution <- map_dfr(multiqc_rseqc_read_distribution.tsv, bind_rows, .id = "database_dir") %>% clean_names()
ipsc_mn_als_datasets_with_lee.pcaData = ipsc_mn_als_datasets_with_lee.pcaData %>% left_join(multiqc_general_stats) %>% left_join(multiqc_rseqc_read_distribution) # need unique sample names - If multiple dataset names within a single multiqc_general_stats.txt then will only join the first dataset name

# Colour by dataset
colourCount = length(unique(mtcars$hp))
getPalette = colorRampPalette(brewer.pal(8, "Accent"))
getPalette(colourCount)
show_col(getPalette(colourCount))
#7FC97F" "#94C09B" "#A9B7B7" "#BEAED4" "#D3B4BA" "#E8BAA0" "#FDC086" "#FDD58C" "#FEEA92" "#FFFF99" "#BCCEA0" "#7A9DA8" 
# "#386CB0" "#75489F" "#B2258F" "#F0027F" "#DF1F5C" "#CF3D39" "#BF5B17" "#A15E31" "#83624B" "#666666"

# Colour by dataset
ipsc_mn_als_datasets_with_lee.pca_dataset <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) +
  scale_color_manual(values = c("AnswerALS"="#94C09B", "neurolincs.iMN"="#A9B7B7", "neurolincs.diMN"="#BEAED4", "bhinge"= "#7FC97F","catanese"="#7FC97F", "dafinca"="#94C09B", "desantis"="#FEEA92", "kapeli"="#A9B7B7", "kiskinis"="#BEAED4", "luisier"="#D3B4BA", "hawkins"="#E8BAA0", "sareen"="#FDC086", "smith"="#FDD58C", "sommer"="#FFFF99", "sterneckert"="#7A9DA8", "wang"="#BCCEA0","lee.CTRL"="#75489F", "lee.ALS"="#386CB0")) +
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "right", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) #+ #coord_fixed() +
  # ggforce::geom_mark_ellipse(aes(label = library_type), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE) + guides(color=guide_legend(title="Dataset", keywidth=0.1, keyheight=0.15, default.unit="inch"))
ipsc_mn_als_datasets_with_lee.pca_dataset

# Colour by library_type
ipsc_mn_als_datasets_with_lee.pca_library_type <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=library_type)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))+labs(colour="Library preparation")
ipsc_mn_als_datasets_with_lee.pca_library_type
# Colour by instrument
ipsc_mn_als_datasets_with_lee.pca_instrument <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=instrument)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_blank(), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_instrument
# Colour by DIV
ipsc_mn_als_datasets_with_lee.pca_DIV <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=DIV)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top"))  + labs(color="Days in vitro") + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_DIV
# Colour by ALS
ipsc_mn_als_datasets_with_lee.pca_als <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=condition)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch", title.position = "top")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_als
# Colour by mutation
ipsc_mn_als_datasets_with_lee.pca_mutation <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=mutation)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_mutation
# Colour by gender
ipsc_mn_als_datasets_with_lee.pca_gender <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=gender)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets_with_lee.pca_gender

```

Fig S2 Merge

```{r}
ipsc_mn_als_datasets_with_lee.pca_merged = plot_grid(
  ipsc_mn_als_datasets_with_lee.vsd.loadings.scatter, ipsc_mn_als_datasets_with_lee.pca_dataset,
  ipsc_mn_als_datasets_with_lee.pca_library_type, ipsc_mn_als_datasets_with_lee.pca_instrument, #ncol = 2, labels = c("b","c")), 
  ipsc_mn_als_datasets_with_lee.pca_gender, ipsc_mn_als_datasets_with_lee.pca_DIV, #ncol = 2, labels = c("d","e")), 
  ipsc_mn_als_datasets_with_lee.pca_als, ipsc_mn_als_datasets_with_lee.pca_mutation, #ncol = 2, labels = c("f","g")),
  nrow = 4, ncol = 2, labels = c("auto"))
# ipsc_mn_als_datasets_with_lee.pca_merged
ggsave(ipsc_mn_als_datasets_with_lee.pca_merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.pca_merged.png"), height = 11.75, width = 8.25)
```

## RNAseq QC metrics

Fig S3

```{r}
### PCA 2

# Colour by strandedness
ipsc_mn_als_datasets_with_lee.pca_strandedness <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=strandedness)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(keywidth=0.2, keyheight=0.2, default.unit="inch")) + 
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_strandedness
# Colour by read depth
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_raw_total_sequences)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6, angle = 0), legend.title=element_text(size=7), legend.position = "top", legend.box = "vertical", legend.spacing = unit(1, 'cm'))  + 
  guides(color = guide_colourbar(title = "read depth", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_raw_total_sequences,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences


# Color by intronic read % intron_read_pct
ipsc_mn_als_datasets_with_lee.pca_read_distribution <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=introns_tag_pct)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "intronic reads %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$introns_tag_pct,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance")) 
ipsc_mn_als_datasets_with_lee.pca_read_distribution
# Colour by biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna
ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "ribosomal rna %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna
# Colour by quali_map_mqc_generalstats_qualimap_5_3_bias
ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=quali_map_mqc_generalstats_qualimap_5_3_bias)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "5' 3' bias", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=1, low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias
# Colour by quali_map_mqc_generalstats_qualimap_reads_aligned
ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_reads_aligned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=quali_map_mqc_generalstats_qualimap_reads_aligned)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads aligned", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$quali_map_mqc_generalstats_qualimap_reads_aligned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_reads_aligned
# Colour by picard_mqc_generalstats_picard_percent_duplication
ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=picard_mqc_generalstats_picard_percent_duplication)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Read duplication %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$picard_mqc_generalstats_picard_percent_duplication,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +  
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication
# Colour by samtools_mqc_generalstats_samtools_reads_mapped
ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mapped)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
  guides(color = guide_colourbar(title = "Reads mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped
# Colour by star_mqc_generalstats_star_uniquely_mapped_percent
ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=star_mqc_generalstats_star_uniquely_mapped_percent)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
  guides(color = guide_colourbar(title = "Reads uniquely mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
  scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$star_mqc_generalstats_star_uniquely_mapped_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent

ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged = ggarrange(
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_strandedness, ipsc_mn_als_datasets_with_lee.pca_read_distribution , ncol = 2, labels = c("a","b")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_raw_total_sequences, ipsc_mn_als_datasets_with_lee.pca_quali_map_mqc_generalstats_qualimap_5_3_bias, ncol = 2, labels = c("c","d")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna, ipsc_mn_als_datasets_with_lee.pca_picard_mqc_generalstats_picard_percent_duplication, ncol = 2, labels = c("e","f")),
  ggarrange(ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped_percent, ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped, ncol = 2, labels = c("g","h")), 
  nrow = 4, heights = c(1,1,1,1))
# ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged
ggsave(ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.pca_nfcore_qc_metrics.merged.png"), height = 11.75, width = 8.25)


# # Colour by samtools_mqc_generalstats_samtools_reads_mapped_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mapped_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mapped_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mapped_percent
# # Colour by samtools_mqc_generalstats_samtools_error_rate
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_error_rate <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_error_rate)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Error rate", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_error_rate,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_error_rate
# # Colour by samtools_mqc_generalstats_samtools_non_primary_alignments
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_non_primary_alignments <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_non_primary_alignments)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Non primary alignments", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_non_primary_alignments,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_non_primary_alignments
# # Colour by samtools_mqc_generalstats_samtools_reads_mq0_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mq0_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_mq0_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Mapping quality 0 %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_mq0_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_mq0_percent
# # Colour by samtools_mqc_generalstats_samtools_flagstat_total
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_flagstat_total <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_flagstat_total)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_flagstat_total
# # Colour by samtools_mqc_generalstats_samtools_mapped_passed
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_mapped_passed <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_mapped_passed)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Mapped reads passed", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_mapped_passed,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_mapped_passed
# # Colour by r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent
# ipsc_mn_als_datasets_with_lee.pca_r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Proper pairs %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent
# # Colour by samtools_mqc_generalstats_samtools_reads_properly_paired_percent
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_properly_paired_percent <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=samtools_mqc_generalstats_samtools_reads_properly_paired_percent)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Properly paired %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$samtools_mqc_generalstats_samtools_reads_properly_paired_percent,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_samtools_mqc_generalstats_samtools_reads_properly_paired_percent
# # Colour by feature_counts_mqc_generalstats_featurecounts_percent_assigned
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_percent_assigned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=feature_counts_mqc_generalstats_featurecounts_percent_assigned)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Assigned reads %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$feature_counts_mqc_generalstats_featurecounts_percent_assigned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_percent_assigned
# # Colour by feature_counts_mqc_generalstats_featurecounts_assigned
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_assigned <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=feature_counts_mqc_generalstats_featurecounts_assigned)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Assigned reads", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$feature_counts_mqc_generalstats_featurecounts_assigned,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_feature_counts_mqc_generalstats_featurecounts_assigned
# # Colour by salmon_mqc_generalstats_salmon_percent_mapped
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_percent_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=salmon_mqc_generalstats_salmon_percent_mapped)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped %", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$salmon_mqc_generalstats_salmon_percent_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_percent_mapped
# # Colour by salmon_mqc_generalstats_salmon_num_mapped
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_num_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=salmon_mqc_generalstats_salmon_num_mapped)) + geom_point(size=1) +
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) +
#   guides(color = guide_colourbar(title = "Reads mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$salmon_mqc_generalstats_salmon_num_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# ipsc_mn_als_datasets_with_lee.pca_salmon_mqc_generalstats_salmon_num_mapped
# # Colour by star_mqc_generalstats_star_uniquely_mapped
# ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped <- ggplot(ipsc_mn_als_datasets_with_lee.pcaData, aes(PC1, PC2, color=star_mqc_generalstats_star_uniquely_mapped)) + geom_point(size=1) + 
#   theme_oz() + theme(legend.text=element_text(size=6), legend.title=element_text(size=6), legend.position = "top", legend.box = "vertical", legend.spacing = unit(0.1, 'cm')) + 
#   guides(color = guide_colourbar(title = "Reads uniquely mapped", barwidth = 15, barheight = 0.5, ticks = FALSE, title.position = "top"))  + 
#   scale_color_gradient2(midpoint=mean(ipsc_mn_als_datasets_with_lee.pcaData$star_mqc_generalstats_star_uniquely_mapped,na.rm=TRUE), low="blue", mid="white", high="red", label=comma) +
#   xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets_with_lee.pcaData, "percentVar"))[2],"% variance"))
# # ipsc_mn_als_datasets_with_lee.pca_star_mqc_generalstats_star_uniquely_mapped

```



# Fig S4-9 Heatmap expression

## Cell type marker heatmap

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","EDNRB",  "FGFR3",  "SLC4A4", "DIO2")#"SOX9", ,"MLC1"
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4", "PECAM1")
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CCL3","IL1A", "TREM2") # "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "CX3CR1","P2RY12", "TNF",  "TMEM119"
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1")

undifferentiated = c("SOX2","POU5F1","UTF1","REX1","ALPL","TERT","ESRG","CNMD","SFRP2")
neural.precursor = c("NES","SOX1","SOX2","CD133","PAX6","NOTCH1","OCLN","CDH1","CDH2","SOX10","VIM","GFAP")
neural.progenitor = c("MAP2","HUC","HUD","NF","NCAM","DCX","SOX2")
neural.crest <- c("SOX10","NGFR","FOXD3","SNAI1","SNAI2","LMX1A")   #NGFR very low in HB9 samples
dorsal.progenitor <- c("ATOH1","OLIG3","MSX1","PAX3","PAX7","DBX1","DBX2","IRX3","ASCL1")
ho.genes = c("SST","DCX","RFX4","ASCL1","HOXB8","NEFH","SCN1A","TNS1","SPOCK3","SNAP25","TDGF1","POU5F1","L1TD1","FXYD5","PHLDA2","TOP2A","DLGAP5","FZD7","ILF2","MSH2")

nonspecific = c("SOX2")
roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
pMN.domain = c("NKX6-1", "OLIG2")
ventral.progenitor = c("NKX2-2","NKX2-8")
floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")

nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

# neuronal.markers = c("NPAS4", "NPY", "SST", "CCK", "GAD1", "SYT1", "STMN2", "SYN1") # "ENO2", 

nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#"NEFH",
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3") #"ETV1", "NOS1",,"RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2"
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

### Cell type markers
celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, oligo.markers, mn.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]
celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte.markers", gene_name %in% endo.markers ~ "endo.markers", 
                           gene_name %in% myeloid.markers ~ "myeloid.markers", gene_name %in% oligo.markers ~ "oligo.markers", 
                           gene_name %in% mn.markers ~ "mn.markers"), 
         group = factor(group, levels = c("astrocyte.markers", "endo.markers", "myeloid.markers", "oligo.markers","mn.markers"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets_with_lee.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = as.factor(dataset)) %>% filter(!grepl("answerals_CASE",dataset_sample))

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
celltype.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))

### Cluster samples ###
celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$dataset_sample]
dataset.samples.df$dataset_sample[!dataset.samples.df$dataset_sample %in% rownames(celltype.markers.marker.centered_mat)]

celltype.markers.rnaseq.cluster.heatmap <- Heatmap(t(celltype.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = celltype.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = celltype.markers.df.filt$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelial", "Myeloid", "Oligodendrocyte","Motor neuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), column_title = NULL, 
          show_row_names = FALSE, #row_names_gp = gpar(fontsize = 6), 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)),  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
celltype.markers.rnaseq.cluster.heatmap
ggsave(as.ggplot(celltype.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.celltype.markers.heatmap.png"), width = 8.25, height = 9)
```

## Neuron type markers

```{r}
nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","SYT4","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#,"NEFH"
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","FOXP1","ALDH1A2","ETV4","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3","ETV1", "NOS1","RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2") #
oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

neuron.type.markers = c(motor.neuron.progenitor, postmitotic.motor.neuron, nonspecific.neuronal,interneuron.markers) 
neuron.type.markers[!neuron.type.markers %in% gtf$gene_name]
neuron.type.markers.df = tibble("gene_name" = neuron.type.markers) %>% filter(gene_name %in% neuron.type.markers) %>% 
  mutate(group = case_when(gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% motor.neuron.progenitor ~ "motor.neuron.progenitor", 
                           gene_name %in% postmitotic.motor.neuron ~ "postmitotic.motor.neuron", #gene_name %in% oculomotor.markers ~ "oculomotor.markers", 
                           gene_name %in% interneuron.markers ~ "interneuron"), 
         group = factor(group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

dataset.samples.df = ipsc_mn_als_datasets_with_lee.metadata %>% select(dataset_sample, dataset) %>% mutate(dataset_sample = gsub("dafinca.tardbp", "dafinca", dataset_sample), dataset_sample = gsub("dafinca.c9orf72", "dafinca", dataset_sample), dataset = gsub("lee.ALS", "lee", dataset), dataset = gsub("lee.CTRL", "lee", dataset), dataset = gsub("mn_d25_", "",dataset), dataset = as.factor(dataset)) %>% filter(!grepl("answerals_CASE",dataset_sample))

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
neuron.type.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% neuron.type.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuron.type.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
neuron.type.markers.df.filt = neuron.type.markers.df %>% filter(gene_name %in% neuron.type.markers.marker$gene_name)
neuron.type.markers.marker.mat <- make_matrix(select(neuron.type.markers.marker,-gene_name), pull(neuron.type.markers.marker,gene_name))

### Cluster samples ###
neuron.type.markers.rnaseq.heatmap <- Heatmap(t(neuron.type.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_order = neuron.type.markers.df.filt$gene_name,
          column_names_gp = gpar(fontsize = 6), 
          column_split = factor(neuron.type.markers.df.filt$group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron")),
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",4)), labels = c("nonspecific\nneuronal", "motor neuron\nprogenitor", "postmitotic\nmotor neuron", "interneuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, 
          show_row_names = FALSE,
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen","Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
neuron.type.markers.rnaseq.heatmap
ggsave(as.ggplot(neuron.type.markers.rnaseq.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.neuron.type.markers.heatmap.png"), width = 8.25, height = 9)

# ## clustering samples
# neuron.type.markers.rnaseq.cluster.heatmap <- Heatmap(t(neuron.type.markers.marker.mat), 
#           border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
#           column_order = neuron.type.markers.df.filt$gene_name,
#           column_names_gp = gpar(fontsize = 6), 
#           column_split = factor(neuron.type.markers.df.filt$group, levels = c("nonspecific.neuronal", "motor.neuron.progenitor", "postmitotic.motor.neuron", "interneuron")),
#           bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",4)), labels = c("nonspecific\nneuronal", "motor\nneuron\nprogenitor", "postmitotic\nmotor\nneuron", "interneuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           column_title = NULL, 
#           show_row_names = TRUE, row_names_gp = gpar(fontsize = 6), 
#           row_title = NULL, cluster_rows = TRUE, cluster_column_slices = FALSE, cluster_columns = FALSE)
# neuron.type.markers.rnaseq.cluster.heatmap
# ggsave(as.ggplot(neuron.type.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets_with_lee.neuron.type.markers.cluster.heatmap.png"), width = 9, height = 16)
# 
## Motor neuron markers only heatmap
# celltype.marker.df = data.frame("gene_name" = mn.markers) %>% mutate(group = case_when(gene_name %in% mn.markers ~ "motor neuron"))
# ipsc_mn_als_datasets_with_lee$vsd.counts %>% filter(gene_name %in% mn.markers) %>% group_by(gene_name) %>% filter(n()>1) # any duplicate gene_names?
# 
# vsd_counts.celltype.marker <-  ipsc_mn_als_datasets_with_lee$vsd.counts  %>% filter(gene_name %in% mn.markers) %>% select(-gene_id) %>% mutate_if(is.numeric, funs((. - 4.505))) #rowwise() %>%
# vsd_counts.celltype.marker.mat <- make_matrix(select(vsd_counts.celltype.marker,-gene_name), pull(vsd_counts.celltype.marker,gene_name))
# celltype.marker.df.filt <- celltype.marker.df %>% filter(gene_name %in% vsd_counts.celltype.marker$gene_name)
# 
# motor.neuron.marker.heatmap <- Heatmap(t(vsd_counts.celltype.marker.mat), 
#           heatmap_legend_param = list(title = ""), 
#           cluster_columns = TRUE, column_order = vsd_counts.celltype.marker$gene_name,
#           show_column_names = TRUE, column_names_gp = gpar(fontsize = 8, fontface = "italic"), #row_labels = celltype.marker.row.labels,
#           top_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = c("white")), labels = c("Motor neuron markers"), labels_gp = gpar(fontsize = 10))), column_title = NULL,
#           show_row_names = FALSE, #column_names_rot = 90, 
#           row_title = NULL, cluster_rows = FALSE,
#           right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",16)), 
#                                                                  labels = c("AnswerALS", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Luisier",  "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "bhinge", "Smith", "Sterneckert", "Wang"), 
#                                                                  labels_gp = gpar(fontsize = 7), labels_rot = 0)), row_split = dataset.samples.df$dataset)
# 
# motor.neuron.marker.heatmap

```


## Dorso-ventral markers

From Fig 2 in https://journals.biologists.com/dev/article/148/15/dev199711/271192/Single-cell-transcriptome-profiling-of-the-human 

```{r}
nonspecific.neuronal = c("ELAVL3","MAP2","STMN2")
dl1.domain = c("BARHL1","BARHL2","LHX2","LHX9","ATOH1")
dl2.domain = c("FOXD3","LHX5")
dl3.domain = c("POU4F1","OTP","TLX3")
dl4.domain = c("LHX1","PAX8","LBX1","PAX2","GBX1","PTF1A")
dl5.domain = c("GSX1","GSX2","PAX3","PAX7","LMX1B","ASCL1")
dl6.domain = c("DMRT3","WT1")
dl.domains = c(dl1.domain,dl2.domain,dl3.domain,dl4.domain,dl5.domain,dl6.domain)
V0.domain = c("EVX1","EVX2","PITX2")
V1 = c("EN1")
V2a.domain = c("LHX3","SOX14","SOX21","VSX2","FOXN4","VSX1")
V2b.domain = c("GATA2","GATA3","TAL1","MSX1")
MN.domain = c("ISL1","ISL2","MNX1","SLC10A4","SLC18A3","OLIG2","ALDH1A2","ARHGAP36")
V3.domain = c("NKX2-2","SIM1")

neuronal.markers = c(dl.domains, V0.domain,V1, V2a.domain, V2b.domain,MN.domain,V3.domain) #nonspecific.neuronal
neuronal.markers[!neuronal.markers %in% gtf$gene_name]
neuronal.markers.df = tibble("gene_name" = neuronal.markers) %>% 
  mutate(group = case_when(gene_name %in% nonspecific.neuronal ~ "nonspecific.neuronal", gene_name %in% dl.domains ~ "dl.domains", 
                           gene_name %in% V0.domain ~ "V0.domain", gene_name %in% V1 ~ "V1", 
                           gene_name %in% V2a.domain ~ "V2a.domain", gene_name %in% V2b.domain ~ "V2b.domain", gene_name %in% MN.domain ~ "MN.domain", gene_name %in% V3.domain ~ "V3.domain"), 
         group = factor(group, levels = c("dl.domains","V0.domain", "V1", "V2a.domain", "V2b.domain", "MN.domain", "V3.domain"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

# together with other datasets
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
neuronal.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
  filter(gene_name %in% neuronal.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = neuronal.markers.df$gene_name)) %>% select(-gene_id) %>%
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
neuronal.markers.marker.mat <- make_matrix(select(neuronal.markers.marker,-gene_name), pull(neuronal.markers.marker,gene_name))

### Cluster samples ###
neuronal.markers.marker.centered_mat = t(scale(t(neuronal.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
dorsoventral.markers.heatmap <- Heatmap(t(neuronal.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = neuronal.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = neuronal.markers.df$group,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",7)), labels =  c("dl\ndomains","V0\ndomain", "V1", "V2a\ndomain", "V2b\ndomain", "MN\ndomain", "V3"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          # show_row_names = TRUE, row_names_gp = gpar(fontsize = 6, col = c(rep("#386CB0",9), rep("#75489F",9), rep("black",90))), #row_names_rot = 90, 
          cluster_rows = FALSE,
          row_title = NULL, 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",18)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)))
dorsoventral.markers.heatmap
ggsave(as.ggplot(dorsoventral.markers.heatmap), filename = here(proj_path,"expression/deseq2/dorsoventral.markers.heatmap.png"), width = 8.25, height = 9)


# # cluster by columns
# Heatmap(neuronal.markers.marker.centered_mat, 
#           border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
#           row_order = neuronal.markers.df$gene_name,
#           row_names_gp = gpar(fontsize = 7), 
#           row_split = factor(neuronal.markers.df$group, levels =  c("nonspecific.neuronal","dl.domains","V0.domain", "V1", "V2a.domain", "V2b.domain", "MN.domain", "V3.domain")),
#           right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = rep("white",8)), labels =  c("nonspecific\nneuronal","dl\ndomains","V0\ndomain", "V1", "V2a\ndomain", "V2b\ndomain", "MN\ndomain", "V3\ndomain"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           row_title = "Neuronal markers", row_title_gp = gpar(fontsize = 8), 
#           show_column_names = TRUE, column_names_gp = gpar(fontsize = 6, col = c(rep("#386CB0",9), rep("#75489F",9), rep("black",90))), #column_names_rot = 90,
#           cluster_rows = FALSE, cluster_row_slices = FALSE, cluster_columns = TRUE,
#           column_title = NULL)

```

Progenitor dosroventral markers


```{r}
# nonspecific = c("SOX2")
# roof.plate = c("LMX1A","MSX1","MSX2","PAX3","WNT1","GDF7")    
# dorsal.progenitor = c("IRX3","IRX5","OLIG3","PAX6","ASCL1","GBX2","GSX2","PAX7","GSX1","DBX2","DBX1","SP8")
# intermediate.progenitor = c("NKX6-2","PRDM12","FOXN4")
# pMN.domain = c("NKX6-1", "OLIG2")
# ventral.progenitor = c("NKX2-2","NKX2-8")
# floor.plate = c("ARX","FERD3L","FOXA2","LMX1B","SHH")
# 
# progenitor.markers = c(nonspecific, roof.plate, dorsal.progenitor,intermediate.progenitor, pMN.domain, ventral.progenitor,floor.plate)
# progenitor.markers[!progenitor.markers %in% gtf$gene_name]
# progenitor.markers.df = tibble("gene_name" = progenitor.markers) %>% 
#   mutate(group = case_when(gene_name %in% nonspecific ~ "nonspecific", gene_name %in% roof.plate ~ "roof.plate", 
#                            gene_name %in% dorsal.progenitor ~ "dorsal.progenitor", gene_name %in% intermediate.progenitor ~ "intermediate.progenitor", 
#                            gene_name %in% pMN.domain ~ "pMN.domain", gene_name %in% ventral.progenitor ~ "ventral.progenitor", gene_name %in% floor.plate ~ "floor.plate"), 
#          group = factor(group, levels = c("nonspecific","roof.plate", "dorsal.progenitor", "intermediate.progenitor", "pMN.domain", "ventral.progenitor", "floor.plate"))) %>% 
#   arrange(group) %>% distinct(gene_name, .keep_all = TRUE)
# 
# # together with other datasets
# min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
# progenitor.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% 
#   filter(gene_name %in% progenitor.markers.df$gene_name) %>% 
#   dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
#   arrange(factor(gene_name, levels = progenitor.markers.df$gene_name)) %>% select(-gene_id) %>%
#   rename_with(~ gsub("dafinca.tardbp", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("dafinca.c9orf72", "dafinca", .x, fixed = TRUE)) %>% rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE))
# progenitor.markers.marker.mat <- make_matrix(select(progenitor.markers.marker,-gene_name), pull(progenitor.markers.marker,gene_name))
# 
# ### Cluster samples ###
# progenitor.markers.marker.centered_mat = t(scale(t(progenitor.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
# progenitor.markers.heatmap <- Heatmap(t(progenitor.markers.marker.mat), 
#           border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
#           column_order = progenitor.markers.df$gene_name,
#           column_names_gp = gpar(fontsize = 7), 
#           column_split = progenitor.markers.df$group,
#           bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",7)), labels =  c("nonspecific","roof\nplate", "dorsal\nprogenitor", "intermediate\nprogenitor", "pMN\ndomain", "ventral\nprogenitor", "floor\nplate"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           column_title ="Progenitor markers", column_title_gp = gpar(fontsize = 8), 
#           show_row_names = FALSE, 
#           # row_order = dataset.samples.df$dataset_sample,
#           row_split = dataset.samples.df$dataset,
#           right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",17)), 
#                                                                  labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
#                                                                  labels_gp = gpar(fontsize = 7), labels_rot = 0)),
#           row_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE)
# progenitor.markers.heatmap
# 

```


## Rostro-caudal markers

HOX markers

```{r}
HOX.markers <- gtf %>% filter(grepl("^HOX[A-D][0-9]+", gene_name)) %>% distinct(gene_name) %>% pull(gene_name)
# forebrain.markers = c("FOXG1","OTX2")
# HOX.markers = c(forebrain.markers, HOX.markers)
HOX.markers.df <- data.frame(row.names = HOX.markers, gene_name = HOX.markers) %>% 
  mutate(group = gsub("HOX[A-D]","", gene_name), group = factor(as.numeric(gsub("-AS$","", group)))) %>% 
  arrange(group) %>% 
  mutate(level = factor(case_when(group %in% c(1,2,3) ~ "Hindbrain", group %in% c(4,5,6,7) ~ "Cervical", group %in% c(8,9) ~ "Thoracic", 
                                  group %in% c(10,11) ~ "Lumbar", TRUE ~ "Sacral"), levels = c("Hindbrain","Cervical","Thoracic","Lumbar","Sacral")))
min_vst = min(ipsc_mn_als_datasets_with_lee$vsd.counts[,2])
HOX.markers.marker <- ipsc_mn_als_datasets_with_lee$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% 
  filter(gene_name %in% HOX.markers.df$gene_name) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst))) %>% 
  arrange(factor(gene_name, levels = HOX.markers.df$gene_name)) %>% select(-gene_id) %>%
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) %>% 
  select(-contains("answerals_CASE"))
HOX.markers.marker.mat <- make_matrix(select(HOX.markers.marker,-gene_name), pull(HOX.markers.marker,gene_name))
HOX.markers.marker.centered.mat = t(scale(t(HOX.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
HOX.markers.heatmap <- Heatmap(t(HOX.markers.marker.mat), 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"),
          column_order = HOX.markers.df$gene_name,
          column_names_gp = gpar(fontsize = 7), 
          column_split = HOX.markers.df$level,
          bottom_annotation = columnAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Hindbrain", "Cervical", "Thoracic","Lumbar","Sacral"), labels_gp = gpar(fontsize = 8), labels_rot = 0)),
          column_title = NULL, column_title_gp = gpar(fontsize = 8), 
          cluster_columns = FALSE, cluster_column_slices = FALSE, 
          cluster_rows = FALSE,
          row_title = NULL, 
          # row_order = dataset.samples.df$dataset_sample,
          row_split = dataset.samples.df$dataset,
          show_row_names = FALSE, 
          right_annotation = rowAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",17)), 
                                                                 labels = c("AnswerALS", "Bhinge", "Catanese", "Dafinca\nC9orf72", "Dafinca\nTARDBP", "De Santis", "Hawkins", "Kapeli", "Kiskinis", "Lee", "Luisier", "NeuroLINCS\ndiMN batch", "NeuroLINCS\niMN batch", "Sareen", "Sommer", "Smith", "Sterneckert", "Wang"), 
                                                                 labels_gp = gpar(fontsize = 7), labels_rot = 0)))
HOX.markers.heatmap
ggsave(as.ggplot(HOX.markers.heatmap), filename = here(proj_path,"expression/deseq2/HOX.markers.heatmap.png"), width = 8.25, height = 9)
```



# Fig 2 hiPSC ALS meta-analysis

## Volcano

```{r}
ipsc_mn_als_datasets$res %>% filter(padj < 0.05)# 43
# ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)#3
ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #23 down, 20 up
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) #0
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # RNASEL
ipsc_mn_als_datasets$res %>% filter(gene_name %in% c("SFPQ","FUS","TARDBP"))# 43

## Volcano
ipsc_mn_als_datasets.als.labels <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.gene.labels <- ipsc_mn_als_datasets$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.volcano <- volcano_plot(ipsc_mn_als_datasets$res, numerator = "ALS", denomenator = "CTRL",
               subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 2, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.als.labels, ipsc_mn_als_datasets.gene.labels))
# ipsc_mn_als_datasets.volcano

# ## MA plot
# ipsc_mn_als_datasets.ma <- ma_plot(ipsc_mn_als_datasets$res, 
#                numerator = "ALS", denomenator = "CTRL", subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 1000, dpi = 300, 
#                gene_labels = c(ipsc_mn_als_datasets.als.labels, ipsc_mn_als_datasets.gene.labels)) #+ xlim(0,10000)
# ipsc_mn_als_datasets.ma

ipsc_mn_als_datasets$res %>% filter(gene_name %in% c("H2AX","TP53BP1","PARP1") | grepl("^RPA[1-4]",gene_name)) # RNASEL
ipsc_mn_als_datasets$res %>% filter(grepl("^RPA[1-9]",gene_name)) # RNASEL

```

## GO terms & GSEA

```{r}
ipsc_mn_als_datasets.gost <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_als_datasets.gprofiles <- ipsc_mn_als_datasets.gost$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
ipsc_mn_als_datasets.gprofiles_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")
ipsc_mn_als_datasets.gprofiles_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(ipsc_mn_als_datasets.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "transcription regulator complex",
# "sequence-specific DNA binding",
# "double-stranded DNA binding",
# "sequence-specific double-stranded DNA binding",
# "chromatin",
"neuron fate commitment",
# "transcription regulatory region nucleic acid binding",
# "transcription cis-regulatory region binding",
# "DNA-binding transcription factor activity",
"cell differentiation in spinal cord",
"DNA-binding transcription factor activity, RNA polymerase II-specific",
"ventral spinal cord development",
"cis-regulatory region sequence-specific DNA binding"
# "Aplasia of the musculature"
)
ipsc_mn_als_datasets.gprofiles_filt_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_als_datasets.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"p53 transcriptional gene network",
"Genotoxicity pathway",
"miRNA regulation of DNA damage response",
"DNA damage response",
"p53 signaling pathway")
ipsc_mn_als_datasets.gprofiles_filt_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_als_datasets.gprofiles_filt_combined <- bind_rows(ipsc_mn_als_datasets.gprofiles_filt_down, ipsc_mn_als_datasets.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
ipsc_mn_als_datasets.gprofiles_filt_combined = ipsc_mn_als_datasets.gprofiles_filt_combined %>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
ipsc_mn_als_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_als_datasets.gprofiles_filt_combined) #+ theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
ipsc_mn_als_datasets.go_curated

# which genes are driving the terms
p53.DDR.repair_gene_set = unique(c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`,gprofiler_database$`DNA repair`))
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway)) # CDKN1A
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% c(gprofiler_database$`DNA Damage Response`)) # TNFRSF10B, CDKN1A, RRM2B, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA Damage Response`) # TNFRSF10B, CDKN1A, RRM2B, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA repair`) # TNFRSF10B, CDKN1A, RRM2B, SESN1

wikipathways.msigdb <- gmtPathways(here(camp_path,"/genomes/gene-ontology/c2.cp.wikipathways.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
hpo.msigdb <- gmtPathways(here(camp_path,"/genomes/gene-ontology/c5.hpo.v7.5.1.symbols.gmt"))# %>% mutate(term = gsub("GO|_", " ", term))
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_MIRNA_REGULATION_OF_DNA_DAMAGE_RESPONSE) # TNFRSF10B, CDKN1A, MDM2, SESN1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`TP53 Network`) # CDKN1A
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`cell fate commitment`) # TBX5, MYOG, LMO4, OLIG1, TBR1, OLIG2, FOXN4, BCL11B
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`neuron fate commitment`) # OLIG1, OLIG2, FOXN4
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_OLIGODENDROCYTE_SPECIFICATION_AND_DIFFERENTIATION_LEADING_TO_MYELIN_COMPONENTS_FOR_CNS) # OLIG1, OLIG2
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% hpo.msigdb$HP_APLASIA_OF_THE_MUSCULATURE) # CHRND, TBX5
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`ventral spinal cord development`) # LMO4, OLIG2, FOXN4
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`DNA-binding transcription activator activity, RNA polymerase II-specific`) # MYOG, TBX5, POU5F1
ipsc_mn_als_datasets$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$chromatin) # MYOG, TBX5, POU5F1, OLIG1, OLIG2, FOXN4

## GSEA: DNA damage response, p53 signalling pathway
ipsc_mn_als_datasets.geneList <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.geneList) <- ipsc_mn_als_datasets$res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ipsc_mn_als_datasets.geneList = sort(ipsc_mn_als_datasets.geneList, decreasing = TRUE)

# p53 signalling pathway
p53.signalling_genes.list <- list("p53.signalling" = ipsc_mn_als_datasets$res$gene_name[ipsc_mn_als_datasets$res$gene_name %in% p53.signalling.pathway])
ipsc_mn_als_datasets.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = ipsc_mn_als_datasets.geneList)
ipsc_mn_als_datasets.p53.signalling_fgseaRes
# pathway         pval         padj   log2err        ES      NES size                                leadingEdge
# 1: p53.signalling 0.0004854782 0.0004854782 0.4984931 0.4073582 1.437247  264 RRM2B,CDKN1A,EDA2R,MDM2,TRIAP1,GADD45A,...
ipsc_mn_als_datasets.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], ipsc_mn_als_datasets.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("p53 signalling") + xlab("") +
  annotate("text", x = 35000, y = 0.3, label = paste0("NES = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$pval, digits = 6)), size = 2.8, hjust = 0) + theme_oz()
ipsc_mn_als_datasets.p53.signalling.gseaplot

```


## DoRoTHEA & PROGENy

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/pw_bk.html 
<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.res.matrix <- ipsc_mn_als_datasets$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_progeny.pathwayActivity <- ggplot(ipsc_mn_als_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "ALS vs CTRL Normalised enrichment") +
    stat_pvalue_manual(ipsc_mn_als_progeny.contrast_acts, label = "p.signif", y.position = 14, xmin = "source", xmax = NULL, size = 3, hide.ns = TRUE) 
ipsc_mn_als_progeny.pathwayActivity

# p53 weights
ipsc_mn_als_progeny.p53weights = net_progeny %>% filter(source == "p53", target %in% ipsc_mn_als_datasets$res$gene_name) %>% left_join(select(ipsc_mn_als_datasets$res, target=gene_name,stat)) %>% 
  mutate(color = case_when(weight > 0 & stat > 0 ~ '1', weight > 0 & stat < 0 ~ '2', weight < 0 & stat > 0 ~ '2', weight < 0 & stat < 0 ~ '1', TRUE ~ "3"))
ipsc_mn_als_progeny.p53weights.plot = ggplot(ipsc_mn_als_progeny.p53weights, aes(x = weight, y = stat, color = color)) + geom_point(size = 0.5) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_progeny.p53weights, weight > 5, stat > 3), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.1) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "p53 pathway weights", x = "p53 pathway gene weight", y = "ALS vs CTRL test statistic") +
  geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted') +
  coord_cartesian(xlim = c(-8,18), ylim = c(-2.5,5)) 
ipsc_mn_als_progeny.p53weights.plot

```

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_dorothea.contrast_acts.labels.up = ipsc_mn_als_dorothea.contrast_acts %>% top_n(score, n = 3) %>% pull(source)
ipsc_mn_als_dorothea.contrast_acts.labels.down = ipsc_mn_als_dorothea.contrast_acts %>% top_n(-score, n = 3) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.contrast_acts, source %in% c(ipsc_mn_als_dorothea.contrast_acts.labels.up, ipsc_mn_als_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_dorothea.contrast_acts.volcano

ipsc_mn_als_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_dorothea.contrast_acts %>% arrange(score)

# # # TP53 target genes
# ipsc_mn_als_dorothea.tp53 <- net_dorothea %>%  filter(source == "TP53") %>% mutate(gene_name = target) %>% left_join(select(ipsc_mn_als_datasets$res, gene_name,stat,log2FoldChange,pvalue)) %>%
#   mutate(color = case_when(mor > 0 & stat > 0 ~ '1', mor > 0 & stat < 0~'2',mor < 0 & stat > 0~'2',mor < 0 & stat < 0~'1',TRUE ~ "3"))
# 
# ipsc_mn_als_dorothea.tp53.volcano = ggplot(ipsc_mn_als_dorothea.tp53, aes(x = log2FoldChange, y = -log10(pvalue), color = color, size=abs(mor))) + geom_point() + scale_size(range = c(0, 3)) +
#   scale_colour_manual(values = c("red","royalblue3","grey")) +
#   geom_text_repel(data = filter(ipsc_mn_als_dorothea.tp53, abs(stat) > 3), aes(label = gene_name, size=1), fontface = "italic", max.overlaps = Inf, size = 2.3) +
#   theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +  geom_vline(xintercept = 0, linetype = 'dotted') + geom_hline(yintercept = 0, linetype = 'dotted') +
#     labs(title = "TP53 regulon", x = "log2 fold change (ALS vs CTRL)", y = expression(-log[10]~P~value)) + ylim(c(0,2)) +   geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.tp53, log2FoldChange > 1), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5)
# ipsc_mn_als_dorothea.tp53.volcano

ipsc_mn_als_dorothea.tp53 %>% arrange(-stat) # TNFRSF10B, SESN1, RRM2B, CDKN1A, MDM2
```

## Merge

```{r}
ipsc_mn_als_datasets.figs.merged = plot_grid(
  plot_grid(ipsc_mn_als_datasets.volcano, ipsc_mn_als_datasets.go_curated, ipsc_mn_als_datasets.p53.signalling.gseaplot, ncol=3, rel_widths = c(1,0.8,0.6), labels = c("a","b","c")), 
  plot_grid(ipsc_mn_als_progeny.pathwayActivity, ipsc_mn_als_progeny.p53weights.plot,ipsc_mn_als_dorothea.contrast_acts.volcano, ncol=3, rel_widths = c(1,0.9,1), labels = c("d","e","f")), 
  nrow = 2, rel_heights = c(1.1,1))
# ipsc_mn_als_datasets.figs.merged
ggsave(ipsc_mn_als_datasets.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.figs.merged.png"), height = 7, width = 10)

```


# Table S3 DGE panALS

```{r}
ipsc_mn_als_datasets.res.sig = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% select(gene_name, everything())
# ipsc_mn_als_datasets.res.sig
write_csv(ipsc_mn_als_datasets.res.sig, here(proj_path,"expression/deseq2/TableS2.ipsc_mn_als_datasets.res.sig.csv"))

```

# Table S4 Dorothea panALS

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
net_dorothea %>% distinct(source) # 429
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% select(-statistic, -condition, transcription_factor = source, normalised_enrichment_score = score) %>% arrange(p_value, -abs(normalised_enrichment_score))
# ipsc_mn_als_dorothea.contrast_acts
write_csv(ipsc_mn_als_dorothea.contrast_acts, here(proj_path,"expression/deseq2/TableS4.ipsc_mn_als_dorothea.csv"))

```

# Fig S8 polyA & total subgroup analyses

## polyA

```{r}
# ipsc_mn_als_datasets.polyA = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.rds"))
# ipsc_mn_als_datasets.total = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.rds"))

ipsc_mn_als_datasets.polyA.metadata %>% count(dataset) # 10
ipsc_mn_als_datasets.polyA.metadata %>% count(condition) # 48 ALS, 43 CTRL
ipsc_mn_als_datasets.total.metadata %>% count(dataset) # 5
ipsc_mn_als_datasets.total.metadata %>% count(condition) # 275 ALS, 63 CTRL

# PCA
ipsc_mn_als_datasets.polyA.pcaData <- plotPCA(ipsc_mn_als_datasets.polyA$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "poly(A)"), dataset_sample, gender)) 
  
ipsc_mn_als_datasets.polyA.pca_dataset <- ggplot(ipsc_mn_als_datasets.polyA.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.polyA.pcaData, "percentVar"))[1],"% variance")) + ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.polyA.pcaData, "percentVar"))[2],"% variance")) + labs(color="polyA\ndatasets")
ipsc_mn_als_datasets.polyA.pca_dataset

## Volcano
ipsc_mn_als_datasets.polyA.metadata %>% count(condition)
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) # 57
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)# 4
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #32 down, 25 up
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) # PRPH2
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # POLR2J3
ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # POLR2J3

ipsc_mn_als_datasets.polyA.als.labels <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.gene.labels <- ipsc_mn_als_datasets.polyA$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.volcano <- volcano_plot(ipsc_mn_als_datasets.polyA$res, numerator = "ALS", denomenator = "CTRL",
               title = "polyA", subtitle = "43 Controls vs 48 ALS", ymax = 10, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.polyA.als.labels, ipsc_mn_als_datasets.polyA.gene.labels,"SESN1","TCEAL6"), distinct_labels = TRUE)
ipsc_mn_als_datasets.polyA.volcano

## DoRoTHEA & PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.polyA.res.matrix <- ipsc_mn_als_datasets.polyA$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.polyA_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.polyA.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.polyA_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.polyA_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "polyA: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.polyA_progeny.contrast_acts, label = "p.signif", y.position = 12, xmin = "source", xmax = NULL, size = 6, hide.ns = TRUE) 
ipsc_mn_als_datasets.polyA_progeny.pathwayActivity

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.polyA.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% filter(p_value < 0.15) %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "polyA: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano

ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% arrange(score)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% filter(source == "TP53")

```


## Total

```{r}
# PCA
ipsc_mn_als_datasets.total.pcaData <- plotPCA(ipsc_mn_als_datasets.total$vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/luscomben/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "Total"), dataset_sample, gender)) 

ipsc_mn_als_datasets.total.pca_dataset <- ggplot(ipsc_mn_als_datasets.total.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(nrow=2, keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.total.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.total.pcaData, "percentVar"))[2],"% variance"))  + labs(color="Total\ndatasets") 
ipsc_mn_als_datasets.total.pca_dataset

## Volcano
ipsc_mn_als_datasets.total.metadata %>% count(condition)
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) # 10
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) %>% left_join(distinct(gtf, ensemblID, .keep_all = T)) %>% filter(seqnames %in% c("X","Y")) %>% print(n=Inf)# 0
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #7 down, 3 up
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) # 
ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) # RNASEL, MRPL18

## Volcano
filter(ipsc_mn_als_datasets.metadata, library_type == "Total") %>% count(condition)
ipsc_mn_als_datasets.total.als.labels <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.total.gene.labels <- ipsc_mn_als_datasets.total$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.total.volcano <- volcano_plot(ipsc_mn_als_datasets.total$res, numerator = "ALS", denomenator = "CTRL",
               title = "Total", subtitle = "63 Controls vs 275 ALS", ymax = 8, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_als_datasets.total.als.labels, ipsc_mn_als_datasets.total.gene.labels,"RNASEL","SESN1", "TCEAL6","MTCO1P12","MTCO2P12","LMO4"))
ipsc_mn_als_datasets.total.volcano

## DoRoTHEA & PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.total.res.matrix <- ipsc_mn_als_datasets.total$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.total_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.total.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.total_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.total_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Total: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.total_progeny.contrast_acts, label = "p.signif", y.position = 8, xmin = "source", xmax = NULL, size = 6, hide.ns = TRUE) 
ipsc_mn_als_datasets.total_progeny.pathwayActivity


net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
# net_dorothea %>% distinct(source) # 429
ipsc_mn_als_datasets.total_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.total.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.total_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Total: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.total_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano

ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% filter(source == "TP53")
ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% arrange(-score)
ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% arrange(score)



## Compare polyA & total
# Venn overlap
ipsc_mn_als_datasets.polyA.res.dge.up <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.res.dge.down <- ipsc_mn_als_datasets.polyA$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
ipsc_mn_als_datasets.total.res.dge.up <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.total.res.dge.down <- ipsc_mn_als_datasets.total$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)

# # ggvenn
ipsc_mn_als_datasets.polyA.total.up_genes <- list("polyA" = ipsc_mn_als_datasets.polyA.res.dge.up, "Total" = ipsc_mn_als_datasets.total.res.dge.up)
ipsc_mn_als_datasets.polyA.total.down_genes <- list("polyA" = ipsc_mn_als_datasets.polyA.res.dge.down, "Total" = ipsc_mn_als_datasets.total.res.dge.down)
ipsc_mn_als_datasets.polyA.total.up_venn <- ggvenn(ipsc_mn_als_datasets.polyA.total.up_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap Up")  + theme(plot.subtitle = element_text(hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, -1, -1, -1), "lines"))
ipsc_mn_als_datasets.polyA.total.up_venn

ipsc_mn_als_datasets.polyA.total.down_venn <- ggvenn(ipsc_mn_als_datasets.polyA.total.down_genes, fill_color = c("dodgerblue3", "firebrick1", "forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.7) + labs(subtitle = "Overlap down")  + theme(plot.subtitle = element_text(hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, -1, -1, -1), "lines"))
ipsc_mn_als_datasets.polyA.total.down_venn

ipsc_mn_als_datasets.polyA.res.dge.up[ipsc_mn_als_datasets.polyA.res.dge.up %in% ipsc_mn_als_datasets.total.res.dge.up]
# 
# # Scatter correlation
ipsc_mn_als_datasets.polyA_total.labels = select(ipsc_mn_als_datasets.total$res, gene_id, gene_name, polyA.stat = stat) %>% left_join(select(ipsc_mn_als_datasets.polyA$res, gene_id, gene_name, total.stat = stat)) %>% filter((polyA.stat > 2.5 & total.stat > 2.5) | (polyA.stat < -2.5 & total.stat < -2.5))
ipsc_mn_als_datasets.polyA_total.list = list("polyA: ALS vs CTRL" = ipsc_mn_als_datasets.polyA$res, "Total: ALS vs CTRL" = ipsc_mn_als_datasets.total$res)
ipsc_mn_als_datasets.polyA_total_stat.corr = plot_de_correlation(model_list1 = ipsc_mn_als_datasets.polyA_total.list, model_list2 = ipsc_mn_als_datasets.polyA_total.list, mutation1 = "polyA: ALS vs CTRL", mutation2 = "Total: ALS vs CTRL", highlight = pull(ipsc_mn_als_datasets.polyA_total.labels,gene_name), plot_corr_line = TRUE)
ipsc_mn_als_datasets.polyA_total_stat.corr

```


## Merge

```{r}
ipsc_mn_als_datasets.polyA_total.figs.merged = plot_grid(
  ipsc_mn_als_datasets.polyA.pca_dataset, ipsc_mn_als_datasets.total.pca_dataset,
  ipsc_mn_als_datasets.polyA.volcano, ipsc_mn_als_datasets.total.volcano,
  ipsc_mn_als_datasets.polyA_progeny.pathwayActivity, ipsc_mn_als_datasets.total_progeny.pathwayActivity,
  ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano, ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano, 
  nrow = 4, ncol=2, rel_heights = c(0.7,1,0.8,1), labels = "auto") 
# ipsc_mn_als_datasets.polyA_total.figs.merged
ggsave(ipsc_mn_als_datasets.polyA_total.figs.merged, filename = here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA_total.figs.merged.png"), height = 11.75, width = 8.25)

# # Table DGE total & polyA
# ipsc_mn_als_datasets.polyA_total.sig = select(ipsc_mn_als_datasets.polyA$res, gene_id, gene_name, padj.polyA=padj, stat.polyA=stat, lfc.polyA=log2FoldChange) %>% left_join(select(ipsc_mn_als_datasets.total$res, gene_id, gene_name, padj.total=padj,stat.total=stat, lfc.total=log2FoldChange)) %>% filter(padj.polyA < 0.05 | padj.total < 0.05) %>% mutate(changed = case_when(padj.polyA < 0.05 & lfc.polyA > 0 ~ "polyA up", padj.polyA < 0.05 & lfc.polyA < 0 ~ "polyA down", padj.total < 0.05 & lfc.total > 0 ~ "total up", padj.total < 0.05 & lfc.total < 0 ~ "total down"))
# ipsc_mn_als_datasets.polyA_total.sig
# ipsc_mn_als_datasets.polyA_total.sig %>% count(changed)
# write_csv(ipsc_mn_als_datasets.polyA_total.sig, here(proj_path,"expression/deseq2/Table.ipsc_mn_als_datasets.polyA_total.sig.csv"))

```



# Fig 3 Compare mutations

Compare sporadic, C9orf72, SOD1, FUS & TARDBP

```{r}
ipsc_mn_sporadic_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sporadic_datasets.rds")) #733M
ipsc_mn_fus_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_fus_datasets.rds")) # 84M
ipsc_mn_sod1_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_sod1_datasets.rds")) # 162M
ipsc_mn_c9orf72_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_c9orf72_datasets.rds")) # 287M
ipsc_mn_tardbp_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_tardbp_datasets.rds")) # 85M

```

### Volcano

```{r}
ipsc_mn_mutants_bind_datasets.res = bind_rows(mutate(ipsc_mn_sporadic_datasets$res, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$res, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$res, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$res, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$res, mutation = "FUS"))#, mutate(ipsc_mn_vcp_datasets$res, mutation = "VCP"))

ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% count(mutation) %>% arrange(-n)
# mutation     n
#   <chr>    <int>
# 1 TARDBP    3547
# 2 FUS        202
# 3 C9orf72    161
# 4 SOD1         7
# 5 Sporadic     4
ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05)

ipsc_mn_mutants_join_datasets.res = full_join(select(ipsc_mn_c9orf72_datasets$res, gene_name, padj.c9orf72 = padj, log2FoldChange.c9orf72 = log2FoldChange),
          select(ipsc_mn_sporadic_datasets$res, gene_name, padj.sals = padj, log2FoldChange.sals = log2FoldChange)) %>%
  full_join(select(ipsc_mn_fus_datasets$res, gene_name, padj.fus = padj, log2FoldChange.fus = log2FoldChange)) %>%
  full_join(select(ipsc_mn_tardbp_datasets$res, gene_name, padj.tardbp = padj, log2FoldChange.tardbp = log2FoldChange)) %>%
  full_join(select(ipsc_mn_sod1_datasets$res, gene_name, padj.sod1 = padj, log2FoldChange.sod1 = log2FoldChange)) #%>% full_join(select(ipsc_mn_vcp_datasets$res, gene_name, padj.vcp = padj, log2FoldChange.vcp = log2FoldChange))
ipsc_mn_mutants_join_datasets.res.labels = ipsc_mn_mutants_join_datasets.res %>% filter(c(padj.c9orf72 < 0.05 & padj.fus < 0.05 & padj.tardbp < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.tardbp < 0.05 & padj.sod1 < 0.05) | 
                                                                                        c(padj.fus < 0.05 & padj.tardbp < 0.05 & padj.c9orf72 < 0.05)) %>% pull(gene_name)

ipsc_mn_tardbp_datasets.metadata %>% count(dataset)
ipsc_mn_tardbp_datasets.metadata %>% count(condition) 
ipsc_mn_tardbp_datasets.labels <- ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.001, gene_name %in% c(ALS_genes,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_tardbp_datasets.volcano <- volcano_plot(ipsc_mn_tardbp_datasets$res, "TARDBP", numerator = "TARDBP", denomenator = "CTRL", subtitle = "48 Controls vs 10 TARDBP", ymax = 15, xmax = 4.5, dpi = 300, gene_labels = c(ipsc_mn_tardbp_datasets.labels,"UPK3BL1","NPIPA8"))
# ipsc_mn_tardbp_datasets.volcano

ipsc_mn_fus_datasets.metadata %>% count(dataset) 
ipsc_mn_fus_datasets.metadata %>% count(condition) 
ipsc_mn_fus_datasets$res %>% filter(padj < 0.01) %>% top_n(n=10, wt = abs(stat))
ipsc_mn_fus_datasets.labels <- ipsc_mn_fus_datasets$res %>% filter(padj < 0.05, gene_name %in% c(p53.DDR.repair_gene_set, ALS_genes,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_fus_datasets.volcano <- volcano_plot(ipsc_mn_fus_datasets$res, "FUS", numerator = "FUS", denomenator = "CTRL", subtitle = "55 Controls vs 14 FUS", ymax = 15, xmax = 4, dpi = 300, gene_labels = c(ipsc_mn_fus_datasets.labels,"UPK3BL1","NPIPA8","ZNF439","HSD11B2","RNF5P1","RNU1-3"))
# ipsc_mn_fus_datasets.volcano

ipsc_mn_c9orf72_datasets.metadata %>% count(dataset)
ipsc_mn_c9orf72_datasets.metadata %>% count(condition) 
ipsc_mn_c9orf72_datasets.labels <- ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.02, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS,"UPK3BL1","NPIPA8")) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_c9orf72_datasets.volcano <- volcano_plot(ipsc_mn_c9orf72_datasets$res, "C9orf72", numerator = "C9orf72", denomenator = "CTRL", subtitle = "83 Controls vs 60 C9orf72", ymax = 9, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_c9orf72_datasets.labels, "C9orf72","UPK3BL1","NPIPA8"))
# ipsc_mn_c9orf72_datasets.volcano

ipsc_mn_sod1_datasets.metadata %>% count(dataset) 
ipsc_mn_sod1_datasets.metadata %>% count(condition) 
ipsc_mn_sod1_datasets.labels <- ipsc_mn_sod1_datasets$res %>% filter(padj < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_sod1_datasets.volcano <- volcano_plot(ipsc_mn_sod1_datasets$res, "SOD1", numerator = "SOD1", denomenator = "CTRL", subtitle = "63 Controls vs 20 SOD1", ymax = 12, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_sod1_datasets.labels))
# ipsc_mn_sod1_datasets.volcano

ipsc_mn_sporadic_datasets.metadata %>% count(dataset) 
ipsc_mn_sporadic_datasets.metadata %>% count(condition) 
ipsc_mn_sporadic_datasets.labels <- ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_sporadic_datasets.volcano <- volcano_plot(ipsc_mn_sporadic_datasets$res, "Sporadic", numerator = "Sporadic", denomenator = "CTRL", subtitle = "56 Controls vs 208 Sporadic", ymax = 12, xmax = 3, dpi = 300, gene_labels = c(ipsc_mn_sporadic_datasets.labels))
# ipsc_mn_sporadic_datasets.volcano

# volcano_multiplot <- ipsc_mn_tardbp_datasets.volcano + ipsc_mn_fus_datasets.volcano + ipsc_mn_c9orf72_datasets.volcano + ipsc_mn_sod1_datasets.volcano + ipsc_mn_sporadic_datasets.volcano +
#   plot_layout(ncol = 3, nrow = 2)
# volcano_multiplot

ipsc_mn_mutants_join_datasets.res.labels
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% ALS_genes]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% RNA_BINDING_PROTEINS]
ipsc_mn_mutants_join_datasets.res.labels[ipsc_mn_mutants_join_datasets.res.labels %in% p53.signalling.pathway]

```


### Correlation heatmap

```{r}
# heatmap of correlation coefficients
ipsc_mn_mutations_stat = select(ipsc_mn_sporadic_datasets$res, gene_id, Sporadic = stat) %>% 
  left_join(select(ipsc_mn_c9orf72_datasets$res, gene_id, C9orf72 = stat)) %>%
  left_join(select(ipsc_mn_sod1_datasets$res, gene_id, SOD1 = stat)) %>%
  left_join(select(ipsc_mn_fus_datasets$res, gene_id, FUS = stat)) %>%
  left_join(select(ipsc_mn_tardbp_datasets$res, gene_id, TARDBP = stat)) %>%
  # left_join(select(ipsc_mn_als_datasets$res, gene_id, panALS = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_id")
cormat <- round(cor(ipsc_mn_mutations_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
dge_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5))
# dge_correlation_ggheatmap

mutation_volcanos_corr_heatmap <- ipsc_mn_tardbp_datasets.volcano + ipsc_mn_fus_datasets.volcano + ipsc_mn_c9orf72_datasets.volcano + ipsc_mn_sod1_datasets.volcano + ipsc_mn_sporadic_datasets.volcano + dge_correlation_ggheatmap +
  plot_layout(ncol = 3, nrow = 2)
# mutation_volcanos_corr_heatmap

ipsc_mn_mutants_join_datasets.res = full_join(select(ipsc_mn_c9orf72_datasets$res, gene_name, gene_id, padj.c9orf72 = padj, log2FoldChange.c9orf72 = log2FoldChange, stat.c9orf72 = stat),
          select(ipsc_mn_sporadic_datasets$res, gene_name, gene_id, padj.sals = padj, log2FoldChange.sals = log2FoldChange, stat.sals = stat)) %>%
  full_join(select(ipsc_mn_fus_datasets$res, gene_name, gene_id, padj.fus = padj, log2FoldChange.fus = log2FoldChange, stat.fus = stat)) %>%
  full_join(select(ipsc_mn_tardbp_datasets$res, gene_name, gene_id, padj.tardbp = padj, log2FoldChange.tardbp = log2FoldChange, stat.tardbp = stat)) %>%
  full_join(select(ipsc_mn_sod1_datasets$res, gene_name, gene_id, padj.sod1 = padj, log2FoldChange.sod1 = log2FoldChange, stat.sod1 = stat)) #%>% full_join(select(ipsc_mn_vcp_datasets$res, gene_name, padj.vcp = padj, log2FoldChange.vcp = log2FoldChange))

ipsc_mn_mutants_join_datasets.res %>% filter(padj.sod1 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # UPK3BL1, NPIPA8
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.sod1 < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.sod1 < 0.05) # 

ipsc_mn_mutants_join_datasets.res %>% filter(stat.sod1 > 1, stat.sals > 1, stat.c9orf72 > 1, stat.fus > 1, stat.tardbp > 1) %>% filter(gene_name %in% c(ALS_genes,p53.signalling.pathway))
ipsc_mn_mutants_join_datasets.res %>% filter(stat.sod1 < -1, stat.sals < -1, stat.c9orf72 < -1, stat.fus < -1, stat.tardbp < -1) %>% filter(gene_name %in% c(ALS_genes,p53.signalling.pathway))


  padj.sod1 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.fus < 0.05) # UPK3BL1, NPIPA8
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.sod1 < 0.05, padj.fus < 0.05) # 
ipsc_mn_mutants_join_datasets.res %>% filter(padj.c9orf72 < 0.05, padj.tardbp < 0.05, padj.sod1 < 0.05) # 


# # Overlap ALL DEGs
# ipsc_mn_mutants_bind_datasets.dir.res = bind_rows(mutate(filter(ipsc_mn_c9orf72_datasets$res, log2FoldChange > 0), mutation = "C9orf72 Up"), mutate(filter(ipsc_mn_c9orf72_datasets$res, log2FoldChange < 0), mutation = "C9orf72 Down"),
#                                               mutate(filter(ipsc_mn_tardbp_datasets$res, log2FoldChange > 0), mutation = "TARDBP Up"), mutate(filter(ipsc_mn_tardbp_datasets$res, log2FoldChange < 0), mutation = "TARDBP Down"),
#                                               mutate(filter(ipsc_mn_fus_datasets$res, log2FoldChange > 0), mutation = "FUS Up"), mutate(filter(ipsc_mn_fus_datasets$res, log2FoldChange < 0), mutation = "FUS Down"),
#                                               mutate(filter(ipsc_mn_sod1_datasets$res, log2FoldChange > 0), mutation = "SOD1 Up"), mutate(filter(ipsc_mn_sod1_datasets$res, log2FoldChange < 0), mutation = "SOD1 Down"),
# ipsc_mn_mutants_bind_datasets.res.direction.upset = ipsc_mn_mutants_bind_datasets.dir.res %>% filter(padj < 0.05) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#   geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.4, hjust = -0.2, size = 2.5, angle = 45) +
#   scale_x_upset(order_by = "degree")  + # n_intersections = 20
#   scale_y_continuous(breaks = NULL, lim = c(0, 2700), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=6), axis.title=element_text(size=8)) +
#   ylab(expression()) +
#   theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.res.direction.upset
# 
# # Overlap UP DEGs
# ipsc_mn_mutants_bind_datasets.upregulated.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05, log2FoldChange > 0) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#       geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, size = 2.8) +
#     scale_x_upset(order_by = "degree")  + # n_intersections = 20
#     scale_y_continuous(breaks = NULL, lim = c(0, 2500), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + 
#       ylab(expression()) +
#     theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.upregulated.res.upset
# 
# # Overlap DOWN DEGs
# ipsc_mn_mutants_bind_datasets.downregulated.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05, log2FoldChange < 0) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
#   ggplot(aes(x = mutations)) + geom_bar() +
#       geom_text(stat='count', aes(label=after_stat(count)), vjust=-1, size = 2.8) +
#     scale_x_upset(order_by = "degree")  + # n_intersections = 20
#     scale_y_continuous(breaks = NULL, lim = c(0, 3000), name = "") +
#   theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + 
#       ylab(expression()) +
#     theme_combmatrix(combmatrix.label.make_space = TRUE)
# ipsc_mn_mutants_bind_datasets.downregulated.res.upset

```


## p53 PROGENy & TP53 Dorothea

```{r}
ipsc_mn_mutations_list = list("Sporadic" = select(ipsc_mn_sporadic_datasets$res,gene_name,stat), "C9orf72" = select(ipsc_mn_c9orf72_datasets$res,gene_name,stat), "TARDBP" = select(ipsc_mn_tardbp_datasets$res,gene_name,stat), "SOD1" = select(ipsc_mn_sod1_datasets$res,gene_name,stat), "FUS" = select(ipsc_mn_fus_datasets$res,gene_name,stat)) #, "VCP" = select(ipsc_mn_vcp_datasets$res,gene_name,stat)) 
mutations <- names(ipsc_mn_mutations_list)
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{drop_na(.x)}) 
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

# ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP","VCP")))) +
#   geom_bar(stat='identity', position = "dodge2") +
#   scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
#   theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
#   geom_hline(yintercept = 0, linetype = 3) +
#   labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment") 
# ipsc_mn_mutations_list.progeny_scores.plot

ipsc_mn_mutations_list.p53.progeny_scores = ipsc_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
ipsc_mn_mutations_list.p53.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.p53.progeny_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
  stat_pvalue_manual(ipsc_mn_mutations_list.p53.progeny_scores, label = "p.signif", y.position = 13, xmin = "mutation", xmax = NULL, size = 6, hide.ns = TRUE) 
ipsc_mn_mutations_list.p53.progeny_scores.plot



# Dorothea
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

ipsc_mn_mutations_list.TP53.dorothea_scores = ipsc_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
ipsc_mn_mutations_list.TP53.dorothea_scores.plot <- ggplot(ipsc_mn_mutations_list.TP53.dorothea_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") + #ylim(c(-11,11)) +
  stat_pvalue_manual(ipsc_mn_mutations_list.TP53.dorothea_scores, label = "p.signif", y.position = 11, xmin = "mutation", xmax = NULL, size = 6, hide.ns = TRUE) 
ipsc_mn_mutations_list.TP53.dorothea_scores.plot
```


## Merge

```{r}
mutations_compared_fig.merged = plot_grid(
  plot_grid(ipsc_mn_tardbp_datasets.volcano, ipsc_mn_fus_datasets.volcano, ipsc_mn_c9orf72_datasets.volcano, ipsc_mn_sod1_datasets.volcano, ncol = 4, labels = c("a","b","c","d")),
  plot_grid(ipsc_mn_sporadic_datasets.volcano, dge_correlation_ggheatmap, ipsc_mn_mutations_list.p53.progeny_scores.plot, ipsc_mn_mutations_list.TP53.dorothea_scores.plot, ncol = 4, rel_widths = c(1,1,1,1), labels = c("e","f","g","h")),
  nrow = 2)
# mutations_compared_fig.merged
ggsave(mutations_compared_fig.merged, filename = here(proj_path,"expression/deseq2/mutations_compared_fig.merged.png"), height = 7, width = 11)

```


# Fig S9 Gene expression overlap mutations 

## Scatters

Scatter correlation of mutants

```{r}
ipsc_mn_mutations_list = list("Sporadic" = ipsc_mn_sporadic_datasets$res, "C9orf72" = ipsc_mn_c9orf72_datasets$res, "TARDBP" = ipsc_mn_tardbp_datasets$res, "SOD1" = ipsc_mn_sod1_datasets$res, "FUS" = ipsc_mn_fus_datasets$res)#, "panALS" = ipsc_mn_als_datasets$res)
# mutations <- names(ipsc_mn_mutations_list)
# 
# ipsc_mn_mutations_list = purrr::map2(
#       .x = ipsc_mn_mutations_list,
#       .y = names(ipsc_mn_mutations_list), ~{
#         .x$mutation <- .y
#         return(.x)
#       })

ipsc_mn_mutations_list_t_stat_plot <- 
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", highlight = "C9orf72")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", highlight = "SOD1")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", highlight = "FUS")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", highlight = "TARDBP")  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", highlight = c("C9orf72","SOD1"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", highlight = c("C9orf72","FUS"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", highlight = c("C9orf72","TARDBP"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", highlight = c("SOD1","FUS"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", highlight = c("SOD1","TARDBP"))  +
plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", highlight = c("TARDBP","FUS"))  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "panALS", highlight = c("SOD1"))  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "panALS")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "panALS", highlight = c("C9orf72"))  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "panALS", highlight = c("FUS"))  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "TARDBP", mutation2 = "panALS", highlight = c("TARDBP")) +
  plot_layout(ncol = 4, nrow = 3) &  
  xlim(-8,8) & ylim(-8,8)
# ipsc_mn_mutations_list_t_stat_plot
# ggsave(ipsc_mn_mutations_list_t_stat_plot, filename = here(proj_path,"expression/deseq2/dge.mutation.stat.scatters.merged.png"), height = 10, width = 7)

# # log2FoldChange
# lfc_plot <- 
#   plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "VCP", col = "log2FoldChange")  +
# plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "TARDBP", mutation2 = "VCP", col = "log2FoldChange") &  
#   xlim(-8,8) & ylim(-8,8)  +
#   plot_layout(ncol = 3, nrow = 5)
# lfc_plot
# ggsave(lfc_plot, filename = here(proj_path,"expression/deseq2/dge.mutation.lfc.scatters.merged.png", height = 13, width = 10)

```

## Upset plot

```{r}
### Upset plot https://github.com/const-ae/ggupset
ipsc_mn_mutants_bind_datasets.res = bind_rows(mutate(ipsc_mn_sporadic_datasets$res, mutation = "Sporadic"), mutate(ipsc_mn_c9orf72_datasets$res, mutation = "C9orf72"), mutate(ipsc_mn_tardbp_datasets$res, mutation = "TARDBP"), mutate(ipsc_mn_sod1_datasets$res, mutation = "SOD1"), mutate(ipsc_mn_fus_datasets$res, mutation = "FUS"))# ,mutate(ipsc_mn_vcp_datasets$res, mutation = "VCP"))
ipsc_mn_mutants_bind_datasets.res.upset = ipsc_mn_mutants_bind_datasets.res %>% filter(padj < 0.05) %>% select(mutation, gene_name) %>% group_by(gene_name) %>% summarise(mutations = list(mutation)) %>%
  ggplot(aes(x = mutations)) + geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.2, hjust = 0.5, size = 3, angle = 0) +
    scale_x_upset(order_by = "degree")  + # n_intersections = 20
    scale_y_continuous(breaks = NULL, lim = c(0, 4000), name = "") +
    theme_classic() +  theme(legend.title = element_blank(), legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) + labs(x = "", y = "") +
    theme_combmatrix(combmatrix.label.make_space = TRUE)
ipsc_mn_mutants_bind_datasets.res.upset


```

## GO terms

```{r}
# c9orf72
ipsc_mn_c9orf72_datasets.dge_genes <- ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt <- ipsc_mn_c9orf72_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " C9orf72 Down ", query == "query_2" ~ " C9orf72 Up "), term_name = as.factor(term_name))
ipsc_mn_c9orf72_datasets.dge_gprofiles_down <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ")
ipsc_mn_c9orf72_datasets.dge_gprofiles_up <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")
paste('"',unique(ipsc_mn_c9orf72_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "hedgehog family protein binding",
"microtubule bundle formation",
"cytoplasmic region",
"cytoskeleton organization",
"axoneme assembly",
"cilium assembly")
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_down <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_c9orf72_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest",
"miRNA regulation of p53 pathway in prostate cancer",
# "miR-517 relationship with ARCN1 and USP1",
# "retrograde vesicle-mediated transport, Golgi to endoplasmic reticulum",
# "phospholipid transporter activity",
"Genotoxicity pathway",
"p53 signaling pathway")
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_up <- ipsc_mn_c9orf72_datasets.dge_gprofiles_filt %>% filter(query == " C9orf72 Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_down, ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" C9orf72 Up ", " C9orf72 Down ")))
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name <- gsub("DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest", "DNA damage response by p53", ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" pathway in prostate cancer", "", ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_c9orf72_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_c9orf72_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
# ipsc_mn_c9orf72_datasets.go_curated

# fus
ipsc_mn_fus_datasets.dge_genes <- ipsc_mn_fus_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_fus_datasets.dge_gprofiles_filt <- ipsc_mn_fus_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " FUS Down ", query == "query_2" ~ " FUS Up "), term_name = as.factor(term_name))
ipsc_mn_fus_datasets.dge_gprofiles_down <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Down ")
ipsc_mn_fus_datasets.dge_gprofiles_up <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Up ")
paste('"',unique(ipsc_mn_fus_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"positive regulation of oxidative phosphorylation",
"BAT3 complex binding",
"cell-cell adhesion via plasma-membrane adhesion molecules",
# "Babinski sign",
# "Abnormal superficial reflex",
# "Abnormality of salivation",
"synaptic signaling",
# "trans-synaptic signaling",
"chemical synaptic transmission")
ipsc_mn_fus_datasets.dge_gprofiles_filt_down <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_fus_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"regulation of RNA metabolic process",
# "DNA binding",
"transcription regulator activity",
# "DNA-binding transcription factor activity, RNA polymerase II-specific",
# "DNA-binding transcription factor activity",
"transcription by RNA polymerase II",
"double-stranded DNA binding",
"transcription cis-regulatory region binding")
ipsc_mn_fus_datasets.dge_gprofiles_filt_up <- ipsc_mn_fus_datasets.dge_gprofiles_filt %>% filter(query == " FUS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_fus_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_fus_datasets.dge_gprofiles_filt_down, ipsc_mn_fus_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" FUS Up ", " FUS Down ")))
ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name <- gsub("transcription cis-regulatory region binding", "transcription regulatory binding", ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name) %>% gsub("positive regulation of ","",.)  %>% gsub("regulation of ","",.) %>% gsub("cell-cell adhesion via plasma-membrane adhesion molecules", "cell membrane adhesion",.)
# ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" pathway", "", ipsc_mn_fus_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_fus_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_fus_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
ipsc_mn_fus_datasets.go_curated

# tardbp
ipsc_mn_tardbp_datasets.dge_genes <- ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_tardbp_datasets.dge_gprofiles_filt <- ipsc_mn_tardbp_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
  mutate(query = case_when(query == "query_1" ~ " TARDBP Down ", query == "query_2" ~ " TARDBP Up "), term_name = as.factor(term_name))
ipsc_mn_tardbp_datasets.dge_gprofiles_down <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Down ")
ipsc_mn_tardbp_datasets.dge_gprofiles_up <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Up ")
paste('"',unique(ipsc_mn_tardbp_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"spliceosomal complex",
"mRNA Splicing",
# "DNA IR-damage and cellular response via ATR",
"regulation of RNA metabolic process",
"nervous system development",
# "DNA metabolic process",
# "microtubule cytoskeleton organization",
"microtubule cytoskeleton",
# "chromosome segregation",
"cell division",
# "mitotic cell cycle",
"protein binding")
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_down <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Down ") %>% filter(term_name %in% down_list)
paste('"',unique(ipsc_mn_tardbp_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
# "neuron differentiation",
"glutamatergic synapse",
# "Neuronal System",
# "neurogenesis",
"synaptic signaling",
# "dendritic tree",
"dendrite",
"axon",
"neuron projection",
"synapse")
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_up <- ipsc_mn_tardbp_datasets.dge_gprofiles_filt %>% filter(query == " TARDBP Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined <- bind_rows(ipsc_mn_tardbp_datasets.dge_gprofiles_filt_down, ipsc_mn_tardbp_datasets.dge_gprofiles_filt_up) %>%
  mutate(query = factor(query, levels = c(" TARDBP Up ", " TARDBP Down ")))
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub("regulation of RNA metabolic process", "RNA metabolic process", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub(" and cellular response via ATR", "", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name <- gsub("cellular response to DNA", "DNA", ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined$term_name)
ipsc_mn_tardbp_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_tardbp_datasets.dge_gprofiles_filt_combined) + theme(legend.position = "none")
# ipsc_mn_tardbp_datasets.go_curated

# # sod1
# ipsc_mn_sod1_datasets.dge_genes <- ipsc_mn_sod1_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_sod1_datasets.dge_gprofiles_filt <- ipsc_mn_sod1_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in sod1 ", query == "query_2" ~ " Up in sod1 "), term_name = as.factor(term_name))
# ipsc_mn_sod1_datasets.dge_gprofiles_down <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Down in sod1 ")
# ipsc_mn_sod1_datasets.dge_gprofiles_up <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Up in sod1 ")
# paste('"',unique(ipsc_mn_sod1_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# ipsc_mn_sod1_datasets.dge_gprofiles_filt_down <- ipsc_mn_sod1_datasets.dge_gprofiles_filt %>% filter(query == " Down in sod1 ") %>% filter(term_name %in% down_list)
# paste('"',unique(ipsc_mn_sod1_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# #no terms for SOD1
# 
# # vcp
# ipsc_mn_vcp_datasets.dge_genes <- ipsc_mn_vcp_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_vcp_datasets.dge_gprofiles_filt <- ipsc_mn_vcp_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA|WP")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in vcp ", query == "query_2" ~ " Up in vcp "), term_name = as.factor(term_name))
# ipsc_mn_vcp_datasets.dge_gprofiles_down <- ipsc_mn_vcp_datasets.dge_gprofiles_filt %>% filter(query == " Down in vcp ")
# ipsc_mn_vcp_datasets.dge_gprofiles_up <- ipsc_mn_vcp_datasets.dge_gprofiles_filt %>% filter(query == " Up in vcp ")
# paste('"',unique(ipsc_mn_vcp_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# paste('"',unique(ipsc_mn_vcp_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# # no terms for VCP
# 
# # sporadic
# ipsc_mn_sporadic_datasets.dge_genes <- ipsc_mn_sporadic_datasets$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
# ipsc_mn_sporadic_datasets.dge_gprofiles_filt <- ipsc_mn_sporadic_datasets.dge_genes$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange(-log10(p_value)) %>%
#   mutate(query = case_when(query == "query_1" ~ " Down in sporadic ", query == "query_2" ~ " Up in sporadic "), term_name = as.factor(term_name))
# ipsc_mn_sporadic_datasets.dge_gprofiles_down <- ipsc_mn_sporadic_datasets.dge_gprofiles_filt %>% filter(query == " Down in sporadic ")
# ipsc_mn_sporadic_datasets.dge_gprofiles_up <- ipsc_mn_sporadic_datasets.dge_gprofiles_filt %>% filter(query == " Up in sporadic ")
# paste('"',unique(ipsc_mn_sporadic_datasets.dge_gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
# paste('"',unique(ipsc_mn_sporadic_datasets.dge_gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
# # none or sporadic

ipsc_mn_c9orf72_fus_tardbp.go_curated.merged =  + ipsc_mn_fus_datasets.go_curated + ipsc_mn_tardbp_datasets.go_curated
ipsc_mn_c9orf72_fus_tardbp.go_curated.merged


ipsc_mn_tardbp_datasets$res %>% filter(padj < 0.05, gene_name %in% wikipathways.msigdb$WP_DNA_IRDAMAGE_AND_CELLULAR_RESPONSE_VIA_ATR) # OLIG1, OLIG2

# # fus
# ipsc_mn_fus_datasets.rrvgo = gprofiler2rrvgo(ipsc_mn_fus_datasets$res , query1 = "Down", query2 = "Up", GO_cat = "BP", sig.filter = "padj", species = "human")
# ipsc_mn_fus_datasets.rrvgo =  ipsc_mn_fus_datasets.rrvgo %>%  mutate(term = gsub("regulation of ", "", term), term = gsub("homophilic cell adhesion via plasma membrane adhesion molecules", "plasma membrane adhesion", term), parentTerm = term)
# ipsc_mn_fus_datasets.rrvgo.plot = rrvgo_plot(gprofiler2rrvgo = ipsc_mn_fus_datasets.rrvgo, n_terms = 10, remove_terms = NULL, colours = 2, cols =  c("firebrick2", "dodgerblue2"), label_angle = 0)
# 
# # tardbp
# ipsc_mn_tardbp_datasets.rrvgo = gprofiler2rrvgo(ipsc_mn_tardbp_datasets$res , query1 = "Down", query2 = "Up", GO_cat = "BP", sig.filter = "padj", species = "human")
# ipsc_mn_tardbp_datasets.rrvgo =  ipsc_mn_tardbp_datasets.rrvgo %>%  mutate(term = gsub("regulation of ", "", term), parentTerm = term)
# ipsc_mn_tardbp_datasets.rrvgo.plot = rrvgo_plot(gprofiler2rrvgo = ipsc_mn_tardbp_datasets.rrvgo, n_terms = 10, remove_terms = NULL, colours = 2, cols =  c("firebrick2", "dodgerblue2"), label_angle = 0)

# de_go_terms_mutations_merged = ggarrange(NULL,ipsc_mn_c9orf72_datasets.rrvgo.plot,NULL, ipsc_mn_sod1_datasets.rrvgo.plot,NULL, ipsc_mn_fus_datasets.rrvgo.plot,NULL, ipsc_mn_tardbp_datasets.rrvgo.plot, nrow = 8, ncol = 1, labels = c("C9orf72", "", "SOD1", "", "FUS", "", "TARDBP",""), heights = c(0.1,0.8,0.1,1.3,0.1,0.9,0.1,1.3)) #, ipsc_mn_vcp_datasets.rrvgo.plot) # no terms for VCP
# de_go_terms_mutations_merged
# ggsave(de_go_terms_mutations_merged, filename = here(proj_path,"expression/deseq2/figS5.de_go_terms_mutations_merged.png", height = 11.75, width = 8.25)

# which genes are driving the terms
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GABA_RECEPTOR_COMPLEX)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_PLASMA_MEMBRANE_REGION)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_GAP_JUNCTION)
ipsc_mn_c9orf72_datasets$res %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT_CONFERRING_COMPRESSION_RESISTANCE)
```


## Merge

```{r}
genetic_subgroups_scatter_upset = plot_grid(
  plot_grid(plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "C9orf72", highlight = "C9orf72"),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "SOD1", highlight = "SOD1"),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "FUS", highlight = "FUS"),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "Sporadic", mutation2 = "TARDBP", highlight = "TARDBP"), ncol = 4, labels = c("a","","","")),
  plot_grid(plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "SOD1", highlight = c("C9orf72","SOD1")),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "FUS", highlight = c("C9orf72","FUS")),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "C9orf72", mutation2 = "TARDBP", highlight = c("C9orf72","TARDBP")),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "FUS", highlight = c("SOD1","FUS")), ncol = 4),
  plot_grid(plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "SOD1", mutation2 = "TARDBP", highlight = c("SOD1","TARDBP")),
  plot_de_correlation(model_list1 = ipsc_mn_mutations_list, model_list2 = ipsc_mn_mutations_list, mutation1 = "FUS", mutation2 = "TARDBP", highlight = c("TARDBP","FUS")), 
  ipsc_mn_mutants_bind_datasets.res.upset, ncol = 3, rel_widths = c(1,1,2), labels = c("","","b")), 
  plot_grid(ipsc_mn_c9orf72_datasets.go_curated, ipsc_mn_fus_datasets.go_curated, ipsc_mn_tardbp_datasets.go_curated, ncol=3, labels = c("c","d","e")), nrow = 4)
# genetic_subgroups_compared.merged
ggsave(genetic_subgroups_scatter_upset, filename = here(proj_path,"expression/deseq2/genetic_subgroups_scatter_upset.png"), height = 9.5, width = 8.25)

```


# Fig S10 DoRoTHEA & PROGENy

<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md
Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
ipsc_mn_mutations_list = list("Sporadic" = select(ipsc_mn_sporadic_datasets$res,gene_name,stat), "C9orf72" = select(ipsc_mn_c9orf72_datasets$res,gene_name,stat), "TARDBP" = select(ipsc_mn_tardbp_datasets$res,gene_name,stat), "SOD1" = select(ipsc_mn_sod1_datasets$res,gene_name,stat), "FUS" = select(ipsc_mn_fus_datasets$res,gene_name,stat))#, "VCP" = select(ipsc_mn_vcp_datasets$res,gene_name,stat)) 
mutations <- names(ipsc_mn_mutations_list)
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{drop_na(.x)}) 
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_mutations_list = map(ipsc_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_mutations_list.progeny_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation, source = as.factor(source), source = fct_reorder(source, score))

ipsc_mn_mutations_list.progeny_scores.plot <- ggplot(ipsc_mn_mutations_list.progeny_scores, aes(x=mutation, y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment") +
  theme_oz() + theme(axis.text.x = element_text(angle = 90, size = 6), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.position = "top") +
  geom_hline(yintercept = 0, linetype = 3) +
  facet_grid(~source) +
  stat_pvalue_manual(ipsc_mn_mutations_list.progeny_scores, label = "p.signif", y.position = 12, xmin = "mutation", xmax = NULL, size = 4, hide.ns = TRUE)
ipsc_mn_mutations_list.progeny_scores.plot

```

Dorothea

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_mutations_list.dorothea_scores = map_df(ipsc_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation) %>% 
  mutate(sig = case_when(p_value < 0.1 & abs(score) < 4 ~ "sig", p_value < 0.1 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.1 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

# Volcano of TFs score vs p_value
ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_mutations_list.dorothea_scores, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
    labs(title = "Transcription Factors", y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
    geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "C9orf72"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "FUS"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "SOD1"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "Sporadic"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_text_repel(fontface = "italic", data = top_n(filter(ipsc_mn_mutations_list.dorothea_scores, p_value < 0.05, mutation == "TARDBP"),n = 3, wt = abs(score)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
    geom_text_repel(fontface = "italic", data = filter(ipsc_mn_mutations_list.dorothea_scores,source %in% c("TP53","ZNF274","GATA3","MAZ","TAL1","TEAD4")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  facet_wrap(~mutation, scales = "free") & ylim(0,2.8)
ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano

# ,"E2F4","EOMES","MAZ","MITF","SOX2","TFDP1","ZBTB33","ZNF263"

ipsc_mn_mutations_list.dorothea_scores %>% filter(source == "TP53")
# mutation statistic  source condition  score p_value p.signif group1   group2   sig        direction class        
#   <chr>    <chr>      <chr>  <chr>      <dbl>   <dbl> <chr>    <chr>    <chr>    <chr>      <chr>     <chr>        
# 1 Sporadic norm_wmean TP53   V1        -0.313   0.254 ""       Sporadic Sporadic non_sig    down      non_sig down 
# 2 C9orf72  norm_wmean TP53   V1         9.27    0.002 "*"      C9orf72  C9orf72  sig_strong up        sig_strong up
# 3 TARDBP   norm_wmean TP53   V1         2.45    0.014 "*"      TARDBP   TARDBP   sig        up        sig up       
# 4 SOD1     norm_wmean TP53   V1        -2.97    0.002 "*"      SOD1     SOD1     sig        down      sig down     
# 5 FUS      norm_wmean TP53   V1         4.02    0.004 "*"      FUS      FUS      sig_strong up        sig_strong up

# commonly changed TFs
# increased
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score > 0) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 ZNF274     4
# decreased
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, score < 0) %>% count(source) %>% arrange(-n)
# source     n
#    <chr>  <int>
#  1 GATA3      4
#  2 MAZ        4
#  3 TAL1       4
#  4 TEAD4      4

ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "MAZ")
ipsc_mn_mutations_list.dorothea_scores %>% filter(p_value < 0.05, source == "TEAD4")

```

Merge

```{r}
ipsc_mn_mutations_list.dorothea_progeny_merged = ggarrange(ipsc_mn_mutations_list.progeny_scores.plot, ipsc_mn_mutations_merged_dorothea.contrast_acts.volcano, nrow = 2, labels = c("a","b"), heights = c(1,1.7))
# ipsc_mn_mutations_list.dorothea_progeny_merged
ggsave(ipsc_mn_mutations_list.dorothea_progeny_merged, filename = here(proj_path, "expression/deseq2/ipsc_mn_mutations_list.dorothea_progeny_merged.png"), height = 10, width = 9)

```



# Table S5 DEGs overlap between genetic backgrounds

```{r}
ipsc_mn_mutants_join_datasets.res = 
  mutate(select(ipsc_mn_sporadic_datasets$res, gene_name, gene_id, padj.sporadic = padj, stat.sporadic = stat, lfc.sporadic = log2FoldChange), direction.sporadic = case_when(padj.sporadic < 0.05 & lfc.sporadic > 0 ~ "up", padj.sporadic < 0.05 & lfc.sporadic < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(ipsc_mn_c9orf72_datasets$res, gene_name, gene_id, padj.c9orf72 = padj, stat.c9orf72 = stat, lfc.c9orf72 = log2FoldChange), direction.c9orf72 = case_when(padj.c9orf72 < 0.05 & lfc.c9orf72 > 0 ~ "up", padj.c9orf72 < 0.05 & lfc.c9orf72 < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_fus_datasets$res, gene_name, gene_id, padj.fus = padj, stat.fus = stat, lfc.fus = log2FoldChange), direction.fus = case_when(padj.fus < 0.05 & lfc.fus > 0 ~ "up", padj.fus < 0.05 & lfc.fus < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_tardbp_datasets$res, gene_name, gene_id,  padj.tardbp = padj, stat.tardbp = stat, lfc.tardbp = log2FoldChange), direction.tardbp = case_when(padj.tardbp < 0.05 & lfc.tardbp > 0 ~ "up", padj.tardbp < 0.05 & lfc.tardbp < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  full_join(mutate(select(ipsc_mn_sod1_datasets$res, gene_name, gene_id, padj.sod1 = padj, stat.sod1 = stat, lfc.sod1 = log2FoldChange), direction.sod1 = case_when(padj.sod1 < 0.05 & lfc.sod1 > 0 ~ "up", padj.sod1 < 0.05 & lfc.sod1 < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  # arrange() %>%
  select(gene_name, gene_id, everything()) %>%
  filter(padj.sporadic < 0.05 | padj.c9orf72 < 0.05 | padj.fus < 0.05 | padj.tardbp < 0.05 | padj.sod1 < 0.05)
ipsc_mn_mutants_join_datasets.res
write_csv(ipsc_mn_mutants_join_datasets.res, here(proj_path,"expression/deseq2/TableS6.overlap_als_individual_datasets.csv"))

```


# Fig S11 Postmortem 

Import NYGC counts from Humphrey et al Zenodo https://zenodo.org/record/6385747#.Yod3Y5PMJqs

## Volcano

```{r}
#### Spinal cord only ###

# metadata
nygc_cervical_cord_metadata = read_tsv(here(shared_path,"nygc-als-consortium/counts/spinal_cord_jack_humphrey/Cervical_Spinal_Cord/Cervical_Spinal_Cord_metadata.tsv")) # 174
nygc_thoracic_cord_metadata = read_tsv(here(shared_path,"nygc-als-consortium/counts/spinal_cord_jack_humphrey/Thoracic_Spinal_Cord/Thoracic_Spinal_Cord_metadata.tsv")) # 52
nygc_lumbar_cord_metadata = read_tsv(here(shared_path,"nygc-als-consortium/counts/spinal_cord_jack_humphrey/Lumbar_Spinal_Cord/Lumbar_Spinal_Cord_metadata.tsv")) # 154
nygc_spinal_cord_metadata.bind = bind_rows(nygc_cervical_cord_metadata, nygc_thoracic_cord_metadata, nygc_lumbar_cord_metadata) %>% # 380
  mutate(condition = case_when(disease == "Control" ~ "ctrl", grepl("ALS",disease) ~ "als"), condition = factor(condition, levels = c("ctrl","als")), library_prep = as.factor(library_prep),
         gender = tolower(sex), mutation = tolower(mutations), mutation = case_when(condition == "als" & mutation == "none" ~ "sporadic", TRUE ~ mutation))
nygc_spinal_cord_metadata.bind %>% count(condition)
nygc_spinal_cord_metadata = nygc_spinal_cord_metadata.bind %>% mutate(tissue = factor(tissue, levels = c("Cervical_Spinal_Cord", "Thoracic_Spinal_Cord", "Lumbar_Spinal_Cord")), sample = rna_id) %>% arrange(tissue) %>% distinct(dna_id, .keep_all = TRUE) %>% column_to_rownames(var = "rna_id") # 203 / 380
nygc_spinal_cord_metadata.sporadic = nygc_spinal_cord_metadata %>% filter(mutation %in% c("sporadic","none"))
nygc_spinal_cord_metadata.c9orf72 = nygc_spinal_cord_metadata %>% filter(mutation %in% c("c9orf72","none"))
nygc_spinal_cord_metadata.sod1 = nygc_spinal_cord_metadata %>% filter(mutation %in% c("sod1","none"))
nygc_spinal_cord_metadata.fus = nygc_spinal_cord_metadata %>% filter(mutation %in% c("fus","none"))

nygc_spinal_cord_metadata %>% count(condition)
nygc_spinal_cord_metadata.sporadic %>% count(condition)
nygc_spinal_cord_metadata.c9orf72 %>% count(condition)
nygc_spinal_cord_metadata.sod1 %>% count(condition)
nygc_spinal_cord_metadata.fus %>% count(condition)

nygc_postmortem_spinal_cord.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.tissue.rds"))
nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) # 14,529
nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) #6417 up, 8112 down
nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(gene_name %in% c("TARDBP","FUS","SFPQ"))

## Volcano
nygc_spinal_cord_metadata
nygc_postmortem_spinal_cord.als_vs_ctrl.als.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(gene_name %in% ALS_genes) %>% top_n(n=5,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.p53.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(gene_name %in% p53.DDR.repair_gene_set) %>% top_n(n=5,wt = abs(stat)) %>% pull(gene_name)
ipsc_mn_als_datasets.res.dge.up <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.res.dge.down <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.ipsn.up.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj<0.05,stat>0,gene_name %in% ipsc_mn_als_datasets.res.dge.up) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.ipsn.down.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj<0.05,stat<0,gene_name %in% ipsc_mn_als_datasets.res.dge.down) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.labels <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(n=10,wt = abs(stat)) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.volcano <- volcano_plot(nygc_postmortem_spinal_cord.als_vs_ctrl$res, numerator = "ALS", denomenator = "CTRL",
               title = "Postmortem spinal cord", subtitle = "80 Controls vs 300 ALS", ymax = 60, xmax = 3, dpi = 300,
               gene_labels = c(nygc_postmortem_spinal_cord.als_vs_ctrl.als.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.ipsn.up.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.ipsn.down.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.labels, nygc_postmortem_spinal_cord.als_vs_ctrl.p53.labels))
# nygc_postmortem_spinal_cord.als_vs_ctrl.volcano

```

## GO

```{r}
nygc_postmortem_spinal_cord.als_vs_ctrl.gost <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles <- nygc_postmortem_spinal_cord.als_vs_ctrl.gost$result %>% filter(!str_detect(source, "TF|CORUM|HPA|MIRNA")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_down <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Down ")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_up <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Up ")
paste('"',unique(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "dendrite",
"transcription by RNA polymerase II",
"regulation of RNA metabolic process",
"cytoskeleton organization",
"axon",
"neuron development",
"synapse",
"neuron projection",
# "generation of neurons",
"neurogenesis")
# "nervous system development")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_down <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
paste('"',unique(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"cellular response to DNA damage stimulus",
"programmed cell death",
# "cell death",
"endoplasmic reticulum",
# "mitochondrial matrix",
"lysosome",
"protein localization",
# "mitochondrial envelope",
# "protein modification process",
"translation",
"response to stress",
# "peptide metabolic process",
# "catabolic process",
"mitochondrion",
# "extracellular vesicle",
"protein metabolic process",
"programmed cell death")
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_up <- nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined <- bind_rows(nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_down, nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined = nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined %>% mutate(term_name = gsub("regulation of ", "", term_name), term_name = gsub("and","&",term_name), term_name = gsub("cellular ","",term_name))
nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated <- curated_profiles(gprofiles = nygc_postmortem_spinal_cord.als_vs_ctrl.gprofiles_filt_combined)# + theme_oz() #theme(legend.position = "none", axis.text = element_text(size=10))
nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated

```


## PROGENY & Dorothea

```{r}
# nygc_postmortem_spinal_cord.als_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.tissue.rds"))

# PROGENy & DOROTHEA
net_progeny <- get_progeny(organism = 'human', top = 100)
nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts <- run_wmean(mat=nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity <- ggplot(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment ALS vs CTRL") +
    stat_pvalue_manual(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.contrast_acts, label = "p.signif", y.position = 6, xmin = "source", xmax = NULL, size = 4, hide.ns = TRUE) 
nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts <- run_wmean(mat=nygc_postmortem_spinal_cord.als_vs_ctrl.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 10000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(sig = case_when(p_value < 0.06 & abs(score) < 4 ~ "sig", p_value < 0.06 & abs(score) >= 4 ~ "sig_strong", p_value > 0.05 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.up = nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts %>% top_n(score, n = 3) %>% pull(source)
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.down = nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts %>% top_n(-score, n = 3) %>% pull(source)

# Volcano of TFs score vs p_value
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano = ggplot(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts, source %in% c(nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.up, nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted')
nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano

nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts %>% filter(source == "TP53")

```

## Scatter postmortem against iPSN

```{r}
# correlate to meta
postmortem_ipsn.als_vs_ctrl = select(nygc_postmortem_spinal_cord.als_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_als_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.als_vs_ctrl.overlapping = postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name) # 50
postmortem_ipsn_de_list = list("Postmortem spinal cord: ALS vs CTRL" = nygc_postmortem_spinal_cord.als_vs_ctrl$res, "iPSN: ALS vs CTRL" = ipsc_mn_als_datasets$res)
postmortem_ipsn.als_vs_ctrl.scatter = plot_de_correlation(model_list1 = postmortem_ipsn_de_list, model_list2 = postmortem_ipsn_de_list, mutation1 = "iPSN: ALS vs CTRL", mutation2 = "Postmortem spinal cord: ALS vs CTRL", highlight = c(postmortem_ipsn.als_vs_ctrl.overlapping, "CFAP410"))
# postmortem_ipsn.als_vs_ctrl.scatter

postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat))
postmortem_ipsn.als_vs_ctrl.overlapping # 
postmortem_ipsn.als_vs_ctrl.overlapping[postmortem_ipsn.als_vs_ctrl.overlapping %in% ALS_genes] # 0
postmortem_ipsn.als_vs_ctrl.overlapping[postmortem_ipsn.als_vs_ctrl.overlapping %in% RNA_BINDING_PROTEINS] # "RNASEL" 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% RNA_BINDING_PROTEINS) 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% gprofiler_database$`DNA Damage Response`) 
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0), gene_name %in% RNA_BINDING_PROTEINS  ) %>% arrange(-nygc.stat + ipsn.stat)
postmortem_ipsn.als_vs_ctrl %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% group_split(nygc.stat > 0)


nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.up <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.down <- nygc_postmortem_spinal_cord.als_vs_ctrl$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
ipsc_mn_als_datasets.res.dge.up <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(gene_name)
ipsc_mn_als_datasets.res.dge.down <- ipsc_mn_als_datasets$res %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(gene_name)
go.obj <- newGeneOverlap(c(ipsc_mn_als_datasets.res.dge.up,ipsc_mn_als_datasets.res.dge.down),
                         c(nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.up,nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.down),
                         genome.size=nrow(nygc_postmortem_spinal_cord.als_vs_ctrl$res))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=43
# listB size=14501
# Intersection size=22
# Overlapping p-value=1.6e-04
# Jaccard Index=0.0

# Fisher exact overlap test
go.obj <- newGeneOverlap(ipsc_mn_als_datasets.res.dge.up,
                         nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.up,
                         genome.size=nrow(nygc_postmortem_spinal_cord.als_vs_ctrl$res))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=20
# listB size=6411
# Intersection size=7
# Overlapping p-value=3.9e-03
# Jaccard Index=0.0
go.obj <- newGeneOverlap(ipsc_mn_als_datasets.res.dge.down,
                         nygc_postmortem_spinal_cord.als_vs_ctrl.res.dge.down,
                         genome.size=nrow(nygc_postmortem_spinal_cord.als_vs_ctrl$res))
testGeneOverlap(go.obj)
# GeneOverlap object:
# listA size=23
# listB size=8092
# Intersection size=9
# Overlapping p-value=2.3e-03
# Jaccard Index=0.0

# hypergeometric: .  Sorry to be pedantic, but an overlap of 15 feels small given the prevalence of differential genes post-mortem (14529) against iPSN (37), and the discrepancy between these two vastly different rates might draw attention to this 
phyper(22-1, 14501, nrow(nygc_postmortem_spinal_cord.als_vs_ctrl$res)-14501, 43, lower.tail=FALSE)
# 0.000158402
```


## Compare mutations postmortem

```{r}
nygc_postmortem_spinal_cord.c9orf72_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.c9orf72_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.fus_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.fus_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sod1_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sod1_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.sporadic_vs_ctrl = readRDS(here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.sporadic_vs_ctrl.rds"))

nygc_postmortem_spinal_cord_mn_mutations_list = list("Sporadic" = select(nygc_postmortem_spinal_cord.sporadic_vs_ctrl$res,gene_name,stat), "C9orf72" = select(nygc_postmortem_spinal_cord.c9orf72_vs_ctrl$res,gene_name,stat), "SOD1" = select(nygc_postmortem_spinal_cord.sod1_vs_ctrl$res,gene_name,stat), "FUS" = select(nygc_postmortem_spinal_cord.fus_vs_ctrl$res,gene_name,stat)) #"TARDBP" = select(nygc_postmortem_spinal_cord.tardbp_vs_ctrl$res,gene_name,stat), 
mutations <- names(nygc_postmortem_spinal_cord_mn_mutations_list)
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{drop_na(.x)}) 
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
nygc_postmortem_spinal_cord_mn_mutations_list = map(nygc_postmortem_spinal_cord_mn_mutations_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores = map_df(nygc_postmortem_spinal_cord_mn_mutations_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores, aes(x=reorder(source, score), y = score, fill = factor(mutation, levels = c("Sporadic","C9orf72","FUS","SOD1","TARDBP")))) +
  geom_bar(stat='identity', position = "dodge2") +
  scale_fill_npg() +    # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) +
  theme_oz() + theme(axis.text.x = element_text(angle = 90), plot.title = element_text(hjust = 0.5), legend.title = element_blank()) +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "Signaling Pathways", x = "", y = "Normalised enrichment")
nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores.plot

nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores = nygc_postmortem_spinal_cord_mn_mutations_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
  theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #ylim(c(-12,12)) +
  stat_pvalue_manual(nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores, label = "p.signif", y.position = 6, xmin = "mutation", xmax = NULL, size = 6, hide.ns = TRUE) 
nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot


# Dorothea
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
nygc_postmortem_spinal_cord_mn_mutations_list.dorothea_scores = map_df(nygc_postmortem_spinal_cord_mn_mutations_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "mutation") %>% filter(statistic == 'norm_wmean')

nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores = nygc_postmortem_spinal_cord_mn_mutations_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = mutation, group2 = mutation)
nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot <- ggplot(nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores, aes(x=reorder(mutation, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 0), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") + #ylim(c(-6,6)) +
  stat_pvalue_manual(nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores, label = "p.signif", y.position = 5, xmin = "mutation", xmax = NULL, size =6, hide.ns = TRUE) 
nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot

```

## Scatter postmortem & iPSN subgroups

```{r}
postmortem_ipsc_mn_mutations_list = list("Sporadic iPSN" = ipsc_mn_sporadic_datasets$res, "C9orf72 iPSN" = ipsc_mn_c9orf72_datasets$res, "SOD1 iPSN" = ipsc_mn_sod1_datasets$res, "FUS iPSN" = ipsc_mn_fus_datasets$res, "Sporadic postmortem" = nygc_postmortem_spinal_cord.sporadic_vs_ctrl$res, "C9orf72 postmortem" = nygc_postmortem_spinal_cord.c9orf72_vs_ctrl$res, "SOD1 postmortem" = nygc_postmortem_spinal_cord.sod1_vs_ctrl$res, "FUS postmortem" = nygc_postmortem_spinal_cord.fus_vs_ctrl$res)

postmortem_ipsn.sporadic.join = select(nygc_postmortem_spinal_cord.sporadic_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_sporadic_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.sporadic.overlapping = postmortem_ipsn.sporadic.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name)
postmortem_ipsn.c9orf72.join = select(nygc_postmortem_spinal_cord.c9orf72_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_c9orf72_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.c9orf72.overlapping = postmortem_ipsn.c9orf72.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name)
postmortem_ipsn.sod1.join = select(nygc_postmortem_spinal_cord.sod1_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_sod1_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.sod1.overlapping = postmortem_ipsn.sod1.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name)
postmortem_ipsn.fus.join = select(nygc_postmortem_spinal_cord.fus_vs_ctrl$res, gene_name, gene_id, nygc.stat = stat, nygc.padj = padj) %>% left_join(select(ipsc_mn_fus_datasets$res, gene_name, gene_id, ipsn.stat = stat, ipsn.padj = padj))
postmortem_ipsn.fus.overlapping = postmortem_ipsn.fus.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_name)

postmortem_ipsc_mn_mutations_list_t_stat_plot <- plot_grid(
plot_de_correlation(model_list1 = postmortem_ipsc_mn_mutations_list, model_list2 = postmortem_ipsc_mn_mutations_list, mutation1 = "Sporadic iPSN", mutation2 = "Sporadic postmortem", highlight = postmortem_ipsn.sporadic.overlapping),
plot_de_correlation(model_list1 = postmortem_ipsc_mn_mutations_list, model_list2 = postmortem_ipsc_mn_mutations_list, mutation1 = "C9orf72 iPSN", mutation2 = "C9orf72 postmortem", highlight = c(postmortem_ipsn.c9orf72.overlapping, "C9orf72")),
plot_de_correlation(model_list1 = postmortem_ipsc_mn_mutations_list, model_list2 = postmortem_ipsc_mn_mutations_list, mutation1 = "SOD1 iPSN", mutation2 = "SOD1 postmortem", highlight = c(postmortem_ipsn.sod1.overlapping, "SOD1")),
plot_de_correlation(model_list1 = postmortem_ipsc_mn_mutations_list, model_list2 = postmortem_ipsc_mn_mutations_list, mutation1 = "FUS iPSN", mutation2 = "FUS postmortem", highlight = c(postmortem_ipsn.fus.overlapping, "FUS")),
  ncol = 4, nrow = 1)# & xlim(-8,8) & ylim(-8,8)
postmortem_ipsc_mn_mutations_list_t_stat_plot
# ggsave(postmortem_ipsc_mn_mutations_list_t_stat_plot, filename = here(proj_path,"expression/deseq2/postmortem_ipsc_mn_mutations_list_t_stat_plot.png"), height = 8, width = 8)

postmortem_ipsn.sporadic.overlapping[postmortem_ipsn.sporadic.overlapping %in% p53.signalling.pathway]
postmortem_ipsn.c9orf72.overlapping[postmortem_ipsn.c9orf72.overlapping %in% p53.signalling.pathway]
postmortem_ipsn.sod1.overlapping[postmortem_ipsn.sod1.overlapping %in% p53.signalling.pathway]
postmortem_ipsn.fus.overlapping[postmortem_ipsn.fus.overlapping %in% p53.signalling.pathway]
postmortem_ipsn.sporadic.overlapping[postmortem_ipsn.sporadic.overlapping %in% RNA_BINDING_PROTEINS]
postmortem_ipsn.c9orf72.overlapping[postmortem_ipsn.c9orf72.overlapping %in% RNA_BINDING_PROTEINS]
postmortem_ipsn.sod1.overlapping[postmortem_ipsn.sod1.overlapping %in% RNA_BINDING_PROTEINS]
postmortem_ipsn.fus.overlapping[postmortem_ipsn.fus.overlapping %in% RNA_BINDING_PROTEINS]
postmortem_ipsn.sporadic.overlapping[postmortem_ipsn.sporadic.overlapping %in% ALS_genes]
postmortem_ipsn.c9orf72.overlapping[postmortem_ipsn.c9orf72.overlapping %in% ALS_genes]
postmortem_ipsn.sod1.overlapping[postmortem_ipsn.sod1.overlapping %in% ALS_genes]
postmortem_ipsn.fus.overlapping[postmortem_ipsn.fus.overlapping %in% ALS_genes]

postmortem_ipsn.c9orf72.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_id) %>% gost() # axonemal dynein complex assembly
postmortem_ipsn.c9orf72.overlapping[postmortem_ipsn.c9orf72.overlapping%in%gprofiler_database$`axonemal dynein complex assembly`]

postmortem_ipsn.fus.join %>% filter(nygc.padj < 0.05, ipsn.padj < 0.05, (nygc.stat > 0 & ipsn.stat > 0) | (nygc.stat < 0 & ipsn.stat < 0)) %>% top_n(n = 20, wt = abs(nygc.stat)) %>% pull(gene_id) %>% gost() # TF Factor: TCF-3; motif: ACATCGRGRCGCTGW

```


## Merge

```{r}
nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged = ggarrange(
  ggarrange(nygc_postmortem_spinal_cord.als_vs_ctrl.volcano, nygc_postmortem_spinal_cord.als_vs_ctrl.go_curated, ncol=2, labels = c("a","b"), widths = c(1,0.8)),
  ggarrange(nygc_postmortem_spinal_cord.als_vs_ctrl_progeny.pathwayActivity, nygc_postmortem_spinal_cord.als_vs_ctrl_dorothea.contrast_acts.volcano, ncol=2, labels = c("c","d")),
  ggarrange(postmortem_ipsn.als_vs_ctrl.scatter, nygc_postmortem_spinal_cord_mn_mutations_list.p53.progeny_scores.plot, nygc_postmortem_spinal_cord_mn_mutations_list.TP53.dorothea_scores.plot, ncol = 3, labels = c("e","f","g"), widths = c(2,1,1)),
  postmortem_ipsc_mn_mutations_list_t_stat_plot,
  nrow = 4, labels = c("","","","h"))
# nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged
ggsave(nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged, filename = here(proj_path,"expression/deseq2/nygc_postmortem_spinal_cord.als_vs_ctrl.fig.merged.png"), height = 12.5, width = 9.5)

```


# Table S6 postmortem & ipsn overlap DGE

```{r}
postmortem_ipsc_overlap.res = 
  mutate(select(nygc_postmortem_spinal_cord.als_vs_ctrl$res, gene_name, gene_id, padj.postmortem = padj, stat.postmortem = stat, lfc.postmortem = log2FoldChange), direction.postmortem = case_when(padj.postmortem < 0.05 & lfc.postmortem > 0 ~ "up", padj.postmortem < 0.05 & lfc.postmortem < 0 ~ "down", TRUE ~ "non-significant")) %>%
  full_join(mutate(select(ipsc_mn_als_datasets$res, gene_name, gene_id, padj.ipsn = padj, stat.ipsn = stat, lfc.ipsn = log2FoldChange), direction.ipsn = case_when(padj.ipsn < 0.05 & lfc.ipsn > 0 ~ "up", padj.ipsn < 0.05 & lfc.ipsn < 0 ~ "down", TRUE ~ "non-significant"))) %>%
  select(gene_name, gene_id, everything()) %>% 
  filter(padj.ipsn < 0.05, padj.postmortem < 0.05, ((stat.ipsn > 0 & stat.postmortem > 0) | (stat.ipsn < 0 & stat.postmortem < 0)))
postmortem_ipsc_overlap.res
write_csv(postmortem_ipsc_overlap.res, here(proj_path,"expression/deseq2/Table.postmortem_ipsc_overlap.csv"))

```


# Fig 4 Splicing pan ALS

MAJIQ 

## Volcano & GO terms

```{r}
majiq_schematic = magick::image_read(here(proj_path,"splicing/majiq/figures/MAJIQ splicing schematic.png"))
majiq_schematic.gg = ggdraw() + draw_image(majiq_schematic)
majiq_schematic.gg

als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
als_ctrl.changing_only.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.changing_only.tsv"), grp1 = "als", grp2 = "ctrl")
pan_als.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/pan_als"), skip_rows = 9)

als_ctrl.changing_only.voila
als_ctrl.voila
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 264
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 161
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 27
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.DDR.repair_gene_set) # MTA1, POLM, PMS1 HUWE1, HDAC1, ZWIM7
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) # MTA1, POLM, PMS1 HUWE1, HDAC1, ZWIM7
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # 0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) # HLA-B
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% sporadic_ALS_genes) #0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_GWAS) #0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_EWAS) # SIRT2
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_modifiers) # 0
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # HDAC1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`DNA repair`) #  MTA1, POLM, PMS1 HUWE1, ZWIM7
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, -log10(tnom) > 4) # TVP23C-CDRT4, SREK1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("SETMAR", "RIF1", "HUS1", "SMC6", "METTL22")) # METTL22

## Volcano
ipsc_mn_als_datasets.polyA.metadata %>% count(dataset) # 10
ipsc_mn_als_datasets.polyA.metadata %>% count(condition) #48 ALS, 43 ctrl
# majiq.als_vs_ctrl.RBP.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.RBP.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("YTHDC2", "THOC1", "PRR3", "STAU2", "PTBP3", "SREK1", "POLDIP3")) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.ALS.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.DDR.junction_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.DDR.repair_gene_set) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.extra_labels <- als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("METTL22","LINC00665")) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, -log10(tnom)>3, -log10(tnom)<4) 

majiq.als_vs_ctrl.volcano.junction_labels = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(majiq.als_vs_ctrl.RBP.junction_labels, majiq.als_vs_ctrl.ALS.junction_labels, majiq.als_vs_ctrl.DDR.junction_labels)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.volcano <- majiq_het_volcano_plot(als_ctrl.voila, "Alternative splicing: pan ALS", numerator = "ALS", denomenator = "CTRL", subtitle = "polyA: 43 CTRL vs 48 ALS", ymax = 5, xmax = 0.32, dpi = 300, junction_labels = c(majiq.als_vs_ctrl.RBP.junction_labels, majiq.als_vs_ctrl.ALS.junction_labels, majiq.als_vs_ctrl.DDR.junction_labels, majiq.als_vs_ctrl.extra_labels))
majiq.als_vs_ctrl.volcano

# GO of splicing events
als_ctrl.voila.gost = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
als_ctrl.voila.gost_filt <- als_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(als_ctrl.voila.gost_filt$query) # 
paste('"',unique(als_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
"Hedgehog signaling pathway",
"regulation of nervous system development",
"positive regulation of synapse maturation",
# "latrotoxin receptor activity",
"protein binding")
als_ctrl.voila.gost_filt.terms <- als_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
als_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", als_ctrl.voila.gost_filt.terms$term_name) 
als_ctrl.voila.gost_filt.terms.curated <- curated_profiles(als_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
als_ctrl.voila.gost_filt.terms.curated

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`protein binding`) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`regulation of nervous system development`) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`positive regulation of synapse maturation`) %>% arrange(-abs(deltapsi))
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% kegg.pathways.msigdb$KEGG_HEDGEHOG_SIGNALING_PATHWAY) %>% arrange(-abs(deltapsi))

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_POSITIVE_REGULATION_OF_PROTEIN_BINDING) %>% arrange(-abs(deltapsi)) # ARHGEF7, NRP1, ADD1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_AXON) %>% arrange(-abs(deltapsi)) %>% distinct(gene_name) # SRRM1, SORBS2, PAN3, MARK2, HUWE1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% arrange(-abs(deltapsi)) %>% print(n=Inf) # FZNF326, RTF1, AUH, NOVA1
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% go.pathways.msigdb$GO_RNA_SPLICING) %>% arrange(-abs(deltapsi)) #THOC1, ZC3H13, PTBP3, SREK1


# Compare splicing with gene expression
als_splicing_events_genes = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
als_degs = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% pull(gene_name)
als_degs[als_degs %in% als_splicing_events_genes] #THB15B


# TDP43 knockdown overlap
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
tdp43kd.als_vs_ctrl.brown.voila.sig = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9)# %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.sig 

tdp43kd.als_vs_ctrl.brown.voila.labels = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)

# # Cryptic Exons == denovo & cassette_exon
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 28 / 264
28/264# 10.6%
pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer) #7 cryptic exons
pan_als.voila_modulizer.cryptic = pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) %>% pull(gene_name)  # "RELCH"         "HOXC4"         "RBM26"         "SLC35B3"       "TENM3"         "TPTEP2-CSNK1E" "ZSCAN29"   
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 59 /264
50/264
# overlap cryptic exons with Brown et al TDP43 knockdown
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
tdp43kd.als_vs_ctrl.brown.voila.sig.cryptic = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, de_novo_junctions == TRUE)# %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.sig.cryptic #121

tdp43kd.als_vs_ctrl.brown.voila.sig.cryptic$gene_name[tdp43kd.als_vs_ctrl.brown.voila.sig.cryptic$gene_name %in% "UNC13A"]
tdp43kd.als_vs_ctrl.brown.voila.labels = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
```

## Modulizer

```{r}
# pan_als.voila_modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq_polyA/modulize/pan_als"), skip_rows = 9)

# Modulizer pie chart of differential events
modulizer_simplfied.sig.summary = pan_als.voila_modulizer %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% count(modulizer_simplified) %>% mutate(modulizer_simplified = gsub("_"," ",modulizer_simplified), modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>% arrange(desc(modulizer_simplified)) %>% 
  mutate(pct = n/sum(n) *100, pct.label = case_when(pct > 8 ~ paste(as.character(n),"\n", as.character(round(pct,1)),"%"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) #%>% print
modulizer_simplfied.sig.summary %>% print
pan_als.voila_modulizer.sig.pie = ggplot(modulizer_simplfied.sig.summary, aes(x = "", y = pct, fill = fct_reorder(modulizer_simplified, n, .desc = TRUE))) + geom_bar(width = 1, stat = "identity", color="white") + coord_polar("y", start=0) + 
  geom_text(aes(x = 1.2, y = ypos, label = pct.label), color = "black", size=2.5) + scale_fill_brewer(palette="Set1") + 
  theme_oz() + theme_void() + theme(legend.title=element_blank(), legend.position = "right", legend.text=element_text(size=7), legend.spacing.y = unit(1, 'mm'), legend.spacing.x = unit(1, 'mm'), legend.key.height = unit(0.4, 'cm'), legend.key.width = unit(0.3, 'cm')) + guides(fill=guide_legend(ncol=1))
pan_als.voila_modulizer.sig.pie

# Intron retention
pan_als.voila_modulizer.ir = pan_als.voila_modulizer %>% filter(modulizer_simplified == "intron_retention")
pan_als.voila_modulizer.ir %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # 60
pan_als.voila_modulizer.ir %>% filter(gene_name %in% ALS_genes, abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # SIRT2
pan_als.voila_modulizer.ir %>% filter(gene_name %in% "SFPQ")


```

## Voila view

```{r}
# Voila View violin plots:
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% arrange(-abs(deltapsi)) %>% print(n=Inf)
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "SYNJ1") %>% glimpse # lsv_junction_id   "ENSG00000159082:t:32673340-32673531:j:32673531-32678645", "ENSG00000159082:t:32673340-32673531:j:32673531-32676332"
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "AGAP1") %>% glimpse # lsv_junction_id   ENSG00000157985:t:235968462-235968623:j:235908906-235968462", "ENSG00000157985:t:235968462-235968623:j:235930923-235968462"
als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "LINC00665") %>% glimpse # lsv_junction_id "ENSG00000232677:t:36321993-36322030:j:36322030-36330253", "ENSG00000232677:t:36321993-36322030:j:36322030-36322130"


als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% arrange(-abs(deltapsi)) %>% print(n=Inf) # HLA-B
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "HLA-B") %>% glimpse # lsv_junction_id   "ENSG00000234745:s:31356688-31356957:j:31356442-31356688", "ENSG00000234745:s:31356688-31356957:j:31271348-31356688"

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% p53.DDR.repair_gene_set) # MTA1, POLM, PMS1 HUWE1, HDAC1, ZWIM7, METTL22
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "METTL22") %>% glimpse # lsv_junction_id  "ENSG00000067365:s:8644437-8645044:j:8644725-8645908", "ENSG00000067365:s:8644437-8645044:j:8644725-8646108"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "MTA1") %>% glimpse # lsv_junction_id  "ENSG00000182979:s:105466707-105467192:j:105467193-105468108"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLM") %>% glimpse # lsv_junction_id  "ENSG00000122678:s:44073625-44073708:j:44073430-44073625", "ENSG00000122678:s:44073625-44073708:j:44073377-44073625"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "PMS1") %>% glimpse # lsv_junction_id "ENSG00000064933:t:189867799-189867929:j:189864360-189867798"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "HUWE1") %>% glimpse # lsv_junction_id "ENSG00000086758:s:53627410-53627515:j:53626034-53627410", "ENSG00000086758:s:53627410-53627515:j:53625258-53627410"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "HDAC1") %>% glimpse # lsv_junction_id  "ENSG00000116478:s:32327536-32327677:j:32327677-32329068", "ENSG00000116478:s:32327536-32327677:j:32327678-32329067"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "ZSWIM7") %>% glimpse # lsv_junction_id  "ENSG00000214941:t:15976560-15977092:j:15977092-15978044", "ENSG00000214941:t:15976560-15977092:j:15977092-15977579"

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% arrange(-abs(deltapsi)) %>% print(n=Inf) # YTHDC2, THOC1, PRR3, STAU2, PTBP3, SREK1, POLDIP3
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLDIP3") %>% glimpse # lsv_junction_id  "ENSG00000100227:s:42602768-42603160:j:42599793-42602770"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "YTHDC2") %>% glimpse # lsv_junction_id"ENSG00000047188:t:113532879-113533045:j:113515362-113532879", "ENSG00000047188:t:113532879-113533045:j:113526785-113532879"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "THOC1") %>% glimpse # lsv_junction_id ENSG00000079134:s:263493-264092:j:260304-264026"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "PRR3") %>% glimpse # lsv_junction_id"ENSG00000204576:s:30558150-30558436:j:30558212-30561328", "ENSG00000204576:s:30558150-30558436:j:30558212-30561834"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "STAU2") %>% glimpse # lsv_junction_id"ENSG00000040341:s:73549604-73552319:j:73422702-73552012"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "PTBP3") %>% glimpse # lsv_junction_id "ENSG00000119314:s:112297832-112297916:j:112276013-112297832"
als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "SREK1") %>% glimpse # lsv_junction_id  "ENSG00000153914:s:66174624-66175041:j:66175041-66177517"


### import violin from voila view
# edit colour and labels in illustrator > export as png
polyA_majiq_violins_merged.png = magick::image_read(here(proj_path,"splicing/majiq/figures/polyA_majiq_violins_merged.png"))
polyA_majiq_violins_merged.gg = ggdraw() + draw_image(polyA_majiq_violins_merged.png)
polyA_majiq_violins_merged.gg
# BID_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/BID_majiq_violin.png"))
# BID_majiq_violins.gg = ggdraw() + draw_image(BID_majiq_violins.png)
# BID_majiq_violins.gg
# HTT_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/HTT_majiq_violin.png"))
# HTT_majiq_violins.gg = ggdraw() + draw_image(HTT_majiq_violins.png)
# HTT_majiq_violins.gg
# GADD45B_majiq_violins.png = magick::image_read(here(proj_path,"splicing/majiq/figures/GADD45B_majiq_violin.png"))
# GADD45B_majiq_violins.gg = ggdraw() + draw_image(GADD45B_majiq_violins.png)
# GADD45B_majiq_violins.gg
# majiq_violins.gg = GADD45B_majiq_violins.gg + CAMTA1_majiq_violins.gg  + BID_majiq_violins.gg + HTT_majiq_violins.gg
# majiq_violins.gg



```



## Merge

```{r}
pan_als_ipsn_splicing_fig_merged = plot_grid(
  majiq_schematic.gg,
  plot_grid(majiq.als_vs_ctrl.volcano, polyA_majiq_violins_merged.gg, ncol = 2, labels = c("b","c"), rel_widths = c(1,0.9)),
  plot_grid(als_ctrl.voila.gost_filt.terms.curated, pan_als.voila_modulizer.sig.pie, ncol = 2, labels = c("d","e")),
  nrow = 3, rel_heights = c(0.5,1,0.7), labels = c("a","","",""))
# pan_als_ipsn_splicing_fig_merged
ggsave(pan_als_ipsn_splicing_fig_merged, filename = here(proj_path,"splicing/majiq/figures/pan_als_ipsn_splicing_fig_merged.png"), height = 7, width = 8.5)

```



# Table S7 MAJIQ panALS datasets

```{r}
# als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
als_ctrl.voila.sig = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% select(-contains("percentile"),-contains("changing"),-ucsc_lsv_link) %>% arrange(-abs(deltapsi))
als_ctrl.voila.sig
write_csv(als_ctrl.voila.sig, here(proj_path,"splicing/majiq/figures/Table.majiq_splicing_panALS.sig.csv"))

# ipsc_mn_als_datasets.polyA.ir.sig = ipsc_mn_als_datasets.polyA$irfinder %>% filter(padj < 0.05) %>% select(Intron.GeneName.GeneID.Coords, log2FoldChange, stat, padj, IRratio.ALS = IRratio.mutant, IRratio.ctrl, IRratio.diff, IRdirection, gene_name, gene_id)
# ipsc_mn_als_datasets.polyA.ir.sig
# write_csv(ipsc_mn_als_datasets.polyA.ir.sig, here(proj_path,"splicing/irfinder/ipsc_mn_als_datasets.polyA.ir.sig.csv"))

```


# Fig S11 TDP43 knockdown POLDIP event

```{r}
als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
tdp43_positive.als_vs_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq/voila-tsv/tdp43_positive.als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl") %>% mutate(mutation = "tdp43_positive")
tdp43kd.als_vs_ctrl.brown.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/public-data/ipsc-mn-tardbp-kd-brown-2022/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv"), grp1 = "tdp43kd", grp2 = "ctrl") %>% mutate(mutation = "tdp43kd")
# neuronal_nuclei_tdp43.liu.voila = read_voila_deltapsi("/camp/project/proj-luscombn-patani/working/public-data/neuronal-nuclei-tdp43-liu-2019/splicing/majiq/voila-tsv/tdp43kd_vs_ctrl.tsv", grp1 = "tdp43pos", grp2 = "tdp43neg") %>% mutate(mutation = "tdp43pos")
# sfpqkd_vs_scramble.voila = read_voila_deltapsi(here(camp_path,"proj-luscombn-patani/working/patani-data/motor-neuron-rbp-knockdown-rnaseq/splicing/majiq/voila/whole_sfpqkd_vs_scramble.tsv"), grp1 = "sfpqkd_whole", grp2 = "scramble_whole") %>% mutate(mutation = "sfpqkd")


# Volcano TDP-43 knockdown
tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) %>% print(n=50)
tdp43kd.als_vs_ctrl.brown.voila.labels = tdp43kd.als_vs_ctrl.brown.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9, gene_name %in% ALS_genes) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
tdp43kd.als_vs_ctrl.brown.voila.volcano = majiq_deltapsi_volcano_plot(tdp43kd.als_vs_ctrl.brown.voila, junction_labels = tdp43kd.als_vs_ctrl.brown.voila.labels, ymax = 4.1, xmax = 0.7, xpos = 0.5, numerator = "TDP-43 KD", denomenator = "Control", dpi = 300)
tdp43kd.als_vs_ctrl.brown.voila.volcano

# Scatter correlation TDP-43 knockdown with pan-ALS
als_tdp43kd_majiq.join = select(als_ctrl.voila, gene_name, lsv_id, lsv_junction_id, lsv_type, junctions_coords, exons_coords, ir_coords, ALS.deltapsi = deltapsi, ALS.changing = changing_dpsi0.1) %>% 
  left_join(select(tdp43kd.als_vs_ctrl.brown.voila, gene_name, lsv_id, lsv_junction_id, lsv_type, junctions_coords, exons_coords, ir_coords, TDP43kd.deltapsi = deltapsi, TDP43kd.changing = changing_dpsi0.1))
als_tdp43kd_majiq.join %>% glimpse # 138,252
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) # 0
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% arrange(-(abs(ALS.deltapsi) + abs(TDP43kd.deltapsi))) %>% print(n=Inf)
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, abs(TDP43kd.deltapsi) > 0.1, ALS.deltapsi > 0 & TDP43kd.deltapsi > 0 | ALS.deltapsi < 0 & TDP43kd.deltapsi < 0) %>% print(n=Inf) # GRIA4, DONSON
# als_tdp43kd_majiq.join = select(als_ctrl.voila, gene_name, lsv_id,  lsv_type, junctions_coords, exons_coords, ir_coords, ALS.deltapsi = deltapsi, ALS.p = tnom) %>% 
#   left_join(select(tdp43kd.als_vs_ctrl.brown.voila, gene_name, lsv_id, lsv_type, junctions_coords, exons_coords, ir_coords, TDP43kd.deltapsi = mean_dpsi_per_lsv_junction, TDP43kd.changing = probability_changing))
# als_tdp43kd_majiq.join %>% glimpse # 229,823

als_tdp43kd_majiq.list = list("pan ALS" = als_ctrl.voila, "TDP-43 KD" = tdp43kd.als_vs_ctrl.brown.voila)
tdp43kd.als_vs_ctrl.brown.voila %>% filter(probability_changing < 0.95, gene_name %in% ALS_genes)

# scatter correlation
als_tdp43kd_majiq.join.labels = als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # "NYNRIN" "SNX2"  
als_tdp43kd_majiq.join.scatter =dpsi_majiq_scatter(model_list1 = als_tdp43kd_majiq.list, model_list2 = als_tdp43kd_majiq.list, mutation1 = "pan ALS", mutation2 = "TDP-43 KD", highlight = als_tdp43kd_majiq.join.labels) + xlim(-0.3,0.3) + ylim(-0.7,0.7)
als_tdp43kd_majiq.join.scatter

# Overlap splicing events in TDP-43 knockdown with pan-ALS
als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% print(n=Inf)# NYNRIN
als_tdp43kd_majiq.join.labels = als_tdp43kd_majiq.join %>% filter(ALS.changing == TRUE, TDP43kd.changing == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # "NYNRIN" "SNX2"  
als_tdp43kd_majiq.join.labels #POLDIP3
als_tdp43kd_majiq.join.labels[als_tdp43kd_majiq.join.labels %in% RNA_BINDING_PROTEINS]

# Overlap at gene level:
pan_als_majiq_genes.sig = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) %>% pull(gene_name)
tdp43kd_brown_majiq_genes.sig = tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) %>% pull(gene_name)
pan_als_majiq_genes.sig[pan_als_majiq_genes.sig %in% tdp43kd_brown_majiq_genes.sig] # "POLDIP3" "CAMK2B"  "CEP290"  "HERC2P3"

als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "CAMK2B") %>% glimpse #"ENSG00000058404:s:44234390-44234461:j:44232866-44234390", "ENSG00000058404:s:44234390-44234461:j:44231054-44234390"
tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name == "CAMK2B") %>% glimpse #"ENSG00000058404:t:44220826-44220953:j:44220901-44231006", "ENSG00000058404:t:44254542-44254607:j:44254607-44258872", "ENSG00000058404:t:44254542-44254607:j:44254607-44258490"


# where in POLDIP3 is the cryptic exon?
tdp43kd.als_vs_ctrl.brown.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name == "POLDIP3") %>% glimpse # "ENSG00000100227:t:42599698-42599793:j:42599793-42602770", "ENSG00000100227:t:42599698-42599793:j:42599793-42601970", "ENSG00000100227:s:42602768-42603160:j:42602056-42602770"

```


# Fig 5 Variants and fusions

## Variants

https://cran.r-project.org/web/packages/vcfR/vignettes/intro_to_vcfR.html

```{r}
# vcf.annotate_script = ipsc_mn_als_datasets.metadata %>%
#   mutate(`#!/bin/bash` = case_when(dataset == "answerals" ~ paste0("zcat /camp/project/proj-luscombn-patani/working/answerals/rnavar/variant_calling/", sample, "/", sample, ".haplotypecaller.filtered.vcf.gz | vcf-annotate --fill-type | grep -oP 'TYPE=\\w+' | sort | uniq -c | sed 's/ TYPE=/,/' > ", dataset_sample, ".vcfstats.csv"),
#   TRUE ~ paste0("zcat /camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/", dataset_sample,"/", dataset_sample, ".haplotypecaller.filtered.vcf.gz | vcf-annotate --fill-type | grep -oP 'TYPE=\\w+' | sort | uniq -c | sed 's/ TYPE=/,/' > ", "/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate/",dataset_sample, ".vcfstats.csv"))) %>% select(`#!/bin/bash`)
# vcf.annotate_script
# vcf.annotate_script %>% pull(`#!/bin/bash`)    # check script: fastq order, job name
# write_tsv(vcf.annotate_script, "/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/sample-details/vcf.annotate_script.sh")

# read_vcfannotate = function(metadata){
#   print("reading in vcf-annotate files with read_csv")
#   vcf.annotate.csv = metadata$rnavar_path %>% map(read_csv, col_names = c("n","type"), col_types = list(n="n",type="c"))
#   names(vcf.annotate.csv) = metadata$dataset_sample # needs to be unique names e.g. dataset_sample
#   vcf.annotate_tab <- map_dfr(vcf.annotate.csv, bind_rows, .id = "dataset_sample") %>%
#     mutate(variant_type = case_when(grepl("del", type) ~ "Deletion", grepl("ins", type) ~ "Insertion", grepl("snp", type) ~ "SNP", grepl("complex", type) ~ "Complex", TRUE ~ "Unknown")) %>%
#     left_join(metadata, by = "dataset_sample")
#   return(vcf.annotate_tab)
# }
# ipsc_mn_als_datasets.metadata = ipsc_mn_als_datasets.metadata %>% mutate(rnavar_path = paste0("/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate/",dataset_sample,".vcfstats.csv"))
# ipsc_mn_als_datasets.vcfannotate = read_vcfannotate(ipsc_mn_als_datasets.metadata)
# ipsc_mn_als_datasets.vcfannotate
# write_csv(ipsc_mn_als_datasets.vcfannotate, here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.csv"))
ipsc_mn_als_datasets.vcfannotate = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.redi.filtered.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(dataset_sample)%>% add_tally(n, name = "total") %>% ungroup %>% select(dataset_sample, n, type, variant_type, total, everything()) # add a column of sum of variants per sample
ipsc_mn_als_datasets.vcfannotate

# left_join multiqc stats
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
multiqc_rseqc_read_distribution.tsv

ipsc_mn_als_datasets.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.vcfannotate

answerals.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% filter(dataset == "answerals")
```

### ALS vs CTRL variant number

```{r}
ipsc_mn_als_datasets.metadata %>% filter(dataset == "answerals") %>% count(condition) # 238 ALS, 42 CTRL

answerals.vcfannotate.samples_no_variants = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% answerals.vcfannotate$dataset_sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

answerals.vcfannotate.totals = answerals.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) #keep 1 row per sample for total number of variants
answerals.vcfannotate.totals %>% glimpse

# generalised linear model spline
# library(rms)
answerals.vcfannotate.totals.glm = glm(data=answerals.vcfannotate.totals, total ~ CONDITION + rms::rcs(total_reads, 3), family = poisson) # + rms::rcs(total_reads, 3)
summary(answerals.vcfannotate.totals.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.063e+01  3.409e-03 3119.19   <2e-16 ***
# CONDITIONALS                          1.020e-02  7.027e-04   14.52   <2e-16 ***
fit <- glm(total ~ CONDITION * rms::rcs(total_reads, 3), family = poisson, data=answerals.vcfannotate.totals)
predicted <- predict(fit, type="response")
ggplot(cbind(answerals.vcfannotate.totals, predicted=predicted), aes(x=total_reads, y=predicted, colour= CONDITION, group = interaction(CONDITION))) + geom_line() + theme_oz()

# Violin total variant events
answerals.vcfannotate.totals.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 75000, xmin = 1, xmax = 2)
answerals.vcfannotate.total.violin <- violin_plot(answerals.vcfannotate.totals, color_by = "CONDITION", continuous = "total", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(50000,76000), dpi = 300, ylabel = "Total Number of Variants", title = "All Variant types") +
  stat_pvalue_manual(answerals.vcfannotate.totals.stats, tip.length = 0.01, size = 4)
answerals.vcfannotate.total.violin

answerals.vcfannotate.totals %>% my_summarise(total, condition) %>% mutate(IQR = Q3_total - Q1_total)
# condition mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <chr>          <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1 als           58935.       58320.    3396.   56877.   60480.     47174     70740  14026509     238 3603 
# 2 ctrl          57615.       57278.    3394.   55198.   58946.     53050     69191   2419840      42 3748.

# # coverage & variant detection scatter
# answerals.vcfannotate.totals_total_reads.scatter <- ggscatter(answerals.vcfannotate.totals, x = "total_reads", y = "total", color = "condition", cor.coef = TRUE, cor.method = "pearson", size = 0.5, cor.coeff.args = list(label.x = 50000000, label.y = 65000, label.sep="\n", size = 2.8),
#           xlab = expression(Total~reads~per~sample), ylab = expression(Total~number~of~variants~per~sample), title = "Answer ALS variant detection") +
#   theme_oz() + theme(legend.position = "right") #+
#   # geom_hline(yintercept = 0, linetype = 2) + geom_vline(xintercept = 0, linetype = 2) +
#   # coord_cartesian(ylim=c(0,80000))
# answerals.vcfannotate.totals_total_reads.scatter

```

### Variant types

```{r}
answerals.vcfannotate %>% group_by(variant_type) %>% tally(n)
# variant_type        n
#   <fct>           <dbl>
# 1 SNP          14114280
# 2 Insertion     1280116
# 3 Deletion      1049471
# 4 Complex          2482


### SNP
answerals.vcfannotate.snp = answerals.vcfannotate %>% filter(variant_type == "SNP") 
# add samples with 0 variants for each variant type
samples_no_SNP = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% pull(answerals.vcfannotate.snp,dataset_sample)] 
samples_no_SNP # 0 samples with 0 SNPS

answerals.vcfannotate.snp.glm = glm(data = answerals.vcfannotate.snp, n ~ CONDITION + rms::rcs(total_reads, 3), family = poisson)
summary(answerals.vcfannotate.snp.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.048e+01  3.680e-03 2847.29   <2e-16 ***
# CONDITIONALS                          8.880e-03  7.581e-04   11.71   <2e-16 ***

answerals.vcfannotate.snp.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 62000, xmin = 1, xmax = 2)
answerals.vcfannotate.snp.violin <- violin_plot(answerals.vcfannotate.snp, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(40000,64000), dpi = 300, ylabel = "Number of Variants", title = "SNV")  +
  stat_pvalue_manual(answerals.vcfannotate.snp.stats, tip.length = 0.01, size = 4)
answerals.vcfannotate.snp.violin

answerals.vcfannotate.snp %>% my_summarise(n, condition) %>% mutate(IQR = Q3_n - Q1_n)
# condition mean_n median_n  sd_n   Q1_n   Q3_n min_n max_n    sum_n   n_n   IQR
#   <chr>      <dbl>    <dbl> <dbl>  <dbl>  <dbl> <dbl> <dbl>    <dbl> <int> <dbl>
# 1 als       50567.    50093 2990. 48884. 51770. 40535 60910 12034889   238 2886 
# 2 ctrl      49509.    49361 2854. 47203. 50772  45465 59326  2079391    42 3569.


### Insertions
answerals.vcfannotate.insertion = answerals.vcfannotate %>% filter(variant_type == "Insertion") 
# add samples with 0 variants for each variant type
samples_no_Insertion = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% pull(answerals.vcfannotate.insertion,dataset_sample)]
samples_no_Insertion # 0 samples with 0 Insertions

answerals.vcfannotate.Insertion.glm = glm(data = answerals.vcfannotate.insertion, n ~ CONDITION + rms::rcs(total_reads, 3), family = poisson)
summary(answerals.vcfannotate.Insertion.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           8.121e+00  1.219e-02 666.031  < 2e-16 ***
# CONDITIONALS                          1.928e-02  2.528e-03   7.624 2.47e-14 ***

answerals.vcfannotate.insertion.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 6400, xmin = 1, xmax = 2)
answerals.vcfannotate.insertion.violin <- violin_plot(answerals.vcfannotate.insertion, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(3500,6700), dpi = 300, ylabel = "", title = "Insertion")  +
  stat_pvalue_manual(answerals.vcfannotate.insertion.stats, tip.length = 0.01, size = 4)
answerals.vcfannotate.insertion.violin

answerals.vcfannotate.insertion %>% my_summarise(n, condition) %>% mutate(IQR = Q3_n - Q1_n)
# condition mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n   sum_n   n_n   IQR
#   <chr>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>   <dbl> <int> <dbl>
# 1 als        4595.    4580.  378. 4365  4818.  3506  6353 1093518   238  452.
# 2 ctrl       4443.    4436.  438. 4104. 4715   3731  5716  186598    42  611.


### Deletions
answerals.vcfannotate.deletion = answerals.vcfannotate %>% filter(variant_type == "Deletion") 

# add samples with 0 variants for each variant type
samples_no_Deletion = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% pull(answerals.vcfannotate.deletion,dataset_sample)] 
samples_no_Deletion # 0 samples with 0 Deletions

answerals.vcfannotate.Deletion.glm = glm(data = answerals.vcfannotate.deletion, n ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(answerals.vcfannotate.Deletion.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           7.858e+00  1.351e-02  581.45  < 2e-16 ***
# CONDITIONALS                          1.701e-02  2.789e-03    6.10 1.06e-09 ***

answerals.vcfannotate.insertion.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "****", y.position = 4900, xmin = 1, xmax = 2)
answerals.vcfannotate.deletion.violin <- violin_plot(answerals.vcfannotate.deletion, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(3000,5000), dpi = 300, ylabel = "", title = "Deletion") +
  stat_pvalue_manual(answerals.vcfannotate.insertion.stats, tip.length = 0.01, size = 4)
answerals.vcfannotate.deletion.violin

answerals.vcfannotate.deletion %>% my_summarise(n, condition) %>% mutate(IQR = Q3_n - Q1_n)
# condition mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n  sum_n   n_n   IQR
#   <chr>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>  <dbl> <int> <dbl>
# 1 als        3765.    3769   211. 3625. 3858.  2994  4486 895987   238  233.
# 2 ctrl       3654.    3676.  217. 3466. 3786   3323  4149 153484    42  320.


# ### Complex
# answerals.vcfannotate_complex = answerals.vcfannotate %>% filter(variant_type == "Complex")
# 
# # add a row for each sample with 0 complex variants to get per iPSN rate
# samples_no_Complex = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% answerals.vcfannotate_complex$dataset_sample]
# samples_no_Complex # 0 samples with 0 Complex variants
# 
# answerals.vcfannotate.Complex.glm = glm(data = answerals.vcfannotate_complex, n ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
# summary(answerals.vcfannotate.Complex.glm)
# # Coefficients:
# #                                        Estimate Std. Error z value Pr(>|z|)    
# # (Intercept)                           1.338e+00  2.887e-01   4.633  3.6e-06 ***
# # CONDITIONALS                         -9.001e-03  5.709e-02  -0.158  0.87471    
# 
# answerals.vcfannotate_complex %>% my_summarise(n, condition) %>% mutate(IQR = Q3_n - Q1_n)
# # condition mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
# #   <chr>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# # 1 als         8.89      9    2.99     7  11       2    20  2115   238  4   
# # 2 ctrl        8.74      8.5  2.66     7  10.8     3    15   367    42  3.75
# 
# answerals.vcfannotate.complex.violin <- violin_plot(answerals.vcfannotate_complex, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"),
#              plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,25), dpi = 300, ylabel = "", title = "Complex")
# answerals.vcfannotate.complex.violin

answerals.vcfannotate.variant_types.violin = plot_grid(answerals.vcfannotate.snp.violin, answerals.vcfannotate.insertion.violin, answerals.vcfannotate.deletion.violin, nrow = 1) 
#answerals.vcfannotate.complex.violin

answerals.vcfannotate %>% my_summarise(n, CONDITION, variant_type)
# CONDITION variant_type   mean_n median_n     sd_n min_n  max_n    sum_n   n_n
#   <fct>     <fct>           <dbl>    <dbl>    <dbl> <dbl>  <dbl>    <dbl> <int>
# 1 CTRL      SNP          45509.     48434. 19130.   13873 193118  4823999   106
# 2 CTRL      Insertion     4433.      4257   2526.     523  11457   469891   106
# 3 CTRL      Deletion      3125.      3554.  1080.     519   4773   331210   106
# 4 CTRL      Complex          8.36       8      4.44     1     18      803    96
# 5 ALS       SNP          49435.     51279   9412.   10534 119364 15967439   323
# 6 ALS       Insertion     4701.      4711   1709.     527  11544  1518500   323
# 7 ALS       Deletion      3634.      3889    796.     517   4673  1173875   323
# 8 ALS       Complex          9.04       9      3.90     1     23     2801   310

```

### Substitutions

```{r}
# http://bioconductor.org/packages/release/bioc/vignettes/MutationalPatterns/inst/doc/Introduction_to_MutationalPatterns.html
# BiocManager::install("MutationalPatterns")
library(MutationalPatterns)
ref_genome <- "BSgenome.Hsapiens.UCSC.hg38"
library(ref_genome, character.only = TRUE)

# # read in vcf files as granges
# answerals.metadata = answerals.metadata %>% mutate(vcf_path = paste0("/camp/project/proj-luscombn-patani/working/answerals/rnavar/variant_calling/",sample, "/", sample, ".redi.filtered.recode.vcf.gz"))
# grl <- read_vcfs_as_granges(answerals.metadata$vcf_path, answerals.metadata$sample, ref_genome, type = "all")
# saveRDS(grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/grl.RDS")
# snv_grl <- get_mut_type(grl, type = "snv") # get types of mutations SNV, indels
# saveRDS(snv_grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/snv_grl.RDS")
# indel_grl <- get_mut_type(grl, type = "indel")
# saveRDS(indel_grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/indel_grl.RDS")
# dbs_grl <- get_mut_type(grl, type = "dbs")
# saveRDS(dbs_grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/dbs_grl.RDS")
# mbs_grl <- get_mut_type(grl, type = "mbs")
# saveRDS(mbs_grl, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/mbs_grl.RDS")
# type_occurrences <- mut_type_occurrences(snv_grl, ref_genome) # count mutation type occurences and base substitutions 
# saveRDS(type_occurrences, "/camp/project/proj-luscombn-patani/working/answerals/rnavar/mutationalpatterns/type_occurrences.RDS")

grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/grl.RDS"))
snv_grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/snv_grl.RDS"))
indel_grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/indel_grl.RDS"))
dbs_grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/dbs_grl.RDS"))
mbs_grl = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/mbs_grl.RDS"))
type_occurrences = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/type_occurrences.RDS"))
type_occurrences
grl

# left_join multiqc stats
answerals.multiqc_rseqc_read_distribution.tsv = read_tsv(here(shared_path,"answerals/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names()

answerals.base_substitutions = type_occurrences %>% as_tibble(rownames = "sample") %>% select(-`C>T at CpG`, -`C>T other`) %>%
  left_join(select(answerals.metadata, sample, condition)) %>% 
  pivot_longer(!c(condition, sample), names_to = "Substitution", values_to = "count") %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), Substitution = as.factor(Substitution)) %>%
  left_join(select(answerals.multiqc_rseqc_read_distribution.tsv, sample, total_reads))
answerals.base_substitutions  


# Plot count of each base substitution type
  
answerals.base_substitutions %>% my_summarise(count, CONDITION, Substitution) %>% mutate(IQR = Q3_count - Q1_count)
# condition Substitution mean_count median_count sd_count Q1_count Q3_count min_count max_count sum_count n_count   IQR
#    <fct>     <chr>             <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <int>     <int>     <int>   <int> <dbl>
#  1 ctrl      C>A               3263.        3236      223.    3112.    3379       2873      3936    137063      42  268.
#  2 ctrl      C>G               4438.        4425      232.    4291.    4556.      4054      5250    186400      42  265 
#  3 ctrl      C>T              16069.       16092.     834.   15431.   16520.     14716     19221    674892      42 1090.
#  4 ctrl      T>A               2076.        2048.     156.    1954     2180.      1877      2576     87197      42  226.
#  5 ctrl      T>C              19309.       19287     1297.   18278.   19822.     17496     23287    810985      42 1544 
#  6 ctrl      T>G               3176.        3169      151.    3076.    3256.      2842      3592    133402      42  179 
#  7 als       C>A               3344.        3330      243.    3170.    3465.      2690      4465    795989     238  295 
#  8 als       C>G               4509.        4473      270.    4351     4618.      3757      5541   1073075     238  267.
#  9 als       C>T              16337.       16226.     996.   15732.   16684.     13438     20056   3888189     238  952.
# 10 als       T>A               2135.        2130.     155.    2035.    2206.      1686      2666    508227     238  171 
# 11 als       T>C              19793.       19599     1278.   18982    20442.     15490     24349   4710814     238 1460.
# 12 als       T>G               3225.        3215      181.    3115     3308.      2600      3880    767497     238  194.

# For count data use glm(count ~ condition, family=poisson) which gives estimates and p-values of the underlying 'rate's of fusion per sample 
### C>A
glm_starfusion_C_A = glm(data = filter(answerals.base_substitutions, Substitution == "C>A"), count ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_C_A)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           7.672e+00  1.443e-02 531.861  < 2e-16 ***
# CONDITIONALS                          8.564e-03  2.952e-03   2.901  0.00372 ** 

### C>G
glm_starfusion_C_G = glm(data = filter(answerals.base_substitutions, Substitution == "C>G"), count ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_C_G)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           8.085e+00  1.230e-02 657.069   <2e-16 ***
# CONDITIONALS                          3.246e-03  2.533e-03   1.281      0.2    

### C>T
glm_starfusion_C_T = glm(data = filter(answerals.base_substitutions, Substitution == "C>T"), count ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_C_T)
# Coefficients:
#                                        Estimate Std. Error  z value Pr(>|z|)    
# (Intercept)                           9.365e+00  6.467e-03 1448.230  < 2e-16 ***
# CONDITIONALS                          4.251e-03  1.331e-03    3.193  0.00141 ** 

### T>A
glm_starfusion_T_A = glm(data = filter(answerals.base_substitutions, Substitution == "T>A"), count ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_T_A)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           7.247e+00  1.803e-02 401.926  < 2e-16 ***
# CONDITIONALS                          1.202e-02  3.700e-03   3.247  0.00117 ** 

### T>C
glm_starfusion_T_C = glm(data = filter(answerals.base_substitutions, Substitution == "T>C"), count ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_T_C)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           9.563e+00  5.869e-03 1629.54   <2e-16 ***
# CONDITIONALS                          1.433e-02  1.214e-03   11.81   <2e-16 ***

### T>G
glm_starfusion_T_G = glm(data = filter(answerals.base_substitutions, Substitution == "T>G"), count ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_T_G)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           7.741e+00  1.456e-02 531.774   <2e-16 ***
# CONDITIONALS                          2.634e-03  2.995e-03   0.879    0.379    


ipsc_mn_als_datasets.Substitution.stat <- answerals.base_substitutions %>% group_by(Substitution) %>% t_test(count ~ CONDITION) %>% mutate(p.signif = c("**","ns","**","**","****","ns"), y.position = c(4500,5500,20500,2700,25000,3900)) 
ipsc_mn_als_datasets.Substitution.stat
# Substitution .y.   group1 group2    n1    n2 statistic    df      p p.signif y.position
#   <fct>        <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>  <dbl> <chr>         <dbl>
# 1 C>A          count CTRL   ALS       42   238     -2.14  59.5 0.0362 **             4500
# 2 C>G          count CTRL   ALS       42   238     -1.77  62.4 0.0812 ns             5500
# 3 C>T          count CTRL   ALS       42   238     -1.86  63.6 0.0672 **            20500
# 4 T>A          count CTRL   ALS       42   238     -2.27  56.2 0.0269 **             2700
# 5 T>C          count CTRL   ALS       42   238     -2.23  56.0 0.0294 ****          25000
# 6 T>G          count CTRL   ALS       42   238     -1.86  63.8 0.0674 ns             3900
answerals.base_substitutions.violin <- ggplot(answerals.base_substitutions, aes(x = CONDITION, y = count, colour = CONDITION)) +
    ggrastr::rasterise(geom_violin(adjust = 1), dpi = 300) +
    ggrastr::rasterise(geom_sina(size = 0.5), dpi = 300) +
    ggrastr::rasterise(geom_boxplot(aes(fill = CONDITION), colour = "black", width=0.2, alpha = 0.5, outlier.shape = NA), dpi = 300) + 
    scale_colour_manual(values = c("dodgerblue2", "firebrick2")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
    theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
    facet_wrap(~ Substitution, scales = "free", nrow = 1)  + labs(y = "Number of Variants", x = "") + #, title = "Base substitution types"
    stat_pvalue_manual(ipsc_mn_als_datasets.Substitution.stat, label = "p.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
answerals.base_substitutions.violin

```


## Fusions

```{r}
# read_starfusion = function(metadata){
#   print("reading in STAR Fusion files with read_tsv")
#   starfusion.tsv = metadata$starfusion_path %>% map(read_tsv, col_types = list(JunctionReadCount="n",SpanningFragCount="n",est_J="n",est_S="n",FFPM="n",LeftBreakEntropy="n",RightBreakEntropy="n",.default="c"))
#   names(starfusion.tsv) = metadata$dataset_sample # needs to be unique names e.g. dataset_sample
#   starfusion_tab <- map_dfr(starfusion.tsv, bind_rows, .id = "dataset_sample") %>% rename(fusion_name = '#FusionName') %>% clean_names() %>%
#     mutate(left_chr = str_split_fixed(left_breakpoint,":",3)[,1], left_coord = str_split_fixed(left_breakpoint,":",3)[,2], left_strand = str_split_fixed(left_breakpoint,":",3)[,3],
#            right_chr = str_split_fixed(right_breakpoint,":",3)[,1], right_coord = str_split_fixed(right_breakpoint,":",3)[,2], right_strand = str_split_fixed(right_breakpoint,":",3)[,3],
#            fusion_type = case_when(grepl("INTERCHROMOSOMAL", annots) ~ "Inter\nChromosomal", grepl("OVERLAP", annots) ~ "Overlapping\nNeighbours", grepl("NEIGHBORS", annots) ~ "Neighbours", grepl("LOCAL_INVERSION", annots) ~ "Local\nInversion", grepl("LOCAL_REARRANGEMENT", annots) ~ "Local\nRearrangement", TRUE ~ "Distant\nProximity"),
#            breakpoints = paste(left_breakpoint, right_breakpoint, sep="_")) %>%
#     left_join(metadata, by = "dataset_sample")
#   return(starfusion_tab)
# }
# ipsc_mn_als_datasets.paired_end.metadata = ipsc_mn_als_datasets.paired_end.metadata %>% mutate(starfusion_path = paste0("/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/rnafusion/starfusion/",dataset_sample,".starfusion.abridged.coding_effect.tsv"))
# ipsc_mn_als_datasets.starfusion = read_starfusion(ipsc_mn_als_datasets.paired_end.metadata)
# ipsc_mn_als_datasets.starfusion
# write_csv(ipsc_mn_als_datasets.starfusion, here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv"))
ipsc_mn_als_datasets.starfusion = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))
ipsc_mn_als_datasets.starfusion

# left_join multiqc stats
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
multiqc_rseqc_read_distribution.tsv

ipsc_mn_als_datasets.starfusion = ipsc_mn_als_datasets.starfusion %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion

fusion_schematic = magick::image_read(here(proj_path,"rnafusion/gene fusion schematic.png"))
fusion_schematic.gg = ggdraw() + draw_image(fusion_schematic)
fusion_schematic.gg

```

### ALS vs CTRL

```{r}
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset)
ipsc_mn_als_datasets.paired_end.metadata %>% count(condition) # 306 ALS, 90 CTRL
ipsc_mn_als_datasets.paired_end.metadata %>% count(mutation)

# Venn overlap unqiue gene fusions
ipsc_mn_als_datasets.starfusion %>% distinct(fusion_name) # 353
ipsc_mn_als_datasets.starfusion %>% distinct(fusion_type) # 6
ipsc_mn_als_datasets.starfusion %>% group_by(condition) %>% distinct(fusion_name) %>% count(condition)
# condition     n
#   <chr>     <int>
# 1 als         292
# 2 ctrl        152
ipsc_mn_als_datasets.starfusion.als_events <- ipsc_mn_als_datasets.starfusion %>% filter(condition == "als") %>% distinct(fusion_name) %>% pull(fusion_name)
ipsc_mn_als_datasets.starfusion.ctrl_events <- ipsc_mn_als_datasets.starfusion %>% filter(condition == "ctrl") %>% distinct(fusion_name) %>% pull(fusion_name)

# # ggvenn unique fusion events in CTRL & ALS
ipsc_mn_als_datasets.starfusion.als_ctrl_events <- list("ALS" = ipsc_mn_als_datasets.starfusion.als_events, "CTRL" = ipsc_mn_als_datasets.starfusion.ctrl_events)
ipsc_mn_als_datasets.starfusion.als_ctrl_events_venn <- ggvenn(ipsc_mn_als_datasets.starfusion.als_ctrl_events, auto_scale = TRUE, fill_color = c("firebrick2", "dodgerblue2"), set_name_color = c("firebrick2", "dodgerblue2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5) + labs(subtitle = "Unique Gene Fusions")  + 
  theme(plot.subtitle = element_text(hjust = 0.5, size = 9), plot.margin = unit(c(0, -1, 0, -1), "lines")) #+ theme_oz()
ipsc_mn_als_datasets.starfusion.als_ctrl_events_venn


### ALS vs CTRL unique fusion events
ipsc_mn_als_datasets.starfusion.n = ipsc_mn_als_datasets.starfusion %>% 
  add_count(dataset_sample) %>% # count number of fusions per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, dataset, total_reads) #keep 1 row per sample

# add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.samples_no_fusions = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% ipsc_mn_als_datasets.starfusion.n$dataset_sample] 
ipsc_mn_als_datasets.starfusion.samples_no_fusions # "smith_tardbp_1" "smith_ctrl_1"  
ipsc_mn_als_datasets.paired_end.metadata.no_fusions = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.samples_no_fusions) %>% 
  left_join(multiqc_rseqc_read_distribution.tsv) %>%
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), n = 0) %>% select(dataset_sample, n, CONDITION, dataset, total_reads)
ipsc_mn_als_datasets.starfusion.n = bind_rows(ipsc_mn_als_datasets.starfusion.n, ipsc_mn_als_datasets.paired_end.metadata.no_fusions)

glm_starfusion = glm(data = ipsc_mn_als_datasets.starfusion.n, n ~ CONDITION + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           2.121e+00  3.330e-01   6.369 1.90e-10 ***
# CONDITIONALS                          1.327e-01  4.282e-02   3.098  0.00195 ** 

# Violin
ipsc_mn_als_datasets.starfusion.n.stats = tibble(group1 = "CTRL", group2 = "ALS", p.signif = "**", y.position = 40, xmin = 1, xmax = 2)
ipsc_mn_als_datasets.starfusion.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.n, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,42), dpi = 300, ylabel = "Number of Fusion Events", title = "All Fusion types") +
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.n.stats, tip.length = 0.01, size = 4)
ipsc_mn_als_datasets.starfusion.violin

ipsc_mn_als_datasets.starfusion.n %>% my_summarise(n, CONDITION) %>% mutate(IQR = Q3_n - Q1_n)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <chr>      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 ALS        10.6        10  5.97     7    13     0    39  3251   306     6
# 2 CTRL        8.17        8  4.33     5    10     0    19   735    90     5


# # PCA of FFPM
# ipsc_mn_als_datasets.starfusion.vsd = ipsc_mn_als_datasets.starfusion %>% mutate(breakpoints = paste(left_breakpoint, right_breakpoint, sep = "_")) %>% select(breakpoints, dataset_sample, ffpm) %>% pivot_wider(names_from = dataset_sample, values_from = ffpm) %>% replace(is.na(.),0)  # samples = dataset_sample, rows = fusion_name, values = ffpm
# ipsc_mn_als_datasets.starfusion.vsd
# library(ggfortify)
# ipsc_mn_als_datasets.starfusion.pca <- prcomp(select(ipsc_mn_als_datasets.starfusion.vsd,-breakpoints), scale. = TRUE)
# autoplot(ipsc_mn_als_datasets.starfusion.pca, data = ipsc_mn_als_datasets.starfusion, colour = "condition")

# # break entropy
# ipsc_mn_als_datasets.starfusion %>% glimpse
# ipsc_mn_als_datasets.starfusion.violin <- violin_plot(ipsc_mn_als_datasets.starfusion, color_by = "CONDITION", continuous = "left_break_entropy", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, stats_test = "t_test", arrow_labels = FALSE, ylims = c(0.9,2.1), dpi = 300, ylabel = "Number of Unique Fusion Events")
# ipsc_mn_als_datasets.starfusion.violin
# ipsc_mn_als_datasets.starfusion.violin <- violin_plot(ipsc_mn_als_datasets.starfusion, color_by = "CONDITION", continuous = "right_break_entropy", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, stats_test = "t_test", arrow_labels = FALSE, ylims = c(0.9,2.1), dpi = 300, ylabel = "Number of Unique Fusion Events")
# ipsc_mn_als_datasets.starfusion.violin
```



## Merge

```{r}
ipsc_mn_als_datasets.gene_fusion_merged = plot_grid(
  plot_grid(answerals.vcfannotate.total.violin, answerals.vcfannotate.variant_types.violin, ncol = 2, rel_widths = c(0.5,1.2), labels = "auto"),
  answerals.base_substitutions.violin,
  plot_grid(fusion_schematic.gg, ipsc_mn_als_datasets.starfusion.als_ctrl_events_venn, ipsc_mn_als_datasets.starfusion.violin, ncol = 3, rel_widths = c(0.9,0.9,1), labels = c("d","e","f")),
  labels = c("","c",""), nrow = 3, rel_heights = c(1,0.9,1))
# ipsc_mn_als_datasets.gene_fusion_merged
ggsave(ipsc_mn_als_datasets.gene_fusion_merged, filename = here(proj_path,"rnafusion/ipsc_mn_als_datasets.gene_fusion_merged.png"), height = 7, width = 9)

```




# Table S8 RNA variants

```{r}
ipsc_mn_als_datasets.vcfannotate = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(dataset_sample)%>% add_tally(n, name = "total") %>% ungroup %>% select(dataset_sample, n, type, variant_type, total, everything()) # add a column of sum of variants per sample

# left_join multiqc stats
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)

ipsc_mn_als_datasets.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% left_join(multiqc_rseqc_read_distribution.tsv)

answerals.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% filter(dataset == "answerals")

answerals.vcfannotate.s8 = answerals.vcfannotate %>% select(dataset, sample, n, variant_type, condition, mutation, Total = total) %>% 
  pivot_wider(names_from = "variant_type", values_from = "n", values_fill = 0) %>% 
  select(dataset, sample, condition, mutation, Total, SNP, Insertion, Deletion) %>% #, Complex
  arrange(condition)
answerals.vcfannotate.s8 %>% print(n=Inf)
write_csv(answerals.vcfannotate.s8, here(proj_path,"rnavar/answerals.vcfannotate.s8.csv"))

```


# Table S9-10 RNA fusions

RNA fusion events in each sample

```{r}
ipsc_mn_als_datasets.starfusion = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))

# left_join multiqc stats
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)

ipsc_mn_als_datasets.starfusion = ipsc_mn_als_datasets.starfusion %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.n = ipsc_mn_als_datasets.starfusion %>% 
  add_count(dataset_sample) %>% # count number of fusions per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset, sample, condition, fusion_number = n, total_reads, dataset_sample, mutation) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.samples_no_fusions = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% ipsc_mn_als_datasets.starfusion.n$dataset_sample] 
ipsc_mn_als_datasets.starfusion.samples_no_fusions # "smith_tardbp_1" "smith_ctrl_1"  
ipsc_mn_als_datasets.paired_end.metadata.no_fusions = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.samples_no_fusions) %>% 
  left_join(multiqc_rseqc_read_distribution.tsv) %>%
  mutate(fusion_number = 0) %>% select(dataset_sample, sample, fusion_number, condition, dataset, total_reads, mutation)
ipsc_mn_als_datasets.starfusion.n = bind_rows(ipsc_mn_als_datasets.starfusion.n, ipsc_mn_als_datasets.paired_end.metadata.no_fusions)

# number of fusions
ipsc_mn_als_datasets.starfusion.n %>% glimpse
ipsc_mn_als_datasets.starfusion.table_s11 = ipsc_mn_als_datasets.starfusion.n %>% select(dataset, sample, condition, mutation, fusion_number, total_reads)
ipsc_mn_als_datasets.starfusion.table_s11
write_csv(ipsc_mn_als_datasets.starfusion.table_s11, here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.table_s11.csv"))

## Fusions with greatest frequency
ipsc_mn_als_datasets.starfusion.als.count = ipsc_mn_als_datasets.starfusion %>% filter(condition == "als") %>% count(breakpoints, name = "als_yes")
ipsc_mn_als_datasets.starfusion.ctrl.count = ipsc_mn_als_datasets.starfusion %>% filter(condition == "ctrl") %>% count(breakpoints, name = "ctrl_yes")
ipsc_mn_als_datasets.starfusion.als_ctrl.prop = ipsc_mn_als_datasets.starfusion.als.count %>% full_join(ipsc_mn_als_datasets.starfusion.ctrl.count)  %>% replace(is.na(.),0) %>% 
  mutate(als_total = 360, ctrl_total = 90) 

ipsc_mn_als_datasets.starfusion.als_ctrl.burden = ipsc_mn_als_datasets.starfusion.als_ctrl.prop %>% rowwise() %>% 
  mutate(p_value = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total), 
         odds_ratio = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$estimate[1],
         lower_ci = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$conf.int[1], 
         upper_ci = ClusterBurden::burden_test(n1 = als_yes, n2 = ctrl_yes, ss1 = als_total, ss2 = ctrl_total, pval = FALSE)$conf.int[2])
ipsc_mn_als_datasets.starfusion.als_ctrl.burden

als_ctrl.voila = read_voila_het(here(proj_path,"splicing/majiq_polyA/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
als_ctrl.voila.splicing_genes = als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) %>% pull(gene_name) # 264

# add back gene fusion names
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.s9 = ipsc_mn_als_datasets.starfusion.als_ctrl.burden %>% left_join(distinct(select(ipsc_mn_als_datasets.starfusion, breakpoints, fusion_name, fusion_type))) %>% arrange(p_value) %>% 
  mutate(gene1 = str_split_fixed(fusion_name, "--",2)[,1], gene2 = str_split_fixed(fusion_name, "--",2)[,2], alterated_splicing = case_when(gene1 %in% als_ctrl.voila.splicing_genes | gene2 %in% als_ctrl.voila.splicing_genes ~ "yes", TRUE ~ "")) %>%
  select(fusion_name, breakpoints, fusion_type, als_number = als_yes, ctrl_number = ctrl_yes, odds_ratio, lower_ci, upper_ci, p_value, alterated_splicing)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.s9
write_csv(ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.s9, here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.s9.csv"))


ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split = ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.s9 %>% mutate(gene1 = str_split_fixed(fusion_name, "--",2)[,1], gene2 = str_split_fixed(fusion_name, "--",2)[,2])
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split

ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene1 %in% ALS_genes)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene2 %in% ALS_genes)

# overlap fusions with degs
ipsc_mn_als_datasets.degs = ipsc_mn_als_datasets$res %>% filter(padj < 0.05) %>% pull(gene_name)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene1 %in% ipsc_mn_als_datasets.degs)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene2 %in% ipsc_mn_als_datasets.degs)

# overlap fusions with splice events
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene1 %in% als_ctrl.voila.splicing_genes)
ipsc_mn_als_datasets.starfusion.als_ctrl.burden_table.genes_split %>% filter(gene2 %in% als_ctrl.voila.splicing_genes)

```


# Fig S13 Variants & fusion types & genetic groups

## Variants

### Base substitution proportions

```{r}
# Proportions of base substitutions
# base_substitutions.als_ctrl.sd <- plot_spectrum(type_occurrences, by = answerals.metadata$condition, CT = FALSE, indv_points = TRUE, error_bars = "stdev") # error bars SD rather than 95% CI
# base_substitutions.als_ctrl.sd

tb_per_sample <- type_occurrences[, seq_len(6)] %>% tibble::rownames_to_column("sample") %>% 
        dplyr::mutate(condition = factor(toupper(answerals.metadata$condition), levels = c("CTRL","ALS"))) %>% tidyr::pivot_longer(c(-sample, 
        -condition), names_to = "variable", values_to = "nmuts") %>% 
        dplyr::group_by(sample) %>% dplyr::mutate(value = nmuts/sum(nmuts)) %>% 
        dplyr::ungroup() %>% dplyr::mutate(sub_type = stringr::str_remove(variable, 
        " .*"), variable = factor(variable, levels = unique(variable)))
tb <- tb_per_sample %>% dplyr::group_by(condition, variable) %>% dplyr::summarise(sub_type = sub_type[[1]], 
        mean = mean(value), stdev = stats::sd(value), total_individuals = sum(value), 
        total_mutations = sum(nmuts)) %>% dplyr::mutate(total_individuals = sum(total_individuals), 
        total_mutations = sum(total_mutations)) %>% dplyr::mutate(sem = stdev/sqrt(total_individuals), 
        error_95 = ifelse(total_individuals > 1, qt(0.975, df = total_individuals - 
            1) * sem, NA)) %>% dplyr::ungroup() %>% dplyr::mutate(total_mutations = prettyNum(total_mutations, 
        big.mark = ","), total_mutations = paste("No. mutations = ", 
        total_mutations), error_pos = mean)
tb_per_sample = left_join(tb_per_sample, tb[, c("condition", "sub_type", "total_mutations")], by = c("condition", "sub_type"))

# stats from t_test
answerals.Substitution.prop.stat <- tb_per_sample %>% group_by(sub_type) %>% t_test(value ~ condition) %>% add_significance() %>% add_xy_position()
answerals.Substitution.prop.stat
# sub_type .y.   group1 group2    n1    n2 statistic    df      p p.signif y.position groups        xmin  xmax
#   <chr>    <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>  <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 C>A      value CTRL   ALS       42   238    -0.649  62.2 0.519  ns           0.130  <chr [2]>        1     2
# 2 C>G      value CTRL   ALS       42   238     1.64   57.8 0.107  ns           0.144  <chr [2]>        1     2
# 3 C>T      value CTRL   ALS       42   238     1.74   62.0 0.0863 ns           0.390  <chr [2]>        1     2
# 4 T>A      value CTRL   ALS       42   238    -1.39   56.9 0.169  ns           0.0952 <chr [2]>        1     2
# 5 T>C      value CTRL   ALS       42   238    -1.14   64.5 0.26   ns           0.505  <chr [2]>        1     2
# 6 T>G      value CTRL   ALS       42   238     1.58   53.9 0.121  ns           0.118  <chr [2]>        1     2
base_substitutions.als_ctrl.sd.type <- ggplot(data = tb_per_sample, aes(x = condition, y = value, colour = condition)) + 
   # geom_bar(stat = "identity") + 
    ggrastr::rasterise(geom_violin(adjust = 1), dpi = 300) +
    ggrastr::rasterise(geom_sina(size = 0.5), dpi = 300) +
    ggrastr::rasterise(geom_boxplot(aes(fill = condition), colour = "black", width=0.2, alpha = 0.5, outlier.shape = NA), dpi = 300) + 
    scale_colour_manual(values = c("dodgerblue2", "firebrick2")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
    theme_bw() + xlab("") + ylab("Relative proportions") + 
    theme_oz() + theme(panel.spacing.x = unit(0.5, "lines"), legend.position = "none") + 
   # geom_jitter(data = tb_per_sample, aes(y = value), height = 0, width = 0.3, shape = 21, colour = "grey23") +
    facet_wrap(~ sub_type, scales = "free", nrow = 1)
base_substitutions.als_ctrl.sd.type
# sub_type .y.   group1 group2    n1    n2 statistic    df      p
# * <chr>    <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>  <dbl>
# 1 C>A      value ctrl   als       42   238    -0.649  62.2 0.519 
# 2 C>G      value ctrl   als       42   238     1.64   57.8 0.107 
# 3 C>T      value ctrl   als       42   238     1.74   62.0 0.0863
# 4 T>A      value ctrl   als       42   238    -1.39   56.9 0.169 
# 5 T>C      value ctrl   als       42   238    -1.14   64.5 0.26  
# 6 T>G      value ctrl   als       42   238     1.58   53.9 0.121 

plot_spectrum.type = function (type_occurrences, CT = FALSE, by = NA, indv_points = FALSE, 
    error_bars = c("95%_CI", "stdev", "SEM", "none"), colors = NA, 
    legend = TRUE, condensed = FALSE) {
    value <- nmuts <- sub_type <- variable <- error_pos <- stdev <- total_mutations <- NULL
    x <- total_individuals <- sem <- error_95 <- NULL
    error_bars <- match.arg(error_bars)
    # if (.is_na(colors)) {
    #     colors <- COLORS7
    # }
    if (length(colors) != 7) {
        stop("Colors parameter: supply color vector with length 7")
    }
    if (CT == FALSE) {
        type_occurrences <- type_occurrences[, seq_len(6)]
    }
    else {
        type_occurrences <- type_occurrences[, c(1, 2, 8, 7, 
            4, 5, 6)]
    }
    # if (.is_na(by)) {
    #     by <- "all"
    # }
    tb_per_sample <- type_occurrences %>% tibble::rownames_to_column("sample") %>% 
        dplyr::mutate(by = by) %>% tidyr::pivot_longer(c(-sample, 
        -by), names_to = "variable", values_to = "nmuts") %>% 
        dplyr::group_by(sample) %>% dplyr::mutate(value = nmuts/sum(nmuts)) %>% 
        dplyr::ungroup() %>% dplyr::mutate(sub_type = stringr::str_remove(variable, 
        " .*"), variable = factor(variable, levels = unique(variable)))
    tb <- tb_per_sample %>% dplyr::mutate(by = factor(by, levels = unique(by))) %>% 
        dplyr::group_by(by, variable) %>% dplyr::summarise(sub_type = sub_type[[1]], 
        mean = mean(value), stdev = stats::sd(value), total_individuals = sum(value), 
        total_mutations = sum(nmuts)) %>% dplyr::mutate(total_individuals = sum(total_individuals), 
        total_mutations = sum(total_mutations)) %>% dplyr::mutate(sem = stdev/sqrt(total_individuals), 
        error_95 = ifelse(total_individuals > 1, qt(0.975, df = total_individuals - 
            1) * sem, NA)) %>% dplyr::ungroup() %>% dplyr::mutate(total_mutations = prettyNum(total_mutations, 
        big.mark = ","), total_mutations = paste("No. mutations = ", 
        total_mutations), error_pos = mean)
    if (CT == FALSE) {
        colors <- colors[c(1, 2, 3, 5, 6, 7)]
    }
    else {
        CpG <- which(tb$variable == "C>T at CpG")
        other <- which(tb$variable == "C>T other")
        tb$error_pos[CpG] <- tb$error_pos[other] + tb$error_pos[CpG]
        CpG <- which(tb_per_sample$variable == "C>T at CpG")
        other <- which(tb_per_sample$variable == "C>T other")
        tb_per_sample$value[CpG] <- tb_per_sample$value[other] + 
            tb_per_sample$value[CpG]
    }
    if (condensed == TRUE) {
        width <- 1
        spacing <- 0
    }
    else {
        width <- 0.9
        spacing <- 0.5
    }
    plot <- ggplot(data = tb, aes(x = total_mutations, y = mean, fill = variable, 
        group = total_mutations, width = width)) + geom_bar(stat = "identity") + 
        scale_fill_manual(values = colors, name = "Point mutation type") + 
        theme_bw() + xlab("") + ylab("Relative contribution") + 
        theme(axis.ticks = element_blank(), axis.text.x = element_blank(), 
            panel.grid.major.x = element_blank(), panel.spacing.x = unit(spacing, 
                "lines"))
    if (indv_points == TRUE) {
        tb_per_sample <- dplyr::left_join(tb_per_sample, tb[, 
            c("by", "variable", "total_mutations")], by = c("by", 
            "variable"))
        plot <- plot + geom_jitter(data = tb_per_sample, aes(y = value), 
            height = 0, width = 0.3, shape = 21, colour = "grey23")
    }
    if (sum(is.na(tb$stdev)) > 0 & error_bars != "none") {
        warning("No error bars can be plotted, because there is only one sample per mutation spectrum.\n              Use the argument: `error_bars = 'none'`, if you want to avoid this warning.", 
            call. = FALSE)
    }
    else {
        if (error_bars == "stdev") {
            plot <- plot + geom_errorbar(aes(ymin = error_pos - 
                stdev, ymax = error_pos + stdev), width = 0.2)
        }
        else if (error_bars == "95%_CI") {
            plot <- plot + geom_errorbar(aes(ymin = error_pos - 
                error_95, ymax = error_pos + error_95), width = 0.2)
        }
        else if (error_bars == "SEM") {
            plot <- plot + geom_errorbar(aes(ymin = error_pos - 
                sem, ymax = error_pos + sem), width = 0.2)
        }
    }
    if (length(by) == 1) {
        plot <- plot + facet_wrap(~ sub_type)
    }
    else {
        plot <- plot + facet_wrap(by ~  sub_type)
    }
    if (legend == FALSE) {
        plot <- plot + guides(fill = "none")
    }
    return(plot)
}

#plot base substitutions
base_substitutions <- plot_spectrum(type_occurrences) # contribution of each base substitution over all samples
base_substitutions
p2 <- plot_spectrum(type_occurrences, CT = TRUE) # with CpG site distinction
p3 <- plot_spectrum(type_occurrences, CT = TRUE, indv_points = TRUE) # plot individual samples on barcharts
p4 <- plot_spectrum(type_occurrences, by = answerals.metadata$condition, CT = TRUE, indv_points = TRUE) # add facet by condition
p5 <- plot_spectrum(type_occurrences, by = answerals.metadata$condition, CT = TRUE, indv_points = TRUE, error_bars = "stdev") # error bars SD rather than 95% CI
p5
base_substitutions.als_ctrl.cpg.sd <- plot_spectrum(type_occurrences, by = answerals.metadata$condition, CT = TRUE, indv_points = TRUE, error_bars = "stdev") # error bars SD rather than 95% CI
base_substitutions.als_ctrl.cpg.sd

base_substitutions.als_ctrl.sd <- plot_spectrum(type_occurrences, by = answerals.metadata$condition, CT = FALSE, indv_points = TRUE, error_bars = "stdev") # error bars SD rather than 95% CI
base_substitutions.als_ctrl.sd

# # 96 mutational profile
# mut_mat =readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/mut_mat.RDS"))
# mut_mat <- mut_matrix(vcf_list = grl, ref_genome = ref_genome) # make 96 trinucleotide mutation count matrix
# head(mut_mat)
# plot_96_profile(mut_mat[, c(1, 7)]) 

# heatmap of base substitutions 
mut_mat_ext_context = readRDS(here(shared_path, "answerals/rnavar/mutationalpatterns/mut_mat_ext_context.RDS")
mut_mat_ext_context.heatmap = plot_profile_heatmap(mut_mat_ext_context, by = answerals.metadata$condition)
mut_mat_ext_context.heatmap

# riverplot of base substitutions
mut_mat_ext_context.river = plot_river(mut_mat_ext_context, by = answerals.metadata$condition)
mut_mat_ext_context.river

# plot indel types
plot_indel_contexts(indel_counts, condensed = TRUE)
plot_main_indel_contexts(indel_counts) # plot raw indel types - does not account for number of repeat units

# DBS doublet based substitutions
plot_dbs_contexts(dbs_counts, same_y = TRUE)
plot_main_dbs_contexts(dbs_counts, same_y = TRUE)

# MBS multiplet based substitutions
plot_mbs_contexts(mbs_counts, same_y = TRUE)

# Mutational signatures

# Genomic distribution
chromosomes <- seqnames(get(ref_genome))[1:22]
plot_rainfall(snv_grl[[1]], title = names(snv_grl[1]), chromosomes = chromosomes, cex = 1.5, ylim = 1e+09) # mutation types and intermutation distance

```

### Genetic subgroups

```{r}
answerals.metadata %>% count(mutation)

answerals.vcfannotate.mutation = answerals.vcfannotate %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic",mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "C9orf72", "SOD1"))) %>% #mutation == "fus" ~ "FUS","FUS",  mutation == "tardbp" ~ "TARDBP", "TARDBP", 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, total, genetic_group, CONDITION, total_reads, mutation) # keep 1 row per sample

answerals.metadata_genetic_subgroup = answerals.metadata %>% filter(mutation %in%  c("ctrl", "iso", "sporadic", "sod1", "c9orf72")) #"fus", "tardbp", 
answerals.vcfannotate.samples_no_variants = answerals.metadata_genetic_subgroup$dataset_sample[!answerals.metadata_genetic_subgroup$dataset_sample %in% answerals.vcfannotate.mutation$dataset_sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

answerals.vcfannotate.mutation.totals = answerals.vcfannotate.mutation %>% distinct(dataset_sample, .keep_all = TRUE) #keep 1 row per sample for total number of variants
answerals.vcfannotate.mutation.totals %>% glimpse

# generalised linear model spline

## sporadic
answerals.vcfannotate.mutation.totals.glm.sporadic = glm(data=filter(answerals.vcfannotate.mutation.totals, genetic_group %in% c("Sporadic", "CTRL")), total ~ genetic_group + rms::rcs(total_reads, 3), family = poisson)
summary(answerals.vcfannotate.mutation.totals.glm.sporadic)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.064e+01  3.849e-03 2764.35   <2e-16 ***
# genetic_groupSporadic                 1.365e-02  7.146e-04   19.11   <2e-16 ***

## c9orf72
answerals.vcfannotate.mutation.totals.glm.c9orf72 = glm(data=filter(answerals.vcfannotate.mutation.totals, genetic_group %in% c("C9orf72", "CTRL")), total ~ genetic_group + rms::rcs(total_reads, 3), family = poisson)
summary(answerals.vcfannotate.mutation.totals.glm.c9orf72)
# Coefficients:
#                                        Estimate Std. Error  z value Pr(>|z|)    
# (Intercept)                           1.064e+01  7.903e-03 1346.465   <2e-16 ***
# genetic_groupC9orf72                 -2.788e-03  1.115e-03   -2.501   0.0124 *  

## SOD1
answerals.vcfannotate.mutation.totals.glm.SOD1 = glm(data=filter(answerals.vcfannotate.mutation.totals, genetic_group %in% c("SOD1", "CTRL")), total ~ genetic_group + rms::rcs(total_reads, 3), family = poisson) # + rms::rcs(total_reads, 3)
summary(answerals.vcfannotate.mutation.totals.glm.SOD1)
# Coefficients:
#                                        Estimate Std. Error  z value Pr(>|z|)    
# (Intercept)                           1.061e+01  8.616e-03 1231.631  < 2e-16 ***
# genetic_groupSOD1                    -9.399e-03  1.662e-03   -5.655 1.56e-08 ***


# Violin total variant events
answerals.vcfannotate.mutation.stat <- answerals.vcfannotate.mutation %>% wilcox_test(total ~ genetic_group) %>% add_significance() %>% add_xy_position() %>% filter(group1 == "CTRL") %>%
  mutate(p.adj.signif = case_when(group2 == "Sporadic" ~ "****", TRUE ~ p.adj.signif), y.position = case_when(group2 == "Sporadic" ~ 75000, TRUE ~ y.position))
answerals.vcfannotate.mutation.stat

answerals.vcfannotate.mutation.violin <- violin_plot(answerals.vcfannotate.mutation.totals, color_by = "genetic_group", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(50000,76000), dpi = 300, ylabel = "Number of Variants") +
  stat_pvalue_manual(answerals.vcfannotate.mutation.stat, tip.length = 0.01, size = 4, hide.ns = TRUE)
answerals.vcfannotate.mutation.violin

answerals.vcfannotate.mutation %>% my_summarise(total, genetic_group) %>% mutate(IQR = Q3_total - Q1_total)
# genetic_group mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <fct>              <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1 CTRL              57615.       57278.    3394.   55198.   58946.     53050     69191   2419840      42 3748.
# 2 Sporadic          59177.       58566     3435.   57063.   60632.     52833     70740  11835412     200 3570.
# 3 C9orf72           57419.       56842     2507.   56066    58166      53495     62532   1205807      21 2100 
# 4 SOD1              58716.       57905     2367.   57448.   60129.     55629     63145    469730       8 2682.


```


## Fusions

### Fusion types

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type_n = ipsc_mn_als_datasets.starfusion %>% 
  add_count(dataset_sample, fusion_type) %>% # count number of fusion types per sample
  distinct(dataset_sample, fusion_type, .keep_all = TRUE) %>% select(dataset_sample, fusion_type, n, condition, dataset, total_reads) #keep 1 row per sample fusion type
ipsc_mn_als_datasets.starfusion.fusion_type_n

# fusions types detected in X samples in ALS & CTRL:
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Neighbours") %>% count(condition) # 89/90 98.9% 302/306 98.7%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Local\nRearrangement") %>% count(condition) # 46/90 51.1% 253/306 82.7%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Distant\nProximity") %>% count(condition) # 64/90 71.1% 254/306 83.0%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Overlapping\nNeighbours") %>% count(condition) #37/90 41.1% 197/306 64.4%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Inter\nChromosomal") %>% count(condition) # 27/90 30% 105/306 34%
filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Local\nInversion") %>% count(condition) #8/90 8.9% 5/306 1.6%

# add samples with 0 fusions for each fusion type
samples_no_neighbours = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Neighbours"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_neighbours = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_neighbours) %>% mutate(fusion_type = "Neighbours", n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type)
ipsc_mn_als_datasets.paired_end.metadata.no_neighbours

samples_no_local_rearrangement = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Local\nRearrangement"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_local_rearrangement = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_local_rearrangement) %>% mutate(fusion_type = "Local\nRearrangement", n = 0) %>%
  select(dataset_sample, n, condition, fusion_type)
ipsc_mn_als_datasets.paired_end.metadata.no_local_rearrangement

samples_no_local_inversion = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Local\nInversion"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_local_inversion = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_local_inversion) %>% mutate(fusion_type = "Local\nInversion", n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type)
ipsc_mn_als_datasets.paired_end.metadata.no_local_inversion

samples_no_overlapping_neighbours = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Overlapping\nNeighbours"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_overlapping_neighbours = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_overlapping_neighbours) %>% mutate(fusion_type = "Overlapping\nNeighbours", n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type)
ipsc_mn_als_datasets.paired_end.metadata.no_overlapping_neighbours

samples_no_distant_proximity = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Distant\nProximity" ),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_distant_proximity = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_distant_proximity) %>% mutate(fusion_type = "Distant\nProximity" , n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type)
ipsc_mn_als_datasets.paired_end.metadata.no_distant_proximity

samples_no_inter_chromosomal = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% pull(filter(ipsc_mn_als_datasets.starfusion.fusion_type_n,fusion_type == "Inter\nChromosomal"),dataset_sample)]
ipsc_mn_als_datasets.paired_end.metadata.no_inter_chromosomal = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% samples_no_inter_chromosomal) %>% mutate(fusion_type = "Inter\nChromosomal", n = 0) %>% 
  select(dataset_sample, n, condition, fusion_type)
ipsc_mn_als_datasets.paired_end.metadata.no_inter_chromosomal

ipsc_mn_als_datasets.starfusion.fusion_type_n = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type_n, ipsc_mn_als_datasets.paired_end.metadata.no_neighbours, ipsc_mn_als_datasets.paired_end.metadata.no_local_rearrangement, ipsc_mn_als_datasets.paired_end.metadata.no_local_inversion, 
                                                          ipsc_mn_als_datasets.paired_end.metadata.no_overlapping_neighbours, ipsc_mn_als_datasets.paired_end.metadata.no_distant_proximity, ipsc_mn_als_datasets.paired_end.metadata.no_inter_chromosomal) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), 
         fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))
ipsc_mn_als_datasets.starfusion.fusion_type_n

ipsc_mn_als_datasets.starfusion.fusion_type_n %>% my_summarise(n, CONDITION, fusion_type) %>% mutate(IQR = Q3_n - Q1_n)
# CONDITION fusion_type               mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#    <fct>     <fct>                      <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
#  1 CTRL      "Local\nInversion"        0.1           0 0.337     0  0        0     2     9    90  0   
#  2 CTRL      "Local\nRearrangement"    1.4           1 1.77      0  2.75     0     7   126    90  2.75
#  3 CTRL      "Neighbours"              4.2           4 2.43      3  5        0    12   378    90  2   
#  4 CTRL      "Overlapping\nNeighbours" 0.422         0 0.519     0  1        0     2    38    90  1   
#  5 CTRL      "Distant\nProximity"      1.47          1 1.38      0  2        0     6   132    90  2   
#  6 CTRL      "Inter\nChromosomal"      0.578         0 1.36      0  1        0     8    52    90  1   
#  7 ALS       "Local\nInversion"        0.0196        0 0.161     0  0        0     2     6   306  0   
#  8 ALS       "Local\nRearrangement"    2.29          2 1.80      1  3        0    10   701   306  2   
#  9 ALS       "Neighbours"              5.29          4 3.35      3  7        0    21  1619   306  4   
# 10 ALS       "Overlapping\nNeighbours" 0.686         1 0.555     0  1        0     3   210   306  1   
# 11 ALS       "Distant\nProximity"      1.71          2 1.30      1  2        0     7   523   306  1   
# 12 ALS       "Inter\nChromosomal"      0.627         0 1.96      0  1        0    22   192   306  1   


# For count data use glm(count ~ condition, family=poisson) which gives estimates and p-values of the underlying 'rate's of fusion per sample 
### Neighbours
glm_starfusion_neighbours = glm(data = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Neighbours"), n ~ CONDITION + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_neighbours)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.317e+00  4.619e-01   2.851  0.00436 ** 
# CONDITIONALS                          9.019e-02  5.977e-02   1.509  0.13132    

### Local_Inversion
glm_starfusion_Local_Inversion = glm(data = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Local\nInversion"), n ~ CONDITION + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_Local_Inversion)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)
# (Intercept)                          -5.246e-01  7.747e+00  -0.068    0.946
# CONDITIONALS                          4.192e-02  6.737e-01   0.062    0.950

### Local_Rearrangement
glm_starfusion_Local_Rearrangement = glm(data = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Local\nRearrangement"), n ~ CONDITION + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_Local_Rearrangement)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.761e+00  4.389e-01   4.013    6e-05 ***
# CONDITIONALS                         -9.295e-03  9.826e-02  -0.095  0.92463    

### Overlapping_Neighbours
glm_starfusion_Overlapping_Neighbours = glm(data = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Overlapping\nNeighbours"), n ~ CONDITION + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_Overlapping_Neighbours)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)
# (Intercept)                          -2.462e-01  9.241e-01  -0.266    0.790
# CONDITIONALS                          4.034e-02  1.846e-01   0.218    0.827

### Distant_Proximity
glm_starfusion_Distant_Proximity = glm(data = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Distant\nProximity"), n ~ CONDITION + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_Distant_Proximity)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)  
# (Intercept)                           6.691e-01  7.221e-01   0.927   0.3541  
# CONDITIONALS                          8.612e-03  1.021e-01   0.084   0.9328 

### Inter_Chromosomal
glm_starfusion_Inter_Chromosomal = glm(data = filter(ipsc_mn_als_datasets.starfusion.fusion_type_n, fusion_type == "Inter\nChromosomal"), n ~ CONDITION + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion_Inter_Chromosomal)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                          -1.546e-01  1.269e+00  -0.122 0.903002    
# CONDITIONALS                          6.354e-01  1.717e-01   3.700 0.000216 ***


ipsc_mn_als_datasets.starfusion.fusion_type.stat <- ipsc_mn_als_datasets.starfusion.fusion_type_n %>% group_by(fusion_type) %>% t_test(n ~ CONDITION) %>% mutate(p.signif = c("ns","ns","ns","ns","ns","***"), y.position = 24) 
ipsc_mn_als_datasets.starfusion.fusion_type.stat
# fusion_type               .y.   group1 group2    n1    n2 statistic    df         p p.signif y.position
#   <fct>                     <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>     <dbl> <chr>         <dbl>
# 1 "Local\nInversion"        n     CTRL   ALS       90   306     2.19   101. 0.0307    ns                3
# 2 "Local\nRearrangement"    n     CTRL   ALS       90   306    -4.19   147. 0.0000479 ns                3
# 3 "Neighbours"              n     CTRL   ALS       90   306    -3.41   199. 0.00078   ns                3
# 4 "Overlapping\nNeighbours" n     CTRL   ALS       90   306    -4.18   154. 0.0000494 ns                3
# 5 "Distant\nProximity"      n     CTRL   ALS       90   306    -1.48   138. 0.141     ns                3
# 6 "Inter\nChromosomal"      n     CTRL   ALS       90   306    -0.273  209. 0.785     ***               3

ipsc_mn_als_datasets.starfusion.fusion_type.violin <- ggplot(ipsc_mn_als_datasets.starfusion.fusion_type_n, aes(x = CONDITION, y = n, colour = CONDITION)) +
    # ggrastr::rasterise(geom_violin(adjust = 2), dpi = 300) +
    ggrastr::rasterise(geom_sina(size = 0.5), dpi = 300) +
    ggrastr::rasterise(geom_boxplot(aes(fill = CONDITION), colour = "black", width=0.2, alpha = 0.5, outlier.shape = NA), dpi = 300) + 
    scale_colour_manual(values = c("dodgerblue2", "firebrick2")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
    theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
    facet_wrap(~ fusion_type, scales = "free", nrow = 1)  + labs(y = "Number of Fusion Events", x = "") +
    stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.stat, label = "p.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) 
ipsc_mn_als_datasets.starfusion.fusion_type.violin
  
# # histogram
# ipsc_mn_als_datasets.starfusion.fusion_type.histogram = ggplot(ipsc_mn_als_datasets.starfusion.fusion_type_n, aes(x=n, color=CONDITION, fill=CONDITION)) + 
#   geom_histogram(alpha=0.5, position="identity", binwidth = 1)+
#   scale_color_manual(values=c("dodgerblue2", "firebrick2"))+ scale_fill_manual(values=c("dodgerblue2", "firebrick2"))+
#   facet_wrap(~ fusion_type, scales = "free", nrow = 1) + theme_oz() #+ ggtitle("Zeros filled")
# ipsc_mn_als_datasets.starfusion.fusion_type.histogram
```



### Genetic subgroups

Sporadic

```{r}
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "sporadic")

ipsc_mn_als_datasets.starfusion.sporadic = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP"))) %>%
  filter(dataset %in% c("answerals","neurolincs.diMN"), mutation %in% c("ctrl","iso","sporadic"))
ipsc_mn_als_datasets.starfusion.sporadic

ipsc_mn_als_datasets.starfusion.sporadic = ipsc_mn_als_datasets.starfusion.sporadic %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.sporadic

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","neurolincs.diMN"), mutation %in% c("ctrl","iso","sporadic")) %>% count(condition) # 208 ALS, 50 CTRL

ipsc_mn_als_datasets.starfusion.sporadic.n = ipsc_mn_als_datasets.starfusion.sporadic %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group, mutation, dataset, total_reads) #keep 1 row per sample
 # add samples with 0 fusions

glm_starfusion.sporadic = glm(data = ipsc_mn_als_datasets.starfusion.sporadic.n, n ~ genetic_group + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion.sporadic)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           2.545e+00  2.669e-01   9.536  < 2e-16 ***
# genetic_groupSporadic                 6.268e-02  5.030e-02   1.246 0.212678  

# Violin
ipsc_mn_als_datasets.starfusion.sporadic.n.stats = tibble(group1 = "CTRL", group2 = "Sporadic", p.signif = "ns", y.position = 40, xmin = 1, xmax = 2)
ipsc_mn_als_datasets.starfusion.sporadic.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.sporadic.n, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "firebrick2"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,43), dpi = 300, ylabel = "Number of Fusion Events") +
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.sporadic.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.sporadic.violin
```

c9orf72

```{r}
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "c9orf72")

ipsc_mn_als_datasets.starfusion.c9orf72 = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP"))) %>%
  filter(dataset %in% c("answerals","neurolincs.diMN","catanese","dafinca.c9orf72","neurolincs.iMN","sommer"), mutation %in% c("ctrl","iso","c9orf72"))
ipsc_mn_als_datasets.starfusion.c9orf72

ipsc_mn_als_datasets.starfusion.c9orf72 = ipsc_mn_als_datasets.starfusion.c9orf72 %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.c9orf72

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","neurolincs.diMN","catanese","dafinca.c9orf72","neurolincs.iMN","sommer"), mutation %in% c("ctrl","iso","c9orf72")) %>% count(condition) # 53 ALS, 76 CTRL

ipsc_mn_als_datasets.starfusion.c9orf72.n = ipsc_mn_als_datasets.starfusion.c9orf72 %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group,  mutation, dataset, total_reads) #keep 1 row per sample
 # add samples with 0 fusions

glm_starfusion.c9orf72 = glm(data = ipsc_mn_als_datasets.starfusion.c9orf72.n, n ~ genetic_group + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion.c9orf72)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           2.783e+00  5.216e-01   5.336 9.50e-08 ***
# CONDITIONALS                          2.401e-01  6.037e-02   3.978 6.96e-05 ***

# Violin
ipsc_mn_als_datasets.starfusion.c9orf72.n.stats = tibble(group1 = "CTRL", group2 = "C9orf72", p.signif = "****", y.position = 40, xmin = 1, xmax = 2)
ipsc_mn_als_datasets.starfusion.c9orf72.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.c9orf72.n, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "forestgreen"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,43), dpi = 300, ylabel = "Number of Unique Fusion Events") +
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.c9orf72.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.c9orf72.violin

```

TARDBP

```{r}
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "tardbp")

ipsc_mn_als_datasets.starfusion.tardbp = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP"))) %>%
  filter(dataset %in% c("answerals","dafinca.tardbp","smith"), mutation %in% c("ctrl","iso","tardbp"))
ipsc_mn_als_datasets.starfusion.tardbp

ipsc_mn_als_datasets.starfusion.tardbp = ipsc_mn_als_datasets.starfusion.tardbp %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.tardbp

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","dafinca.tardbp","smith"), mutation %in% c("ctrl","iso","tardbp")) %>% count(condition) # 48 ALS, 10 CTRL

ipsc_mn_als_datasets.starfusion.tardbp.n = ipsc_mn_als_datasets.starfusion.tardbp %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group, mutation, dataset, total_reads) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.paired_end.metadata.tardbp = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","dafinca.tardbp","smith"), mutation %in% c("ctrl","iso","tardbp"))
ipsc_mn_als_datasets.starfusion.samples_no_fusions.tardbp = ipsc_mn_als_datasets.paired_end.metadata.tardbp$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.tardbp$dataset_sample %in% ipsc_mn_als_datasets.starfusion.tardbp.n$dataset_sample] 
ipsc_mn_als_datasets.starfusion.samples_no_fusions.tardbp # "smith_tardbp_1" "smith_ctrl_1"  

ipsc_mn_als_datasets.paired_end.metadata.no_fusions.tardbp = ipsc_mn_als_datasets.paired_end.metadata.tardbp %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.samples_no_fusions.tardbp) %>% 
  left_join(multiqc_rseqc_read_distribution.tsv) %>%
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), n = 0, genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP"))) %>% select(dataset_sample, n, CONDITION, mutation, dataset, total_reads, genetic_group)
ipsc_mn_als_datasets.starfusion.tardbp.n = bind_rows(ipsc_mn_als_datasets.starfusion.tardbp.n, ipsc_mn_als_datasets.paired_end.metadata.no_fusions.tardbp)

glm_starfusion.tardbp = glm(data = ipsc_mn_als_datasets.starfusion.tardbp.n, n ~ genetic_group + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion.tardbp)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)   
# (Intercept)                           3.461e+00  1.578e+00   2.193  0.02829 * 
# CONDITIONALS                          2.857e-01  1.922e-01   1.487  0.13713   

# Violin
ipsc_mn_als_datasets.starfusion.tardbp.n.stats = tibble(group1 = "CTRL", group2 = "TARDBP", p.signif = "ns", y.position = 40, xmin = 1, xmax = 2)
ipsc_mn_als_datasets.starfusion.tardbp.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.tardbp.n, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "purple"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,43), dpi = 300, ylabel = "Number of Fusion Events") +
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.tardbp.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.tardbp.violin

```

fus

```{r}
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "fus")

ipsc_mn_als_datasets.starfusion.fus = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP"))) %>%
  filter(dataset %in% c("catanese","desantis"), mutation %in% c("ctrl","iso","fus"))
ipsc_mn_als_datasets.starfusion.fus

ipsc_mn_als_datasets.starfusion.fus = ipsc_mn_als_datasets.starfusion.fus %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.fus

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("catanese","desantis"), mutation %in% c("ctrl","iso","fus")) %>% count(condition) # 9 ALS, 9 CTRL

ipsc_mn_als_datasets.starfusion.fus.n = ipsc_mn_als_datasets.starfusion.fus %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group, mutation, dataset, total_reads) #keep 1 row per sample
 # add samples with 0 fusions

glm_starfusion.fus = glm(data = ipsc_mn_als_datasets.starfusion.fus.n, n ~ genetic_group + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion.fus)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)  
# (Intercept)                          -2.586e+00  1.819e+00  -1.421   0.1552  
# CONDITIONALS                          1.124e-01  1.842e-01   0.610   0.5416  

# Violin
ipsc_mn_als_datasets.starfusion.fus.n.stats = tibble(group1 = "CTRL", group2 = "FUS", p.signif = "ns", y.position = 40, xmin = 1, xmax = 2)
ipsc_mn_als_datasets.starfusion.fus.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fus.n, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "grey"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,43), dpi = 300, ylabel = "Number of Fusion Events") +
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fus.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fus.violin

```

sod1

```{r}
ipsc_mn_als_datasets.paired_end.metadata %>% count(dataset, mutation) %>% filter(mutation == "sod1")

ipsc_mn_als_datasets.starfusion.sod1 = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")),
         genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "fus" ~ "FUS", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), 
         genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "C9orf72", "FUS", "TARDBP"))) %>%
  filter(dataset %in% c("answerals","neurolincs.diMN", "kiskinis","wang"), mutation %in% c("ctrl","iso","sod1"))
ipsc_mn_als_datasets.starfusion.sod1

ipsc_mn_als_datasets.starfusion.sod1 = ipsc_mn_als_datasets.starfusion.sod1 %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.starfusion.sod1

### ALS vs CTRL fusion events
ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset %in% c("answerals","neurolincs.diMN", "kiskinis","wang"), mutation %in% c("ctrl","iso","sod1")) %>% count(condition) # 208 ALS, 50 CTRL

ipsc_mn_als_datasets.starfusion.sod1.n = ipsc_mn_als_datasets.starfusion.sod1 %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, genetic_group, mutation, dataset, total_reads) #keep 1 row per sample
 # add samples with 0 fusions

glm_starfusion.sod1 = glm(data = ipsc_mn_als_datasets.starfusion.sod1.n, n ~ genetic_group + dataset + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion.sod1)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           2.748e+00  1.073e+00   2.562   0.0104 *  
# CONDITIONALS                          3.339e-01  8.367e-02   3.991 6.59e-05 ***

# Violin
ipsc_mn_als_datasets.starfusion.sod1.n.stats = tibble(group1 = "CTRL", group2 = "SOD1", p.signif = "****", y.position = 40, xmin = 1, xmax = 2)
ipsc_mn_als_datasets.starfusion.sod1.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.sod1.n, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2", "gold2"), plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,43), dpi = 300, ylabel = "Number of Fusion Events") +
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.sod1.n.stats, tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.sod1.violin

```

## Merge

```{r}
answerals.variant_fusion.genetic_groups = plot_grid(
  base_substitutions.als_ctrl.sd.type,
  answerals.vcfannotate.mutation.violin,
  ipsc_mn_als_datasets.starfusion.fusion_type.violin,
  plot_grid(ipsc_mn_als_datasets.starfusion.sporadic.violin, ipsc_mn_als_datasets.starfusion.c9orf72.violin, ipsc_mn_als_datasets.starfusion.sod1.violin, ipsc_mn_als_datasets.starfusion.tardbp.violin, ipsc_mn_als_datasets.starfusion.fus.violin, ncol = 5), 
  nrow = 4, rel_heights = c(1,1,1,1), labels = "auto")
# answerals.variant_fusion.genetic_groups
ggsave(answerals.variant_fusion.genetic_groups, filename = here(proj_path,"rnafusion/answerals.variant_fusion.genetic_groups.png"), height = 9, width = 8.25)

# , #ipsc_mn_als_datasets.starfusion.fusion_type.histogram

```


# *** ADDITIONAL UNUSED ***


# Fusion other tools

```{r}
# test Catanese dataset

# Arriba
c9orf72_1.arriba.fusions.tsv = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/arriba/c9orf72_1.arriba.fusions.tsv"))
c9orf72_1.arriba.fusions.tsv
c9orf72_1.arriba.fusions.tsv %>% glimpse

# FusionCatcher
c9orf72_1.fusioncatcher.fusion_genes.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/fusioncatcher/c9orf72_1.fusioncatcher.fusion-genes.txt"))
c9orf72_1.fusioncatcher.fusion_genes.txt
c9orf72_1.fusioncatcher.fusion_genes.txt %>% glimpse

# kallisto
c9orf72_1.kallisto_quant.fusions.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/kallisto/c9orf72_1.kallisto_quant.fusions.txt"))
c9orf72_1.kallisto_quant.fusions.txt
c9orf72_1.kallisto_quant.fusions.txt %>% glimpse

# pizzly
c9orf72_1.pizzly.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/pizzly/c9orf72_1.pizzly.txt"))
c9orf72_1.pizzly.txt
c9orf72_1.pizzly.txt %>% glimpse

# squid
c9orf72_1.squid.fusions.annotated.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/squid/c9orf72_1.squid.fusions.annotated.txt"))
c9orf72_1.squid.fusions.annotated.txt
c9orf72_1.squid.fusions.annotated.txt %>% glimpse

# squid
c9orf72_1.squid.fusions_sv.txt = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/squid/c9orf72_1.squid.fusions_sv.txt"))
c9orf72_1.squid.fusions_sv.txt

# starfusion
c9orf72_1.starfusion.abridged.coding_effect.tsv = read_tsv(here(shared_path, "public-data/ipsc-mn-c9orf72-fus-catanese-2021/rnafusion/starfusion/c9orf72_1.starfusion.abridged.coding_effect.tsv"))
c9orf72_1.starfusion.abridged.coding_effect.tsv
c9orf72_1.starfusion.abridged.coding_effect.tsv %>% glimpse

read_starfusion = function(metadata){
  print("reading in STAR Fusion files with read_tsv")
  starfusion.tsv = metadata$starfusion_path %>% map(read_tsv, col_types = list(JunctionReadCount="n",SpanningFragCount="n",est_J="n",est_S="n",FFPM="n",LeftBreakEntropy="n",RightBreakEntropy="n",.default="c"))#, show_col_types = FALSE) # ! Can't combine `ExcludedBases` <double> and `ExcludedBases` <character>.
  names(starfusion.tsv) = metadata$sample #dataset_sample # needs to be unique names e.g. dataset_sample
  starfusion_tab <- map_dfr(starfusion.tsv, bind_rows, .id = "sample") %>% rename(fusion_name = '#FusionName') %>% clean_names() %>% #dataset_sample
    mutate(left_chr = str_split_fixed(left_breakpoint,":",3)[,1], left_coord = str_split_fixed(left_breakpoint,":",3)[,2], left_strand = str_split_fixed(left_breakpoint,":",3)[,3],
           right_chr = str_split_fixed(right_breakpoint,":",3)[,1], right_coord = str_split_fixed(right_breakpoint,":",3)[,2], right_strand = str_split_fixed(right_breakpoint,":",3)[,3],
           fusion_type = case_when(grepl("INTERCHROMOSOMAL", annots) ~ "inter_chromosomal", grepl("NEIGHBORS_OVERLAP", annots) ~ "overlapping_neighbors", grepl("NEIGHBORS", annots) ~ "neighbors", grepl("LOCAL_INVERSION", annots) ~ "local_inversion", grepl("LOCAL_REARRANGEMENT", annots) ~ "local_rearrangement", TRUE ~ "distant_proximity")) %>%
    left_join(metadata, by = "sample") #dataset_sample
  return(starfusion_tab)
}
catanese.metadata = catanese.metadata %>% mutate(starfusion_path = paste0(database_dir,"/rnafusion/starfusion/",sample,".starfusion.abridged.coding_effect.tsv"))
catanese.starfusion = read_starfusion(catanese.metadata)
catanese.starfusion


# The intra-chromosomal fusions were further classified into four subtypes: (1) local rearrangements, which were fusions where the genes are in an unexpected order given the strand of each gene in the pair; (2) not close proximity, which encompassed genes >100 kb apart; (3) neighbors, which were fusions that encompassed genes <100 kb apart and did not show evidence of gene orientation rearrangement; and (4) overlapping neighbors, which encompassed genes whose spans overlapped by at least one base pair.

# Violin unique fusion events in CTRL & ALS
catanese.starfusion %>% add_count(sample) %>% violin_plot(color_by = "condition", continuous = "n", stats_test = FALSE)

catanese.starfusion.n = catanese.starfusion %>% add_count(sample)
catanese.starfusion.violin <- violin_plot(catanese.starfusion.n, color_by = "condition", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, stats_test = "t_test", arrow_labels = FALSE, ylims = c(0,20), dpi = 300, ylabel = "Number of Unique Fusion Events")
catanese.starfusion.violin
```


# Fig S15 Fusions genetic subgroups

```{r}
# left_join multiqc stats
multiqc_rseqc_read_distribution.tsv = read_tsv(here(proj_path,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"), show_col_types = FALSE) %>% clean_names() %>% rename(dataset_sample = sample)
multiqc_rseqc_read_distribution.tsv
```



## Fusion Types

Neighbours

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Neighbours") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Neighbours = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Neighbours = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Neighbours) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Neighbours)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.stat$y.position = c(22,21,21,21,21)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.stat
# .y.   group1 group2      n1    n2 statistic     df        p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl>  <dbl>    <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    -3.64  209.   0.000349 0.005 **                 22.4 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -1.23   23.6  0.232    1     ns                 24.0 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9    -0.654   9.52 0.529    1     ns                 25.5 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10     1.53   10.2  0.156    1     ns                 27.1 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53    -1.97   68.6  0.053    0.641 ns                 28.6 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,22), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Neighbours") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL            4.2       4    2.43  3     5        0    12   378    90  2   
# 2 Sporadic        5.40      5    3.03  3     7        0    21  1124   208  4   
# 3 SOD1            5         5    2.54  3     5.75     2    13    90    18  2.75
# 4 FUS             4.78      5    2.54  3     6        2     9    43     9  3   
# 5 TARDBP          2.6       1.5  3.20  0.25  3        0    10    26    10  2.75
# 6 C9orf72         5.57      3    4.70  3     8        1    20   295    53  5   

```

Local Rearrangement

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Local\nRearrangement") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Local_Rearrangement = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Local_Rearrangement = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Local_Rearrangement) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Local_Rearrangement)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.stat$y.position = c(9,21,21,21,21)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.stat
# .y.   group1 group2      n1    n2 statistic    df            p      p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl>        <dbl>      <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    -5.67  153.  0.0000000678 0.00000102 ****               10.8 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -1.33   24.4 0.197        0.985      ns                 11.7 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9     2.48   13.5 0.027        0.245      ns                 12.6 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10     1.27   10.8 0.23         0.985      ns                 13.5 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53    -0.794  92.2 0.429        1          ns                 14.4 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,10), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Local Rearrangement") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL           1.4        1    1.77     0  2.75     0     7   126    90  2.75
# 2 Sporadic       2.62       2    1.58     2  4        0     8   546   208  2   
# 3 SOD1           2          1.5  1.75     1  3        0     6    36    18  2   
# 4 FUS            0.444      0    1.01     0  0        0     3     4     9  0   
# 5 TARDBP         0.6        0    1.90     0  0        0     6     6    10  0   
# 6 C9orf72        1.68       1    2.17     0  2        0    10    89    53  2   

```

Distant Proximity

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Distant\nProximity") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Distant_Proximity = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Distant_Proximity = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Distant_Proximity) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Distant_Proximity)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.stat$y.position = c(22,21,8,21,21)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.stat
#  .y.   group1 group2      n1    n2 statistic    df     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    -2.21  154.  0.028 0.283 ns                   22 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -1.31   22.3 0.203 1     ns                   21 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9     3.61   23.7 0.001 0.02  *                    21 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10     0.134  10.8 0.896 1     ns                   21 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53     0.526 108.  0.6   1     ns                   21 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,8), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Distant Proximity") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL           1.47         1  1.38  0        2     0     6   132    90  2   
# 2 Sporadic       1.84         2  1.24  1        3     0     7   383   208  2   
# 3 SOD1           2            2  1.61  1        2     0     7    36    18  1   
# 4 FUS            0.667        1  0.5   0        1     0     1     6     9  1   
# 5 TARDBP         1.4          1  1.51  0.25     2     0     5    14    10  1.75
# 6 C9orf72        1.34         1  1.40  0        2     0     5    71    53  2   

```


Overlapping Neighbours

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Overlapping\nNeighbours") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Overlapping_Neighbours = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Overlapping_Neighbours = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Overlapping_Neighbours) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Overlapping_Neighbours)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.stat$y.position = c(3.2,21,3.4,21,21)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.stat
# .y.   group1 group2      n1    n2 statistic    df        p    p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    -5.34  173.  2.95e- 7 3.54e- 6 ****                 22 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -2.05   22.9 5.2 e- 2 4.06e- 1 ns                   21 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9     7.72   89   1.62e-11 2.27e-10 ****                 21 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10     0.753  11.4 4.66e- 1 1   e+ 0 ns                   21 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53    -0.945 105.  3.47e- 1 1   e+ 0 ns                   21 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,3.5), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Overlapping Neighbours") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) + 
  geom_jitter(size = 0.5, width = 0.2, height = 0.2) # geom_sina doesnt work if all values the same so specify geom_jitter https://githubmemory.com/repo/thomasp85/ggforce/issues/212
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL           0.422        0 0.519     0  1        0     2    38    90  1   
# 2 Sporadic       0.774        1 0.531     0  1        0     3   161   208  1   
# 3 SOD1           0.722        1 0.575     0  1        0     2    13    18  1   
# 4 FUS            0            0 0         0  0        0     0     0     9  0   
# 5 TARDBP         0.3          0 0.483     0  0.75     0     1     3    10  0.75
# 6 C9orf72        0.509        0 0.541     0  1        0     2    27    53  1   

```

Inter Chromosomal

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Inter\nChromosomal") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Inter_Chromosomal = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Inter_Chromosomal = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Inter_Chromosomal) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Inter_Chromosomal)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.stat
#  .y.   group1 group2      n1    n2 statistic    df     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208     1.19  108.  0.237     1 ns                 24.4 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18    -1.80   17.2 0.089     1 ns                 27.0 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9    -0.565  11.4 0.583     1 ns                 29.5 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10    -0.411  15.1 0.687     1 ns                 32.1 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53     0.172 141.  0.864     1 ns                 34.7 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,4), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Inter Chromosomal") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
# genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL           0.578      0   1.36      0   1       0     8    52    90   1  
# 2 Sporadic       0.399      0   0.659     0   1       0     3    83   208   1  
# 3 SOD1           3.61       0   7.12      0   2.5     0    22    65    18   2.5
# 4 FUS            0.778      1   0.972     0   1       0     3     7     9   1  
# 5 TARDBP         0.7        0.5 0.823     0   1       0     2     7    10   1  
# 6 C9orf72        0.547      0   0.774     0   1       0     3    29    53   1  
```

Local Inversion

```{r}
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion = filter(ipsc_mn_als_datasets.starfusion, fusion_type == "Local\nInversion") %>% 
  mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72"))) %>% 
  drop_na(genetic_group) %>% # remove all other mutants n < 3.
  add_count(dataset_sample) %>% # count number of fusion types per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, genetic_group) #keep 1 row per sample fusion type

# fill missing 0s
ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Local_Inversion = ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup$dataset_sample %in% ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion$dataset_sample] #  "answerals_CASE_NEUZE432DZM" "dafinca.tardbp_tardbp_6"    "dafinca.tardbp_tardbp_4"    "smith_tardbp_1"             "smith_ctrl_1"    
ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Local_Inversion = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.genetic_subgroup.samples_no_Local_Inversion) %>% mutate(genetic_group = case_when(mutation == "c9orf72" ~ "C9orf72", mutation == "fus" ~ "FUS", mutation == "sod1" ~ "SOD1", mutation == "sporadic" ~ "Sporadic", mutation == "tardbp" ~ "TARDBP", mutation %in% c("ctrl","iso") ~ "CTRL"), genetic_group = factor(genetic_group, levels = c("CTRL", "Sporadic", "SOD1", "FUS", "TARDBP", "C9orf72")), n = 0) %>% select(dataset_sample, n, genetic_group)
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion = bind_rows(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion, ipsc_mn_als_datasets.paired_end.metadata.genetic_subgroup.no_Local_Inversion)

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.stat <- ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion %>% t_test(n ~ genetic_group) %>% add_significance()  %>% add_xy_position() %>% filter(group1 == "CTRL")
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.stat
#  .y.   group1 group2      n1    n2 statistic    df     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    90   208    2.50   95.6  0.014 0.169 ns                 2.24 <chr [2]>        1     2
# 2 n     CTRL   SOD1        90    18   -0.0953 20.6  0.925 1     ns                 2.50 <chr [2]>        1     3
# 3 n     CTRL   FUS         90     9   -0.808   8.96 0.44  1     ns                 2.75 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      90    10    2.82   89    0.006 0.084 ns                 3.01 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     90    53    2.82   89    0.006 0.084 ns                 3.27 <chr [2]>        1     6
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion, color_by = "genetic_group", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,2), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Local Inversion") + 
  stat_pvalue_manual(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) +
    geom_jitter(size = 0.5, width = 0.2, height = 0.2) # geom_sina doesnt work if all values the same so specify geom_jitter https://githubmemory.com/repo/thomasp85/ggforce/issues/212
ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.violin

ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
#  genetic_group  mean_n median_n   sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>           <dbl>    <dbl>  <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL          0.1            0 0.337      0     0     0     2     9    90     0
# 2 Sporadic      0.00962        0 0.0978     0     0     0     1     2   208     0
# 3 SOD1          0.111          0 0.471      0     0     0     2     2    18     0
# 4 FUS           0.222          0 0.441      0     0     0     1     2     9     0
# 5 TARDBP        0              0 0          0     0     0     0     0    10     0
# 6 C9orf72       0              0 0          0     0     0     0     0    53     0


# # histogram
# ipsc_mn_als_datasets.starfusion.fusion_type.histogram = ggplot(ipsc_mn_als_datasets.starfusion.fusion_type_n, aes(x=n, color=CONDITION, fill=CONDITION)) + 
#   geom_histogram(alpha=0.5, position="identity", binwidth = 1)+
#   scale_color_manual(values=c("dodgerblue2", "firebrick2"))+ scale_fill_manual(values=c("dodgerblue2", "firebrick2"))+
#   facet_wrap(~ fusion_type, scales = "free", nrow = 1) + theme_oz() #+ ggtitle("Zeros filled")
# ipsc_mn_als_datasets.starfusion.fusion_type.histogram

```


## Merge

```{r}
ipsc_mn_als_datasets.fusion.genetic_groups =  plot_grid(ipsc_mn_als_datasets.starfusion.mutation.violin,
                                                        plot_grid(ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Neighbours.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Rearrangement.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Distant_Proximity.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Overlapping_Neighbours.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Inter_Chromosomal.violin,
                                                                  ipsc_mn_als_datasets.starfusion.fusion_type.genetic_subgroup_Local_Inversion.violin, 
                                                                  ncol = 2, nrow = 3, labels = c("b","c","d","e","f","g")),
                                                        nrow = 2, rel_heights = c(1,3), labels = c("a",""))
# ipsc_mn_als_datasets.fusion.genetic_groups
ggsave(ipsc_mn_als_datasets.fusion.genetic_groups, filename = here(proj_path,"rnafusion/ipsc_mn_als_datasets.fusion.genetic_groups.png"), height = 11.75, width = 8.25)

                                                        
```



# Answer ALS alone

## Variants

```{r}
# ipsc_mn_als_datasets.vcfannotate = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.csv")) %>% 
#   mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
#   group_by(dataset_sample)%>% add_tally(n, name = "total") %>% ungroup %>% select(dataset_sample, n, type, variant_type, total, everything()) # add a column of sum of variants per sample
# ipsc_mn_als_datasets.vcfannotate

ipsc_mn_als_datasets.vcfannotate.answerals = ipsc_mn_als_datasets.vcfannotate %>% filter(dataset == "answerals")

# Violin total variant events in CTRL & ALS
ipsc_mn_als_datasets.vcfannotate.answerals.totals = ipsc_mn_als_datasets.vcfannotate.answerals %>% distinct(dataset_sample, .keep_all = TRUE) #keep 1 row per sample for total number of variants
ipsc_mn_als_datasets.vcfannotate.answerals.totals

ipsc_mn_als_datasets.vcfannotate.answerals.totals.glm = glm(data = ipsc_mn_als_datasets.vcfannotate.answerals.totals, total ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(ipsc_mn_als_datasets.vcfannotate.answerals.totals.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.067e+01  3.343e-03 3191.93   <2e-16 ***
# CONDITIONALS                          1.072e-02  6.892e-04   15.55   <2e-16 ***


ipsc_mn_als_datasets.vcfannotate.answerals.total.violin <- violin_plot(ipsc_mn_als_datasets.vcfannotate.answerals.totals, color_by = "CONDITION", continuous = "total", cols = c("dodgerblue2", "firebrick2"), 
                                                             one_sample = FALSE, stats_test = "wilcox_test", arrow_labels = FALSE, ylims = c(50000,80000), dpi = 300, ylabel = "Total Number of Variants", title = "AnswerALS: All Variant types") 
ipsc_mn_als_datasets.vcfannotate.answerals.total.violin
#  .y.   group1 group2    n1    n2 statistic      p p.signif
#   <chr> <chr>  <chr>  <int> <int>     <dbl>  <dbl> <chr>   
# 1 lfc   CTRL   ALS       42   238     3762. 0.0106 *       
# # A tibble: 1 × 7
#   .y.   group1 group2 effsize    n1    n2 magnitude
# * <chr> <chr>  <chr>    <dbl> <int> <int> <ord>    
# 1 lfc   CTRL   ALS     -0.394    42   238 small    

ipsc_mn_als_datasets.vcfannotate.answerals.totals %>% my_summarise(total, condition) %>% mutate(IQR = Q3_total - Q1_total)
# condition mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <chr>          <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1 als           61291.       60689     3546.   59168.   62950.     48715     73472  14587359     238 3782.
# 2 ctrl          59890.       59646.    3601.   57313    61359.     54971     72087   2515399      42 4046.



ipsc_mn_als_datasets.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% left_join(multiqc_rseqc_read_distribution.tsv)
ipsc_mn_als_datasets.vcfannotate

answerals.vcfannotate = ipsc_mn_als_datasets.vcfannotate %>% filter(dataset == "answerals")

### ALS vs CTRL variant number
answerals.vcfannotate.samples_no_variants = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% answerals.vcfannotate$dataset_sample] # 0 i.e. all samples have at least 1 variant
answerals.vcfannotate.samples_no_variants # 0

# Violin total variant events in CTRL & ALS
answerals.vcfannotate.totals = answerals.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) #keep 1 row per sample for total number of variants
answerals.vcfannotate.totals %>% glimpse

# generalised linear model spline
# library(rms)
answerals.vcfannotate.totals.glm = glm(total ~ CONDITION + rms::rcs(total_reads, 3), data=answerals.vcfannotate.totals, family=poisson) # + rms::rcs(total_reads, 3)
summary(answerals.vcfannotate.totals.glm)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           1.067e+01  3.343e-03 3192.16   <2e-16 ***
# CONDITIONALS                          1.072e-02  6.892e-04   15.55   <2e-16 ***


```


## Fusions


```{r}
# ipsc_mn_als_datasets.starfusion = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% 
#   mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))
# ipsc_mn_als_datasets.starfusion

ipsc_mn_als_datasets.starfusion.answerals = ipsc_mn_als_datasets.starfusion %>% filter(dataset == "answerals")

# Venn overlap unqiue gene fusions
ipsc_mn_als_datasets.starfusion.answerals %>% distinct(fusion_name) # 173
ipsc_mn_als_datasets.starfusion.answerals %>% distinct(fusion_type) # 6
ipsc_mn_als_datasets.starfusion.answerals %>% group_by(condition) %>% distinct(fusion_name) %>% count(condition)
# condition     n
#   <chr>     <int>
# 1 als         166
# 2 ctrl         71
ipsc_mn_als_datasets.starfusion.answerals.als_events <- ipsc_mn_als_datasets.starfusion.answerals %>% filter(condition == "als") %>% distinct(fusion_name) %>% pull(fusion_name)
ipsc_mn_als_datasets.starfusion.answerals.ctrl_events <- ipsc_mn_als_datasets.starfusion.answerals %>% filter(condition == "ctrl") %>% distinct(fusion_name) %>% pull(fusion_name)

# # ggvenn unique fusion events in CTRL & ALS
ipsc_mn_als_datasets.starfusion.answerals.als_ctrl_events <- list("ALS" = ipsc_mn_als_datasets.starfusion.answerals.als_events, "CTRL" = ipsc_mn_als_datasets.starfusion.answerals.ctrl_events)
ipsc_mn_als_datasets.starfusion.answerals.als_ctrl_events_venn <- ggvenn(ipsc_mn_als_datasets.starfusion.answerals.als_ctrl_events, auto_scale = TRUE, fill_color = c("firebrick2", "dodgerblue2"), set_name_color = c("firebrick2", "dodgerblue2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 3.5) + labs(subtitle = "Unique Gene Fusions")  + 
  theme(plot.subtitle = element_text(hjust = 0.5, size = 8), plot.margin = unit(c(0, -1, 0, -1), "lines")) #+ theme_oz()
ipsc_mn_als_datasets.starfusion.answerals.als_ctrl_events_venn

### ALS vs CTRL unique fusion events
ipsc_mn_als_datasets.starfusion.answerals.n = ipsc_mn_als_datasets.starfusion.answerals %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, total_reads) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.answerals.samples_no_fusions = answerals.metadata$dataset_sample[!answerals.metadata$dataset_sample %in% ipsc_mn_als_datasets.starfusion.answerals.n$dataset_sample] # 0 
# answerals.metadata.no_fusions = answerals.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.answerals.samples_no_fusions) %>% 
  # mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), n = 0) %>% select(dataset_sample, n, CONDITION)
# ipsc_mn_als_datasets.starfusion.answerals.n = bind_rows(ipsc_mn_als_datasets.starfusion.answerals.n, answerals.metadata.no_fusions)

glm_starfusion = glm(data = ipsc_mn_als_datasets.starfusion.answerals.n, n ~ CONDITION + rms::rcs(total_reads, 3), family=poisson)
summary(glm_starfusion)
# Coefficients:
#                                        Estimate Std. Error z value Pr(>|z|)    
# (Intercept)                           2.746e+00  2.282e-01  12.034   <2e-16 ***
# CONDITIONALS                          9.116e-02  5.096e-02   1.789   0.0736 .

ipsc_mn_als_datasets.starfusion.answerals.n %>% my_summarise(n, CONDITION) %>% mutate(IQR = Q3_n - Q1_n)
# CONDITION mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>      <dbl>    <dbl> <dbl> <dbl> <dbl> <int> <int> <int> <int> <dbl>
# 1 CTRL        10.9     10    3.83     8    13     6    19   457    42     5
# 2 ALS         11.5     10.5  5.61     8    14     2    39  2747   238     6

# Violin
ipsc_mn_als_datasets.starfusion.answerals.violin <- violin_plot(ipsc_mn_als_datasets.starfusion.answerals.n, color_by = "CONDITION", continuous = "n", cols = c("dodgerblue2", "firebrick2"), one_sample = FALSE, stats_test = "t_test", arrow_labels = FALSE, ylims = c(0,43), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "Answer ALS: All Fusion types")
ipsc_mn_als_datasets.starfusion.answerals.violin
#  .y.   group1 group2    n1    n2 statistic    df     p p.signif
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <chr>   
# 1 lfc   CTRL   ALS       42   238    -0.952  75.9 0.344 ns      
# # A tibble: 1 × 7
#   .y.   group1 group2 effsize    n1    n2 magnitude 
# * <chr> <chr>  <chr>    <dbl> <int> <int> <ord>     
# 1 lfc   CTRL   ALS     -0.123    42   238 negligible

```


# scatter number of variants aginst read coverage

```{r}
# left_join multiqc
# multiqc stats
multiqc_general_stats_stat_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_general_stats_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), 
     multiqc_general_stats_stat_paths = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project","/Volumes/lab-luscomben/home/users/ziffo",multiqc_general_stats_stat_paths)) %>% pull(multiqc_general_stats_stat_paths)
multiqc_general_stats.tsv = multiqc_general_stats_stat_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_general_stats.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_general_stats <- map_dfr(multiqc_general_stats.tsv, bind_rows, .id = "database_dir") %>% clean_names()
multiqc_rseqc_read_distribution_paths = ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir) %>% mutate(multiqc_rseqc_read_distribution_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt"),
     multiqc_rseqc_read_distribution_paths = gsub("/camp/lab/luscomben/home/users/ziffo|/camp/project","/Volumes/lab-luscomben/home/users/ziffo",multiqc_rseqc_read_distribution_paths)) %>% pull(multiqc_rseqc_read_distribution_paths)
multiqc_rseqc_read_distribution.tsv = multiqc_rseqc_read_distribution_paths %>% map(read_tsv, show_col_types = FALSE)
names(multiqc_rseqc_read_distribution.tsv) =  ipsc_mn_als_datasets_with_lee.metadata %>% distinct(database_dir,.keep_all=TRUE) %>% pull(database_dir)
multiqc_rseqc_read_distribution <- map_dfr(multiqc_rseqc_read_distribution.tsv, bind_rows, .id = "database_dir") %>% clean_names()
ipsc_mn_als_datasets_with_lee.pcaData = ipsc_mn_als_datasets_with_lee.pcaData %>% left_join(multiqc_general_stats) %>% left_join(multiqc_rseqc_read_distribution) # need unique sample names - If multiple dataset names within a single multiqc_general_stats.txt then will only join the first dataset name

# variants
ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads = select(distinct(ipsc_mn_als_datasets.vcfannotate, dataset_sample, .keep_all = TRUE), database_dir, sample, total, dataset, condition, library_type) %>% left_join(select(multiqc_rseqc_read_distribution, database_dir, sample, total_reads))
ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads

ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads.scatter <- ggscatter(ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads, x = "total_reads", y = "total", color = "condition", cor.coef = TRUE, cor.method = "pearson", size = 0.5, cor.coeff.args = list(label.x = 90000000, label.y = 20000, label.sep="\n", size = 2.8),
          xlab = expression(Total~reads~per~sample), ylab = expression(Total~number~of~variants~per~sample)) +
  theme_oz() + theme(legend.position = "right") +
  geom_hline(yintercept = 0, linetype = 2) + geom_vline(xintercept = 0, linetype = 2) +
  coord_cartesian(ylim=c(0,80000))
ipsc_mn_als_datasets.vcfannotate.total.multiqc_total_reads.scatter

# fusions

ipsc_mn_als_datasets.starfusion
ipsc_mn_als_datasets.starfusion.n = ipsc_mn_als_datasets.starfusion %>% 
  add_count(dataset_sample) %>% # count number of fusion per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, CONDITION, dataset, total_reads) #keep 1 row per sample
 # add samples with 0 fusions
ipsc_mn_als_datasets.starfusion.samples_no_fusions = ipsc_mn_als_datasets.paired_end.metadata$dataset_sample[!ipsc_mn_als_datasets.paired_end.metadata$dataset_sample %in% ipsc_mn_als_datasets.starfusion.n$dataset_sample] # "smith_tardbp_1" "smith_ctrl_1"  
ipsc_mn_als_datasets.paired_end.metadata.no_fusions = ipsc_mn_als_datasets.paired_end.metadata %>% filter(dataset_sample %in% ipsc_mn_als_datasets.starfusion.samples_no_fusions) %>% 
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), n = 0) %>% select(dataset_sample, n, CONDITION, dataset)
ipsc_mn_als_datasets.starfusion.n = bind_rows(ipsc_mn_als_datasets.starfusion.n, ipsc_mn_als_datasets.paired_end.metadata.no_fusions)


ipsc_mn_als_datasets.starfusion.scatter <- ggscatter(ipsc_mn_als_datasets.starfusion.n, x = "total_reads", y = "n", color = "dataset", cor.coef = TRUE, cor.method = "pearson", size = 0.5, cor.coeff.args = list(label.x = 90000000, label.y = 50, label.sep="\n", size = 2.8),
          xlab = expression(Total~reads~per~sample), ylab = expression(Total~number~of~fusions~per~sample)) +
  theme_oz() + theme(legend.position = "right") +
  geom_hline(yintercept = 0, linetype = 2) + geom_vline(xintercept = 0, linetype = 2) #+  coord_cartesian(ylim=c(0,80000))
ipsc_mn_als_datasets.starfusion.scatter


```


# NeuroLINCS iPSC vs iMNs d32 vs diMN d18

Metadata of iPSC & iMN batches from same donors

```{r}
neurolincs.ipsc_iMN_diMN.metadata = read_csv(here(shared_path,"neurolincs/sample-details/samplesheet.csv")) %>% distinct(sample, .keep_all = TRUE) %>%
  mutate(database_dir = here(shared_path,"neurolincs"), condition = factor(condition, levels = c("ctrl", "als")),
         dataset = case_when(dataset == "neurolincs0" ~ "neurolincs.diMN",dataset == "neurolincsA" ~ "neurolincs.iMN", dataset == "neurolincsI" ~ "neurolincs.iPSC"), dataset_sample = paste0(dataset,"_",sample), study_accession = dataset)
neurolincs.ipsc_iMN_diMN.metadata %>% glimpse
neurolincs.ipsc_iMN_diMN.metadata %>% group_by(gap_subject_id)
neurolincs.ipsc_iMN_diMN.metadata %>% group_by(submitted_subject_id) # submitted_subject_id is equivelent to gap_subject_id
neurolincs.ipsc_iMN_diMN.metadata %>% group_by(DIV) %>% count(submitted_subject_id) %>% print(n=Inf)
neurolincs.ipsc_iMN_diMN.metadata %>% select(submitted_subject_id, gap_subject_id) %>% distinct()

```

ALS vs CTRL in iPSC

```{r}
# neurolincs.ipsc = DESeq.analysis(metadata = neurolincs.ipsc.metadata, unique_names = "sample", design = ~ condition, contrast = "condition_als_vs_ctrl")
# saveRDS(neurolincs.ipsc, here(proj_path,"expression/deseq2/neurolincs.ipsc.rds"))
neurolincs.ipsc = readRDS(here(proj_path,"expression/deseq2/neurolincs.ipsc.rds"))
net_progeny <- get_progeny(organism = 'human', top = 100)
neurolincs.ipsc.res.matrix <- neurolincs.ipsc$res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
neurolincs.ipsc_progeny.contrast_acts <- run_wmean(mat=neurolincs.ipsc.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
neurolincs.ipsc_progeny.pathwayActivity <- ggplot(neurolincs.ipsc_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
  geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
  theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "NeuroLINCS iPSC Signaling Pathways", x = "", y = "ALS vs CTRL Normalised enrichment") +
  stat_pvalue_manual(neurolincs.ipsc_progeny.contrast_acts, label = "p.signif", y.position = 14, xmin = "source", xmax = NULL, size = 5, hide.ns = TRUE) 
neurolincs.ipsc_progeny.pathwayActivity

```

## rnavar

vcf-annotate

```{r}
# vcf.annotate_script = neurolincs.ipsc_iMN_diMN.metadata %>% filter(DIV == 0) %>%
#   mutate(`#!/bin/bash` = paste0("zcat /camp/project/proj-luscombn-patani/working/neurolincs/rnavar/variant_calling/", sample,"/", sample, ".haplotypecaller.filtered.vcf.gz | vcf-annotate --fill-type | grep -oP 'TYPE=\\w+' | sort | uniq -c | sed 's/ TYPE=/,/' > ", "/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate/",dataset_sample, ".vcfstats.csv")) %>% select(`#!/bin/bash`)
# vcf.annotate_script
# vcf.annotate_script %>% pull(`#!/bin/bash`)    # check script: fastq order, job name
# write_tsv(vcf.annotate_script, "/camp/project/proj-luscombn-patani/working/neurolincs/sample-details/neurolincs.ipsc.vcf.annotate_script.sh")

# # run script
# ml VCFtools
# cd /camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate
# bash /camp/project/proj-luscombn-patani/working/neurolincs/sample-details/neurolincs.ipsc.vcf.annotate_script.sh

# read_vcfannotate = function(metadata){
#   print("reading in vcf-annotate files with read_csv")
#   vcf.annotate.csv = metadata$rnavar_path %>% map(read_csv, col_names = c("n","type"), col_types = list(n="n",type="c"))
#   names(vcf.annotate.csv) = metadata$dataset_sample # needs to be unique names e.g. dataset_sample
#   vcf.annotate_tab <- map_dfr(vcf.annotate.csv, bind_rows, .id = "dataset_sample") %>%
#     mutate(variant_type = case_when(grepl("del", type) ~ "Deletion", grepl("ins", type) ~ "Insertion", grepl("snp", type) ~ "SNP", grepl("complex", type) ~ "Complex", TRUE ~ "Unknown")) %>%
#     left_join(metadata, by = "dataset_sample")
#   return(vcf.annotate_tab)
# }
# neurolincs.ipsc.metadata = neurolincs.ipsc_iMN_diMN.metadata %>% filter(DIV == 0)
# neurolincs.ipsc.metadata = neurolincs.ipsc.metadata %>% mutate(rnavar_path = paste0("/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling/vcfannotate/",dataset_sample,".vcfstats.csv"))
# neurolincs.ipsc.metadata.vcfannotate = read_vcfannotate(neurolincs.ipsc.metadata)
# neurolincs.ipsc.metadata.vcfannotate
# write_csv(neurolincs.ipsc.metadata.vcfannotate, here(proj_path,"rnavar/neurolincs.ipsc.metadata.vcfannotate.csv"))

neurolincs.ipsc.metadata.vcfannotate.csv = read_csv(here(proj_path,"rnavar/neurolincs.ipsc.metadata.vcfannotate.csv"))
neurolincs.imn_dimn.metadata.vcfannotate.csv = read_csv(here(proj_path,"rnavar/ipsc_mn_als_datasets.vcfannotate.csv")) %>% filter(dataset %in% c("neurolincs.diMN","neurolincs.iMN"))

neurolincs.ipsc_imn_dimn.metadata.vcfannotate = bind_rows(neurolincs.ipsc.metadata.vcfannotate.csv, neurolincs.imn_dimn.metadata.vcfannotate.csv) %>%
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), variant_type = factor(variant_type, levels = c("SNP", "Insertion", "Deletion", "Complex"))) %>% 
  group_by(dataset_sample)%>% add_tally(n, name = "total") %>% ungroup %>% select(dataset_sample, n, type, variant_type, total, everything()) # add a column of sum of variants per sample
neurolincs.ipsc_imn_dimn.metadata.vcfannotate
```

Total Fusions

```{r}
# Distinct row per samples
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals = neurolincs.ipsc_imn_dimn.metadata.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, total, DIV, condition, submitted_subject_id) # keep 1 row per sample

# Fill in samples with 0 counts
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.samples_no_variants = neurolincs.ipsc_iMN_diMN.metadata$dataset_sample[!neurolincs.ipsc_iMN_diMN.metadata$dataset_sample %in% neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals$dataset_sample] # 0 i.e. all samples have at least 1 variant
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.samples_no_variants # 0

# Violin total variant events in CTRL & ALS at iPSC, iMN and dIMN
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat <- neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals %>% wilcox_test(total ~ DIV) %>% add_significance()  %>% add_xy_position() #%>% filter(group1 == "0")
# neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat$y.position = c(75000,80000,80000,80000,80000)
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat
# .y.   group1 group2    n1    n2 statistic     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 total 0      18        20    30      276  0.645     1 ns               67487. <chr [2]>        1     2
# 2 total 0      32        20    14      125  0.616     1 ns               69069. <chr [2]>        1     3
# 3 total 18     32        30    14      190. 0.614     1 ns               70650. <chr [2]>        2     3
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.violin <- violin_plot(mutate(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals, DIV = as.factor(DIV)), color_by = "DIV", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"),  plot_stats = FALSE, arrow_labels = FALSE, ylims = c(40000,70000), dpi = 300, ylabel = "Number of Variants", title = "NeuroLINCS iPSC, diMN, iMN: Total Variants", xlab = "DIV") + 
  stat_pvalue_manual(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat, label = "p.adj.signif", tip.length = 0.001, size = 4, hide.ns = TRUE)
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.violin

ggline(mutate(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals,submitted_subject_id=as.factor(submitted_subject_id)), "DIV", "total", lintype = "submitted_subject_id", color = "submitted_subject_id")

neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals %>% filter(condition == "als") %>% my_summarise(total, DIV) %>% mutate(IQR = Q3_total - Q1_total)
# DIV mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <dbl>      <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1     0     54008        53930.    1646.   52845    55419.     51093     56690    648096      12 2574.
# 2    18     53870.       54638     3194.   52325.   55914.     42761     58352   1185133      22 3589.
# 3    32     54014.       54492     2304.   53279.   54866.     49614     57671    432111       8 1586.

# restrict to ALS samples only
# Distinct row per samples
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.totals = neurolincs.ipsc_imn_dimn.metadata.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) %>% filter(condition == "als") %>% select(dataset_sample, total, DIV) # keep 1 row per sample

# Violin total variant events in ALS at iPSC, iMN and dIMN
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.stat <- neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.totals %>% wilcox_test(total ~ DIV) %>% add_significance()  %>% add_xy_position() #%>% filter(group1 == "0")
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.stat
# .y.   group1 group2    n1    n2 statistic     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 total 0      18        12    22       126 0.845     1 ns               58551. <chr [2]>        1     2
# 2 total 0      32        12     8        46 0.91      1 ns               58851. <chr [2]>        1     3
# 3 total 18     32        22     8        93 0.836     1 ns               59150. <chr [2]>        2     3
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.violin <- violin_plot(mutate(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.totals, DIV = as.factor(DIV)), color_by = "DIV", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"),  plot_stats = FALSE, arrow_labels = FALSE, ylims = c(40000,60000), dpi = 300, ylabel = "Number of Variants", title = "NeuroLINCS ALS only iPSC, diMN, iMN: Total Variants", xlab = "DIV")
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.violin

neurolincs.ipsc_imn_dimn.metadata.vcfannotate.als.totals %>% my_summarise(total, DIV) %>% mutate(IQR = Q3_total - Q1_total)
# DIV mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <dbl>      <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1     0     54008        53930.    1646.   52845    55419.     51093     56690    648096      12 2574.
# 2    18     53870.       54638     3194.   52325.   55914.     42761     58352   1185133      22 3589.
# 3    32     54014.       54492     2304.   53279.   54866.     49614     57671    432111       8 1586.

# merge iMN & diMN
# Distinct row per samples
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals = neurolincs.ipsc_imn_dimn.metadata.vcfannotate %>% distinct(dataset_sample, .keep_all = TRUE) %>% mutate(celltype = case_when(DIV == 0 ~ "iPSC", TRUE ~ "iPSN")) %>% select(dataset_sample, total, celltype, condition) # keep 1 row per sample

# Violin total variant events in CTRL & ALS at iPSC, iMN and dIMN
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat <- neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals %>% wilcox_test(total ~ celltype) %>% add_significance()  %>% add_xy_position() #%>% filter(group1 == "0")
# neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat$y.position = c(75000,80000,80000,80000,80000)
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.stat
# .y.   group1 group2    n1    n2 statistic     p p.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 total iPSC   iPSN      20    44       401 0.577 ns           67487. <chr [2]>        1     2
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.violin <- violin_plot(neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals, color_by = "celltype", continuous = "total", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"),  plot_stats = FALSE, arrow_labels = FALSE, ylims = c(40000,70000), dpi = 300, ylabel = "Number of Variants", title = "NeuroLINCS iPSC vs iPSN: Total Variants", xlab = "celltype") 
neurolincs.ipsc_imn_dimn.metadata.vcfannotate.violin

neurolincs.ipsc_imn_dimn.metadata.vcfannotate.totals %>% filter(condition == "als") %>% my_summarise(total, celltype) %>% mutate(IQR = Q3_total - Q1_total)
# celltype mean_total median_total sd_total Q1_total Q3_total min_total max_total sum_total n_total   IQR
#   <chr>         <dbl>        <dbl>    <dbl>    <dbl>    <dbl>     <dbl>     <dbl>     <dbl>   <int> <dbl>
# 1 iPSC         54008        53930.    1646.   52845    55419.     51093     56690    648096      12 2574.
# 2 iPSN         53908.       54498     2945.   52394.   55224.     42761     58352   1617244      30 2830 
```


## rnafusion

```{r}
# read_starfusion = function(metadata){
#   print("reading in STAR Fusion files with read_tsv")
#   starfusion.tsv = metadata$starfusion_path %>% map(read_tsv, col_types = list(JunctionReadCount="n",SpanningFragCount="n",est_J="n",est_S="n",FFPM="n",LeftBreakEntropy="n",RightBreakEntropy="n",.default="c"))
#   names(starfusion.tsv) = metadata$dataset_sample # needs to be unique names e.g. dataset_sample
#   starfusion_tab <- map_dfr(starfusion.tsv, bind_rows, .id = "dataset_sample") %>% rename(fusion_name = '#FusionName') %>% clean_names() %>%
#     mutate(left_chr = str_split_fixed(left_breakpoint,":",3)[,1], left_coord = str_split_fixed(left_breakpoint,":",3)[,2], left_strand = str_split_fixed(left_breakpoint,":",3)[,3],
#            right_chr = str_split_fixed(right_breakpoint,":",3)[,1], right_coord = str_split_fixed(right_breakpoint,":",3)[,2], right_strand = str_split_fixed(right_breakpoint,":",3)[,3],
#            fusion_type = case_when(grepl("INTERCHROMOSOMAL", annots) ~ "Inter\nChromosomal", grepl("OVERLAP", annots) ~ "Overlapping\nNeighbours", grepl("NEIGHBORS", annots) ~ "Neighbours", grepl("LOCAL_INVERSION", annots) ~ "Local\nInversion", grepl("LOCAL_REARRANGEMENT", annots) ~ "Local\nRearrangement", TRUE ~ "Distant\nProximity"),
#            breakpoints = paste(left_breakpoint, right_breakpoint, sep="_")) %>%
#     left_join(metadata, by = "dataset_sample")
#   return(starfusion_tab)
# }
# neurolincs.ipsc.metadata = neurolincs.ipsc_iMN_diMN.metadata %>% filter(DIV == 0)
# neurolincs.ipsc.metadata = neurolincs.ipsc.metadata %>% mutate(starfusion_path = paste0("/camp/project/proj-luscombn-patani/working/neurolincs/rnafusion/starfusion/",sample,".starfusion.abridged.coding_effect.tsv"))
# neurolincs.ipsc.metadata.starfusion = read_starfusion(neurolincs.ipsc.metadata)
# neurolincs.ipsc.metadata.starfusion
# write_csv(neurolincs.ipsc.metadata.starfusion, here(proj_path,"rnafusion/neurolincs.ipsc.metadata.starfusion.csv"))

neurolincs.ipsc.metadata.starfusion.csv = read_csv(here(proj_path,"rnafusion/neurolincs.ipsc.metadata.starfusion.csv")) 
neurolincs.imn_dimn.metadata.starfusion.csv = read_csv(here(proj_path,"rnafusion/ipsc_mn_als_datasets.starfusion.csv")) %>% filter(dataset %in% c("neurolincs.diMN","neurolincs.iMN"))

neurolincs.ipsc_imn_dimn.starfusion = bind_rows(neurolincs.ipsc.metadata.starfusion.csv, neurolincs.imn_dimn.metadata.starfusion.csv) %>%
  mutate(CONDITION = factor(toupper(condition), levels = c("CTRL","ALS")), fusion_type = factor(fusion_type, levels = c("Local\nInversion", "Local\nRearrangement", "Neighbours", "Overlapping\nNeighbours", "Distant\nProximity", "Inter\nChromosomal")))
neurolincs.ipsc_imn_dimn.starfusion

```

```{r}
neurolincs.ipsc_imn_dimn.starfusion.n = neurolincs.ipsc_imn_dimn.starfusion %>% add_count(dataset_sample) %>% # count number of fusions per sample
  distinct(dataset_sample, .keep_all = TRUE) %>% select(dataset_sample, n, DIV, condition, submitted_subject_id) # keep 1 row per sample

# Fill in samples with 0 counts
neurolincs.ipsc_imn_dimn.metadata.starfusion.samples_no_fusion = neurolincs.ipsc_iMN_diMN.metadata$dataset_sample[!neurolincs.ipsc_iMN_diMN.metadata$dataset_sample %in% neurolincs.ipsc_imn_dimn.starfusion.n$dataset_sample] # 0 i.e. all samples have at least 1 fusion
neurolincs.ipsc_imn_dimn.metadata.starfusion.samples_no_fusion # 0


neurolincs.ipsc_imn_dimn.starfusion.stat <- neurolincs.ipsc_imn_dimn.starfusion.n %>% t_test(n ~ DIV) %>% add_significance()  %>% add_xy_position()
# neurolincs.ipsc_imn_dimn.starfusion.stat$y.position = c(40,38,40,42,42)
neurolincs.ipsc_imn_dimn.starfusion.stat
# .y.   group1 group2    n1    n2 statistic    df     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     0      18        20    30     -2.04  47.7 0.047 0.095 ns                 16.8 <chr [2]>        1     2
# 2 n     0      32        20    14      1.52  23.8 0.142 0.142 ns                 18.1 <chr [2]>        1     3
# 3 n     18     32        30    14      3.05  32.7 0.005 0.014 *                  19.4 <chr [2]>        2     3
neurolincs.ipsc_imn_dimn.starfusion.violin <- violin_plot(mutate(neurolincs.ipsc_imn_dimn.starfusion.n, DIV = as.factor(DIV)), color_by = "DIV", continuous = "n", cols = c("dodgerblue2","firebrick2","forestgreen","gold2","purple","grey"), violin_adjust = 2, plot_stats = FALSE, arrow_labels = FALSE, ylims = c(0,18), dpi = 300, ylabel = "Number of Unique Fusion Events", title = "NeuroLINCS iPSC, diMN, iMN: All gene Fusion Types") + 
  stat_pvalue_manual(neurolincs.ipsc_imn_dimn.starfusion.stat, label = "p.adj.signif", tip.length = 0.01, size = 4, hide.ns = TRUE) #y.position = 0.95*{{ylims[2]}}, 
neurolincs.ipsc_imn_dimn.starfusion.violin
#   .y.   group1 group2      n1    n2 statistic     df          p    p.adj p.adj.signif y.position groups        xmin  xmax
#   <chr> <chr>  <chr>    <int> <int>     <dbl>  <dbl>      <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 n     CTRL   Sporadic    89   208    -4.92  192.   0.00000187 0.000028 ****                 36 <chr [2]>        1     2
# 2 n     CTRL   SOD1        89    18    -2.41   18.6  0.026      0.292    ns                   38 <chr [2]>        1     3
# 3 n     CTRL   FUS         89     9     1.23   11.4  0.244      1        ns                   40 <chr [2]>        1     4
# 4 n     CTRL   TARDBP      89     9     0.942   8.75 0.371      1        ns                   42 <chr [2]>        1     5
# 5 n     CTRL   C9orf72     89    53    -1.18   70.4  0.243      1        ns                   42 <chr [2]>        1     6

ggline(neurolincs.ipsc_imn_dimn.starfusion.n, "DIV", "n", lintype = "submitted_subject_id")

neurolincs.ipsc_imn_dimn.starfusion.n %>% my_summarise(n, genetic_group) %>% mutate(IQR = Q3_n - Q1_n)
#   genetic_group mean_n median_n  sd_n  Q1_n  Q3_n min_n max_n sum_n   n_n   IQR
#   <fct>          <dbl>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>
# 1 CTRL            8.17      8    4.33   5   10        0    19   735    90  5   
# 2 Sporadic       11.1      10    4.96   8   13        2    36  2299   208  5   
# 3 SOD1           13.4      11.5  8.93   9   13.8      4    38   242    18  4.75
# 4 FUS             6.89      7    3.06   5    9        2    11    62     9  4   
# 5 TARDBP          5.6       5    6.29   1.5  5.75     0    22    56    10  4.25
# 6 C9orf72         9.64      8    7.90   3   12        2    39   511    53  9   


```


# Copy number variations

## RNAseqCNV

https://github.com/honzee/RNAseqCNV
https://www.nature.com/articles/s41375-022-01547-8

```{r}
devtools::install_github(repo = "honzee/RNAseqCNV")
library(RNAseqCNV)

# inputs: gene counts & GATK SNV MAFs

#config:
out_dir = "/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/rnaseqcnv"
count_dir = "/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/nfcore/star_salmon"
snv_dir = "/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/rnavar/variant_calling" 

# metadata: dataset_sample; count_file; vcf_file


RNAseqCNV_wrapper(config = "path/to/config", metadata = "path/to/metadata", snv_format = "vcf", genom_version = "hg38")




```

## CNVkit

## SuperFreq

## CaSpER

https://github.com/akdess/CaSpER/

```{r}

ipsc_mn_als_datasets = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.rds")) # 1.26G
centromere <- read.delim("/camp/lab/luscomben/home/users/ziffo/genomes/CaSpER/cytoBand_centromere.txt", header=F)
annotation <- CaSpER::generateAnnotation(id_type="ensembl_gene_id", genes=rownames(assay(ipsc_mn_als_datasets$vsd)), ishg19=FALSE, centromere = centromere)

loh <- CaSpER::readBAFExtractOutput(path="/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/cnv", sequencing.type="bulk", suffix = "baf")
names(loh) <- gsub(".baf", "", names(loh))

object <- CaSpER::CreateCasperObject(raw.data=counts(ipsc_mn_als_datasets$dds,normalized=FALSE),annotation=annotation, control.sample.ids=pull(filter(ipsc_mn_als_datasets.metadata,condition == "ctrl"),dataset_sample), 
                                     cytoband=cytoband, loh.name.mapping=pull(ipsc_mn_als_datasets.metadata,dataset_sample), method="iterative",
                                     cnv.scale=3, loh.scale=3, loh=loh, sequencing.type="bulk", expr.cutoff=0.1, log.transformed = FALSE, genomeVersion = "hg38",
                                     matrix.type="raw",  filter="median")
object

final.objects <- CaSpER::runCaSpER(object, removeCentromere=T, cytoband=cytoband, method="iterative")
final.objects
# Large-Scale CNV Summarisation
finalChrMat <- CaSpER::extractLargeScaleEvents(final.objects, thr=0.75) 
finalChrMat
# Segment based CNV Summarisation
gamma <- 6
all.segments <- do.call(rbind, lapply(final.objects, function(x) x@segments))
segment.summary <- extractSegmentSummary (final.objects)
loss <- segment.summary$all.summary.loss
gain <- segment.summary$all.summary.gain
loh <- segment.summary$all.summary.loh
loss.final <- loss[loss$count>gamma, ]
gain.final <- gain[gain$count>gamma, ]
loh.final <- loh[loh$count>gamma, ]
# Gene based CNV Summarisation
all.summary<- rbind(loss.final, gain.final)
colnames(all.summary) [2:4] <- c("Chromosome", "Start",   "End")
rna <-  GRanges(seqnames = Rle(gsub("q", "", gsub("p", "", all.summary$Chromosome))), 
                IRanges(all.summary$Start, all.summary$End))   
ann.gr <- makeGRangesFromDataFrame(final.objects[[1]]@annotation.filt, keep.extra.columns = TRUE, seqnames.field="Chr")
hits <- findOverlaps(geno.rna, ann.gr)
genes <- splitByOverlap(ann.gr, geno.rna, "GeneSymbol")
genes.ann <- lapply(genes, function(x) x[!(x=="")])
all.genes <- unique(final.objects[[1]]@annotation.filt[,2])
all.samples <- unique(as.character(final.objects[[1]]@segments$ID))
rna.matrix <- gene.matrix(seg=all.summary, all.genes=all.genes, all.samples=all.samples, genes.ann=genes.ann)

# Visualisation
obj <- final.objects[[9]]
plotHeatmap(object=obj, fileName="/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/heatmap.png",cnv.scale= 3, cluster_cols = F, cluster_rows = T, show_rownames = T, only_soi = T)
plotLargeScaleEvent(object=obj, fileName="/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/large.scale.events.png") 
plotGEAndGT(chrMat=finalChrMat, genoMat=genoMat, fileName="/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/RNASeqAndGT.png") # only for small sample size
plotBAFAllSamples(loh = obj@loh.median.filtered.data, fileName="/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/LOHAllSamples.png") 
# plotBAFOneSample(object, fileName="/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/LOHPlotsAllScales.pdf") 
# plotBAFInSeperatePages (loh=obj@loh.median.filtered.data, folderName="/camp/lab/luscomben/home/users/ziffo/projects/ipsc-mn-als-meta/cnv/LOHPlots") # plot each sample on separate page
plotGEAndBAFOneSample (object=obj, cnv.scale=3, loh.scale=3, sample= "MN-5") # visualise gene expression and BAF signal for one sample
## calculate significant mutual exclusive and co-occurent events
results <- extractMUAndCooccurence(finalChrMat, loh, loh.name.mapping)
## visualize mutual exclusive and co-occurent events
plotMUAndCooccurence(results)



ipsc_mn_als_datasets$vsd.counts %>% select(-gene_name)

generateAnnotation <- function(id_type = "ensembl_gene_id", genes, ishg19, centromere, host="www.ensembl.org") {

    if (ishg19) {
        mart <- useDataset("hsapiens_gene_ensembl", useEnsembl(biomart = "ensembl", GRCh = 37, host = host))
    } else {
        mart <- useDataset("hsapiens_gene_ensembl", useEnsembl(biomart = "ensembl", host = host))
    }
    G_list <- getBM(filters = id_type, attributes = c(id_type, 
        "hgnc_symbol", "chromosome_name", "start_position", 
        "end_position", "band"), values = genes, 
        mart = mart, useCache=F)
    common <- intersect(genes, G_list[, id_type])
    ord <- match(common, G_list[, id_type])
    annotation <- G_list[ord, ]
    annotation <- annotation[order(annotation$start_position), ]
    annotation$cytoband <- paste0(annotation$chromosome_name, substr((annotation$band), 0, 1))
    
    chr.list <- as.character(1:22, "X")
    idx <- unlist(unique(as.vector(sapply(chr.list, function(x) as.vector(unlist(which(as.character(annotation$chromosome_name) == 
        x)))))))
    annotation <- as.data.frame(annotation[idx, ])
    
    colnames(annotation)[c(1:5, 7)] <- c("Gene", "GeneSymbol",  "Chr", "start", "end", "cytoband")
    
    annotation$isCentromer <- rep("no", nrow(annotation))
    
    centromere_snps <- NULL
    for (k in 1:(dim(centromere)[1])) {
        annotation$isCentromer[which(as.character(annotation$Chr) == gsub("chr", "", as.character(centromere$V1[k])) & (as.numeric(as.character(annotation$Position)) >= 
            centromere$V2[k] & as.numeric(as.character(annotation$Position)) <= centromere$V3[k]))] <- "yes"
    }
    

```

# Microsatelite instability

https://github.com/WangX-Lab/PreMSIm

```{r}
premsim_genes = c("DDX27", "EPM2AIP1", "HENMT1", "LYG1", "MLH1", "MSH4", "NHLRC1", "NOL4L", "RNLS", "RPL22L1", "RTF2", "SHROOM4", "SMAP1", "TTC30A", "ZSWIM3")

# import FPKM/TPM counts per sample
premsim_gene_counts = read_tsv(here(shared_path,"answerals/nfcore/star_salmon/salmon.merged.gene_tpm.tsv")) %>% 
  filter(gene_name %in% premsim_genes)

# ipsc_mn_als_datasets.metadata = ipsc_mn_als_datasets.metadata %>% mutate(salmon_path = paste0(database_dir, "/nfcore/star_salmon/salmon.merged.gene_tpm.tsv"))
# ipsc_mn_als_datasets.metadata.filt = ipsc_mn_als_datasets.metadata %>% distinct(salmon_path, .keep_all = TRUE) 
# salmon.merged.gene_tpm.tsv = ipsc_mn_als_datasets.metadata.filt %>% pull(salmon_path) %>% map(read_tsv) #, col_names = c("n","type"), col_types = list(n="n",type="c"))
# names(salmon.merged.gene_tpm.tsv) = ipsc_mn_als_datasets.metadata.filt$dataset
# salmon.merged.gene_tpm.tab <- reduce(salmon.merged.gene_tpm.tsv, left_join, by = c("gene_id", "gene_name"), suffix = c())
# write_csv(salmon.merged.gene_tpm.tab, here(proj_path,"nfcore/ipsc_mn_als_datasets.tpm_counts.csv"))
salmon.merged.gene_tpm.tab = read_csv(here(proj_path,"nfcore/ipsc_mn_als_datasets.tpm_counts.csv"))
salmon.merged.gene_tpm.premsim = salmon.merged.gene_tpm.tab %>% filter(gene_name %in% premsim_genes)

salmon.merged.gene_tpm.premsim.data_pre = PreMSIm::data_pre(salmon.merged.gene_tpm.premsim)

filter(ipsc_mn_als_datasets$res, gene_name %in% premsim_genes)
ipsc_mn_als_datasets.premism.volcano <- volcano_plot(filter(ipsc_mn_als_datasets$res, gene_name %in% premsim_genes), numerator = "ALS", denomenator = "CTRL",
               subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 2, dpi = 300, gene_labels = c(premsim_genes))
ipsc_mn_als_datasets.premism.volcano


library(PreMSIm)


premsim_genes

LYG1, MSH4, and RPL22L1
data_pre <- function(deseq_res, input.path, type = c("gene_name", "gene_id")) {

	## Check arguments
	if (missing(input.path) || class(input.path) != "character")
		stop("'input.path' is missing or incorrect")
        type <- match.arg(type)   
	# ## Read input data
	# a <- read.table(input.path, stringsAsFactors = FALSE, header = TRUE, row.names = 1, sep = "\t", check.names = FALSE)
	
	## Filter and min-max scaling
	if (TRUE %in% is.na(match(feature[,type], rownames(a)))) {
		if (sum(!is.na(match(feature[,type], rownames(a)))) >= 5) {
			feature <- feature[!is.na(match(feature[,type], rownames(a))),]
			rownames(feature) <- NULL
		} else {
		stop("Some features of the current test set are missing!")
		}
	}
	a <- a[match(feature[,type], rownames(a)),, drop = FALSE]
	if (FALSE %in% complete.cases(a))
		stop("Predictor variables with missing values are presented in the current test set")
	if (dim(a)[2] != 1) {
		a <- apply(t(a), 2, function(x) {
			(x - min(x)) / (max(x) - min(x))
		})
		a[is.nan(a)] <- 1
	} else {
		a <- t(a)
	}
	if (type == "ID") {
		colnames(a) <- as.character(feature[,"Symbol"])
	}
	return(a)
}

```




