---
title: "Pan-ALS vs control differential expression"
author: |
  | Dr Oliver Ziff
  | The Francis Crick Institute
  | Institute of Neurology
  | University College London, UK
date: "`r Sys.Date()`"
output: 
 rmarkdown::html_document:
   theme: spacelab
   highlight: haddock
   code_folding: hide
   toc: true
   toc_float: true
   smooth_scroll: true
   number_sections: false
   self_contained: true  
---


```{r setup, include=FALSE}
library(Cairo)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, dev="CairoPNG", dpi=72, fig.height = 8, fig.width = 10)
library(here)
proj_path = here("/camp/lab/patanir/home/users/ziffo/projects/ipsc-mn-als-meta")
load(here(proj_path,"expression/deseq2/ipsc_mn_als_meta.metadata.RData")) # load metadata before functions to avoid overwriting with old functions
source("/camp/lab/patanir/home/users/ziffo/scripts/functions/R_workspace.R")

ipsc_mn_als_datasets.res = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.res.rds"))
ipsc_mn_als_datasets.polyA.res = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.res.rds"))
ipsc_mn_als_datasets.polyA.vsd = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.polyA.vsd.rds"))
ipsc_mn_als_datasets.total.res = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.res.rds"))
ipsc_mn_als_datasets.total.vsd = readRDS(here(proj_path,"expression/deseq2/ipsc_mn_als_datasets.total.vsd.rds"))
```



# Fig 2 iPSN ALS meta-analysis

## Volcano

```{r}
ipsc_mn_als_datasets.als.labels <- ipsc_mn_als_datasets.res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.gene.labels <- ipsc_mn_als_datasets.res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.volcano <- volcano_plot(ipsc_mn_als_datasets.res, numerator = "ALS", denomenator = "CTRL",
               subtitle = "106 Controls vs 323 ALS", ymax = 8, xmax = 2, dpi = 72, gene_labels = c(ipsc_mn_als_datasets.als.labels, ipsc_mn_als_datasets.gene.labels))
ipsc_mn_als_datasets.volcano

```

## Functional enrichment terms

```{r}
ipsc_mn_als_datasets.gost <- ipsc_mn_als_datasets.res %>% filter(padj < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
ipsc_mn_als_datasets.gprofiles <- ipsc_mn_als_datasets.gost$result %>% filter(!str_detect(source, "TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ " ALS Down ", query == "query_2" ~ " ALS Up "), term_name = as.factor(term_name))
ipsc_mn_als_datasets.gprofiles_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")
ipsc_mn_als_datasets.gprofiles_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")
# paste('"',unique(ipsc_mn_als_datasets.gprofiles_down$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"neuron fate commitment",
"cell differentiation in spinal cord",
"DNA-binding transcription factor activity, RNA polymerase II-specific",
"ventral spinal cord development",
"cis-regulatory region sequence-specific DNA binding"
)
ipsc_mn_als_datasets.gprofiles_filt_down <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Down ")  %>% filter(term_name %in% down_list)
# paste('"',unique(ipsc_mn_als_datasets.gprofiles_up$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"p53 transcriptional gene network",
"Genotoxicity pathway",
"miRNA regulation of DNA damage response",
"DNA damage response",
"p53 signaling pathway")
ipsc_mn_als_datasets.gprofiles_filt_up <- ipsc_mn_als_datasets.gprofiles %>% filter(query == " ALS Up ")  %>% filter(term_name %in% up_list)
ipsc_mn_als_datasets.gprofiles_filt_combined <- bind_rows(ipsc_mn_als_datasets.gprofiles_filt_down, ipsc_mn_als_datasets.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c(" ALS Up ", " ALS Down ")))
ipsc_mn_als_datasets.gprofiles_filt_combined = ipsc_mn_als_datasets.gprofiles_filt_combined %>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
ipsc_mn_als_datasets.go_curated <- curated_profiles(gprofiles = ipsc_mn_als_datasets.gprofiles_filt_combined)
ipsc_mn_als_datasets.go_curated
```

## GSEA of p53 signalling pathway

```{r}
ipsc_mn_als_datasets.geneList <- ipsc_mn_als_datasets.res %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.geneList) <- ipsc_mn_als_datasets.res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ipsc_mn_als_datasets.geneList = sort(ipsc_mn_als_datasets.geneList, decreasing = TRUE)

# p53 signalling pathway
p53.signalling_genes.list <- list("p53.signalling" = ipsc_mn_als_datasets.res$gene_name[ipsc_mn_als_datasets.res$gene_name %in% go.pathways.msigdb$GO_SIGNAL_TRANSDUCTION_BY_P53_CLASS_MEDIATOR])
ipsc_mn_als_datasets.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = ipsc_mn_als_datasets.geneList)
ipsc_mn_als_datasets.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], ipsc_mn_als_datasets.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("p53 signalling") + xlab("") +
  annotate("text", x = 35000, y = 0.3, label = paste0("NES = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(ipsc_mn_als_datasets.p53.signalling_fgseaRes$pval, digits = 6)), size = 2.8, hjust = 0) + theme_oz()
ipsc_mn_als_datasets.p53.signalling.gseaplot

```


## PROGENy pathways

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/pw_bk.html 
<https://github.com/saezlab/transcriptutorial/blob/master/scripts/03_Pathway_activity_with_Progeny.md> 
<https://saezlab.github.io/progeny/>

Pathway RespOnsive GENes (PROGENy) estimates signaling pathway activity

```{r}
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.res.matrix <- ipsc_mn_als_datasets.res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "***", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_progeny.pathwayActivity <- ggplot(ipsc_mn_als_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Signaling Pathways", x = "", y = "ALS vs CTRL Normalised enrichment") +
    stat_pvalue_manual(ipsc_mn_als_progeny.contrast_acts, label = "p.signif", y.position = 14, xmin = "source", xmax = NULL, size = 3, hide.ns = TRUE) 
ipsc_mn_als_progeny.pathwayActivity
```

p53 weights

```{r}
ipsc_mn_als_progeny.p53weights = net_progeny %>% filter(source == "p53", target %in% ipsc_mn_als_datasets.res$gene_name) %>% left_join(select(ipsc_mn_als_datasets.res, target=gene_name,stat)) %>% 
  mutate(color = case_when(weight > 0 & stat > 0 ~ '1', weight > 0 & stat < 0 ~ '2', weight < 0 & stat > 0 ~ '2', weight < 0 & stat < 0 ~ '1', TRUE ~ "3"))
ipsc_mn_als_progeny.p53weights.plot = ggplot(ipsc_mn_als_progeny.p53weights, aes(x = weight, y = stat, color = color)) + geom_point(size = 0.5) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_progeny.p53weights, weight > 5, stat > 3), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.1) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
  labs(title = "p53 pathway weights", x = "p53 pathway gene weight", y = "ALS vs CTRL test statistic") +
  geom_vline(xintercept = 0, linetype = 'dotted') +  geom_hline(yintercept = 0, linetype = 'dotted') +
  coord_cartesian(xlim = c(-8,18), ylim = c(-2.5,5)) 
ipsc_mn_als_progeny.p53weights.plot

```

## DoRoTHEA transcription factors

https://www.bioconductor.org/packages/release/bioc/vignettes/decoupleR/inst/doc/tf_bk.html
https://github.com/saezlab/transcriptutorial/blob/master/scripts/04_TranscriptionFactor_activity_with_Dorothea.md

```{r}
net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))

ipsc_mn_als_dorothea.contrast_acts.labels.up = ipsc_mn_als_dorothea.contrast_acts %>% top_n(score, n = 3) %>% pull(source)
ipsc_mn_als_dorothea.contrast_acts.labels.down = ipsc_mn_als_dorothea.contrast_acts %>% top_n(-score, n = 3) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.contrast_acts, source %in% c(ipsc_mn_als_dorothea.contrast_acts.labels.up, ipsc_mn_als_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_dorothea.contrast_acts.volcano
```

TP53 target genes

```{r}
ipsc_mn_als_dorothea.tp53 <- net_dorothea %>%  filter(source == "TP53") %>% mutate(gene_name = target) %>% left_join(select(ipsc_mn_als_datasets.res, gene_name,stat,log2FoldChange,pvalue)) %>%
  mutate(color = case_when(mor > 0 & stat > 0 ~ '1', mor > 0 & stat < 0~'2',mor < 0 & stat > 0~'2',mor < 0 & stat < 0~'1',TRUE ~ "3"))

ipsc_mn_als_dorothea.tp53.volcano = ggplot(ipsc_mn_als_dorothea.tp53, aes(x = log2FoldChange, y = -log10(pvalue), color = color, size=abs(mor))) + geom_point() + scale_size(range = c(0, 3)) +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_text_repel(data = filter(ipsc_mn_als_dorothea.tp53, abs(stat) > 3), aes(label = gene_name, size=1), fontface = "italic", max.overlaps = Inf, size = 2.3) +
  theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +  geom_vline(xintercept = 0, linetype = 'dotted') + geom_hline(yintercept = 0, linetype = 'dotted') +
    labs(title = "TP53 regulon", x = "log2 fold change (ALS vs CTRL)", y = expression(-log[10]~P~value)) + ylim(c(0,2)) +   
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_dorothea.tp53, log2FoldChange > 0.25, -log10(pvalue) > 1.0), aes(label = target), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.5)
ipsc_mn_als_dorothea.tp53.volcano

```


# Table S3 DGE panALS

```{r}
ipsc_mn_als_datasets.res.sig = ipsc_mn_als_datasets.res %>% filter(padj < 0.05) %>% select(gene_name, everything())
kbl(ipsc_mn_als_datasets.res.sig) %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), fixed_thead = T) %>% scroll_box(width = "100%", height = "500px")

```

# Table S4 Dorothea panALS

```{r}
ipsc_mn_als_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 100, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% select(-statistic, -condition, transcription_factor = source, normalised_enrichment_score = score) %>% arrange(p_value, -abs(normalised_enrichment_score)) %>% top_n(n = 20, wt = abs(normalised_enrichment_score))
kbl(ipsc_mn_als_dorothea.contrast_acts) %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), fixed_thead = T) %>% scroll_box(width = "100%", height = "500px")

```

# Fig S8 GSEA DNA damage 

```{r}
ipsc_mn_als_datasets.geneList <- ipsc_mn_als_datasets.res %>% drop_na(stat) %>% pull(stat)
names(ipsc_mn_als_datasets.geneList) <- ipsc_mn_als_datasets.res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
ipsc_mn_als_datasets.geneList = sort(ipsc_mn_als_datasets.geneList, decreasing = TRUE)

gprofiler_database.DNA_damage_pathways = gprofiler_database[grepl('DNA damage|p53', names(gprofiler_database))]
gprofiler_database.DNA_damage_pathways_fgseaRes <- fgsea(pathways = gprofiler_database.DNA_damage_pathways, stats = ipsc_mn_als_datasets.geneList)
gprofiler_database.DNA_damage.filt = filter(gprofiler_database.DNA_damage_pathways_fgseaRes, size >7, !grepl("uncertain|negative|prostate",pathway)) %>% mutate(p.signif = case_when(pval < 0.003 ~ "***", pval < 0.009 ~ "***",pval < 0.01 ~ "**",pval < 0.05 ~ "*", TRUE ~ ""), group1 = "pathway", group2 = "pathway")
gprofiler_database.DNA_damage_pathways_fgseaRes.barplot <- ggplot(gprofiler_database.DNA_damage.filt, aes(x=reorder(pathway, NES), y = NES)) +
    geom_bar(aes(fill = NES), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "DNA damage pathways", y = "ALS vs CTRL iPSMNs\nNormalised Enrichment Score", x = "") + geom_hline(yintercept = 0, linetype = 3) + 
    stat_pvalue_manual(gprofiler_database.DNA_damage.filt, label = "p.signif", y.position = 2, size = 4, xmin = "pathway", xmax = NULL, hide.ns = TRUE) + coord_flip()
gprofiler_database.DNA_damage_pathways_fgseaRes.barplot
```

# Fig S9 polyA & total subgroup analyses

## polyA

### PCA

```{r}
ipsc_mn_als_datasets.polyA.pcaData <- plotPCA(ipsc_mn_als_datasets.polyA.vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/patanir/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "poly(A)"), dataset_sample, gender)) 
  
ipsc_mn_als_datasets.polyA.pca_dataset <- ggplot(ipsc_mn_als_datasets.polyA.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.polyA.pcaData, "percentVar"))[1],"% variance")) + ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.polyA.pcaData, "percentVar"))[2],"% variance")) + labs(color="polyA\ndatasets") + 
  ggforce::geom_mark_ellipse(aes(label = dataset), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE)
ipsc_mn_als_datasets.polyA.pca_dataset
```

### Volcano

```{r}
ipsc_mn_als_datasets.polyA.als.labels <- ipsc_mn_als_datasets.polyA.res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.gene.labels <- ipsc_mn_als_datasets.polyA.res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.polyA.volcano <- volcano_plot(ipsc_mn_als_datasets.polyA.res, numerator = "ALS", denomenator = "CTRL",
               title = "polyA", subtitle = "43 Controls vs 48 ALS", ymax = 10, xmax = 3, dpi = 72, gene_labels = c(ipsc_mn_als_datasets.polyA.als.labels, ipsc_mn_als_datasets.polyA.gene.labels,"SESN1","TCEAL6"), distinct_labels = TRUE)
ipsc_mn_als_datasets.polyA.volcano
```

### DoRoTHEA & PROGENy

```{r}
net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_als_datasets.polyA.res.matrix <- ipsc_mn_als_datasets.polyA.res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.polyA_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.polyA.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.polyA_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.polyA_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "polyA: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.polyA_progeny.contrast_acts, label = "p.signif", y.position = 12, xmin = "source", xmax = NULL, size = 6, hide.ns = TRUE) 
ipsc_mn_als_datasets.polyA_progeny.pathwayActivity

net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.polyA.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% filter(p_value < 0.15) %>% top_n(score, n = 4) %>% pull(source)
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.polyA_dorothea.contrast_acts %>% top_n(-score, n = 4) %>% pull(source)

# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "polyA: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.labels.down)), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_datasets.polyA_dorothea.contrast_acts.volcano

```


## Total

### PCA

```{r}
ipsc_mn_als_datasets.total.pcaData <- plotPCA(ipsc_mn_als_datasets.total.vsd, intgroup=c("sample", "dataset", "database_dir", "condition", "mutation", "dataset_sample", "strandedness", "onset_site", "study_accession", "DIV", "instrument","library_type"), returnData=TRUE) %>% as_tibble() %>%
  mutate(dataset = as.character(dataset), dataset = gsub("dafianca","dafinca",dataset), dataset_sample = gsub("dafianca","dafinca",dataset_sample), dataset = case_when(dataset == "neurolincs0" & grepl("^0007",sample) ~ "neurolincs_diMN", dataset == "neurolincsA" & grepl("^A-042",sample) ~ "neurolincs_iMN", dataset == "answerals" ~ "AnswerALS",  dataset %in% c("dafinca.c9orf72","dafinca.tardbp") ~ "dafinca", TRUE ~  dataset), 
         DIV = as.character(DIV), DIV = case_when(is.na(DIV) ~ "30", TRUE ~  DIV), database_dir = gsub("/camp/lab/patanir/home/shared/projects/patani-collab/public-data","/camp/project/proj-luscombn-patani/working/public-data",database_dir),
         library_type = case_when(dataset %in% c("catanese","dafinca","kapeli","kiskinis","luisier","mitchell","sareen","sommer","smith","sterneckert","wang") ~ "poly(A)", TRUE ~ "Total")) %>%
  left_join(select(filter(ipsc_mn_als_datasets.metadata, library_type == "Total"), dataset_sample, gender)) 

ipsc_mn_als_datasets.total.pca_dataset <- ggplot(ipsc_mn_als_datasets.total.pcaData, aes(PC1, PC2, color=dataset)) + geom_point(size=1) + 
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + guides(color=guide_legend(nrow=2, keywidth=0.1, keyheight=0.1, default.unit="inch")) +
  xlab(paste0("PC1: ",round(100 * attr(ipsc_mn_als_datasets.total.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(ipsc_mn_als_datasets.total.pcaData, "percentVar"))[2],"% variance"))  + labs(color="Total\ndatasets") +
    ggforce::geom_mark_ellipse(aes(label = dataset), label.fontsize = 8, label.fontface = "plain", con.type = "straight", con.cap = 0.5, show.legend = FALSE)
ipsc_mn_als_datasets.total.pca_dataset
```

### Volcano

```{r}
ipsc_mn_als_datasets.total.als.labels <- ipsc_mn_als_datasets.total.res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes, RNA_BINDING_PROTEINS, p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) %>% distinct(gene_name) %>% pull(gene_name)
ipsc_mn_als_datasets.total.gene.labels <- ipsc_mn_als_datasets.total.res %>% filter(!grepl("^AC|^AL", gene_name)) %>% top_n(20, wt = -pvalue) %>% pull(gene_name)
ipsc_mn_als_datasets.total.volcano <- volcano_plot(ipsc_mn_als_datasets.total.res, numerator = "ALS", denomenator = "CTRL",
               title = "Total", subtitle = "63 Controls vs 275 ALS", ymax = 8, xmax = 3, dpi = 72, gene_labels = c(ipsc_mn_als_datasets.total.als.labels, ipsc_mn_als_datasets.total.gene.labels,"RNASEL","SESN1", "TCEAL6","MTCO1P12","MTCO2P12","LMO4"))
ipsc_mn_als_datasets.total.volcano
```

### DoRoTHEA & PROGENy

```{r}
ipsc_mn_als_datasets.total.res.matrix <- ipsc_mn_als_datasets.total.res %>% select(gene_name, stat) %>%  drop_na(stat) %>% distinct(gene_name, .keep_all = TRUE) %>% tibble_to_matrix(stat, row_names = "gene_name")
ipsc_mn_als_datasets.total_progeny.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.total.res.matrix, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = source, group2 = source)
ipsc_mn_als_datasets.total_progeny.pathwayActivity <- ggplot(ipsc_mn_als_datasets.total_progeny.contrast_acts, aes(x=reorder(source, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
    scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + 
    theme_oz() + theme(axis.text.x = element_text(angle = 90), legend.position = "none", plot.title = element_text(hjust = 0.5)) +
    labs(title = "Total: Signaling Pathways", x = "", y = "ALS vs CTRL enrichment") +
    stat_pvalue_manual(ipsc_mn_als_datasets.total_progeny.contrast_acts, label = "p.signif", y.position = 8, xmin = "source", xmax = NULL, size = 6, hide.ns = TRUE) 
ipsc_mn_als_datasets.total_progeny.pathwayActivity

ipsc_mn_als_datasets.total_dorothea.contrast_acts <- run_wmean(mat=ipsc_mn_als_datasets.total.res.matrix, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5) %>% filter(statistic == 'norm_wmean') %>% 
  mutate(p.signif = case_when(p_value < 0.05 ~ "*", TRUE ~ ""), sig = case_when(p_value < 0.135 & abs(score) < 4 ~ "sig", p_value < 0.135 & abs(score) >= 4 ~ "sig_strong", p_value >= 0.135 ~ "non_sig"), direction = ifelse(score > 0, "up", "down"), class = paste(sig, direction))
ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.up = ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% top_n(score, n = 5) %>% pull(source)
ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.down = ipsc_mn_als_datasets.total_dorothea.contrast_acts %>% top_n(-score, n = 5) %>% pull(source)
# Volcano of TFs score vs p_value
ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano = ggplot(ipsc_mn_als_datasets.total_dorothea.contrast_acts, aes(x = score, y = -log10(p_value))) + geom_point(aes(colour = class)) +
    scale_colour_manual(values = c("non_sig up" = "gray", "non_sig down" = "gray",  "sig up" = "#F36F6F", "sig_strong up" = "#EB4445", "sig down" = "#A6CEE3", "sig_strong down" = "#79B1D3")) +
  labs(y = expression(-log[10]~P~value), x = "Normalised enrichment ALS vs CTRL", title = "Total: Transcription Factors") +
    guides(colour = "none") +
    theme_oz() +  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5), panel.border = element_blank(), axis.ticks = element_line(colour = "black") ) +
  # scale_y_continuous(expand = c(0,0), limits = c(0,ymax)) + scale_x_continuous(limits = c(-xmax,xmax))
  geom_text_repel(fontface = "italic", data = filter(ipsc_mn_als_datasets.total_dorothea.contrast_acts, source %in% c(ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.up, ipsc_mn_als_datasets.total_dorothea.contrast_acts.labels.down, "TP53")), aes(label = source), max.overlaps = Inf, min.segment.length = unit(0, "lines"), size = 2.3) +  
  geom_vline(xintercept = 0, linetype = 'dotted') + ylim(c(0,2.8))
ipsc_mn_als_datasets.total_dorothea.contrast_acts.volcano
```


### Compare polyA & total

```{r}
# # Scatter correlation
ipsc_mn_als_datasets.polyA_total.labels = select(ipsc_mn_als_datasets.total.res, gene_id, gene_name, polyA.stat = stat) %>% left_join(select(ipsc_mn_als_datasets.polyA.res, gene_id, gene_name, total.stat = stat)) %>% filter((polyA.stat > 2.5 & total.stat > 2.5) | (polyA.stat < -2.5 & total.stat < -2.5))
ipsc_mn_als_datasets.polyA_total.list = list("polyA: ALS vs CTRL" = ipsc_mn_als_datasets.polyA.res, "Total: ALS vs CTRL" = ipsc_mn_als_datasets.total.res)
ipsc_mn_als_datasets.polyA_total_stat.corr = plot_de_correlation(model_list1 = ipsc_mn_als_datasets.polyA_total.list, model_list2 = ipsc_mn_als_datasets.polyA_total.list, mutation1 = "polyA: ALS vs CTRL", mutation2 = "Total: ALS vs CTRL", gene_labels = pull(ipsc_mn_als_datasets.polyA_total.labels,gene_name), plot_corr_line = TRUE, plot_p_value = TRUE)
ipsc_mn_als_datasets.polyA_total_stat.corr

```

# Fig S10 Datasets independently


```{r}
load(here(proj_path, "expression/deseq2/datasets_separated.RData"))

```

## Volcano

```{r}
ipsc_mn_datasets_list.res = list("AnswerALS" = answerals$res, "neurolincs.diMN" = neurolincs.diMN$res, "neurolincs.iMN" = neurolincs.iMN$res, "catanese" = catanese$res, "dafinca C9orf72" = dafinca.c9orf72$res, "dafinca tardbp" = dafinca.tardbp$res, "luisier" = luisier$res, "wang" = wang$res, "kiskinis" = kiskinis$res, "sareen" = sareen$res, "sommer" = sommer$res, "sterneckert" = sterneckert$res, "kapeli" = kapeli$res, "desantis" = desantis$res, "smith" = smith$res, "bhinge" = bhinge$res, "hawkins" = hawkins$res)
ipsc_mn_datasets.res = bind_rows(ipsc_mn_datasets_list.res, .id = "dataset")

datasets_separated.volcano = volcano_plot(ipsc_mn_datasets.res, arrow_labels = FALSE, dpi = 300) + facet_wrap(~dataset, scales = "free", ncol = 6)
datasets_separated.volcano

```


## Correlation heatmap

```{r}
# heatmap of correlation coefficients
ipsc_mn_datasets_stat = select(answerals$res, gene_id, answerals = stat) %>% 
  left_join(select(neurolincs.diMN$res, gene_id, neurolincs.diMN = stat)) %>%
  left_join(select(neurolincs.iMN$res, gene_id, neurolincs.iMN = stat)) %>%
  left_join(select(catanese$res, gene_id, catanese = stat)) %>%
  left_join(select(dafinca.c9orf72$res, gene_id, dafinca.c9orf72 = stat)) %>%
  left_join(select(dafinca.tardbp$res, gene_id, dafinca.tardbp = stat)) %>%
  left_join(select(luisier$res, gene_id, luisier = stat)) %>%
  left_join(select(wang$res, gene_id, wang = stat)) %>%
  left_join(select(kiskinis$res, gene_id, kiskinis = stat)) %>%
  left_join(select(sareen$res, gene_id, sareen = stat)) %>%
  left_join(select(sommer$res, gene_id, sommer = stat)) %>%
  left_join(select(sterneckert$res, gene_id, sterneckert = stat)) %>%
  left_join(select(kapeli$res, gene_id, kapeli = stat)) %>%
  left_join(select(desantis$res, gene_id, desantis = stat)) %>%
  left_join(select(smith$res, gene_id, smith = stat)) %>%
  left_join(select(bhinge$res, gene_id, bhinge = stat)) %>%
  left_join(select(hawkins$res, gene_id, hawkins = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_id")
cormat <- round(cor(ipsc_mn_datasets_stat),2)
melted_cormat <- melt(cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
reorder_cormat <- function(cormat){ # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
}
cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) # Melt the correlation matrix
dataset_dge_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = c(0.25,0.75), legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text.x = element_text(angle = 30, hjust = 1, vjust = 1), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5))
dataset_dge_correlation_ggheatmap


```


## p53 PROGENy & TP53 Dorothea

```{r}
ipsc_mn_datasets_list = list("AnswerALS" = select(answerals$res,gene_name,stat), "neurolincs.diMN" = select(neurolincs.diMN$res,gene_name,stat), "neurolincs.iMN" = select(neurolincs.iMN$res,gene_name,stat), "catanese" = select(catanese$res,gene_name,stat), "dafinca C9orf72" = select(dafinca.c9orf72$res,gene_name,stat), "dafinca tardbp" = select(dafinca.tardbp$res,gene_name,stat), "luisier" = select(luisier$res,gene_name,stat), "wang" = select(wang$res,gene_name,stat), "kiskinis" = select(kiskinis$res,gene_name,stat), "sareen" = select(sareen$res,gene_name,stat), "sommer" = select(sommer$res,gene_name,stat), "sterneckert" = select(sterneckert$res,gene_name,stat), "kapeli" = select(kapeli$res,gene_name,stat), "desantis" = select(desantis$res,gene_name,stat), "smith" = select(smith$res,gene_name,stat), "bhinge" = select(bhinge$res,gene_name,stat), "hawkins" = select(hawkins$res,gene_name,stat))
datasets <- names(ipsc_mn_datasets_list)
ipsc_mn_datasets_list = map(ipsc_mn_datasets_list, ~{drop_na(.x)}) 
ipsc_mn_datasets_list = map(ipsc_mn_datasets_list, ~{distinct(.x, gene_name, .keep_all = TRUE)})
ipsc_mn_datasets_list = map(ipsc_mn_datasets_list, ~{tibble_to_matrix(.x, stat, row_names = "gene_name")})

# PROGENy
# net_progeny <- get_progeny(organism = 'human', top = 100)
ipsc_mn_datasets_list.progeny_scores = map_df(ipsc_mn_datasets_list, ~{run_wmean(.x, net=net_progeny, .source='source', .target='target', .mor='weight', times = 1000, minsize = 5)}, .id = "dataset") %>% filter(statistic == 'norm_wmean')

ipsc_mn_datasets_list.p53.progeny_scores = ipsc_mn_datasets_list.progeny_scores %>% filter(source == "p53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = dataset, group2 = dataset)
ipsc_mn_datasets_list.p53.progeny_scores.plot <- ggplot(ipsc_mn_datasets_list.p53.progeny_scores, aes(x=reorder(dataset, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "p53 pathway activity", x = "", y = "Normalised enrichment") + #
  stat_pvalue_manual(ipsc_mn_datasets_list.p53.progeny_scores, label = "p.signif", y.position = 11.5, xmin = "dataset", xmax = NULL, size = 4, hide.ns = TRUE) + ylim(c(-12,12))
ipsc_mn_datasets_list.p53.progeny_scores.plot


# Dorothea
# net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C'))
ipsc_mn_datasets_list.dorothea_scores = map_df(ipsc_mn_datasets_list, ~{run_wmean(.x, net=net_dorothea, .source='source', .target='target', .mor='mor', times = 1000, minsize = 5)}, .id = "dataset") %>% filter(statistic == 'norm_wmean')

ipsc_mn_datasets_list.TP53.dorothea_scores = ipsc_mn_datasets_list.dorothea_scores %>% filter(source == "TP53") %>% mutate(p.signif = case_when(p_value < 0.003 ~ "****", p_value < 0.009 ~ "***",p_value < 0.01 ~ "**",p_value < 0.05 ~ "*", TRUE ~ ""), group1 = dataset, group2 = dataset)
ipsc_mn_datasets_list.TP53.dorothea_scores.plot <- ggplot(ipsc_mn_datasets_list.TP53.dorothea_scores, aes(x=reorder(dataset, score), y = score)) +
    geom_bar(aes(fill = score), stat = "identity") +
  scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +  
  theme_oz() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(title = "TP53 Transcription Factor activity", x = "", y = "Normalised enrichment") +
  stat_pvalue_manual(ipsc_mn_datasets_list.TP53.dorothea_scores, label = "p.signif", y.position = 13, xmin = "dataset", xmax = NULL, size = 4, hide.ns = TRUE) + ylim(c(-13,13))
ipsc_mn_datasets_list.TP53.dorothea_scores.plot

```

# Fig S11 Proteomics AnswerALS


```{r}
answerals.proteomics.als_vs_ctrl.res = read_csv(here(shared_path,"/answerals/proteomics/answerals_proteomics_als_vs_ctrl.csv"))
```


## Volcano & GO & GSEA
 
```{r}
answerals.proteomics.als_vs_ctrl.volcano = volcano_plot(answerals.proteomics.als_vs_ctrl.res, title = "Protein Expression", subtitle = "171 ALS vs 33 CTRL", 
  gene_labels = c(pull(top_n(answerals.proteomics.als_vs_ctrl.res,n=10,wt=stat),gene_name), pull(filter(answerals.proteomics.als_vs_ctrl.res, pvalue < 0.05, gene_name %in% c(ALS_genes,p53.signalling.pathway)),gene_name), "TARDBP","FUS","SFPQ","SOD1", "BRD4", "EEF2"),
                  ymax = 4,xmax = 2,numerator = "ALS",denomenator = "CTRL", significance_threshold = "pvalue", dpi = 300)
answerals.proteomics.als_vs_ctrl.volcano

# GO terms
answerals.proteomics.als_vs_ctrl.gost <- answerals.proteomics.als_vs_ctrl.res %>% filter(pvalue < 0.05) %>% group_split(log2FoldChange > 0) %>%  map("gene_id") %>% gost()
answerals.proteomics.als_vs_ctrl.gprofiles <- answerals.proteomics.als_vs_ctrl.gost$result %>% filter(!str_detect(source, "TF|CORUM|HPA")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "ALS\nDown", query == "query_2" ~ "ALS\nUp"), term_name = as.factor(term_name))
answerals.proteomics.als_vs_ctrl.gprofiles_down <- answerals.proteomics.als_vs_ctrl.gprofiles %>% filter(query == "ALS\nDown")
answerals.proteomics.als_vs_ctrl.gprofiles_up <- answerals.proteomics.als_vs_ctrl.gprofiles %>% filter(query == "ALS\nUp")
down_list <- c(
"Schaffer collateral - CA1 synapse",
"Golgi transport complex")
answerals.proteomics.als_vs_ctrl.gprofiles_filt_down <- answerals.proteomics.als_vs_ctrl.gprofiles_down %>% filter(query == "ALS\nDown")  %>% filter(term_name %in% down_list)
up_list <- c(
"proteasome complex",
"Axon guidance",
"nucleocytoplasmic transport",
"Cholesterol biosynthesis",
"Amyotrophic lateral sclerosis",
"protein modification process",
"Metabolism of proteins",
"Cellular responses to stress",
"protein localization",
"endoplasmic reticulum",
"protein binding",
"Metabolism of RNA",
"translation",
"protein metabolic process")
answerals.proteomics.als_vs_ctrl.gprofiles_filt_up <- answerals.proteomics.als_vs_ctrl.gprofiles_up %>% filter(query == "ALS\nUp")  %>% filter(term_name %in% up_list)
answerals.proteomics.als_vs_ctrl.gprofiles_filt_combined <- bind_rows(answerals.proteomics.als_vs_ctrl.gprofiles_filt_down, answerals.proteomics.als_vs_ctrl.gprofiles_filt_up) %>% mutate(query = factor(query, levels = c("ALS\nUp", "ALS\nDown")))
answerals.proteomics.als_vs_ctrl.go_curated <- curated_profiles(gprofiles = answerals.proteomics.als_vs_ctrl.gprofiles_filt_combined,label_angle = 0) + theme(legend.position = "none", axis.text = element_text(size=10))
answerals.proteomics.als_vs_ctrl.go_curated

## GSEA: DNA damage response, p53 signalling pathway
answerals.proteomics.als_vs_ctrl.geneList <- answerals.proteomics.als_vs_ctrl.res %>% drop_na(stat) %>% pull(stat)
names(answerals.proteomics.als_vs_ctrl.geneList) <- answerals.proteomics.als_vs_ctrl.res %>% drop_na(stat) %>% pull(gene_name) %>% as.character()
answerals.proteomics.als_vs_ctrl.geneList = sort(answerals.proteomics.als_vs_ctrl.geneList, decreasing = TRUE)

# p53 signalling pathway
p53.signalling_genes.list <- list("p53.signalling" = answerals.proteomics.als_vs_ctrl.res$gene_name[answerals.proteomics.als_vs_ctrl.res$gene_name %in% p53.signalling.pathway])
answerals.proteomics.als_vs_ctrl.p53.signalling_fgseaRes <- fgsea(pathways = p53.signalling_genes.list, stats = answerals.proteomics.als_vs_ctrl.geneList)
answerals.proteomics.als_vs_ctrl.p53.signalling_fgseaRes
# pathway      pval      padj    log2err        ES       NES size                             leadingEdge
# 1: p53.signalling 0.5122995 0.5122995 0.04763873 0.2633203 0.9831828  103 RBBP7,CSNK2B,PRMT1,CNOT9,RPL23,RRM2,...
answerals.proteomics.als_vs_ctrl.p53.signalling.gseaplot = plotEnrichment(p53.signalling_genes.list[["p53.signalling"]], answerals.proteomics.als_vs_ctrl.geneList) + 
  theme(axis.text=element_text(size=8), axis.title=element_text(size=10)) + ggtitle("p53 signalling") + xlab("") +
  annotate("text", x = 3000, y = 0.2, label = paste0("NES = ", round(answerals.proteomics.als_vs_ctrl.p53.signalling_fgseaRes$NES, digits = 3),"\nP = ", round(answerals.proteomics.als_vs_ctrl.p53.signalling_fgseaRes$pval, digits = 6)), size = 2.8, hjust = 0) + theme_oz()
answerals.proteomics.als_vs_ctrl.p53.signalling.gseaplot

```


## Protein & RNA correlation

```{r}
# Compare RNAseq to mass spec
ipsc_mn_rnaseq_mass_spec.join = select(answerals.proteomics.als_vs_ctrl.res, gene_name, gene_id, MS.lfc = log2FoldChange, MS.pvalue = pvalue, MS.padj = padj, MS.stat = stat) %>% 
  left_join(select(ipsc_mn_als_datasets.res, gene_name, gene_id, RNA.lfc = log2FoldChange, RNA.pvalue = pvalue, RNA.padj = padj, RNA.stat = stat))

# Correlate RNAseq & Mass Spec
ipsc_mn_rnaseq_mass_spec = list("RNA-seq: ALS vs CTRL" = ipsc_mn_als_datasets.res, "Mass Spec: ALS vs CTRL" = answerals.proteomics.als_vs_ctrl.res)
ipsc_mn_rnaseq_mass_spec.scatter = plot_de_correlation(model_list1 = ipsc_mn_rnaseq_mass_spec, model_list2 = ipsc_mn_rnaseq_mass_spec, mutation1 = "RNA-seq: ALS vs CTRL", mutation2 = "Mass Spec: ALS vs CTRL", col = "log2FoldChange", gene_labels = c("MACROD1","EEFSEC","EIF1AY","H3C4","CEMIP","PTGR1","HOXC8","KRT1","IGFBP5","TPX2", "TARDBP","FUS","SFPQ","SOD1"), plot_p_value = TRUE) & xlim(-0.9,0.6) & ylim(-1.8,1.9)
ipsc_mn_rnaseq_mass_spec.scatter

```

